diff -ruN glibc/glibc-2.3.5/localedata/Makefile glibc-2.3.5.locale/localedata/Makefile
--- glibc/glibc-2.3.5/localedata/Makefile	2004-08-07 18:38:13.000000000 -0500
+++ glibc-2.3.5.locale/localedata/Makefile	2005-11-02 13:50:00.000000000 -0600
@@ -222,6 +222,7 @@
 	echo -n '...'; \
 	input=`echo $$locale | sed 's/\([^.]*\)[^@]*\(.*\)/\1\2/'`; \
 	$(LOCALEDEF) --alias-file=../intl/locale.alias \
+		     --no-archive \
 		     -i locales/$$input -c -f charmaps/$$charset \
 		     $(addprefix --prefix=,$(install_root)) $$locale; \
 	echo ' done'; \
