--- kdeutils/printer-applet/printer-applet.py	2010-08-03 21:07:17.000000000 +0100
+++ kdeutils/printer-applet/printer-applet.py	2010-09-22 16:49:04.628075748 +0100
@@ -319,7 +319,7 @@
 
     def notify_new_printer (self, printer, title, text):
         self.sysTray.setStatus(KStatusNotifierItem.Active)
-        KNotification.event(title, text, KIcon("konqueror").pixmap(QSize(22,22)))
+        KNotification.event(title, text, KIcon("printer").pixmap(QSize(22,22)))
 
     """unused, see set_special_statusicon
     def set_statusicon_from_pixbuf (self, pb):
@@ -721,7 +721,7 @@
         """
 
         (title, text) = reason.get_description ()
-        KNotification.event("Other", text, KIcon("konqueror").pixmap(QSize(22,22)))
+        KNotification.event("Other", text, KIcon("printer").pixmap(QSize(22,22)))
         self.set_statusicon_visibility ()
 
     ## monitor.Watcher interface
--- kdeutils/printer-applet/monitor.py	2010-02-09 14:29:43.000000000 +0000
+++ kdeutils/printer-applet/monitor.py	2010-09-22 17:41:33.194708031 +0100
@@ -327,6 +327,7 @@
         deferred_calls = []
         jobs = self.jobs.copy ()
         for event in notifications['events']:
+            print "event: "+ str(type(event)) + str(event)
             seq = event['notify-sequence-number']
             self.sub_seq = seq
             nse = event['notify-subscribed-event']
@@ -373,7 +374,13 @@
                 continue
 
             # Job events
-            jobid = event['notify-job-id']
+            if not nse.startswith ("job-"):
+                # Some versions of CUPS give empty
+                # notify-subscribed-event attributes (STR #3608).
+                debugprint ("Unhandled nse %s" % repr (nse))
+                continue
+
+            jobid = event['notify-job-id']
             if (nse == 'job-created' or
                 (nse == 'job-state-changed' and
                  not jobs.has_key (jobid) and
