--- src/Makefile.orig	2008-10-22 07:06:31.000000000 -0400
+++ src/Makefile	2008-10-22 07:10:54.000000000 -0400
@@ -497,13 +497,13 @@ install: $(LIBNAME) $(DSONAME) \
 	$(INSTALL_LIB) $(LIBNAME) $(DESTDIR)$(libdir)
 	$(INSTALL_LIB) $(FLLIBNAME) $(DESTDIR)$(libdir)
 	$(INSTALL_LIB) $(IMGLIBNAME) $(DESTDIR)$(libdir)
-	$(RANLIB) $(DESTDIR)$(libdir)/$(LIBNAME)
-	$(RANLIB) $(DESTDIR)$(libdir)/$(FLLIBNAME)
+	$(RANLIB) $(DESTDIR)$(libdir)/`basename $(LIBNAME)`
+	$(RANLIB) $(DESTDIR)$(libdir)/`basename $(FLLIBNAME)`
 	if test x$(GLLIBNAME) != x; then \
 		$(INSTALL_LIB) $(GLLIBNAME) $(DESTDIR)$(libdir); \
-		$(RANLIB) $(DESTDIR)$(libdir)/$(GLLIBNAME); \
+		$(RANLIB) $(DESTDIR)$(libdir)/`basename $(GLLIBNAME)`; \
 	fi
-	$(RANLIB) $(DESTDIR)$(libdir)/$(IMGLIBNAME)
+	$(RANLIB) $(DESTDIR)$(libdir)/`basename $(IMGLIBNAME)`
 	if test x$(DSONAME) = xlibfltk.so.1.1; then\
 		$(RM) $(DESTDIR)$(libdir)/libfltk.so*;\
 		$(INSTALL_LIB) libfltk.so.1.1 $(DESTDIR)$(libdir); \
