diff -Naur ppp-2.4.4.orig/chat/Makefile.linux ppp-2.4.4/chat/Makefile.linux
--- ppp-2.4.4.orig/chat/Makefile.linux	2007-05-15 16:42:59.000000000 -0400
+++ ppp-2.4.4/chat/Makefile.linux	2007-05-15 16:44:29.000000000 -0400
@@ -1,8 +1,9 @@
 #	$Id: Makefile.linux,v 1.15 2006/06/04 05:07:46 paulus Exp $
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+PREFIX = $(DESTDIR)
+BINDIR = $(INSTROOT)$(PREFIX)/sbin
+MANDIR = $(INSTROOT)$(PREFIX)/share/man/man8
 
 CDEF1=	-DTERMIOS			# Use the termios structure
 CDEF2=	-DSIGTYPE=void			# Standard definition
diff -Naur ppp-2.4.4.orig/pppd/Makefile.linux ppp-2.4.4/pppd/Makefile.linux
--- ppp-2.4.4.orig/pppd/Makefile.linux	2007-05-15 16:42:59.000000000 -0400
+++ ppp-2.4.4/pppd/Makefile.linux	2007-05-15 16:54:46.000000000 -0400
@@ -5,10 +5,11 @@
 
 # Default installation locations
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
-INCDIR = $(DESTDIR)/include
-LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)
+PREFIX = $(DESTDIR)
+BINDIR = $(INSTROOT)$(PREFIX)/sbin
+MANDIR = $(INSTROOT)$(PREFIX)/share/man/man8
+INCDIR = $(INSTROOT)$(PREFIX)/include
+LIBDIR = $(INSTROOT)$(PREFIX)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)
 
 TARGETS = pppd
 
diff -Naur ppp-2.4.4.orig/pppd/Makefile.linux~ ppp-2.4.4/pppd/Makefile.linux~
--- ppp-2.4.4.orig/pppd/Makefile.linux~	2007-05-15 16:42:59.000000000 -0400
+++ ppp-2.4.4/pppd/Makefile.linux~	2007-05-15 16:42:59.000000000 -0400
@@ -8,6 +8,7 @@
 BINDIR = $(DESTDIR)/sbin
 MANDIR = $(DESTDIR)/share/man/man8
 INCDIR = $(DESTDIR)/include
+LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)
 
 TARGETS = pppd
 
@@ -32,7 +33,7 @@
 
 CC = gcc
 #
-COPTS = -Wall $(RPM_OPT_FLAGS)
+COPTS = -Wall $(RPM_OPT_FLAGS) -DLIBDIR=\""$(LIBDIR)"\"
 LIBS = -lutil
 
 # Uncomment the next 2 lines to include support for Microsoft's
@@ -68,7 +69,7 @@
 PLUGIN=y
 
 # Enable Microsoft proprietary Callback Control Protocol
-#CBCP=y
+CBCP=y
 
 # Enable EAP SRP-SHA1 authentication (requires libsrp)
 #USE_SRP=y
diff -Naur ppp-2.4.4.orig/pppd/Makefile.linux.rej ppp-2.4.4/pppd/Makefile.linux.rej
--- ppp-2.4.4.orig/pppd/Makefile.linux.rej	1969-12-31 19:00:00.000000000 -0500
+++ ppp-2.4.4/pppd/Makefile.linux.rej	2007-05-15 16:44:29.000000000 -0400
@@ -0,0 +1,25 @@
+***************
+*** 5,14 ****
+  
+  # Default installation locations
+  DESTDIR = $(INSTROOT)@DESTDIR@
+- BINDIR = $(DESTDIR)/sbin
+- MANDIR = $(DESTDIR)/share/man/man8
+  INCDIR = $(DESTDIR)/include
+  
+  TARGETS = pppd
+  
+  PPPDSRCS = main.c magic.c fsm.c lcp.c ipcp.c upap.c chap-new.c md5.c ccp.c \
+--- 5,16 ----
+  
+  # Default installation locations
+  DESTDIR = $(INSTROOT)@DESTDIR@
++ PREFIX = $(DESTDIR)
++ BINDIR = $(INSTROOT)$(PREFIX)/sbin
++ MANDIR = $(INSTROOT)$(PREFIX)/share/man/man8
+  INCDIR = $(DESTDIR)/include
+  
++ 
+  TARGETS = pppd
+  
+  PPPDSRCS = main.c magic.c fsm.c lcp.c ipcp.c upap.c chap-new.c md5.c ccp.c \
diff -Naur ppp-2.4.4.orig/pppd/plugins/Makefile.linux ppp-2.4.4/pppd/plugins/Makefile.linux
--- ppp-2.4.4.orig/pppd/plugins/Makefile.linux	2007-05-15 16:42:59.000000000 -0400
+++ ppp-2.4.4/pppd/plugins/Makefile.linux	2007-05-15 16:48:30.000000000 -0400
@@ -5,9 +5,10 @@
 INSTALL	= install
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
-LIBDIR = $(DESTDIR)/lib/$(shell $(CC) -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
+PREFIX = $(DESTDIR)
+BINDIR = $(INSTROOT)$(PREFIX)/sbin
+MANDIR = $(INSTROOT)$(PREFIX)/share/man/man8
+LIBDIR = $(INSTROOT)$(PREFIX)/lib/$(shell $(CC) -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
 
 SUBDIRS := rp-pppoe pppoatm
 # Uncomment the next line to include the radius authentication plugin
diff -Naur ppp-2.4.4.orig/pppd/plugins/Makefile.linux~ ppp-2.4.4/pppd/plugins/Makefile.linux~
--- ppp-2.4.4.orig/pppd/plugins/Makefile.linux~	1969-12-31 19:00:00.000000000 -0500
+++ ppp-2.4.4/pppd/plugins/Makefile.linux~	2007-05-15 16:42:59.000000000 -0400
@@ -0,0 +1,41 @@
+CC	= gcc
+COPTS	= $(RPM_OPT_FLAGS)
+CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
+LDFLAGS	= -shared
+INSTALL	= install
+
+DESTDIR = $(INSTROOT)@DESTDIR@
+BINDIR = $(DESTDIR)/sbin
+MANDIR = $(DESTDIR)/share/man/man8
+LIBDIR = $(DESTDIR)/lib/$(shell $(CC) -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
+
+SUBDIRS := rp-pppoe pppoatm
+# Uncomment the next line to include the radius authentication plugin
+SUBDIRS += radius
+PLUGINS := minconn.so passprompt.so passwordfd.so winbind.so
+
+# include dependencies if present
+ifeq (.depend,$(wildcard .depend))
+include .depend
+endif
+
+all:	$(PLUGINS)
+	for d in $(SUBDIRS); do $(MAKE) $(MFLAGS) -C $$d all; done
+
+%.so: %.c
+	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) $^
+
+VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../patchlevel.h)
+
+install: $(PLUGINS)
+	$(INSTALL) -d $(LIBDIR)
+	$(INSTALL) $? $(LIBDIR)
+	for d in $(SUBDIRS); do $(MAKE) $(MFLAGS) -C $$d install; done
+
+clean:
+	rm -f *.o *.so *.a
+	for d in $(SUBDIRS); do $(MAKE) $(MFLAGS) -C $$d clean; done
+
+depend:
+	$(CPP) -M $(CFLAGS) *.c >.depend
+	for d in $(SUBDIRS); do $(MAKE) $(MFLAGS) -C $$d depend; done
diff -Naur ppp-2.4.4.orig/pppd/plugins/pppoatm/Makefile.linux ppp-2.4.4/pppd/plugins/pppoatm/Makefile.linux
--- ppp-2.4.4.orig/pppd/plugins/pppoatm/Makefile.linux	2007-05-15 16:42:59.000000000 -0400
+++ ppp-2.4.4/pppd/plugins/pppoatm/Makefile.linux	2007-05-15 16:50:16.000000000 -0400
@@ -7,7 +7,8 @@
 #***********************************************************************
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
+PREFIX = $(DESTDIR)
+LIBDIR = $(INSTROOT)$(PREFIX)/lib/$(shell $(CC) -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
 
diff -Naur ppp-2.4.4.orig/pppd/plugins/radius/Makefile.linux ppp-2.4.4/pppd/plugins/radius/Makefile.linux
--- ppp-2.4.4.orig/pppd/plugins/radius/Makefile.linux	2007-05-15 16:42:59.000000000 -0400
+++ ppp-2.4.4/pppd/plugins/radius/Makefile.linux	2007-05-15 16:49:30.000000000 -0400
@@ -4,8 +4,10 @@
 #
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-MANDIR = $(DESTDIR)/share/man/man8
-LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
+PREFIX = $(DESTDIR)
+BINDIR = $(INSTROOT)$(PREFIX)/sbin
+MANDIR = $(INSTROOT)$(PREFIX)/share/man/man8
+LIBDIR = $(INSTROOT)$(PREFIX)/lib/$(shell $(CC) -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
 
diff -Naur ppp-2.4.4.orig/pppd/plugins/rp-pppoe/Makefile.linux ppp-2.4.4/pppd/plugins/rp-pppoe/Makefile.linux
--- ppp-2.4.4.orig/pppd/plugins/rp-pppoe/Makefile.linux	2007-05-15 16:42:59.000000000 -0400
+++ ppp-2.4.4/pppd/plugins/rp-pppoe/Makefile.linux	2007-05-15 16:50:22.000000000 -0400
@@ -14,8 +14,8 @@
 #***********************************************************************
 
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
-LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)/pppd/$(PPPDVERSION)
+PREFIX = $(DESTDIR)
+LIBDIR = $(INSTROOT)$(PREFIX)/lib/$(shell $(CC) -print-multi-os-directory 2> /dev/null)/pppd/$(PPPDVERSION)
 
 PPPDVERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
 
diff -Naur ppp-2.4.4.orig/pppdump/Makefile.linux ppp-2.4.4/pppdump/Makefile.linux
--- ppp-2.4.4.orig/pppdump/Makefile.linux	2007-05-15 16:42:59.000000000 -0400
+++ ppp-2.4.4/pppdump/Makefile.linux	2007-05-15 16:53:32.000000000 -0400
@@ -1,6 +1,7 @@
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+PREFIX = $(DESTDIR)
+BINDIR = $(INSTROOT)$(PREFIX)/sbin
+MANDIR = $(INSTROOT)$(PREFIX)/share/man/man8
 
 CFLAGS= $(RPM_OPT_FLAGS) -I../include/net
 OBJS = pppdump.o bsd-comp.o deflate.o zlib.o
diff -Naur ppp-2.4.4.orig/pppstats/Makefile.linux ppp-2.4.4/pppstats/Makefile.linux
--- ppp-2.4.4.orig/pppstats/Makefile.linux	2007-05-15 16:42:59.000000000 -0400
+++ ppp-2.4.4/pppstats/Makefile.linux	2007-05-15 16:52:36.000000000 -0400
@@ -3,8 +3,9 @@
 # $Id: Makefile.linux,v 1.9 2006/06/04 05:07:46 paulus Exp $
 #
 DESTDIR = $(INSTROOT)@DESTDIR@
-BINDIR = $(DESTDIR)/sbin
-MANDIR = $(DESTDIR)/share/man/man8
+PREFIX = $(DESTDIR)
+BINDIR = $(INSTROOT)$(PREFIX)/sbin
+MANDIR = $(INSTROOT)$(PREFIX)/share/man/man8
 
 PPPSTATSRCS = pppstats.c
 PPPSTATOBJS = pppstats.o
