diff -ur vsftpd-2.0.3-old/vsftpd.conf vsftpd-2.0.3/vsftpd.conf
--- vsftpd-2.0.3-old/vsftpd.conf	2006-07-28 13:36:12.000000000 -0400
+++ vsftpd-2.0.3/vsftpd.conf	2006-07-28 13:38:00.000000000 -0400
@@ -103,3 +103,5 @@
 # the presence of the "-R" option, so there is a strong case for enabling it.
 #ls_recurse_enable=YES
 
+# Comment out this line if you are using xinetd to run this service
+listen=YES
diff -ur vsftpd-2.0.3-old/xinetd.d/vsftpd vsftpd-2.0.3/xinetd.d/vsftpd
--- vsftpd-2.0.3-old/xinetd.d/vsftpd	2006-07-28 13:36:12.000000000 -0400
+++ vsftpd-2.0.3/xinetd.d/vsftpd	2006-07-28 13:37:21.000000000 -0400
@@ -3,6 +3,9 @@
 #   The vsftpd FTP server serves FTP connections. It uses
 #   normal, unencrypted usernames and passwords for authentication.
 # vsftpd is designed to be secure.
+
+# If you want to use xinetd to start vsftpd, please comment out "listen=YES" in /etc/vsftpd/vsftpd.conf
+
 service ftp
 {
         socket_type             = stream
