*** sendmail-8.12.10/devtools/bin/Build	2003-08-20 14:08:07.000000000 -0400
--- sendmail-8.12.10/devtools/bin/Build.new	2004-05-26 11:21:52.000000000 -0400
***************
*** 326,331 ****
--- 326,334 ----
  	os=`uname -s | sed -e 's/\//-/g' -e 's/ //g'`
  	rel=`uname -r | sed -e 's/(/-/g' -e 's/)//g' -e 's/ //g'`
  fi
+ arch=%(unamearch)s
+ os=Linux
+ rel=""
  
  #
  #  Tweak the values we have already got.  PLEASE LIMIT THESE to
