--- groff-1.18.1/src/roff/troff/node.cpp.orig	2003-01-08 20:29:35.000000000 +0200
+++ groff-1.18.1/src/roff/troff/node.cpp	2003-01-08 22:37:46.000000000 +0200
@@ -988,8 +988,10 @@
   set_font(tf);
   unsigned char c = ci->get_ascii_code();
   if (c == '\0') {
-    glyph_color(gcol);
-    fill_color(fcol);
+    if (gcol)
+      glyph_color(gcol);
+    if (fcol)
+      fill_color(fcol);
     flush_tbuf();
     do_motion();
     check_charinfo(tf, ci);
