--- mkconfig.rb.orig    2009-03-27 15:54:22.000000000 -0400
+++ mkconfig.rb 2009-03-27 15:54:39.000000000 -0400
@@ -102,7 +102,7 @@
 
 drive = File::PATH_SEPARATOR == ';'
 
-prefix = '/lib/ruby/' + RUBY_VERSION + '/' + RUBY_PLATFORM
+prefix = (/x86_64/=~RUBY_PLATFORM ? '/lib64/ruby/' : '/lib/ruby/') + RUBY_VERSION.sub(/\.\d+$/, '') + '/' + RUBY_PLATFORM
 print "  TOPDIR = File.dirname(__FILE__).chomp!(#{prefix.dump})\n"
 print "  DESTDIR = ", (drive ? "TOPDIR && TOPDIR[/\\A[a-z]:/i] || " : ""), "'' unless defined? DESTDIR\n"
 print "  CONFIG = {}\n"

