--- a/src/splashy_video.c	2009-01-09 23:09:41.000000000 +1100
+++ b/src/splashy_video.c	2009-01-09 23:09:51.000000000 +1100
@@ -145,7 +145,7 @@
          * as F2 is pressed. Let's give'm a show for their
          * money!
          */
-        if (visible)
+        if (visible && video.textbox)  /* close #473230 thanks John Hughes */
         {
                 video.textbox->offscreen->Blit (video.textbox->surface,
                                                 video.textbox->offscreen,
