--- src/splashy_config-functions.c.orig	2008-09-26 21:47:39.000000000 +0300
+++ src/splashy_config-functions.c	2008-09-26 21:48:35.000000000 +0300
@@ -70,8 +70,8 @@ set_new_theme (gchar * theme_name)
         g_string_append (new_theme, G_DIR_SEPARATOR_S);
         g_string_append (new_theme, theme_name);
 
-        g_print (_(">Set theme as: "));
-        g_print (theme_name);
+        //g_print (_(">Set theme as: "));
+        //g_print (theme_name);
         /*
          ** Check if the theme exist
          */
@@ -91,9 +91,7 @@ set_new_theme (gchar * theme_name)
                 ret = RETURN_OK;
         }
 
-        if (ret == RETURN_OK)
-                PRINT_DONE;
-        else
+        if (ret != RETURN_OK)
                 PRINT_FAIL;
               
         g_string_free (new_theme, TRUE);
@@ -299,12 +297,12 @@ remove_theme (gchar * theme_name)
 {
         need_root ();
         gint ret = RETURN_ERROR;
-        g_print (_(">Remove theme"));
+        //g_print (_(">Remove theme"));
         GString *path =
                 g_string_new (splashy_get_config_string (SPL_THEMES_DIR));
         g_string_append (path, G_DIR_SEPARATOR_S);
         g_string_append (path, theme_name);
-        g_print (" %s", theme_name);
+        //g_print (" %s", theme_name);
 
 
         const gchar *file = NULL;
@@ -358,7 +356,7 @@ remove_theme (gchar * theme_name)
                 }
                 else
                 {
-                        PRINT_DONE;
+                        //PRINT_DONE;
                         ret = RETURN_OK;
                 }
         }
