diff -ruN cups-1.1.20/cups/Makefile cups.pic/cups/Makefile
--- cups-1.1.20/cups/Makefile	2004-12-28 15:05:25.312035773 -0600
+++ cups.pic/cups/Makefile	2004-12-28 15:07:23.218929232 -0600
@@ -25,6 +25,7 @@
 #
 
 include ../Makedefs
+CFLAGS += -fpic
 
 #
 # Object files...
diff -ruN cups-1.1.20/filter/Makefile cups.pic/filter/Makefile
--- cups-1.1.20/filter/Makefile	2004-12-28 15:05:25.366023818 -0600
+++ cups.pic/filter/Makefile	2004-12-28 15:13:42.654957982 -0600
@@ -37,10 +37,14 @@
 		image-sgi.o image-sgilib.o image-sun.o image-tiff.o \
 		image-zoom.o image.o raster.o
 FORMOBJS =	form-attr.o form-main.o form-ps.o form-text.o form-tree.o
-OBJS	=	$(HPGLOBJS) $(IMAGEOBJS) $(FORMOBJS) \
+LIBOBJS =	$(HPGLOBJS) $(IMAGEOBJS) $(FORMOBJS)
+BINOBJS	=	$(HPGLOBJS) $(IMAGEOBJS) $(FORMOBJS) \
 		gziptoany.o imagetops.o imagetoraster.o common.o pstops.o \
 		rastertodymo.o rastertoepson.o rastertohp.o \
 		texttops.o textcommon.o
+OBJS =		$(LIBOBJS) $(BINOBJS)
+
+$(LIBOBJS):    CFLAGS += -fpic
 
 #
 # Make all targets...
diff -ruN cups-1.1.20/pdftops/Makefile cups.pic/pdftops/Makefile
--- cups-1.1.20/pdftops/Makefile	2004-12-28 15:05:25.400016290 -0600
+++ cups.pic/pdftops/Makefile	2004-12-28 15:10:08.101432116 -0600
@@ -10,6 +10,7 @@
 
 include ../Makedefs
 
+
 #
 # Object files...
 #
@@ -58,6 +59,9 @@
 		UnicodeMap.o \
 		XRef.o
 
+# For libxpdf.a:
+$(LIBOBJS): CXXFLAGS += -fPIC
+$(LIBOBJS): CFLAGS += -fPIC
 
 OBJS	=	pdftops.o $(LIBOBJS)
 
