diff -Naur raa-branding-2.1.0.orig/raa/templates/login.kid raa-branding-2.1.0/raa/templates/login.kid
--- raa-branding-2.1.0.orig/raa/templates/login.kid	2007-04-23 14:57:27.000000000 -0400
+++ raa-branding-2.1.0/raa/templates/login.kid	2007-06-28 16:10:18.000000000 -0400
@@ -19,7 +19,7 @@
         addLoadEvent(doFocus);
     </script>
 </head>
-<body>
+<body id="pageLogin">
     <p py:if="message != 'None'" class="message">${message}</p>
     <div id="loginBoxRight" class="palette">
         <img class="left" src="${tg.url('/static/images/header_orange_left.png')}" alt="" />
@@ -27,20 +27,20 @@
         <div class="loginBoxHeader">Sign In</div>
             <div class="loginBox">
             <form action="${previous_url}" method="POST">
-                <table width="100%">
+                <table>
                 <tr>
-                    <td class="label" width="40%">
+                    <td class="label">
                         <label for="user_name">User Name:</label>
                     </td>
-                    <td class="field" width="60%">
+                    <td class="field">
                         <input type="text" id="user_name" name="user_name" />
                     </td>
                 </tr>
                 <tr>
-                    <td class="label" width="40%">
+                    <td class="label">
                         <label for="password">Password:</label>
                     </td>
-                    <td class="field" width="60%">
+                    <td class="field">
                         <input type="password" id="password" name="password" />
                     </td>
                 </tr>
@@ -58,6 +58,10 @@
         </div>
     </div>
 
+    <div>
+        The default login username is <b>admin</b> and password is <b>password</b>.
+    </div>
+
     <div py:if="locals().has_key('user') and locals().has_key('pwd')">
         The default login username is <b>${user}</b> and password is <b>${pwd}</b>.
     </div>
