diff -r 8890a9c9a487 Makefile.am
--- a/Makefile.am	Fri May 02 10:06:05 2008 -0400
+++ b/Makefile.am	Fri May 02 10:09:29 2008 -0400
@@ -26,6 +26,7 @@
 	sed	-e 's|[@]BINDIR[@]|$(bindir)|g' \
 		-e 's|[@]LIBEXECDIR[@]|$(libexecdir)|g' \
 		-e 's|[@]PREFIX[@]|$(prefix)|g' \
+		-e 's|[@]LIBDIR[@]|$(libdir)|g' \
 		< $(srcdir)/nm-openvpn-service.name.in \
 		> nm-openvpn-service.name
 
diff -r 8890a9c9a487 nm-openvpn-service.name.in
--- a/nm-openvpn-service.name.in	Fri May 02 10:06:05 2008 -0400
+++ b/nm-openvpn-service.name.in	Fri May 02 10:09:29 2008 -0400
@@ -5,5 +5,5 @@
 
 [GNOME]
 auth-dialog=@LIBEXECDIR@/nm-openvpn-auth-dialog
-properties=@PREFIX@/${LIB}/libnm-openvpn-properties
+properties=@LIBDIR@/libnm-openvpn-properties
 
