? config.guess.cdbs-orig
? config.sub.cdbs-orig
? debian
Index: src/nm-openvpn-service.c
===================================================================
RCS file: /cvs/gnome/NetworkManager/vpn-daemons/openvpn/src/nm-openvpn-service.c,v
retrieving revision 1.7
diff -u -r1.7 nm-openvpn-service.c
--- src/nm-openvpn-service.c	11 Dec 2005 01:11:54 -0000	1.7
+++ src/nm-openvpn-service.c	16 Dec 2005 18:48:59 -0000
@@ -45,7 +45,6 @@
 #include <ctype.h>
 
 #include <NetworkManager/NetworkManager.h>
-#include <NetworkManager/NetworkManagerVPN.h>
 
 #include "nm-openvpn-service.h"
 #include "nm-utils.h"
@@ -1242,6 +1241,7 @@
   guint32		ip4_nbns_len;
   gboolean		success = FALSE;
   char *                empty = "";
+  char *                login_banner = "Login Successful";
 
   g_return_if_fail (data != NULL);
   g_return_if_fail (con != NULL);
@@ -1283,7 +1283,7 @@
 				DBUS_TYPE_ARRAY, DBUS_TYPE_UINT32, &ip4_dns, ip4_dns_len,
 				DBUS_TYPE_ARRAY, DBUS_TYPE_UINT32, &ip4_nbns, ip4_nbns_len,
 				DBUS_TYPE_STRING, &empty,
-				DBUS_TYPE_STRING, &empty,
+				DBUS_TYPE_STRING, &login_banner,
 				DBUS_TYPE_INVALID);
 
       if (!dbus_connection_send (data->con, signal, NULL))
