diff -r caed5b64d318 src/nm-openvpn-service.c
--- a/src/nm-openvpn-service.c	Thu Sep 18 20:33:18 2008 -0400
+++ b/src/nm-openvpn-service.c	Thu Sep 18 20:35:32 2008 -0400
@@ -646,13 +646,6 @@
 	add_openvpn_arg (args, "--syslog");
 	add_openvpn_arg (args, "nm-openvpn");
 
-	/* Punch script security in the face; this option was added to OpenVPN 2.1-rc9
-	 * and defaults to disallowing any scripts, a behavior change from previous
-	 * versions.
-	 */
-	add_openvpn_arg (args, "--script-security");
-	add_openvpn_arg (args, "2");
-
 	/* Up script, called when connection has been established or has been restarted */
 	add_openvpn_arg (args, "--up");
 	add_openvpn_arg (args, NM_OPENVPN_HELPER_PATH);
