--- qbittorrent/qBittorrent-release-3.1.0/src/Icons/qBittorrent.desktop	2013-10-12 16:52:45.000000000 +0200
+++ qbittorrent/qBittorrent-release-3.1.0/src/Icons/qBittorrent-fix.desktop	2013-10-17 19:22:36.395735953 +0200
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Categories=Network;FileTransfer;P2P;Qt;
-Exec=qbittorrent %U
+Exec=qbittorrent
 GenericName=BitTorrent client
 Comment=Download and share files over BitTorrent
 Icon=qbittorrent
