diff --git a/src/statsdialog.ui b/src/statsdialog.ui
index 7829b15..9d8d555 100644
--- a/src/statsdialog.ui
+++ b/src/statsdialog.ui
@@ -34,7 +34,7 @@
         </property>
        </widget>
       </item>
-      <item row="11" column="1" alignment="Qt::AlignRight">
+      <item row="11" column="1">
        <widget class="QLabel" name="labelPeers">
         <property name="text">
          <string notr="true">TextLabel</string>
@@ -48,14 +48,14 @@
         </property>
        </widget>
       </item>
-      <item row="1" column="1" alignment="Qt::AlignRight">
+      <item row="1" column="1">
        <widget class="QLabel" name="labelAlltimeUL">
         <property name="text">
          <string notr="true">TextLabel</string>
         </property>
        </widget>
       </item>
-      <item row="0" column="1" alignment="Qt::AlignRight">
+      <item row="0" column="1">
        <widget class="QLabel" name="labelAlltimeDL">
         <property name="text">
          <string notr="true">TextLabel</string>
@@ -69,7 +69,7 @@
         </property>
        </widget>
       </item>
-      <item row="3" column="1" alignment="Qt::AlignRight">
+      <item row="3" column="1">
        <widget class="QLabel" name="labelGlobalRatio">
         <property name="text">
          <string notr="true">TextLabel</string>
@@ -83,7 +83,7 @@
         </property>
        </widget>
       </item>
-      <item row="2" column="1" alignment="Qt::AlignRight">
+      <item row="2" column="1">
        <widget class="QLabel" name="labelWaste">
         <property name="text">
          <string notr="true">TextLabel</string>
@@ -106,14 +106,14 @@
         </property>
        </widget>
       </item>
-      <item row="1" column="1" alignment="Qt::AlignRight">
+      <item row="1" column="1">
        <widget class="QLabel" name="labelCacheHits">
         <property name="text">
          <string notr="true">TextLabel</string>
         </property>
        </widget>
       </item>
-      <item row="2" column="1" alignment="Qt::AlignRight">
+      <item row="2" column="1">
        <widget class="QLabel" name="labelTotalBuf">
         <property name="text">
          <string notr="true">TextLabel</string>
@@ -136,28 +136,28 @@
       <string>Performance statistics</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
-      <item row="3" column="1" alignment="Qt::AlignRight">
+      <item row="3" column="1">
        <widget class="QLabel" name="labelJobsTime">
         <property name="text">
          <string notr="true">TextLabel</string>
         </property>
        </widget>
       </item>
-      <item row="2" column="1" alignment="Qt::AlignRight">
+      <item row="2" column="1">
        <widget class="QLabel" name="labelQueuedJobs">
         <property name="text">
          <string notr="true">TextLabel</string>
         </property>
        </widget>
       </item>
-      <item row="0" column="1" alignment="Qt::AlignRight">
+      <item row="0" column="1">
        <widget class="QLabel" name="labelWriteStarve">
         <property name="text">
          <string notr="true">TextLabel</string>
         </property>
        </widget>
       </item>
-      <item row="1" column="1" alignment="Qt::AlignRight">
+      <item row="1" column="1">
        <widget class="QLabel" name="labelReadStarve">
         <property name="text">
          <string notr="true">TextLabel</string>
@@ -199,7 +199,7 @@
         </property>
        </widget>
       </item>
-      <item row="4" column="1" alignment="Qt::AlignRight">
+      <item row="4" column="1">
        <widget class="QLabel" name="labelQueuedBytes">
         <property name="text">
          <string notr="true">TextLabel</string>
