traw.orig/setup.py 2006-04-02 01:44:29.000000000 -0500
+++ straw/setup.py      2006-08-16 01:20:07.000000000 -0400
@@ -73,12 +73,7 @@
     # the 'gnome' module. But as of 2.10, the gtkhtml2 module was
     # removed in 'gnome' and added in 'gnome-extras' module.
     # The egg.trayicon module is also included in g-p-e, so no need to check here.
-    mod_list = [
-        ('gnome', 'python-gnome2', 0),
-        ('gtkhtml2', "python-gnome2-extras", 0),
-        ('gconf', "python-gnome2", 1),
-        ('bsddb.db', 'python-bsddb', 0),
-        ('dbus', 'python-dbus', 0)]
+    mod_list = [ ]

     for m, w, x in mod_list:
        if nogtk == 1 and m == 'gtkhtml2':
@@ -91,16 +86,6 @@
                 sys.exit("Error: '%s' module in package '%s' is required to install %s" \
                       % (m,w,strawenv['APPNAME']))

-    # Check for dbus version
-    if getattr(dbus, 'version', None) is None:
-        print "Warning: Recent versions of  DBus is needed to run Straw"
-
-    # gtk.glade needs special care (ugly...)
-    path = imp.find_module('gtk')
-    if not os.path.exists(path[1] + '/glade.so'):
-        sys.exit('Error: %s module is required to install %s' \
-                 % ("PyGTK's glade", strawenv['APPNAME']))
-
     # Check for ADNS
     try:
         import adns, ADNS
