diff -Naur gnome-main-menu-0.6.3.svn20070119.orig/main-menu/etc/main-menu-rug.desktop.in gnome-main-menu-0.6.3.svn20070119/main-menu/etc/main-menu-rug.desktop.in
--- gnome-main-menu-0.6.3.svn20070119.orig/main-menu/etc/main-menu-rug.desktop.in	2007-01-19 23:15:55.000000000 -0500
+++ gnome-main-menu-0.6.3.svn20070119/main-menu/etc/main-menu-rug.desktop.in	2007-01-19 23:33:46.000000000 -0500
@@ -1,7 +1,7 @@
 [Desktop Entry]
 X-SuSE-translate=true
 _Name=Software Update
-Exec=gnomesu rug update unstable
+Exec=gnome-open http://localhost:8004/rAA/updatetroves/UpdateTroves/
 Icon=system-software-update
 Terminal=true
 Type=Application
