diff -Naur gnome-main-menu-0.6.3.svn20070119.orig/control-center/src/control-center.c gnome-main-menu-0.6.3.svn20070119/control-center/src/control-center.c
--- gnome-main-menu-0.6.3.svn20070119.orig/control-center/src/control-center.c	2007-01-19 23:15:55.000000000 -0500
+++ gnome-main-menu-0.6.3.svn20070119/control-center/src/control-center.c	2007-01-20 22:20:11.000000000 -0500
@@ -144,7 +144,7 @@
 	}
 
 	AppShellData *app_data = appshelldata_new (
-		"preferences.menu", NULL, CONTROL_CENTER_PREFIX, GTK_ICON_SIZE_DIALOG, TRUE);
+		"settings.menu", NULL, CONTROL_CENTER_PREFIX, GTK_ICON_SIZE_DIALOG, TRUE);
 	generate_categories (app_data);
 
 	GSList *actions = get_actions_list ();
diff -Naur gnome-main-menu-0.6.3.svn20070119.orig/main-menu/etc/main-menu-rug.desktop.in gnome-main-menu-0.6.3.svn20070119/main-menu/etc/main-menu-rug.desktop.in
--- gnome-main-menu-0.6.3.svn20070119.orig/main-menu/etc/main-menu-rug.desktop.in	2007-01-19 23:15:55.000000000 -0500
+++ gnome-main-menu-0.6.3.svn20070119/main-menu/etc/main-menu-rug.desktop.in	2007-01-19 23:33:46.000000000 -0500
@@ -1,7 +1,7 @@
 [Desktop Entry]
 X-SuSE-translate=true
 _Name=Software Update
-Exec=gnomesu rug update unstable
+Exec=gnome-open http://localhost:8004/rAA/updatetroves/UpdateTroves/
 Icon=system-software-update
 Terminal=true
 Type=Application
diff -Naur gnome-main-menu-0.6.3.svn20070119.orig/main-menu/etc/slab.schemas.in.in gnome-main-menu-0.6.3.svn20070119/main-menu/etc/slab.schemas.in.in
--- gnome-main-menu-0.6.3.svn20070119.orig/main-menu/etc/slab.schemas.in.in	2007-01-19 23:15:55.000000000 -0500
+++ gnome-main-menu-0.6.3.svn20070119/main-menu/etc/slab.schemas.in.in	2007-01-20 12:29:03.000000000 -0500
@@ -6,7 +6,7 @@
 			<owner>gnome-main-menu</owner>
 			<type>list</type>
 			<list_type>string</list_type>
-			<default>[MozillaFirefox.desktop,evolution.desktop,banshee.desktop,writer.desktop,f-spot.desktop,nautilus-home.desktop]</default>
+			<default>[MozillaFirefox.desktop,evolution.desktop,banshee.desktop,writer.desktop,f-spot.desktop,gaim.desktop,nautilus-home.desktop]</default>
 			<locale name="C">
 				<short>.desktop files for "Favorite Applications"</short>
 				<long>.desktop files for "Favorite Applications"</long>
