diff -r 3c76a568ad33 main-menu/etc/system-items.xbel
--- a/main-menu/etc/system-items.xbel	Thu May 22 16:25:03 2008 -0400
+++ b/main-menu/etc/system-items.xbel	Thu May 22 16:28:21 2008 -0400
@@ -30,7 +30,7 @@
       </metadata>
     </info>
   </bookmark>
-  <bookmark href="package-manager.desktop" added="2007-01-16T05:53:36Z" modified="2007-01-16T05:53:36Z" visited="2007-01-16T05:53:36Z">
+  <bookmark href="gpk-application.desktop" added="2007-01-16T05:53:36Z" modified="2007-01-16T05:53:36Z" visited="2007-01-16T05:53:36Z">
     <info>
       <metadata owner="http://freedesktop.org">
         <mime:mime-type type="application/x-desktop"/>
@@ -38,7 +38,20 @@
           <bookmark:group>rank-2</bookmark:group>
         </bookmark:groups>
         <bookmark:applications>
-          <bookmark:application name="Install Software" exec="package-manager %F" timestamp="1168926816" count="1"/>
+          <bookmark:application name="Install Software" exec="gpk-application %F" timestamp="1168926816" count="1"/>
+        </bookmark:applications>
+      </metadata>
+    </info>
+  </bookmark>
+  <bookmark href="gpk-update-viewer.desktop" added="2007-01-16T05:53:36Z" modified="2007-01-16T05:53:36Z" visited="2007-01-16T05:53:36Z">
+    <info>
+      <metadata owner="http://freedesktop.org">
+        <mime:mime-type type="application/x-desktop"/>
+        <bookmark:groups>
+          <bookmark:group>rank-2</bookmark:group>
+        </bookmark:groups>
+        <bookmark:applications>
+          <bookmark:application name="Update System" exec="gpk-update-viewer %F" timestamp="1168926816" count="1"/>
         </bookmark:applications>
       </metadata>
     </info>
