Only in powernowd-0.96: powernowd
diff -u powernowd-0.96-orig/powernowd.c powernowd-0.96/powernowd.c
--- powernowd-0.96-orig/powernowd.c	2005-05-09 08:24:55.000000000 -0500
+++ powernowd-0.96/powernowd.c	2005-06-06 14:46:55.000000000 -0500
@@ -853,7 +853,7 @@
 	/* Now the main program loop */
 	while(1) {
 		usleep(poll*1000);
-		for(i=0; i<num_real_cpus; i+=j) {
+		for(i=0; i<ncpus; i+=j) {
 			change = LOWER;
 			/* handle SMT/CMP here */
 			for (j=0; j<cpu[i].threads_per_core; j++) {
Only in powernowd-0.96: powernowd.patch
