diff -r 20cf0910967d src/lxc/start.c
--- a/src/lxc/start.c	Thu Mar 05 09:59:24 2009 -0500
+++ b/src/lxc/start.c	Thu Mar 05 10:25:49 2009 -0500
@@ -48,7 +48,31 @@
 #ifdef HAVE_SYS_SIGNALFD_H 
 #include <sys/signalfd.h>
 #else
-extern int signalfd (int fd, const sigset_t *mask, int flags);
+int signalfd (int fd, const sigset_t *mask, int flags)
+{
+  int retval;
+#ifndef __NR_signalfd4
+/* assume kernel headers are too old */
+#  if __i386__
+#  define __NR_signalfd4 327
+#  elif __x86_64__
+#  define __NR_signalfd4 289
+#  endif
+#endif
+#ifndef __NR_signalfd
+/* assume kernel headers are too old */
+#  if __i386__
+#  define __NR_signalfd 321
+#  elif __x86_64__
+#  define __NR_signalfd 282
+#  endif
+#endif
+  retval = syscall (__NR_signalfd4, fd, mask, _NSIG / 8, flags);
+  if ((errno == ENOSYS) && flags == 0) {
+      retval = syscall (__NR_signalfd, fd, mask, _NSIG / 8);
+  }
+  return retval;
+}
 #endif
 
 #include "error.h"
