--- src/Makefile.in	2007-07-24 23:35:21.000000000 +0200
+++ src/Makefile.in.new	2007-12-18 12:15:51.000000000 +0100
@@ -196,7 +196,7 @@
 
 # Damit Dinge nach /usr/games installiert werden können...
 # (Da gehört das Binary laut BRL hin)
-gamesdir = $(prefix)/games
+gamesdir = $(prefix)/bin
 cuyo_SOURCES = bilddatei.cpp blopbesitzer.cpp blop.cpp blopgitter.cpp \
 	configdatei.cpp fall.cpp cuyo.cpp kiplayer.cpp leveldaten.cpp \
 	main.cpp nachbariterator.cpp sorte.cpp spielfeld.cpp \
