--- evolution-data-server-1.2.0/camel/providers/local/camel-local-folder.c.orig	2005-03-15 13:48:11.461676880 -0500
+++ evolution-data-server-1.2.0/camel/providers/local/camel-local-folder.c	2005-03-15 13:48:36.346893752 -0500
@@ -377,8 +377,8 @@
 					/* $HOME relative path + protocol string */
 					folder->description = g_strdup_printf(_("~%%s (%%s)"), path+strlen(tmp),
 									      ((CamelService *)folder->parent_store)->url->protocol);
-				else if ((tmp = "/var/spool/mail") && strncmp(tmp, path, strlen(tmp)) == 0)
-					/* /var/spool/mail relative path + protocol */
+				else if ((tmp = "%(servicedir)s/spool/mail") && strncmp(tmp, path, strlen(tmp)) == 0)
+					/* %(servicedir)s/spool/mail relative path + protocol */
 					folder->description = g_strdup_printf(_("mailbox:%%s (%%s)"), path+strlen(tmp),
 									      ((CamelService *)folder->parent_store)->url->protocol);
 				else if ((tmp = "/var/mail") && strncmp(tmp, path, strlen(tmp)) == 0)
