Use Debian executable name
Index: b/ruby/qtruby/bin/rbqtapi
===================================================================
--- a/ruby/qtruby/bin/rbqtapi	2010-07-10 01:48:45.855182000 +0200
+++ b/ruby/qtruby/bin/rbqtapi	2010-07-10 01:48:57.731182001 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby1.8
 
 # Note: this program is part of qtruby and makes use of its internal functions.
 #       You should not rely on those in your own programs.
