Index: b/csharp/plasma/examples/applets/CMakeLists.txt
===================================================================
--- a/csharp/plasma/examples/applets/CMakeLists.txt	2010-07-10 01:48:45.663182007 +0200
+++ b/csharp/plasma/examples/applets/CMakeLists.txt	2010-07-10 01:48:57.983182000 +0200
@@ -1,2 +1,3 @@
 add_subdirectory( extendertutorial )
-add_subdirectory( tiger )
+#add_subdirectory( tiger )
+
