Kimono external library should be imported as libkimono

Index: b/csharp/kimono/kde/KUrlExtras.cs
===================================================================
--- a/csharp/kimono/kde/KUrlExtras.cs	2010-07-10 01:48:45.615182001 +0200
+++ b/csharp/kimono/kde/KUrlExtras.cs	2010-07-10 01:48:58.143182000 +0200
@@ -13,13 +13,13 @@
 		public class List : List<KUrl> {
 			delegate bool GetNextDictionaryEntryFn(ref IntPtr key, ref IntPtr value);
 			
-			[DllImport("kimono", CharSet=CharSet.Ansi)]
+			[DllImport("libkimono", CharSet=CharSet.Ansi)]
 			static extern void KUrlListPopulateMimeData(SmokeMarshallers.NoArgs getNextItem, IntPtr mimeData, GetNextDictionaryEntryFn getNextDictionaryEntry, uint flags);
-			[DllImport("kimono", CharSet=CharSet.Ansi)]
+			[DllImport("libkimono", CharSet=CharSet.Ansi)]
 			static extern void KUrlListMimeDataTypes(SmokeMarshallers.FromIntPtr fn);
-			[DllImport("kimono", CharSet=CharSet.Ansi)]
+			[DllImport("libkimono", CharSet=CharSet.Ansi)]
 			static extern bool KUrlListCanDecode(IntPtr mimeData);
-			[DllImport("kimono", CharSet=CharSet.Ansi)]
+			[DllImport("libkimono", CharSet=CharSet.Ansi)]
 			static extern void KUrlListFromMimeData(SmokeMarshallers.FromIntPtr addfn, IntPtr mimeData, GetNextDictionaryEntryFn getNextDictionaryEntry);
 			
 			public List() {}
Index: b/csharp/kimono/kde/KPluginFactoryExtras.cs
===================================================================
--- a/csharp/kimono/kde/KPluginFactoryExtras.cs	2010-07-10 01:48:45.627181998 +0200
+++ b/csharp/kimono/kde/KPluginFactoryExtras.cs	2010-07-10 01:48:58.143182000 +0200
@@ -5,7 +5,7 @@
 	using Qyoto;
 
 	public partial class KPluginFactory : QObject, IDisposable {
-		[DllImport("kimono", CharSet=CharSet.Ansi)]
+		[DllImport("libkimono", CharSet=CharSet.Ansi)]
 		static extern IntPtr KPluginFactory_Create(IntPtr self, string classname, IntPtr parentWidget, IntPtr parent, IntPtr[] args, int numArgs, string keyword);
 
 		public T Create<T>() where T: class {
