Index: b/python/pykde4/kpythonpluginfactory/kpythonpluginfactory.cpp
===================================================================
--- a/python/pykde4/kpythonpluginfactory/kpythonpluginfactory.cpp	2010-07-10 01:48:57.855182000 +0200
+++ b/python/pykde4/kpythonpluginfactory/kpythonpluginfactory.cpp	2010-07-10 01:48:58.567182001 +0200
@@ -307,7 +307,7 @@
 {
     QLibrary *pythonLib = new QLibrary();
     pythonLib->setLoadHints(QLibrary::ExportExternalSymbolsHint);
-    pythonLib->setFileName(LIB_PYTHON);
+    pythonLib->setFileName(LIB_PYTHON ".1");
     pythonLib->load();
     return pythonLib;
 }
