Index: csharp/qttest/qttest-sharp.pc
===================================================================
--- csharp/qttest/qttest-sharp.pc	(revision 0)
+++ csharp/qttest/qttest-sharp.pc	(revision 929133)
@@ -0,0 +1,4 @@
+Name: Qyoto - QtTest
+Description: Qt .NET/Mono bindings (QtTest)
+Version: 2.2.0
+Libs: -r:%(libdir)s/mono/2.0/qttest.dll
Index: csharp/qttest/CMakeLists.txt
===================================================================
--- csharp/qttest/CMakeLists.txt	(revision 929132)
+++ csharp/qttest/CMakeLists.txt	(revision 929133)
@@ -22,3 +22,4 @@
 
 INSTALL(TARGETS qttest-sharp LIBRARY DESTINATION ${LIB_INSTALL_DIR})
 INSTALL_GAC(qttest)
+INSTALL(FILES ${PROJECT_NAME}.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
Index: csharp/qtwebkit/qtwebkit-sharp.pc
===================================================================
--- csharp/qtwebkit/qtwebkit-sharp.pc	(revision 0)
+++ csharp/qtwebkit/qtwebkit-sharp.pc	(revision 929133)
@@ -0,0 +1,4 @@
+Name: Qyoto - QtWebKit
+Description: Qt .NET/Mono bindings (QtWebKit)
+Version: 2.2.0
+Libs: -r:%(libdir)s/mono/2.0/qtwebkit.dll
Index: csharp/qtwebkit/CMakeLists.txt
===================================================================
--- csharp/qtwebkit/CMakeLists.txt	(revision 929132)
+++ csharp/qtwebkit/CMakeLists.txt	(revision 929133)
@@ -22,3 +22,4 @@
 
 INSTALL(TARGETS qtwebkit-sharp LIBRARY DESTINATION ${LIB_INSTALL_DIR})
 INSTALL_GAC(qtwebkit)
+INSTALL(FILES ${PROJECT_NAME}.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
Index: csharp/qtuitools/qtuitools-sharp.pc
===================================================================
--- csharp/qtuitools/qtuitools-sharp.pc	(revision 0)
+++ csharp/qtuitools/qtuitools-sharp.pc	(revision 929133)
@@ -0,0 +1,4 @@
+Name: Qyoto - QtUiTools
+Description: Qt .NET/Mono bindings (QtUiTools)
+Version: 2.2.0
+Libs: -r:%(libdir)s/mono/2.0/qtuitools.dll
Index: csharp/qtuitools/CMakeLists.txt
===================================================================
--- csharp/qtuitools/CMakeLists.txt	(revision 929132)
+++ csharp/qtuitools/CMakeLists.txt	(revision 929133)
@@ -22,3 +22,4 @@
 
 INSTALL(TARGETS qtuitools-sharp LIBRARY DESTINATION ${LIB_INSTALL_DIR})
 INSTALL_GAC(qtuitools)
+INSTALL(FILES ${PROJECT_NAME}.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
Index: csharp/qtscript/qtscript-sharp.pc
===================================================================
--- csharp/qtscript/qtscript-sharp.pc	(revision 0)
+++ csharp/qtscript/qtscript-sharp.pc	(revision 929133)
@@ -0,0 +1,4 @@
+Name: Qyoto - QtScript
+Description: Qt .NET/Mono bindings (QtScript)
+Version: 2.2.0
+Libs: -r:%(libdir)s/mono/2.0/qtscript.dll
Index: csharp/qtscript/CMakeLists.txt
===================================================================
--- csharp/qtscript/CMakeLists.txt	(revision 929132)
+++ csharp/qtscript/CMakeLists.txt	(revision 929133)
@@ -23,3 +23,4 @@
 
 INSTALL(TARGETS qtscript-sharp LIBRARY DESTINATION ${LIB_INSTALL_DIR})
 INSTALL_GAC(qtscript)
+INSTALL(FILES ${PROJECT_NAME}.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
Index: csharp/qyoto/qyoto.pc
===================================================================
--- csharp/qyoto/qyoto.pc	(revision 0)
+++ csharp/qyoto/qyoto.pc	(revision 929133)
@@ -0,0 +1,4 @@
+Name: Qyoto
+Description: Qt .NET/Mono bindings
+Version: 4.4.0
+Libs: -r:%(libdir)s/mono/2.0/qt-dotnet.dll
Index: csharp/qyoto/CMakeLists.txt
===================================================================
--- csharp/qyoto/CMakeLists.txt	(revision 929132)
+++ csharp/qyoto/CMakeLists.txt	(revision 929133)
@@ -59,6 +59,8 @@
 src/qyoto.h src/qyotosmokebinding.h src/signalreturnvalue.h src/slotreturnvalue.h
 src/smokeqyoto.h src/virtualmethodcall.h src/virtualmethodreturnvalue.h DESTINATION include/qyoto)
 INSTALL_GAC(qt-dotnet)
+INSTALL(FILES ${PROJECT_NAME}.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
 
+
 ADD_SUBDIRECTORY(tools)
 ADD_SUBDIRECTORY(examples EXCLUDE_FROM_ALL)
