--- kdebindings-4.4.80/smoke/qsci/CMakeLists.txt	2010/05/22 12:19:50	1129443
+++ kdebindings-4.4.80/smoke/qsci/CMakeLists.txt	2010/05/22 12:24:17	1129444
@@ -17,7 +17,7 @@
 	${CMAKE_CURRENT_BINARY_DIR}/x_10.cpp

     COMMAND smokegen
-    ARGS -config "${CMAKE_CURRENT_BINARY_DIR}/../qt/config.xml" -smokeconfig "${CMAKE_CURRENT_SOURCE_DIR}/smokeconfig.xml"
+    ARGS -config "${CMAKE_CURRENT_BINARY_DIR}/../qtgui/config.xml" -smokeconfig "${CMAKE_CURRENT_SOURCE_DIR}/smokeconfig.xml"
          -I "${QSCINTILLA_INCLUDE_DIR}" -- ${CMAKE_CURRENT_SOURCE_DIR}/qscintilla2_includes.h
     DEPENDS smokegen generator_smoke WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
