Index: b/generator/parser/CMakeLists.txt
===================================================================
--- a/generator/parser/CMakeLists.txt	2010-07-10 01:48:45.471181996 +0200
+++ b/generator/parser/CMakeLists.txt	2010-07-10 01:48:58.691182000 +0200
@@ -36,6 +36,7 @@
 
 add_definitions(-D__CPPPARSER_BUILDING)
 add_library(cppparser SHARED ${cppparser_LIB_SRCS})
+set_target_properties(cppparser PROPERTIES VERSION 0a.0.0 SOVERSION 0a)
 target_link_libraries(cppparser ${QT_QTCORE_LIBRARY})
 
 install(TARGETS cppparser LIBRARY DESTINATION ${LIB_INSTALL_DIR})
