diff -Naur usb-modeswitch-1.1.2/Makefile usb-modeswitch-1.1.2.dir/Makefile
--- usb-modeswitch-1.1.2/Makefile	2010-04-17 23:57:11.000000000 +0530
+++ usb-modeswitch-1.1.2.dir/Makefile	2010-04-20 10:28:04.345041264 +0530
@@ -7,7 +7,7 @@
 PREFIX      = $(DESTDIR)/usr
 ETCDIR      = $(DESTDIR)/etc
 UDEVDIR     = $(DESTDIR)/lib/udev
-SBINDIR     = $(PREFIX)/sbin
+SBINDIR     = $(PREFIX)/bin
 MANDIR      = $(PREFIX)/share/man/man1
 
 .PHONY:     clean
