--- nautilus-burn-drive.c.orig	2005-02-08 13:01:56.445214680 +0100
+++ nautilus-burn-drive.c	2005-02-08 13:02:51.579832944 +0100
@@ -628,6 +628,8 @@
 } UnmountData;
 
 static const char *umount_known_locations [] = {
+	"/usr/bin/pumount",
+	"/bin/pumount",
 	"/sbin/umount",
 	"/bin/umount",
 	"/usr/sbin/umount",
@@ -693,7 +695,7 @@
 	int i;
 
 	for (i = 0; known_locations [i]; i++){
-		if (g_file_test (known_locations [i], G_FILE_TEST_EXISTS))
+		if (g_file_test (known_locations [i], G_FILE_TEST_IS_EXECUTABLE))
 			return known_locations [i];
 	}
 	return NULL;
