diff -Naur nautilus-cd-burner-2.15.5.orig/src/nautilus-burn-drive.c nautilus-cd-burner-2.15.5/src/nautilus-burn-drive.c
--- nautilus-cd-burner-2.15.5.orig/src/nautilus-burn-drive.c	2006-06-12 12:34:08.000000000 -0400
+++ nautilus-cd-burner-2.15.5/src/nautilus-burn-drive.c	2006-07-30 10:28:54.000000000 -0400
@@ -610,6 +610,7 @@
 
 #ifndef USE_GNOME_MOUNT
 static const char *umount_known_locations [] = {
+	"/usr/bin/pumount",
 	"/sbin/umount",
 	"/bin/umount",
 	"/usr/sbin/umount",
