diff -ur nautilus-cd-burner-2.10.2.orig/nautilus-burn-drive.c nautilus-cd-burner-2.10.2/nautilus-burn-drive.c
--- nautilus-cd-burner-2.10.2.orig/nautilus-burn-drive.c	2005-04-24 16:13:30.000000000 -0400
+++ nautilus-cd-burner-2.10.2/nautilus-burn-drive.c	2005-07-25 16:41:45.000000000 -0400
@@ -737,6 +737,8 @@
 } UnmountData;
 
 static const char *umount_known_locations [] = {
+	"/usr/bin/pumount",
+	"/bin/pumount",
 	"/sbin/umount",
 	"/bin/umount",
 	"/usr/sbin/umount",
@@ -802,7 +804,7 @@
 	int i;
 
 	for (i = 0; known_locations [i]; i++){
-		if (g_file_test (known_locations [i], G_FILE_TEST_EXISTS))
+		if (g_file_test (known_locations [i], G_FILE_TEST_IS_EXECUTABLE))
 			return known_locations [i];
 	}
 	return NULL;
Only in nautilus-cd-burner-2.10.2: nautilus-burn-drive.c~
