diff -Naur nautilus-cd-burner-2.13.5.orig/nautilus-burn-drive.c nautilus-cd-burner-2.13.5/nautilus-burn-drive.c
--- nautilus-cd-burner-2.13.5.orig/nautilus-burn-drive.c	2006-01-14 08:44:37.000000000 -0500
+++ nautilus-cd-burner-2.13.5/nautilus-burn-drive.c	2006-01-19 10:44:44.000000000 -0500
@@ -1314,6 +1314,8 @@
 }
 
 static const char *umount_known_locations [] = {
+	"/usr/bin/pumount",
+	"/bin/pumount",
 	"/sbin/umount",
 	"/bin/umount",
 	"/usr/sbin/umount",
