--- pywbem/tupleparse.py.orig	2008-06-12 14:17:21.000000000 -0400
+++ pywbem/tupleparse.py	2008-06-12 14:16:21.000000000 -0400
@@ -867,7 +867,7 @@
     val = unpack_value(tt)
     a = attrs(tt)
     embedded_object=None
-    if a.has_key('EmbeddedObject'):
+    if a.has_key('EmbeddedObject') and val is not None:
         val = parse_embeddedObject(val)
         embedded_object=a['EmbeddedObject']
 
@@ -904,7 +904,7 @@
     values = unpack_value(tt)
     a = attrs(tt)
     embedded_object = None
-    if a.has_key('EmbeddedObject'):
+    if a.has_key('EmbeddedObject') and values is None:
         values = parse_embeddedObject(values)
         embedded_object = a['EmbeddedObject']
 
