diff -ru mtr-0.69-old/configure.in mtr-0.69/configure.in
--- mtr-0.69-old/configure.in	2005-05-20 15:42:31.881692200 -0400
+++ mtr-0.69/configure.in	2005-05-20 15:41:55.828173168 -0400
@@ -61,11 +61,12 @@
                            AC_DEFINE(NO_GTK)
                            GTK_OBJ="")
         else
-		AM_PATH_GTK(1.0.0, CFLAGS="$CFLAGS $GTK_CFLAGS"
-                           LIBS="$LIBS $GTK_LIBS",
-                           AC_MSG_WARN(Building without GTK+ display support)
-                           AC_DEFINE(NO_GTK)
-                           GTK_OBJ="")
+                AC_DEFINE(NO_GTK)
+		#AM_PATH_GTK(1.0.0, CFLAGS="$CFLAGS $GTK_CFLAGS"
+                #           LIBS="$LIBS $GTK_LIBS",
+                #           AC_MSG_WARN(Building without GTK+ display support)
+                #           AC_DEFINE(NO_GTK)
+                #           GTK_OBJ="")
 	fi
 else
 	AC_DEFINE(NO_GTK)
