diff -r dd5bb08782d3 pitivi/project.py
--- a/pitivi/project.py	Thu Dec 04 12:47:46 2008 -0500
+++ b/pitivi/project.py	Thu Dec 04 13:07:07 2008 -0500
@@ -99,11 +99,12 @@
             gst.warning("Already loaded !!!")
             return True
         try:
-            res = self._load()
-        except:
-            gst.error("An Exception was raised during loading !")
-            traceback.print_exc()
-            res = False
+            try:
+                res = self._load()
+            except:
+                gst.error("An Exception was raised during loading !")
+                traceback.print_exc()
+                res = False
         finally:
             return res
 
@@ -121,11 +122,12 @@
             path = gst.uri_get_location(self.uri)
             fileobj = open(path, "r")
             try:
-                tree = loader.openFromFile(fileobj)
-                self.fromDataFormat(tree)
-            except ProjectLoadError:
-                gst.error("Error while loading the project !!!")
-                return False
+                try:
+                    tree = loader.openFromFile(fileobj)
+                    self.fromDataFormat(tree)
+                except ProjectLoadError:
+                    gst.error("Error while loading the project !!!")
+                    return False
             finally:
                 fileobj.close()
             self.format = saveformat
