diff -Nur hal/build-tree/hal-0.5.7.1/hald/linux2/blockdev.c hal.new/build-tree/hal-0.5.7.1/hald/linux2/blockdev.c
--- hal-0.5.7.1/hald/linux2/blockdev.c	2006-02-24 05:29:06.000000000 +0100
+++ hal-0.5.7.1/hald/linux2/blockdev.c	2006-10-06 10:44:08.000000000 +0200
@@ -205,6 +205,14 @@
 	while ((mnte = getmntent_r (f, &mnt, buf, sizeof(buf))) != NULL) {
 		struct stat statbuf;
 
+		/* If this is a nfs mount (fstype == 'nfs') ignore the mount. Reason:
+		 *  1. we don't list nfs devices in HAL
+		 *  2. more problematic: stat on mountpoints with 'stale nfs handle' never come
+		 *     back and block complete HAL and all applications using HAL fail.
+		 */
+		if (!strcmp(mnt.mnt_type, "nfs"))
+			continue;
+
 		/* check the underlying device of the mount point */
 		if (stat (mnt.mnt_dir, &statbuf) != 0)
 			continue;
