diff -aur hal-0.5.8.vanilla/hal.conf.in hal-0.5.8/hal.conf.in
--- hal-0.5.8.vanilla/hal.conf.in	2006-09-11 23:56:39.000000000 +0200
+++ hal-0.5.8/hal.conf.in	2006-09-17 12:32:49.000000000 +0200
@@ -54,6 +54,16 @@
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
 
+  <!-- Debian groups policies -->
+  <policy group="powerdev">
+    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+  </policy>
+  <policy group="plugdev">
+    <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+  </policy>
+
   <!-- You can change this to a more suitable user, or make per-group -->
   <policy user="0">
     <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
