diff -ruN hal-0.4.7-old/hald/linux/block_class_device.c hal-0.4.7/hald/linux/block_class_device.c
--- hal-0.4.7-old/hald/linux/block_class_device.c	2005-01-21 16:46:55.000000000 +0100
+++ hal-0.4.7/hald/linux/block_class_device.c	2005-02-24 11:50:55.000000000 +0100
@@ -298,14 +298,19 @@
 
 	/* Check handling */
 	fd = open (device_file, O_RDONLY | O_NONBLOCK);
-	if (fd < 0)
+	if (fd < 0) {
+                HAL_ERROR(("cdrom_get_properties: could not open device %s: %s\n",
+                            device_file, strerror(errno)));
 		return;
+        }
 		
 	if( ioctl (fd, CDROM_SET_OPTIONS, CDO_USE_FFLAGS) < 0 )
 	    HAL_ERROR(("CDROM_SET_OPTIONS failed: %s\n", strerror(errno)));
 		
 	capabilities = ioctl (fd, CDROM_GET_CAPABILITY, 0);
 	if (capabilities < 0) {
+                HAL_ERROR(("cdrom_get_properties: CDROM_GET_CAPABILITY ioctl failed for device %s: %s\n",
+                            device_file, strerror(errno)));
 		close(fd);
 		return;
 	}
