diff -ruN hal-0.4.7-old/fdi/90defaultpolicy/storage-policy.fdi hal-0.4.7/fdi/90defaultpolicy/storage-policy.fdi
--- hal-0.4.7-old/fdi/90defaultpolicy/storage-policy.fdi	2005-03-17 12:53:44.847773000 +0100
+++ hal-0.4.7/fdi/90defaultpolicy/storage-policy.fdi	2005-03-17 12:57:03.976501360 +0100
@@ -15,6 +15,17 @@
     </match>
   </device>
 
+  <!-- non-root hald cannot read harddrive partitions; workaround for correct fsusage -->
+  <device>
+    <match key="info.category" string="volume">
+      <match key="block.is_volume" bool="true">
+        <match key="volume.fsusage" string="">
+          <merge key="volume.fsusage" type="string">filesystem</merge>
+        </match>
+      </match>
+    </match>
+  </device>
+
   <device>
     <!-- Whitelist bus types of storage devices we care about  -->
     <match key="info.category" string="storage">
diff -ruN hal-0.4.7-old/libhal-storage/libhal-storage.c hal-0.4.7/libhal-storage/libhal-storage.c
--- hal-0.4.7-old/libhal-storage/libhal-storage.c	2005-01-25 03:37:02.000000000 +0100
+++ hal-0.4.7/libhal-storage/libhal-storage.c	2005-03-17 12:54:08.864122488 +0100
@@ -592,10 +595,6 @@
 	if (mount_point == NULL)
 		mount_point = target_mount_point;
 
-	/* bail out if we don't know the filesystem */
-	if (fstype == NULL)
-		goto out;
-
 	/* blacklist fhs2.3 top level mount points */
 	if (mount_point != NULL) {
 		for (i = 0; fhs23_toplevel_mount_points[i] != NULL; i++) {
@@ -605,7 +604,7 @@
 	}
 
 	/* blacklist partitions with name 'bootstrap' of type HFS (Apple uses that) */
-	if (label != NULL && strcmp (label, "bootstrap") == 0 && strcmp (fstype, "hfs") == 0)
+	if (label != NULL && fstype != NULL && strcmp (label, "bootstrap") == 0 && strcmp (fstype, "hfs") == 0)
 		goto out;
 
 	/* only the real lucky mount points will make it this far :-) */
