diff -ruN hal-0.5.7-old/tools/device-manager/DeviceManager.py hal-0.5.7/tools/device-manager/DeviceManager.py
--- hal-0.5.7-old/tools/device-manager/DeviceManager.py	2005-11-08 16:47:36.000000000 +0100
+++ hal-0.5.7/tools/device-manager/DeviceManager.py	2006-03-27 16:40:35.000000000 +0200
@@ -360,7 +360,7 @@
 	    category.set_label("Unknown")
 
 	if device.properties.has_key("info.capabilities"):
-            capabilities.set_label("%s"%device.properties["info.capabilities"])
+            capabilities.set_label(", ".join(device.properties["info.capabilities"]))
 	else:
 	    capabilities.set_label("Unknown")
 
@@ -454,6 +454,8 @@
                     store.set(iter, 0, p, 1, "bool", 2, "false")
             elif ptype==float:
                 store.set(iter, 0, p, 1, "float", 2, "%f"%val)
+	    elif ptype==list:
+		store.set(iter, 0, p, 1, "list", 2, ", ".join(val))
 	    else:
 		# assume strlist
 		store.set(iter, 0, p, 1, "strlist", 2, val)
