diff -Nur hal-0.5.7/tools/hal-system-power-hibernate hal-0.5.7.new/build-tree/hal-0.5.7/tools/hal-system-power-hibernate
--- hal-0.5.7/tools/hal-system-power-hibernate	2006-05-08 15:04:50.789766728 +0100
+++ hal-0.5.7/tools/hal-system-power-hibernate	2006-05-08 15:05:19.234442480 +0100
@@ -53,4 +53,15 @@
 		fi
 	fi
 
+#Refresh devices as a resume can do funny things
+for type in button battery ac_adapter
+do
+	devices=`hal-find-by-capability --capability $type`
+	for device in $devices
+	do
+		dbus-send --system --print-reply --dest=org.freedesktop.Hal \
+			  $device org.freedesktop.Hal.Device.Rescan
+	done
+done
+
 exit $RET
diff -Nur hal-0.5.7/tools/hal-system-power-suspend hal-0.5.7.new/build-tree/hal-0.5.7/tools/hal-system-power-suspend
--- hal-0.5.7/tools/hal-system-power-suspend	2006-05-08 15:04:50.789766728 +0100
+++ hal-0.5.7/tools/hal-system-power-suspend	2006-05-08 15:06:21.701945976 +0100
@@ -71,4 +71,16 @@
 	echo "org.freedesktop.Hal.Device.SystemPowerManagement.NotSupported" >&2
 	exit 1
 fi
+
+#Refresh devices as a resume can do funny things
+for type in button battery ac_adapter
+do
+	devices=`hal-find-by-capability --capability $type`
+	for device in $devices
+	do
+		dbus-send --system --print-reply --dest=org.freedesktop.Hal \
+			  $device org.freedesktop.Hal.Device.Rescan
+	done
+done
+
 exit $RET
