diff -ruN hal-0.5.7-old/hald/linux2/probing/probe-volume.c hal-0.5.7/hald/linux2/probing/probe-volume.c
--- hal-0.5.7-old/hald/linux2/probing/probe-volume.c	2006-02-22 22:27:52.000000000 +0100
+++ hal-0.5.7/hald/linux2/probing/probe-volume.c	2006-05-10 18:20:39.000000000 +0200
@@ -67,6 +67,7 @@
 	char *endchar;
 	char *newstr;
 	char *s;
+	unsigned fixes = 0;
 
 	if (str == NULL)
 		return NULL;
@@ -75,6 +76,13 @@
 	s = newstr;
 	while (!g_utf8_validate (s, -1, (const char **) &endchar)) {
 		*endchar = '_';
+		++fixes;
+	}
+
+	/* If we had to fix more than 20% of the characters, give up */
+	if (fixes > 0 && g_utf8_strlen (newstr, -1) / fixes < 5) {
+	    g_free (newstr);
+	    newstr = g_strdup("");
 	}
 
 	return newstr;
