diff -Nur hal-0.5.7/build-tree/hal-0.5.7/hald/linux2/acpi.c hal-0.5.7.new/build-tree/hal-0.5.7/hald/linux2/acpi.c
--- hal-0.5.7/build-tree/hal-0.5.7/hald/linux2/acpi.c	2006-02-18 23:11:23 +0000
+++ hal-0.5.7/hald/linux2/acpi.c	2006-05-16 12:39:31 +0100
@@ -697,7 +697,7 @@
 	} else if (acpi_type == ACPI_TYPE_OMNIBOOK_DISPLAY) {
 		type = "omnibook";
 		desc = "Omnibook LCD Panel";
-		br_levels = 8;
+		br_levels = 11;
 	} else {
 		type = "unknown";
 		desc = "Unknown LCD Panel";
