diff -aur hal-0.5.8.vanilla/tools/linux/hal-system-lcd-get-brightness-linux hal-0.5.8/tools/linux/hal-system-lcd-get-brightness-linux
--- hal-0.5.8.vanilla/tools/linux/hal-system-lcd-get-brightness-linux	2006-09-11 23:56:40.000000000 +0200
+++ hal-0.5.8/tools/linux/hal-system-lcd-get-brightness-linux	2006-09-17 12:20:05.000000000 +0200
@@ -55,7 +55,7 @@
 	# spicctrl -B
 	# 70
 	# 0..255
-	value="`/usr/sbin/spicctrl -B`"
+	value="`/usr/bin/spicctrl -B`"
 	RETVAL=$?
 	if [ $RETVAL != 0 ]; then
 		echo "org.freedesktop.Hal.Device.LaptopPanel.NotSupported" >&2
diff -aur hal-0.5.8.vanilla/tools/linux/hal-system-lcd-set-brightness-linux hal-0.5.8/tools/linux/hal-system-lcd-set-brightness-linux
--- hal-0.5.8.vanilla/tools/linux/hal-system-lcd-set-brightness-linux	2006-09-11 23:56:40.000000000 +0200
+++ hal-0.5.8/tools/linux/hal-system-lcd-set-brightness-linux	2006-09-17 12:20:17.000000000 +0200
@@ -50,7 +50,7 @@
 	# spicctrl -B
 	# 70
 	# 0..255
-	value="`/usr/sbin/spicctrl -B`"
+	value="`/usr/bin/spicctrl -B`"
 	RETVAL=$?
 	if [ $RETVAL != 0 ]; then
 		echo "org.freedesktop.Hal.Device.LaptopPanel.NotSupported" >&2
