diff -Nur hal/build-tree/hal-0.5.7.1/fdi/policy/10osvendor/10-laptop-panel-mgmt-policy.fdi hal.new/build-tree/hal-0.5.7.1/fdi/policy/10osvendor/10-laptop-panel-mgmt-policy.fdi
--- hal-0.5.7.1/fdi/policy/10osvendor/10-laptop-panel-mgmt-policy.fdi	2005-09-03 17:55:11.000000000 +0200
+++ hal-0.5.7.1/fdi/policy/10osvendor/10-laptop-panel-mgmt-policy.fdi	2006-10-13 11:24:21.000000000 +0200
@@ -15,5 +15,20 @@
       <append key="org.freedesktop.Hal.Device.LaptopPanel.method_execpaths" type="strlist">hal-system-lcd-get-brightness</append>
     </match>
   </device>
+  <device>
+    <match key="info.category" string="laptop_panel">
+      <match key="/org/freedesktop/Hal/devices/computer:smbios.system.manufacturer" string="LENOVO">
+        <match key="/org/freedesktop/Hal/devices/computer:smbios.system.version" contains="ThinkPad X">
+          <merge key="laptop_panel.brightness_in_hardware" type="bool">true</merge>
+        </match>
+        <match key="/org/freedesktop/Hal/devices/computer:smbios.system.version" contains="ThinkPad Z">
+          <merge key="laptop_panel.brightness_in_hardware" type="bool">true</merge>
+        </match>
+        <match key="/org/freedesktop/Hal/devices/computer:smbios.system.version" contains="ThinkPad T">
+          <merge key="laptop_panel.brightness_in_hardware" type="bool">true</merge>
+        </match>
+      </match>
+    </match>
+  </device>
 
 </deviceinfo>
