diff -Nur hal/build-tree/hal-0.5.7.1/fdi/information/10freedesktop/10-usb-card-readers.fdi hal.new/build-tree/hal-0.5.7.1/fdi/information/10freedesktop/10-usb-card-readers.fdi
--- hal-0.5.7.1/fdi/information/10freedesktop/10-usb-card-readers.fdi	2006-01-15 17:54:05.000000000 +0100
+++ hal-0.5.7.1/fdi/information/10freedesktop/10-usb-card-readers.fdi	2006-10-09 10:28:57.000000000 +0200
@@ -66,6 +66,9 @@
       <match key="storage.model" contains="Storage-MMC">
         <merge key="storage.drive_type" type="string">sd_mmc</merge>
       </match>
+      <match key="storage.model" contains="Storage-SDC">
+        <merge key="storage.drive_type" type="string">sd_mmc</merge>
+      </match>
 
       <!-- Lexar CF Reader -->
       <match key="@storage.physical_device:usb.vendor_id" int="0x05dc">
