diff -Nur hal/build-tree/hal-0.5.7.1/tools/hal-system-power-hibernate hal.new/build-tree/hal-0.5.7.1/tools/hal-system-power-hibernate
--- hal-0.5.7.1/tools/hal-system-power-hibernate	2006-10-09 10:31:16.000000000 +0200
+++ hal-0.5.7.1/tools/hal-system-power-hibernate	2006-10-09 10:32:56.000000000 +0200
@@ -30,12 +30,12 @@
 
 #Other distros just need to have *any* tools installed
 else
-	if [ -x "/usr/bin/powersave" ] ; then
-	        $POWERSAVED_SUSPEND2DISK
-		RET=$?
-	elif [ -x "/usr/sbin/pmi" ] ; then
+	if [ -x "/usr/sbin/pmi" ] ; then
 		/usr/sbin/pmi action hibernate force
 		RET=$?
+	elif [ -x "/usr/bin/powersave" ] ; then
+	        $POWERSAVED_SUSPEND2DISK
+		RET=$?
 	elif [ -x "/usr/sbin/pm-hibernate" ] ; then
 		/usr/sbin/pm-hibernate
 		RET=$?
diff -Nur hal/build-tree/hal-0.5.7.1/tools/hal-system-power-suspend hal.new/build-tree/hal-0.5.7.1/tools/hal-system-power-suspend
--- hal-0.5.7.1/tools/hal-system-power-suspend	2006-10-09 10:31:16.000000000 +0200
+++ hal-0.5.7.1/tools/hal-system-power-suspend	2006-10-09 10:32:28.000000000 +0200
@@ -47,16 +47,16 @@
 
 #Other distros just need to have *any* tools installed
 else
-	if [ -x "/usr/bin/powersave" ] ; then
-	    $POWERSAVED_SUSPEND2RAM
-	    RET=$?
-	elif [ -x "/usr/sbin/hibernate" ] ; then
+	if [ -x "/usr/sbin/hibernate" ] ; then
 		# Use hibernate configured for suspend-to-ram
 		/usr/sbin/hibernate -F/etc/hibernate/ram.conf
 		RET=$?
 	elif [ -x "/usr/sbin/pmi" ] ; then
 	    /usr/sbin/pmi action suspend force
 	    RET=$?
+	elif [ -x "/usr/bin/powersave" ] ; then
+	    $POWERSAVED_SUSPEND2RAM
+	    RET=$?
 	elif  [ "$HAL_PROP_POWER_MANAGEMENT_TYPE" = "pmu" ]; then
 		#PMU systems cannot use /sys/power/state yet, so use a helper to issue an ioctl
 		hal-system-power-pmu sleep
