diff -Nur hal-0.5.8.1/build-tree/hal-0.5.8.1/hald/linux/addons/addon-storage.c hal-0.5.8.1.new/build-tree/hal-0.5.8.1/hald/linux/addons/addon-storage.c
--- hal-0.5.8.1/hald/linux/addons/addon-storage.c	2006-09-19 22:23:25.000000000 +0200
+++ hal-0.5.8.1/hald/linux/addons/addon-storage.c	2006-10-26 20:24:38.000000000 +0200
@@ -281,6 +281,8 @@
 	 */
         /*drop_privileges (1);*/
 
+        drop_privileges (1);
+
 	if ((udi = getenv ("UDI")) == NULL)
 		goto out;
 	if ((device_file = getenv ("HAL_PROP_BLOCK_DEVICE")) == NULL)
