diff -Nur hal-0.5.8.1/build-tree/hal-0.5.8.1/hald/linux/addons/addon-keyboard.c hal-0.5.8.1.new/build-tree/hal-0.5.8.1/hald/linux/addons/addon-keyboard.c
--- hal-0.5.8.1/hald/linux/addons/addon-keyboard.c	2006-09-19 22:23:25.000000000 +0200
+++ hal-0.5.8.1/hald/linux/addons/addon-keyboard.c	2006-10-26 20:27:23.000000000 +0200
@@ -6,6 +6,7 @@
  * Copyright (C) 2005 David Zeuthen, <david@fubar.dk>
  * Copyright (C) 2005 Ryan Lortie <desrt@desrt.ca>
  * Copyright (C) 2006 Matthew Garrett <mjg59@srcf.ucam.org>
+ * Copyright (C) 2006 Paul Sladen <ubuntu@paul.sladen.org>
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -41,8 +42,19 @@
 
 static char *udi;
 
+/* Waiting to will be assigned upstream in 'input.h' */
+#ifndef KEY_BATTERY
+#define KEY_BATTERY 236
+#endif
+
 static char *key_name[KEY_MAX + 1] = {
 	[0 ... KEY_MAX] = NULL,
+	[KEY_BATTERY] = "battery",
+	[KEY_MACRO] = "macro",
+	[KEY_MUTE] = "mute",
+	[KEY_VOLUMEDOWN] = "volumedown",
+	[KEY_VOLUMEUP] = "volumeup",
+	[KEY_POWER] = "power",
 	[KEY_STOP] = "stop",
 	[KEY_AGAIN] = "again",
 	[KEY_PROPS] = "props",
@@ -102,11 +114,11 @@
 	[KEY_F16] = "f16",
 	[KEY_F17] = "f17",
 	[KEY_F18] = "f18",
-	[KEY_F19] = "f19",
-	[KEY_F20] = "f20",
-	[KEY_F21] = "f21",
-	[KEY_F22] = "f22",
-	[KEY_F23] = "f23",
+	[KEY_F19] = "light",  // Unassigned, aliased key
+	[KEY_F20] = "videoout",  // Unassigned, aliased key
+	[KEY_F21] = "rotatescreen",  // Unassigned, aliased key
+	[KEY_F22] = "videomodecycle",  // Unassigned, aliased key
+	[KEY_F23] = "presentation",  // Unassigned, aliased key
 	[KEY_F24] = "f24",
 	[KEY_PLAYCD] = "play-cd",
 	[KEY_PAUSECD] = "pause-cd",
