diff -Nur hal-0.5.8.1/build-tree/hal-0.5.8.1/hald/linux/acpi.c hal-0.5.8.1.new/build-tree/hal-0.5.8.1/hald/linux/acpi.c
--- hal-0.5.8.1/hald/linux/acpi.c	2006-09-19 22:23:25.000000000 +0200
+++ hal-0.5.8.1/hald/linux/acpi.c	2006-10-26 22:38:15.000000000 +0200
@@ -777,7 +777,7 @@
 	} else if (acpi_type == ACPI_TYPE_OMNIBOOK_DISPLAY) {
 		type = "omnibook";
 		desc = "Omnibook LCD Panel";
-		br_levels = 8;
+		br_levels = 11;
 	} else if (acpi_type == ACPI_TYPE_SONYPI_DISPLAY) {
 		type = "sonypi";
 		desc = "Sony LCD Panel";
