diff -Nur hal/build-tree/hal-0.5.7.1/fdi/information/10freedesktop/10-usb-music-players.fdi hal.new/build-tree/hal-0.5.7.1/fdi/information/10freedesktop/10-usb-music-players.fdi
--- hal-0.5.7.1/fdi/information/10freedesktop/10-usb-music-players.fdi	2006-10-09 10:35:04.000000000 +0200
+++ hal-0.5.7.1/fdi/information/10freedesktop/10-usb-music-players.fdi	2006-10-09 10:36:50.000000000 +0200
@@ -853,6 +853,66 @@
 
 	</match>
       </match>
+
+      <!-- Philips --> 
+      <match key="@storage.physical_device:usb.vendor_id" int="0x471">
+        <!-- Philips GoGear SA1330 -->
+        <match key="@storage.physical_device:usb.product_id" int="0x165">
+           <append key="info.capabilities" type="strlist">portable_audio_player</append>
+           <merge key="info.category" type="string">portable_audio_player</merge>
+           <merge key="portable_audio_player.type" type="string">generic</merge>
+           <merge key="portable_audio_player.access_method" type="string">storage</merge>
+           <merge key="portable_audio_player.storage_device" type="copy_property">info.udi</merge>
+           <append key="portable_audio_player.output_formats" type="strlist">audio/mpeg</append>
+           <append key="portable_audio_player.output_formats" type="strlist">audio/x-ms-wma</append>
+           <append key="portable_audio_player.output_formats" type="strlist">audio/x-wav</append>
+           <append key="portable_audio_player.input_formats" type="strlist">audio/x-wav</append>
+	</match>
+      </match>
+
+      <!-- Motorola iTunes enabled phone -->
+      <match key="@storage.physical_device:usb.vendor_id" int="0x22b8">
+        <match key="@storage.physical_device:usb.product_id" int="0x4810">
+          <append key="info.capabilities" type="strlist">portable_audio_player</append>
+          <merge key="info.category" type="string">portable_audio_player</merge>
+          <merge key="portable_audio_player.access_method" type="string">storage</merge>
+          <merge key="portable_audio_player.type" type="string">ipod</merge>
+          <merge key="portable_audio_player.storage_device" type="copy_property">info.udi</merge>
+          <append key="portable_audio_player.output_formats" type="strlist">audio/mpeg</append>
+          <append key="portable_audio_player.output_formats" type="strlist">audio/aac</append>
+          <merge key="storage.requires_eject" type="bool">true</merge>
+        </match>
+      </match>
+
+      <!-- Sandisk Corp. Sansa m250 (maybe others as well) -->
+      <match key="@storage.physical_device:usb.vendor_id" int="0x0781">
+        <match key="@storage.physical_device:usb.product_id" int="0x7401">
+          <append key="info.capabilities" type="strlist">portable_audio_player</append>
+          <merge key="info.category" type="string">portable_audio_player</merge>
+          <merge key="portable_audio_player.type" type="string">generic</merge>
+          <merge key="portable_audio_player.access_method" type="string">storage</merge>
+          <merge key="portable_audio_player.storage_device" type="copy_property">info.udi</merge>
+          <append key="portable_audio_player.output_formats" type="strlist">audio/mpeg</append>
+          <append key="portable_audio_player.output_formats" type="strlist">audio/x-ms-wma</append>
+          <append key="portable_audio_player.input_formats" type="strlist">audio/mpeg</append>
+          <append key="portable_audio_player.input_formats" type="strlist">audio/x-ms-wma</append>
+        </match>
+      </match>
+
+        <!-- TrekStor i.Beat jump MP3 Player -->
+      <match key="@storage.physical_device:usb.vendor_id" int="0x066f">
+        <match key="@storage.physical_device:usb.product_id" int="0x835d">
+          <append key="info.capabilities" type="strlist">portable_audio_player</append>
+          <merge key="info.category" type="string">portable_audio_player</merge>
+          <merge key="portable_audio_player.type" type="string">generic</merge>
+          <merge key="portable_audio_player.access_method" type="string">storage</merge>
+          <merge key="portable_audio_player.storage_device" type="copy_property">info.udi</merge>
+          <append key="portable_audio_player.output_formats" type="strlist">audio/mpeg</append>
+          <append key="portable_audio_player.output_formats" type="strlist">audio/x-ms-wma</append>
+          <append key="portable_audio_player.input_formats" type="strlist">audio/x-wav</append>
+        </match>
+      </match>
+
     </match>      
 
     <!-- USB devices that are music players but uses a user-space library -->
@@ -1019,6 +1079,20 @@
         </match>
       </match>
 
+  <!-- Sandisk Corp. Sansa e270 (maybe others as well) -->
+      <match key="@storage.physical_device:usb.vendor_id" int="0x0781">
+        <match key="@storage.physical_device:usb.product_id" int="0x7421">
+           <append key="info.capabilities" type="strlist">portable_audio_player</append>
+           <merge key="info.category" type="string">portable_audio_player</merge>
+           <merge key="portable_audio_player.type" type="string">generic</merge>
+           <merge key="portable_audio_player.access_method" type="string">storage</merge>
+           <merge key="portable_audio_player.storage_device" type="copy_property">info.udi</merge>
+           <append key="portable_audio_player.output_formats" type="strlist">audio/mpeg</append>
+           <append key="portable_audio_player.output_formats" type="strlist">audio/x-ms-wma</append>
+           <append key="portable_audio_player.input_formats" type="strlist">audio/mpeg</append>
+           <append key="portable_audio_player.input_formats" type="strlist">audio/x-ms-wma</append>
+         </match>
+      </match>
     </match>
 
   </device>
