diff -Nur hal-0.5.8.1/build-tree/hal-0.5.8.1/tools/linux/hal-system-power-hibernate-linux hal-0.5.8.1.new/build-tree/hal-0.5.8.1/tools/linux/hal-system-power-hibernate-linux
--- hal-0.5.8.1/tools/linux/hal-system-power-hibernate-linux	2006-10-26 23:21:43.000000000 +0200
+++ hal-0.5.8.1/tools/linux/hal-system-power-hibernate-linux	2006-10-26 23:27:24.000000000 +0200
@@ -39,12 +39,12 @@
 
 #Other distros just need to have *any* tools installed
 else
-	if [ -x "/usr/bin/powersave" ] ; then
-	        $POWERSAVED_SUSPEND2DISK
-		RET=$?
-	elif [ -x "/usr/sbin/pmi" ] ; then
+	if [ -x "/usr/sbin/pmi" ] ; then
 		/usr/sbin/pmi action hibernate force
 		RET=$?
+	elif [ -x "/usr/bin/powersave" ] ; then
+	        $POWERSAVED_SUSPEND2DISK
+		RET=$?
 	elif [ -x "/usr/sbin/pm-hibernate" ] ; then
 		/usr/sbin/pm-hibernate
 		RET=$?
diff -Nur hal-0.5.8.1/build-tree/hal-0.5.8.1/tools/linux/hal-system-power-suspend-linux hal-0.5.8.1.new/build-tree/hal-0.5.8.1/tools/linux/hal-system-power-suspend-linux
--- hal-0.5.8.1/tools/linux/hal-system-power-suspend-linux	2006-10-26 23:21:45.000000000 +0200
+++ hal-0.5.8.1/tools/linux/hal-system-power-suspend-linux	2006-10-26 23:26:27.000000000 +0200
@@ -80,12 +80,12 @@
 
 #Other distros just need to have *any* tools installed
 else
-	if [ -x "/usr/bin/powersave" ] ; then
-	    $POWERSAVED_SUSPEND2RAM
-	    RET=$?
-	elif [ -x "/usr/sbin/pmi" ] ; then
+	if [ -x "/usr/sbin/pmi" ] ; then
 	    /usr/sbin/pmi action suspend force
 	    RET=$?
+	elif [ -x "/usr/bin/powersave" ] ; then
+	    $POWERSAVED_SUSPEND2RAM
+	    RET=$?
 	elif [ -x "/usr/sbin/s2ram" ] ; then
 	    /usr/sbin/s2ram
 	   RET=$?
