diff -ruN hal-0.5.7-old/po/POTFILES.in hal-0.5.7/po/POTFILES.in
--- hal-0.5.7-old/po/POTFILES.in	2006-01-26 15:32:37.000000000 +0100
+++ hal-0.5.7/po/POTFILES.in	2006-03-27 16:34:50.000000000 +0200
@@ -1,4 +1,5 @@
 # List of source files containing translatable strings.
 # Please keep this file sorted alphabetically.
+../../debian/hal-device-manager.desktop.in
 libhal-storage/libhal-storage.c
 tools/device-manager/hal-device-manager.glade
