Index: hald/device_info.c
===================================================================
RCS file: /cvs/hal/hal/hald/device_info.c,v
retrieving revision 1.16
diff -u -r1.16 device_info.c
--- hal/hald/device_info.c	14 Oct 2004 18:37:28 -0000	1.16
+++ hal/hald/device_info.c	28 Oct 2004 18:57:31 -0000
@@ -1176,7 +1176,10 @@
 di_search_and_merge (HalDevice *d)
 {
 
-	return scan_fdi_files (PACKAGE_DATA_DIR "/hal/fdi", d);
+	dbus_bool_t ret;
+
+	ret = scan_fdi_files (PACKAGE_DATA_DIR "/hal/fdi", d);
+	return scan_fdi_files (PACKAGE_SYSCONF_DIR "/hal/fdi", d) || ret;
 }
 
 /** @} */
