diff -ruN hal-0.4.3-old/configure hal-0.4.3/configure
--- hal-0.4.3-old/configure	2005-01-03 23:12:22.000000000 +0100
+++ hal-0.4.3/configure	2005-01-04 18:01:26.000000000 +0100
@@ -21525,48 +21525,48 @@
 fi;
 
 # Check whether --with-hwdata or --without-hwdata was given.
-if test "${with_hwdata+set}" = set; then
-  withval="$with_hwdata"
-
-fi;
-if ! test -z "$with_hwdata" ; then
-   HWDATA_DIR="$with_hwdata"
-else
-   for dir in /usr/share/hwdata /usr/share/misc /usr/share; do
-      as_ac_File=`echo "ac_cv_file_$dir/pci.ids" | $as_tr_sh`
-echo "$as_me:$LINENO: checking for $dir/pci.ids" >&5
-echo $ECHO_N "checking for $dir/pci.ids... $ECHO_C" >&6
-if eval "test \"\${$as_ac_File+set}\" = set"; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
-if test -r "$dir/pci.ids"; then
-  eval "$as_ac_File=yes"
-else
-  eval "$as_ac_File=no"
-fi
-fi
-echo "$as_me:$LINENO: result: `eval echo '${'$as_ac_File'}'`" >&5
-echo "${ECHO_T}`eval echo '${'$as_ac_File'}'`" >&6
-if test `eval echo '${'$as_ac_File'}'` = yes; then
-  HWDATA_DIR=$dir
-fi
-
-   done
-fi
-if test -z "$HWDATA_DIR"; then
-   { { echo "$as_me:$LINENO: error: cannot find pci.ids. Use --with-hwdata to specify location" >&5
-echo "$as_me: error: cannot find pci.ids. Use --with-hwdata to specify location" >&2;}
-   { (exit 1); exit 1; }; }
-fi
+#if test "${with_hwdata+set}" = set; then
+#  withval="$with_hwdata"
+#
+#fi;
+#if ! test -z "$with_hwdata" ; then
+#   HWDATA_DIR="$with_hwdata"
+#else
+#   for dir in /usr/share/hwdata /usr/share/misc /usr/share; do
+#      as_ac_File=`echo "ac_cv_file_$dir/pci.ids" | $as_tr_sh`
+#echo "$as_me:$LINENO: checking for $dir/pci.ids" >&5
+#echo $ECHO_N "checking for $dir/pci.ids... $ECHO_C" >&6
+#if eval "test \"\${$as_ac_File+set}\" = set"; then
+#  echo $ECHO_N "(cached) $ECHO_C" >&6
+#else
+#  test "$cross_compiling" = yes &&
+#  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+#echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+#   { (exit 1); exit 1; }; }
+#if test -r "$dir/pci.ids"; then
+#  eval "$as_ac_File=yes"
+#else
+#  eval "$as_ac_File=no"
+#fi
+#fi
+#echo "$as_me:$LINENO: result: `eval echo '${'$as_ac_File'}'`" >&5
+#echo "${ECHO_T}`eval echo '${'$as_ac_File'}'`" >&6
+#if test `eval echo '${'$as_ac_File'}'` = yes; then
+#  HWDATA_DIR=$dir
+#fi
+#
+#   done
+#fi
+#if test -z "$HWDATA_DIR"; then
+#   { { echo "$as_me:$LINENO: error: cannot find pci.ids. Use --with-hwdata to specify location" >&5
+#echo "$as_me: error: cannot find pci.ids. Use --with-hwdata to specify location" >&2;}
+#   { (exit 1); exit 1; }; }
+#fi
 
 
-cat >>confdefs.h <<_ACEOF
-#define HWDATA_DIR "$HWDATA_DIR"
-_ACEOF
+#cat >>confdefs.h <<_ACEOF
+##define HWDATA_DIR "$HWDATA_DIR"
+#_ACEOF
 
 
 
diff -ruN hal-0.4.3-old/hald/linux/pci_bus_device.c hal-0.4.3/hald/linux/pci_bus_device.c
--- hal-0.4.3-old/hald/linux/pci_bus_device.c	2004-10-14 21:21:41.000000000 +0200
+++ hal-0.4.3/hald/linux/pci_bus_device.c	2005-01-04 17:50:21.000000000 +0100
@@ -283,7 +283,6 @@
 
 	fp = fopen (path, "r");
 	if (fp == NULL) {
-		HAL_ERROR (("couldn't open PCI database at %s,", path));
 		return FALSE;
 	}
 
@@ -369,11 +368,24 @@
 static void
 pci_device_init (BusDeviceHandler *self)
 {
+	dbus_bool_t ret = FALSE;
+	int i;
+	char *locations[] = { "/usr/share/hwdata/pci.ids",
+                        "/usr/share/misc/pci.ids",
+                        "/usr/share/pci.ids",
+                        "/var/lib/misc/pci.ids",
+                        NULL
+	};
 	/* get all drivers under /sys/bus/pci/drivers */
 	drivers_collect ("pci");
 
 	/* Load /usr/share/hwdata/pci.ids */
-	pci_ids_load (HWDATA_DIR "/pci.ids");
+	for (i=0 ; locations[i] != NULL && !ret ; i++) {
+		ret = pci_ids_load (locations[i]);
+	}
+	if (!ret) {
+		HAL_ERROR (("couldn't open PCI database"));
+	}
 }
 
 /** Shutdown function for PCI handling
diff -ruN hal-0.4.3-old/hald/linux/usb_bus_device.c hal-0.4.3/hald/linux/usb_bus_device.c
--- hal-0.4.3-old/hald/linux/usb_bus_device.c	2004-10-14 20:51:12.000000000 +0200
+++ hal-0.4.3/hald/linux/usb_bus_device.c	2005-01-04 17:50:21.000000000 +0100
@@ -549,15 +549,26 @@
 static void
 usb_device_init ()
 {
+	dbus_bool_t ret = FALSE;
+	int i;
+	char *locations[] = { "/usr/share/hwdata/usb.ids",
+                        "/usr/share/misc/usb.ids",
+                        "/usr/share/usb.ids",
+                        "/var/lib/misc/usb.ids",
+                        NULL
+	};
 
 	/* get all drivers under /sys/bus/usb/drivers */
 	drivers_collect ("usb");
 
 	/* Load /usr/share/hwdata/usb.ids */
-	usb_ids_load (HWDATA_DIR "/usb.ids");
-
+	for (i=0; locations[i] != NULL && !ret ; i++) {
+		ret = usb_ids_load (locations[i]);
+	}
+	if (!ret) {
+		HAL_ERROR (("couldn't open USB database"));
+	}
 	/* Parse /proc/bus/usb/devices */
-
 	usb_proc_parse ();
 }
 
