diff -ruN hal-0.5.4-old/tools/device-manager/DeviceManager.py hal-0.5.4/tools/device-manager/DeviceManager.py
--- hal-0.5.4-old/tools/device-manager/DeviceManager.py	2005-07-15 22:28:57.000000000 +0200
+++ hal-0.5.4/tools/device-manager/DeviceManager.py	2005-10-18 14:03:05.000000000 +0200
@@ -1,5 +1,6 @@
 """This file contains the DeviceManager class."""
  
+import LaunchpadIntegration
 import sys
 import gobject
 import gtk
@@ -40,6 +41,11 @@
         """Init the GUI and connect to the HAL daemon."""
         LibGladeApplication.__init__(self, Const.DATADIR + "/hal-device-manager.glade")
 
+        LaunchpadIntegration.set_sourcepackagename("hal")
+	widget = self.xml.get_widget("help1_menu")
+	print widget
+        LaunchpadIntegration.add_items (widget, -1, False, True);
+
         ver = getattr(dbus, 'version', (0, 0, 0))
         if ver < (0, 40, 0):
             dialog = gtk.MessageDialog(None, gtk.DIALOG_MODAL, 
