diff -Nur hal/build-tree/hal-0.5.7.1/tools/hal-system-storage-eject hal.new/build-tree/hal-0.5.7.1/tools/hal-system-storage-eject
--- hal-0.5.7.1/tools/hal-system-storage-eject	2006-02-10 04:41:54.000000000 +0100
+++ hal-0.5.7.1/tools/hal-system-storage-eject	2006-08-04 09:57:41.000000000 +0200
@@ -31,7 +31,18 @@
 read GIVEN_EJECTOPTIONS
 GIVEN_EJECTOPTIONS=${GIVEN_EJECTOPTIONS//[^a-zA-Z0-9_=[:space:]]/_}
 
-RESULT=$(eject "$HAL_PROP_BLOCK_DEVICE" 2>&1)
+# check validity of user id
+if [ "$HAL_METHOD_INVOKED_BY_UID" = "0" ]; then
+    echo "Script was called with HAL_METHOD_INVOKED_BY_UID=0, aborting." >&2
+    exit 1
+fi
+PWD=`getent passwd "$HAL_METHOD_INVOKED_BY_UID"` || {
+    echo "User id $HAL_METHOD_INVOKED_BY_UID does not exist." >&2
+    exit 1
+}
+USER="`echo "$PWD" | cut -f 1 -d: | head -n 1`"
+
+RESULT=$(su -c "eject '$HAL_PROP_BLOCK_DEVICE'" "$USER" 2>&1)
 if [ $? -ne 0 ]; then
     case "$RESULT" in
 	*busy*)
diff -Nur hal/build-tree/hal-0.5.7.1/tools/hal-system-storage-mount hal.new/build-tree/hal-0.5.7.1/tools/hal-system-storage-mount
--- hal-0.5.7.1/tools/hal-system-storage-mount	2006-02-14 19:49:55.000000000 +0100
+++ hal-0.5.7.1/tools/hal-system-storage-mount	2006-08-04 09:57:32.000000000 +0200
@@ -15,6 +15,20 @@
     exit 1
 fi
 
+# check validity of user id
+if [ "$HAL_METHOD_INVOKED_BY_UID" = "0" ]; then
+    echo "Script was called with HAL_METHOD_INVOKED_BY_UID=0, aborting." >&2
+    exit 1
+fi
+PWD=`getent passwd "$HAL_METHOD_INVOKED_BY_UID"` || {
+    echo "User id $HAL_METHOD_INVOKED_BY_UID does not exist." >&2
+    exit 1
+}
+USER="`echo "$PWD" | cut -f 1 -d:`"
+
+su -c "pmount-hal \"$HAL_PROP_BLOCK_DEVICE\"" "$USER"
+exit $?
+
 # check if device is already mounted
 if [ "$HAL_PROP_VOLUME_IS_MOUNTED" = "true" ]; then
     echo "org.freedesktop.Hal.Device.Volume.AlreadyMounted" >&2
diff -Nur hal/build-tree/hal-0.5.7.1/tools/hal-system-storage-unmount hal.new/build-tree/hal-0.5.7.1/tools/hal-system-storage-unmount
--- hal-0.5.7.1/tools/hal-system-storage-unmount	2006-02-13 08:29:44.000000000 +0100
+++ hal-0.5.7.1/tools/hal-system-storage-unmount	2006-08-04 09:57:39.000000000 +0200
@@ -13,6 +13,20 @@
     exit 1
 fi
 
+# check validity of user id
+if [ "$HAL_METHOD_INVOKED_BY_UID" = "0" ]; then
+    echo "Script was called with HAL_METHOD_INVOKED_BY_UID=0, aborting." >&2
+    exit 1
+fi
+PWD=`getent passwd "$HAL_METHOD_INVOKED_BY_UID"` || {
+    echo "User id $HAL_METHOD_INVOKED_BY_UID does not exist." >&2
+    exit 1
+}
+USER="`echo "$PWD" | cut -f 1 -d:`"
+
+su -c "pumount \"$HAL_PROP_BLOCK_DEVICE\"" "$USER"
+exit $?
+
 MOUNT_POINT="$HAL_PROP_INFO_HAL_MOUNT_CREATED_MOUNT_POINT"
 if [ "$MOUNT_POINT" == "" ]; then
     MOUNT_POINT="$HAL_PROP_VOLUME_MOUNT_POINT"
