diff -ruN hal-0.4.7-old/hald/linux/class_device.c hal-0.4.7/hald/linux/class_device.c
--- hal-0.4.7-old/hald/linux/class_device.c	2004-12-01 04:39:01.000000000 +0100
+++ hal-0.4.7/hald/linux/class_device.c	2005-02-15 10:10:17.060885680 +0100
@@ -447,9 +447,12 @@
 	sysfs_path = hal_device_property_get_string (d, ".udev.sysfs_path");
 	assert (sysfs_path != NULL);
 	class_device = sysfs_open_class_device_path (sysfs_path);
-	if (class_device == NULL)
-		DIE (("Coulnd't get sysfs class device object for path %s", 
+	if (class_device == NULL) {
+		HAL_WARNING (("Coulnd't get sysfs class device object for path %s", 
 		      sysfs_path));
+		hal_device_store_remove (hald_get_tdl (), d);
+		return;
+	}
 	self->pre_process (self, d, sysfs_path, class_device);
 	sysfs_close_class_device (class_device);
 
