--- xorg-cf-files-1.0.2/lnxLib.rules.jau	2004-04-23 14:41:58.000000000 -0400
+++ xorg-cf-files-1.0.2/lnxLib.rules	2007-03-06 11:22:28.000000000 -0500
@@ -417,13 +417,13 @@
 JUMP_HOST_ICE = $(JUMP_ROOT_X11)
 JUMP_DIR_ICE = $(JUMP_DIR_X11)
 
-XCOMM libXt.so (Xt, Xmu, Xext, Xi, Xtst, Xp)
+XCOMM libXt.so (Xt, Xmu, Xext, Xi, Xtst)
 JUMP_ROOT_Xt = $(TOOLKITSRC)
 JUMP_IS_HOST_Xt = YES
-JUMP_STUBLIBS_Xt = libXt.sa libXmu.sa libXext.sa libXi.sa libXtst.sa libXp.sa
+JUMP_STUBLIBS_Xt = libXt.sa libXmu.sa libXext.sa libXi.sa libXtst.sa
 JUMP_SUBDIRS_Xt = $(JUMP_ROOT_X11)
 JUMP_SIBDIRS_Xt = $(JUMP_ROOT_Xmu) $(JUMP_ROOT_Xext) \
-	$(JUMP_ROOT_Xi) $(JUMP_ROOT_Xtst) $(JUMP_ROOT_Xp)
+	$(JUMP_ROOT_Xi) $(JUMP_ROOT_Xtst)
 JUMP_DIR_Xt = $(JUMP_ROOT_Xt)/shared
 JUMP_VARS_Xt = $(JUMP_ROOT_Xt)/jump_vars
 JUMP_FUNCS_Xt = $(JUMP_ROOT_Xt)/jump_funcs
@@ -433,9 +433,9 @@
 JUMP_ADDRESS_Xt = 0x602a0000
 JUMP_JUMPTABLESIZE_Xt = 0x4000
 JUMP_GOTSIZE_Xt = 4096
-JUMP_STUBNAMES_Xt = libXt libXmu libXext libXi libXtst libXp
+JUMP_STUBNAMES_Xt = libXt libXmu libXext libXi libXtst
 JUMP_STUBS_IMPORT_Xt = $(JUMP_STUBS_IMPORT_X11)
-JUMP_SIBARS_Xt = jump/libXmu.a~ jump/libXext.a~ jump/libXi.a~ jump/libXtst.a~ jump/libXp.a~
+JUMP_SIBARS_Xt = jump/libXmu.a~ jump/libXext.a~ jump/libXi.a~ jump/libXtst.a~
 JUMP_LDLIBS_Xt = $(JUMP_SIBARS_Xt) $(JUMP_ROOT_ICE)/libICE.sa \
 	$(JUMP_ROOT_SM)/libSM.sa $(JUMP_ROOT_X11)/libX11.sa $(JUMP_LDLIBS_libc)
 
@@ -463,12 +463,6 @@
 JUMP_HOST_Xtst = $(JUMP_ROOT_Xt)
 JUMP_DIR_Xtst = $(JUMP_DIR_Xt)
 
-XCOMM libXp (part of libXt.so)
-JUMP_ROOT_Xp = $(XPLIBSRC)
-JUMP_IS_HOST_Xp = NO
-JUMP_HOST_Xp = $(JUMP_ROOT_Xt)
-JUMP_DIR_Xp = $(JUMP_DIR_Xt)
-
 XCOMM libXaw.so (Xaw)
 JUMP_ROOT_Xaw = $(AWIDGETSRC)
 JUMP_IS_HOST_Xaw = YES
