--- xorg-cf-files-1.0.2/X11.tmpl.jau	2006-05-19 14:39:39.000000000 -0400
+++ xorg-cf-files-1.0.2/X11.tmpl	2007-03-06 11:58:49.000000000 -0500
@@ -3929,7 +3929,7 @@
 
 #ifndef ExtraXawReqs
 # if BuildXaw
-#  define ExtraXawReqs $(XPLIB)
+#  define ExtraXawReqs /* $(XPLIB) */
 # else
 #  define ExtraXawReqs /**/
 # endif
@@ -3939,7 +3939,7 @@
 
 #ifndef ExtraXawClientDepLibs
 # if BuildXaw
-#  define ExtraXawClientDepLibs $(DEPXPLIB)
+#  define ExtraXawClientDepLibs /* $(DEPXPLIB) */
 # else
 #  define ExtraXawClientDepLibs /**/
 # endif
@@ -3953,7 +3953,7 @@
 
 #ifndef ExtraXawClientLibs
 # if BuildXaw
-#  define ExtraXawClientLibs $(XPLIB)
+#  define ExtraXawClientLibs /* $(XPLIB) */
 # else
 #  define ExtraXawClientLibs /**/
 # endif
