diff -up glib-2.22.1/gio/Makefile.am.fix-runtime-libdir glib-2.22.1/gio/Makefile.am
--- glib-2.22.1/gio/Makefile.am.fix-runtime-libdir	2009-09-22 13:43:48.000000000 -0400
+++ glib-2.22.1/gio/Makefile.am	2009-09-29 22:52:56.044577337 -0400
@@ -430,7 +430,7 @@ if HAVE_GLIB_RUNTIME_LIBDIR
 install-data-hook:
 	mkdir -p $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	mv $(DESTDIR)$(libdir)/libgio-2.0.so.0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
-	mv $(DESTDIR)$(libdir)/libgio-2.0.so.0.$(LT_CURRENT).0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
+	mv $(DESTDIR)$(libdir)/libgio-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	rm -f $(DESTDIR)$(libdir)/libgio-2.0.so
-	ln -s $(GLIB_RUNTIME_LIBDIR)/libgio-2.0.so.0.$(LT_CURRENT).0 $(DESTDIR)$(libdir)/libgio-2.0.so
+	ln -s $(GLIB_RUNTIME_LIBDIR)/libgio-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/libgio-2.0.so
 endif
diff -up glib-2.22.1/glib/Makefile.am.fix-runtime-libdir glib-2.22.1/glib/Makefile.am
--- glib-2.22.1/glib/Makefile.am.fix-runtime-libdir	2009-09-29 18:06:56.000000000 -0400
+++ glib-2.22.1/glib/Makefile.am	2009-09-29 22:52:56.040593799 -0400
@@ -385,7 +385,7 @@ install-data-hook: libglib-gdb.py
 if HAVE_GLIB_RUNTIME_LIBDIR
 	mkdir -p $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	mv $(DESTDIR)$(libdir)/libglib-2.0.so.0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
-	mv $(DESTDIR)$(libdir)/libglib-2.0.so.0.$(LT_CURRENT).0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
+	mv $(DESTDIR)$(libdir)/libglib-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	rm -f $(DESTDIR)$(libdir)/libglib-2.0.so
-	ln -s $(GLIB_RUNTIME_LIBDIR)/libglib-2.0.so.0.$(LT_CURRENT).0 $(DESTDIR)$(libdir)/libglib-2.0.so
+	ln -s $(GLIB_RUNTIME_LIBDIR)/libglib-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/libglib-2.0.so
 endif
diff -up glib-2.22.1/gmodule/Makefile.am.fix-runtime-libdir glib-2.22.1/gmodule/Makefile.am
--- glib-2.22.1/gmodule/Makefile.am.fix-runtime-libdir	2009-09-29 22:56:34.846827363 -0400
+++ glib-2.22.1/gmodule/Makefile.am	2009-09-29 22:57:06.690827788 -0400
@@ -121,7 +121,7 @@ if HAVE_GLIB_RUNTIME_LIBDIR
 install-data-hook:
 	mkdir -p $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	mv $(DESTDIR)$(libdir)/libgmodule-2.0.so.0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
-	mv $(DESTDIR)$(libdir)/libgmodule-2.0.so.0.$(LT_CURRENT).0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
+	mv $(DESTDIR)$(libdir)/libgmodule-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	rm -f $(DESTDIR)$(libdir)/libgmodule-2.0.so
-	ln -s $(GLIB_RUNTIME_LIBDIR)/libgmodule-2.0.so.0.$(LT_CURRENT).0 $(DESTDIR)$(libdir)/libgmodule-2.0.so
+	ln -s $(GLIB_RUNTIME_LIBDIR)/libgmodule-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/libgmodule-2.0.so
 endif
diff -up glib-2.22.1/gobject/Makefile.am.fix-runtime-libdir glib-2.22.1/gobject/Makefile.am
--- glib-2.22.1/gobject/Makefile.am.fix-runtime-libdir	2009-09-29 18:06:56.000000000 -0400
+++ glib-2.22.1/gobject/Makefile.am	2009-09-29 22:52:56.041593367 -0400
@@ -279,7 +279,7 @@ install-data-hook: libgobject-gdb.py
 if HAVE_GLIB_RUNTIME_LIBDIR
 	mkdir -p $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	mv $(DESTDIR)$(libdir)/libgobject-2.0.so.0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
-	mv $(DESTDIR)$(libdir)/libgobject-2.0.so.0.$(LT_CURRENT).0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
+	mv $(DESTDIR)$(libdir)/libgobject-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	rm -f $(DESTDIR)$(libdir)/libgobject-2.0.so
-	ln -s $(GLIB_RUNTIME_LIBDIR)/libgobject-2.0.so.0.$(LT_CURRENT).0 $(DESTDIR)$(libdir)/libgobject-2.0.so
+	ln -s $(GLIB_RUNTIME_LIBDIR)/libgobject-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/libgobject-2.0.so
 endif
diff -up glib-2.22.1/gthread/Makefile.am.fix-runtime-libdir glib-2.22.1/gthread/Makefile.am
--- glib-2.22.1/gthread/Makefile.am.fix-runtime-libdir	2009-09-29 22:57:23.832578375 -0400
+++ glib-2.22.1/gthread/Makefile.am	2009-09-29 22:57:51.978582942 -0400
@@ -96,7 +96,7 @@ if HAVE_GLIB_RUNTIME_LIBDIR
 install-data-hook:
 	mkdir -p $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	mv $(DESTDIR)$(libdir)/libgthread-2.0.so.0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
-	mv $(DESTDIR)$(libdir)/libgthread-2.0.so.0.$(LT_CURRENT).0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
+	mv $(DESTDIR)$(libdir)/libgthread-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
 	rm -f $(DESTDIR)$(libdir)/libgthread-2.0.so
-	ln -s $(GLIB_RUNTIME_LIBDIR)/libgthread-2.0.so.0.$(LT_CURRENT).0 $(DESTDIR)$(libdir)/libgthread-2.0.so
+	ln -s $(GLIB_RUNTIME_LIBDIR)/libgthread-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/libgthread-2.0.so
 endif
