--- lxde-common/lxde-common-0.5.5/lxpanel/panel.in	2011-07-29 16:04:03.000000000 +0200
+++ lxde-common/lxde-common-0.5.5/lxpanel/copy-panel.in	2012-04-06 22:33:46.000000000 +0200
@@ -50,10 +50,16 @@
     type = launchbar
     Config {
         Button {
-            id=pcmanfm.desktop
+            id=spacefm.desktop
         }
         Button {
-            id=firefox.desktop
+            id=gnome-terminal.desktop
+        }
+        Button {
+            id=net-chromium-browser.desktop
+        }
+        Button {
+            id=net-thunderbird.desktop
         }
     }
 }
@@ -70,6 +76,7 @@
     Config {
         Button1=iconify
         Button2=shade
+        Toggle=1
     }
 }
 
@@ -110,10 +117,6 @@
 }
 
 Plugin {
-    type = netstat
-}
-
-Plugin {
     type = cpu
 }
 
@@ -124,7 +127,7 @@
 Plugin {
     type = dclock
     Config {
-        ClockFmt=%R
+        ClockFmt=%R %D
         TooltipFmt=%A %x
         BoldFont=0
     }
@@ -134,7 +137,7 @@
     type = launchbar
     Config {
         Button {
-            id=lxde-logout.desktop
+            id=net-lxde-logout.desktop
         }
     }
 }
