--- lxde-common/lxde-common-0.5.5/lxpanel/panel.in	2011-07-29 16:04:03.000000000 +0200
+++ lxde-common/lxde-common-0.5.5/lxpanel/foresight-panel.in	2012-03-22 22:40:22.000000000 +0100
@@ -53,7 +53,13 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=firefox.desktop
+            id=gnome-terminal.desktop
+        }
+        Button {
+            id=net-chromium-browser.desktop
+        }
+        Button {
+            id=net-thunderbird.desktop
         }
     }
 }
@@ -70,6 +76,7 @@
     Config {
         Button1=iconify
         Button2=shade
+        Toggle=1
     }
 }
 
@@ -109,12 +116,9 @@
     }
 }
 
-Plugin {
-    type = netstat
-}
 
 Plugin {
-    type = cpu
+    type = batt
 }
 
 Plugin {
@@ -134,7 +138,7 @@
     type = launchbar
     Config {
         Button {
-            id=lxde-logout.desktop
+            id=net-lxde-logout.desktop
         }
     }
 }
