--- lxde-common/lxde-common-0.5.5/lxpanel/panel.in	2012-03-21 21:23:10.000000000 +0100
+++ lxde-common/lxde-common-0.5.5/lxpanel/foresight-panel.in	2012-03-21 21:22:57.000000000 +0100
@@ -53,7 +53,10 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=firefox.desktop
+            id=lxterminal.desktop
+        }
+        Button {
+            id=net-chromium-browser.desktop
         }
     }
 }
@@ -110,7 +113,7 @@
 }
 
 Plugin {
-    type = netstat
+    type = batt
 }
 
 Plugin {
@@ -134,7 +137,7 @@
     type = launchbar
     Config {
         Button {
-            id=lxde-logout.desktop
+            id=net-lxde-logout.desktop
         }
     }
 }
