diff -r e74f2f16b63e src/launcher-sidebar.c
--- a/src/launcher-sidebar.c	Wed Nov 05 15:44:01 2008 -0500
+++ b/src/launcher-sidebar.c	Wed Nov 05 15:48:31 2008 -0500
@@ -952,6 +952,7 @@
   return shortcut;
 }
 
+/*
 static void 
 on_quit_clicked (ClutterActor *shortcut, gpointer null)
 {
@@ -963,6 +964,7 @@
 
   gtk_widget_destroy (quit);
 }
+*/
 
 static void
 load_volumes (LauncherSidebar *sidebar)
@@ -1010,10 +1012,13 @@
   clutter_container_add_actor (CLUTTER_CONTAINER (sidebar), shortcut);
   clutter_actor_set_position (shortcut, 0, 
                               cfg->win_height - (bar_height)-(PANEL_HEIGHT*2));
-  clutter_actor_set_name (shortcut, "gnome-session-save --gui --kill");
+  clutter_actor_set_name (shortcut, "gnome-shutdown");
   clutter_actor_show_all (shortcut);
   priv->links = g_list_append (priv->links, shortcut);
+  /*
   g_signal_connect (shortcut, "clicked", G_CALLBACK (on_quit_clicked), NULL);
+  */
+  g_signal_connect (shortcut, "clicked", G_CALLBACK (on_link_clicked), NULL);
 }
 
 static void
