diff -Naur PolicyKit-0.5.orig/polkit-dbus/polkit-dbus.c PolicyKit-0.5/polkit-dbus/polkit-dbus.c
--- PolicyKit-0.5.orig/polkit-dbus/polkit-dbus.c	2007-08-21 10:53:24.000000000 -0400
+++ PolicyKit-0.5/polkit-dbus/polkit-dbus.c	2007-08-22 12:53:15.000000000 -0400
@@ -607,13 +607,16 @@
 	/* only get the context if we are enabled */
 	selinux_context = NULL;
 	if (is_selinux_enabled () != 0) {
-		if (getpidcon (pid, &secon) != 0) {
-			g_warning ("Cannot lookup SELinux context for pid %d: %s", pid, strerror (errno));
-			goto out;
-		}
-		selinux_context = g_strdup (secon);
-		freecon (secon);
-	}
+        /* only get the context if we are enabled */
+        selinux_context = NULL;
+        if (is_selinux_enabled () != 0) {
+            if (getpidcon (pid, &secon) != 0) {
+                g_warning ("Cannot lookup SELinux context for pid %d: %s", pid, strerror (errno));
+                goto out;
+            }
+            selinux_context = g_strdup (secon);
+            freecon (secon);
+        }
 #else
         selinux_context = NULL;
 #endif
