diff -Naur PolicyKit-0.5.orig/data/PolicyKit.conf.in PolicyKit-0.5/data/PolicyKit.conf.in
--- PolicyKit-0.5.orig/data/PolicyKit.conf.in	2007-08-21 07:23:07.000000000 -0400
+++ PolicyKit-0.5/data/PolicyKit.conf.in	2007-08-22 16:58:23.000000000 -0400
@@ -6,4 +6,5 @@
 <!-- See the manual page PolicyKit.conf(5) for file format -->
 
 <config version="0.1">
+    <define_admin_auth group="wheel" />
 </config>
