diff -Naur PolicyKit-0.5.orig/data/PolicyKit.conf.in PolicyKit-0.5/data/PolicyKit.conf.in
--- PolicyKit-0.5.orig/data/PolicyKit.conf.in	2007-08-21 07:23:07.000000000 -0400
+++ PolicyKit-0.5/data/PolicyKit.conf.in	2007-08-22 13:25:58.000000000 -0400
@@ -6,4 +6,7 @@
 <!-- See the manual page PolicyKit.conf(5) for file format -->
 
 <config version="0.1">
+    <match group="wheel">
+        <return result="yes"/>
+    </match>
 </config>
