--- kdebase-3.5.9.orig/startkde 2008-05-29 16:48:00.000000000 -0400
+++ kdebase-3.5.9/startkde      2008-05-30 02:10:24.000000000 -0400
@@ -14,6 +14,11 @@
        exit 1
 fi

+# Before we do anything else, launch pulseaudio daemon, so that artsd doesn't
+# throw a fit when it starts and pulse is not running
+# https://issues.foresightlinux.org/browse/KDE-26
+pulseaudio -D
+
 # Set the background to plain grey.
 # The standard X background is nasty, causing moire effects and exploding
 # people's heads. We use colours from the standard KDE palette for those with
