diff -uNr kdebase-3.5.9-orig/kdm/config.def kdebase-3.5.9/kdm/config.def
--- kdebase-3.5.9-orig/kdm/config.def	2008-05-14 14:08:30.000000000 -0400
+++ kdebase-3.5.9/kdm/config.def	2008-05-14 14:09:24.000000000 -0400
@@ -2375,7 +2375,7 @@
 Type: string
 Default: *KDMCONF "/backgroundrc"
 User: greeter-c
-Instance: #*/""
+Instance: */"/usr/share/kde3-profiles/default/share/apps/kdm/backgroundrc"
 Update: handBgCfg
 Comment:
  The configuration file to be used by krootimage.
@@ -2599,7 +2599,7 @@
 Type: bool
 Default: false
 User: greeter
-Instance: #*/true
+Instance: */true
 Comment: &
 Description:
  Whether the greeter should be themed.
@@ -2608,7 +2608,7 @@
 Type: string
 Default: ""
 User: greeter
-Instance: */KDMDATA "/themes/circles"
+Instance: */"/usr/share/kde3-profiles/default/share/apps/kdm/themes/Clean_Drops"
 Comment: &
 Description:
  The theme to use for the greeter. Can point to either a directory or an XML
