diff -ur kdebase-3.5.9.orig/kdm/kfrontend/genkdmconf.c kdebase-3.5.9/kdm/kfrontend/genkdmconf.c
--- kdebase-3.5.9.orig/kdm/kfrontend/genkdmconf.c	2008-04-23 12:12:17.000000000 -0400
+++ kdebase-3.5.9/kdm/kfrontend/genkdmconf.c	2008-04-23 12:40:55.000000000 -0400
@@ -565,14 +565,7 @@
 #ifdef XDMCP
 static const char def_willing[] =
 "#! /bin/sh\n"
-"# The output of this script is displayed in the chooser window\n"
-"# (instead of \"Willing to manage\").\n"
-"\n"
-"load=`uptime|sed -e 's/^.*load[^0-9]*//'`\n"
-"nrusers=`who|cut -c 1-8|sort -u|wc -l|sed 's/^[ \t]*//'`\n"
-"s=\"\"; [ \"$nrusers\" != 1 ] && s=s\n"
-"\n"
-"echo \"${nrusers} user${s}, load: ${load}\"\n";
+". /etc/X11/xdm/Xwilling\n";
 #endif
 
 static const char def_setup[] =
@@ -594,25 +587,7 @@
 "#\n"
 "#chown $USER /dev/console\n"
 "\n"
-#ifdef _AIX
-"# We create a pseudodevice for finger.  (host:0 becomes xdm/host_0)\n"
-"# Without it, finger errors out with \"Can't stat /dev/host:0\".\n"
-"#\n"
-"#devname=`echo $DISPLAY | cut -c1-8`\n"
-"#if [ ! -d /dev/xdm ]; then\n"
-"#  mkdir /dev/xdm\n"
-"#  chmod 755 /dev/xdm\n"
-"#fi\n"
-"#touch /dev/xdm/$devname\n"
-"#chmod 644 /dev/xdm/$devname\n"
-"#exec sessreg -a -l xdm/$devname -h \"`echo $DISPLAY | cut -d: -f1`\""
-#else
-"#exec sessreg -a -l $DISPLAY -h \"`echo $DISPLAY | cut -d: -f1`\""
-# ifdef BSD
-" -x " KDMCONF "/Xservers"
-# endif
-#endif /* _AIX */
-" $USER\n"
+". /etc/X11/xdm/Xstartup\n"
 "\n# NOTE: The session is aborted if the last command returns non-zero.\n";
 
 static const char def_reset[] =
@@ -625,16 +600,7 @@
 "#chown root /dev/console\n"
 "#chmod 622 /dev/console\n"
 "\n"
-#ifdef _AIX
-"#devname=`echo $DISPLAY | cut -c1-8`\n"
-"#exec sessreg -d -l xdm/$devname -h \"`echo $DISPLAY | cut -d: -f1`\""
-#else
-"#exec sessreg -d -l $DISPLAY -h \"`echo $DISPLAY | cut -d: -f1`\""
-# ifdef BSD
-" -x " KDMCONF "/Xservers"
-# endif
-#endif /* _AIX */
-" $USER\n";
+". /etc/X11/xdm/Xstartup\n";
 
 static const char def_session1[] =
 "#! /bin/sh\n"
@@ -683,27 +649,8 @@
 "    ;;\n"
 "esac\n"
 "\n"
-"[ -f /etc/xprofile ] && . /etc/xprofile\n"
-"[ -f $HOME/.xprofile ] && . $HOME/.xprofile\n"
-"\n"
-"case $session in\n"
-"  \"\")\n"
-"    exec xmessage -center -buttons OK:0 -default OK \"Sorry, $DESKTOP_SESSION is no valid session.\"\n"
-"    ;;\n"
-"  failsafe)\n"
-"    exec xterm -geometry 80x24-0-0\n"
-"    ;;\n"
-"  custom)\n"
-"    exec $HOME/.xsession\n"
-"    ;;\n"
-"  default)\n"
-"    exec " KDE_BINDIR "/startkde\n"
-"    ;;\n"
-"  *)\n"
-"    eval exec \"$session\"\n"
-"    ;;\n"
-"esac\n"
-"exec xmessage -center -buttons OK:0 -default OK \"Sorry, cannot execute $session. Check $DESKTOP_SESSION.desktop.\"\n";
+"# Invoke xdm session script\n"
+". /etc/X11/xdm/Xsession\n";
 
 static const char def_background[] =
 "[Desktop0]\n"
