diff -ur kdebase-3.5.9.orig/kdm/config.def kdebase-3.5.9/kdm/config.def
--- kdebase-3.5.9.orig/kdm/config.def	2008-04-23 12:45:59.000000000 -0400
+++ kdebase-3.5.9/kdm/config.def	2008-04-23 11:39:30.000000000 -0400
@@ -1361,7 +1361,7 @@
 Type: string
 Default: ""
 User: core
-Instance: :*/"-nolisten tcp"
+Instance: :*/"-nolisten tcp -dpi 96"
 Comment: &
 Description:
  Additional arguments for the &X-Server;s for local sessions.
