diff -uNr kdebase-3.5.9.orig/kdm/config.def kdebase-3.5.9/kdm/config.def
--- kdebase-3.5.9.orig/kdm/config.def   2008-06-03 15:16:01.000000000 -0400
+++ kdebase-3.5.9/kdm/config.def        2008-06-03 15:20:00.000000000 -0400
@@ -2011,7 +2011,7 @@
 Type: string
 Default: ""
 User: greeter(logo)
-Instance: */KDMDATA "/pics/kdelogo.png"
+Instance: */"/usr/share/kde3-profiles/default/share/apps/kdm/foresight.png"
 Comment:
  The image to show when LogoArea=Logo.
 Description:
