--- a/apps/kdepasswd/kcm/main.cpp	2009-08-01 14:43:42 +0000
+++ b/apps/kdepasswd/kcm/main.cpp	2009-08-01 16:05:58 +0000
@@ -22,6 +22,12 @@
 
 #include "main.h"
 
+// for getpwuid(getuid())
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+
+
 #include <QtGui/QLayout>
 #include <QtGui/QLabel>
 #include <QtGui/QLineEdit>
@@ -143,7 +149,14 @@
 
 	_kes->setProfile(_kes->defaultProfileName());
 
-	_mw->leRealname->setText( _kes->getSetting( KEMailSettings::RealName ));
+	QString realName = _kes->getSetting( KEMailSettings::RealName );
+        if (realName.isEmpty()) {
+		QString gecos( getpwuid(getuid())->pw_gecos );
+		realName = gecos.left(gecos.indexOf(','));
+		_kes->setSetting( KEMailSettings::RealName, realName );
+	}
+	_mw->leRealname->setText( realName );
+
 	_mw->leEmail->setText( _kes->getSetting( KEMailSettings::EmailAddress ));
 	_mw->leOrganization->setText( _kes->getSetting( KEMailSettings::Organization ));
 	_mw->leSMTP->setText( _kes->getSetting( KEMailSettings::OutServer ));

--- a/apps/kdepasswd/kcm/main_widget.ui	2009-08-01 14:43:42 +0000
+++ b/apps/kdepasswd/kcm/main_widget.ui	2009-08-01 15:31:23 +0000
@@ -130,6 +130,9 @@
       </item>
       <item row="0" column="1" >
        <widget class="KLineEdit" name="leRealname" >
+        <property name="enabled">
+         <bool>false</bool>
+        </property>
         <property name="showClearButton" stdset="0" >
          <bool>true</bool>
         </property>

