Index: kdebase-4.4.0/apps/plasma/applets/folderview/folderview.cpp
===================================================================
--- kdebase-4.4.0.orig/apps/plasma/applets/folderview/folderview.cpp	2010-02-21 10:24:10.042000060 +0100
+++ kdebase-4.4.0/apps/plasma/applets/folderview/folderview.cpp	2010-02-21 10:27:49.382499808 +0100
@@ -576,6 +576,15 @@
     }
 }
 
+void FolderView::configChanged()
+{
+    setUrl(config().readEntry("url", KUrl(m_url))); 
+    m_dirModel->dirLister()->openUrl(m_url);
+
+    delete m_itemActions;
+    m_itemActions = 0;
+}
+
 void FolderView::configAccepted()
 {
     KUrl url;
Index: kdebase-4.4.0/apps/plasma/applets/folderview/folderview.h
===================================================================
--- kdebase-4.4.0.orig/apps/plasma/applets/folderview/folderview.h	2010-02-21 10:24:12.150000731 +0100
+++ kdebase-4.4.0/apps/plasma/applets/folderview/folderview.h	2010-02-21 10:26:45.514498870 +0100
@@ -107,6 +107,7 @@
     void dragMoveEvent(QGraphicsSceneDragDropEvent *event);
     void dropEvent(QGraphicsSceneDragDropEvent *event);
     QSizeF sizeHint(Qt::SizeHint which, const QSizeF &constraint) const;
+    void configChanged();
 
 protected slots:
     // These slots are for KonqPopupMenu
