diff -Nur kdebase-4.2.85/apps/dolphin/src/settings/dolphin_directoryviewpropertysettings.kcfg kdebase-4.2.85.new/apps/dolphin/src/settings/dolphin_directoryviewpropertysettings.kcfg
--- kdebase-4.2.85/apps/dolphin/src/settings/dolphin_directoryviewpropertysettings.kcfg	2009-04-28 15:46:07.000000000 +0200
+++ kdebase-4.2.85.new/apps/dolphin/src/settings/dolphin_directoryviewpropertysettings.kcfg	2009-06-02 16:54:12.000000000 +0200
@@ -26,7 +26,7 @@
         <entry name="ShowPreview" type="Bool" >
             <label context="@label">Show preview</label>
             <whatsthis context="@info:whatsthis">When this option is enabled, a preview of the file content is shown as an icon.</whatsthis>
-            <default>false</default>
+            <default>true</default>
         </entry>
 
         <entry name="CategorizedSorting" type="Bool" >
