--- autofs-5.0.0_beta5/samples/rc.autofs.in.orig	2006-06-22 12:59:04.000000000 -0400
+++ autofs-5.0.0_beta5/samples/rc.autofs.in	2006-06-22 13:00:03.000000000 -0400
@@ -7,7 +7,10 @@
 # On most distributions, this file should be called:
 # /etc/rc.d/init.d/autofs or /etc/init.d/autofs or /etc/rc.d/rc.autofs
 #
-
+# chkconfig: 345 28 72
+# processname: /usr/sbin/automount
+# config: /etc/auto.master
+# description: Automounts filesystems on demand
 #
 # Location of the automount daemon and the init directory
 #
