diff -ur gnomebaker-0.4.orig/help/C/Makefile.in gnomebaker-0.4/help/C/Makefile.in
--- gnomebaker-0.4.orig/help/C/Makefile.in	2005-06-02 17:38:43.000000000 -0400
+++ gnomebaker-0.4/help/C/Makefile.in	2005-07-20 10:28:06.000000000 -0400
@@ -212,8 +212,8 @@
 # ************* Begin of section some packagers may need to modify  **************
 # This variable (docdir) specifies where the documents should be installed.
 # This default value should work for most packages.
-docdir = $(datadir)/@PACKAGE@/doc/$(docname)/$(lang)
-#docdir = $(datadir)/gnome/help/$(docname)/$(lang)
+#docdir = $(datadir)/@PACKAGE@/doc/$(docname)/$(lang)
+docdir = $(datadir)/gnome/help/$(docname)/$(lang)
 
 # **************  You should not have to edit below this line  *******************
 # ** GSt did ... 
diff -ur gnomebaker-0.4.orig/xmldocs.make gnomebaker-0.4/xmldocs.make
--- gnomebaker-0.4.orig/xmldocs.make	2005-02-06 08:43:51.000000000 -0500
+++ gnomebaker-0.4/xmldocs.make	2005-07-20 10:22:10.000000000 -0400
@@ -1,8 +1,8 @@
 # ************* Begin of section some packagers may need to modify  **************
 # This variable (docdir) specifies where the documents should be installed.
 # This default value should work for most packages.
-docdir = $(datadir)/@PACKAGE@/doc/$(docname)/$(lang)
-#docdir = $(datadir)/gnome/help/$(docname)/$(lang)
+#docdir = $(datadir)/@PACKAGE@/doc/$(docname)/$(lang)
+docdir = $(datadir)/gnome/help/$(docname)/$(lang)
 
 # **************  You should not have to edit below this line  *******************
 # ** GSt did ... 
