diff --git a/COPYING b/COPYING
diff --git a/fusion-icon.desktop b/fusion-icon.desktop
index b2568ce..d1e779e 100644
--- a/fusion-icon.desktop
+++ b/fusion-icon.desktop
@@ -1,9 +1,9 @@
 [Desktop Entry]
-Name=Compiz Fusion Icon
+Name=Compiz Fusion
 Comment=Start and manage Compiz Fusion
 Exec=fusion-icon
 Icon=fusion-icon
 Terminal=false
 Type=Application
-Categories=System;Application;
+Categories=GNOME;GTK;Settings;DesktopSettings;
 Encoding=UTF-8
diff --git a/src/libfusionicon.py b/src/libfusionicon.py
index 2788c18..9a6cc38 100644
--- a/src/libfusionicon.py
+++ b/src/libfusionicon.py
@@ -21,7 +21,7 @@ import ConfigParser, compizconfig, commands, os, subprocess, time
 mesa_libgl_locations = ('/usr/lib/fglrx/libGL.so.1.2.xlibmesa', '/opt/mesa-xgl/lib/libGL.so.1.2')
 tfp = 'GLX_EXT_texture_from_pixmap'
 
-apps = ('compiz.real', 'ccsm', 'compiz', 'gtk-window-decorator', 'kde-window-decorator', 'emerald', 'metacity', 'kwin', 'xfwm4') 
+apps = ('compiz.real', 'ccsm', 'compiz-start', 'gtk-window-decorator', 'kde-window-decorator', 'emerald', 'metacity', 'kwin', 'xfwm4') 
 
 wmlist = ['xfwm4', 'kwin', 'metacity'] #add compiz later when it's defined
 wms_to_kill = ['xfwm4', 'kwin']
@@ -233,7 +233,7 @@ for app in apps:
 # If compiz.real exists (which means that compiz is just a wrapper script), use that.
 compiz = ''
 if is_installed('compiz.real'): compiz = 'compiz.real'
-elif is_installed('compiz'): compiz = 'compiz'
+elif is_installed('compiz-start'): compiz = 'compiz-start'
 wmlist.append(compiz)
 
 # Check presence of decorators:
