diff -r f0570fd53b6e lib/sensors-interface.c
--- a/lib/sensors-interface.c	Tue May 10 09:14:33 2011 +0200
+++ b/lib/sensors-interface.c	Tue May 10 09:25:07 2011 +0200
@@ -34,7 +34,7 @@
 #include <gtk/gtk.h>
 
 /* Xfce includes */
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 /* Package includes */
 #include <sensors-interface-common.h>
diff -r f0570fd53b6e src/actions.c
--- a/src/actions.c	Tue May 10 09:14:33 2011 +0200
+++ b/src/actions.c	Tue May 10 09:25:07 2011 +0200
@@ -28,7 +28,7 @@
 #include <gtk/gtk.h>
 
 /* Xfce includes */
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 /* Package includes */
 #include <middlelayer.h>
diff -r f0570fd53b6e src/callbacks.c
--- a/src/callbacks.c	Tue May 10 09:14:33 2011 +0200
+++ b/src/callbacks.c	Tue May 10 09:25:07 2011 +0200
@@ -27,7 +27,7 @@
 #include <stdlib.h>
 
 /* Xfce includes */
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 /* Package includes */
 #include <sensors-interface.h>
diff -r f0570fd53b6e src/interface.c
--- a/src/interface.c	Tue May 10 09:14:33 2011 +0200
+++ b/src/interface.c	Tue May 10 09:25:07 2011 +0200
@@ -24,7 +24,7 @@
 #endif
 
 /* Xfce includes */
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 /* Package includes */
 #include <cpu.h>
