diff -up xf86-video-mga-20071210/src/mga_dri.c.jx xf86-video-mga-20071210/src/mga_dri.c
--- xf86-video-mga-20071210/src/mga_dri.c.jx	2007-12-10 14:30:30.000000000 -0500
+++ xf86-video-mga-20071210/src/mga_dri.c	2008-01-02 11:04:02.000000000 -0500
@@ -38,11 +38,6 @@
 
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
-#define PSZ 8
-#include "cfb.h"
-#undef PSZ
-#include "cfb16.h"
-#include "cfb32.h"
 
 #include "miline.h"
 
diff -up xf86-video-mga-20071210/src/mga_storm.c.jx xf86-video-mga-20071210/src/mga_storm.c
--- xf86-video-mga-20071210/src/mga_storm.c.jx	2007-12-10 14:30:30.000000000 -0500
+++ xf86-video-mga-20071210/src/mga_storm.c	2008-01-02 11:04:43.000000000 -0500
@@ -1,7 +1,3 @@
-/* $XFree86: xc/programs/Xserver/hw/xfree86/drivers/mga/mga_storm.c,v 1.99tsi Exp $ */
-
-#define PSZ 8
-
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
@@ -27,7 +23,6 @@
 #include "servermd.h"
 
 #ifdef XF86DRI
-#include "cfb.h"
 #include "GL/glxtokens.h"
 #endif
 
diff -up xf86-video-mga-20071210/src/mga_driver.c.jx xf86-video-mga-20071210/src/mga_driver.c
--- xf86-video-mga-20071210/src/mga_driver.c.jx	2008-01-02 11:04:02.000000000 -0500
+++ xf86-video-mga-20071210/src/mga_driver.c	2008-01-02 11:04:02.000000000 -0500
@@ -96,8 +96,6 @@
 #include "shadowfb.h"
 #include "fbdevhw.h"
 
-#include "cfb8_32.h"
-
 #ifdef XF86DRI
 #include "dri.h"
 #endif
@@ -357,12 +355,6 @@ static const char *fbSymbols[] = {
     NULL
 };
 
-static const char *xf8_32bppSymbols[] = {
-    "cfb8_32ScreenInit",
-    "xf86Overlay8Plus32Init",
-    NULL
-};
-
 #ifdef USE_EXA
 static const char *exaSymbols[] = {
     "exaDriverInit",
@@ -559,7 +551,7 @@ mgaSetup(pointer module, pointer opts, i
 #ifdef USE_EXA
                           exaSymbols,
 #endif
-			  xf8_32bppSymbols, ramdacSymbols,
+			  ramdacSymbols,
 			  ddcSymbols, i2cSymbols, shadowSymbols,
 			  fbdevHWSymbols, vbeSymbols,
 			  fbSymbols, int10Symbols,
@@ -2641,11 +2633,7 @@ MGAPreInit(ScrnInfoPtr pScrn, int flags)
 
     /* Load the required framebuffer */
     if (pMga->Overlay8Plus24) {
-	if (!xf86LoadSubModule(pScrn, "xf8_32bpp")) {
-	    MGAFreeRec(pScrn);
-	    return FALSE;
-	}
-	xf86LoaderReqSymLists(xf8_32bppSymbols, NULL);
+	return FALSE;
     } else {
 	if (!xf86LoadSubModule(pScrn, "fb")) {
 	    MGAFreeRec(pScrn);
@@ -3680,16 +3668,9 @@ MGAScreenInit(int scrnIndex, ScreenPtr p
 #endif
 
 
-    if (pMga->Overlay8Plus24) {
-	ret = cfb8_32ScreenInit(pScreen, FBStart,
-			width, height,
-			pScrn->xDpi, pScrn->yDpi,
-			displayWidth);
-    } else {
 	ret = fbScreenInit(pScreen, FBStart, width, height,
 			   pScrn->xDpi, pScrn->yDpi,
 			   displayWidth, pScrn->bitsPerPixel);
-    }
 
     if (!ret)
 	return FALSE;
@@ -3773,11 +3754,6 @@ MGAScreenInit(int scrnIndex, ScreenPtr p
 	NULL, f))
 	return FALSE;
 
-    if(pMga->Overlay8Plus24) { /* Must come after colormap initialization */
-	if(!xf86Overlay8Plus32Init(pScreen))
-	    return FALSE;
-    }
-
     if(pMga->ShadowFB) {
 	RefreshAreaFuncPtr refreshArea = MGARefreshArea;
 
