diff -ruN xen-3.0.4_1-src.orig/tools/blktap/drivers/Makefile xen-3.0.4_1-src.new/tools/blktap/drivers/Makefile
--- xen-3.0.4_1-src.orig/tools/blktap/drivers/Makefile	2007-01-08 10:00:46.000000000 -0500
+++ xen-3.0.4_1-src.new/tools/blktap/drivers/Makefile	2007-01-18 11:26:37.000000000 -0500
@@ -8,7 +8,7 @@
 INST_DIR  = /usr/sbin
 LIBAIO_DIR   = ../../libaio/src
 
-CFLAGS   += -Werror
+#CFLAGS   += -Werror
 CFLAGS   += -Wno-unused
 CFLAGS   += -fno-strict-aliasing
 CFLAGS   += -I $(XEN_LIBXC) -I $(LIBAIO_DIR)
diff -ruN xen-3.0.4_1-src.orig/tools/blktap/lib/Makefile xen-3.0.4_1-src.new/tools/blktap/lib/Makefile
--- xen-3.0.4_1-src.orig/tools/blktap/lib/Makefile	2007-01-08 10:00:47.000000000 -0500
+++ xen-3.0.4_1-src.new/tools/blktap/lib/Makefile	2007-01-18 11:26:45.000000000 -0500
@@ -14,7 +14,7 @@
 SRCS     :=
 SRCS     += xenbus.c blkif.c xs_api.c
 
-CFLAGS   += -Werror
+#CFLAGS   += -Werror
 CFLAGS   += -Wno-unused
 CFLAGS   += -fno-strict-aliasing -fPIC
 # get asprintf():
diff -ruN xen-3.0.4_1-src.orig/tools/console/Makefile xen-3.0.4_1-src.new/tools/console/Makefile
--- xen-3.0.4_1-src.orig/tools/console/Makefile	2007-01-08 10:00:47.000000000 -0500
+++ xen-3.0.4_1-src.new/tools/console/Makefile	2007-01-18 11:26:00.000000000 -0500
@@ -5,7 +5,7 @@
 DAEMON_INSTALL_DIR = /usr/sbin
 CLIENT_INSTALL_DIR = /usr/$(LIBDIR)/xen/bin
 
-CFLAGS  += -Werror
+#CFLAGS  += -Werror
 
 CFLAGS  += -I $(XEN_LIBXC)
 CFLAGS  += -I $(XEN_XENSTORE)
diff -ruN xen-3.0.4_1-src.orig/tools/xenstore/Makefile xen-3.0.4_1-src.new/tools/xenstore/Makefile
--- xen-3.0.4_1-src.orig/tools/xenstore/Makefile	2007-01-08 10:00:52.000000000 -0500
+++ xen-3.0.4_1-src.new/tools/xenstore/Makefile	2007-01-18 11:25:48.000000000 -0500
@@ -6,7 +6,7 @@
 MINOR = 0
 
 PROFILE=#-pg
-BASECFLAGS=-Werror
+BASECFLAGS=
 # Make gcc generate dependencies.
 BASECFLAGS += -Wp,-MD,.$(@F).d
 PROG_DEP = .*.d
