--- src/modules/module-udev-detect.c.orig	2009-12-11 17:38:09.164403736 +0000
+++ src/modules/module-udev-detect.c	2009-12-14 11:55:29.185232248 +0000
@@ -555,10 +555,17 @@
     if (u->inotify_fd >= 0)
         return 0;
 
+#ifdef IN_NONBLOCK
     if ((u->inotify_fd = inotify_init1(IN_CLOEXEC|IN_NONBLOCK)) < 0) {
         pa_log("inotify_init1() failed: %s", pa_cstrerror(errno));
         return -1;
     }
+#else
+    if ((u->inotify_fd = inotify_init()) < 0) {
+        pa_log("inotify_init() failed: %s", pa_cstrerror(errno));
+        return -1;
+    }
+#endif
 
     dev_snd = pa_sprintf_malloc("%s/snd", udev_get_dev_path(u->udev));
     r = inotify_add_watch(u->inotify_fd, dev_snd, IN_ATTRIB|IN_CLOSE_WRITE|IN_DELETE_SELF|IN_MOVE_SELF);
