--- module-udev-detect.c.orig	2009-12-11 17:38:09.164403736 +0000
+++ module-udev-detect.c	2009-12-11 17:44:27.620639170 +0000
@@ -39,6 +39,14 @@
 
 #include "module-udev-detect-symdef.h"
 
+#ifndef IN_CLOEXEC
+#define IN_CLOEXEC O_CLOEXEC
+#endif
+
+#ifndef IN_NONBLOCK
+#define IN_NONBLOCK O_NONBLOCK
+#endif
+
 PA_MODULE_AUTHOR("Lennart Poettering");
 PA_MODULE_DESCRIPTION("Detect available audio hardware and load matching drivers");
 PA_MODULE_VERSION(PACKAGE_VERSION);
