diff -upr gst-plugins-0.8.10-old/ext/cdparanoia/gstcdparanoia.c gst-plugins-0.8.10-new/ext/cdparanoia/gstcdparanoia.c
--- gst-plugins-0.8.10-old/ext/cdparanoia/gstcdparanoia.c	2005-06-14 11:37:53.000000000 -0500
+++ gst-plugins-0.8.10-new/ext/cdparanoia/gstcdparanoia.c	2005-08-30 01:27:31.000000000 -0500
@@ -1210,7 +1210,7 @@ cdparanoia_convert (GstPad * pad,
         if (src_value < 0 || src_value > src->d->tracks)
           return FALSE;
 
-        sector = get_first_sector (src, src_value);
+        sector = cdda_track_firstsector (src->d, src_value + 1);
       } else if (src_format == sector_format) {
         sector = src_value;
       } else {
@@ -1283,9 +1283,15 @@ cdparanoia_query (GstPad * pad, GstQuery
         *value = src->d->tracks;
         res = TRUE;
       } else {
-        res = gst_pad_convert (src->srcpad, sector_format,
-            get_first_sector (src, src->cur_track + 1) -
-            get_first_sector (src, src->cur_track), format, value);
+        if (src->track_mode) {
+          /* Current track*/
+          res = gst_pad_convert (src->srcpad, sector_format,
+              get_first_sector (src, src->cur_track + 1) -
+              get_first_sector (src, src->cur_track), format, value);
+				} else {
+          /* CD total */
+          res = gst_pad_convert (src->srcpad, sector_format, src->last_sector + 1, format, value);
+				}
       }
       break;
     case GST_QUERY_POSITION:
