diff -ur booty-0.31.1.orig/bootloaderInfo.py booty-0.31.1/bootloaderInfo.py
--- booty-0.31.1.orig/bootloaderInfo.py	2003-08-15 16:50:35.000000000 -0400
+++ booty-0.31.1/bootloaderInfo.py	2005-04-17 00:45:51.000000000 -0400
@@ -94,6 +94,9 @@
             if str: str = str + " "
             str = str + ("%s=ide-scsi" % device)
 
+	# Add args for bootsplash
+	str = str + "splash=silent vga=791 "
+
         # look for kernel arguments we know should be preserved and add them
         ourargs = ["speakup_synth=", "apic", "noapic", "apm=", "ide=nodma", "noht", "acpi=", "video="]
         f = open("/proc/cmdline")
