--- booty-0.31.1/bootloaderInfo.py.orig	2005-03-03 09:16:26.719667800 -0500
+++ booty-0.31.1/bootloaderInfo.py	2005-03-03 09:16:41.434430816 -0500
@@ -17,7 +17,7 @@
 
 import os, sys
 import crypt
-import whrandom
+import random
 import butil
 import string
 
@@ -585,8 +585,8 @@
             saltLen = 8
 
 	    for i in range(saltLen):
-                salt = salt + whrandom.choice (string.letters +
-                                               string.digits + './')
+                salt = salt + random.choice (string.letters +
+                                             string.digits + './')
 
             self.password = crypt.crypt (val, salt)
             self.pure = val
