--- booty-0.31.1/bootloaderInfo.py.2.6	2005-03-30 18:35:39.180960864 -0500
+++ booty-0.31.1/bootloaderInfo.py	2005-03-30 18:35:52.643914184 -0500
@@ -88,12 +88,6 @@
                 except Exception, e:
                     pass
 
-        import isys
-        cdrw = isys.ideCdRwList()
-        for device in cdrw:
-            if str: str = str + " "
-            str = str + ("%s=ide-scsi" % device)
-
         # look for kernel arguments we know should be preserved and add them
         ourargs = ["speakup_synth=", "apic", "noapic", "apm=", "ide=nodma", "noht", "acpi=", "video="]
         f = open("/proc/cmdline")
