--- indicator-applet-0.2.0.orig/src/applet-main.c
+++ indicator-applet-0.2.0/src/applet-main.c
@@ -212,6 +212,13 @@
 	GtkWidget *menubar;
 	gint i;
 	gint indicators_loaded = 0;
+
+	/* check if we are running stracciatella session */
+	if (g_strcmp0(g_getenv("GDMSESSION"), "gnome-stracciatella") == 0) {
+		g_debug("Running stracciatella GNOME session, disabling myself");
+		return TRUE;
+	}
+  
 	static gboolean first_time = FALSE;
 
 	if (!first_time)
