--- nash/block.c.orig	2006-10-30 13:05:29.000000000 -0500
+++ nash/block.c	2006-10-30 13:06:37.000000000 -0500
@@ -28,7 +28,6 @@
 #include <argz.h>
 #include <envz.h>
 
-#include <selinux/selinux.h>
 #include <blkid/blkid.h>
 
 #include <linux/blkpg.h>
--- nash/nash.c.orig	2008-04-02 23:38:35.000000000 +0300
+++ nash/nash.c	2008-04-02 23:39:52.000000000 +0300
@@ -60,8 +60,6 @@
 
 #include <libdevmapper.h>
 
-#include <selinux/selinux.h>
-
 #include <bdevid.h>
 
 #include <nash.h>
@@ -1724,6 +1722,7 @@ mkdirCommand(char * cmd, char * end)
 static int
 loadPolicyCommand(char * cmd, char *end)
 {
+#if 0
     int enforce = 0;
     int rootfd;
 
@@ -1766,6 +1765,8 @@ loadPolicyCommand(char * cmd, char *end)
 
     close(rootfd);
     return 0;
+#endif
+    return 1;
 }
 
 static int
--- mkinitrd.orig	2008-04-03 00:06:05.000000000 +0300
+++ mkinitrd	2008-04-03 00:07:11.000000000 +0300
@@ -1866,7 +1866,8 @@ if [ -n "$dhclient_leases_cmd" ]; then
     emit "$dhclient_leases_cmd"
 fi
 
-emit "loadpolicy"
+# didn't compile nash with selinux, loadpolicy wouldn't work
+#emit "loadpolicy"
 emit "echo Switching to new root and running init."
 emit "switchroot"
 emit "echo Booting has failed."
