--- nash/block.c.orig	2006-10-30 13:05:29.000000000 -0500
+++ nash/block.c	2006-10-30 13:06:37.000000000 -0500
@@ -28,7 +28,6 @@
 #include <argz.h>
 #include <envz.h>
 
-#include <selinux/selinux.h>
 #include <blkid/blkid.h>
 
 #include <linux/blkpg.h>
--- nash/nash.c.orig	2008-04-16 13:29:50.000000000 +0300
+++ nash/nash.c	2008-04-16 13:31:56.000000000 +0300
@@ -60,8 +60,6 @@
 
 #include <libdevmapper.h>
 
-#include <selinux/selinux.h>
-
 /* for showElfInterpCommand */
 #include <gelf.h>
 #include <libelf.h>
@@ -1730,6 +1728,7 @@ mkdirCommand(char * cmd, char * end)
 static int
 loadPolicyCommand(char * cmd, char *end)
 {
+#if 0
     int enforce = 0;
     int rootfd;
 
@@ -1772,6 +1771,8 @@ loadPolicyCommand(char * cmd, char *end)
 
     close(rootfd);
     return 0;
+#endif
+    return 1;
 }
 
 static int
--- mkinitrd.orig	2008-04-16 13:32:23.000000000 +0300
+++ mkinitrd	2008-04-16 13:32:54.000000000 +0300
@@ -1846,8 +1846,10 @@ if [ -n "$dhclient_leases_cmd" ]; then
     emit "$dhclient_leases_cmd"
 fi
 
-emit "loadpolicy"
-emit "plymouth --newroot=/sysroot"
+# didn't compile nash with selinux, loadpolicy wouldn't work
+#emit "loadpolicy"
+# not using plymouth
+#emit "plymouth --newroot=/sysroot"
 emit "echo Switching to new root and running init."
 emit "switchroot"
 emit "echo Booting has failed."
