--- mkinitrd/mkinitrd-6.0.45/mkinitrd	2008-09-04 11:18:29.000000000 -0500
+++ mkinitrd-6.0.45.fw/mkinitrd	2008-09-04 11:20:24.000000000 -0500
@@ -1423,7 +1423,7 @@ if [ -z "$MNTIMAGE" -o -z "$IMAGE" ]; th
 fi
 
 mkdir -p $MNTIMAGE
-mkdir -p $MNTIMAGE/firmware
+mkdir -p $MNTIMAGE/lib/firmware
 mkdir -p $MNTIMAGE/lib/modules/$kernel
 mkdir -p $MNTIMAGE/bin
 mkdir -p $MNTIMAGE/etc
@@ -1459,7 +1459,7 @@ installmodule()
     fi
     for fw in $(/sbin/modinfo -F firmware $MODULE 2>/dev/null); do
         if [ -f /lib/firmware/$fw ]; then
-            inst "/lib/firmware/$fw" "$MNTIMAGE" "/firmware/$fw"
+            inst "/lib/firmware/$fw" "$MNTIMAGE" "/lib/firmware/$fw"
         fi
     done
 }
