--- mkinitrd.orig	2009-02-05 11:00:42.000000000 -0500
+++ mkinitrd	2009-02-05 11:01:50.000000000 -0500
@@ -1868,6 +1868,25 @@
     emit "losetup /dev/loop0 /tmpmount/$looppath"
 fi
 
+/sbin/libata-migrate-links
+for i in /boot/grub/grub.conf /etc/bootloader.conf /etc/fstab \
+         /etc/smartd.conf /boot/grub/device.map /etc/mtools.conf $RCFILE
+do  
+    [ -f $i ] && DEVSUB="$DEVSUB $i" 
+done
+rm -f /etc/sysconfig/devsub # or it may not modify $RCFILE
+[ -n "$DEVSUB" ] && /usr/bin/devsub /etc/sysconfig $DEVSUB
+rm -f /etc/blkid.tab /etc/lvm/cache/.cache
+
+inst /bin/ash.static "$MNTIMAGE" /bin/ash
+ln -s ash "$MNTIMAGE"/bin/bash
+for i in /sbin/libata-migrate-links /bin/ln /bin/mkdir /bin/basename \
+         /bin/cat /bin/sed /sbin/*.libata-migrate
+do
+    inst $i "$MNTIMAGE"
+done
+emit "/sbin/libata-migrate-links"
+
 emit "echo Creating root device."
 # mkrootdev does "echo /dev/root /sysroot ext3 defaults,ro 0 0 >/etc/fstab"
 emit "mkrootdev -t $rootfs -o $rootopts $rootdev"
