--- mkinitrd-6.0.71/mkinitrd	2010-05-26 17:17:30.000000000 -0400
+++ mkinitrd-6.0.71/mkinitrd.new	2010-05-26 17:17:17.000000000 -0400
@@ -1107,6 +1107,14 @@
     findmodule -dasd_fba_mod
 fi
 
+# Only forceload libata modules if we have libata-migrate-links
+if [ -x /sbin/libata-migrate-links -a -f /lib/modules/$kernel/modules.libata ]; then
+    MODS="$MODS $(cat /lib/modules/$kernel/modules.libata |sed -e 's/.ko//')"
+    for m in $MODS; do
+        findmodule $m
+    done
+fi
+
 # Loopback root support
 # loopdev : device or nfs server:path file is on
 # looppath : filename
