--- mkinitrd/mkinitrd-6.0.45/mkinitrd	2008-05-06 15:48:37.000000000 -0500
+++ mkinitrd-6.0.45.pcifilter/mkinitrd	2008-05-06 16:00:42.000000000 -0500
@@ -275,6 +275,7 @@
     DSO_DEPS="${FILES[@]}"
 }
 
+stabalize_scsi="no"
 scsi_wait_scan="no"
 findmodule() {
     skiperrors=""
@@ -1747,8 +1748,14 @@
         echo "options $module $options" >> $MNTIMAGE/etc/modprobe.conf
     fi
 
-    emit "echo \"Loading $module module\""
-    emit "modprobe -q $module"
+    if [ "${module::5}" == "pata_" -o "$module" == "ata_piix" -o "$module" == "ahci" -o "${module::5}" == "sata_" -o "$module" == "ibmvscsic" ]; then
+        stabalize_scsi="yes"
+        emit "echo \"Loading $module module\""
+        emit "modprobe --pci-filter -q $module"
+    else
+        emit "echo \"Loading $module module\""
+        emit "modprobe -q $module"
+    fi
         
     # Hack - we need a delay after loading usb-storage to give things
     #        time to settle down before we start looking a block devices
@@ -1783,10 +1790,6 @@
 EOF
         done
     fi
-    if [ "${module::5}" == "pata_" -o "$module" == "ata_piix" -o "$module" == "ahci" -o "${module::5}" == "sata_" -o "$module" == "ibmvscsic" ]; then
-        emit "echo Waiting for driver initialization."
-        emit "stabilized --hash --interval 250 /proc/scsi/scsi"
-    fi
 done
 unset usb_mounted
 
@@ -1812,6 +1815,11 @@
 
 emit_iscsi
 
+if [ "$stabalize_scsi" == "yes" ]; then
+        emit "echo Waiting for driver initialization."
+        emit "stabilized --hash --interval 250 /proc/scsi/scsi"
+fi
+
 if [ "$scsi_wait_scan" == "yes" ]; then
     emit "modprobe scsi_wait_scan"
     emit "rmmod scsi_wait_scan"
