--- mkinitrd.orig	2007-12-27 12:58:21.000000000 +0200
+++ mkinitrd	2007-12-27 15:57:42.000000000 +0200
@@ -258,7 +258,7 @@
     fi
 
     # only need to add each module once
-    [[ "$(eval echo \${$modlist})" =~ "$modName" ]] || \
+    [[ "$(eval echo \${$modlist})" =~ $modName ]] || \
         eval $modlist=\"\${$modlist} $modName\"
 
     # need to handle prescsimods here -- they need to go _after_ scsi_mod
@@ -331,7 +331,7 @@
         sysfs=$(readlink ${sysfs%/*})
     fi
 
-    if [[ ! "$sysfs" =~ '^/sys/devices/.*/block/.*$' ]]; then
+    if [[ ! "$sysfs" =~ ^/sys/devices/.*/block/.*$ ]]; then
         error "WARNING: $sysfs is a not a block sysfs path, skipping"
         return
     fi
@@ -342,12 +342,12 @@
         *) handleddevices="$handleddevices $sysfs" ;;
     esac
 
-    if [[ "$sysfs" =~ '^/sys/devices/virtual/block/md[0-9]+$' ]]; then
+    if [[ "$sysfs" =~ ^/sys/devices/virtual/block/md[0-9]+$ ]]; then
         local raid=${sysfs##*/}
         vecho "Found MDRAID component $raid"
         handleraid $raid
     fi
-    if [[ "$sysfs" =~ '^/sys/devices/virtual/block/dm-[0-9]+$' ]]; then
+    if [[ "$sysfs" =~ ^/sys/devices/virtual/block/dm-[0-9]+$ ]]; then
         vecho "Found DeviceMapper component ${sysfs##*/}"
         handledm $(cat $sysfs/dev |cut -d : -f 1) $(cat $sysfs/dev |cut -d : -f 2)
     fi
@@ -1164,7 +1164,7 @@
 # loopfs : filesystem of loopdev
 # loopots : options to mount loopfs
 
-if [ -n "${loopfs}" ] || [[ "$rootopts" =~ "loop" ]]; then
+if [ -n "${loopfs}" ] || [[ "$rootopts" =~ loop ]]; then
     	# FIXME: probe this somehow?
 	
 	rootdev=/dev/loop0
@@ -1224,7 +1224,7 @@
                     # requires real parsing... 
                     # XXX also covered by #132001
                     for mod in $TABLE ; do 
-                        DMMODS="$DMMODS $([[ "$mod" =~ "[[:alpha:]]" ]] && echo "$mod")"
+                        DMMODS="$DMMODS $([[ "$mod" =~ [[:alpha:]] ]] && echo "$mod")"
                     done
                     DMDEVS="$DMDEVS $NAME"
                     ;;
@@ -1451,12 +1451,12 @@
 
         # we mount usbfs before the first module *after* the HCDs
         if [ "$usb_mounted" == "prep" ]; then
-            if [[ "$module" =~ ".hci[_-]hcd" ]]; then
+            if [[ "$module" =~ .hci[_-]hcd ]]; then
                 usb_mounted="no"
                 found_hcd_module="yes"
             fi
         elif [ "$usb_mounted" == "no" ]; then
-            if [[ ! "$module" =~ ".hci[_-]hcd" ]]; then
+            if [[ ! "$module" =~ .hci[_-]hcd ]]; then
                 usb_mounted=yes
                 emit "mount -t usbfs /proc/bus/usb /proc/bus/usb"
             fi
