--- mkinitrd-6.0.45/mkinitrd.old	2009-11-03 11:22:56.000000000 -0500
+++ mkinitrd-6.0.45/mkinitrd	2009-12-15 15:29:03.000000000 -0500
@@ -528,6 +528,9 @@
                     ;;
                 *)
                     slavestried="$slavestried $slavename"
+                    if [ -L $slave ]; then
+                        slave=$(readlink $slave)
+                    fi
                     qpushd $slave
                     findstoragedriverinsys
                     qpopd
