diff -u --recursive mkinitrd-6.0.87.orig/mkinitrd mkinitrd-6.0.87/mkinitrd
--- mkinitrd-6.0.87.orig/mkinitrd	2009-08-25 11:48:00.794259000 +0100
+++ mkinitrd-6.0.87/mkinitrd	2009-08-25 11:52:37.012005042 +0100
@@ -66,6 +66,7 @@
 ncryptolvs=0
 ncryptoraids=0
 root=""
+stabalize_scsi="no"
 scsi_wait_scan="no"
 
 NET_LIST=""
@@ -1377,8 +1378,14 @@
             echo "options $module $options" >> $MNTIMAGE/etc/modprobe.conf
         fi
 
-        emit "echo \"Loading $module module\""
-        emit "modprobe -q $module"
+#        if [ "${module::5}" == "pata_" -o "$module" == "ata_piix" -o "$module" == "ahci" -o "${module::5}" == "sata_" -o "$module" == "ibmvscsic" ]; then
+#            stabalize_scsi="yes"
+#            emit "echo \"Loading $module module\""
+#            emit "modprobe --pci-filter -q $module"
+#        else
+#            emit "echo \"Loading $module module\""
+#            emit "modprobe -q $module"
+#        fi
 
         # Hack - we need a delay after loading usb-storage to give things
         #        time to settle down before we start looking a block devices
@@ -1609,6 +1616,11 @@
 
 emit_iscsi
 
+if [ "$stabalize_scsi" == "yes" ]; then
+        emit "echo Waiting for driver initialization."
+        emit "stabilized --hash --interval 250 /proc/scsi/scsi"
+fi
+
 if [ "$scsi_wait_scan" == "yes" ]; then
     emit "modprobe scsi_wait_scan"
     emit "rmmod scsi_wait_scan"
