--- mkinitrd	2009-03-16 18:41:42.000000000 -0400
+++ mkinitrd	2009-03-16 18:45:21.000000000 -0400
@@ -67,6 +67,7 @@
 ncryptoraids=0
 root=""
 scsi_wait_scan="no"
+usb_storage="no"
 
 NET_LIST=""
 LD_SO_CONF=/etc/ld.so.conf
@@ -385,6 +386,16 @@
     # only need to add each module once
     [[ "$MODULES" =~ $modName ]] || MODULES="$MODULES $modName"
 
+    # add usb storage module if we're including usb support
+    if [    "$modName" = "uhci-hcd" \
+         -o "$modName" = "ohci-hcd" \
+         -o "$modName" = "ehci-hcd" ]; then
+        locatemodule usb-storage
+        if [ -n "$fmPath" -a -f "$fmPath" ]; then
+            usb_storage="yes"
+        fi
+    fi
+
     # need to handle prescsimods here -- they need to go _after_ scsi_mod
     if [ "$modName" = "scsi_mod" ]; then
         for n in $PRESCSIMODS ; do
@@ -1717,6 +1717,11 @@
     installmodule scsi_wait_scan
 fi
 
+if [ "$usb_storage" == "yes" ]; then
+    vecho "Adding module usb-storage"
+    installmodule usb-storage
+fi
+
 /sbin/depmod -a -b $MNTIMAGE $kernel
 if [ $? -ne 0 ]; then
     error "\"/sbin/depmod -a $kernel\" failed."
