--- mkinitrd.orig	2007-12-27 12:58:21.000000000 +0200
+++ mkinitrd	2007-12-27 15:57:42.000000000 +0200
@@ -350,8 +350,8 @@
     fi
 
     # only need to add each module once
-    [[ "$(eval echo \${$modlist})" =~ "$modName" ]] || \
+    [[ "$(eval echo \${$modlist})" =~ $modName ]] || \
         eval $modlist=\"\${$modlist} $modName\"
 
     if [ -f /etc/mkinitrd-firmware.d/$modName ] ; then
         FIRMWARE=" $(cat /etc/mkinitrd-firmware.d/$modName)"
@@ -522,7 +522,7 @@
                 continue ;;
             *) handleddevices="$handleddevices $device" ;;
         esac
-        if [[ "$device" =~ "^md[0-9]+" ]]; then
+        if [[ "$device" =~ ^md[0-9]+ ]]; then
             vecho "Found RAID component $device"
             handleraid "$device"
             continue
@@ -1164,7 +1164,7 @@
 # loopfs : filesystem of loopdev
 # loopots : options to mount loopfs
 
-if [ -n "${loopfs}" ] || [[ "$rootopts" =~ "loop" ]]; then
+if [ -n "${loopfs}" ] || [[ "$rootopts" =~ loop ]]; then
     	# FIXME: probe this somehow?
 	
 	rootdev=/dev/loop0
@@ -1224,7 +1224,7 @@
                     # requires real parsing... 
                     # XXX also covered by #132001
                     for mod in $TABLE ; do 
-                        DMMODS="$DMMODS $([[ "$mod" =~ "[[:alpha:]]" ]] && echo "$mod")"
+                        DMMODS="$DMMODS $([[ "$mod" =~ [[:alpha:]] ]] && echo "$mod")"
                     done
                     DMDEVS="$DMDEVS $NAME"
                     ;;
@@ -1478,12 +1478,12 @@
 
         # we mount usbfs before the first module *after* the HCDs
         if [ "$usb_mounted" == "prep" ]; then
-            if [[ "$module" =~ ".hci[_-]hcd" ]]; then
+            if [[ "$module" =~ .hci[_-]hcd ]]; then
                 usb_mounted="no"
                 found_hcd_module="yes"
             fi
         elif [ "$usb_mounted" == "no" ]; then
-            if [[ ! "$module" =~ ".hci[_-]hcd" ]]; then
+            if [[ ! "$module" =~ .hci[_-]hcd ]]; then
                 usb_mounted=yes
                 emit "mount -t usbfs /proc/bus/usb /proc/bus/usb"
         fi
