--- mkinitrd.orig	2008-03-31 16:47:40.000000000 +0300
+++ mkinitrd	2008-03-31 16:48:24.000000000 +0300
@@ -547,6 +547,9 @@ findstoragedriver () {
             sysfs=$(finddevnoinsys $majmin)
             handledm $(echo "$majmin" |cut -d : -f 1) $(echo "$majmin" |cut -d : -f 2)
         else
+            if [[ "$device" =~ ^/sys ]]; then
+                device=${device##*/}
+            fi
             sysfs=$(findone -type d /sys/block -name $device)
         fi
         [ -z "$sysfs" ] && return
