--- mkinitrd.orig	2007-11-02 20:52:05.000000000 +0200
+++ mkinitrd	2007-11-02 20:53:28.000000000 +0200
@@ -386,7 +386,8 @@ inst() {
     indent_chars="${indent_chars}  "
     indent=${indent_chars:2}
 
-    mkdir -p "$root/$(dirname $dest)"
+    local destdir=$(dirname "$dest")
+    mkdir -p "$root/$destdir"
 
     local RET=0
     local target=""
@@ -1337,6 +1338,20 @@ for i in 0 1 2 3 ; do
     emit "mknod /dev/ttyS$i c 4 $(($i + 64))"
 done
 
+if [ -f /usr/sbin/splashy ] ; then
+    emit "mknod /dev/fb0 c 29 0"
+    emit "mkdir /dev/.initramfs"
+    emit "/usr/sbin/splashy boot"
+
+    inst /usr/sbin/splashy "$MNTIMAGE"
+    inst /etc/splashy/config.xml "$MNTIMAGE"
+
+    SPLASHY_THEMES_DIR="$(splashy_config --get-key /splashy/themes)"
+    SPLASHY_CURRENT_THEME="$(splashy_config --get-key /splashy/current_theme)"
+
+    inst "$SPLASHY_THEMES_DIR/$SPLASHY_CURRENT_THEME" "$MNTIMAGE"
+fi
+
 emit "echo Setting up hotplug."
 emit "hotplug"
 
