--- mkinitrd-6.0.9/mkinitrd.orig	2007-07-23 14:53:50.000000000 -0400
+++ mkinitrd-6.0.9/mkinitrd	2007-07-28 01:07:50.000000000 -0400
@@ -67,6 +67,7 @@
 withusb=1
 rc=0
 dynamic=""
+usplash=1
 
 IMAGESIZE=8000
 PRESCSIMODS="sd_mod"
@@ -108,7 +109,8 @@
     $cmd "       [--image-version] [--force-raid-probe | --omit-raid-modules]"
     $cmd "       [--with=<module>] [--force-lvm-probe | --omit-lvm-modules]"
     $cmd "       [--builtin=<module>] [--omit-dmraid] [--net-dev=<interface>]"
-    $cmd "       [--fstab=<fstab>] [--nocompress] <initrd-image> <kernel-version>"
+    $cmd "       [--fstab=<fstab>] [--nocompress] [--nousplash] <initrd-image>"
+    $cmd "       <kernel-version>"
     $cmd ""
     $cmd "       (ex: `basename $0` /boot/initrd-2.2.5-15.img 2.2.5-15)"
 
@@ -893,6 +897,9 @@
                 shift
             fi
 	    ;;
+        --nousplash)
+            usplash=0
+            ;;
         --help)
             usage -n
             ;;
@@ -1335,6 +1342,24 @@
     emit "mknod /dev/ttyS$i c 4 $(($i + 64))"
 done
 
+if [ $usplash -eq 1 ] && [ -f /sbin/usplash ] ; then
+    emit "mknod /dev/fb0 c 29 0"
+    emit "mkdir /dev/.initramfs"
+    emit "usplash-start"
+
+    inst /sbin/usplash "$MNTIMAGE"
+    inst /sbin/usplash_write "$MNTIMAGE"
+    inst /sbin/usplash-start "$MNTIMAGE"
+    if [ -d /usr/lib64 ] ; then
+        inst /usr/lib64/usplash/usplash-artwork.so "$MNTIMAGE"
+    else
+        inst /usr/lib/usplash/usplash-artwork.so "$MNTIMAGE"
+    fi
+    inst /bin/sh "$MNTIMAGE"
+    inst /bin/sleep "$MNTIMAGE"
+    inst /bin/cat "$MNTIMAGE"
+fi
+
 emit "echo Setting up hotplug."
 emit "hotplug"
 
