--- nash/util.h.jau	2008-01-09 12:55:54.000000000 -0500
+++ nash/util.h	2008-01-09 12:58:37.000000000 -0500
@@ -439,6 +439,12 @@
     __attribute__((format(printf, 1, 2)));
 #endif
 
+#define save_errno(tmp, unsafe_code) ({     \
+        (tmp) = errno ;                     \
+        unsafe_code ;                       \
+        errno = (tmp) ;                     \
+    })
+
 #include <sys/syscall.h>
 #include <sys/poll.h>
 static inline int nash_ppoll_fallback(struct pollfd *fds, nfds_t nfds,
@@ -475,12 +481,6 @@
         __en;})
 #endif
 
-#define save_errno(tmp, unsafe_code) ({     \
-        (tmp) = errno ;                     \
-        unsafe_code ;                       \
-        errno = (tmp) ;                     \
-    })
-
 #define xfree(x) ({if (x) { free(x); x = NULL; }})
 #define movptr(x, y) ({(x) = (y); (y) = NULL; })
 #define xmovptr(x, y) ({if (!(y) && (x)) movptr((x),(y));})
