diff --git a/midori/main.c b/midori/main.c
index 384ed30..9d6c842 100644
--- a/midori/main.c
+++ b/midori/main.c
@@ -886,6 +886,7 @@ midori_soup_session_prepare (SoupSession*       session,
     {
         "/etc/pki/tls/certs/ca-bundle.crt",
         "/etc/ssl/certs/ca-certificates.crt",
+        "/etc/ssl/certs/ca-bundle.crt",
         "/usr/local/share/certs/ca-root-nss.crt",
         NULL
     };
