diff --git a/midori/main.c b/midori/main.c
index 9c84c31..dead395 100644
--- a/midori/main.c
+++ b/midori/main.c
@@ -1655,7 +1655,7 @@ speeddial_new_from_file (const gchar* config,
         gchar* val;
         gchar** values = g_strsplit (parts[i], "\"", -1);
 
-        if (*values[1])
+        if (values[1] && *values[1])
         {
             if (!g_strcmp0 (values[1], "shortcuts"))
             {
@@ -1734,12 +1734,12 @@ speeddial_new_from_file (const gchar* config,
                 columns = atoi (val);
             }
 
-            i++;
 
             g_free (key);
             g_free (val);
-            g_strfreev (values);
         }
+        i++;
+        g_strfreev (values);
     }
 
     /* Default to 3 x 3 grid. Calculate rows here, columns is initialized as 3 */
