commit 2b2df20dd087be8d6719bc93dcefdfa37a22abb4
Author: Paweł Forysiuk <tuxator@o2.pl>
Date:   Tue May 3 14:06:43 2011 +0200

    Use proper markup for empty dials

diff --git a/midori/midori-view.c b/midori/midori-view.c
index e2df2a0..789ca10 100644
--- a/midori/midori-view.c
+++ b/midori/midori-view.c
@@ -3970,7 +3970,8 @@ prepare_speed_dial_html (MidoriView* view)
         else
             position = g_strdup ("");
 
-        if (g_key_file_has_group (key_file, dial_entry))
+        if (g_key_file_has_group (key_file, dial_entry)
+        && g_strcmp0 (g_key_file_get_string (key_file, dial_entry, "uri", NULL), "#"))
         {
             gchar* slot_id = g_strdup_printf ("s%d", slot);
             gchar* uri = g_key_file_get_string (key_file, dial_entry, "uri", NULL);
