--- xscreensaver-4.18/driver/XScreenSaver.ad.in.change-defaults	2004-08-15 01:10:03.000000000 -0400
+++ xscreensaver-4.18/driver/XScreenSaver.ad.in	2005-01-04 19:11:05.011379792 -0500
@@ -36,25 +36,25 @@
 *dpmsStandby:		2:00:00
 *dpmsSuspend:		2:00:00
 *dpmsOff:		4:00:00
-*grabDesktopImages:	True
+*grabDesktopImages:	False
 *grabVideoFrames:	False
-*chooseRandomImages:	False
-*imageDirectory:	
+*chooseRandomImages:	True
+*imageDirectory:        @GNOME_DATADIR@/backgrounds/images
 *nice:			10
 *memoryLimit:		0
-*lock:			False
-*lockVTs:		True
+*lock:			True
+*lockVTs:		False
 *verbose:		False
 *timestamp:		True
 *fade:			True
 *unfade:		False
 *fadeSeconds:		0:00:03
 *fadeTicks:		20
-*splash:		True
+*splash:		False
 *splashDuration:	0:00:05
 *visualID:		default
-*captureStderr: 	True
-*ignoreUninstalledPrograms: False
+*captureStderr: 	False
+*ignoreUninstalledPrograms: True
 
 *overlayTextForeground:	#FFFF00
 *overlayTextBackground:	#000000
@@ -62,14 +62,14 @@
 *font:			*-medium-r-*-140-*-m-*
 
 ! The default is to use these extensions if available (as noted.)
-*sgiSaverExtension:	True
-*mitSaverExtension:	False
-*xidleExtension:	True
-*procInterrupts:	True
+*sgiSaverExtension:	False
+*mitSaverExtension:	True
+*xidleExtension:	False
+*procInterrupts:	False
 
 ! Set this to True if you are experiencing longstanding XFree86 bug #421
 ! (xscreensaver not covering the whole screen)
-GetViewPortIsFullOfLies: False
+GetViewPortIsFullOfLies: True
 
 ! This is what the "Demo" button on the splash screen runs (/bin/sh syntax.)
 *demoCommand: xscreensaver-demo
@@ -82,8 +82,7 @@
 
 ! This is how the "Help" button loads URLs (/bin/sh syntax.)
 ! The "helpURL" will be substituted for up to two occurrences of "%s".
-@GNOMEHELP_N@*loadURL: mozilla '%s' || netscape '%s'
-@GNOMEHELP_Y@*loadURL: gnome-url-show '%s' || gnome-moz-remote --newwin '%s'
+*loadURL: htmlview '%s'
 
 ! This is what the "Manual" button in xscreensaver-demo runs (/bin/sh syntax.)
 @GNOMEHELP_N@*manualCommand: xterm -sb -fg black -bg gray75 -T '%s manual' \
@@ -109,6 +108,7 @@
 !
 *installColormap:	True
 
+*mode:                  blank
 
 ! This is the list of installed screen saver modes.  See "man xscreensaver"
 ! for the syntax used here.
@@ -394,35 +394,35 @@
 ! Resources for the password and splash-screen dialog boxes of
 ! the "xscreensaver" daemon.
 !
-*Dialog.headingFont:		*-times-bold-r-*-*-*-180-*-*-*-iso8859-1
+*Dialog.headingFont:		*-helvetica-bold-r-*-*-*-180-*-*-*-iso8859-1
 *Dialog.bodyFont:		*-helvetica-bold-r-*-*-*-140-*-*-*-iso8859-1
 *Dialog.labelFont:		*-helvetica-bold-r-*-*-*-140-*-*-*-iso8859-1
 *Dialog.buttonFont:		*-helvetica-bold-r-*-*-*-140-*-*-*-iso8859-1
-*Dialog.dateFont:		*-courier-medium-r-*-*-*-80-*-*-*-iso8859-1
+*Dialog.dateFont:		*-clean-medium-r-*-*-*-80-*-*-*-iso8859-1
 *Dialog.foreground:		#000000
-*Dialog.background:		#D6D6D6
+*Dialog.background:		#E6E6E6
 *Dialog.Button.foreground:	#000000
-*Dialog.Button.background:	#EAEAEA
+*Dialog.Button.background:	#F5F5F5
 !*Dialog.Button.background:	#D6D6D6
 !*Dialog.Button.pointBackground: #EAEAEA
 !*Dialog.Button.clickBackground: #C3C3C3
 *Dialog.text.foreground:	#000000
 *Dialog.text.background:	#FFFFFF
-*passwd.thermometer.foreground:	#FF0000
+*passwd.thermometer.foreground:	#4464AC
 *passwd.thermometer.background:	#FFFFFF
 *Dialog.topShadowColor:		#FFFFFF
-*Dialog.bottomShadowColor:	#666666
+*Dialog.bottomShadowColor:	#CECECE
 *Dialog.logo.width:		210
 *Dialog.logo.height:		210
-*Dialog.internalBorderWidth:	30
+*Dialog.internalBorderWidth:	24
 *Dialog.borderWidth:		1
-*Dialog.shadowThickness:	2
+*Dialog.shadowThickness:	1
 
-*passwd.heading.label:		XScreenSaver %s
-*passwd.body.label:		This display is locked.
-*passwd.user.label:		User:
+*passwd.heading.label:		Screen Locked
+*passwd.body.label:		Please enter your password.
+*passwd.user.label:		Username:
 *passwd.passwd.label:		Password:
-*passwd.passwdFont:		*-courier-medium-r-*-*-*-140-*-*-*-iso8859-1
+*passwd.passwdFont:		*-helvetica-medium-r-*-*-*-140-*-*-*-iso8859-1
 *passwd.thermometer.width:	8
 
 *splash.heading.label:		XScreenSaver %s
