--- xscreensaver-4.16/driver/xscreensaver.pam.orig	2004-06-21 15:43:10.498467325 -0400
+++ xscreensaver-4.16/driver/xscreensaver.pam	2004-06-21 15:44:53.503882600 -0400
@@ -1,7 +1,7 @@
 #%PAM-1.0
 
 # Red Hat says this is right for them, as of 7.3:
-auth       required	/lib/security/pam_stack.so service=system-auth
+auth       required	pam_stack.so service=system-auth
 
 # SuSE 9.0 uses this (along with "--with-passwd-helper" of "unix2_chkpwd")
 # auth     required       pam_unix2.so	nullok
