--- xscreensaver-4.18/hacks/piecewise.c.stuff-piecewise-leak	2003-02-18 16:39:20.000000000 -0500
+++ xscreensaver-4.18/hacks/piecewise.c	2004-10-19 12:06:18.119796835 -0400
@@ -370,6 +370,7 @@
   if (!*eq) {
     e->l = e->r = 0;
     *eq = e;
+    return;
     }
 
   event_cut_y = e->y;
@@ -381,6 +382,8 @@
       e->r = 0;             /* doing this instead of dying might be dangerous */
       (*eq)->l = e;           
       }
+    else
+      free(e);
     }
   else if (e->y < (*eq)->y) {
     e->l = (*eq)->l;
