--- xscreensaver-4.16/hacks/glx/glsnake.c	2004-02-09 04:08:27.000000000 -0500
+++ xscreensaver-4.16.new/hacks/glx/glsnake.c	2004-08-06 15:29:22.102854416 -0400
@@ -528,7 +528,7 @@
 	PIN, ZERO, RIGHT, RIGHT, ZERO, PIN, PIN, ZERO, PIN, PIN, ZERO,
 	RIGHT, ZERO }
     },
-    { "k's turd",
+    { "caterpillar",
       { RIGHT, RIGHT, PIN, RIGHT, LEFT, RIGHT, PIN, RIGHT, LEFT,
 	RIGHT, PIN, RIGHT, LEFT, RIGHT, PIN, RIGHT, LEFT, RIGHT, PIN,
 	RIGHT, LEFT, RIGHT, PIN, ZERO }
@@ -543,7 +543,7 @@
 	RIGHT, LEFT, RIGHT, ZERO, LEFT, RIGHT, LEFT, ZERO, RIGHT, LEFT,
 	RIGHT, ZERO, LEFT, ZERO }
     },
-    { "arse gegl",
+    { "gegl",
       { ZERO, PIN, PIN, ZERO, ZERO, ZERO, PIN, PIN, ZERO, LEFT, ZERO,
 	PIN, PIN, ZERO, RIGHT, LEFT, ZERO, PIN, ZERO, PIN, PIN, ZERO,
 	PIN, ZERO }
@@ -553,22 +553,22 @@
 	ZERO, PIN, ZERO, ZERO, ZERO, ZERO, PIN, ZERO, ZERO, ZERO, ZERO,
 	ZERO, ZERO, ZERO }
     },
-    { "kissy box",
+    { "ribbon",
       { PIN, ZERO, ZERO, ZERO, PIN, ZERO, ZERO, ZERO, ZERO, ZERO,
 	ZERO, PIN, ZERO, ZERO, ZERO, ZERO, PIN, ZERO, ZERO, ZERO, ZERO,
 	ZERO, PIN, ZERO }
     },
-    { "erect penis",     /* thanks benno */
+    { "shuffle board",     /* thanks benno */
       { PIN, ZERO, PIN, PIN, ZERO, ZERO, PIN, ZERO, ZERO, ZERO, PIN,
 	PIN, ZERO, ZERO, ZERO, RIGHT, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO,
 	ZERO, ZERO }
     },
-    { "flaccid penis",
+    { "anchor",
       { PIN, ZERO, PIN, PIN, ZERO, ZERO, PIN, ZERO, ZERO, ZERO, PIN,
 	PIN, ZERO, ZERO, ZERO, RIGHT, PIN, ZERO, ZERO, ZERO, ZERO, ZERO,
 	ZERO, ZERO }
     },
-    { "vagina",
+    { "engagement ring",
       { RIGHT, ZERO, ZERO, ZERO, RIGHT, ZERO, ZERO, PIN, ZERO, ZERO,
 	LEFT, ZERO, ZERO, ZERO, LEFT, ZERO, LEFT, PIN, LEFT, PIN, RIGHT,
 	PIN, RIGHT, ZERO }
@@ -984,7 +984,7 @@
     { "Parrot",
         { ZERO, ZERO, ZERO, ZERO, RIGHT, RIGHT, ZERO, LEFT, PIN, RIGHT, ZERO, RIGHT, ZERO, RIGHT, ZERO, RIGHT, PIN, LEFT, ZERO, RIGHT, LEFT, ZERO, PIN }
     },
-    { "Penis",
+    { "Shuttle",
         { PIN, PIN, RIGHT, ZERO, PIN, PIN, ZERO, PIN, ZERO, ZERO, RIGHT, PIN, LEFT, ZERO, ZERO, PIN, ZERO, PIN, PIN, ZERO, LEFT, PIN, PIN }
     },
     { "PictureCommingSoon",
--- xscreensaver-4.16/hacks/barcode.c	2003-07-22 22:25:05.000000000 -0400
+++ xscreensaver-4.16.new/hacks/barcode.c	2004-08-14 12:00:35.779783288 -0400
@@ -122,8 +122,6 @@
   "belly",
   "bliss",
   "bogosity",
-  "boobies",
-  "boobs",
   "booty",
   "bread",
   "bubba",
@@ -136,7 +134,6 @@
   "children",
   "chocolate",
   "CLONE",
-  "cock",
   "constriction",
   "contrition",
   "cop",
@@ -176,7 +173,6 @@
   "fear",
   "fever",
   "filth",
-  "flatulence",
   "fluff",
   "fnord",
   "freedom",
@@ -193,9 +189,7 @@
   "happiness",
   "hate",
   "helplessness",
-  "hemorrhoid",
   "hermaphrodite",
-  "heroin",
   "heroine",
   "hope",
   "hysteria",
@@ -270,7 +264,6 @@
   "punishment",
   "punk rock",
   "punk",
-  "pussy",
   "quagmire",
   "quarantine",
   "quartz",
@@ -279,7 +272,6 @@
   "rage",
   "readout",
   "reality",
-  "rectum",
   "reject",
   "rejection",
   "respect",
@@ -338,8 +330,6 @@
   "vegetarian",
   "venom",
   "verifiability",
-  "viagra",
-  "vibrator",
   "victim",
   "vignette",
   "villainy",
