diff -r 036fa6a75a9f libsoup/soup-socket.c
--- a/libsoup/soup-socket.c	Thu Apr 28 14:04:01 2011 +0200
+++ b/libsoup/soup-socket.c	Thu Apr 28 14:05:58 2011 +0200
@@ -932,7 +932,7 @@
 				       "server-identity", identity,
 				       "use-system-certdb", FALSE,
 				       "require-close-notify", FALSE,
-				       "use-ssl3", TRUE,
+				       "use-ssl3", FALSE,
 				       NULL);
 		g_object_unref (identity);
 
