diff --git a/src/gpk-update-viewer.c b/src/gpk-update-viewer.c
index c07e0d8..be742f8 100644
--- a/src/gpk-update-viewer.c
+++ b/src/gpk-update-viewer.c
@@ -247,7 +247,7 @@ gpk_update_viewer_undisable_packages ()
 	while (valid) {
 		gtk_list_store_set (list_store_updates, &iter,
 				    GPK_UPDATES_COLUMN_SENSITIVE, TRUE,
-				    GPK_UPDATES_COLUMN_CLICKABLE, FALSE,
+				    GPK_UPDATES_COLUMN_CLICKABLE, TRUE,
 				    -1);
 		valid = gtk_tree_model_iter_next (model, &iter);
 	}
@@ -413,8 +413,7 @@ gpk_update_viewer_button_install_cb (GtkWidget *widget, gpointer data)
 
 	/* get packages that also have to be updated */
 #if PK_CHECK_VERSION(0,5,2)
-	//ret = pk_client_simulate_update_packages (client_primary, package_ids, &error);
-    ret =pk_client_update_system(client_primary,FALSE,FALSE);
+	ret = pk_client_simulate_update_packages (client_primary, package_ids, &error);
 #else
 	ret = pk_client_get_depends (client_primary, pk_bitfield_value (PK_FILTER_ENUM_NOT_INSTALLED), package_ids, TRUE, &error);
 #endif
@@ -933,7 +932,7 @@ gpk_update_viewer_treeview_update_toggled (GtkCellRendererToggle *cell, gchar *p
 			    GPK_UPDATES_COLUMN_ID, &package_id, -1);
 
 	/* unstage */
-	update =  1;
+	update ^= 1;
 
 	egg_debug ("update %s[%i]", package_id, update);
 	g_free (package_id);
