diff --git a/src/gpk-update-viewer.c b/src/gpk-update-viewer.c
index be742f8..72c77f0 100644
--- a/src/gpk-update-viewer.c
+++ b/src/gpk-update-viewer.c
@@ -413,7 +413,8 @@ gpk_update_viewer_button_install_cb (GtkWidget *widget, gpointer data)
 
 	/* get packages that also have to be updated */
 #if PK_CHECK_VERSION(0,5,2)
-	ret = pk_client_simulate_update_packages (client_primary, package_ids, &error);
+	//ret = pk_client_simulate_update_packages (client_primary, package_ids, &error);
+    ret =pk_client_update_system(client_primary,FALSE,FALSE);
 #else
 	ret = pk_client_get_depends (client_primary, pk_bitfield_value (PK_FILTER_ENUM_NOT_INSTALLED), package_ids, TRUE, &error);
 #endif
