diff --git a/src/gpk-application.c b/src/gpk-application.c
--- a/src/gpk-application.c
+++ b/src/gpk-application.c
@@ -2061,7 +2061,7 @@
 		ret = pk_client_simulate_remove_packages (application->priv->client_primary, package_ids, &error);
 #else
 		application->priv->dep_check_info_only = FALSE;
-		ret = pk_client_get_requires (application->priv->client_primary, pk_bitfield_value (PK_FILTER_ENUM_INSTALLED), package_ids, TRUE, &error);
+		ret = pk_client_remove_packages (application->priv->client_primary, package_ids, FALSE, FALSE, &error);
 #endif
 		if (!ret) {
 			egg_warning ("failed to get requires: %s", error->message);
