diff --git a/src/gpk-application.c b/src/gpk-application.c
index 9501c62..7a930f2 100644
--- a/src/gpk-application.c
+++ b/src/gpk-application.c
@@ -2046,7 +2046,7 @@ gpk_application_button_apply_cb (GtkWidget *widget, GpkApplication *application)
 
 		/* install */
 #if PK_CHECK_VERSION(0,5,2)
-		ret = pk_client_simulate_install_packages (application->priv->client_primary, package_ids, &error);
+		//ret = pk_client_simulate_install_packages (application->priv->client_primary, package_ids, &error);
 #else
 		application->priv->dep_check_info_only = FALSE;
 		ret = pk_client_get_depends (application->priv->client_primary, pk_bitfield_value (PK_FILTER_ENUM_NOT_INSTALLED), package_ids, TRUE, &error);
@@ -2072,7 +2072,7 @@ gpk_application_button_apply_cb (GtkWidget *widget, GpkApplication *application)
 
 		/* remove */
 #if PK_CHECK_VERSION(0,5,2)
-		ret = pk_client_simulate_remove_packages (application->priv->client_primary, package_ids, &error);
+		//ret = pk_client_simulate_remove_packages (application->priv->client_primary, package_ids, &error);
 #else
 		application->priv->dep_check_info_only = FALSE;
 		ret = pk_client_get_requires (application->priv->client_primary, pk_bitfield_value (PK_FILTER_ENUM_INSTALLED), package_ids, TRUE, &error);
diff --git a/src/gpk-dbus-task.c b/src/gpk-dbus-task.c
index c0c2b93..31e3694 100644
--- a/src/gpk-dbus-task.c
+++ b/src/gpk-dbus-task.c
@@ -560,7 +560,7 @@ gpk_dbus_task_install_package_ids_dep_check (GpkDbusTask *task)
 
 	/* find out if this would drag in other packages */
 #if PK_CHECK_VERSION(0,5,2)
-	ret = pk_client_simulate_install_packages (task->priv->client_primary, task->priv->package_ids, &error_local);
+	//ret = pk_client_simulate_install_packages (task->priv->client_primary, task->priv->package_ids, &error_local);
 #else
 	ret = pk_client_get_depends (task->priv->client_primary, pk_bitfield_value (PK_FILTER_ENUM_NOT_INSTALLED), task->priv->package_ids, TRUE, &error_local);
 #endif
@@ -2047,7 +2047,7 @@ gpk_dbus_task_install_files_dep_check (GpkDbusTask *task)
 
 #if (PK_CHECK_VERSION(0,5,2))
 	/* find out if this would drag in other packages */
-	ret = pk_client_simulate_install_files (task->priv->client_primary, task->priv->files, &error_local);
+	//ret = pk_client_simulate_install_files (task->priv->client_primary, task->priv->files, &error_local);
 	if (!ret) {
 		/* TRANSLATORS: error: could not get the extra package list when installing a package */
 		gpk_dbus_task_error_msg (task, _("Could not work out what packages would be also installed"), error_local);
diff --git a/src/gpk-update-viewer.c b/src/gpk-update-viewer.c
index 73d4cd3..cd55b52 100644
--- a/src/gpk-update-viewer.c
+++ b/src/gpk-update-viewer.c
@@ -433,7 +433,7 @@ gpk_update_viewer_button_install_cb (GtkWidget *widget, gpointer data)
 
 	/* get packages that also have to be updated */
 #if PK_CHECK_VERSION(0,5,2)
-	ret = pk_client_simulate_update_packages (client_primary, package_ids, &error);
+	//et = pk_client_simulate_update_packages (client_primary, package_ids, &error);
 #else
 	ret = pk_client_get_depends (client_primary, pk_bitfield_value (PK_FILTER_ENUM_NOT_INSTALLED), package_ids, TRUE, &error);
 #endif
