diff --git a/src/gpk-application.c b/src/gpk-application.c
index 64e0ad3..fb7943d 100644
--- a/src/gpk-application.c
+++ b/src/gpk-application.c
@@ -585,7 +585,7 @@ gpk_application_menu_run_cb (GtkAction *action, GpkApplication *application)
 	/* only if installed */
 	if (gpk_application_state_installed (state)) {
 		/* run this single package id */
-		array = g_strsplit (package_id, "|", 1);
+		array = g_strsplit (package_id, "%", 1);
 		exec = gpk_client_run_show (array);
 		if (exec != NULL) {
 			ret = g_spawn_command_line_async (exec, &error);
diff --git a/src/gpk-client.c b/src/gpk-client.c
index b6bd614..343590d 100644
--- a/src/gpk-client.c
+++ b/src/gpk-client.c
@@ -1754,7 +1754,7 @@ gpk_client_install_mime_type (GpkClient *gclient, const gchar *mime_type, GError
 	}
 
 	/* install this specific package */
-	package_ids = g_strsplit (package_id, "|", 1);
+	package_ids = g_strsplit (package_id, "%", 1);
 	ret = gpk_client_install_package_ids (gclient, package_ids, error);
 out:
 	if (list != NULL) {
@@ -1822,7 +1822,7 @@ gpk_client_install_font (GpkClient *gclient, const gchar *font_desc, GError **er
 	}
 
 	/* install this specific package */
-	package_ids = g_strsplit (package_id, "|", 1);
+	package_ids = g_strsplit (package_id, "%", 1);
 	ret = gpk_client_install_package_ids (gclient, package_ids, error);
 out:
 	if (list != NULL) {
