diff --git a/src/gpk-application.c b/src/gpk-application.c
--- a/src/gpk-application.c
+++ b/src/gpk-application.c
@@ -2035,7 +2035,7 @@
 		ret = pk_client_simulate_install_packages (application->priv->client_primary, package_ids, &error);
 #else
 		application->priv->dep_check_info_only = FALSE;
-		ret = pk_client_get_depends (application->priv->client_primary, pk_bitfield_value (PK_FILTER_ENUM_NOT_INSTALLED), package_ids, TRUE, &error);
+		ret = pk_client_install_packages (application->priv->client_primary, package_ids, &error);
 #endif
 		if (!ret) {
 			egg_warning ("failed to get depends: %s", error->message);
diff --git a/src/gpk-dbus-task.c b/src/gpk-dbus-task.c
--- a/src/gpk-dbus-task.c
+++ b/src/gpk-dbus-task.c
@@ -562,7 +562,7 @@
 #if PK_CHECK_VERSION(0,5,2)
 	ret = pk_client_simulate_install_packages (task->priv->client_primary, task->priv->package_ids, &error_local);
 #else
-	ret = pk_client_get_depends (task->priv->client_primary, pk_bitfield_value (PK_FILTER_ENUM_NOT_INSTALLED), task->priv->package_ids, TRUE, &error_local);
+	ret = pk_client_install_packages (task->priv->client_primary, task->priv->package_ids, &error_local);
 #endif
 	if (!ret) {
 		/* TRANSLATORS: error: could not get the extra package list when installing a package */
diff --git a/src/gpk-update-viewer.c b/src/gpk-update-viewer.c
--- a/src/gpk-update-viewer.c
+++ b/src/gpk-update-viewer.c
@@ -435,7 +435,7 @@
 #if PK_CHECK_VERSION(0,5,2)
 	ret = pk_client_simulate_update_packages (client_primary, package_ids, &error);
 #else
-	ret = pk_client_get_depends (client_primary, pk_bitfield_value (PK_FILTER_ENUM_NOT_INSTALLED), package_ids, TRUE, &error);
+	ret = pk_client_update_packages (client_primary, package_ids, &error);
 #endif
 	if (!ret) {
 		egg_warning ("cannot get depends for updates: %s", error->message);
