diff --git a/src/gpk-check-update.c b/src/gpk-check-update.c
--- a/src/gpk-check-update.c
+++ b/src/gpk-check-update.c
@@ -1128,7 +1128,7 @@
 		goto out;
 	}
 
-	ret = pk_client_get_distro_upgrades (cupdate->priv->client_primary, &error);
+	ret = pk_client_get_updates (cupdate->priv->client_primary, NULL,  &error);
 	if (!ret) {
 		egg_warning ("cannot get updates: %s", error->message);
 		g_error_free (error);
