diff -r a4c7e9dbfebf configure.ac
--- a/configure.ac	Wed Mar 18 13:56:28 2009 -0600
+++ b/configure.ac	Wed Mar 18 13:57:23 2009 -0600
@@ -32,7 +32,7 @@
 	CPPFLAGS="$CPPFLAGS -DG_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED"
 	CPPFLAGS="$CPPFLAGS -DGDK_MULTIHEAD_SAFE -DGTK_MULTIHEAD_SAFE"
 	CPPFLAGS="$CPPFLAGS -Wall -Wreturn-type"
-	CPPFLAGS="$CPPFLAGS -Werror"
+#	CPPFLAGS="$CPPFLAGS -Werror"
 #	CPPFLAGS="$CPPFLAGS -Wextra"
 	CPPFLAGS="$CPPFLAGS -Wdeclaration-after-statement"
 #	CPPFLAGS="$CPPFLAGS -Wmissing-declarations"
