--- prefdm.orig	2009-12-18 18:43:19.593767923 +0000
+++ prefdm	2009-12-18 18:44:05.330855012 +0000
@@ -43,15 +43,15 @@
 done
 
 
-[ -n "$preferred" ] && exec $preferred "$@" >/dev/null 2>&1 </dev/null
+[ -n "$preferred" ] && exec ck-launch-session $preferred "$@" >/dev/null 2>&1 </dev/null
 
 splash_quit
 
 # Fallbacks, in order
-exec gdm "$@" >/dev/null 2>&1 </dev/null
-exec kdm "$@" >/dev/null 2>&1 </dev/null
-exec wdm "$@" >/dev/null 2>&1 </dev/null
-exec xdm "$@" >/dev/null 2>&1 </dev/null
+exec ck-launch-session gdm "$@" >/dev/null 2>&1 </dev/null
+exec ck-launch-session kdm "$@" >/dev/null 2>&1 </dev/null
+exec ck-launch-session wdm "$@" >/dev/null 2>&1 </dev/null
+exec ck-launch-session xdm "$@" >/dev/null 2>&1 </dev/null
 # catch all exit error
 exit 1
 
