--- initscripts-7.53/prefdm.fast	2004-03-16 23:26:00.000000000 -0500
+++ initscripts-7.53/prefdm	2004-09-23 17:14:42.000000000 -0400
@@ -34,6 +34,11 @@
 	fi
 fi
 
+
+while ! [ -f /var/lock/subsys/prefdm ] ; do
+        sleep 0.25
+done
+
 if [ -n "$preferred" ] && which $preferred >/dev/null 2>&1; then 
 	exec `which $preferred` $* >/dev/null 2>&1
 fi
--- initscripts-7.53/inittab.fast	2003-01-24 12:36:32.000000000 -0500
+++ initscripts-7.53/inittab	2004-09-23 16:56:43.000000000 -0400
@@ -25,7 +25,7 @@
 l2:2:wait:/etc/rc.d/rc 2
 l3:3:wait:/etc/rc.d/rc 3
 l4:4:wait:/etc/rc.d/rc 4
-l5:5:wait:/etc/rc.d/rc 5
+l5:5:once:/etc/rc.d/rc 5
 l6:6:wait:/etc/rc.d/rc 6
 
 # Trap CTRL-ALT-DELETE
@@ -42,7 +42,7 @@
 
 
 # Run gettys in standard runlevels
-1:2345:respawn:/sbin/mingetty tty1
+1:234:respawn:/sbin/mingetty tty1
 2:2345:respawn:/sbin/mingetty tty2
 3:2345:respawn:/sbin/mingetty tty3
 4:2345:respawn:/sbin/mingetty tty4
