diff -ruN initscripts-8.11.1/inittab initscripts-8.11.1.new/inittab
--- initscripts-8.11.1/inittab	2005-04-15 13:44:24.000000000 -0400
+++ initscripts-8.11.1.new/inittab	2005-07-11 17:14:52.000000000 -0400
@@ -25,7 +25,7 @@
 l2:2:wait:/etc/rc.d/rc 2
 l3:3:wait:/etc/rc.d/rc 3
 l4:4:wait:/etc/rc.d/rc 4
-l5:5:wait:/etc/rc.d/rc 5
+l5:5:once:/etc/rc.d/rc 5
 l6:6:wait:/etc/rc.d/rc 6
 
 # Trap CTRL-ALT-DELETE
diff -ruN initscripts-8.11.1/prefdm initscripts-8.11.1.new/prefdm
--- initscripts-8.11.1/prefdm	2005-04-15 13:48:44.000000000 -0400
+++ initscripts-8.11.1.new/prefdm	2005-07-11 17:17:10.000000000 -0400
@@ -37,6 +37,11 @@
 	pidof xdm >/dev/null 2>&1 && exit 0
 fi
 
+while ! [ -f /var/lock/subsys/prefdm ] ; do
+         # some locales want 0,25 instead
+         LANG=C sleep 0.25
+done
+
 if [ -n "$preferred" ]; then
 	$preferred "$@" >/dev/null 2>&1
 	if [ $? -ne 127 ]; then
