diff -ruN initscripts-8.11.1/rc.d/init.d/functions initscripts-8.11.1.new/rc.d/init.d/functions
--- initscripts-8.11.1/rc.d/init.d/functions	2005-04-29 11:51:48.000000000 -0400
+++ initscripts-8.11.1.new/rc.d/init.d/functions	2005-07-12 11:05:10.000000000 -0400
@@ -183,8 +183,8 @@
 		done
 	fi
 	if [ -z "$pid" ]; then
-		pid=`pidof -c -o $$ -o $PPID -o %PPID -x $1 || \
-			pidof -c -o $$ -o $PPID -o %PPID -x $base`
+		pid=`pidof -o $$ -o $PPID -o %PPID -x $1 || \
+			pidof -o $$ -o $PPID -o %PPID -x $base`
 	fi
 
         # Kill it.
@@ -272,8 +272,8 @@
 	                return 0
 	        fi
 	fi
-	pidof -c -o $$ -o $PPID -o %PPID -x $1 || \
-		pidof -c -o $$ -o $PPID -o %PPID -x $base
+	pidof -o $$ -o $PPID -o %PPID -x $1 || \
+		pidof -o $$ -o $PPID -o %PPID -x $base
 }
 
 status() {
