diff -ruN initscripts.foresight/initscripts-8.76.2/rc.d/rc.sysinit initscripts-8.76.2.mdfix/rc.d/rc.sysinit
--- initscripts.foresight/initscripts-8.76.2/rc.d/rc.sysinit	2008-07-28 11:45:17.000000000 -0500
+++ initscripts-8.76.2.mdfix/rc.d/rc.sysinit	2008-07-28 11:54:35.000000000 -0500
@@ -335,6 +335,13 @@
 splash_progress 7
 action $"Setting hostname ${HOSTNAME}: " hostname ${HOSTNAME}
 
+# RAID setup
+update_boot_stage RCraid
+splash_progress 8
+if [ -f /etc/mdadm.conf ]; then
+    /sbin/mdadm -A -s --auto=yes
+fi
+
 # Device mapper & related initialization
 if ! __fgrep "device-mapper" /proc/devices >/dev/null 2>&1 ; then
 	modprobe dm-mod >/dev/null 2>&1
