--- initscripts-7.53/rc.d/rc.sysinit.old	2004-09-04 19:59:11.000000000 -0400
+++ initscripts-7.53/rc.d/rc.sysinit	2004-09-04 20:01:05.000000000 -0400
@@ -687,7 +687,6 @@
 	   rm -f $afile
 	fi
 done
-rm -f /var/lib/rpm/__db*
 
 # Reset pam_console permissions
 [ -x /sbin/pam_console_apply ] && /sbin/pam_console_apply -r
@@ -817,12 +816,6 @@
   fi
 fi
 
-# Generate a header that defines the boot kernel.
-# remove after Fedora Core 2
-if [ "${version[0]}" -lt "3" -a "${version[1]}" -lt "6" ]; then
- /sbin/mkkerneldoth
-fi
-
 # Now that we have all of our basic modules loaded and the kernel going,
 # let's dump the syslog ring somewhere so we can find it later
 dmesg -s 131072 > /var/log/dmesg
