diff -ur initscripts-8.33-old/rc.d/init.d/halt initscripts-8.33/rc.d/init.d/halt
--- initscripts-8.33-old/rc.d/init.d/halt	2006-03-03 14:05:42.000000000 -0500
+++ initscripts-8.33/rc.d/init.d/halt	2006-05-17 10:44:51.000000000 -0400
@@ -23,7 +23,7 @@
 }
 
 halt_get_remaining() {
-	awk '$2 ~ /^\/$|^\/proc|^\/dev/{next}
+	awk '$2 ~ /^\/$|^\/proc|^\/initrd|^\/dev/{next}
 	     $3 == "tmpfs" || $3 == "proc" {print $2 ; next}
 	     /(^#|loopfs|autofs|sysfs|^none|^\/dev\/ram|^\/dev\/root)/ {next}
 	     {print $2}' /proc/mounts
@@ -133,8 +133,8 @@
 
 # Unmount file systems, killing processes if we have to.
 # Unmount loopback stuff first
-remaining=`awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $2}' /proc/mounts`
-devremaining=`awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $1}' /proc/mounts`
+remaining=`awk '$1 ~ /^\/dev\/loop/ && $2 != "/" && $2 !~ /^\/initrd/ {print $2}' /proc/mounts`
+devremaining=`awk '$1 ~ /^\/dev\/loop/ && $2 != "/" && $2 !~ /^\/initrd/ {print $1}' /proc/mounts`
 [ -n "$remaining" ] && {
 	sig=
 	retry=3
@@ -149,8 +149,8 @@
 			losetup $dev > /dev/null 2>&1 && \
 				runcmd $"Detaching loopback device $dev: " losetup -d $dev
 		done
-		remaining=`awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $2}' /proc/mounts`
-		devremaining=`awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $1}' /proc/mounts`
+                remaining=`awk '$1 ~ /^\/dev\/loop/ && $2 != "/" && $2 !~ /^\/initrd/ {print $2}' /proc/mounts`
+                devremaining=`awk '$1 ~ /^\/dev\/loop/ && $2 != "/" && $2 !~ /^\/initrd/ {print $1}' /proc/mounts`
 		[ -z "$remaining" ] && break
 		/sbin/fuser -k -m $sig $remaining >/dev/null
 		sleep 5
diff -ur initscripts-8.33-old/rc.d/init.d/netfs initscripts-8.33/rc.d/init.d/netfs
--- initscripts-8.33-old/rc.d/init.d/netfs	2006-05-17 10:36:46.000000000 -0400
+++ initscripts-8.33/rc.d/init.d/netfs	2006-05-17 10:37:52.000000000 -0400
@@ -88,9 +88,20 @@
 	action $"Mounting other filesystems: " mount -a -t nonfs,nfs4,smbfs,cifs,ncpfs,gfs
 	;;
   stop)
-        # Unmount loopback stuff first
-	remaining=`LC_ALL=C awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $2}' /proc/mounts`
-	devremaining=`LC_ALL=C awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $1}' /proc/mounts`
+        # Unmount loopback mounts of anonymous devices, which may be mounted
+        # over files contained on network filesystems.
+        network_mounted(){
+            loopback_over_network=""
+            mounted="$(LC_ALL=C awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $1}' /proc/mounts)"
+            for loopdevice in $mounted; do
+                device_type="$(losetup $loopdevice | sed -e 's/.*\[\(..\)..\].*/\1/')"
+                [ "$device_type" == "00" ] && loopback_over_network="$loopback_over_network $loopdevice"
+            done
+            echo $loopback_over_network
+        }
+
+        remaining=$(network_mounted)
+	devremaining=$(for dev in $(network_mounted);do cat /proc/mounts | fgrep $dev | awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $2}';done)
 	[ -n "$remaining" ] && {
 	        sig=
 		retry=3
@@ -105,8 +116,8 @@
 				losetup $dev >/dev/null 2>&1 && \
 				action $"Detaching loopback device $dev: " losetup -d $dev
 			done	
-			remaining=`LC_ALL=C awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $2}' /proc/mounts`
-			devremaining=`LC_ALL=C awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $1}' /proc/mounts`
+                        remaining=$(network_mounted)
+                        devremaining=$(for dev in $(network_mounted);do cat /proc/mounts | fgrep $dev | awk '$1 ~ /^\/dev\/loop/ && $2 != "/" {print $2}';done)
 			[ -z "$remaining" ] && break
 			/sbin/fuser -k -m $sig $remaining >/dev/null
 			sleep 5
diff -ur initscripts-8.33-old/rc.d/rc initscripts-8.33/rc.d/rc
--- initscripts-8.33-old/rc.d/rc	2006-05-17 10:36:46.000000000 -0400
+++ initscripts-8.33/rc.d/rc	2004-10-31 15:08:15.000000000 -0500
@@ -95,4 +95,3 @@
 if [ -x /usr/bin/rhgb-client ] && /usr/bin/rhgb-client --ping ; then
   /usr/bin/rhgb-client --quit
 fi
-
diff -ur initscripts-8.33-old/rc.d/rc.sysinit initscripts-8.33/rc.d/rc.sysinit
--- initscripts-8.33-old/rc.d/rc.sysinit	2006-05-17 10:36:46.000000000 -0400
+++ initscripts-8.33/rc.d/rc.sysinit	2006-05-17 11:35:57.000000000 -0400
@@ -342,7 +342,10 @@
 				dirs)
 					mount_dirs $path
 					;;
+                                \#)
+                                        ;;
 				*)
+                                        mount_dirs $type
 					;;
 			esac
 			[ -n "$SELINUX_STATE" ] && restorecon -R "$1"
@@ -437,7 +440,7 @@
 # Remount the root filesystem read-write.
 update_boot_stage RCmountfs
 state=`LC_ALL=C awk '/ \/ / && ($3 !~ /rootfs/) { print $4 }' /proc/mounts`
-[ "$state" != "rw" -a "$READONLY" != "yes" ] && \
+[ "${state:0:2}" != "rw" -a "$READONLY" != "yes" ] && \
   action $"Remounting root filesystem in read-write mode: " mount -n -o remount,rw /
 
 # Clean up SELinux labels
