diff -Naur initscripts-8.33.orig/rc.d/init.d/functions initscripts-8.33/rc.d/init.d/functions
--- initscripts-8.33.orig/rc.d/init.d/functions	2006-03-27 17:34:21.000000000 -0500
+++ initscripts-8.33/rc.d/init.d/functions	2006-08-09 16:13:16.000000000 -0400
@@ -103,8 +103,8 @@
 
 # Output PIDs of matching processes, found using pidof
 __pids_pidof() {
-	pidof -c -o $$ -o $PPID -o %PPID -x "$1" || \
-		pidof -c -o $$ -o $PPID -o %PPID -x "${1##*/}"
+	pidof -o $$ -o $PPID -o %PPID -x "$1" || \
+		pidof -o $$ -o $PPID -o %PPID -x "${1##*/}"
 }
 
 
