--- initscripts-8.33/rc.d/rc.sysinit.kvermodules	2006-06-06 14:53:39.000000000 -0400
+++ initscripts-8.33/rc.d/rc.sysinit	2006-06-06 14:54:05.000000000 -0400
@@ -162,6 +162,10 @@
 for file in /etc/sysconfig/modules/*.modules ; do
   [ -x $file ] && $file
 done
+# Do kernel-version-specific module loading as well
+for file in /lib/modules/$unamer/modules.d/*.modules ; do
+  [ -x $file ] && $file
+done
 
 # Load modules (for backward compatibility with VARs)
 if [ -f /etc/rc.modules ]; then
