diff -Naur initscripts-8.33.orig/rc.d/rc.sysinit initscripts-8.33/rc.d/rc.sysinit
--- initscripts-8.33.orig/rc.d/rc.sysinit	2006-04-20 16:10:16.000000000 -0400
+++ initscripts-8.33/rc.d/rc.sysinit	2007-06-13 21:52:36.000000000 -0400
@@ -570,7 +570,7 @@
 		*/screen)	rm -rf $afile/* ;;
 		*/cvs)		rm -rf $afile/* ;;
 		*/dovecot)	rm -f $afile/*/* ;;
-		*)		rm -f $afile/* ;;
+		*)		rm -rf $afile/* ;;
 	   esac
 	else
 	   rm -f $afile
