--- sysconfig/network-scripts/ifup-eth.orig	2008-08-24 05:30:40.000000000 -0400
+++ sysconfig/network-scripts/ifup-eth	2008-08-23 17:57:26.000000000 -0400
@@ -175,10 +175,10 @@ if [ -n "${DYNCONFIG}" -a -x /sbin/dhcli
         cat /etc/dhclient-${DEVICE}.conf.ifupnew > /etc/dhclient-${DEVICE}.conf
         rm -f /etc/dhclient-${DEVICE}.conf.ifupnew
     fi
-    if [[ "${PERSISTENT_DHCLIENT}" =  [yY1]* ]]; then
-       ONESHOT="";
-    else
+    if [[ "${PERSISTENT_DHCLIENT}" =  [nN0]* ]]; then
        ONESHOT="-1";
+    else
+       ONESHOT="";
     fi;
     if [ -n "${DHCP_HOSTNAME}" ]; then
        # Send a host-name to the DHCP server (requ. by some dhcp servers).
