diff -ur initscripts-7.53.orig/rc.d/rc.sysinit initscripts-7.53/rc.d/rc.sysinit
--- initscripts-7.53.orig/rc.d/rc.sysinit	2004-05-07 00:26:43.000000000 -0400
+++ initscripts-7.53/rc.d/rc.sysinit	2005-03-18 11:07:23.048335696 -0500
@@ -13,6 +13,9 @@
 # If we're using devfs, start devfsd now - we need the old device names
 [ -e /dev/.devfsd -a -x /sbin/devfsd ] && /sbin/devfsd /dev
 
+# If we're using udev, start udev now 
+[ -e /dev/.udev.tdb ] && /sbin/udevstart </dev/null >/dev/null 2>&1
+
 HOSTNAME=`/bin/hostname`
 HOSTTYPE=`uname -m`
 unamer=`uname -r`
