diff -ur gparted-0.0.8.orig/gparted.desktop gparted-0.0.8/gparted.desktop
--- gparted-0.0.8.orig/gparted.desktop	2005-05-27 23:04:38.000000000 -0400
+++ gparted-0.0.8/gparted.desktop	2005-05-27 23:16:11.000000000 -0400
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=GParted
+Name=Hard Disk Utility
 Name[ca]=GParted
 Name[cs]=GParted
 Name[de]=GParted
@@ -34,8 +34,9 @@
 Comment[sv]=Gnome-partitionsredigerare
 Comment[uk]=Редактор розділів дисків для GNOME
 Comment[zh_CN]=Gnome 分区编辑器
-Exec=gparted
+Exec=gksudo gparted
 Terminal=false
 Type=Application
 Categories=GNOME;Application;System;
 StartupNotify=true
+Icon=gparted.png
