diff -Naur gparted-0.3.orig/gparted.desktop gparted-0.3/gparted.desktop
--- gparted-0.3.orig/gparted.desktop	2006-09-02 07:01:22.000000000 -0400
+++ gparted-0.3/gparted.desktop	2006-09-05 06:36:21.000000000 -0400
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=GParted
+Name=Disk Partition Editor
 Name[bg]=GParted
 Name[ca]=GParted
 Name[cs]=GParted
@@ -51,9 +51,9 @@
 Comment[sv]=Partitionsredigerare för GNOME
 Comment[tr]=GNOME Bölümleme Düzenleyicisi
 Comment[vi]=Bộ hiệu chỉnh phân vùng Gnome
-Exec=gparted
+Exec=gksudo gparted
 Icon=gparted.png
 Terminal=false
 Type=Application
-Categories=GNOME;Application;System;
+Categories=GNOME;Settings;System;
 StartupNotify=true
diff -Naur gparted-0.3.orig/gparted.desktop.in gparted-0.3/gparted.desktop.in
--- gparted-0.3.orig/gparted.desktop.in	2006-01-26 07:50:45.000000000 -0500
+++ gparted-0.3/gparted.desktop.in	2006-09-05 06:36:31.000000000 -0400
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=GParted
+_Name=Disk Partition Editor
 _Comment=GNOME Partition Editor
-Exec=gparted
+Exec=gksudo gparted
 Icon=gparted.png
 Terminal=false
 Type=Application
-Categories=GNOME;Application;System;
+Categories=GNOME;Settings;System;
 StartupNotify=true
