--- vlc/vlc-2.0.0/bin/Makefile.am	2011-12-08 19:00:25.000000000 +0100
+++ vlc/vlc-2.0.0/bin/Makefile-no-vlc-cache-gen.am	2012-02-19 10:49:52.000000000 +0100
@@ -69,18 +69,4 @@
 vlc_cache_gen_DEPENDENCIES = vlc_win32_rc.$(OBJEXT)
 endif
 
-#
-# Plug-ins cache
-#
-noinst_DATA += ../modules/plugins.dat
-MOSTLYCLEANFILES = $(noinst_DATA)
 
-.PHONY: ../modules/plugins.dat
-
-../modules/plugins.dat: vlc-cache-gen$(EXEEXT)
-	$(AM_V_at)rm -f ../modules/plugins.dat
-	$(AM_V_GEN)if test "$(build)" = "$(host)"; then \
-		./vlc-cache-gen$(EXEEXT) ../modules ; \
-	else \
-		echo "Cross-compilation: cache generation skipped!" ; \
-	fi
