diff -uNr layout/applications.menu newlayout/applications.menu
--- layout/applications.menu	2005-06-28 09:16:39.000000000 +0100
+++ newlayout/applications.menu	2006-01-08 16:45:07.000000000 +0000
@@ -136,6 +136,17 @@
     </Include>
   </Menu>   <!-- End System Tools -->
 
+  <!-- Foresight -->
+  <Menu>
+    <Name>Foresight</Name>
+    <Directory>Foresight.directory</Directory>
+    <Include>
+      <And>
+        <Category>Foresight</Category>
+      </And>
+    </Include>
+  </Menu>   <!-- End Foresight -->
+
   <!-- Other -->
   <Menu>
     <Name>Other</Name>
