diff -Naur gnome-menus-2.18.2.orig/layout/applications.menu gnome-menus-2.18.2/layout/applications.menu
--- gnome-menus-2.18.2.orig/layout/applications.menu	2007-05-28 07:41:35.000000000 -0400
+++ gnome-menus-2.18.2/layout/applications.menu	2007-06-28 09:17:00.000000000 -0400
@@ -102,6 +102,29 @@
     </Include>
   </Menu>   <!-- End Internet -->
 
+  <!-- Foresight -->
+  <Menu>
+    <Name>Foresight</Name>
+    <Directory>Foresight.directory</Directory>
+    <Include>
+      <And>
+        <Category>Foresight</Category>
+      </And>
+    </Include>
+  </Menu>   <!-- End Foresight -->
+
+  <!-- Google -->
+  <Menu>
+    <Name>Google Desktop</Name>
+    <Directory>google-gdl.directory</Directory>
+    <Include>
+      <And>
+        <Category>Google</Category>
+      </And>
+    </Include> 
+  </Menu>   <!-- End Google -->
+
+
   <!-- Multimedia -->
   <Menu>
     <Name>Multimedia</Name>
