--- a/src/mail-server.c
+++ b/src/mail-server.c
@@ -139,7 +139,7 @@
        
   priv = server->priv = MAIL_SERVER_GET_PRIVATE (server);
 
-  indicate_server_set_desktop_file(INDICATE_SERVER(server), "/usr/share/applications/evolution-mail.desktop");
+  indicate_server_set_desktop_file(INDICATE_SERVER(server), "/usr/share/applications/evolution.desktop");
   indicate_server_set_type(INDICATE_SERVER(server), "message.mail");
 }
 
