--- src/evolution-indicator.c	2009-09-17 11:24:20.000000000 +0200
+++ src/evolution-indicator.c.new	2009-09-22 10:58:27.104136721 +0200
@@ -59,7 +59,7 @@
 #define ACCOUNT_DIR "/apps/evolution/mail"
 #define ACCOUNTS    ACCOUNT_DIR"/accounts"
 
-#define EVOLUTION_DESKTOP_FILE "/usr/share/applications/evolution-mail.desktop"
+#define EVOLUTION_DESKTOP_FILE "/usr/share/applications/evolution.desktop"
 
 #define UNREAD_DATA "unread"
 
