diff -Naur gnome-system-tools-1.4.0.orig/src/disks/disks-gui.c gnome-system-tools-1.4.0/src/disks/disks-gui.c
--- gnome-system-tools-1.4.0.orig/src/disks/disks-gui.c	2005-08-08 15:00:02.000000000 -0400
+++ gnome-system-tools-1.4.0/src/disks/disks-gui.c	2005-10-30 22:12:27.000000000 -0500
@@ -1058,7 +1058,7 @@
 	gst_disks_gui_setup_mounted (status_label, mount_button, mounted);
 
 	if (mounted) {
-		gtk_widget_set_sensitive (browse_button, TRUE);
+		/* gtk_widget_set_sensitive (browse_button, TRUE); */
 		
 		gtk_progress_bar_set_fraction (
 			GTK_PROGRESS_BAR (size_progress), 1.0);
