diff -Naur gnome-system-tools-1.4.0.orig/src/disks/callbacks.c gnome-system-tools-1.4.0/src/disks/callbacks.c
--- gnome-system-tools-1.4.0.orig/src/disks/callbacks.c	2005-08-08 15:00:02.000000000 -0400
+++ gnome-system-tools-1.4.0/src/disks/callbacks.c	2005-10-30 12:10:33.000000000 -0500
@@ -483,7 +483,7 @@
 		if (gtk_tree_model_get_n_columns (model) == PARTITION_LIST_LAST) {
 			gtk_tree_model_get (model, &iter, PARTITION_LIST_POINTER, &part, -1);
 			if (GST_IS_DISKS_PARTITION (part)) {
-				gst_disks_partition_browse (part);
+				/*gst_disks_partition_browse (part);*/
 			}
 		} else if (gtk_tree_model_get_n_columns (model) == STORAGE_LIST_LAST) {
 			gtk_tree_model_get (model, &iter, STORAGE_LIST_POINTER, &cdrom, -1);
