diff -r 5332db63ff9d src/common/gst-service-role.c
--- a/src/common/gst-service-role.c	Mon Sep 22 15:02:43 2008 -0400
+++ b/src/common/gst-service-role.c	Mon Sep 22 15:03:13 2008 -0400
@@ -158,7 +158,6 @@
 	{ "rgmanager",              GST_ROLE_CLUSTER_MANAGEMENT },
 	{ "rsyncd",                 GST_ROLE_REMOTE_BACKUP },
 	{ "rsync",                  GST_ROLE_REMOTE_BACKUP },
-	{ "samba",                  GST_ROLE_FILE_SERVER_SMB },
 	{ "schoolbell",             GST_ROLE_WEB_CALENDAR_SERVER },
 	{ "schooltool",             GST_ROLE_SCHOOL_MANAGEMENT_PLATFORM },
 	{ "screen",                 GST_ROLE_TERMINAL_MULTIPLEXOR },
@@ -167,6 +166,7 @@
 	{ "shorewall",              GST_ROLE_FIREWALL_MANAGEMENT },
 	{ "slapd",                  GST_ROLE_LDAP_SERVER },
 	{ "smartmontools",          GST_ROLE_HARDWARE_MONITORING },
+	{ "smb",                    GST_ROLE_FILE_SERVER_SMB },
 	{ "snmpd",                  GST_ROLE_SNMP_SERVER },
 	{ "spamassassin",           GST_ROLE_SPAM_FILTER },
 	{ "squid",                  GST_ROLE_PROXY_CACHE },
