diff -Naur --exclude=.svn configure.in.up configure.in
+++ configure.in.up	2011-03-19 19:35:01.828390047 +0000
--- configure.in	2011-03-19 19:35:17.140390047 +0000
@@ -30,12 +30,6 @@
 AC_SUBST(GLIB_CFLAGS)
 AC_SUBST(GLIB_LIBS)
 
-PKG_CHECK_MODULES(GLIB2_26,[glib-2.0 >= 2.26.0],[with_gconf=no with_gsettings=yes with_gio=yes GLIB2_26_DEFINES="-DGLIB2_26_ENABLED" AC_DEFINE([GLIB2_26_ENABLED],[1],[Define to 1 if you have GLIB 2.26 or higher])],AC_MSG_NOTICE([Using GLIB < 2.26 specific features]))
-AC_SUBST(GLIB2_26_DEFINES)
-AC_SUBST(ALSA_CFLAGS)
-AC_SUBST(PULSE_CFLAGS)
-
-
 AC_ARG_WITH([xulrunner-sdk],
     [AS_HELP_STRING([--with-xulrunner-sdk=path_to_sdk],
         [Path to xulrunner SDK for build --with-xulrunner-sdk=/path/to/sdk])],
