--- a/lib/ec2/common/curl.rb	2008-12-09 10:16:45.000000000 -0500
+++ b/lib/ec2/common/curl.rb	2008-12-11 13:13:03.000000000 -0500
@@ -63,7 +63,7 @@
       end
       
       def self.invoke(command, debug=false)
-        invocation =  "curl -sSL #{command}"
+        invocation =  "curl -ksSL #{command}"
 #         invocation =  "curl -vsSL #{command}" if debug
         invocation << ' -w "Response-Code: %{http_code}\nContent-Type: %{content_type}"'
         STDERR.puts invocation if debug
