diff -ur bin/ec2-ami-tools-version bin/ec2-ami-tools-version
--- a/bin/ec2-ami-tools-version	2008-12-09 10:20:32.000000000 -0500
+++ b/bin/ec2-ami-tools-version	2008-12-17 10:28:56.000000000 -0500
@@ -1,4 +1,3 @@
 #!/bin/bash
 
-home=${EC2_AMITOOL_HOME:-${EC2_HOME:?Neither of EC2_AMITOOL_HOME or EC2_HOME environment variables are set}}
-ruby -I $home/lib ${home}/lib/ec2/amitools/showversion.rb $*
+ruby %(ec2home)s/amitools/showversion.rb $*
diff -ur bin/ec2-bundle-image bin/ec2-bundle-image
--- a/bin/ec2-bundle-image	2008-12-09 10:20:32.000000000 -0500
+++ b/bin/ec2-bundle-image	2008-12-17 10:29:02.000000000 -0500
@@ -1,4 +1,3 @@
 #!/bin/bash
 
-home=${EC2_AMITOOL_HOME:-${EC2_HOME:?Neither of EC2_AMITOOL_HOME or EC2_HOME environment variables are set}}
-ruby -I $home/lib ${home}/lib/ec2/amitools/bundleimage.rb $*
+ruby %(ec2home)s/amitools/bundleimage.rb $*
diff -ur bin/ec2-bundle-vol bin/ec2-bundle-vol
--- a/bin/ec2-bundle-vol	2008-12-09 10:20:32.000000000 -0500
+++ b/bin/ec2-bundle-vol	2008-12-17 10:29:07.000000000 -0500
@@ -1,4 +1,3 @@
 #!/bin/bash
 
-home=${EC2_AMITOOL_HOME:-${EC2_HOME:?Neither of EC2_AMITOOL_HOME or EC2_HOME environment variables are set}}
-ruby -I $home/lib ${home}/lib/ec2/amitools/bundlevol.rb $*
+ruby %(ec2home)s/amitools/bundlevol.rb $*
diff -ur bin/ec2-delete-bundle bin/ec2-delete-bundle
--- a/bin/ec2-delete-bundle	2008-12-09 10:20:32.000000000 -0500
+++ b/bin/ec2-delete-bundle	2008-12-17 10:29:14.000000000 -0500
@@ -1,4 +1,3 @@
 #!/bin/bash
 
-home=${EC2_AMITOOL_HOME:-${EC2_HOME:?Neither of EC2_AMITOOL_HOME or EC2_HOME environment variables are set}}
-ruby -I $home/lib ${home}/lib/ec2/amitools/deletebundle.rb $*
+ruby %(ec2home)s/amitools/deletebundle.rb $*
diff -ur bin/ec2-download-bundle bin/ec2-download-bundle
--- a/bin/ec2-download-bundle	2008-12-09 10:20:32.000000000 -0500
+++ b/bin/ec2-download-bundle	2008-12-17 10:26:24.000000000 -0500
@@ -1,4 +1,3 @@
 #!/bin/bash
 
-home=${EC2_AMITOOL_HOME:-${EC2_HOME:?Neither of EC2_AMITOOL_HOME or EC2_HOME environment variables are set}}
-ruby -I $home/lib ${home}/lib/ec2/amitools/downloadbundle.rb $*
+ruby %(ec2home)s/amitools/downloadbundle.rb $*
diff -ur bin/ec2-migrate-bundle bin/ec2-migrate-bundle
--- a/bin/ec2-migrate-bundle	2008-12-09 10:20:32.000000000 -0500
+++ b/bin/ec2-migrate-bundle	2008-12-17 10:26:49.000000000 -0500
@@ -1,4 +1,3 @@
 #!/bin/bash
 
-home=${EC2_AMITOOL_HOME:-${EC2_HOME:?Neither of EC2_AMITOOL_HOME or EC2_HOME environment variables are set}}
-ruby -I $home/lib ${home}/lib/ec2/amitools/migratebundle.rb $*
+ruby %(ec2home)s/amitools/migratebundle.rb $*
diff -ur bin/ec2-migrate-manifest bin/ec2-migrate-manifest
--- a/bin/ec2-migrate-manifest	2008-12-09 10:20:32.000000000 -0500
+++ b/bin/ec2-migrate-manifest	2008-12-17 10:27:22.000000000 -0500
@@ -1,4 +1,3 @@
 #!/bin/bash
 
-home=${EC2_AMITOOL_HOME:-${EC2_HOME:?Neither of EC2_AMITOOL_HOME or EC2_HOME environment variables are set}}
-ruby -I $home/lib ${home}/lib/ec2/amitools/migratemanifest.rb $*
+ruby %(ec2home)s/amitools/migratemanifest.rb $*
diff -ur bin/ec2-unbundle bin/ec2-unbundle
--- a/bin/ec2-unbundle	2008-12-09 10:20:32.000000000 -0500
+++ b/bin/ec2-unbundle	2008-12-17 10:27:56.000000000 -0500
@@ -1,4 +1,3 @@
 #!/bin/bash
 
-home=${EC2_AMITOOL_HOME:-${EC2_HOME:?Neither of EC2_AMITOOL_HOME or EC2_HOME environment variables are set}}
-ruby -I $home/lib ${home}/lib/ec2/amitools/unbundle.rb $*
+ruby %(ec2home)s/amitools/unbundle.rb $*
diff -ur bin/ec2-upload-bundle bin/ec2-upload-bundle
--- a/bin/ec2-upload-bundle	2008-12-09 10:20:32.000000000 -0500
+++ b/bin/ec2-upload-bundle	2008-12-17 10:28:24.000000000 -0500
@@ -1,4 +1,3 @@
 #!/bin/bash
 
-home=${EC2_AMITOOL_HOME:-${EC2_HOME:?Neither of EC2_AMITOOL_HOME or EC2_HOME environment variables are set}}
-ruby -I $home/lib ${home}/lib/ec2/amitools/uploadbundle.rb $*
+ruby %(ec2home)s/amitools/uploadbundle.rb $*
