--- gimp-print-4.2.6/src/foomatic/foomatic-generator.in.orig	2004-09-09 14:01:17.987224152 -0400
+++ gimp-print-4.2.6/src/foomatic/foomatic-generator.in	2004-09-09 14:01:52.943909928 -0400
@@ -206,7 +206,7 @@
 
 open PRINTERS, $mapfile or die "Cannot open mapfile $mapfile: $!\n";
 for (<PRINTERS>) {
-    if (m!^\#\s*gptofoo\s+([^\s]+)\s+([^\s]+)!) {
+    if (m!^\#gptofoo\s+(\S+)\s+(\S+)!) {
 	my $foomaticid = $2;
 	my $deviceclass = $1;
 	if ($opt_o) {
