diff -Naur system-tools-backends-2.4.1.orig/Time/TimeDate.pm system-tools-backends-2.4.1/Time/TimeDate.pm
--- system-tools-backends-2.4.1.orig/Time/TimeDate.pm	2007-10-03 20:14:46.000000000 -0400
+++ system-tools-backends-2.4.1/Time/TimeDate.pm	2008-01-03 14:49:32.000000000 -0500
@@ -144,7 +144,7 @@
     unlink $localtime;  # Important, since it might be a symlink.
     
     &Utils::Report::enter ();
-    $res = symlink ($tz, $localtime);
+    $res = copy ($tz, $localtime);
     &Utils::Report::leave ();
     return -1 unless $res;
     return 0;
