diff -Naur system-tools-backends-1.9.7.orig/Utils/Platform.pm system-tools-backends-1.9.7/Utils/Platform.pm
--- system-tools-backends-1.9.7.orig/Utils/Platform.pm	2006-10-05 18:12:06.000000000 -0400
+++ system-tools-backends-1.9.7/Utils/Platform.pm	2006-10-09 15:52:06.000000000 -0400
@@ -99,6 +99,7 @@
   "fedora-3"        => [ "Fedora Core", "3", "Heidelberg" ],
   "fedora-4"        => [ "Fedora Core", "4", "Stentz" ],
   "rpath"           => [ "rPath Linux" ],
+  "rpath"           => [ "Foresight Desktop Linux" ],
   "ark"             => [ "Ark Linux" ],
   "solaris-2.11"    => [ "Solaris / OpenSolaris", "2.11", "Nevada" ],
   "nexenta-1.0"     => [ "Nexenta GNU/Solaris", "1.0", "Ellate" ],
