diff -Naur system-tools-backends-1.4.0.orig/disks-conf.in system-tools-backends-1.4.0/disks-conf.in
--- system-tools-backends-1.4.0.orig/disks-conf.in	2005-01-20 13:48:52.000000000 -0500
+++ system-tools-backends-1.4.0/disks-conf.in	2005-10-31 11:35:41.000000000 -0500
@@ -66,7 +66,7 @@
 @platforms = ("redhat-5.2", "redhat-6.0", "redhat-6.1", "redhat-6.2", "redhat-7.0",
               "redhat-7.1", "mandrake-7.2", "debian-2.2", "debian-woody", "debian-sarge",
 	      "debian-3.1", "suse-7.0", "suse-9.1", "suse-1.0", "unitedlinux-1.0", 
-	      "turbolinux-7.0");
+	      "turbolinux-7.0", "foresight");
 
 $description =<<"end_of_description;";
        Configures locally mounted partitioned media.
diff -Naur system-tools-backends-1.4.0.orig/network-conf.in system-tools-backends-1.4.0/network-conf.in
--- system-tools-backends-1.4.0.orig/network-conf.in	2005-08-27 11:21:20.000000000 -0400
+++ system-tools-backends-1.4.0/network-conf.in	2005-10-31 11:35:05.000000000 -0500
@@ -70,7 +70,7 @@
               "debian-2.2", "debian-3.0", "debian-sarge",
 	      "ubuntu-5.04",
               "suse-7.0", "suse-9.0", "suse-9.1",
-              "turbolinux-7.0", "fedora-1", "fedora-2", "fedora-3", "specifix",
+              "turbolinux-7.0", "fedora-1", "fedora-2", "fedora-3", "foresight",
               "pld-1.0", "pld-1.1", "pld-1.99",
               "conectiva-9", "conectiva-10",
               "vine-3.0", "vine-3.1",
diff -Naur system-tools-backends-1.4.0.orig/network.pl.in system-tools-backends-1.4.0/network.pl.in
--- system-tools-backends-1.4.0.orig/network.pl.in	2005-08-27 10:53:53.000000000 -0400
+++ system-tools-backends-1.4.0/network.pl.in	2005-10-31 11:35:05.000000000 -0500
@@ -516,7 +516,7 @@
        "fedora-1"     => "redhat-6.2",
        "fedora-2"     => "redhat-6.2",
        "fedora-3"     => "redhat-6.2",
-       "specifix"     => "redhat-6.2",
+       "foresight"     => "redhat-6.2",
        "vine-3.0"     => "redhat-6.2",
        "vine-3.1"     => "redhat-6.2",
        "slackware-9.1.0" => "redhat-6.2",
@@ -1798,7 +1798,7 @@
     "fedora-1"     => \&gst_network_rh72_get_file,
     "fedora-2"     => \&gst_network_rh72_get_file,
     "fedora-3"     => \&gst_network_rh72_get_file,
-    "specifix"     => \&gst_network_rh72_get_file,
+    "foresight"     => \&gst_network_rh72_get_file,
     "vine-3.0"     => \&gst_network_rh62_get_file,
     "vine-3.1"     => \&gst_network_rh62_get_file,
     "slackware-9.1.0" => \&gst_network_deb22_get_file,
@@ -3901,7 +3901,7 @@
     "fedora-1"     => "",
     "fedora-2"     => "",
     "fedora-3"     => "",
-    "specifix"     => "",
+    "foresight"     => "",
     "vine-3.0"     => "lo",
     "vine-3.1"     => "lo",
     "slackware-9.1.0" => "",
@@ -4057,7 +4057,7 @@
     "fedora-1"     => "redhat-7.2",
     "fedora-2"     => "redhat-7.2",
     "fedora-3"     => "redhat-7.2",
-    "specifix"     => "redhat-7.2",
+    "foresight"     => "redhat-7.2",
     "vine-3.0"     => "redhat-7.0",
     "vine-3.1"     => "redhat-7.0",
     "slackware-9.1.0" => "slackware-9.1.0",
@@ -4501,7 +4501,7 @@
     "fedora-1"     => "redhat-7.2",
     "fedora-2"     => "redhat-7.2",
     "fedora-3"     => "redhat-7.2",
-    "specifix"     => "redhat-7.2",
+    "foresight"     => "redhat-7.2",
     "vine-3.0"     => "vine-3.0",
     "vine-3.1"     => "vine-3.0",
     "slackware-9.1.0" => "slackware-9.1.0",
@@ -5375,7 +5375,7 @@
     "fedora-1"     => "redhat-7.2",
     "fedora-2"     => "redhat-7.2",
     "fedora-3"     => "redhat-7.2",
-    "specifix"     => "redhat-7.2",
+    "foresight"     => "redhat-7.2",
     "vine-3.0"     => "redhat-7.0",
     "vine-3.1"     => "redhat-7.0",
     "slackware-9.1.0" => "slackware-9.1.0",
@@ -5805,7 +5805,7 @@
     "fedora-1"     => "redhat-7.2",
     "fedora-2"     => "redhat-7.2",
     "fedora-3"     => "redhat-7.2",
-    "specifix"     => "redhat-7.2",
+    "foresight"     => "redhat-7.2",
     "vine-3.0"     => "vine-3.0",
     "vine-3.1"     => "vine-3.0",
     "slackware-9.1.0" => "slackware-9.1.0",
diff -Naur system-tools-backends-1.4.0.orig/platform.pl.in system-tools-backends-1.4.0/platform.pl.in
--- system-tools-backends-1.4.0.orig/platform.pl.in	2005-07-04 18:21:29.000000000 -0400
+++ system-tools-backends-1.4.0/platform.pl.in	2005-10-31 11:35:05.000000000 -0500
@@ -88,7 +88,7 @@
   "fedora-1"        => "Fedora Core 1 (Yarrow)",
   "fedora-2"        => "Fedora Core 2 (Tettnang)",
   "fedora-3"        => "Fedora Core 3 (Heidelberg)",
-  "specifix"        => "Specifix Linux",
+  "foresight"        => "Foresight Desktop Linux",
 };
   
 sub check_lsb
@@ -261,7 +261,7 @@
     return -1;
 }
 
-sub check_specifix
+sub check_foresight
 {
   open SPECIFIX, "$gst_prefix/etc/distro-release" or return -1;
 
@@ -270,10 +270,10 @@
     $ver = $_;
     chomp ($ver);
 
-    if ($ver =~ /^Specifix Linux release (\S+)/)
+    if ($ver =~ /^Foresight Desktop Linux release (\S+)/)
     {
       close SPECIFIX;
-      return "specifix";
+      return "foresight";
     }
   }
 
@@ -478,7 +478,7 @@
     # Red Hat check must run after Vine, Mandrake and Fedora, and Mandrake after BlackPanther
     "Linux" => [ \&check_lsb,      \&check_debian,   \&check_caldera, \&check_suse, \&check_blackpanther, \&check_vine,
                  \&check_fedora, \&check_mandrake, \&check_conectiva, \&check_linuxppc, \&check_redhat,  \&check_openna,
-                 \&check_turbolinux, \&check_slackware, \&check_vlos, \&check_gentoo, \&check_pld, \&check_specifix, \&check_archlinux ],
+                 \&check_turbolinux, \&check_slackware, \&check_vlos, \&check_gentoo, \&check_pld, \&check_foresight, \&check_archlinux ],
     "FreeBSD" => [ \&check_freebsd ],
     "SunOS"    => [ \&check_solaris ]
                );
diff -Naur system-tools-backends-1.4.0.orig/service.pl.in system-tools-backends-1.4.0/service.pl.in
--- system-tools-backends-1.4.0.orig/service.pl.in	2005-07-04 18:21:29.000000000 -0400
+++ system-tools-backends-1.4.0/service.pl.in	2005-10-31 11:35:05.000000000 -0500
@@ -92,7 +92,7 @@
        "fedora-2"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
        "fedora-3"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
 
-       "specifix"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
+       "foresight"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
 
        "vine-3.0"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
        "vine-3.1"     => ["$gst_prefix/etc/rc.d", "$gst_prefix/etc/rc.d/init.d", "../init.d"],
@@ -154,7 +154,7 @@
        "fedora-2"       => [3, 5],
        "fedora-3"       => [3, 5],
 
-       "specifix"       => [3, 5],
+       "foresight"       => [3, 5],
        
        "vine-3.0"       => [3, 5],
        "vine-3.1"       => [3, 5],
@@ -213,7 +213,7 @@
      "fedora-2"       => "redhat-5.2",
      "fedora-3"       => "redhat-5.2",
 
-     "specifix"       => "redhat-5.2",
+     "foresight"       => "redhat-5.2",
 
      "vine-3.0"       => "redhat-5.2",
      "vine-3.1"       => "redhat-5.2",
@@ -1561,7 +1561,7 @@
        "fedora-1"     => \&gst_service_redhat_get_status,
        "fedora-2"     => \&gst_service_redhat_get_status,
        "fedora-3"     => \&gst_service_redhat_get_status,
-       "specifix"     => \&gst_service_redhat_get_status,
+       "foresight"     => \&gst_service_redhat_get_status,
 
        "suse-9.0"     => \&gst_service_suse_get_status,
        "suse-9.1"     => \&gst_service_suse_get_status,
diff -Naur system-tools-backends-1.4.0.orig/services-conf.in system-tools-backends-1.4.0/services-conf.in
--- system-tools-backends-1.4.0.orig/services-conf.in	2005-06-26 13:43:10.000000000 -0400
+++ system-tools-backends-1.4.0/services-conf.in	2005-10-31 11:35:05.000000000 -0500
@@ -51,7 +51,7 @@
               
               "debian-2.2", "debian-3.0", "debian-sarge",
 
-              "suse-7.0", "turbolinux-7.0", "fedora-1", "fedora-2", "fedora-3", "specifix",
+              "suse-7.0", "turbolinux-7.0", "fedora-1", "fedora-2", "fedora-3", "foresight",
               "pld-1.0", "pld-1.1", "pld-1.99",
               "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0",
               "vine-3.0", "vine-3.1",
diff -Naur system-tools-backends-1.4.0.orig/shares-conf.in system-tools-backends-1.4.0/shares-conf.in
--- system-tools-backends-1.4.0.orig/shares-conf.in	2005-06-21 13:37:10.000000000 -0400
+++ system-tools-backends-1.4.0/shares-conf.in	2005-10-31 11:35:05.000000000 -0500
@@ -547,7 +547,7 @@
      "fedora-1"     => "redhat-6.2",
      "fedora-2"     => "redhat-6.2",
      "fedora-3"     => "redhat-6.2",
-     "specifix"     => "redhat-6.2",
+     "foresight"     => "redhat-6.2",
      "vine-3.0"     => "redhat-6.2",
      "vine-3.1"     => "redhat-6.2",
      "slackware-9.1.0" => "slackware-9.1.0",
diff -Naur system-tools-backends-1.4.0.orig/time-conf.in system-tools-backends-1.4.0/time-conf.in
--- system-tools-backends-1.4.0.orig/time-conf.in	2005-06-21 13:39:32.000000000 -0400
+++ system-tools-backends-1.4.0/time-conf.in	2005-10-31 11:35:05.000000000 -0500
@@ -69,7 +69,7 @@
               "debian-2.2", "debian-3.0", "debian-sarge",
               "suse-7.0", "suse-9.0", "suse-9.1", "turbolinux-7.0",
               "slackware-8.0.0", "slackware-8.1", "slackware-9.0.0", "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0",
-              "gentoo", "vlos-1.2", "pld-1.0", "pld-1.1", "pld-1.99", "fedora-1", "fedora-2", "fedora-3", "specifix",
+              "gentoo", "vlos-1.2", "pld-1.0", "pld-1.1", "pld-1.99", "fedora-1", "fedora-2", "fedora-3", "foresight",
               "vine-3.0", "vine-3.1",
               "freebsd-5", "freebsd-6");
 
@@ -544,7 +544,7 @@
    "fedora-2"        => "redhat-7.0",
    "fedora-3"        => "redhat-7.0",
    
-   "specifix"        => "redhat-7.0",
+   "foresight"        => "redhat-7.0",
 
    "vine-3.0"        => "redhat-7.0",
    "vine-3.1"        => "redhat-7.0",
@@ -775,7 +775,7 @@
    "fedora-1"        => "redhat-7.0",
    "fedora-2"        => "redhat-7.0",
    "fedora-3"        => "redhat-7.0",
-   "specifix"        => "redhat-7.0",
+   "foresight"        => "redhat-7.0",
 
    "vine-3.0"        => "redhat-7.0",
    "vine-3.1"        => "redhat-7.0",
diff -Naur system-tools-backends-1.4.0.orig/users-conf.in system-tools-backends-1.4.0/users-conf.in
--- system-tools-backends-1.4.0.orig/users-conf.in	2005-08-08 13:30:23.000000000 -0400
+++ system-tools-backends-1.4.0/users-conf.in	2005-10-31 11:35:05.000000000 -0500
@@ -79,7 +79,7 @@
               "freebsd-4", "freebsd-5", "freebsd-6",
               "gentoo", "vlos-1.2",
               "archlinux-0.7",
-              "pld-1.0", "pld-1.1", "pld-1.99", "fedora-1", "fedora-2", "fedora-3", "specifix", "vine-3.0", "vine-3.1");
+              "pld-1.0", "pld-1.1", "pld-1.99", "fedora-1", "fedora-2", "fedora-3", "foresight", "vine-3.0", "vine-3.1");
 
 $description =<<"end_of_description;";
        Manages system users.
@@ -358,7 +358,7 @@
   'fedora-1'        => $rh_logindefs_defaults,
   'fedora-2'        => $rh_logindefs_defaults,
   'fedora-3'        => $rh_logindefs_defaults,
-  'specifix'        => $rh_logindefs_defaults,
+  'foresight'        => $rh_logindefs_defaults,
   'debian-2.2'      => $rh_logindefs_defaults,
   'debian-3.0'      => $rh_logindefs_defaults,
   'debian-sarge'    => $rh_logindefs_defaults,
