cvs diff: Diffing Users
Index: Users/Users.pm
===================================================================
RCS file: /cvs/system-tools-backends/system-tools-backends/Users/Users.pm,v
retrieving revision 1.11
diff -u -p -r1.11 Users.pm
--- Users/Users.pm      25 Sep 2007 23:43:27 -0000      1.11
+++ Users/Users.pm      14 Oct 2007 16:58:20 -0000
@@ -579,7 +579,8 @@ sub add_user
 
     if ($cmd_adduser &&
         $Utils::Backend::tool{"platform"} !~ /^slackware/ &&
-        $Utils::Backend::tool{"platform"} !~ /^archlinux/)
+        $Utils::Backend::tool{"platform"} !~ /^archlinux/ &&
+        $Utils::Backend::tool{"platform"} !~ /^redhat/)
     {
       # use adduser if available and valid (slackware one is b0rk)
       # set empty gecos fields and password, they will be filled out later

