diff -r 338c813c783d Users/Users.pm
--- a/Users/Users.pm    Mon Mar 05 01:53:18 2007 +0000
+++ b/Users/Users.pm    Mon Mar 05 03:09:10 2007 +0000
@@ -577,39 +577,16 @@ sub add_user
     $home_parents =~ s/\/+[^\/]+\/*$//;
     &Utils::File::run ("$tool_mkdir -p $home_parents");
 
-    if ($cmd_adduser)
-    {
-      # use adduser if available, set empty gecos fields
-      # and password, they will be filled out later
-      $command = "$cmd_adduser --gecos '' --disabled-password" .
-          " --home \'"  . $$user[$HOME]   . "\'" .
-          " --gid \'"   . $$user[$GID]    . "\'" .
-          " --shell \'" . $$user[$SHELL]  . "\'" .
-          " --uid \'"   . $$user[$UID]    . "\'" .
-          " \'"         . $$user[$LOGIN]  . "\'";
-
-      &Utils::File::run ($command);
-
-      # password can't be set in non-interactive
-      # mode with adduser, call usermod instead
-      $command = "$cmd_usermod " .
-          " -p '" . $$user[$PASSWD] . "' " . $$user[$LOGIN];
-
-      &Utils::File::run ($command);
-    }
-    else
-    {
-      # fallback to useradd
-      $command = "$cmd_useradd -m" .
-          " --home \'"     . $$user[$HOME]   . "\'" .
-          " --gid \'"      . $$user[$GID]    . "\'" .
-          " --password \'" . $$user[$PASSWD] . "\'" .
-          " --shell \'"    . $$user[$SHELL]  . "\'" .
-          " --uid \'"      . $$user[$UID]    . "\'" .
-          " \'"            . $$user[$LOGIN]  . "\'";
-
-      &Utils::File::run ($command);
-    }
+    # fallback to useradd
+    $command = "$cmd_useradd -m" .
+        " -d \'"     . $$user[$HOME]   . "\'" .
+        " -g \'"      . $$user[$GID]    . "\'" .
+        " -p \'" . $$user[$PASSWD] . "\'" .
+        " -s \'"    . $$user[$SHELL]  . "\'" .
+        " -u \'"      . $$user[$UID]    . "\'" .
+        " \'"            . $$user[$LOGIN]  . "\'";
+
+    &Utils::File::run ($command);
   }
 
   &change_user_chfn ($$user[$LOGIN], undef, $$user[$COMMENT]);

