diff -up c-ares-1.6.0/configure~ c-ares-1.6.0/configure
--- c-ares-1.6.0/configure~	2008-12-10 00:31:08.000000000 +0200
+++ c-ares-1.6.0/configure	2009-07-25 20:26:11.000000000 +0300
@@ -14950,7 +14950,7 @@ echo "$as_me: WARNING: compiler options 
     ac_var_stripped=""
   for word1 in $tmp_CFLAGS; do
     ac_var_strip_word="no"
-    for word2 in $flags_dbg_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_strip_word="yes"
       fi
@@ -14966,7 +14966,7 @@ echo "$as_me: WARNING: compiler options 
     ac_var_stripped=""
   for word1 in $tmp_CPPFLAGS; do
     ac_var_strip_word="no"
-    for word2 in $flags_dbg_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_strip_word="yes"
       fi
@@ -14982,12 +14982,12 @@ echo "$as_me: WARNING: compiler options 
     if test "$want_debug" = "yes"; then
       { echo "$as_me:$LINENO: checking if compiler accepts debug enabling options" >&5
 echo $ECHO_N "checking if compiler accepts debug enabling options... $ECHO_C" >&6; }
-      tmp_options="$flags_dbg_yes"
+      tmp_options=""
     fi
     if test "$want_debug" = "no"; then
       { echo "$as_me:$LINENO: checking if compiler accepts debug disabling options" >&5
 echo $ECHO_N "checking if compiler accepts debug disabling options... $ECHO_C" >&6; }
-      tmp_options="$flags_dbg_off"
+      tmp_options=""
     fi
     #
     CPPFLAGS="$tmp_CPPFLAGS"
@@ -15221,7 +15221,7 @@ echo $ECHO_N "checking if compiler optim
 
   ac_var_match_word="no"
   for word1 in $tmp_CFLAGS; do
-    for word2 in $flags_opt_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_match_word="yes"
       fi
@@ -15239,7 +15239,7 @@ echo $ECHO_N "checking if compiler optim
 
   ac_var_match_word="no"
   for word1 in $tmp_CPPFLAGS; do
-    for word2 in $flags_opt_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_match_word="yes"
       fi
@@ -15270,7 +15270,7 @@ echo "${ECHO_T}$honor_optimize_option" >
     ac_var_stripped=""
   for word1 in $tmp_CFLAGS; do
     ac_var_strip_word="no"
-    for word2 in $flags_opt_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_strip_word="yes"
       fi
@@ -15286,7 +15286,7 @@ echo "${ECHO_T}$honor_optimize_option" >
     ac_var_stripped=""
   for word1 in $tmp_CPPFLAGS; do
     ac_var_strip_word="no"
-    for word2 in $flags_opt_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_strip_word="yes"
       fi
@@ -15301,12 +15301,12 @@ echo "${ECHO_T}$honor_optimize_option" >
       if test "$want_optimize" = "yes"; then
         { echo "$as_me:$LINENO: checking if compiler accepts optimizer enabling options" >&5
 echo $ECHO_N "checking if compiler accepts optimizer enabling options... $ECHO_C" >&6; }
-        tmp_options="$flags_opt_yes"
+        tmp_options=""
       fi
       if test "$want_optimize" = "no"; then
         { echo "$as_me:$LINENO: checking if compiler accepts optimizer disabling options" >&5
 echo $ECHO_N "checking if compiler accepts optimizer disabling options... $ECHO_C" >&6; }
-        tmp_options="$flags_opt_off"
+        tmp_options=""
       fi
       CPPFLAGS="$tmp_CPPFLAGS"
       CFLAGS="$tmp_CFLAGS $tmp_options"
