Add support for python 2.4

Author: Michael Tharp <gxti@partiallystapled.com>

diff --git a/_pylibmcmodule.c b/_pylibmcmodule.c
--- a/_pylibmcmodule.c
+++ b/_pylibmcmodule.c
@@ -33,6 +33,10 @@
 
 #include "_pylibmcmodule.h"
 
+#if PY_VERSION_HEX < 0x02050000
+typedef int Py_ssize_t;
+#endif
+
 /* {{{ _pylibmc.client implementation */
 /* {{{ Type methods */
 static PylibMC_Client *PylibMC_ClientType_new(PyTypeObject *type,
