--- nautilus-dropbox/nautilus-dropbox-0.7.1/configure	2011-12-22 23:43:38.000000000 +0100
+++ nautilus-dropbox/nautilus-dropbox-0.7.1/configure-pygtk	2011-12-22 23:45:53.000000000 +0100
@@ -10796,28 +10796,6 @@
 
 
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for pygtk" >&5
-$as_echo_n "checking for pygtk... " >&6; }
-
-cat <<EOF | python
-try:
- import gtk
-except:
- exit(1)
-else:
- exit(0)
-EOF
-
-if test $? -ne 0; then
-   { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-   as_fn_error "couldn't find pygtk" "$LINENO" 5
-else
-   { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-fi
-
-
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for gobject" >&5
 $as_echo_n "checking for gobject... " >&6; }
 
