diff -Naur rhpxl-0.8.orig/rhpxl/videocard.py rhpxl-0.8/rhpxl/videocard.py
--- rhpxl-0.8.orig/rhpxl/videocard.py	2005-11-21 14:29:32.000000000 -0500
+++ rhpxl-0.8/rhpxl/videocard.py	2005-12-12 16:57:01.000000000 -0500
@@ -78,12 +78,12 @@
 	if string.find(line,'\t') == -1:
 	    continue
 	( driver, desc ) = string.splitfields (line, '\t', 1)
-        if os.access('/usr/%s/xorg/modules/drivers/%s_drv.so' % (libdir, driver), os.R_OK) or os.access('/usr/%s/xorg/modules/drivers/%s_drv.o' % (libdir, driver), os.R_OK):
+        if os.access('/usr/X11R6/lib/modules/drivers/%s_drv.so' % (driver), os.R_OK) or os.access('/usr/X11R6/lib/modules/drivers/%s_drv.o' % (driver), os.R_OK):
 	    drivers[driver] = VideoDriver(driver = driver, description = desc)
 	    
     # Add drivers that do exist that we don't have entries for.
-    if os.access('/usr/%s/xorg/modules/drivers' % (libdir,) , os.R_OK):
-	d = os.listdir('/usr/%s/xorg/modules/drivers' %(libdir,))
+    if os.access('/usr/X11R6/lib/modules/drivers', os.R_OK):
+	d = os.listdir('/usr/X11R6/lib/modules/drivers')
 	    
 	for module in d:
 	    if module.endswith('_drv.so') or module.endswith('_drv.o'):
