--- qt4--qt-copy/tools/qdbus/tools/qdbuscpp2xml/qdbuscpp2xml.cpp.orig	2007-04-24 01:45:45.000000000 +0400
+++ qt4--qt-copy/tools/qdbus/tools/qdbuscpp2xml/qdbuscpp2xml.cpp	2007-08-14 18:27:35.184204487 +0400
@@ -376,7 +376,7 @@
         else {
             // run moc on this file
             QProcess proc;
-            proc.start(QLibraryInfo::location(QLibraryInfo::BinariesPath) + QLatin1String("/moc"), QStringList() << QFile::decodeName(argv[i]), QIODevice::ReadOnly | QIODevice::Text);
+            proc.start(QLibraryInfo::location(QLibraryInfo::BinariesPath) + QLatin1String("/moc-qt4"), QStringList() << QFile::decodeName(argv[i]), QIODevice::ReadOnly | QIODevice::Text);
 
             if (!proc.waitForStarted()) {
                 fprintf(stderr, PROGRAMNAME ": could not execute moc! Aborting.\n");
