--- qt4-4.6.2-14-g0ad4a1f/tools/assistant/tools/assistant/assistant.pro.orig	2010-03-12 12:52:23.968542830 +0300
+++ qt4-4.6.2-14-g0ad4a1f/tools/assistant/tools/assistant/assistant.pro	2010-03-12 12:51:35.202542944 +0300
@@ -20,7 +20,7 @@
 
 ### Work around a qmake issue when statically linking to
 ### not-yet-installed plugins
-LIBS += -L$$QT_BUILD_TREE/plugins/sqldrivers
+LIBS += -L$$QT_BUILD_TREE/plugins/sqldrivers -Wl,-rpath-link,$$QT_BUILD_TREE/lib
 
 HEADERS += helpviewer.h \
            mainwindow.h \
--- qt4-4.6.2-14-g0ad4a1f/tools/assistant/tools/qcollectiongenerator/qcollectiongenerator.pro.orig	2010-03-12 12:54:47.422542077 +0300
+++ qt4-4.6.2-14-g0ad4a1f/tools/assistant/tools/qcollectiongenerator/qcollectiongenerator.pro	2010-03-12 12:55:00.735541698 +0300
@@ -5,6 +5,8 @@
 CONFIG += qt warn_on help console
 CONFIG -= app_bundle
 
+LIBS += -Wl,-rpath-link,$$QT_BUILD_TREE/lib
+
 target.path=$$[QT_INSTALL_BINS]
 INSTALLS += target
 
--- qt4-4.6.2-14-g0ad4a1f/tools/assistant/tools/qhelpconverter/qhelpconverter.pro.orig	2010-03-12 12:58:08.199541715 +0300
+++ qt4-4.6.2-14-g0ad4a1f/tools/assistant/tools/qhelpconverter/qhelpconverter.pro	2010-03-12 12:58:32.237541933 +0300
@@ -4,6 +4,8 @@
 DESTDIR = ../../../../bin
 CONFIG += qt warn_on help
 
+LIBS += -Wl,-rpath-link,$$QT_BUILD_TREE/lib
+
 target.path=$$[QT_INSTALL_BINS]
 INSTALLS += target
 
--- qt4-4.6.2-14-g0ad4a1f/tools/assistant/tools/qhelpgenerator/qhelpgenerator.pro.orig	2010-03-12 12:53:05.695542229 +0300
+++ qt4-4.6.2-14-g0ad4a1f/tools/assistant/tools/qhelpgenerator/qhelpgenerator.pro	2010-03-12 12:53:21.470542301 +0300
@@ -8,6 +8,8 @@
 target.path=$$[QT_INSTALL_BINS]
 INSTALLS += target
 
+LIBS += -Wl,-rpath-link,$$QT_BUILD_TREE/lib
+
 SOURCES += ../shared/helpgenerator.cpp \
            main.cpp
 
