diff -up dovecot-1.0.15/src/plugins/acl/acl-cache.c.cve_2008_4577 dovecot-1.0.15/src/plugins/acl/acl-cache.c
--- dovecot-1.0.15/src/plugins/acl/acl-cache.c.cve_2008_4577	2008-10-29 08:53:23.872610374 +0100
+++ dovecot-1.0.15/src/plugins/acl/acl-cache.c	2008-10-29 08:53:23.873610244 +0100
@@ -375,10 +375,8 @@ acl_cache_my_current_rights_recalculate(
 			   rights. */
 			right_size = obj_cache->my_neg_rights[i]->size;
 			p = buffer_get_space_unsafe(bitmask, 0, right_size);
-			for (j = 0; j < right_size; j++) {
-				p[j] |=
-					obj_cache->my_neg_rights[i]->mask[j];
-			}
+			for (j = 0; j < right_size; j++)
+				p[j] &= ~obj_cache->my_neg_rights[i]->mask[j];
 		}
 	}
 
