diff -ru xfce4-settings-4.6.4.orig/dialogs/keyboard-settings/xfce-keyboard-settings.c xfce4-settings-4.6.4/dialogs/keyboard-settings/xfce-keyboard-settings.c
--- xfce4-settings-4.6.4.orig/dialogs/keyboard-settings/xfce-keyboard-settings.c	2010-01-23 14:58:17.000000000 +0000
+++ xfce4-settings-4.6.4/dialogs/keyboard-settings/xfce-keyboard-settings.c	2010-01-23 14:58:35.000000000 +0000
@@ -457,7 +457,7 @@
 
 #ifdef HAVE_LIBXKLAVIER
   /* Stop xklavier engine */
-  xkl_engine_stop_listen (settings->priv->xkl_engine);
+  xkl_engine_stop_listen (settings->priv->xkl_engine, XKLL_TRACK_KEYBOARD_STATE);
 #endif /* HAVE_LIBXKLAVIER */
 
   g_object_unref (settings->priv->provider);
