--- /images/conary/builds/chkconfig/chkconfig-1.3.32/leveldb.c	2007-02-01 23:49:09.000000000 -0500
+++ /tmp/leveldb.c	2010-03-12 16:30:12.000000000 -0500
@@ -81,7 +81,7 @@
 	    } else ++start;
 
 	    while (isspace(*start) && start < end) start++;
-	    if (start == end) {
+	    if (start >= end) {
 		return 1;
 	    }
           {
@@ -201,6 +201,8 @@
 				if (!serv.desc) {
 					if (eng_desc)
 					  serv.desc = eng_desc;
+					else
+					  serv.desc = strdup(name);
 				} else if (eng_desc)
 					  free (eng_desc);
 			}
@@ -384,7 +386,7 @@
 	}
 
 	if (!strncmp(start, "chkconfig:", 10)) {
-	    int spri, kpri;
+	    int spri, kpri, rc;
 	     
 	    start += 10;
 	    while (isspace(*start) && start < end) start++;
@@ -394,9 +396,9 @@
 		return 1;
 	    }
 
-	    if ((sscanf(start, "%s %d %d%c", levelbuf,
-			&spri, &kpri, &overflow) != 4) ||
-		 !isspace(overflow)) {
+	    rc = sscanf(start, "%19s %d %d%c", levelbuf,
+			&spri, &kpri, &overflow);
+            if (!(rc == 3 || (rc == 4 && isspace(overflow)))) {
 		if (serv.desc) free(serv.desc);
 		free(bufstart);
 		return 1;
@@ -544,6 +546,8 @@
     if (!serv.desc) {
       if (english_desc)
 	serv.desc = english_desc;
+      else
+	serv.desc = strdup(name);
     } else if (english_desc)
 	free (english_desc);
 
