--- chkconfig.c.orig	2008-09-25 15:38:35.000000000 +0300
+++ chkconfig.c	2008-09-25 15:38:50.000000000 +0300
@@ -203,7 +203,6 @@ static void frobDependencies(struct serv
 			continue;
 		}
 		if (!S_ISREG(sb.st_mode)) continue;
-		if (!strcmp(ent->d_name, s->name)) continue;
 		servs = realloc(servs, (numservs+1) * sizeof(struct service));
 		if (!readServiceInfo(ent->d_name, servs + numservs, 0))
 			numservs++;
