--- vim61/runtime/filetype.vim.syntx	Wed Oct  2 05:03:48 2002
+++ vim61/runtime/filetype.vim	Wed Oct  2 05:06:21 2002
@@ -88,7 +88,7 @@
 au BufNewFile,BufRead proftpd.conf*		setf apachestyle
 
 " Apache config file
-au BufNewFile,BufRead httpd.conf*,srm.conf*,access.conf*,.htaccess,apache.conf* setf apache
+au BufNewFile,BufRead httpd.conf*,srm.conf*,access.conf*,.htaccess,apache.conf*,auth_mysql.conf*,auth_pgsql.conf*,ssl.conf*,perl.conf*,php.conf*,python.conf*,squirrelmail.conf* setf apache
 
 " Applix ELF
 au BufNewFile,BufRead *.am
