--- gthumb-2.6.9/src/gthumb-window.c.panes	2006-02-17 03:49:30.000000000 -0500
+++ gthumb-2.6.9/src/gthumb-window.c	2006-07-30 06:41:47.000000000 -0400
@@ -5360,8 +5360,8 @@
 
 	/* image preview, comment, exif data. */
 
-	image_pane_paned1 = gtk_vpaned_new ();
-	gtk_paned_set_position (GTK_PANED (image_pane_paned1), eel_gconf_get_integer (PREF_UI_WINDOW_HEIGHT, DEF_WIN_HEIGHT) / 2);
+	image_pane_paned1 = gtk_hpaned_new ();
+	gtk_paned_set_position (GTK_PANED (image_pane_paned1), eel_gconf_get_integer (PREF_UI_WINDOW_WIDTH, DEF_WIN_WIDTH) * 3 / 4);
 	gtk_box_pack_start (GTK_BOX (image_vbox), image_pane_paned1, TRUE, TRUE, 0);
 
 	/**/
@@ -5387,8 +5387,8 @@
 
 	/**/
 
-	window->preview_widget_data_comment = image_pane_paned2 = gtk_hpaned_new ();
-	gtk_paned_set_position (GTK_PANED (image_pane_paned2), eel_gconf_get_integer (PREF_UI_WINDOW_WIDTH, DEF_WIN_WIDTH) / 2);
+	window->preview_widget_data_comment = image_pane_paned2 = gtk_vpaned_new ();
+	gtk_paned_set_position (GTK_PANED (image_pane_paned2), eel_gconf_get_integer (PREF_UI_WINDOW_HEIGHT, DEF_WIN_HEIGHT) * 3 / 4);
 	gtk_paned_pack2 (GTK_PANED (image_pane_paned1), image_pane_paned2, TRUE, FALSE);
 
 	window->preview_widget_comment = scrolled_win = gtk_scrolled_window_new (NULL, NULL);
