--- grub-0.97/stage2/boot.c.jau	2007-04-13 11:21:13.000000000 -0400
+++ grub-0.97/stage2/boot.c	2007-04-13 11:32:19.000000000 -0400
@@ -403,7 +403,20 @@
 	  {
 	    char *src = skip_to (0, arg);
 	    char *dest = linux_data_tmp_addr + LINUX_CL_OFFSET;
-	
+	    char *boot_image = "BOOT_IMAGE=";
+	    char *boot_kernel = kernel;
+
+	    while (dest < linux_data_tmp_addr + LINUX_CL_END_OFFSET &&
+		   *boot_image)
+	      *(dest++) = *(boot_image++);
+
+	    while (dest < linux_data_tmp_addr + LINUX_CL_END_OFFSET &&
+		   *boot_kernel && *boot_kernel != ' ')
+	      *(dest++) = *(boot_kernel++);
+
+	    if (dest < linux_data_tmp_addr + LINUX_CL_END_OFFSET)
+	      *(dest++) = ' ';
+
 	    while (dest < linux_data_tmp_addr + LINUX_CL_END_OFFSET && *src)
 	      *(dest++) = *(src++);
 	
