--- src/core/input.c.orig	2008-05-20 18:21:27.000000000 +0300
+++ src/core/input.c	2008-05-20 18:23:58.000000000 +0300
@@ -454,7 +454,9 @@ dfb_input_core_shutdown( DFBInputCore *d
           fusion_call_destroy( &devshared->call );
           fusion_skirmish_destroy( &devshared->lock );
 
-          driver->funcs->CloseDevice( device->driver_data );
+          if (device->driver_data) { // be extra safe
+               driver->funcs->CloseDevice( device->driver_data );
+          }
 
           if (!--driver->nr_devices) {
                direct_module_unref( driver->module );
@@ -535,7 +537,10 @@ dfb_input_core_suspend( DFBInputCore *da
                       driver->info.version.major,
                       driver->info.version.minor, driver->info.vendor );
 
-          driver->funcs->CloseDevice( device->driver_data );
+          if (device->driver_data) { // be extra safe
+              driver->funcs->CloseDevice( device->driver_data );
+              device->driver_data = NULL; // freed by CloseDevice()
+          }
 
           flush_keys( device );
      }
