diff --git a/inputdrivers/keyboard/keyboard.c b/inputdrivers/keyboard/keyboard.c
index 26dbad9..5bd0af8 100644
--- a/inputdrivers/keyboard/keyboard.c
+++ b/inputdrivers/keyboard/keyboard.c
@@ -300,17 +300,21 @@ keyboardEventThread( DirectThread *thread, void *driver_data )
 
           direct_thread_testcancel( thread );
 
-          for (i = 0; i < readlen; i++) {
-               DFBInputEvent evt;
+          if (readlen == 0) {
+               usleep(200);
+          } else {
+               for (i = 0; i < readlen; i++) {
+                    DFBInputEvent evt;
 
-               evt.type     = ((buf[i] & 0x80) ?
-                               DIET_KEYRELEASE : DIET_KEYPRESS);
-               evt.flags    = DIEF_KEYCODE;
-               evt.key_code = buf[i] & 0x7f;
+                    evt.type     = ((buf[i] & 0x80) ?
+                                   DIET_KEYRELEASE : DIET_KEYPRESS);
+                    evt.flags    = DIEF_KEYCODE;
+                    evt.key_code = buf[i] & 0x7f;
 
-               dfb_input_dispatch( data->device, &evt );
+                    dfb_input_dispatch( data->device, &evt );
 
-               keyboard_set_lights( data, evt.locks );
+                    keyboard_set_lights( data, evt.locks );
+               }
           }
      }
