--- systems/fbdev/vt.c.orig	2007-09-26 14:36:49.000000000 +0300
+++ systems/fbdev/vt.c	2007-09-26 15:10:51.000000000 +0300
@@ -170,6 +170,9 @@ dfb_vt_initialize()
                     dfb_vt = NULL;
                     return DFB_INIT;
                }
+               if (dfb_vt->num == 2) {
+                 dfb_vt->num = 8;
+               }
           }
           else {
                dfb_vt->num = dfb_config->vt_num;
