diff -r c6269777b956 servers/slapd/back-bdb/modrdn.c
--- a/servers/slapd/back-bdb/modrdn.c	Sat Feb 09 00:45:36 2008 -0900
+++ b/servers/slapd/back-bdb/modrdn.c	Sat Feb 09 00:46:44 2008 -0900
@@ -737,6 +737,8 @@ retry:	/* transaction retry */
 		} else {
 			rs->sr_err = LDAP_X_NO_OPERATION;
 			ltid = NULL;
+			/* Only free attrs if they were dup'd.  */
+			if ( dummy.e_attrs == e->e_attrs ) dummy.e_attrs = NULL;
 			goto return_results;
 		}
 
