diff -uNr e2fsimage-0.2.2.orig/src/copy.c e2fsimage-0.2.2/src/copy.c
--- e2fsimage-0.2.2.orig/src/copy.c	2006-02-06 04:27:05.497446750 -0500
+++ e2fsimage-0.2.2/src/copy.c	2006-02-06 05:42:14.959270500 -0500
@@ -85,7 +85,8 @@
 	/* open the source file */
 	fp = fopen(e2c->curr_path, "r");
 	/* unlikely, cause we stated it just some msec before... */
-	ERRNO_ERR(ret, "Error opening: ", e2c->curr_path);
+	if (!fp)
+		ERRNO_ERR(errno, "Error opening: ", e2c->curr_path);
 
 	/* read the input data and write it to the e2 file */
 	ptr = e2c->cp_buf;

