diff -uNr e2fsimage-0.2.2.orig/src/e2fsimage.h e2fsimage-0.2.2/src/e2fsimage.h
--- e2fsimage-0.2.2.orig/src/e2fsimage.h	2006-02-06 15:50:24.002708750 -0500
+++ e2fsimage-0.2.2/src/e2fsimage.h	2006-02-06 15:37:55.591936000 -0500
@@ -144,7 +144,9 @@
 int read_special_file(e2i_ctx_t *e2c);
 
 /* functions from util.c */
+#ifndef _GNU_SOURCE
 const char *basename(const char *path);
+#endif
 void init_inode(e2i_ctx_t *e2c, struct ext2_inode *i, struct stat *s);
 int e2link(e2i_ctx_t *e2c, const char *fname, ext2_ino_t e2ino, int mode);
 __u16 mode2filetype(mode_t m);
diff -uNr e2fsimage-0.2.2.orig/src/mke2fs.c e2fsimage-0.2.2/src/mke2fs.c
--- e2fsimage-0.2.2.orig/src/mke2fs.c	2006-01-11 17:08:58.000000000 -0500
+++ e2fsimage-0.2.2/src/mke2fs.c	2006-02-06 14:35:43.102670000 -0500
@@ -69,7 +69,7 @@
 	}
 	memset(buf, 0, 1024 );
 
-	fseek (fp,(size-1)*1024,SEEK_SET);
+	fseeko (fp,(off_t)(size-1)*1024,SEEK_SET);
 	fwrite (buf,1024,1,fp);
 	fclose(fp);
 
diff -uNr e2fsimage-0.2.2.orig/src/util.c e2fsimage-0.2.2/src/util.c
--- e2fsimage-0.2.2.orig/src/util.c	2006-02-06 15:50:24.006709000 -0500
+++ e2fsimage-0.2.2/src/util.c	2006-02-06 15:38:19.905455500 -0500
@@ -42,6 +42,7 @@
 #include "e2fsimage.h"
 #include <string.h>
 
+#ifndef _GNU_SOURCE
 const char *basename(const char *path)
 {
 	const char *bn;
@@ -56,6 +57,7 @@
 	/* now ptr points to the basename of 'path' */
 	return bn;
 }
+#endif
 
 int filename(char *str, char *estr, char ** ptrptr, char ** eptrptr)
 {
