--- minicom-2.00.0/src/vt100.c.orig	2003-08-20 16:59:03.000000000 -0400
+++ minicom-2.00.0/src/vt100.c	2003-08-20 17:01:48.000000000 -0400
@@ -268,6 +268,8 @@
   vt_charset = 0;
   vt_trans[0] = savetrans[0] = vt_map[0];
   vt_trans[1] = savetrans[1] = vt_map[1];
+  savetrans[0] = vt_trans[0];
+  savetrans[1] = vt_trans[1];
 #endif
 #if OLD
   ptr = -2;
@@ -1021,7 +1023,7 @@
 			fputc(P_CONVCAP[0]=='Y'? vt_inmap[c]: c, vt_capfp);
 	        c = vt_inmap[c];    /* conversion 04.09.97 / jl */
 #if TRANSLATE
-		if (vt_type == VT100 && vt_asis == 0) {
+		if (vt_type == VT100 && vt_trans[vt_charset] && vt_asis == 0) {
 			if (vt_insert)
 				winschar2(vt_win, vt_trans[vt_charset][c], 1);
 			else
