--- src/nm-netlink-monitor.c.orig	2006-11-14 22:44:25.000000000 +0100
+++ src/nm-netlink-monitor.c	2006-11-14 22:47:07.000000000 +0100
@@ -49,6 +49,15 @@
 #define NM_NETLINK_MONITOR_DISCONNECT_CONDITIONS \
 	((GIOCondition) (G_IO_HUP))
 
+#include <linux/if_addr.h>
+#ifndef IFLA_RTA
+#define IFLA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifinfomsg))))
+#endif
+
+#ifndef IFLA_PAYLOAD
+#define IFLA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifaddrmsg))
+#endif
+
 struct _NmNetlinkMonitorPrivate 
 {
 	GMainContext	*context;
