diff -Nru network-manager-0.6.4/gnome/applet/applet.c network-manager-0.6.4.static/gnome/applet/applet.c
--- gnome/applet/applet.c	2006-07-10 22:30:27.000000000 +0200
+++ gnome/applet/applet.c	2007-01-17 15:52:54.000000000 +0100
@@ -1252,6 +1252,23 @@
 	return (found_dev);
 }
 
+/*
+ * nma_menu_network_admin_activate
+ *
+ * Signal function called when user clicks "Static configuration..."
+ *
+ */
+static void nma_menu_network_admin_activate (GtkMenuItem *item, gpointer user_data)
+{
+	NMApplet	*applet = (NMApplet *)user_data;
+	const char *argv[] = { "network-admin", NULL};
+
+	g_return_if_fail (item != NULL);
+	g_return_if_fail (applet != NULL);
+
+	g_spawn_async ("/", argv, NULL, G_SPAWN_SEARCH_PATH, NULL, NULL, NULL, NULL);
+}
+
 
 /*
  * nma_menu_item_activate
@@ -1662,6 +1679,25 @@
 	g_free (add_networks_cb);
 }
 
+/*
+ * nma_menu_add_devices
+ *
+ */
+static void nma_menu_add_network_admin_item (GtkWidget *menu, NMApplet *applet)
+{
+	GtkMenuItem	*item;
+	GSList		*elt;
+
+	g_return_if_fail (menu != NULL);
+	g_return_if_fail (applet != NULL);
+
+	item = GTK_MENU_ITEM (gtk_menu_item_new_with_mnemonic (_("_Static configuration...")));
+
+	g_signal_connect (G_OBJECT (item), "activate", G_CALLBACK (nma_menu_network_admin_activate), applet);
+	gtk_menu_shell_append (GTK_MENU_SHELL (menu), GTK_WIDGET (item));
+	gtk_widget_show_all (GTK_WIDGET (item));
+}
+
 
 /*
  * nma_menu_add_devices
@@ -1903,6 +1930,7 @@
 		nma_menu_add_custom_essid_item (menu, applet);
 		nma_menu_add_create_network_item (menu, applet);
 	}
+	nma_menu_add_network_admin_item (menu, applet);
 }
 
 
