diff -Nur network-manager-0.6.4/src/backends/NetworkManagerDebian.c network-manager-0.6.4.new/src/backends/NetworkManagerDebian.c
--- network-manager-0.6.4/src/backends/NetworkManagerDebian.c	2007-03-30 18:09:56.000000000 +0200
+++ network-manager-0.6.4.new/src/backends/NetworkManagerDebian.c	2007-03-30 18:23:54.000000000 +0200
@@ -556,6 +556,28 @@
 			blacklist = FALSE;
 			}
 
+	/* handle some ipv6 situations */
+
+	curr_b = curr_device;
+
+	/* NM does not know how to handle ipv6 interfaces */
+	if (ifparser_getkey(curr_b, "inet6")) {
+		blacklist = TRUE;
+		goto out;
+	}
+
+	/* if we have a have an inet dhcp interface and later an inet6 stanza
+	 * we blacklist it.
+	 */
+	if (strcmp (ifparser_getkey(curr_b, "inet"),"dhcp") == 0) {
+		while (curr_b->nextsame != 0) {
+			curr_b = curr_b->nextsame;
+			if (ifparser_getkey(curr_b, "inet6")) {
+				blacklist = TRUE;
+			}
+		}
+	}
+
 	/* If the interface has no options other than just "inet dhcp"
 	 * it's probably ok to fiddle with it. [DISABLED]
 	 */
