diff -ur NetworkManager-0.4.orig/gnome/libnm_glib/libnm_glib.c NetworkManager-0.4/gnome/libnm_glib/libnm_glib.c
--- NetworkManager-0.4.orig/gnome/libnm_glib/libnm_glib.c	2005-05-05 17:04:24.000000000 -0400
+++ NetworkManager-0.4/gnome/libnm_glib/libnm_glib.c	2005-07-14 14:56:18.000000000 -0400
@@ -200,7 +200,7 @@
 		dbus_connection_disconnect (ctx->dbus_con);
 		libnm_glib_schedule_dbus_watcher (ctx);
 	}
-#if ((DBUS_VERSION_MAJOR == 0) && ((DBUS_VERSION_MINOR == 30) || (DBUS_VERSION_MINOR == 31) || (DBUS_VERSION_MINOR == 32) || (DBUS_VERSION_MINOR == 33)))
+#if ((DBUS_VERSION_MAJOR == 0) && ((DBUS_VERSION_MINOR == 30) || (DBUS_VERSION_MINOR == 31) || (DBUS_VERSION_MINOR == 32) || (DBUS_VERSION_MINOR == 33) || (DBUS_VERSION_MINOR == 34)))
 	else if (dbus_message_is_signal (message, DBUS_INTERFACE_DBUS, "NameOwnerChanged"))
 	{
 		/* New signal for dbus 0.23... */
