--- NetworkManager-0.6.3-orig/src/nm-device-802-11-wireless.c 2006-06-07 23:30:25.000000000 +0600
+++ NetworkManager-0.6.3/src/nm-device-802-11-wireless.c 2006-06-09 20:05:21.000000000 +0600
@@ -2424,16 +2424,15 @@
 supplicant_exec (NMDevice80211Wireless *self)
 {
 	gboolean	success = FALSE;
-	char *	argv[5];
+	char *	argv[4];
 	GError *	error = NULL;
 	GPid		pid = -1;
 	int		sup_stdout;
 
 	argv[0] = WPA_SUPPLICANT_BIN;
-	argv[1] = "-dd";
-	argv[2] = "-g";
-	argv[3] = WPA_SUPPLICANT_GLOBAL_SOCKET;
-	argv[4] = NULL;
+	argv[1] = "-g";
+	argv[2] = WPA_SUPPLICANT_GLOBAL_SOCKET;
+	argv[3] = NULL;
 
 	success = g_spawn_async_with_pipes ("/", argv, NULL, 0, NULL, NULL, &pid, NULL, &sup_stdout, NULL, &error);
