Index: src/NetworkManager.c
===================================================================
--- src/NetworkManager.c	(revision 3790)
+++ src/NetworkManager.c	(working copy)
@@ -116,8 +116,10 @@
 		case SIGBUS:
 		case SIGILL:
 		case SIGABRT:
+                        /*
 			nm_warning ("Caught signal %d.  Generating backtrace...", signo);
 			nm_logging_backtrace ();
+                        */
 			exit (1);
 			break;
 
@@ -126,8 +128,10 @@
 			/* let the fatal signals interrupt us */
 			--in_fatal;
 
+                        /*
 			nm_warning ("Caught signal %d, shutting down abnormally.  Generating backtrace...", signo);
 			nm_logging_backtrace ();
+                        */
 			g_main_loop_quit (main_loop);
 			break;
 
@@ -136,8 +140,10 @@
 			/* let the fatal signals interrupt us */
 			--in_fatal;
 
+                        /*
 			nm_warning ("Caught signal %d, shutting down normally.", signo);
 			quit_early = TRUE;
+                        */
 			g_main_loop_quit (main_loop);
 			break;
 
