--- valgrind-3.1.1/coregrind/m_syswrap/syswrap-amd64-linux.c.orig       2006-06-07 10:39:04.733043500 -0400
+++ valgrind-3.1.1/coregrind/m_syswrap/syswrap-amd64-linux.c    2006-06-07 12:26:35.652200750 -0400
@@ -362,6 +362,7 @@
 DECL_TEMPLATE(amd64_linux, sys_pwrite64);
 DECL_TEMPLATE(amd64_linux, sys_fadvise64);
 DECL_TEMPLATE(amd64_linux, sys_mmap);
+DECL_TEMPLATE(amd64_linux, sys_flock);


 PRE(sys_clone)
@@ -943,6 +944,13 @@
    SET_STATUS_from_SysRes(r);
 }

+PRE(sys_flock)
+{
+   *flags |= SfMayBlock;
+   PRINT("sys_flock ( %d, %d )", ARG1, ARG2);
+   PRE_REG_READ2(int, "flock", int, fd, int, operation);
+}
+
 #undef PRE
 #undef POST


