diff -r ca2786576ab8 Makefile.am
--- a/Makefile.am	Tue Mar 03 16:52:42 2009 -0500
+++ b/Makefile.am	Tue Mar 03 17:24:53 2009 -0500
@@ -76,9 +76,7 @@
 NTOPDATA = ntop-cert.pem \
            $(ETTER_PASSIVE) \
            oui.txt.gz \
-           specialMAC.txt.gz \
-	   GeoIPASNum.dat \
-	   GeoLiteCity.dat
+           specialMAC.txt.gz
 
 NTOPHTML = html html/*.js html/*.html  html/*.gif html/*.jpg html/*.ico html/*.png \
            html/*.css html/*.dtd \
diff -r ca2786576ab8 configure.in
--- a/configure.in	Tue Mar 03 16:52:42 2009 -0500
+++ b/configure.in	Tue Mar 03 17:24:53 2009 -0500
@@ -1887,20 +1887,6 @@
 fi
 
 
-if test -f "GeoLiteCity.dat"; then
-   echo "GeoLiteCity.dat already present"
-else
-   wget http://www.maxmind.com/download/geoip/database/GeoLiteCity.dat.gz
-   gunzip GeoLiteCity.dat.gz
-fi
-
-if test -f "GeoIPASNum.dat"; then
-   echo "GeoIPASNum.dat already present"
-else
-   wget http://www.maxmind.com/download/geoip/database/asnum/GeoIPASNum.dat.gz
-   gunzip GeoIPASNum.dat.gz
-fi
-
 LDFLAGS="$LDFLAGS -lGeoIP"
 
 dnl> NTOPCONFIGDEBUG_SETTINGS([precet])
diff -r ca2786576ab8 globals-core.c
--- a/globals-core.c	Tue Mar 03 16:52:42 2009 -0500
+++ b/globals-core.c	Tue Mar 03 17:24:53 2009 -0500
@@ -68,6 +68,7 @@
 #else
 				   "/etc",
 #endif
+                                   "/usr/share/GeoIP",
 				   NULL };
 
 /*
