--- pinfo-0.6.8/src/video.c.bak	2002-07-21 21:42:00.000000000 +0300
+++ pinfo-0.6.8/src/video.c	2004-11-11 03:34:41.634249068 +0200
@@ -67,6 +67,7 @@ showscreen (char **message, char *type, 
   for (i = pos; (i < lines) && (i < pos + maxy - 2); i++)
     {
       int tmp = strlen (message[i]) - 1;
+      if (tmp == -1) continue;
       message[i][tmp] = 0;
       if(tmp>column)
         mvaddstr (i + 1 - pos, 0, message[i]+column);
