Index: src/gpm-control.c
===================================================================
--- src/gpm-control.c	(revision 3009)
+++ src/gpm-control.c	(working copy)
@@ -225,7 +225,7 @@
 	g_return_val_if_fail (can, FALSE);
 	*can = FALSE;
 	fg = gpm_control_check_foreground_console (control);
-	polkit_ok = gpm_control_is_user_privileged (control, "hal-power-shutdown");
+	polkit_ok = gpm_control_is_user_privileged (control, "org.freedesktop.hal.power-management.shutdown");
 	if (polkit_ok && fg) {
 		*can = TRUE;
 	}
@@ -249,7 +249,7 @@
 	g_return_val_if_fail (can, FALSE);
 	*can = FALSE;
 	fg = gpm_control_check_foreground_console (control);
-	polkit_ok = gpm_control_is_user_privileged (control, "hal-power-reboot");
+	polkit_ok = gpm_control_is_user_privileged (control, "org.freedesktop.hal.power-management.reboot");
 	if (polkit_ok && fg) {
 		*can = TRUE;
 	}
Index: ChangeLog
===================================================================
--- ChangeLog	(revision 3010)
+++ ChangeLog	(working copy)
@@ -1,5 +1,12 @@
 2008-10-06  Richard Hughes  <richard@hughsie.com>
 
+	* src/gpm-control.c: (gpm_control_allowed_shutdown),
+	(gpm_control_allowed_reboot):
+	Correct the name of the HAL PolicyKit methods now the deprecated names
+	are no longer valid.
+
+2008-10-06  Richard Hughes  <richard@hughsie.com>
+
 	* src/egg-color.c: (egg_color_from_rgb), (egg_color_to_rgb),
 	(egg_color_test):
 	* src/egg-test.c: (egg_test_assert), (egg_test_title_assert):
