--- libgnome-2.9.1/schemas/desktop_gnome_background.schemas.in.orig	2005-02-12 21:15:18.989330880 -0500
+++ libgnome-2.9.1/schemas/desktop_gnome_background.schemas.in	2005-02-12 21:14:06.008425664 -0500
@@ -29,7 +29,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default></default>
+	<default>/usr/share/pixmaps/backgrounds/default.png</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image</long>
@@ -51,7 +51,7 @@
         <applyto>/desktop/gnome/background/primary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#76848F</default>
+        <default>#404760</default>
         <locale name="C">
         <short>Primary Color</short>
         <long>Left or Top color when drawing gradients, or the solid color.</long>
@@ -62,7 +62,7 @@
         <applyto>/desktop/gnome/background/secondary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#7F7F7F</default>
+        <default>#435699</default>
         <locale name="C">
         <short>Secondary Color</short>
         <long>Right or Bottom color when drawing gradients, not used for solid color.</long>
