diff -Naur libgnome-2.14.1.old/schemas/desktop_gnome_interface.schemas libgnome-2.14.1/schemas/desktop_gnome_interface.schemas
--- libgnome-2.14.1.old/schemas/desktop_gnome_interface.schemas	2006-04-10 10:40:25.000000000 -0400
+++ libgnome-2.14.1/schemas/desktop_gnome_interface.schemas	2006-06-14 22:12:17.000000000 -0400
@@ -5905,7 +5905,7 @@
       <key>/schemas/desktop/gnome/interface/document_font_name</key>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>Trebuchet MS 10<</default>
       <locale name="C">
 	<short>Document font</short>
 	<long>Name of the default font used for reading documents</long>
diff -Naur libgnome-2.14.1.old/schemas/desktop_gnome_interface.schemas.in libgnome-2.14.1/schemas/desktop_gnome_interface.schemas.in
--- libgnome-2.14.1.old/schemas/desktop_gnome_interface.schemas.in	2006-01-11 02:01:48.000000000 -0500
+++ libgnome-2.14.1/schemas/desktop_gnome_interface.schemas.in	2006-06-14 22:13:25.000000000 -0400
@@ -196,7 +196,7 @@
       <key>/schemas/desktop/gnome/interface/document_font_name</key>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>Trebuchet MS 10</default>
       <locale name="C">
 	<short>Document font</short>
 	<long>Name of the default font used for reading documents</long>
