diff -Naur libgnome-2.12.0.1.orig/schemas/desktop_gnome_background.schemas libgnome-2.12.0.1/schemas/desktop_gnome_background.schemas
--- libgnome-2.12.0.1.orig/schemas/desktop_gnome_background.schemas	2005-09-07 11:29:01.000000000 -0400
+++ libgnome-2.12.0.1/schemas/desktop_gnome_background.schemas	2005-09-09 11:09:44.000000000 -0400
@@ -331,7 +331,7 @@
         <applyto>/desktop/gnome/background/picture_options</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>wallpaper</default>
+        <default>scaled</default>
         <locale name="C">
         <short>Picture Options</short>
         <long>Determines how the image set by wallpaper_filename is
@@ -674,7 +674,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default></default>
+	<default>/usr/share/pixmaps/backgrounds/gnome/translucent/logo-foresight-trans.png</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image</long>
@@ -1005,7 +1005,7 @@
         <applyto>/desktop/gnome/background/picture_opacity</applyto>
         <owner>gnome</owner>
         <type>int</type>
-	<default>100</default>
+	<default>50</default>
         <locale name="C">
         <short>Picture Opacity</short>
         <long>Opacity with which to draw the background picture</long>
@@ -1326,7 +1326,7 @@
         <applyto>/desktop/gnome/background/primary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#76848F</default>
+        <default>#679167</default>
         <locale name="C">
         <short>Primary Color</short>
         <long>Left or Top color when drawing gradients, or the solid color.</long>
@@ -1662,7 +1662,7 @@
         <applyto>/desktop/gnome/background/secondary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#7F7F7F</default>
+        <default>#679167</default>
         <locale name="C">
         <short>Secondary Color</short>
         <long>Right or Bottom color when drawing gradients, not used for solid color.</long>
diff -Naur libgnome-2.12.0.1.orig/schemas/desktop_gnome_background.schemas.in libgnome-2.12.0.1/schemas/desktop_gnome_background.schemas.in
--- libgnome-2.12.0.1.orig/schemas/desktop_gnome_background.schemas.in	2003-10-27 09:49:19.000000000 -0500
+++ libgnome-2.12.0.1/schemas/desktop_gnome_background.schemas.in	2005-09-09 11:08:48.000000000 -0400
@@ -16,7 +16,7 @@
         <applyto>/desktop/gnome/background/picture_options</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>wallpaper</default>
+        <default>scaled</default>
         <locale name="C">
         <short>Picture Options</short>
         <long>Determines how the image set by wallpaper_filename is
@@ -29,7 +29,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default></default>
+	<default>/usr/share/pixmaps/backgrounds/gnome/translucent/logo-foresight-trans.png</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image</long>
@@ -40,7 +40,7 @@
         <applyto>/desktop/gnome/background/picture_opacity</applyto>
         <owner>gnome</owner>
         <type>int</type>
-	<default>100</default>
+	<default>50</default>
         <locale name="C">
         <short>Picture Opacity</short>
         <long>Opacity with which to draw the background picture</long>
@@ -51,7 +51,7 @@
         <applyto>/desktop/gnome/background/primary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#76848F</default>
+        <default>#679167</default>
         <locale name="C">
         <short>Primary Color</short>
         <long>Left or Top color when drawing gradients, or the solid color.</long>
@@ -62,7 +62,7 @@
         <applyto>/desktop/gnome/background/secondary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#7F7F7F</default>
+        <default>#679167</default>
         <locale name="C">
         <short>Secondary Color</short>
         <long>Right or Bottom color when drawing gradients, not used for solid color.</long>
diff -Naur libgnome-2.12.0.1.orig/schemas/desktop_gnome_interface.schemas libgnome-2.12.0.1/schemas/desktop_gnome_interface.schemas
--- libgnome-2.12.0.1.orig/schemas/desktop_gnome_interface.schemas	2005-09-07 11:29:01.000000000 -0400
+++ libgnome-2.12.0.1/schemas/desktop_gnome_interface.schemas	2005-09-09 11:02:37.000000000 -0400
@@ -3629,7 +3629,7 @@
       <applyto>/desktop/gnome/interface/icon_theme</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>gnome</default>
+      <default>nuoveXT-gnome-1.5</default>
       <locale name="C">
         <short>Icon Theme</short>
         <long>Icon theme to use for the panel, nautilus etc.</long>
diff -Naur libgnome-2.12.0.1.orig/schemas/desktop_gnome_interface.schemas.in libgnome-2.12.0.1/schemas/desktop_gnome_interface.schemas.in
--- libgnome-2.12.0.1.orig/schemas/desktop_gnome_interface.schemas.in	2005-08-01 14:00:43.000000000 -0400
+++ libgnome-2.12.0.1/schemas/desktop_gnome_interface.schemas.in	2005-09-09 11:02:23.000000000 -0400
@@ -130,7 +130,7 @@
       <applyto>/desktop/gnome/interface/icon_theme</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>gnome</default>
+      <default>nuoveXT-gnome-1.5</default>
       <locale name="C">
         <short>Icon Theme</short>
         <long>Icon theme to use for the panel, nautilus etc.</long>
