diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_background.schemas libgnome-2.20.0/schemas/desktop_gnome_background.schemas
--- libgnome-2.20.0.orig/schemas/desktop_gnome_background.schemas	2007-09-17 08:25:52.000000000 -0400
+++ libgnome-2.20.0/schemas/desktop_gnome_background.schemas	2007-09-17 17:15:11.000000000 -0400
@@ -780,7 +780,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default>/usr/share/pixmaps/backgrounds/gnome/background-default.jpg</default>
+	<default>/usr/share/pixmaps/backgrounds/gnome/foresight/foresight_green.png</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image</long>
@@ -1537,7 +1537,7 @@
         <applyto>/desktop/gnome/background/primary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#66ba00</default>
+        <default>#73d216</default>
         <locale name="C">
         <short>Primary Color</short>
         <long>Left or Top color when drawing gradients, or the solid color.</long>
@@ -1923,7 +1923,7 @@
         <applyto>/desktop/gnome/background/secondary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#66ba00</default>
+        <default>#73d216</default>
         <locale name="C">
         <short>Secondary Color</short>
         <long>Right or Bottom color when drawing gradients, not used for solid color.</long>
diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_background.schemas.in libgnome-2.20.0/schemas/desktop_gnome_background.schemas.in
--- libgnome-2.20.0.orig/schemas/desktop_gnome_background.schemas.in	2007-09-17 08:25:50.000000000 -0400
+++ libgnome-2.20.0/schemas/desktop_gnome_background.schemas.in	2007-09-17 17:13:48.000000000 -0400
@@ -29,7 +29,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default>/usr/share/pixmaps/backgrounds/gnome/background-default.jpg</default>
+	<default>/usr/share/pixmaps/backgrounds/gnome/foresight/foresight_green.png</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image</long>
@@ -51,7 +51,7 @@
         <applyto>/desktop/gnome/background/primary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#66ba00</default>
+        <default>#73d216</default>
         <locale name="C">
         <short>Primary Color</short>
         <long>Left or Top color when drawing gradients, or the solid color.</long>
@@ -62,7 +62,7 @@
         <applyto>/desktop/gnome/background/secondary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#66ba00</default>
+        <default>#73d216</default>
         <locale name="C">
         <short>Secondary Color</short>
         <long>Right or Bottom color when drawing gradients, not used for solid color.</long>
