diff -Naur libgnome-2.18.0.orig/schemas/desktop_gnome_background.schemas libgnome-2.18.0/schemas/desktop_gnome_background.schemas
--- libgnome-2.18.0.orig/schemas/desktop_gnome_background.schemas	2007-03-12 07:25:29.000000000 -0400
+++ libgnome-2.18.0/schemas/desktop_gnome_background.schemas	2007-04-06 10:48:40.000000000 -0400
@@ -371,7 +371,7 @@
         <applyto>/desktop/gnome/background/picture_options</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>wallpaper</default>
+        <default>stretched</default>
         <locale name="C">
         <short>Picture Options</short>
         <long>Determines how the image set by wallpaper_filename is
@@ -754,7 +754,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default></default>
+	<default>/usr/share/pixmaps/backgrounds/gnome/foresight/foresight_green.png</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image</long>
@@ -1481,7 +1481,7 @@
         <applyto>/desktop/gnome/background/primary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#76848F</default>
+        <default>#4e9a06</default>
         <locale name="C">
         <short>Primary Color</short>
         <long>Left or Top color when drawing gradients, or the solid color.</long>
@@ -1852,7 +1852,7 @@
         <applyto>/desktop/gnome/background/secondary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#7F7F7F</default>
+        <default>#4e9a06</default>
         <locale name="C">
         <short>Secondary Color</short>
         <long>Right or Bottom color when drawing gradients, not used for solid color.</long>
diff -Naur libgnome-2.18.0.orig/schemas/desktop_gnome_background.schemas.in libgnome-2.18.0/schemas/desktop_gnome_background.schemas.in
--- libgnome-2.18.0.orig/schemas/desktop_gnome_background.schemas.in	2007-01-01 18:57:13.000000000 -0500
+++ libgnome-2.18.0/schemas/desktop_gnome_background.schemas.in	2007-04-06 10:48:24.000000000 -0400
@@ -16,7 +16,7 @@
         <applyto>/desktop/gnome/background/picture_options</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>wallpaper</default>
+        <default>stretched</default>
         <locale name="C">
         <short>Picture Options</short>
         <long>Determines how the image set by wallpaper_filename is
@@ -29,7 +29,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default></default>
+	<default>/usr/share/pixmaps/backgrounds/gnome/foresight/foresight_green.png</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image</long>
@@ -51,7 +51,7 @@
         <applyto>/desktop/gnome/background/primary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#76848F</default>
+        <default>#4e9a06</default>
         <locale name="C">
         <short>Primary Color</short>
         <long>Left or Top color when drawing gradients, or the solid color.</long>
@@ -62,7 +62,7 @@
         <applyto>/desktop/gnome/background/secondary_color</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>#7F7F7F</default>
+        <default>#4e9a06</default>
         <locale name="C">
         <short>Secondary Color</short>
         <long>Right or Bottom color when drawing gradients, not used for solid color.</long>
