--- configure.in	2008-12-24 16:04:17.000000000 -0500
+++ configure.in	2009-02-10 09:31:36.000000000 -0500
@@ -248,6 +248,10 @@
 	AC_SEARCH_LIBS(Py_Main, $PyName python, : , python="no")
       fi
     fi
+
+    # check for openpty and forkpty
+    AC_CHECK_FUNCS(openpty,, AC_CHECK_LIB(util,openpty, [AC_DEFINE(HAVE_OPENPTY)] [LIBS="$LIBS -lutil"]))
+    AC_CHECK_FUNCS(forkpty,, AC_CHECK_LIB(util,forkpty, [AC_DEFINE(HAVE_FORKPTY)] [LIBS="$LIBS -lutil"]))
   else
     python="no"
   fi
