--- kipi-plugins-0.1.3/kipi-plugins/ipodexport/Makefile.in.orig	2007-01-25 00:04:37.000000000 +0300
+++ kipi-plugins-0.1.3/kipi-plugins/ipodexport/Makefile.in	2007-05-03 07:12:55.560694749 +0400
@@ -401,7 +401,7 @@
                                    ipodlistitem.cpp \
                                    imagelist.cpp
 
-kipiplugin_ipodexport_la_LIBADD = $(LIBKIPI_LIBS) $(LIB_KIO) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT)
+kipiplugin_ipodexport_la_LIBADD = -lgobject-2.0 $(LIBKIPI_LIBS) $(LIB_KIO) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT)
 kipiplugin_ipodexport_la_LDFLAGS = -module $(KDE_PLUGIN) $(LIBGPOD_LIBS) $(all_libraries)
 kde_services_DATA = kipiplugin_ipodexport.desktop
 #>- all: all-am
