diff -r 369ee9cd24b6 twisted/spread/pb.py
--- a/twisted/spread/pb.py	Mon Jun 09 14:05:38 2008 -0400
+++ b/twisted/spread/pb.py	Mon Jun 09 14:07:56 2008 -0400
@@ -1056,7 +1056,7 @@
         """
         Build the broker instance, passing the security options to it.
         """
-        p = self.protocol(isClient=True, security=self.security)
+        p = self.protocol()
         p.factory = self
         return p
 
@@ -1206,7 +1206,7 @@
         """
         Return a Broker attached to the factory (as the service provider).
         """
-        proto = self.protocol(isClient=False, security=self.security)
+        proto = self.protocol()
         proto.factory = self
         proto.setNameForLocal("root", self.root.rootObject(proto))
         return proto
