diff -ur metacity-2.9.21.old/src/metacity.schemas metacity-2.9.21/src/metacity.schemas
--- metacity-2.9.21.old/src/metacity.schemas	2005-02-21 21:55:30.000000000 -0500
+++ metacity-2.9.21/src/metacity.schemas	2005-03-02 15:32:38.308635856 -0500
@@ -1799,7 +1799,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Simple</default>
+      <default>Glider</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
