--- util-linux-2.11y/fdisk/fdisk.c.sopwith	2003-01-13 16:41:36.000000000 -0500
+++ util-linux-2.11y/fdisk/fdisk.c	2003-01-13 16:41:57.000000000 -0500
@@ -2494,7 +2494,7 @@
 	}
 
 	if (opts) {
-		long size;
+		unsigned long size;
 
 		nowarn = 1;
 		type_open = O_RDONLY;
@@ -2511,9 +2511,9 @@
 				fatal(ioctl_error);
 			close(fd);
 			if (opts == 1)
-				printf("%ld\n", size/2);
+				printf("%lu\n", size/2);
 			else
-				printf("%s: %ld\n", argv[j], size/2);
+				printf("%s: %lu\n", argv[j], size/2);
 		}
 		exit(0);
 	}
--- util-linux-2.11y/fdisk/sfdisk.c.sopwith	2003-01-13 16:42:08.000000000 -0500
+++ util-linux-2.11y/fdisk/sfdisk.c	2003-01-13 16:43:05.000000000 -0500
@@ -396,7 +396,7 @@
 static struct geometry
 get_geometry(char *dev, int fd, int silent) {
     struct hd_geometry g;
-    long size;
+    unsigned long size;
     struct geometry R;
 
     if (ioctl(fd, BLKGETSIZE, &size)) {
@@ -2396,7 +2396,7 @@
 static void do_unhide(char **av, int ac, char *arg);
 static void do_activate(char **av, int ac, char *arg);
 
-int total_size;
+unsigned long total_size;
 
 int
 main(int argc, char **argv) {
@@ -2536,7 +2536,7 @@
 	}
 
 	if (opt_size)
-	  printf(_("total: %d blocks\n"), total_size);
+	  printf(_("total: %lu blocks\n"), total_size);
 
 	exit(exit_status);
     }
@@ -2662,7 +2662,7 @@
 static void
 do_size (char *dev, int silent) {
     int fd;
-    long size;
+    unsigned long size;
 
     fd = my_open(dev, 0, silent);
     if (fd < 0)
@@ -2683,9 +2683,9 @@
       return;
 
     if (silent)
-      printf("%s: %9ld\n", dev, size);
+      printf("%s: %9lu\n", dev, size);
     else
-      printf("%ld\n", size);
+      printf("%lu\n", size);
 
     total_size += size;
 }
