--- util-linux-2.11n/mkcramfs.c.pgsize	Thu Sep 19 15:24:16 2002
+++ util-linux-2.11n/mkcramfs.c	Thu Sep 19 15:34:43 2002
@@ -28,9 +28,10 @@
 
 static void usage(void)
 {
-	fprintf(stderr, "Usage: '%s [--verbose] dirname outfile'\n"
-		" where <dirname> is the root of the\n"
-		" filesystem to be compressed.\n", progname);
+	fprintf(stderr, "Usage: '%s [--verbose] [--blocksize size] dirname outfile'\n"
+		" where <dirname> is the root of the filesystem to be compressed.\n"
+		" The blocksize must equal the page size of the platform the cramfs\n"
+		" image will be used on.\n", progname);
 	exit(1);
 }
 
@@ -583,7 +584,7 @@
  */
 #define MAXFSLEN ((((1 << OFFSET_WIDTH) - 1) << 2) /* offset */ \
 		  + (1 << SIZE_WIDTH) - 1 /* filesize */ \
-		  + (1 << SIZE_WIDTH) * 4 / PAGE_CACHE_SIZE /* block pointers */ )
+		  + (1 << SIZE_WIDTH) * 4 / blksize /* block pointers */ )
 
 
 /*
@@ -613,6 +614,16 @@
 	if (argv[1] && (!strcmp(argv[1], "--verbose") || !strcmp(argv[1], "-v")))
 	    be_verbose = 1, argc--, argv++;
 
+	if (argv[1] && (!strcmp(argv[1], "--blocksize") || !strcmp(argv[1], "-b"))) {
+	    if (argv[2])
+	        blksize = atoi(argv[2]);
+	    else
+	        usage();
+	    if (blksize <= 0)
+	        usage();
+	    argc-=2, argv+=2;
+	}
+
 	if (argc != 3)
 		usage();
 
