--- util-linux-2.11y/mount/umount.c.sopwith	2003-01-14 11:33:48.000000000 -0500
+++ util-linux-2.11y/mount/umount.c	2003-01-14 11:34:22.000000000 -0500
@@ -279,7 +279,9 @@
 	if (force) {		/* only supported for NFS */
 		res = umount2 (node, MNT_FORCE);
 		if (res == -1) {
+		        int saved_errno = errno;
 			perror("umount2");
+			errno = saved_errno;
 			if (errno == ENOSYS) {
 				if (verbose)
 					printf(_("no umount2, trying umount...\n"));
