--- util-linux-2.11y/sys-utils/readprofile.c.sopwith	2003-04-11 13:04:48.000000000 -0400
+++ util-linux-2.11y/sys-utils/readprofile.c	2003-04-11 13:05:10.000000000 -0400
@@ -50,7 +50,7 @@
 static char *prgname;
 
 /* These are the defaults */
-static char defaultmap[]="/usr/src/linux/System.map";
+static char defaultmap[]="/boot/System.map";
 static char defaultpro[]="/proc/profile";
 static char optstring[]="M:m:np:itvarVb";
 
