--- util-linux-2.11y/fdisk/sfdisk.c.sopwith	Fri Sep 19 13:03:54 2003
+++ util-linux-2.11y/fdisk/sfdisk.c	Fri Sep 19 13:04:12 2003
@@ -2376,8 +2376,6 @@
 			    &ma, &mi, &sz, ptname) != 4)
 			continue;
 		for(s = ptname; *s; s++);
-		if (isdigit(s[-1]))
-			continue;
 		snprintf(devname, sizeof(devname), "/dev/%s", ptname);
 		return devname;
 	}
@@ -2535,9 +2533,6 @@
 		do_list(dev, 1);
 	}
 
-	if (opt_size)
-	  printf(_("total: %lu blocks\n"), total_size);
-
 	exit(exit_status);
     }
 
