--- util-linux-2.11y/mount/mount.c.sopwith	2003-07-14 16:39:11.000000000 -0400
+++ util-linux-2.11y/mount/mount.c	2003-07-14 16:39:42.000000000 -0400
@@ -841,7 +841,9 @@
 
 #ifdef HAVE_NFS
   if (mnt_err && types && streq (types, "nfs")) {
-      if (nfs_mount_version == 4) {
+      if (nfs_mount_version == 4
+	  && mnt_err != ENOENT
+	  /* Add other known-local errnos here */) {
 	  if (verbose)
 	    printf(_("mount: failed with nfs mount version 4, trying 3..\n"));
 	  nfs_mount_version = 3;
