

--- util-linux-2.11n/login-utils/login.c.utmp	Tue Jun 18 16:10:03 2002
+++ util-linux-2.11n/login-utils/login.c	Tue Jun 18 16:46:11 2002
@@ -924,9 +924,8 @@ Michael Riepe <michael@stud.uni-hannover
 	if (utp == NULL) {
 	     setutent();
 	     ut.ut_type = LOGIN_PROCESS;
-	     strncpy(ut.ut_id, tty_number, sizeof(ut.ut_id));
 	     strncpy(ut.ut_line, tty_name, sizeof(ut.ut_line));
-	     utp = getutid(&ut);
+	     utp = getutline(&ut);
 	}
 	
 	if (utp) {
