--- util-linux-2.11y/mount/umount.c.sopwith	2003-01-29 09:33:52.000000000 -0500
+++ util-linux-2.11y/mount/umount.c	2003-01-29 09:34:19.000000000 -0500
@@ -625,7 +625,7 @@
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
 
-	umask(033);
+	umask(022);
 
 	while ((c = getopt_long (argc, argv, "adfhlnrt:O:vV",
 				 longopts, NULL)) != -1)
--- util-linux-2.11y/mount/mount.c.sopwith	2003-01-29 09:33:56.000000000 -0500
+++ util-linux-2.11y/mount/mount.c	2003-01-29 09:34:13.000000000 -0500
@@ -1419,7 +1419,7 @@
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
 
-	umask(033);
+	umask(022);
 
 	/* People report that a mount called from init without console
 	   writes error messages to /etc/mtab
