--- util-linux-2.11w/text-utils/Makefile.sopwith	Tue Oct 29 07:22:25 2002
+++ util-linux-2.11w/text-utils/Makefile	Tue Oct 29 07:23:51 2002
@@ -41,15 +41,15 @@
 # more and pg and ul use curses - maybe we can't compile them
 ifeq "$(HAVE_NCURSES)" "yes"
 # Have ncurses - make more and pg and ul
-more pg ul:
+pg ul:
 	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@ $(LIBCURSES)
-more: more.o $(LIB)/xstrncpy.o
 pg: pg.o
 ul: ul.o
 else
 # Do not have ncurses - give up on pg and ul
 pg ul:
 	@echo $@ not made since it requires ncurses
+endif
 # For more we can also try termcap
 ifeq "$(HAVE_TERMCAP)" "yes"
 more: more.o $(LIB)/xstrncpy.o
@@ -58,7 +58,6 @@
 more:
 	@echo $@ not made since it requires ncurses or termcap
 endif
-endif
 
 # Rules for hexdump
 hexdump: hexdump.o conv.o display.o hexsyntax.o odsyntax.o parse.o
