--- util-linux-2.11y/login-utils/login.c.64bit	2003-04-10 19:36:13.000000000 -0400
+++ util-linux-2.11y/login-utils/login.c	2003-04-10 19:38:14.000000000 -0400
@@ -1373,8 +1373,9 @@
 	if (!quiet) {
 	    if (read(fd, (char *)&ll, sizeof(ll)) == sizeof(ll) &&
 		ll.ll_time != 0) {
+		time_t ll_time = (time_t) ll.ll_time;
 		printf(_("Last login: %.*s "),
-		       24-5, (char *)ctime(&ll.ll_time));
+		       24-5, (char *)ctime(&ll_time));
 		
 		if (*ll.ll_host != '\0')
 		  printf(_("from %.*s\n"),
