--- util-linux-2.11y/sys-utils/ipcs.c.ipcs	Sun Nov 24 17:21:44 2002
+++ util-linux-2.11y/sys-utils/ipcs.c	Fri Apr 11 09:13:34 2003
@@ -25,6 +25,7 @@
 #include <time.h>
 #include <pwd.h>
 #include <grp.h>
+#include <unistd.h>
 #include "nls.h"
 
 /* X/OPEN tells us to use <sys/{types,ipc,sem}.h> for semctl() */
@@ -279,8 +280,8 @@
 			(long) shminfo.shmmni);
 		printf (_("max seg size (kbytes) = %ld\n"),
 			(long) (shminfo.shmmax >> 10));
-		printf (_("max total shared memory (kbytes) = %ld\n"),
-			(long) shminfo.shmall << 2);
+		printf (_("max total shared memory (kbytes) = %lu\n"),
+			getpagesize()*((unsigned long) shminfo.shmall/1024UL));
 		printf (_("min seg size (bytes) = %ld\n"),
 			(long) shminfo.shmmin);
 		return;
@@ -360,12 +361,12 @@
 				printf ("%-10d %-10.10s", shmid, pw->pw_name);
 			else
 				printf ("%-10d %-10d", shmid, ipcp->uid);
-			printf ("%-10o %-10ld %-10ld %-6s %-6s\n", 
+			printf ("%-10o %-10lu %-10ld %-6s %-6s\n", 
 				ipcp->mode & 0777,
 				/*
 				 * earlier: int, Austin has size_t
 				 */
-				(long) shmseg.shm_segsz,
+				((unsigned long)shmseg.shm_segsz),
 				/*
 				 * glibc-2.1.3 and earlier has unsigned short;
 				 * Austin has shmatt_t
