Enable fsck.cramfs -x

diff --git a/disk-utils/fsck.cramfs.c b/disk-utils/fsck.cramfs.c
--- a/disk-utils/fsck.cramfs.c
+++ b/disk-utils/fsck.cramfs.c
@@ -34,7 +34,7 @@
  */
 
 /* compile-time options */
-//#define INCLUDE_FS_TESTS	/* include cramfs checking and extraction */
+#define INCLUDE_FS_TESTS	/* include cramfs checking and extraction */
 
 #define _GNU_SOURCE
 #include <stdio.h>
@@ -710,7 +710,8 @@
 	test_super(&start, &length);
 	test_crc(start);
 #ifdef INCLUDE_FS_TESTS
-	test_fs(start);
+	if (extract_dir != NULL)
+		test_fs(start);
 #endif /* INCLUDE_FS_TESTS */
 
 	if (opt_verbose) {
