--- util-linux-2.11w/mount/swapon.c.swaponsymlink	Tue Oct 29 08:07:20 2002
+++ util-linux-2.11w/mount/swapon.c	Tue Oct 29 08:10:18 2002
@@ -143,9 +143,16 @@
 is_in_proc_swaps(char *fname) {
 	int i;
 
-	for (i = 0; i < numSwaps; i++)
-		if (!strcmp(fname, swapFiles[i]))
+	for (i = 0; i < numSwaps; i++) {
+	        struct stat swapped, swapping;
+
+		if (!strcmp(swapFiles[i], fname)
+		    || (!stat (swapFiles[i], &swapped)
+			&& !stat (fname, &swapping)
+			&& (swapped.st_dev == swapping.st_dev
+			    && swapped.st_ino == swapping.st_ino)))
 			return 1;
+	}
 	return 0;
 }
 
