--- util-linux-2.11y/mount/linux_fs.h.s390x	2003-03-25 11:25:13.000000000 +0100
+++ util-linux-2.11y/mount/linux_fs.h	2003-03-25 11:25:48.000000000 +0100
@@ -15,9 +15,13 @@
 
 #include <inttypes.h>
 #ifndef BLKGETSIZE64
+#ifdef __s390x__
+#define BLKGETSIZE64 BLKGETSIZE
+#else
 #include <sys/ioctl.h>
 #define BLKGETSIZE64 _IOR(0x12,114,sizeof(uint64_t))
 #endif
+#endif
 
 #define MINIX_SUPER_MAGIC   0x137F         /* minix v1, 14 char names */
 #define MINIX_SUPER_MAGIC2  0x138F         /* minix v1, 30 char names */
