diff -r 51cdc5b85e87 libproxy/cmake/modules/pacrunner_webkit.cmk
--- a/libproxy/cmake/modules/pacrunner_webkit.cmk	Mon Oct 10 11:53:30 2011 +0200
+++ b/libproxy/cmake/modules/pacrunner_webkit.cmk	Mon Oct 10 12:01:54 2011 +0200
@@ -12,5 +12,8 @@
     set(WEBKIT_FOUND 1)
   endif()
 else()
-  px_check_modules(WEBKIT webkit-1.0)
-endif()
\ No newline at end of file
+  px_check_modules(WEBKIT "javascriptcoregtk-1.0 >= 1.5.0")
+  if(NOT WEBKIT_LIBRARIES)
+    px_check_modules(WEBKIT "webkit-1.0 < 1.5.0")
+  endif(NOT WEBKIT_LIBRARIES)
+endif()
