Subject: acpi-cope-with-pnpacpi-tables-missing-an-rtc-entry-fix
From: Andrew Morton <akpm@linux-foundation.org>

Cc: <stable@kernel.org>
Cc: Adam Belay <abelay@novell.com>
Cc: Alessandro Zummo <a.zummo@towertech.it>
Cc: Bjorn Helgaas <bjorn.helgaas@hp.com>
Cc: Chuck Ebbert <cebbert@redhat.com>
Cc: David Brownell <dbrownell@users.sourceforge.net>
Cc: Ingo Molnar <mingo@elte.hu>
Cc: Len Brown <lenb@kernel.org>
Cc: Rik Theys <rik.theys@esat.kuleuven.be>
Cc: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/asm-x86/mc146818rtc.h |    2 ++
 1 file changed, 2 insertions(+)

diff -puN
include/asm-x86/mc146818rtc.h~acpi-cope-with-pnpacpi-tables-missing-an-rtc-entry-fix
include/asm-x86/mc146818rtc.h
---
a/include/asm-x86/mc146818rtc.h~acpi-cope-with-pnpacpi-tables-missing-an-rtc-entry-fix
+++ a/include/asm-x86/mc146818rtc.h
@@ -9,6 +9,8 @@
 #include <asm/processor.h>
 #include <linux/mc146818rtc.h>

+struct device;
+
 #ifndef RTC_PORT
 #define RTC_PORT(x) (0x70 + (x))
 #define RTC_ALWAYS_BCD 1 /* RTC operates in binary mode */
