diff -up linux-2.6.31.noarch/arch/x86/Kconfig.dave linux-2.6.31.noarch/arch/x86/Kconfig
--- linux-2.6.31.noarch/arch/x86/Kconfig.dave	2009-09-09 18:13:59.000000000 -0400
+++ linux-2.6.31.noarch/arch/x86/Kconfig	2009-10-23 17:04:30.000000000 -0400
@@ -1444,12 +1444,8 @@ config SECCOMP
 
 	  If unsure, say Y. Only embedded should say N here.
 
-config CC_STACKPROTECTOR_ALL
-	bool
-
 config CC_STACKPROTECTOR
 	bool "Enable -fstack-protector buffer overflow detection (EXPERIMENTAL)"
-	select CC_STACKPROTECTOR_ALL
 	---help---
 	  This option turns on the -fstack-protector GCC feature. This
 	  feature puts, at the beginning of functions, a canary value on
diff -up linux-2.6.31.noarch/arch/x86/Makefile.dave linux-2.6.31.noarch/arch/x86/Makefile
--- linux-2.6.31.noarch/arch/x86/Makefile.dave	2009-09-09 18:13:59.000000000 -0400
+++ linux-2.6.31.noarch/arch/x86/Makefile	2009-10-23 17:04:30.000000000 -0400
@@ -74,7 +74,6 @@ ifdef CONFIG_CC_STACKPROTECTOR
 	cc_has_sp := $(srctree)/scripts/gcc-x86_$(BITS)-has-stack-protector.sh
         ifeq ($(shell $(CONFIG_SHELL) $(cc_has_sp) $(CC)),y)
                 stackp-y := -fstack-protector
-                stackp-$(CONFIG_CC_STACKPROTECTOR_ALL) += -fstack-protector-all
                 KBUILD_CFLAGS += $(stackp-y)
         else
                 $(warning stack protector enabled but no compiler support)
