diff --git a/drivers/gpu/drm/i915/i915_dma.c b/drivers/gpu/drm/i915/i915_dma.c
index 1ce3296..e3b4ee2 100644
--- a/drivers/gpu/drm/i915/i915_dma.c
+++ b/drivers/gpu/drm/i915/i915_dma.c
@@ -717,7 +717,7 @@ static int i915_getparam(struct drm_device *dev, void *data,
 		value = dev->pci_device;
 		break;
 	case I915_PARAM_HAS_GEM:
-		value = 1;
+		value = dev_priv->has_gem;
 		break;
 	default:
 		DRM_ERROR("Unknown parameter %d\n", param->param);
@@ -830,6 +830,17 @@ int i915_driver_load(struct drm_device *dev, unsigned long flags)
 
 	dev_priv->regs = ioremap(base, size);
 
+	/* don't enable GEM on PAE or 8xx */
+#ifdef CONFIG_HIGHMEM64G
+	dev_priv->has_gem = 0;
+#else
+	if (IS_I830(dev) || IS_845G(dev) || IS_I85X(dev) ||
+	    IS_I855(dev) || IS_I865G(dev))
+		dev_priv->has_gem = 0;
+	else
+		dev_priv->has_gem = 1;
+#endif
+
 	i915_gem_load(dev);
 
 	/* Init HWS */
diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index c829f24..fd4f467 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -111,6 +111,8 @@ typedef struct drm_i915_private {
 
 	void __iomem *regs;
 
+	int has_gem;
+
 	drm_i915_ring_buffer_t ring;
 
 	drm_dma_handle_t *status_page_dmah;
