diff --git a/drivers/gpu/drm/i915/i915_suspend.c b/drivers/gpu/drm/i915/i915_suspend.c
index ce8a213..7e90be3 100644
--- a/drivers/gpu/drm/i915/i915_suspend.c
+++ b/drivers/gpu/drm/i915/i915_suspend.c
@@ -28,6 +28,7 @@
 #include "drm.h"
 #include "i915_drm.h"
 #include "i915_drv.h"
+#include <drm/drm_crtc_helper.h>
 
 static bool i915_pipe_enabled(struct drm_device *dev, enum pipe pipe)
 {
@@ -532,6 +533,9 @@ int i915_restore_state(struct drm_device *dev)
 
 	i915_restore_vga(dev);
 
+	if (drm_core_check_feature(dev, DRIVER_MODESET))
+		drm_helper_resume_force_mode(dev);
+
 	return 0;
 }
 
