Index: linux-2.6.29.noarch/include/linux/Kbuild
===================================================================
--- linux-2.6.29.noarch.orig/include/linux/Kbuild
+++ linux-2.6.29.noarch/include/linux/Kbuild
@@ -67,6 +67,7 @@ header-y += falloc.h
 header-y += fd.h
 header-y += fdreg.h
 header-y += fib_rules.h
+header-y += fiemap.h
 header-y += firewire-cdev.h
 header-y += firewire-constants.h
 header-y += fuse.h
