From: Arjan van de Ven <arjan@linux.intel.com>

--- linux-2.6.19/arch/i386/kernel/nmi.c.org	2007-01-07 10:40:52.000000000 -0800
+++ linux-2.6.19/arch/i386/kernel/nmi.c	2007-01-07 10:42:41.000000000 -0800
@@ -55,7 +55,7 @@ static DEFINE_PER_CPU(unsigned long, evn
  */
 atomic_t nmi_active = ATOMIC_INIT(0);		/* oprofile uses this */
 
-unsigned int nmi_watchdog = NMI_DEFAULT;
+unsigned int nmi_watchdog = NMI_NONE;
 static unsigned int nmi_hz = HZ;
 
 struct nmi_watchdog_ctlblk {
--- linux-2.6.19/arch/x86_64/kernel/nmi.c.org	2007-01-07 10:42:55.000000000 -0800
+++ linux-2.6.19/arch/x86_64/kernel/nmi.c	2007-01-07 10:42:58.000000000 -0800
@@ -55,7 +55,7 @@ static DEFINE_PER_CPU(unsigned, evntsel_
 atomic_t nmi_active = ATOMIC_INIT(0);		/* oprofile uses this */
 int panic_on_timeout;
 
-unsigned int nmi_watchdog = NMI_DEFAULT;
+unsigned int nmi_watchdog = NMI_NONE;
 static unsigned int nmi_hz = HZ;
 
 struct nmi_watchdog_ctlblk {

