diff -up linux-2.6.29.noarch/arch/x86/xen/enlighten.c.orig linux-2.6.29.noarch/arch/x86/xen/enlighten.c
--- linux-2.6.29.noarch/arch/x86/xen/enlighten.c.orig	2009-05-18 14:27:53.000000000 +0200
+++ linux-2.6.29.noarch/arch/x86/xen/enlighten.c	2009-05-18 14:32:07.000000000 +0200
@@ -218,6 +218,8 @@ static void xen_cpuid(unsigned int *ax, 
 			    (1 << X86_FEATURE_MCE)  |  /* disable MCE */
 			    (1 << X86_FEATURE_MCA)  |  /* disable MCA */
 			    (1 << X86_FEATURE_ACC));   /* thermal monitoring */
+	else if (*ax == 0x80000001)
+		maskedx = ~((1 << (X86_FEATURE_GBPAGES&31))); /* disable GB pages */
 
 	asm(XEN_EMULATE_PREFIX "cpuid"
 		: "=a" (*ax),
