diff --git a/arch/x86/xen/enlighten.c b/arch/x86/xen/enlighten.c
index f09e8c3..9a93a7b 100644
--- a/arch/x86/xen/enlighten.c
+++ b/arch/x86/xen/enlighten.c
@@ -185,7 +185,8 @@ static void xen_cpuid(unsigned int *ax, unsigned int *bx,
 	if (*ax == 1) {
 		maskecx = cpuid_leaf1_ecx_mask;
 		maskedx = cpuid_leaf1_edx_mask;
-	}
+	} else if (*ax == 0x80000001)
+		maskedx = ~((1 << (X86_FEATURE_GBPAGES&31))); /* disable GB pages */
 
 	asm(XEN_EMULATE_PREFIX "cpuid"
 		: "=a" (*ax),
