diff -up linux-2.6.27.noarch/drivers/pci/setup-res.c.jx linux-2.6.27.noarch/drivers/pci/setup-res.c
--- linux-2.6.27.noarch/drivers/pci/setup-res.c.jx	2008-10-09 18:13:53.000000000 -0400
+++ linux-2.6.27.noarch/drivers/pci/setup-res.c	2008-10-16 14:12:13.000000000 -0400
@@ -138,10 +138,10 @@ int pci_assign_resource(struct pci_dev *
 
 	align = resource_alignment(res);
 	if (!align) {
-		dev_err(&dev->dev, "BAR %d: can't allocate resource (bogus "
-			"alignment) [%#llx-%#llx] flags %#lx\n",
-			resno, (unsigned long long)res->start,
-			(unsigned long long)res->end, res->flags);
+		dev_info(&dev->dev, "BAR %d: can't allocate resource (bogus "
+			 "alignment) [%#llx-%#llx] flags %#lx\n",
+			 resno, (unsigned long long)res->start,
+			 (unsigned long long)res->end, res->flags);
 		return -EINVAL;
 	}
 
@@ -162,11 +162,11 @@ int pci_assign_resource(struct pci_dev *
 	}
 
 	if (ret) {
-		dev_err(&dev->dev, "BAR %d: can't allocate %s resource "
-			"[%#llx-%#llx]\n", resno,
-			res->flags & IORESOURCE_IO ? "I/O" : "mem",
-			(unsigned long long)res->start,
-			(unsigned long long)res->end);
+		dev_info(&dev->dev, "BAR %d: can't allocate %s resource "
+			 "[%#llx-%#llx]\n", resno,
+			 res->flags & IORESOURCE_IO ? "I/O" : "mem",
+			 (unsigned long long)res->start,
+			 (unsigned long long)res->end);
 	} else {
 		res->flags &= ~IORESOURCE_STARTALIGN;
 		if (resno < PCI_BRIDGE_RESOURCES)
diff -up linux-2.6.27.noarch/arch/x86/pci/i386.c.jx linux-2.6.27.noarch/arch/x86/pci/i386.c
--- linux-2.6.27.noarch/arch/x86/pci/i386.c.jx	2008-10-09 18:13:53.000000000 -0400
+++ linux-2.6.27.noarch/arch/x86/pci/i386.c	2008-10-23 10:21:14.000000000 -0400
@@ -128,7 +128,7 @@ static void __init pcibios_allocate_bus_
 				pr = pci_find_parent_resource(dev, r);
 				if (!r->start || !pr ||
 				    request_resource(pr, r) < 0) {
-					dev_err(&dev->dev, "BAR %d: can't allocate resource\n", idx);
+					dev_info(&dev->dev, "BAR %d: can't allocate resource\n", idx);
 					/*
 					 * Something is wrong with the region.
 					 * Invalidate the resource to prevent
@@ -169,7 +169,7 @@ static void __init pcibios_allocate_reso
 					r->flags, disabled, pass);
 				pr = pci_find_parent_resource(dev, r);
 				if (!pr || request_resource(pr, r) < 0) {
-					dev_err(&dev->dev, "BAR %d: can't allocate resource\n", idx);
+					dev_info(&dev->dev, "BAR %d: can't allocate resource\n", idx);
 					/* We'll assign a new address later */
 					r->end -= r->start;
 					r->start = 0;
