--- linux-2.6/include/linux/sched.h
+++ linux-2.6/include/linux/sched.h
@@ -365,7 +365,8 @@ extern int get_dumpable(struct mm_struct
 #define MMF_DUMP_FILTER_MASK \
 	(((1 << MMF_DUMP_FILTER_BITS) - 1) << MMF_DUMP_FILTER_SHIFT)
 #define MMF_DUMP_FILTER_DEFAULT \
-	((1 << MMF_DUMP_ANON_PRIVATE) |	(1 << MMF_DUMP_ANON_SHARED))
+	((1 << MMF_DUMP_ANON_PRIVATE) |	(1 << MMF_DUMP_ANON_SHARED) | \
+	 (1 << MMF_DUMP_ELF_HEADERS))
 
 struct mm_struct {
 	struct vm_area_struct * mmap;		/* list of VMAs */

