diff -r ffeb9d2be572 linux/drivers/media/video/gspca/vc032x.c
--- linux/drivers/media/video/gspca/vc032x.c	Mon Dec 08 10:41:04 2008 +0100
+++ linux/drivers/media/video/gspca/vc032x.c	Wed Dec 10 10:10:55 2008 +0100
@@ -1977,6 +1977,13 @@
 				data, len);
 		return;
 	}
+
+	/* The vc0321 sends some additional data after sending the complete
+	   frame, we ignore this. */
+	if (sd->bridge == BRIDGE_VC0321 && ((frame->data_end - frame->data)
+			+ len) > frame->v4l2_buf.length)
+		len = frame->v4l2_buf.length - (frame->data_end - frame->data);
+
 	gspca_frame_add(gspca_dev, INTER_PACKET, frame, data, len);
 }
 
