Atomic failures aren't too interesting.

--- linux-2.6.21.noarch/include/linux/gfp.h~	2007-05-27 23:21:35.000000000 -0400
+++ linux-2.6.21.noarch/include/linux/gfp.h	2007-05-27 23:21:51.000000000 -0400
@@ -58,7 +58,7 @@ struct vm_area_struct;
 /* This equals 0, but use constants in case they ever change */
 #define GFP_NOWAIT	(GFP_ATOMIC & ~__GFP_HIGH)
 /* GFP_ATOMIC means both !wait (__GFP_WAIT not set) and use emergency pool */
-#define GFP_ATOMIC	(__GFP_HIGH)
+#define GFP_ATOMIC	(__GFP_HIGH | __GFP_NOWARN)
 #define GFP_NOIO	(__GFP_WAIT)
 #define GFP_NOFS	(__GFP_WAIT | __GFP_IO)
 #define GFP_KERNEL	(__GFP_WAIT | __GFP_IO | __GFP_FS)
