Index: linux-2.6.27.noarch/fs/ioctl.c
===================================================================
--- linux-2.6.27.noarch.orig/fs/ioctl.c	2008-10-13 12:58:54.000000000 -0500
+++ linux-2.6.27.noarch/fs/ioctl.c	2008-10-13 13:08:06.861000990 -0500
@@ -352,7 +352,12 @@ static int file_ioctl(struct file *filp,
 	case FIBMAP:
 		return ioctl_fibmap(filp, p);
 	case FS_IOC_FIEMAP:
+#if 0
+		/* Until this is released in 2.6.28 we won't turn it on */
 		return ioctl_fiemap(filp, arg);
+#else
+		return -EINVAL;
+#endif
 	case FIGETBSZ:
 		return put_user(inode->i_sb->s_blocksize, p);
 	case FIONREAD:
