From 54583bf4efda79388fc13163e35c016c8bc5de81 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Sun, 14 Oct 2007 21:21:30 +1000
Subject: [PATCH] i915: fix vbl swap allocation size.

Oops...

Signed-off-by: Dave Airlie <airlied@linux.ie>
---
 drivers/char/drm/i915_irq.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/char/drm/i915_irq.c b/drivers/char/drm/i915_irq.c
index 380c3f3..a443f4a 100644
--- a/drivers/char/drm/i915_irq.c
+++ b/drivers/char/drm/i915_irq.c
@@ -540,7 +540,7 @@ int i915_vblank_swap(struct drm_device *dev, void *data,
 		return -EBUSY;
 	}
 
-	vbl_swap = drm_calloc(1, sizeof(vbl_swap), DRM_MEM_DRIVER);
+	vbl_swap = drm_calloc(1, sizeof(*vbl_swap), DRM_MEM_DRIVER);
 
 	if (!vbl_swap) {
 		DRM_ERROR("Failed to allocate memory to queue swap\n");
-- 
1.5.2.4

