--- linux-2.6.26.noarch/include/linux/slab.h~	2008-09-01 00:33:55.000000000 -0400
+++ linux-2.6.26.noarch/include/linux/slab.h	2008-09-01 00:34:19.000000000 -0400
@@ -272,7 +272,7 @@ static inline void *kmem_cache_zalloc(st
  * @size: how many bytes of memory are required.
  * @flags: the type of memory to allocate (see kmalloc).
  */
-static inline void *kzalloc(size_t size, gfp_t flags)
+static __always_inline void *kzalloc(size_t size, gfp_t flags)
 {
 	return kmalloc(size, flags | __GFP_ZERO);
 }
