--- linux-2.6.32.noarch.orig/kernel/auditsc.c
+++ linux-2.6.32.noarch/kernel/auditsc.c
@@ -1893,14 +1893,14 @@ static int audit_inc_name_count(struct a
 				const struct inode *inode)
 {
 	if (context->name_count >= AUDIT_NAMES) {
-		if (inode)
+		if (inode) {
 			printk(KERN_DEBUG "name_count maxed, losing inode data: "
-			       "dev=%02x:%02x, inode=%lu\n",
+			       "dev=%02x:%02x, inode=%lu, fs=%s\n",
 			       MAJOR(inode->i_sb->s_dev),
 			       MINOR(inode->i_sb->s_dev),
-			       inode->i_ino);
-
-		else
+			       inode->i_ino, inode->i_sb->s_type->name);
+			WARN_ONCE(1, "lost inode data");
+		} else
 			printk(KERN_DEBUG "name_count maxed, losing inode data\n");
 		return 1;
 	}
