diff -up linux-2.6.28.noarch/drivers/acpi/osl.c~ linux-2.6.28.noarch/drivers/acpi/osl.c
--- linux-2.6.28.noarch/drivers/acpi/osl.c~	2009-01-23 22:57:27.000000000 +0100
+++ linux-2.6.28.noarch/drivers/acpi/osl.c	2009-01-23 22:57:27.000000000 +0100
@@ -1077,7 +1077,7 @@ __setup("acpi_wake_gpes_always_on", acpi
 #define ENFORCE_RESOURCES_LAX    1
 #define ENFORCE_RESOURCES_NO     0
 
-static unsigned int acpi_enforce_resources = ENFORCE_RESOURCES_LAX;
+static unsigned int acpi_enforce_resources = ENFORCE_RESOURCES_STRICT;
 
 static int __init acpi_enforce_resources_setup(char *str)
 {
