diff -up linux-2.6.23.noarch/drivers/net/wireless/ath5k/hw.c.orig linux-2.6.23.noarch/drivers/net/wireless/ath5k/hw.c
--- linux-2.6.23.noarch/drivers/net/wireless/ath5k/hw.c.orig	2007-10-17 14:35:44.000000000 -0400
+++ linux-2.6.23.noarch/drivers/net/wireless/ath5k/hw.c	2007-10-17 14:35:35.000000000 -0400
@@ -491,8 +491,18 @@ const struct ath5k_rate_table *ath5k_hw_
 {
 	AR5K_TRACE;
 
-	if (!test_bit(mode, hal->ah_capabilities.cap_mode))
+	if (!test_bit(mode, hal->ah_capabilities.cap_mode)) {
+#if 0
+		printk(KERN_ERR "ath5k_hw_get_rate_table: mode=%u, cap_mode=[", mode);
+		for (mode = 0; mode < NUM_DRIVER_MODES; mode++) {
+			if (test_bit(mode, hal->ah_capabilities.cap_mode))
+				printk("%u,", mode);
+		}
+		printk("]\n");
+		dump_stack();
+#endif
 		return NULL;
+	}
 
 	/* Get rate tables */
 	switch (mode) {
@@ -667,6 +667,8 @@ int ath5k_hw_reset(struct ath_hw *hal, e
 				/*Get rate table for this operation mode*/
 				rt = ath5k_hw_get_rate_table(hal,
 						MODE_IEEE80211B);
+				if (!rt)
+					return -EINVAL;
 
 				/*Write rate duration table*/
 				for (i = 0; i < rt->rate_count; i++) {
@@ -692,6 +694,8 @@ int ath5k_hw_reset(struct ath_hw *hal, e
 				rt = ath5k_hw_get_rate_table(hal,
 				    channel->val & CHANNEL_TURBO ?
 				    MODE_ATHEROS_TURBO : MODE_ATHEROS_TURBOG);
+				if (!rt)
+					return -EINVAL;
 
 				/* Write rate duration table */
 				for (i = 0; i < rt->rate_count; i++)
