https://bugzilla.redhat.com/show_bug.cgi?id=454575

Signed-off-by: Dave Jones <davej@redhat.com>

--- linux-2.6.26.noarch/drivers/net/tulip/tulip_core.c~	2008-07-17 14:04:46.000000000 -0400
+++ linux-2.6.26.noarch/drivers/net/tulip/tulip_core.c	2008-07-17 14:07:38.000000000 -0400
@@ -504,11 +504,11 @@ tulip_open(struct net_device *dev)
 {
 	int retval;
 
+	tulip_init_ring(dev);
+
 	if ((retval = request_irq(dev->irq, &tulip_interrupt, IRQF_SHARED, dev->name, dev)))
 		return retval;
 
-	tulip_init_ring (dev);
-
 	tulip_up (dev);
 
 	netif_start_queue (dev);
