diff -uNrp kernel-2.6.27.orig/net/bluetooth/hidp/core.c kernel-2.6.27.new/net/bluetooth/hidp/core.c
--- kernel-2.6.27.orig/net/bluetooth/hidp/core.c	2008-10-10 00:13:53.000000000 +0200
+++ kernel-2.6.27.new/net/bluetooth/hidp/core.c	2009-01-29 20:51:24.000000000 +0100
@@ -684,6 +684,10 @@ static const struct {
 } hidp_blacklist[] = {
 	/* Apple wireless Mighty Mouse */
 	{ 0x05ac, 0x030c, HID_QUIRK_MIGHTYMOUSE | HID_QUIRK_INVERT_HWHEEL },
+	/* Apple Wireless Keyboard */
+	{ 0x05ac, 0x022c, HID_QUIRK_APPLE_NUMLOCK_EMULATION | HID_QUIRK_APPLE_HAS_FN },
+	{ 0x05ac, 0x022d, HID_QUIRK_APPLE_NUMLOCK_EMULATION | HID_QUIRK_APPLE_HAS_FN | HID_QUIRK_APPLE_ISO_KEYBOARD },
+	{ 0x05ac, 0x022e, HID_QUIRK_APPLE_NUMLOCK_EMULATION | HID_QUIRK_APPLE_HAS_FN },
 
 	{ }	/* Terminating entry */
 };
