diff -up linux-2.6.27.noarch/drivers/net/wireless/hostap/hostap_wlan.h.orig linux-2.6.27.noarch/drivers/net/wireless/hostap/hostap_wlan.h
--- linux-2.6.27.noarch/drivers/net/wireless/hostap/hostap_wlan.h.orig	2008-10-28 13:05:26.000000000 -0400
+++ linux-2.6.27.noarch/drivers/net/wireless/hostap/hostap_wlan.h	2008-10-28 13:05:30.000000000 -0400
@@ -918,9 +918,12 @@ struct hostap_interface {
 
 /*
  * TX meta data - stored in skb->cb buffer, so this must not be increased over
- * the 40-byte limit
+ * the 48-byte limit.
+ * THE PADDING THIS STARTS WITH IS A HORRIBLE HACK THAT SHOULD NOT LIVE
+ * TO SEE THE DAY.
  */
 struct hostap_skb_tx_data {
+	unsigned int __padding_for_default_qdiscs;
 	u32 magic; /* HOSTAP_SKB_TX_DATA_MAGIC */
 	u8 rate; /* transmit rate */
 #define HOSTAP_TX_FLAGS_WDS BIT(0)
