diff -ur linux-2.6.27.noarch/drivers/pci/hotplug/pciehp_hpc.c linux-2.6.27.noarch.hack/drivers/pci/hotplug/pciehp_hpc.c
--- linux-2.6.27.noarch/drivers/pci/hotplug/pciehp_hpc.c	2008-11-12 22:24:32.000000000 +0000
+++ linux-2.6.27.noarch.hack/drivers/pci/hotplug/pciehp_hpc.c	2008-11-12 22:23:35.000000000 +0000
@@ -234,6 +234,9 @@
 		return 0;
 	}
 
+	if (!irq)
+		return -ENODEV;
+
 	/* Installs the interrupt handler */
 	retval = request_irq(irq, pcie_isr, IRQF_SHARED, MY_NAME, ctrl);
 	if (retval)
Only in linux-2.6.27.noarch.hack/drivers/pci/hotplug: pciehp_hpc.c~
