https://bugzilla.redhat.com/show_bug.cgi?id=454575

Signed-off-by: Dave Jones <davej@redhat.com>

diff --git a/drivers/net/tulip/tulip_core.c b/drivers/net/tulip/tulip_core.c
index cafa89e..daa83a7 100644
--- a/drivers/net/tulip/tulip_core.c
+++ b/drivers/net/tulip/tulip_core.c
@@ -502,11 +502,11 @@ tulip_open(struct net_device *dev)
 {
 	int retval;
 
+	tulip_init_ring(dev);
+
 	if ((retval = request_irq(dev->irq, &tulip_interrupt, IRQF_SHARED, dev->name, dev)))
 		return retval;
 
-	tulip_init_ring (dev);
-
 	tulip_up (dev);
 
 	netif_start_queue (dev);
