diff -up linux-2.6.23.noarch/drivers/net/wireless/ath5k/base.c.orig linux-2.6.23.noarch/drivers/net/wireless/ath5k/base.c
--- linux-2.6.23.noarch/drivers/net/wireless/ath5k/base.c.orig	2007-12-01 13:00:52.000000000 -0500
+++ linux-2.6.23.noarch/drivers/net/wireless/ath5k/base.c	2007-12-01 13:01:04.000000000 -0500
@@ -2830,7 +2830,6 @@ ath5k_set_key(struct ieee80211_hw *hw, e
 
 	switch(key->alg) {
 	case ALG_WEP:
-		break;
 	case ALG_TKIP:
 	case ALG_CCMP:
 		return -EOPNOTSUPP;
