diff -up linux-2.6.23.noarch/drivers/net/wireless/zd1211rw/zd_usb.c.orig linux-2.6.23.noarch/drivers/net/wireless/zd1211rw/zd_usb.c
--- linux-2.6.23.noarch/drivers/net/wireless/zd1211rw/zd_usb.c.orig	2007-12-05 20:59:52.000000000 -0500
+++ linux-2.6.23.noarch/drivers/net/wireless/zd1211rw/zd_usb.c	2007-12-05 21:00:36.000000000 -0500
@@ -92,6 +92,7 @@ MODULE_AUTHOR("Ulrich Kunitz");
 MODULE_AUTHOR("Daniel Drake");
 MODULE_VERSION("1.0");
 MODULE_DEVICE_TABLE(usb, usb_ids);
+MODULE_ALIAS("zd1211rw-mac80211");
 
 #define FW_ZD1211_PREFIX	"zd1211/zd1211_"
 #define FW_ZD1211B_PREFIX	"zd1211/zd1211b_"
