---
 drivers/ssb/main.c |    2 ++
 1 file changed, 2 insertions(+)

--- linux-2.6.22.noarch.orig/drivers/ssb/main.c
+++ linux-2.6.22.noarch/drivers/ssb/main.c
@@ -884,6 +884,8 @@ static u32 ssb_tmslow_reject_bitmask(str
 	case SSB_IDLOW_SSBREV_23:
 		return SSB_TMSLOW_REJECT_23;
 	default:
+		printk(KERN_ERR "ssb: rev %x\n", 
+		       ssb_read32(dev, SSB_IDLOW) & SSB_IDLOW_SSBREV);
 		WARN_ON(1);
 	}
 	return (SSB_TMSLOW_REJECT_22 | SSB_TMSLOW_REJECT_23);
