diff -up linux-2.6.24.noarch/drivers/net/wireless/ipw2200.c.orig linux-2.6.24.noarch/drivers/net/wireless/ipw2200.c
--- linux-2.6.24.noarch/drivers/net/wireless/ipw2200.c.orig	2008-04-01 22:28:02.000000000 -0400
+++ linux-2.6.24.noarch/drivers/net/wireless/ipw2200.c	2008-04-01 22:28:07.000000000 -0400
@@ -10349,7 +10349,9 @@ static int ipw_tx_skb(struct ipw_priv *p
 					 remaining_bytes,
 					 PCI_DMA_TODEVICE));
 
-			le32_add_cpu(&tfd->u.data.num_chunks, 1);
+			tfd->u.data.num_chunks =
+			    cpu_to_le32(le32_to_cpu(tfd->u.data.num_chunks) +
+					1);
 		}
 	}
 
diff -up linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2400pci.c.orig linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2400pci.c
--- linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2400pci.c.orig	2008-04-01 22:40:04.000000000 -0400
+++ linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2400pci.c	2008-04-01 22:40:52.000000000 -0400
@@ -29,6 +29,7 @@
 #include <linux/init.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
+#include <linux/version.h>
 #include <linux/pci.h>
 #include <linux/eeprom_93cx6.h>
 
@@ -262,6 +263,7 @@ static void rt2400pci_brightness_set(str
 	rt2x00pci_register_write(led->rt2x00dev, LEDCSR, reg);
 }
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 static int rt2400pci_blink_set(struct led_classdev *led_cdev,
 			       unsigned long *delay_on,
 			       unsigned long *delay_off)
@@ -277,6 +279,7 @@ static int rt2400pci_blink_set(struct le
 
 	return 0;
 }
+#endif
 #endif /* CONFIG_RT2400PCI_LEDS */
 
 /*
@@ -1302,8 +1305,10 @@ static int rt2400pci_init_eeprom(struct 
 	rt2x00dev->led_radio.type = LED_TYPE_RADIO;
 	rt2x00dev->led_radio.led_dev.brightness_set =
 	    rt2400pci_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 	rt2x00dev->led_radio.led_dev.blink_set =
 	    rt2400pci_blink_set;
+#endif
 	rt2x00dev->led_radio.flags = LED_INITIALIZED;
 
 	if (value == LED_MODE_TXRX_ACTIVITY) {
@@ -1311,8 +1316,10 @@ static int rt2400pci_init_eeprom(struct 
 		rt2x00dev->led_radio.type = LED_TYPE_ACTIVITY;
 		rt2x00dev->led_qual.led_dev.brightness_set =
 		    rt2400pci_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 		rt2x00dev->led_qual.led_dev.blink_set =
 		    rt2400pci_blink_set;
+#endif
 		rt2x00dev->led_qual.flags = LED_INITIALIZED;
 	}
 #endif /* CONFIG_RT2400PCI_LEDS */
diff -up linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2500usb.c.orig linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2500usb.c
--- linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2500usb.c.orig	2008-04-01 22:40:04.000000000 -0400
+++ linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2500usb.c	2008-04-01 22:41:43.000000000 -0400
@@ -29,6 +29,7 @@
 #include <linux/init.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
+#include <linux/version.h>
 #include <linux/usb.h>
 
 #include "rt2x00.h"
@@ -301,6 +302,7 @@ static void rt2500usb_brightness_set(str
 	rt2500usb_register_write(led->rt2x00dev, MAC_CSR20, reg);
 }
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 static int rt2500usb_blink_set(struct led_classdev *led_cdev,
 			       unsigned long *delay_on,
 			       unsigned long *delay_off)
@@ -316,6 +318,7 @@ static int rt2500usb_blink_set(struct le
 
 	return 0;
 }
+#endif
 #endif /* CONFIG_RT2500USB_LEDS */
 
 /*
@@ -1388,8 +1391,10 @@ static int rt2500usb_init_eeprom(struct 
 	rt2x00dev->led_radio.type = LED_TYPE_RADIO;
 	rt2x00dev->led_radio.led_dev.brightness_set =
 	    rt2500usb_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 	rt2x00dev->led_radio.led_dev.blink_set =
 	    rt2500usb_blink_set;
+#endif
 	rt2x00dev->led_radio.flags = LED_INITIALIZED;
 
 	if (value == LED_MODE_TXRX_ACTIVITY) {
@@ -1397,8 +1402,10 @@ static int rt2500usb_init_eeprom(struct 
 		rt2x00dev->led_radio.type = LED_TYPE_ACTIVITY;
 		rt2x00dev->led_qual.led_dev.brightness_set =
 		    rt2500usb_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 		rt2x00dev->led_qual.led_dev.blink_set =
 		    rt2500usb_blink_set;
+#endif
 		rt2x00dev->led_qual.flags = LED_INITIALIZED;
 	}
 #endif /* CONFIG_RT2500USB_LEDS */
diff -up linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2500pci.c.orig linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2500pci.c
--- linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2500pci.c.orig	2008-04-01 22:40:04.000000000 -0400
+++ linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2500pci.c	2008-04-01 22:41:17.000000000 -0400
@@ -29,6 +29,7 @@
 #include <linux/init.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
+#include <linux/version.h>
 #include <linux/pci.h>
 #include <linux/eeprom_93cx6.h>
 
@@ -262,6 +263,7 @@ static void rt2500pci_brightness_set(str
 	rt2x00pci_register_write(led->rt2x00dev, LEDCSR, reg);
 }
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 static int rt2500pci_blink_set(struct led_classdev *led_cdev,
 			       unsigned long *delay_on,
 			       unsigned long *delay_off)
@@ -277,6 +279,7 @@ static int rt2500pci_blink_set(struct le
 
 	return 0;
 }
+#endif
 #endif /* CONFIG_RT2500PCI_LEDS */
 
 /*
@@ -1479,8 +1482,10 @@ static int rt2500pci_init_eeprom(struct 
 	rt2x00dev->led_radio.type = LED_TYPE_RADIO;
 	rt2x00dev->led_radio.led_dev.brightness_set =
 	    rt2500pci_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 	rt2x00dev->led_radio.led_dev.blink_set =
 	    rt2500pci_blink_set;
+#endif
 	rt2x00dev->led_radio.flags = LED_INITIALIZED;
 
 	if (value == LED_MODE_TXRX_ACTIVITY) {
@@ -1488,8 +1493,10 @@ static int rt2500pci_init_eeprom(struct 
 		rt2x00dev->led_radio.type = LED_TYPE_ACTIVITY;
 		rt2x00dev->led_qual.led_dev.brightness_set =
 		    rt2500pci_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 		rt2x00dev->led_qual.led_dev.blink_set =
 		    rt2500pci_blink_set;
+#endif
 		rt2x00dev->led_qual.flags = LED_INITIALIZED;
 	}
 #endif /* CONFIG_RT2500PCI_LEDS */
diff -up linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2x00leds.c.orig linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2x00leds.c
--- linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2x00leds.c.orig	2008-04-01 22:28:02.000000000 -0400
+++ linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt2x00leds.c	2008-04-01 22:30:06.000000000 -0400
@@ -25,6 +25,7 @@
 
 #include <linux/kernel.h>
 #include <linux/module.h>
+#include <linux/version.h>
 
 #include "rt2x00.h"
 #include "rt2x00lib.h"
@@ -127,8 +128,10 @@ void rt2x00leds_register(struct rt2x00_d
 	char dev_name[16];
 	char name[32];
 	int retval;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 	unsigned long on_period;
 	unsigned long off_period;
+#endif
 
 	snprintf(dev_name, sizeof(dev_name), "%s-%s",
 		 rt2x00dev->ops->name, wiphy_name(rt2x00dev->hw->wiphy));
@@ -163,6 +166,7 @@ void rt2x00leds_register(struct rt2x00_d
 			goto exit_fail;
 	}
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 	/*
 	 * Initialize blink time to default value:
 	 * On period: 70ms
@@ -174,6 +178,7 @@ void rt2x00leds_register(struct rt2x00_d
 		rt2x00dev->led_radio.led_dev.blink_set(
 		    &rt2x00dev->led_radio.led_dev, &on_period, &off_period);
 	}
+#endif
 
 	return;
 
diff -up linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt61pci.c.orig linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt61pci.c
--- linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt61pci.c.orig	2008-04-01 22:40:04.000000000 -0400
+++ linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt61pci.c	2008-04-01 22:42:09.000000000 -0400
@@ -30,6 +30,7 @@
 #include <linux/init.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
+#include <linux/version.h>
 #include <linux/pci.h>
 #include <linux/eeprom_93cx6.h>
 
@@ -315,6 +316,7 @@ static void rt61pci_brightness_set(struc
 	}
 }
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 static int rt61pci_blink_set(struct led_classdev *led_cdev,
 			     unsigned long *delay_on,
 			     unsigned long *delay_off)
@@ -330,6 +332,7 @@ static int rt61pci_blink_set(struct led_
 
 	return 0;
 }
+#endif
 #endif /* CONFIG_RT61PCI_LEDS */
 
 /*
@@ -2073,16 +2076,20 @@ static int rt61pci_init_eeprom(struct rt
 	rt2x00dev->led_radio.type = LED_TYPE_RADIO;
 	rt2x00dev->led_radio.led_dev.brightness_set =
 	    rt61pci_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 	rt2x00dev->led_radio.led_dev.blink_set =
 	    rt61pci_blink_set;
+#endif
 	rt2x00dev->led_radio.flags = LED_INITIALIZED;
 
 	rt2x00dev->led_assoc.rt2x00dev = rt2x00dev;
 	rt2x00dev->led_assoc.type = LED_TYPE_ASSOC;
 	rt2x00dev->led_assoc.led_dev.brightness_set =
 	    rt61pci_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 	rt2x00dev->led_assoc.led_dev.blink_set =
 	    rt61pci_blink_set;
+#endif
 	rt2x00dev->led_assoc.flags = LED_INITIALIZED;
 
 	if (value == LED_MODE_SIGNAL_STRENGTH) {
@@ -2090,8 +2097,10 @@ static int rt61pci_init_eeprom(struct rt
 		rt2x00dev->led_radio.type = LED_TYPE_QUALITY;
 		rt2x00dev->led_qual.led_dev.brightness_set =
 		    rt61pci_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 		rt2x00dev->led_qual.led_dev.blink_set =
 		    rt61pci_blink_set;
+#endif
 		rt2x00dev->led_qual.flags = LED_INITIALIZED;
 	}
 
diff -up linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt73usb.c.orig linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt73usb.c
--- linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt73usb.c.orig	2008-04-01 22:40:04.000000000 -0400
+++ linux-2.6.24.noarch/drivers/net/wireless/rt2x00/rt73usb.c	2008-04-01 22:42:33.000000000 -0400
@@ -30,6 +30,7 @@
 #include <linux/init.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
+#include <linux/version.h>
 #include <linux/usb.h>
 
 #include "rt2x00.h"
@@ -320,6 +321,7 @@ static void rt73usb_brightness_set(struc
 	}
 }
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 static int rt73usb_blink_set(struct led_classdev *led_cdev,
 			     unsigned long *delay_on,
 			     unsigned long *delay_off)
@@ -335,6 +337,7 @@ static int rt73usb_blink_set(struct led_
 
 	return 0;
 }
+#endif
 #endif /* CONFIG_RT73USB_LEDS */
 
 /*
@@ -1633,16 +1636,20 @@ static int rt73usb_init_eeprom(struct rt
 	rt2x00dev->led_radio.type = LED_TYPE_RADIO;
 	rt2x00dev->led_radio.led_dev.brightness_set =
 	    rt73usb_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 	rt2x00dev->led_radio.led_dev.blink_set =
 	    rt73usb_blink_set;
+#endif
 	rt2x00dev->led_radio.flags = LED_INITIALIZED;
 
 	rt2x00dev->led_assoc.rt2x00dev = rt2x00dev;
 	rt2x00dev->led_assoc.type = LED_TYPE_ASSOC;
 	rt2x00dev->led_assoc.led_dev.brightness_set =
 	    rt73usb_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 	rt2x00dev->led_assoc.led_dev.blink_set =
 	    rt73usb_blink_set;
+#endif
 	rt2x00dev->led_assoc.flags = LED_INITIALIZED;
 
 	if (value == LED_MODE_SIGNAL_STRENGTH) {
@@ -1650,8 +1657,10 @@ static int rt73usb_init_eeprom(struct rt
 		rt2x00dev->led_radio.type = LED_TYPE_QUALITY;
 		rt2x00dev->led_qual.led_dev.brightness_set =
 		    rt73usb_brightness_set;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,24))
 		rt2x00dev->led_qual.led_dev.blink_set =
 		    rt73usb_blink_set;
+#endif
 		rt2x00dev->led_qual.flags = LED_INITIALIZED;
 	}
 
