diff -up linux-2.6.24.noarch/drivers/char/drm/radeon_state.c.da linux-2.6.24.noarch/drivers/char/drm/radeon_state.c
--- linux-2.6.24.noarch/drivers/char/drm/radeon_state.c.da	2008-03-31 11:28:35.000000000 +1000
+++ linux-2.6.24.noarch/drivers/char/drm/radeon_state.c	2008-03-31 11:28:48.000000000 +1000
@@ -3088,7 +3088,8 @@ static int radeon_cp_setparam(struct drm
 			dev_priv->gart_info.table_size = RADEON_PCIGART_TABLE_SIZE;
 		break;
 	case RADEON_SETPARAM_VBLANK_CRTC:
-		return radeon_vblank_crtc_set(dev, sp->value);
+		if (dev_priv)
+			return radeon_vblank_crtc_set(dev, sp->value);
 		break;
 	default:
 		DRM_DEBUG("Invalid parameter %d\n", sp->param);
