diff --git a/arch/x86/kernel/alternative.c b/arch/x86/kernel/alternative.c
index d6405e0..2d00597 100644
--- a/arch/x86/kernel/alternative.c
+++ b/arch/x86/kernel/alternative.c
@@ -151,8 +151,10 @@ static const struct nop {
 } noptypes[] = {
 	{ X86_FEATURE_K8, k8_nops },
 	{ X86_FEATURE_K7, k7_nops },
+#ifndef CONFIG_X86_GENERIC
 	{ X86_FEATURE_P4, p6_nops },
 	{ X86_FEATURE_P3, p6_nops },
+#endif
 	{ -1, NULL }
 };
 
