--- a/include/drm/drmP.h~	2009-10-21 00:07:36.000000000 -0400
+++ b/include/drm/drmP.h	2009-10-21 00:08:22.000000000 -0400
@@ -255,9 +255,10 @@
 do {									\
 	if (!_DRM_LOCK_IS_HELD(file_priv->master->lock.hw_lock->lock) ||		\
 	    file_priv->master->lock.file_priv != file_priv)	{			\
-		DRM_ERROR( "%s called without lock held, held  %d owner %p %p\n",\
-			   __func__, _DRM_LOCK_IS_HELD(file_priv->master->lock.hw_lock->lock),\
-			   file_priv->master->lock.file_priv, file_priv);		\
+		if (printk_ratelimit())							\
+			DRM_ERROR( "%s called without lock held, held  %d owner %p %p\n",\
+				   __func__, _DRM_LOCK_IS_HELD(file_priv->master->lock.hw_lock->lock),\
+				   file_priv->master->lock.file_priv, file_priv);		\
 		return -EINVAL;						\
 	}								\
 } while (0)
