Index: linux-2.6.29.noarch/init/do_mounts_rd.c
===================================================================
--- linux-2.6.29.noarch.orig/init/do_mounts_rd.c
+++ linux-2.6.29.noarch/init/do_mounts_rd.c
@@ -110,6 +110,7 @@ identify_ramdisk_image(int fd, int start
 	}
 
 	/* squashfs is at block zero too */
+#if defined(CONFIG_SQUASHFS) || defined(CONFIG_SQUASHFS_MODULE)
 	if (le32_to_cpu(squashfsb->s_magic) == SQUASHFS_MAGIC) {
 		printk(KERN_NOTICE
 		       "RAMDISK: squashfs filesystem found at block %d\n",
@@ -118,8 +119,7 @@ identify_ramdisk_image(int fd, int start
 			 >> BLOCK_SIZE_BITS;
 		goto done;
 	}
-
-	/* squashfs is at block zero too */
+#else  /* v3 */
 	if (squashfsb->s_magic == SQUASHFS_MAGIC) {
 		printk(KERN_NOTICE
 		       "RAMDISK: squashfs filesystem found at block %d\n",
@@ -130,6 +130,7 @@ identify_ramdisk_image(int fd, int start
 			nblocks = (squashfsb->bytes_used+BLOCK_SIZE-1)>>BLOCK_SIZE_BITS;
 		goto done;
 	}
+#endif
 
 	/*
 	 * Read block 1 to test for minix and ext2 superblock
