--- tools/power/x86/x86_energy_perf_policy/Makefile.orig	2013-03-13 16:41:37.149775903 +0000
+++ tools/power/x86/x86_energy_perf_policy/Makefile	2013-03-13 16:42:37.343777583 +0000
@@ -6,5 +6,7 @@
 	rm -f x86_energy_perf_policy
 
 install :
-	install x86_energy_perf_policy ${DESTDIR}/usr/bin/
-	install x86_energy_perf_policy.8 ${DESTDIR}/usr/share/man/man8/
+	install -d  $(DESTDIR)$(PREFIX)/bin
+	install x86_energy_perf_policy ${DESTDIR}$(PREFIX)/bin/
+	install -d  $(DESTDIR)$(PREFIX)/share/man/man8
+	install x86_energy_perf_policy.8 ${DESTDIR}$(PREFIX)/share/man/man8/

