--- httpAdapter.c.orig	2009-02-11 19:12:30.000000000 -0500
+++ httpAdapter.c	2009-04-15 13:25:27.000000000 -0400
@@ -54,6 +54,10 @@
 #include <sys/ipc.h>
 #include <sys/sem.h>
 
+#ifdef HAVE_SYS_STAT_H
+#include <sys/stat.h>
+#endif
+
 #ifdef HAVE_SYS_FSUID_H
 #include <sys/fsuid.h>
 #endif
@@ -64,7 +68,6 @@
 
 #ifdef HAVE_UDS
 #include <grp.h>
-#include <sys/stat.h>
 #endif
 
 unsigned long exFlags = 0;
@@ -1441,6 +1444,19 @@
 
 }
 
+int isDir(const char *path)
+{
+#if defined (HAVE_SYS_STAT_H) && defined (USE_SSL)
+   struct stat sb;
+   int rc;
+   rc = stat(path, &sb);
+   if (rc == -1)
+      intSSLerror("Error opening the client trust store");
+   return (sb.st_mode & S_IFMT) == S_IFDIR;
+#else
+   return 0;
+#endif
+}
 
 int httpDaemon(int argc, char *argv[], int sslMode, int sfcbPid)
 {
@@ -1713,6 +1729,7 @@
 #if defined USE_SSL
     if (sfcbSSLMode) {
        char *fnc,*fnk, *fnt, *fnl;
+       int rc;
        ctx = SSL_CTX_new(SSLv23_method());
        getControlChars("sslCertificateFilePath", &fnc);
        _SFCB_TRACE(1,("---  sslCertificateFilePath = %s",fnc));
@@ -1740,9 +1757,15 @@
        } 
        getControlChars("sslClientTrustStore", &fnt);
        _SFCB_TRACE(1,("---  sslClientTrustStore = %s",fnt));
-       if (ccVerifyMode != CC_VERIFY_IGNORE &&
-	   SSL_CTX_load_verify_locations(ctx, fnt, NULL) != 1)
-	 intSSLerror("Error locating the client trust store");
+
+       if (ccVerifyMode != CC_VERIFY_IGNORE) {
+           if (isDir(fnt))
+              rc = SSL_CTX_load_verify_locations(ctx, NULL, fnt);
+           else
+              rc = SSL_CTX_load_verify_locations(ctx, fnt, NULL);
+           if (rc != 1)
+              intSSLerror("Error locating the client trust store");
+       }
 
        /* SSLv2 is pretty old; no one should be needing it any more */
        SSL_CTX_set_options(ctx, SSL_OP_ALL | SSL_OP_NO_SSLv2 | 
--- man/sfcbd.1.pre.in.orig	2009-04-15 13:26:47.000000000 -0400
+++ man/sfcbd.1.pre.in	2009-04-15 13:41:48.000000000 -0400
@@ -174,10 +174,19 @@
 set to \fItrue\fR. Default=\fI@sysconfdir@/sfcb/file.pem\fR
 .TP
 .B sslClientTrustStore
-Specify the name of the file containing the either the CA certificate
+Specify the name of the file or directory containing either the CA certificate
 issuing client certificates or self-signed certificates of the clients.
-This file must be in PEM format and is only required if sslClientCertificate
-is set to \fIaccept\fR or \fIrequire\fR. Default=\fI@sysconfdir@/sfcb/client.pem\fR
+This setting is only required if sslClientCertificate is set to \fIaccept\fR
+or \fIrequire\fR.
+
+If a file, it must be in PEM format.
+
+If a directory, it must be in "hash format": each file in the directory should
+be named \fIhash.0\fR, where \fIhash\fR is the output of 
+\fIopenssl x509 -hash\fR. See the \fI-CApath\fR option in \fIverify(1)\fR
+from the \fIopenssl\fR package for more information about the hash format.
+
+Default=\fI@sysconfdir@/sfcb/client.pem\fR
 .TP
 .B sslClientCertificate
 Specify the way sfcb handles client certificate based authentication. If set to
