diff -ur gaim-1.2.1.orig/src/log.c gaim-1.2.1/src/log.c
--- gaim-1.2.1.orig/src/log.c	2005-02-09 19:07:22.000000000 -0500
+++ gaim-1.2.1/src/log.c	2005-04-11 11:28:18.000000000 -0400
@@ -336,8 +336,8 @@
 void gaim_log_init(void)
 {
 	gaim_prefs_add_none("/core/logging");
-	gaim_prefs_add_bool("/core/logging/log_ims", FALSE);
-	gaim_prefs_add_bool("/core/logging/log_chats", FALSE);
+	gaim_prefs_add_bool("/core/logging/log_ims", TRUE);
+	gaim_prefs_add_bool("/core/logging/log_chats", TRUE);
 	gaim_prefs_add_bool("/core/logging/log_system", FALSE);
 	gaim_prefs_add_bool("/core/logging/log_signon_signoff", FALSE);
 	gaim_prefs_add_bool("/core/logging/log_idle_state", FALSE);
