diff -urp empathy-2.27.92.OLD/libempathy/libempathy.pc.in empathy-2.27.92/libempathy/libempathy.pc.in
--- empathy-2.27.92.OLD/libempathy/libempathy.pc.in	2009-09-01 04:57:31.000000000 -0400
+++ empathy-2.27.92/libempathy/libempathy.pc.in	2009-09-08 12:55:21.078269583 -0400
@@ -5,8 +5,7 @@ includedir=@includedir@
 
 Name: libempathy
 Description:  Empathy base library
-Requires: pkg-config >= 0.21
-Requires.private: glib-2.0, gobject-2.0, libxml-2.0, \
+Requires: glib-2.0, gobject-2.0, libxml-2.0, \
                   gio-2.0, gio-unix-2.0, telepathy-glib
 Version: @VERSION@
 Libs: -L${libdir} -lempathy
diff -urp empathy-2.27.92.OLD/libempathy-gtk/libempathy-gtk.pc.in empathy-2.27.92/libempathy-gtk/libempathy-gtk.pc.in
--- empathy-2.27.92.OLD/libempathy-gtk/libempathy-gtk.pc.in	2009-09-01 04:57:31.000000000 -0400
+++ empathy-2.27.92/libempathy-gtk/libempathy-gtk.pc.in	2009-09-08 12:55:40.418269081 -0400
@@ -5,8 +5,7 @@ includedir=@includedir@
 
 Name: libempathy-gtk
 Description:  Empathy interface library
-Requires: pkg-config >= 0.21
-Requires.private: glib-2.0, gobject-2.0, gtk+-2.0, \
+Requires: glib-2.0, gobject-2.0, gtk+-2.0, \
                   libempathy, telepathy-glib, libcanberra-gtk
 Version: @VERSION@
 Libs: -L${libdir} -lempathy-gtk
