--- stund/stun.cxx.old	2010-06-05 20:06:40.000000000 -0400
+++ stund/stun.cxx	2010-06-05 20:07:37.000000000 -0400
@@ -658,28 +658,9 @@
 		
       UInt64 tick;
 		
-#if defined(WIN32) 
-      volatile unsigned int lowtick=0,hightick=0;
-      __asm
-         {
-            rdtsc 
-               mov lowtick, eax
-               mov hightick, edx
-               }
-      tick = hightick;
-      tick <<= 32;
-      tick |= lowtick;
-#elif defined(__GNUC__) && ( defined(__i686__) || defined(__i386__) )
-      asm("rdtsc" : "=A" (tick));
-#elif defined (__SUNPRO_CC) || defined( __sparc__ )	
-      tick = gethrtime();
-#elif defined(__MACH__) 
-      int fd=open("/dev/random",O_RDONLY);
+      int fd=open("/dev/urandom",O_RDONLY);
       read(fd,&tick,sizeof(tick));
       closesocket(fd);
-#else
-#     error Need some way to seed the random number generator 
-#endif 
       int seed = int(tick);
 #ifdef WIN32
       srand(seed);
