Index: src/Extensions/Banshee.NotificationArea/Banshee.NotificationArea/GtkNotificationAreaBox.cs
===================================================================
--- src/Extensions/Banshee.NotificationArea/Banshee.NotificationArea/GtkNotificationAreaBox.cs	(revision 3460)
+++ src/Extensions/Banshee.NotificationArea/Banshee.NotificationArea/GtkNotificationAreaBox.cs	(working copy)
@@ -58,7 +58,7 @@
         
         public GtkNotificationAreaBox (BaseClientWindow window)
         {
-            IconName = "music-player-banshee";
+            IconName = Banshee.ServiceStack.Application.IconName;
             
             Tooltip = window.Title;
             window.TitleChanged += delegate { Tooltip = window.Title; };
Index: src/Extensions/Banshee.NotificationArea/Banshee.NotificationArea/NotificationAreaService.cs
===================================================================
--- src/Extensions/Banshee.NotificationArea/Banshee.NotificationArea/NotificationAreaService.cs	(revision 3460)
+++ src/Extensions/Banshee.NotificationArea/Banshee.NotificationArea/NotificationAreaService.cs	(working copy)
@@ -249,7 +249,7 @@
         {
             try {
                 if (NotifyOnCloseSchema.Get ()) {
-                    Gdk.Pixbuf image = IconThemeUtils.LoadIcon (48, "music-player-banshee");
+                    Gdk.Pixbuf image = IconThemeUtils.LoadIcon (48, Banshee.ServiceStack.Application.IconName);
                     if (image != null) {
                         image = image.ScaleSimple (42, 42, Gdk.InterpType.Bilinear);
                     }
Index: src/Extensions/Banshee.NotificationArea/Banshee.NotificationArea/X11NotificationAreaBox.cs
===================================================================
--- src/Extensions/Banshee.NotificationArea/Banshee.NotificationArea/X11NotificationAreaBox.cs	(revision 3460)
+++ src/Extensions/Banshee.NotificationArea/Banshee.NotificationArea/X11NotificationAreaBox.cs	(working copy)
@@ -59,7 +59,7 @@
         {
             event_box = new EventBox ();
             image = new Image ();
-            image.IconName = "music-player-banshee";
+            image.IconName = Banshee.ServiceStack.Application.IconName;
             event_box.Add (image);
             Add (event_box);
             
Index: src/Core/Banshee.ThickClient/Banshee.Gui/GtkBaseClient.cs
===================================================================
--- src/Core/Banshee.ThickClient/Banshee.Gui/GtkBaseClient.cs	(revision 3460)
+++ src/Core/Banshee.ThickClient/Banshee.Gui/GtkBaseClient.cs	(working copy)
@@ -57,7 +57,7 @@
         
         private string default_icon_name;
         
-        protected GtkBaseClient () : this (true, "music-player-banshee")
+        protected GtkBaseClient () : this (true, Application.IconName)
         {
         }
         
Index: src/Core/Banshee.Services/Banshee.ServiceStack/Application.cs
===================================================================
--- src/Core/Banshee.Services/Banshee.ServiceStack/Application.cs	(revision 3460)
+++ src/Core/Banshee.Services/Banshee.ServiceStack/Application.cs	(working copy)
@@ -167,6 +167,10 @@
             get { return "banshee"; }
         }
         
+        public static string IconName {
+            get { return "media-player-banshee"; }
+        }
+        
         private static string version;
         public static string Version {
             get { 
