diff -ruN SDL/SDL-1.2.11/configure.in SDL-1.2.11.dyn/configure.in
--- SDL/SDL-1.2.11/configure.in	2006-06-26 23:48:33.000000000 -0500
+++ SDL-1.2.11.dyn/configure.in	2008-05-07 15:36:42.000000000 -0500
@@ -845,7 +845,8 @@
                     ;;
                 *)
                     x11_lib_spec=[`echo $X_LIBS | sed 's/.*-L\([^ ]*\).*/\1/'`]
-                    for path in $x11_lib_path /usr/lib /usr/X11/lib /usr/X11R6/lib; do
+                    pfx=${libdir##*/}
+                    for path in $x11_lib_path /usr/${pfx} /usr/lib /usr/X11/lib /usr/X11R6/lib; do
                         if test "x$x11_lib" = "x"; then
                             x11_lib=[`ls -- $path/libX11.so.[0-9] 2>/dev/null | sort -r | sed 's/.*\/\(.*\)/\1/; q'`]
                             if test "x$x11_lib" = "x"; then
