diff -ur gnome-backgrounds-2.10.0.orig/translucent/gnome-translucent.xml.in.in gnome-backgrounds-2.10.0/translucent/gnome-translucent.xml.in.in
--- gnome-backgrounds-2.10.0.orig/translucent/gnome-translucent.xml.in.in	2005-01-24 12:18:25.000000000 -0500
+++ gnome-backgrounds-2.10.0/translucent/gnome-translucent.xml.in.in	2005-03-19 21:08:44.398876560 -0500
@@ -25,4 +25,12 @@
     <pcolor>#76848F</pcolor>
     <scolor>#76848F</scolor>
   </wallpaper>
+  <wallpaper deleted="false">
+    <_name>Foresight (Translucent)</_name>
+    <filename>@datadir@/pixmaps/backgrounds/gnome/translucent/logo-foresight-trans.png</filename>
+    <options>centered</options>
+    <shade_type>horizontal-gradient</shade_type>
+    <pcolor>#76848F</pcolor>
+    <scolor>#76848F</scolor>
+  </wallpaper>
 </wallpapers>
Only in gnome-backgrounds-2.10.0/translucent: logo-foresight-trans.png
diff -ur gnome-backgrounds-2.10.0.orig/translucent/Makefile.am gnome-backgrounds-2.10.0/translucent/Makefile.am
--- gnome-backgrounds-2.10.0.orig/translucent/Makefile.am	2005-01-24 12:18:08.000000000 -0500
+++ gnome-backgrounds-2.10.0/translucent/Makefile.am	2005-03-19 21:07:11.614981856 -0500
@@ -3,7 +3,8 @@
 background_DATA =\
 	ellipsis.png \
 	rects.png \
-	tentacles.png
+	tentacles.png \
+	logo-foresight-trans.png
 
 metadata_in_files = gnome-translucent.xml.in
 metadatadir = $(datadir)/gnome-background-properties
diff -ur gnome-backgrounds-2.10.0.orig/translucent/Makefile.in gnome-backgrounds-2.10.0/translucent/Makefile.in
--- gnome-backgrounds-2.10.0.orig/translucent/Makefile.in	2005-03-07 13:14:28.000000000 -0500
+++ gnome-backgrounds-2.10.0/translucent/Makefile.in	2005-03-19 21:06:57.124184792 -0500
@@ -110,7 +110,7 @@
 
 backgrounddir = $(datadir)/pixmaps/backgrounds/gnome/translucent
 
-background_DATA =  	ellipsis.png 	rects.png 	tentacles.png
+background_DATA =  	ellipsis.png 	rects.png 	tentacles.png	logo-foresight-trans.png
 
 
 metadata_in_files = gnome-translucent.xml.in
