diff -Naur gnome-backgrounds-2.14.0.orig/tiles/gnome-tiles.xml.in.in gnome-backgrounds-2.14.0/tiles/gnome-tiles.xml.in.in
--- gnome-backgrounds-2.14.0.orig/tiles/gnome-tiles.xml.in.in	2006-03-11 13:30:33.000000000 -0500
+++ gnome-backgrounds-2.14.0/tiles/gnome-tiles.xml.in.in	2006-03-14 21:19:33.000000000 -0500
@@ -8,8 +8,8 @@
     <filename>@datadir@/pixmaps/backgrounds/gnome/tiles/pattern-boats.svg</filename>
     <options>wallpaper</options>
     <shade_type>vertical-gradient</shade_type>
-    <pcolor>#FCAF3E</pcolor>
-    <scolor>#555753</scolor>
+    <pcolor>#679167</pcolor>
+    <scolor>#679167</scolor>
   </wallpaper>
   <wallpaper deleted="false">
     <!-- Translators: This is the name of a desktop background image that shows a clouds pattern.
@@ -18,8 +18,8 @@
     <filename>@datadir@/pixmaps/backgrounds/gnome/tiles/pattern-clouds.svg</filename>
     <options>wallpaper</options>
     <shade_type>vertical-gradient</shade_type>
-    <pcolor>#729FCF</pcolor>
-    <scolor>#AED5FF</scolor>
+    <pcolor>#679167</pcolor>
+    <scolor>#679167</scolor>
   </wallpaper>
   <wallpaper deleted="false">
     <!-- Translators: This is the name of a desktop background image that shows a stars pattern.
@@ -28,7 +28,7 @@
     <filename>@datadir@/pixmaps/backgrounds/gnome/tiles/pattern-stars.svg</filename>
     <options>wallpaper</options>
     <shade_type>vertical-gradient</shade_type>
-    <pcolor>#332133</pcolor>
-    <scolor>#75507B</scolor>
+    <pcolor>#679167</pcolor>
+    <scolor>#679167</scolor>
   </wallpaper>
 </wallpapers>
diff -Naur gnome-backgrounds-2.14.0.orig/translucent/gnome-translucent.xml.in.in gnome-backgrounds-2.14.0/translucent/gnome-translucent.xml.in.in
--- gnome-backgrounds-2.14.0.orig/translucent/gnome-translucent.xml.in.in	2006-03-02 18:17:40.000000000 -0500
+++ gnome-backgrounds-2.14.0/translucent/gnome-translucent.xml.in.in	2006-03-14 21:22:56.000000000 -0500
@@ -4,11 +4,20 @@
   <wallpaper deleted="false">
     <!-- Translators: This is the name of a desktop background image that shows some ellipses.
          You might want to translate it into the equivalent words of your language  -->
+    <_name>Foresight (Translucent)</_name>
+    <filename>@datadir@/pixmaps/backgrounds/gnome/translucent/logo-foresight-trans.png</filename>
+    <options>stretched</options>
+    <shade_type>solid</shade_type>
+    <pcolor>#679167</pcolor>
+  </wallpaper>
+  <wallpaper deleted="false">
+    <!-- Translators: This is the name of a desktop background image that shows some ellipses.
+         You might want to translate it into the equivalent words of your language  -->
     <_name>Ellipses (Translucent)</_name>
     <filename>@datadir@/pixmaps/backgrounds/gnome/translucent/ellipsis.svg</filename>
     <options>stretched</options>
     <shade_type>solid</shade_type>
-    <pcolor>#BABDB6</pcolor>
+    <pcolor></pcolor>
   </wallpaper>
   <wallpaper deleted="false">
     <!-- Translators: This is the name of a desktop background image that shows some wavey lines that appear like tentacles.
@@ -17,6 +26,6 @@
     <filename>@datadir@/pixmaps/backgrounds/gnome/translucent/tentacles.svg</filename>
     <options>stretched</options>
     <shade_type>solid</shade_type>
-    <pcolor>#366B04</pcolor>
+    <pcolor>#679167</pcolor>
   </wallpaper>
 </wallpapers>
diff -Naur gnome-backgrounds-2.14.0.orig/translucent/Makefile.am gnome-backgrounds-2.14.0/translucent/Makefile.am
--- gnome-backgrounds-2.14.0.orig/translucent/Makefile.am	2006-01-28 07:19:48.000000000 -0500
+++ gnome-backgrounds-2.14.0/translucent/Makefile.am	2006-03-14 21:23:33.000000000 -0500
@@ -1,8 +1,9 @@
 backgrounddir = $(datadir)/pixmaps/backgrounds/gnome/translucent
 
 background_DATA =\
-	ellipsis.svg \
-	tentacles.svg
+	ellipsis.svg  \
+	tentacles.svg \
+       logo-foresight-trans.png
 
 metadata_in_files = gnome-translucent.xml.in
 metadatadir = $(datadir)/gnome-background-properties
diff -Naur gnome-backgrounds-2.14.0.orig/translucent/Makefile.in gnome-backgrounds-2.14.0/translucent/Makefile.in
--- gnome-backgrounds-2.14.0.orig/translucent/Makefile.in	2006-03-13 15:03:18.000000000 -0500
+++ gnome-backgrounds-2.14.0/translucent/Makefile.in	2006-03-14 21:24:52.000000000 -0500
@@ -174,8 +174,9 @@
 target_alias = @target_alias@
 backgrounddir = $(datadir)/pixmaps/backgrounds/gnome/translucent
 background_DATA = \
-	ellipsis.svg \
-	tentacles.svg
+	ellipsis.svg  \
+	tentacles.svg \
+       logo-foresight-trans.png
 
 metadata_in_files = gnome-translucent.xml.in
 metadatadir = $(datadir)/gnome-background-properties
