diff -r 4f32642992c8 scripts/mk-images
--- a/scripts/mk-images	Mon Jul 26 12:59:17 2010 -0400
+++ b/scripts/mk-images	Thu Aug 12 09:09:31 2010 -0400
@@ -699,6 +699,10 @@
                 mkdir -p $MBD_DIR/modules/firmware/zd1211
                 cp $KERNELROOT/lib/firmware/zd1211/* $MBD_DIR/modules/firmware/zd1211
                 ;;
+            bnx2)
+                mkdir -p $MBD_DIR/modules/firmware/bnx2
+                cp $KERNELROOT/lib/firmware/bnx2/* $MBD_DIR/modules/firmware/bnx2
+                ;;
         esac
     done
 
