diff -r 35685b4edd2f Makefile.inc
--- a/Makefile.inc	Mon Jul 27 14:39:16 2009 -0400
+++ b/Makefile.inc	Wed Aug 19 15:53:59 2009 +0200
@@ -32,5 +32,5 @@
 
 # uncomment to build 32bit on an 64bit host
 #M32FLAGS = -m32
-CFLAGS = $(RPM_OPT_FLAGS) -Wall -Werror -D_GNU_SOURCE=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 $(SELINUXFLAGS) $(M32FLAGS)
+CFLAGS = $(RPM_OPT_FLAGS) -Wall -D_GNU_SOURCE=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 $(SELINUXFLAGS) $(M32FLAGS)
 LDFLAGS = $(M32FLAGS)
diff -r 35685b4edd2f isys/isys.c
--- a/isys/isys.c	Mon Jul 27 14:39:16 2009 -0400
+++ b/isys/isys.c	Wed Aug 19 15:53:59 2009 +0200
@@ -4,8 +4,6 @@
 #include <dirent.h>
 #include <errno.h>
 #define u32 __u32
-#include <linux/ext2_fs.h>
-#include <linux/ext3_fs.h>
 #include <ext2fs/ext2fs.h>
 #include <fcntl.h>
 #include <popt.h>
diff -r 35685b4edd2f loader2/Makefile
--- a/loader2/Makefile	Mon Jul 27 14:39:16 2009 -0400
+++ b/loader2/Makefile	Wed Aug 19 15:53:59 2009 +0200
@@ -138,7 +138,7 @@
 	install -m 644 loader.tr $(DESTDIR)/$(RUNTIMEDIR)/loader
 
 dirbrowser: dirbrowser.c
-	gcc -DSTANDALONE -D_FORTIFY_SOURCE=2 -Wall -Werror -ggdb -o dirbrowser dirbrowser.c -lnewt -lslang
+	gcc -DSTANDALONE -D_FORTIFY_SOURCE=2 -Wall -ggdb -o dirbrowser dirbrowser.c -lnewt -lslang
 
 ifeq (.depend,$(wildcard .depend))
 include .depend
