diff -ur anaconda-10.0.orig/product.py anaconda-10.0/product.py
--- anaconda-10.0.orig/product.py	2004-09-10 15:52:35.000000000 -0400
+++ anaconda-10.0/product.py	2005-03-01 16:32:26.885018032 -0500
@@ -22,21 +22,21 @@
 if path is None:
     productName = "anaconda"
     productVersion = "bluesky"
-    productPath = "Specifix"
+    productPath = "Foresight"
 else:
     f = open(path, "r")
     lines = f.readlines()
     if len(lines) < 4:
         productName = "anaconda"
         productVersion = "bluesky"
-	productPath = "Specifix"
+	productPath = "Foresight"
     else:
         productName = lines[1][:-1]
         productVersion = lines[2][:-1]
 	productPath = lines[3][:-1]
 
 # XXX hardcoded
-if productPath == "Specifix":
+if productPath == "Foresight":
     productPackagePath=productPath + "/changesets"
 else:
     productPackagePath=productPath + "/RPMS"
diff -ur anaconda-10.0.orig/scripts/mk-images anaconda-10.0/scripts/mk-images
--- anaconda-10.0.orig/scripts/mk-images	2005-02-23 15:09:34.000000000 -0500
+++ anaconda-10.0/scripts/mk-images	2005-03-01 16:30:29.832812672 -0500
@@ -610,7 +610,8 @@
     
     if [ $CONARY -eq 1 ]; then
         # XXX hard code for now
-	echo "installLabelPath conary.specifix.com@spx:devel contrib.specifix.com@spx:devel" > $MBD_DIR/etc/conaryrc
+        echo "autoResolve     True" > $MBD_DIR/etc/conaryrc
+        echo "installLabelPath conary.foresightlinux.com@fl:desktop conary.specifix.com@spx:devel contrib.specifix.com@spx:devel" >> $MBD_DIR/etc/conaryrc
     fi
 
     if [ "$BUILDARCH" = "ppc" -o "$BUILDARCH" = "ppc64" ]; then
diff -ur anaconda-10.0.orig/text.py anaconda-10.0/text.py
--- anaconda-10.0.orig/text.py	2005-01-31 15:14:46.000000000 -0500
+++ anaconda-10.0/text.py	2005-03-01 16:31:16.544711384 -0500
@@ -344,7 +344,7 @@
 	return WaitWindow(self.screen, title, text)
 
     def drawFrame(self):
-        self.welcomeText = _("%s (C) 2004-2005 Specifix, Inc.") % (productName,)
+        self.welcomeText = _("%s (C) 2005 Foresight Linux Project") % (productName,)
         self.screen.drawRootText (0, 0, self.welcomeText)
 	self.screen.drawRootText (len(_(self.welcomeText)), 0,
                                   (self.screen.width -
