diff -ur anaconda-10.0.orig/anaconda.conf anaconda-10.0/anaconda.conf
--- anaconda-10.0.orig/anaconda.conf	2004-04-19 11:41:49.000000000 -0400
+++ anaconda-10.0/anaconda.conf	2005-04-20 11:33:40.000000000 -0400
@@ -9,5 +9,5 @@
 Splashscreen		 	pixmaps/first.png
 WelcomeScreen			pixmaps/splash.png
 TitleBar		 	pixmaps/anaconda_header.png
-Title			  	Red Hat Linux Beta
+Title			  	Foresight Desktop Linux
 
diff -ur anaconda-10.0.orig/dispatch.py anaconda-10.0/dispatch.py
--- anaconda-10.0.orig/dispatch.py	2005-02-16 15:21:16.000000000 -0500
+++ anaconda-10.0/dispatch.py	2005-04-20 11:33:40.000000000 -0400
@@ -57,7 +57,7 @@
 
 installSteps = [
     ("welcome", ("id.configFileData",)),
-    ("betanag", betaNagScreen, ("intf", "dir")),
+    #("betanag", betaNagScreen, ("intf", "dir")),
     ("language", ("intf", "id.instLanguage")),
     ("keyboard", ("id.instLanguage.getDefaultKeyboard()", "id.keyboard")),
     ("checkmonitorok", checkMonitorOK, ("id.monitor", "dispatch")),
diff -ur anaconda-10.0.orig/installclass.py anaconda-10.0/installclass.py
--- anaconda-10.0.orig/installclass.py	2004-09-01 15:41:25.000000000 -0400
+++ anaconda-10.0/installclass.py	2005-04-20 11:33:45.000000000 -0400
@@ -31,7 +31,7 @@
     # default to not being hidden
     hidden = 0
     pixmap = None
-    showMinimal = 1
+    showMinimal = 0
     showLoginChoice = 0
     description = None
     name = "base"
diff -ur anaconda-10.0.orig/packages.py anaconda-10.0/packages.py
--- anaconda-10.0.orig/packages.py	2004-10-14 11:37:55.000000000 -0400
+++ anaconda-10.0/packages.py	2005-04-21 10:43:58.000000000 -0400
@@ -38,6 +38,7 @@
 from rhpl.log import log
 from rhpl.translate import _
 import rhpl.arch
+import bootloader
 
 def queryUpgradeContinue(intf, dir):
     if dir == DISPATCH_FORWARD:
@@ -1429,6 +1430,20 @@
                              "--depmod", "--install", kernelTag ],
                            stdout = None, stderr = None,
                            searchPath = 1, root = instRoot)
+    # get the path to the initrd
+    initrd = bootloader.makeInitrd("-%s" % (kernelTag,), instRoot)
+    # check to see if it exists, and that we can write to it
+    if not os.access(initrd, os.W_OK):
+        # no initrd file, nothing left to do
+        return
+    # open the initrd file, append to it
+    f = open(initrd, 'a')
+    iutil.execWithRedirect('/sbin/splash',
+                           ('/sbin/splash', '-s', '-f',
+            '/usr/share/bootsplash/themes/FDL/config/bootsplash-1024x768.cfg'),
+                           stdout = f, stderr = None,
+                           searchPath = 1, root = instRoot)
+    
 
 # XXX Deprecated.  Is this ever called anymore?
 def depmodModules(comps, instPath):
diff -ur anaconda-10.0.orig/product.py anaconda-10.0/product.py
--- anaconda-10.0.orig/product.py	2004-09-10 15:52:35.000000000 -0400
+++ anaconda-10.0/product.py	2005-04-20 11:33:40.000000000 -0400
@@ -22,21 +22,21 @@
 if path is None:
     productName = "anaconda"
     productVersion = "bluesky"
-    productPath = "Specifix"
+    productPath = "Foresight"
 else:
     f = open(path, "r")
     lines = f.readlines()
     if len(lines) < 4:
         productName = "anaconda"
         productVersion = "bluesky"
-	productPath = "Specifix"
+	productPath = "Foresight"
     else:
         productName = lines[1][:-1]
         productVersion = lines[2][:-1]
 	productPath = lines[3][:-1]
 
 # XXX hardcoded
-if productPath == "Specifix":
+if productPath == "Foresight":
     productPackagePath=productPath + "/changesets"
 else:
     productPackagePath=productPath + "/RPMS"
diff -ur anaconda-10.0.orig/scripts/mk-images anaconda-10.0/scripts/mk-images
--- anaconda-10.0.orig/scripts/mk-images	2005-02-23 15:09:34.000000000 -0500
+++ anaconda-10.0/scripts/mk-images	2005-04-20 11:33:40.000000000 -0400
@@ -610,7 +610,8 @@
     
     if [ $CONARY -eq 1 ]; then
         # XXX hard code for now
-	echo "installLabelPath conary.specifix.com@spx:devel contrib.specifix.com@spx:devel" > $MBD_DIR/etc/conaryrc
+        echo "autoResolve     True" > $MBD_DIR/etc/conaryrc
+        echo "installLabelPath conary.foresightlinux.com@fl:desktop conary.specifix.com@spx:devel contrib.specifix.com@spx:devel" >> $MBD_DIR/etc/conaryrc
     fi
 
     if [ "$BUILDARCH" = "ppc" -o "$BUILDARCH" = "ppc64" ]; then
diff -ur anaconda-10.0.orig/scripts/upd-instroot anaconda-10.0/scripts/upd-instroot
--- anaconda-10.0.orig/scripts/upd-instroot	2005-04-01 11:15:52.000000000 -0500
+++ anaconda-10.0/scripts/upd-instroot	2005-04-20 11:33:40.000000000 -0400
@@ -175,7 +175,8 @@
   	        procps anaconda kudzu hwdata bzip2 dosfstools pciutils
   	        gcc parted pyparted sed busybox:anaconda booty hdparm lvm2 rhpl
   	        libxml2 glib elfutils sqlite bogl krb5 jfsutils conary
-  	        db openssl pyxf86config device-mapper"
+  	        db openssl pyxf86config device-mapper gtk-theme-clearlooks
+		distro-release"
 
 # xpm is missing
 
@@ -219,7 +220,8 @@
                  ttfonts-ja ttfonts-zh_TW bitmap-fonts-cjk urw-fonts
                  fonts-arabic fonts-bengali libjpeg tcp_wrappers
                  system-config-date xfsprogs xfsdump dmapi attr acl
-                 xorg-x11-xfs xorg-x11 xorg-x11-fonts" 
+                 xorg-x11-xfs xorg-x11 xorg-x11-fonts 
+		 gtk-theme-clearlooks distro-release" 
 
 #
 # stuff ONLY included for rescue mode
@@ -648,7 +650,7 @@
 usr/$LIBDIR/gconv/*
 usr/$LIBDIR/gdk-pixbuf/loaders/*png*
 usr/$LIBDIR/gdk-pixbuf/loaders/*la*
-usr/$LIBDIR/gtk-2.0/*/engines/libbluecurve.so
+usr/$LIBDIR/gtk-2.0/*/engines/libclearlooks.so
 usr/$LIBDIR/gtk-2.0/*/loaders/*png*
 usr/$LIBDIR/gtk-2.0/*/loaders/*la*
 usr/$LIBDIR/gtk-2.0/immodules/
@@ -687,9 +689,7 @@
 usr/sbin/smartctl
 usr/share/anaconda/*
 usr/share/icons/default/*
-usr/share/icons/Bluecurve/index.theme
-usr/share/icons/Bluecurve/cursors/*
-usr/share/icons/Bluecurve/*/stock/*
+usr/share/icons/gnome/*
 usr/share/fonts/bitmap-fonts/fangsongti*
 usr/share/fonts/default/Type1/n019003l*
 usr/share/fonts/default/Type1/n019004l*
@@ -725,7 +725,7 @@
 usr/share/terminfo/x/xterm
 usr/share/terminfo/v/vt100
 usr/share/terminfo/v/vt100-nav
-usr/share/themes/Bluecurve/gtk-2.0/*
+usr/share/themes/Clearlooks/gtk-2.0/*
 usr/share/vte/termcap/xterm
 usr/share/zoneinfo/zone.tab
 EOF
diff -ur anaconda-10.0.orig/text.py anaconda-10.0/text.py
--- anaconda-10.0.orig/text.py	2005-01-31 15:14:46.000000000 -0500
+++ anaconda-10.0/text.py	2005-04-20 11:33:40.000000000 -0400
@@ -344,7 +344,7 @@
 	return WaitWindow(self.screen, title, text)
 
     def drawFrame(self):
-        self.welcomeText = _("%s (C) 2004-2005 Specifix, Inc.") % (productName,)
+        self.welcomeText = _("%s (C) 2005 Foresight Linux Project") % (productName,)
         self.screen.drawRootText (0, 0, self.welcomeText)
 	self.screen.drawRootText (len(_(self.welcomeText)), 0,
                                   (self.screen.width -
