diff -ui anaconda-10.0.orig/anaconda.conf anaconda-10.0/anaconda.conf
--- anaconda-10.0.orig/anaconda.conf	2004-04-19 11:41:49.000000000 -0400
+++ anaconda-10.0/anaconda.conf	2005-04-20 11:33:40.000000000 -0400
@@ -9,5 +9,5 @@
 Splashscreen		 	pixmaps/first.png
 WelcomeScreen			pixmaps/splash.png
 TitleBar		 	pixmaps/anaconda_header.png
-Title			  	Red Hat Linux Beta
+Title			  	Foresight Desktop Linux
 
Common subdirectories: anaconda-10.0.orig/bootdisk and anaconda-10.0/bootdisk
Common subdirectories: anaconda-10.0.orig/command-stubs and anaconda-10.0/command-stubs
diff -ui anaconda-10.0.orig/dispatch.py anaconda-10.0/dispatch.py
--- anaconda-10.0.orig/dispatch.py	2005-02-16 15:21:16.000000000 -0500
+++ anaconda-10.0/dispatch.py	2005-04-20 11:33:40.000000000 -0400
@@ -57,7 +57,7 @@
 
 installSteps = [
     ("welcome", ("id.configFileData",)),
-    ("betanag", betaNagScreen, ("intf", "dir")),
+    #("betanag", betaNagScreen, ("intf", "dir")),
     ("language", ("intf", "id.instLanguage")),
     ("keyboard", ("id.instLanguage.getDefaultKeyboard()", "id.keyboard")),
     ("checkmonitorok", checkMonitorOK, ("id.monitor", "dispatch")),
Common subdirectories: anaconda-10.0.orig/docs and anaconda-10.0/docs
Common subdirectories: anaconda-10.0.orig/fonts and anaconda-10.0/fonts
Common subdirectories: anaconda-10.0.orig/installclasses and anaconda-10.0/installclasses
diff -ui anaconda-10.0.orig/installclass.py anaconda-10.0/installclass.py
--- anaconda-10.0.orig/installclass.py	2004-09-01 15:41:25.000000000 -0400
+++ anaconda-10.0/installclass.py	2005-04-20 11:33:45.000000000 -0400
@@ -31,7 +31,7 @@
     # default to not being hidden
     hidden = 0
     pixmap = None
-    showMinimal = 1
+    showMinimal = 0
     showLoginChoice = 0
     description = None
     name = "base"
Common subdirectories: anaconda-10.0.orig/isomd5sum and anaconda-10.0/isomd5sum
Common subdirectories: anaconda-10.0.orig/isys and anaconda-10.0/isys
Common subdirectories: anaconda-10.0.orig/iw and anaconda-10.0/iw
Common subdirectories: anaconda-10.0.orig/keymaps and anaconda-10.0/keymaps
Common subdirectories: anaconda-10.0.orig/loader2 and anaconda-10.0/loader2
diff -ui anaconda-10.0.orig/packages.py anaconda-10.0/packages.py
--- anaconda-10.0.orig/packages.py	2004-10-14 11:37:55.000000000 -0400
+++ anaconda-10.0/packages.py	2005-04-21 10:43:58.000000000 -0400
@@ -38,6 +38,7 @@
 from rhpl.log import log
 from rhpl.translate import _
 import rhpl.arch
+import bootloader
 
 def queryUpgradeContinue(intf, dir):
     if dir == DISPATCH_FORWARD:
@@ -1429,6 +1430,20 @@
                              "--depmod", "--install", kernelTag ],
                            stdout = None, stderr = None,
                            searchPath = 1, root = instRoot)
+    # get the path to the initrd
+    initrd = bootloader.makeInitrd("-%s" % (kernelTag,), instRoot)
+    # check to see if it exists, and that we can write to it
+    if not os.access(initrd, os.W_OK):
+        # no initrd file, nothing left to do
+        return
+    # open the initrd file, append to it
+    f = open(initrd, 'a')
+    iutil.execWithRedirect('/sbin/splash',
+                           ('/sbin/splash', '-s', '-f',
+            '/usr/share/bootsplash/themes/FDL/config/bootsplash-1024x768.cfg'),
+                           stdout = f, stderr = None,
+                           searchPath = 1, root = instRoot)
+    
 
 # XXX Deprecated.  Is this ever called anymore?
 def depmodModules(comps, instPath):
Common subdirectories: anaconda-10.0.orig/pixmaps and anaconda-10.0/pixmaps
Common subdirectories: anaconda-10.0.orig/po and anaconda-10.0/po
diff -ui anaconda-10.0.orig/product.py anaconda-10.0/product.py
--- anaconda-10.0.orig/product.py	2004-09-10 15:52:35.000000000 -0400
+++ anaconda-10.0/product.py	2005-04-20 11:33:40.000000000 -0400
@@ -22,21 +22,21 @@
 if path is None:
     productName = "anaconda"
     productVersion = "bluesky"
-    productPath = "Specifix"
+    productPath = "Foresight"
 else:
     f = open(path, "r")
     lines = f.readlines()
     if len(lines) < 4:
         productName = "anaconda"
         productVersion = "bluesky"
-	productPath = "Specifix"
+	productPath = "Foresight"
     else:
         productName = lines[1][:-1]
         productVersion = lines[2][:-1]
 	productPath = lines[3][:-1]
 
 # XXX hardcoded
-if productPath == "Specifix":
+if productPath == "Foresight":
     productPackagePath=productPath + "/changesets"
 else:
     productPackagePath=productPath + "/RPMS"
Common subdirectories: anaconda-10.0.orig/scripts and anaconda-10.0/scripts
Common subdirectories: anaconda-10.0.orig/stubs and anaconda-10.0/stubs
diff -ui anaconda-10.0.orig/text.py anaconda-10.0/text.py
--- anaconda-10.0.orig/text.py	2005-01-31 15:14:46.000000000 -0500
+++ anaconda-10.0/text.py	2005-04-20 11:33:40.000000000 -0400
@@ -344,7 +344,7 @@
 	return WaitWindow(self.screen, title, text)
 
     def drawFrame(self):
-        self.welcomeText = _("%s (C) 2004-2005 Specifix, Inc.") % (productName,)
+        self.welcomeText = _("%s (C) 2005 Foresight Linux Project") % (productName,)
         self.screen.drawRootText (0, 0, self.welcomeText)
 	self.screen.drawRootText (len(_(self.welcomeText)), 0,
                                   (self.screen.width -
Common subdirectories: anaconda-10.0.orig/textw and anaconda-10.0/textw
Common subdirectories: anaconda-10.0.orig/utils and anaconda-10.0/utils
