--- tarextract.py.orig	2010-09-15 18:44:55.641274412 +0100
+++ tarextract.py	2010-09-15 18:46:28.422274412 +0100
@@ -7,6 +7,9 @@
 from subprocess import Popen, PIPE
 
 import logging
+import threading
+
+lock = threading.Lock()
 
 log = logging.getLogger('anaconda')
 
@@ -25,9 +28,11 @@
         self._proc = self._getPipe()
         self._bufferSize = 16*1024
 
+        global lock
+
     def _getPipe(self):
         proc = Popen(['tar', '--exclude', '/var/swap', 'vvvzxf', '-'], stdin=PIPE, stdout=self._stdoutlog,
-                     stderr=self._stderrlog, cwd=self.root)
+                     stderr=self._stderrlog, cwd=self.root, close_fds=True)
         return proc
 
     def extractFile(self, file):
@@ -52,9 +57,14 @@
             self.progress.startWriteBuffer()
 
             data = chunkfh.read(size)
+
+            lock.acquire()
+
             self._extract(data)
             sha1.update(data)
 
+            lock.release()
+
             self.progress.stopWriteBuffer()
 
         self.progress.stopChunk()
