--- tarextract.py.orig	2010-09-14 18:00:16.726281199 +0100
+++ tarextract.py	2009-09-14 18:01:04.670281199 +0100
@@ -27,6 +27,8 @@
 
     def _getPipe(self):
         proc = Popen(['tar', '--exclude', '/var/swap', 'vvvzxf', '-'], stdin=PIPE, stdout=self._stdoutlog,
+                     preexec_fn=lambda:
+                         signal.signal(signal.SIGPIPE, signal.SIG_DFL),
                      stderr=self._stderrlog, cwd=self.root)
         return proc
 
