--- tarextract.py.orig	2010-09-15 02:35:07.772281206 +0100
+++ tarextract.py	2010-09-15 02:35:24.458281206 +0100
@@ -26,7 +26,7 @@
         self._bufferSize = 16*1024
 
     def _getPipe(self):
-        proc = Popen(['tar', '--exclude', '/var/swap', 'vvvzxf', '-'], stdin=PIPE, stdout=self._stdoutlog,
+        proc = Popen(['tar', '--exclude', '/var/swap', 'vvvizxf', '-'], stdin=PIPE, stdout=self._stdoutlog,
                      stderr=self._stderrlog, cwd=self.root)
         return proc
 
