--- tarextract.py.orig	2010-09-15 13:50:14.310281206 +0100
+++ tarextract.py	2010-09-15 13:50:41.512281206 +0100
@@ -27,7 +27,7 @@
 
     def _getPipe(self):
         proc = Popen(['tar', '--exclude', '/var/swap', 'vvvzxf', '-'], stdin=PIPE, stdout=self._stdoutlog,
-                     stderr=self._stderrlog, cwd=self.root)
+                     stderr=self._stderrlog, cwd=self.root, close_fds=True)
         return proc
 
     def extractFile(self, file):
