diff -ur anaconda-10.0.orig/installclass.py anaconda-10.0/installclass.py
--- anaconda-10.0.orig/installclass.py	2004-09-01 15:41:25.000000000 -0400
+++ anaconda-10.0/installclass.py	2005-04-11 08:57:09.000000000 -0400
@@ -31,7 +31,7 @@
     # default to not being hidden
     hidden = 0
     pixmap = None
-    showMinimal = 1
+    showMinimal = 0
     showLoginChoice = 0
     description = None
     name = "base"
