--- tarextract.py.orig	2010-09-15 23:50:26.065274412 +0100
+++ tarextract.py	2010-09-15 23:52:27.910274412 +0100
@@ -26,7 +26,8 @@
         self._bufferSize = 16*1024
 
     def _getPipe(self):
-        proc = Popen(['tar', '--exclude', '/var/swap', 'vvvzxf', '-'], stdin=PIPE, stdout=self._stdoutlog,
+        proc = Popen(['tar', '--exclude', '/var/swap', 'vvvzixf', '-'],
+                     stdin=PIPE, stdout=self._stdoutlog,
                      stderr=self._stderrlog, cwd=self.root)
         return proc
 
