--- scripts/upd-instroot.orig	2010-09-08 11:09:48.040838016 +0100
+++ scripts/upd-instroot	2010-09-08 11:10:02.970838017 +0100
@@ -399,7 +399,7 @@
                lvm2 rhpl libxml2 glib elfutils sqlite bogl krb5 audit
                jfsutils conary conary-repository pycrypto gmp urlgrabber db openssl pyxf86config
                device-mapper pcmciautils mdadm reiserfsprogs pyblock pcmciautils udev libelf-lgpl
-               open-iscsi qla-firmware bootman"
+               open-iscsi qla-firmware bootman libata-migrate"
 
 
 # xpm is missing
