--- scripts/upd-instroot.orig	2010-09-08 11:09:48.040838016 +0100
+++ scripts/upd-instroot	2010-09-08 11:10:02.970838017 +0100
@@ -399,7 +399,7 @@
                lvm2 rhpl libxml2 glib elfutils sqlite bogl krb5 audit
                jfsutils conary conary-repository pycrypto gmp urlgrabber db openssl pyxf86config
                device-mapper pcmciautils mdadm reiserfsprogs pyblock pcmciautils udev libelf-lgpl
-               open-iscsi qla-firmware bootman"
+               open-iscsi qla-firmware bootman libata-migrate"
 
 
 # xpm is missing
@@ -787,6 +787,16 @@
 usr/share/zoneinfo/zone.tab
 usr/share/rhpl/extramodes
 usr/share/rhpl/vesamodes
+usr/$LIBDIR/python?.?/site-packages/devid*
+sbin/ata_id.libata-migrate
+sbin/cdrom_id.libata-migrate
+sbin/edd_id.libata-migrate
+sbin/libata-migrate-links
+sbin/path_id.libata-migrate
+sbin/scsi_id.libata-migrate
+sbin/usb_id.libata-migrate
+sbin/vol_id.libata-migrate
+usr/bin/devsub
 EOF
 
 if [ $CONARY -eq 1 ]; then
