diff -Naur anaconda-10.2.1.5.rpath/conaryinstall.py anaconda-10.2.1.5/conaryinstall.py
--- anaconda-10.2.1.5.rpath/conaryinstall.py	2005-08-28 17:30:57.000000000 -0400
+++ anaconda-10.2.1.5/conaryinstall.py	2005-09-01 11:20:42.000000000 -0400
@@ -237,6 +237,10 @@
 
     conaryInstallPost(id, intf, instPath)
 
+    for (n, tag) in id.grpset.kernelVersionList():
+        log('making initrd for %s', n)
+        packages.recreateInitrd(n, instPath)
+
     method.filesDone()
 
 def doConaryInstall(method, id, intf, instPath):
diff -Naur anaconda-10.2.1.5.rpath/packages.py anaconda-10.2.1.5/packages.py
--- anaconda-10.2.1.5.rpath/packages.py	2005-08-20 23:55:41.000000000 -0400
+++ anaconda-10.2.1.5/packages.py	2005-09-01 16:01:06.000000000 -0400
@@ -38,6 +38,7 @@
 from rhpl.log import log
 from rhpl.translate import _
 import rhpl.arch
+import bootloader
 
 def queryUpgradeContinue(intf, dir):
     if dir == DISPATCH_FORWARD:
@@ -1516,6 +1517,22 @@
                            stdout = None, stderr = None,
                            searchPath = 1, root = instRoot)
 
+    # get the path to the initrd
+    initrd = '%s/%s' %(instRoot, bootloader.makeInitrd("-%s" % (kernelTag,), None))
+    # check to see if it exists, and that we can write to it
+    if not os.access(initrd, os.W_OK):
+        # no initrd file, nothing left to do
+        return
+
+    # open the initrd file, append to it
+    log("Apending bootsplash to initrd for %s" % (kernelTag,))
+    f = open(initrd, 'a')
+    iutil.execWithRedirect('/sbin/splash',
+                           ('/sbin/splash', '-s', '-f',
+            '/usr/share/bootsplash/themes/FDL/config/bootsplash-1024x768.cfg'),
+                           stdout = f.fileno(), stderr = None,
+                           searchPath = 1, root = instRoot)
+
 # XXX Deprecated.  Is this ever called anymore?
 def depmodModules(comps, instPath):
     kernelVersions = comps.kernelVersionList()
diff -Naur anaconda-10.2.1.5.rpath/product.py anaconda-10.2.1.5/product.py
--- anaconda-10.2.1.5.rpath/product.py	2005-08-20 23:39:24.000000000 -0400
+++ anaconda-10.2.1.5/product.py	2005-09-01 11:20:42.000000000 -0400
@@ -22,7 +22,7 @@
 
 productName = "anaconda"
 productVersion = "bluesky"
-productPath = "rpath"
+productPath = "Foresight"
 bugUrl = "your distribution provided bug reporting tool."    
 
 if path is not None:
diff -Naur anaconda-10.2.1.5.rpath/scripts/mk-images anaconda-10.2.1.5/scripts/mk-images
--- anaconda-10.2.1.5.rpath/scripts/mk-images	2005-08-18 15:51:41.000000000 -0400
+++ anaconda-10.2.1.5/scripts/mk-images	2005-09-01 11:20:42.000000000 -0400
@@ -628,7 +628,8 @@
 
     if [ $CONARY -eq 1 ]; then
         # XXX hard code for now
-	echo "installLabelPath conary.rpath.com@rpl:devel contrib.rpath.com@rpl:devel" > $MBD_DIR/etc/conaryrc
+        echo "autoResolve     True" > $MBD_DIR/etc/conaryrc
+        echo "installLabelPath conary.foresightlinux.com@fl:desktop conary.rpath.com@rpl:devel contrib.rpath.com@rpl:devel" >> $MBD_DIR/etc/conaryrc
     fi
 
     if [ "$BUILDARCH" = "ppc" -o "$BUILDARCH" = "ppc64" ]; then
diff -Naur anaconda-10.2.1.5.rpath/scripts/upd-instroot anaconda-10.2.1.5/scripts/upd-instroot
--- anaconda-10.2.1.5.rpath/scripts/upd-instroot	2005-08-29 14:05:08.000000000 -0400
+++ anaconda-10.2.1.5/scripts/upd-instroot	2005-09-01 16:04:46.000000000 -0400
@@ -197,7 +197,7 @@
   	        gcc parted pyparted sed busybox:anaconda booty hdparm
   	        lvm2 rhpl libxml2 glib elfutils sqlite bogl krb5
   	        jfsutils conary urlgrabber db openssl pyxf86config
-  	        device-mapper pcmcia-cs mdadm"
+  	        device-mapper pcmcia-cs mdadm distro-release" 
 
 # xpm is missing
 
@@ -237,7 +237,8 @@
            xorg-x11-libs-data xorg-x11-xfs prelink redhat-logos fedora-logos
            ttfonts-ta ttfonts-bn ttfonts-gu ttfonts-hi ttfonts-pa bogl-devel
            fonts-japanese fonts-chinese fonts-korean fonts-indic 
-           fonts-gujarati fonts-hindi fonts-punjabi fonts-tamil synaptics"
+           fonts-gujarati fonts-hindi fonts-punjabi fonts-tamil synaptics
+           distro-release"
 
 CONARYPACKAGESGR="anaconda bash ncurses glibc readline libpng gtk
                  esound audiofile libgnome e2fsprogs coreutils
@@ -250,7 +251,7 @@
                  fonts-arabic fonts-bengali libjpeg tcp_wrappers
                  system-config-date xfsprogs xfsdump dmapi attr acl
                  xorg-x11-xfs xorg-x11 xorg-x11-fonts reiserfsprogs
-                 libglade gtk-engines bogl"
+                 libglade gtk-engines bogl distro-release"
 
 #
 # stuff ONLY included for rescue mode
