diff -Naur anaconda-10.2.1.5.rpath/conaryinstall.py anaconda-10.2.1.5/conaryinstall.py
--- anaconda-10.2.1.5.rpath/conaryinstall.py	2005-08-25 17:16:45.000000000 -0400
+++ anaconda-10.2.1.5/conaryinstall.py	2005-08-29 11:07:53.000000000 -0400
@@ -237,6 +237,10 @@
 
     conaryInstallPost(id, intf, instPath)
 
+    for (n, tag) in id.grpset.kernelVersionList():
+        log('making initrd for %s', n)
+        packages.recreateInitrd(n, instPath)
+
     method.filesDone()
 
 def doConaryInstall(method, id, intf, instPath):
diff -Naur anaconda-10.2.1.5.rpath/packages.py anaconda-10.2.1.5/packages.py
--- anaconda-10.2.1.5.rpath/packages.py	2005-08-20 23:55:41.000000000 -0400
+++ anaconda-10.2.1.5/packages.py	2005-08-29 11:06:32.000000000 -0400
@@ -1516,6 +1516,15 @@
                            stdout = None, stderr = None,
                            searchPath = 1, root = instRoot)
 
+    # open the initrd file, append to it
+    log("Apending bootsplash to initrd for %s" % (kernelTag,))
+    f = open(initrd, 'a')
+    iutil.execWithRedirect('/sbin/splash',
+                           ('/sbin/splash', '-s', '-f',
+            '/usr/share/bootsplash/themes/FDL/config/bootsplash-1024x768.cfg'),
+                           stdout = f.fileno(), stderr = None,
+                           searchPath = 1, root = instRoot)
+
 # XXX Deprecated.  Is this ever called anymore?
 def depmodModules(comps, instPath):
     kernelVersions = comps.kernelVersionList()
diff -Naur anaconda-10.2.1.5.rpath/product.py anaconda-10.2.1.5/product.py
--- anaconda-10.2.1.5.rpath/product.py	2005-08-20 23:39:24.000000000 -0400
+++ anaconda-10.2.1.5/product.py	2005-08-29 11:10:18.000000000 -0400
@@ -22,7 +22,7 @@
 
 productName = "anaconda"
 productVersion = "bluesky"
-productPath = "rpath"
+productPath = "Foresight"
 bugUrl = "your distribution provided bug reporting tool."    
 
 if path is not None:
diff -Naur anaconda-10.2.1.5.rpath/scripts/mk-images anaconda-10.2.1.5/scripts/mk-images
--- anaconda-10.2.1.5.rpath/scripts/mk-images	2005-08-18 15:51:41.000000000 -0400
+++ anaconda-10.2.1.5/scripts/mk-images	2005-08-29 11:10:18.000000000 -0400
@@ -628,7 +628,8 @@
 
     if [ $CONARY -eq 1 ]; then
         # XXX hard code for now
-	echo "installLabelPath conary.rpath.com@rpl:devel contrib.rpath.com@rpl:devel" > $MBD_DIR/etc/conaryrc
+        echo "autoResolve     True" > $MBD_DIR/etc/conaryrc
+        echo "installLabelPath conary.foresightlinux.com@fl:desktop conary.rpath.com@rpl:devel contrib.rpath.com@rpl:devel" >> $MBD_DIR/etc/conaryrc
     fi
 
     if [ "$BUILDARCH" = "ppc" -o "$BUILDARCH" = "ppc64" ]; then
