diff -ur quick-lounge-applet-2.12.1.orig/src/quick-lounge.c quick-lounge-applet-2.12.1/src/quick-lounge.c
--- quick-lounge-applet-2.12.1.orig/src/quick-lounge.c	2005-11-10 08:07:53.000000000 -0500
+++ quick-lounge-applet-2.12.1/src/quick-lounge.c	2005-11-30 15:37:27.000000000 -0500
@@ -1415,8 +1415,16 @@
 {
 	static char * default_items [] = { 
 		/* FIXME : put default items here. */
+		"search.desktop",
 		"nautilus.desktop",
 		"gnome-terminal.desktop",
+		"evolution-2.4.desktop",
+		"net-firefox.desktop",
+		"xchat-gnome.desktop",
+		"gaim.desktop",
+		"f-spot.desktop",
+		"banshee.desktop",
+		"blam.desktop",
 		"yelp.desktop",
 		NULL
 	};
Only in quick-lounge-applet-2.12.1/src: quick-lounge.c~
diff -ur quick-lounge-applet-2.12.1.orig/src/quick-lounge.schemas quick-lounge-applet-2.12.1/src/quick-lounge.schemas
--- quick-lounge-applet-2.12.1.orig/src/quick-lounge.schemas	2005-11-10 08:07:53.000000000 -0500
+++ quick-lounge-applet-2.12.1/src/quick-lounge.schemas	2005-11-30 15:37:44.000000000 -0500
@@ -5,7 +5,7 @@
         <key>/schemas/apps/quick-lounge-applet/prefs/min_size</key>
         <owner>quick-lounge-applet</owner>
         <type>int</type>
-        <default>4</default>
+        <default>10</default>
         <locale name="C">
           <short>Minimum number of columns to display</short>
           <long></long>
diff -ur quick-lounge-applet-2.12.1.orig/src/quick-lounge.schemas.in quick-lounge-applet-2.12.1/src/quick-lounge.schemas.in
--- quick-lounge-applet-2.12.1.orig/src/quick-lounge.schemas.in	2005-11-10 08:07:53.000000000 -0500
+++ quick-lounge-applet-2.12.1/src/quick-lounge.schemas.in	2005-11-30 15:37:55.000000000 -0500
@@ -5,7 +5,7 @@
         <key>/schemas/apps/quick-lounge-applet/prefs/min_size</key>
         <owner>quick-lounge-applet</owner>
         <type>int</type>
-        <default>4</default>
+        <default>10</default>
         <locale name="C">
           <short>Minimum number of columns to display</short>
           <long></long>
