From c6d35410db6577e1a49166effe114cc64e1b2d3c Mon Sep 17 00:00:00 2001
From: Lionel Le Folgoc <mrpouit@gmail.com>
Date: Wed, 09 Mar 2011 13:12:55 +0000
Subject: Retrieve the translated desktop file name

---
diff --git a/src/xfdesktop-file-utils.c b/src/xfdesktop-file-utils.c
index 33ffe64..366a3d3 100644
--- a/src/xfdesktop-file-utils.c
+++ b/src/xfdesktop-file-utils.c
@@ -212,10 +212,11 @@ xfdesktop_file_utils_get_display_name(GFile *file,
         key_file = xfdesktop_file_utils_query_key_file(file, NULL, NULL);
         if(key_file) {
             /* try to parse the display name */
-            display_name = g_key_file_get_string(key_file, 
-                                                 G_KEY_FILE_DESKTOP_GROUP,
-                                                 G_KEY_FILE_DESKTOP_KEY_NAME,
-                                                 NULL);
+            display_name = g_key_file_get_locale_string(key_file, 
+                                                        G_KEY_FILE_DESKTOP_GROUP,
+                                                        G_KEY_FILE_DESKTOP_KEY_NAME,
+                                                        NULL,
+                                                        NULL);
             
             /* free the key file */
             g_key_file_free (key_file);
--
cgit v0.8.3.4
