diff --git a/Makefile b/Makefile
index 55a28cb..59dff5a 100644
--- a/Makefile
+++ b/Makefile
@@ -33,8 +33,9 @@ install-mkdirs:
 
 install: install-mkdirs
 	install -m 755 foresight-user-guide $(DESTDIR)$(bindir)
-	install -m 755 guide/*.xml $(DESTDIR)$(datadir)/foresight-user-guide/help/C/
-	install -m 755 guide/figures/*.png $(DESTDIR)$(datadir)/foresight-user-guide/help/C/figures/
+	install -m 755 help/C/*.xml $(DESTDIR)$(datadir)/foresight-user-guide/help/C/
+	install -m 755 help/C/*.page $(DESTDIR)$(datadir)/foresight-user-guide/help/C/
+	install -m 755 help/C/figures/*.png $(DESTDIR)$(datadir)/foresight-user-guide/help/C/figures/
 	install -m 755 foresight-user-guide.desktop $(DESTDIR)$(datadir)/applications/
 	install -m 644 foresight-icon.png $(DESTDIR)$(datadir)/pixmaps/
 
