--- ypbind.init.jau	2008-01-04 11:18:42.000000000 -0500
+++ ypbind.init	2008-01-04 11:40:40.000000000 -0500
@@ -27,11 +27,18 @@
 start() {
 	DOMAINNAME=`domainname`
 	if [ "$DOMAINNAME" = "(none)" -o "$DOMAINNAME" = "" ]; then
-	   if [ -n "$NISDOMAIN" ]; then
+            if [ -n "$NISDOMAIN" ]; then
 		action $"Setting NIS domain name $NISDOMAIN: " domainname $NISDOMAIN
-	   else
-	        exit 1
-	   fi
+            else
+                NISDOMAIN=`grep "domain" /etc/yp.conf | grep -v ^# | awk '{print $2}'`
+                if [ -n "$NISDOMAIN" ]; then
+                    action $"NIS domain is $NISDOMAIN: " domainname $NISDOMAIN
+                else
+                    action $"NIS domain not found" /bin/false
+                    logger -t ypbind $"NIS domain not found"
+                    return 1
+                fi
+            fi
 	fi
 	echo -n $"Binding to the NIS domain: "
 	daemon ypbind $OTHER_YPBIND_OPTS
