--- dvb_channels.c	2008-01-02 17:17:19.000000000 +0100
+++ dvb_channels-copy.c	2009-10-24 23:30:59.000000000 +0200
@@ -19,6 +19,10 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  * *****************************************************************/
 
+// Dirty hack to avoid conflicting #include:
+#define _INTTYPES_H 1
+#define _SYS_TYPES_H 1
+
 #include <linux/dvb/frontend.h>
 
 #include <avdec_private.h>
