--- vixie-cron/vixie-cron-4.1/crontab.c	2008-10-29 12:40:52.000000000 -0500
+++ vixie-cron-4.1.fix/crontab.c	2008-10-29 12:44:02.000000000 -0500
@@ -31,6 +31,7 @@ static char rcsid[] = "$Id: crontab.c,v 
 #define	MAIN_PROGRAM
 
 #include "cron.h"
+#include <sys/resource.h>
 #ifdef WITH_SELINUX
 #include <selinux/selinux.h>
 #include <selinux/flask.h>
@@ -109,6 +110,10 @@ usage(const char *msg) {
 int
 main(int argc, char *argv[]) {
 	int exitstatus;
+	struct rlimit rlim;
+
+	rlim.rlim_cur = rlim.rlim_max = 8192*1024;
+	setrlimit(RLIMIT_STACK, &rlim);
 
 	Pid = getpid();
 	ProgramName = argv[0];
