diff -up amarok-2.2.2/src/CMakeLists.txt.no_var_tracking amarok-2.2.2/src/CMakeLists.txt
--- amarok-2.2.2/src/CMakeLists.txt.no_var_tracking	2010-01-06 16:19:46.000000000 -0600
+++ amarok-2.2.2/src/CMakeLists.txt	2010-01-06 17:43:11.257848590 -0600
@@ -714,6 +714,8 @@ set( amaroklib_DEPENDS "amarokpud" )
 
 kde4_add_kcfg_files(amaroklib_LIB_SRCS amarokconfig.kcfgc)
 
+set_source_files_properties(amarokconfig.cpp PROPERTIES COMPILE_FLAGS -fno-var-tracking-assignments )
+
 kde4_add_ui_files(amaroklib_LIB_SRCS
     aboutdialog/OcsPersonItem.ui
     dialogs/EditCoverSearchDialog.ui
