diff --git a/usr/lib64/python2.6/site-packages/rbuild/facade/apifinder.py b/usr/lib64/python2.6/site-packages/rbuild/facade/apifinder.py
--- a/rbuild/facade/apifinder.py
+++ b/rbuild/facade/apifinder.py
@@ -58,10 +58,13 @@
         GET a URL, using the OpenerDirector if supplied.
         '''
         url = self._secureUrl(url)
+        # XXX get around rBO ubber CRAZYNESS XXX
+        req = urllib2.Request(url)
+        req.add_header('User-Agent', 'Mozilla/5.0 (Windows; U; Windows NT 5.1; en-GB; rv:1.9.0.3) Gecko/2008092417 Firefox/3.0.3')
         if self.opener:
-            return self.opener.open(url).read()
+            return self.opener.open(req).read()
         else:
-            return urllib2.urlopen(url).read()
+            return urllib2.urlopen(req).read()
 
     def _secureUrl(self, url):
         '''
