--- tk8.4.15/unix/tcl.m4.jau	2007-05-27 17:31:34.000000000 -0400
+++ tk8.4.15/unix/tcl.m4	2007-05-27 17:44:26.000000000 -0400
@@ -1345,7 +1345,7 @@
 	    #CFLAGS_OPTIMIZE="${CFLAGS_OPTIMIZE} -D__NO_STRING_INLINES -D__NO_MATH_INLINES"
 
 	    if test "$have_dl" = yes; then
-		SHLIB_LD='${CC} -shared ${CFLAGS} ${LDFLAGS}'
+		SHLIB_LD='${CC} -shared ${CFLAGS} ${LDFLAGS} -Wl,-soname,${@}'
 		DL_OBJS="tclLoadDl.o"
 		DL_LIBS="-ldl"
 		LDFLAGS="$LDFLAGS -Wl,--export-dynamic"
