Index: kdebase-workspace-4.3.90/plasma/desktop/applets/tasks/abstracttaskitem.cpp
===================================================================
--- kdebase-workspace-4.3.90.orig/plasma/desktop/applets/tasks/abstracttaskitem.cpp	2010-01-06 12:00:18.000000000 -0500
+++ kdebase-workspace-4.3.90/plasma/desktop/applets/tasks/abstracttaskitem.cpp	2010-01-06 20:08:40.850005656 -0500
@@ -64,6 +64,9 @@
 #include "tasks.h"
 #include "taskgroupitem.h"
 
+#include <X11/Xlib.h>
+#include <X11/Xatom.h>
+
 static const int HOVER_EFFECT_TIMEOUT = 900;
 
 AbstractTaskItem::AbstractTaskItem(QGraphicsWidget *parent, Tasks *applet)
