--- kdebase-workspace/kwin/compositingprefs.cpp	2009-12-04 23:09:28.000000000 +0000
+++ kdebase-workspace/kwin/compositingprefs.cpp	2010-03-11 00:20:42.000000000 +0000
@@ -329,8 +329,10 @@
         kDebug( 1212 ) << "intel driver, disabling vsync, enabling direct";
         mEnableVSync = false;
         mEnableDirectRendering = true;
+	/*
         if( mVersion >= Version( "20061017" ))
             { 
+	*/
             if( mGLRenderer.contains( "Intel(R) 9" ))
                 { // Enable compositing by default on 900-series cards
                 kDebug( 1212 ) << "intel >= 20061017 and 900-series card, enabling compositing";
@@ -342,7 +344,9 @@
                 mRecommendCompositing = true;
                 }
             }
+        /*
         }
+        */
     else if( mDriver == "nvidia" )
         {
         mStrictBinding = false;
