Index: kdebase-workspace-4.5.85/startkde.cmake
===================================================================
--- kdebase-workspace-4.5.85.orig/startkde.cmake	2010-12-13 09:43:30.542222358 -0500
+++ kdebase-workspace-4.5.85/startkde.cmake	2011-01-06 11:53:34.828287264 -0500
@@ -197,6 +197,56 @@
   unset KLOCALE_LANGUAGES
 fi
 
+# configuration of the gtk2-engines-oxygen if not already set
+
+# First we need to remove previous gtk2-engines-qtcurve config if any
+if [ -e $kdehome/env/gtk2-engines-qtcurve.rc.sh ] && [ ! -e /var/lib/dpkg/info/gtk2-engines-qtcurve.list ] ; then
+  rm -f $kdehome/env/gtk2-engines-qtcurve.rc.sh
+  # We also have to remove .gtkrc-2.0-kde4 since it'll have gtk2-engines-qtcurve config in it
+  if [ -e $HOME/.gtkrc-2.0-kde4 ] ; then
+    rm -f $HOME/.gtkrc-2.0-kde4
+  fi
+fi
+
+if [ -e /var/lib/dpkg/info/gtk2-engines-oxygen.list ] && [ -e /var/lib/dpkg/info/kubuntu-default-settings.list ] ; then
+  # The theme and k-d-s are installed, configuring...
+  if [ ! -e $kdehome/env/gtk2-default-theme.rc.sh ] && [ -e /usr/share/kubuntu-default-settings/gtk2-default-theme.rc.sh ] ; then
+    mkdir -p $kdehome/env
+    cp -f /usr/share/kubuntu-default-settings/gtk2-default-theme.rc.sh $kdehome/env
+    chmod 755 $kdehome/env/gtk2-default-theme.rc.sh
+  fi
+
+  if [ ! -e $HOME/.gtkrc-2.0-kde4 ] && [ -e /usr/share/kubuntu-default-settings/dot-gtkrc-2.0-kde4 ] ; then
+    cp -f /usr/share/kubuntu-default-settings/dot-gtkrc-2.0-kde4 $HOME/.gtkrc-2.0-kde4
+  fi
+else
+  # Either the theme or k-d-s are missing, not forcing the configuration then... 
+  if [ -e $kdehome/env/gtk2-default-theme.rc.sh ] ; then
+    rm -f $kdehome/env/gtk2-default-theme.rc.sh
+  fi
+fi
+
+# In Kubuntu start plasma-netbook if the screen is small or if "netbook" parameter were passed to startkde.
+# Starting plasma-mobile if "mobile" parameter were passed to startkde
+if [ -e /usr/share/kubuntu-netbook-default-settings/share/autostart/plasma-netbook.desktop ] ; then
+  HEIGHT=`xdpyinfo | grep dimensions | awk '{print $2}' | sed s,.*x,,`
+  CDROM=`udisks --dump | grep optical_cd`
+  if `laptop-detect` && [ $HEIGHT -lt 700 ] && [ $HEIGHT -gt 0 ] && [ -z $CDROM ] || [ $1 = "netbook" ]; then
+    if [ ! -e ~/.config/autostart/plasma-netbook.desktop ] || `grep -q Hidden=false ~/.config/autostart/plasma-netbook.desktop`; then
+      echo Small screen resolution detected, using plasma-netbook;
+      export KDEDIRS=/usr/share/kubuntu-netbook-default-settings/:/usr/share/kubuntu-default-settings/kde4-profile/default/
+    fi
+  fi
+fi
+
+# Starting plasma-mobile if "mobile" parameter were passed to startkde.
+if [ -e /usr/share/kubuntu-mobile-default-settings/share/autostart/plasma-mobile.desktop ]; then
+  if [ "$1" = "mobile" ] ; then
+    # FIXME: Find a way to decrease the startup time when adding kubuntu-default-settings as fallback.
+    export KDEDIRS=/usr/share/kubuntu-mobile-default-settings/:/usr/share/kubuntu-default-settings/kde4-profile/default/
+  fi
+fi
+
 # Source scripts found in <localprefix>/env/*.sh and <prefixes>/env/*.sh
 # (where <localprefix> is $KDEHOME or ~/.kde, and <prefixes> is where KDE is installed)
 #
