Index: kdebase-workspace-4.4.85/CMakeLists.txt
===================================================================
--- kdebase-workspace-4.4.85.orig/CMakeLists.txt	2010-06-07 01:10:44.311466674 +0200
+++ kdebase-workspace-4.4.85/CMakeLists.txt	2010-06-07 01:10:54.851466662 +0200
@@ -162,7 +162,7 @@
 if(NOT WIN32)
   macro_optional_add_subdirectory( cursors )
 endif(NOT WIN32)
-macro_optional_add_subdirectory( doc )
+#macro_optional_add_subdirectory( doc )
 macro_optional_add_subdirectory( wallpapers )
 
 if(X11_Xau_FOUND AND X11_Xdmcp_FOUND)
