Index: kdebase-workspace-4.3.90/plasma/desktop/shell/desktopcorona.cpp
===================================================================
--- kdebase-workspace-4.3.90.orig/plasma/desktop/shell/desktopcorona.cpp	2009-10-06 17:37:23.000000000 -0400
+++ kdebase-workspace-4.3.90/plasma/desktop/shell/desktopcorona.cpp	2010-01-14 09:28:24.356392453 -0500
@@ -335,6 +335,8 @@
                     config.writeEntry("url", "desktop:/");
                 }
             }
+            loadDefaultApplet("twitter", c);
+            //loadDefaultApplet("opendesktop", c);
         }
 
         emit containmentAdded(c);
@@ -366,7 +368,10 @@
     }
 
     loadDefaultApplet("pager", panel);
+    loadDefaultApplet("quickaccess", panel);
     loadDefaultApplet("tasks", panel);
+    loadDefaultApplet("showdesktop", panel);
+    loadDefaultApplet("indicatordisplay", panel);
     Plasma::Applet *sysTray = loadDefaultApplet("systemtray", panel);
 
     if (sysTray) {
@@ -376,6 +381,10 @@
         }
     }
 
+//     if (QFile::exists("/usr/share/kde4/services/plasma-applet-networkmanagement.desktop")) {
+//       loadDefaultApplet("networkmanagement", panel);
+//     }
+
     loadDefaultApplet("digital-clock", panel);
     emit containmentAdded(panel);
 
Index: kdebase-workspace-4.3.90/plasma/generic/applets/digital-clock/clock.cpp
===================================================================
--- kdebase-workspace-4.3.90.orig/plasma/generic/applets/digital-clock/clock.cpp	2009-12-31 08:28:32.000000000 -0500
+++ kdebase-workspace-4.3.90/plasma/generic/applets/digital-clock/clock.cpp	2010-01-14 09:25:19.748392200 -0500
@@ -48,7 +48,7 @@
       m_plainClockFont(KGlobalSettings::generalFont()),
       m_useCustomColor(false),
       m_plainClockColor(),
-      m_showDate(false),
+      m_showDate(true),
       m_showYear(false),
       m_showDay(false),
       m_showSeconds(false),
@@ -78,7 +78,7 @@
 
     kDebug() << "showTimezone:" << m_showTimezone;
 
-    m_showDate = cg.readEntry("showDate", false);
+    m_showDate = cg.readEntry("showDate", true);
     m_showYear = cg.readEntry("showYear", false);
     m_showDay = cg.readEntry("showDay", true);
 
