--- kdebase-workspace/plasma/netbook/shell/plasma-netbook.desktop	2010-01-19 23:19:10.000000000 +0000
+++ kdebase-workspace/plasma/netbook/shell/plasma-netbook.desktop	2010-01-27 13:27:25.000000000 +0000
@@ -89,4 +89,4 @@
 X-KDE-StartupNotify=false
 OnlyShowIn=KDE;
 X-KDE-autostart-phase=0
-
+Hidden=true
--- kdebase-workspace/plasma/netbook/shell/CMakeLists.txt	2009-10-26 09:14:50.000000000 +0000
+++ kdebase-workspace/plasma/netbook/shell/CMakeLists.txt	2010-02-01 23:36:44.000000000 +0000
@@ -22,5 +22,5 @@
 install(TARGETS kdeinit_plasma-netbook DESTINATION ${LIB_INSTALL_DIR})
 install(TARGETS plasma-netbook ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES plasma-default-layoutrc DESTINATION ${DATA_INSTALL_DIR}/plasma-netbook/)
-#install(FILES plasma-netbook.desktop DESTINATION ${AUTOSTART_INSTALL_DIR})
+install(FILES plasma-netbook.desktop DESTINATION ${AUTOSTART_INSTALL_DIR})
 
