diff -r c76148f9a13b configure.in
--- a/configure.in	Tue Sep 23 21:19:43 2008 +0000
+++ b/configure.in	Tue Sep 23 21:20:17 2008 +0000
@@ -381,7 +381,7 @@
   fail=0
   old_libs="$LIBS"
   AC_CHECK_HEADER([gnutls/gnutls.h], [], [fail=1])
-  AC_CHECK_LIB([gnutls], [gnutls_handshake],[], [fail=1], [-lgcrypt])
+  AC_CHECK_LIB([gnutls], [gnutls_handshake],[], [fail=1], [-lgcrypt -lz])
 
   test $fail = 1 &&
     AC_MSG_ERROR([You must install the GnuTLS library in order to compile and run libvirt])
