--- gnu-efi/gnu-efi-3.0/inc/x86_64/efibind.h	2011-08-24 15:02:00.000000000 +0200
+++ gnu-efi/gnu-efi-3.0/inc/x86_64/efibind.h	2011-09-26 13:51:20.000000000 +0200
@@ -1,3 +1,5 @@
+#ifndef EFIBIND_H
+#define EFIBIND_H
 /*++
 
 Copyright (c) 1998  Intel Corporation
@@ -279,4 +281,4 @@
 #ifdef _MSC_EXTENSIONS
 #pragma warning ( disable : 4731 )  // Suppress warnings about modification of EBP
 #endif
-
+#endif
