--- gnu-efi-3.0/Make.defaults	2011-09-26 14:55:48.000000000 +0200
+++ gnu-efi-3.0/Make.defaults	2011-09-26 14:56:10.000000000 +0200
@@ -39,7 +39,7 @@
 # lib and include under the root
 #
 INSTALLROOT=/usr/local
-LIBDIR=lib
+LIBDIR=lib/gnuefi
 
 TOPDIR    := $(shell if [ "$$PWD" != "" ]; then echo $$PWD; else pwd; fi)
 
@@ -73,7 +73,7 @@
 
 ifeq ($(ARCH), x86_64)
   CFLAGS += -mno-red-zone
-  LIBDIR = lib
+  LIBDIR = lib64/gnuefi
   ifeq ($(HOSTARCH), ia32)
     ARCH3264 = -m64
   endif
