diff -r -U 3 udev-063/udev.nodes udev-063-patch/udev.nodes
--- udev-063/udev.nodes	2005-02-18 14:07:41.000000000 +0100
+++ udev-063-patch/udev.nodes	2005-09-04 20:13:52.000000000 +0200
@@ -26,3 +26,6 @@
 console 
 null 
 zero
+ptmx
+rtc
+inotify
diff -r -U 3 udev-063/udev.rules.persistent udev-063-patch/udev.rules.persistent
--- udev-063/udev.rules.persistent	2005-05-12 12:19:24.000000000 +0200
+++ udev-063-patch/udev.rules.persistent	2005-09-04 20:12:56.000000000 +0200
@@ -1,18 +1,25 @@
-BUS=="scsi", PROGRAM="/sbin/udev.get_persistent_device_name.sh", SYMLINK="%c{1+}"
-BUS=="usb", PROGRAM="/sbin/udev.get_persistent_device_name.sh", SYMLINK="%c{1+}"
-BUS=="ide", PROGRAM="/sbin/udev.get_persistent_device_name.sh", SYMLINK="%c{1+}"
+BUS="scsi", PROGRAM="/sbin/udev.get_persistent_device_name.sh", NAME="%k" SYMLINK="%c{1+}"
+BUS="usb", PROGRAM="/sbin/udev.get_persistent_device_name.sh", NAME="%k" SYMLINK="%c{1+}"
+BUS="ide", PROGRAM="/sbin/udev.get_persistent_device_name.sh", NAME="%k" SYMLINK="%c{1+}"
 
 # S/390 zfcp, full disk
-BUS=="ccw", PROGRAM="/sbin/scsi_id" SYSFS{cutype}="1731/03" SYSFS{devtype}="1732/03" KERNEL="sd*[!0-9]", SYMLINK="disk/by-path/ccw-%s{hba_id}-fcp-%s{wwpn}:%s{fcp_lun} disk/by-serial/fcp-%c{1}%c{2}-%c{3}"
+BUS="ccw", PROGRAM="/sbin/scsi_id" SYSFS{cutype}="1731/03" SYSFS{devtype}="1732/03" KERNEL="sd*[!0-9]", NAME="%k" SYMLINK="disk/by-path/ccw-%s{hba_id}-fcp-%s{wwpn}:%s{fcp_lun} disk/by-serial/fcp-%c{1}%c{2}-%c{3}"
 
 # S/390 zfcp, disk partition
-BUS=="ccw", PROGRAM="/sbin/scsi_id" SYSFS{cutype}="1731/03" SYSFS{devtype}="1732/03" KERNEL="sd*[0-9]", SYMLINK="disk/by-path/ccw-%s{hba_id}-fcp-%s{wwpn}:%s{fcp_lun}-part%n disk/by-serial/fcp-%c{1}%c{2}-%c{3}-part%n"
+BUS="ccw", PROGRAM="/sbin/scsi_id" SYSFS{cutype}="1731/03" SYSFS{devtype}="1732/03" KERNEL="sd*[0-9]", NAME="%k" SYMLINK="disk/by-path/ccw-%s{hba_id}-fcp-%s{wwpn}:%s{fcp_lun}-part%n disk/by-serial/fcp-%c{1}%c{2}-%c{3}-part%n"
 
 # S/390 DASD, full disk
-BUS=="ccw", SYSFS_discipline="ECKD" KERNEL="dasd*[!0-9]", SYMLINK="disk/by-path/ccw-%b"
+BUS="ccw", SYSFS_discipline="ECKD" KERNEL="dasd*[!0-9]", NAME="%k" SYMLINK="disk/by-path/ccw-%b"
 
 # S/390 DASD, disk partition
-BUS=="ccw", SYSFS_discipline="ECKD" KERNEL="dasd*[0-9]", SYMLINK="disk/by-path/ccw-%b-part%n"
+BUS="ccw", SYSFS_discipline="ECKD" KERNEL="dasd*[0-9]", NAME="%k" SYMLINK="disk/by-path/ccw-%b-part%n"
 
 # KERNEL="ntibm*[0-9]", SYMLINK="tape/%b/non-rewinding"
 # KERNEL="rtibm*[0-9]", SYMLINK="tape/%b/rewinding"
+KERNEL="null", MODE="0666"
+KERNEL="zero", MODE="0666"
+KERNEL="urandom", MODE="0666"
+KERNEL="random", MODE="0666"
+KERNEL="ptmx", MODE="0666"
+KERNEL="tty",MODE="0666"
+KERNEL="inotify",MODE="0666"
