--- extras/rule_generator/75-cd-aliases-generator.rules.orig	2007-10-14 09:17:32.000000000 -0400
+++ extras/rule_generator/75-cd-aliases-generator.rules	2009-11-30 13:00:44.000000000 -0500
@@ -1,6 +1,11 @@
 # these rules generate rules for the /dev/{cdrom,dvd,...} symlinks
 
 # the path of removable devices changes frequently
-ACTION=="add", SUBSYSTEM=="block", SUBSYSTEMS=="usb|ieee1394", ENV{ID_CDROM}=="?*", ENV{GENERATED}!="?*", PROGRAM="write_cd_rules by-id", SYMLINK+="%c"
+ACTION=="add", SUBSYSTEM=="block", SUBSYSTEMS=="usb|ieee1394", \
+  ENV{ID_CDROM}=="?*", ENV{GENERATED}!="?*", PROGRAM="write_cd_rules by-id", \
+  SYMLINK+="%c", GOTO="persistent_cd_end"
 
-ACTION=="add", SUBSYSTEM=="block", ENV{ID_CDROM}=="?*", ENV{GENERATED}!="?*", PROGRAM="write_cd_rules", SYMLINK+="%c"
+ACTION=="add", SUBSYSTEM=="block", ENV{ID_CDROM}=="?*", ENV{GENERATED}!="?*", \
+  PROGRAM="write_cd_rules", SYMLINK+="%c"
+
+LABEL="persistent_cd_end"
