--- zsnes/src/linux/sdllink.c.orig   2008-07-27 22:13:02.000000000 +0200
+++ zsnes/src/linux/sdllink.c   2009-02-10 00:50:17.000000000 +0100
@@ -344,6 +344,8 @@
           case SDL_HAT_UP:
             pressed[offset + 3] = 1;
             pressed[offset + 2] = 0;
+            pressed[offset + 1] = 0;
+            pressed[offset + 0] = 0;
             break;
           case SDL_HAT_RIGHTUP:
             pressed[offset] = 1;
@@ -354,6 +356,8 @@
           case SDL_HAT_RIGHT:
             pressed[offset] = 1;
             pressed[offset + 1] = 0;
+            pressed[offset + 2] = 0;
+            pressed[offset + 3] = 0;
             break;
           case SDL_HAT_RIGHTDOWN:
             pressed[offset] = 1;
@@ -363,6 +367,8 @@
             break;
           case SDL_HAT_DOWN:
             pressed[offset + 2] = 1;
+            pressed[offset + 0] = 0;
+            pressed[offset + 1] = 0;
             pressed[offset + 3] = 0;
             break;
           case SDL_HAT_LEFTDOWN:
@@ -374,6 +380,8 @@
           case SDL_HAT_LEFT:
             pressed[offset + 1] = 1;
             pressed[offset] = 0;
+            pressed[offset + 2] = 0;
+            pressed[offset + 3] = 0;
             break;
           case SDL_HAT_LEFTUP:
             pressed[offset + 1] = 1;
