Index: libgksu/libgksu.c
===================================================================
--- libgksu/libgksu.c	(revisão 751)
+++ libgksu/libgksu.c	(revisão 754)
@@ -658,6 +658,10 @@
       password = g_locale_from_utf8 (found->secret,
 				     password_length,
 				     NULL, NULL, NULL);
+      password_length = strlen (password);
+
+      if (password[password_length-1] == '\n')
+	password[password_length-1] = '\0';
       return password;
     }
 
