diff -urN gnome-pilot-conduits-2.0.10/email/Makefile.in gnome-pilot-conduits-2.0.10.fixed/email/Makefile.in
--- gnome-pilot-conduits-2.0.10/email/Makefile.in	2003-07-16 11:44:36.000000000 -0500
+++ gnome-pilot-conduits-2.0.10.fixed/email/Makefile.in	2004-06-01 21:51:21.147956439 -0500
@@ -393,7 +393,9 @@
 
 
 sendmail.conduit: sendmail.conduit.in Makefile
-	sed -e 's^\@prefix\@^$(prefix)^g' < $(srcdir)/sendmail.conduit.in > sendmail.conduit.tmp \
+	sed -e 's^\@libdir\@^$(libdir)^g' \
+	    -e 's^\@prefix\@^$(prefix)^g' \
+            < $(srcdir)/sendmail.conduit.in > sendmail.conduit.tmp \
 	&& mv sendmail.conduit.tmp sendmail.conduit
 
 install-data-local:
diff -urN gnome-pilot-conduits-2.0.10/email/sendmail.conduit.in gnome-pilot-conduits-2.0.10.fixed/email/sendmail.conduit.in
--- gnome-pilot-conduits-2.0.10/email/sendmail.conduit.in	2001-02-28 16:43:03.000000000 -0600
+++ gnome-pilot-conduits-2.0.10.fixed/email/sendmail.conduit.in	2004-06-01 21:17:21.856048937 -0500
@@ -1,5 +1,5 @@
 <gnome-pilot-conduit version="1.0">
-<conduit id="gpsendmail1" type="shlib" location="@prefix@/lib/gnome-pilot/conduits/libemail_conduit.so"/>
+<conduit id="gpsendmail1" type="shlib" location="@libdir@/gnome-pilot/conduits/libemail_conduit.so"/>
 <name value="Sendmail"/>
 <conduit-attribute name="description" value="Mails your Palm emails via sendmail"/>
 <conduit-attribute name="default-synctype" value="custom"/>
diff -urN gnome-pilot-conduits-2.0.10/expense/expense.conduit.in gnome-pilot-conduits-2.0.10.fixed/expense/expense.conduit.in
--- gnome-pilot-conduits-2.0.10/expense/expense.conduit.in	2001-06-28 03:05:05.000000000 -0500
+++ gnome-pilot-conduits-2.0.10.fixed/expense/expense.conduit.in	2004-06-01 21:18:53.294849539 -0500
@@ -1,5 +1,5 @@
 <gnome-pilot-conduit version="1.0">
-	<conduit id="gpexpense1" type="shlib" location="@prefix@/lib/gnome-pilot/conduits/libexpense_conduit.so"/>
+	<conduit id="gpexpense1" type="shlib" location="@libdir@/gnome-pilot/conduits/libexpense_conduit.so"/>
 	<name value="Expense"/>
 	<conduit-attribute name="description" value="Extracts data from the expenses database"/>
 	<conduit-attribute name="default-synctype" value="synchronize"/>
diff -urN gnome-pilot-conduits-2.0.10/expense/Makefile.in gnome-pilot-conduits-2.0.10.fixed/expense/Makefile.in
--- gnome-pilot-conduits-2.0.10/expense/Makefile.in	2003-07-16 11:44:36.000000000 -0500
+++ gnome-pilot-conduits-2.0.10.fixed/expense/Makefile.in	2004-06-01 21:52:45.095705020 -0500
@@ -384,7 +384,9 @@
 
 
 expense.conduit: expense.conduit.in Makefile
-	sed -e 's^\@prefix\@^$(prefix)^g' < $(srcdir)/expense.conduit.in > expense.conduit.tmp \
+	sed -e 's^\@libdir\@^$(libdir)^g' \
+	    -e 's^\@prefix\@^$(prefix)^g' \
+	    < $(srcdir)/expense.conduit.in > expense.conduit.tmp \
 	&& mv expense.conduit.tmp expense.conduit
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -urN gnome-pilot-conduits-2.0.10/mal-conduit/src/Makefile.in gnome-pilot-conduits-2.0.10.fixed/mal-conduit/src/Makefile.in
--- gnome-pilot-conduits-2.0.10/mal-conduit/src/Makefile.in	2003-07-16 11:44:40.000000000 -0500
+++ gnome-pilot-conduits-2.0.10.fixed/mal-conduit/src/Makefile.in	2004-06-01 21:54:39.520987793 -0500
@@ -447,7 +447,9 @@
 
 
 mal.conduit: mal.conduit.in Makefile
-	sed -e 's^\@prefix\@^$(prefix)^g' < $(srcdir)/mal.conduit.in > mal.conduit.tmp \
+	sed -e 's^\@libdir\@^$(libdir)^g' \
+	    -e 's^\@prefix\@^$(prefix)^g' \
+	    < $(srcdir)/mal.conduit.in  > mal.conduit.tmp \
 	&& mv mal.conduit.tmp mal.conduit
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -urN gnome-pilot-conduits-2.0.10/mal-conduit/src/mal.conduit.in gnome-pilot-conduits-2.0.10.fixed/mal-conduit/src/mal.conduit.in
--- gnome-pilot-conduits-2.0.10/mal-conduit/src/mal.conduit.in	2001-05-11 22:07:19.000000000 -0500
+++ gnome-pilot-conduits-2.0.10.fixed/mal-conduit/src/mal.conduit.in	2004-06-01 21:21:44.377035445 -0500
@@ -1,5 +1,5 @@
 <gnome-pilot-conduit version="1.0">
-	<conduit id="malconduit1" type="shlib" location="@prefix@/lib/gnome-pilot/conduits/libmal_conduit.so"/>
+	<conduit id="malconduit1" type="shlib" location="@libdir@/gnome-pilot/conduits/libmal_conduit.so"/>
 <name value="MAL"/>
 	<conduit-attribute name="description" value="Mobile Application Link conduit"/>
 	<conduit-attribute name="url" value="http://www.avantgo.com"/>
diff -urN gnome-pilot-conduits-2.0.10/memo_file/Makefile.in gnome-pilot-conduits-2.0.10.fixed/memo_file/Makefile.in
--- gnome-pilot-conduits-2.0.10/memo_file/Makefile.in	2003-07-16 11:44:41.000000000 -0500
+++ gnome-pilot-conduits-2.0.10.fixed/memo_file/Makefile.in	2004-06-01 21:55:40.922009453 -0500
@@ -392,7 +392,9 @@
 
 
 memofile.conduit: memofile.conduit.in Makefile
-	sed -e 's^\@prefix\@^$(prefix)^g' < $(srcdir)/memofile.conduit.in > memofile.conduit.tmp \
+	sed -e 's^\@libdir\@^$(libdir)^g' \
+	    -e 's^\@prefix\@^$(prefix)^g' \
+	    < $(srcdir)/memofile.conduit.in > memofile.conduit.tmp \
 	&& mv memofile.conduit.tmp memofile.conduit
 
 install-data-local:
diff -urN gnome-pilot-conduits-2.0.10/memo_file/memofile.conduit.in gnome-pilot-conduits-2.0.10.fixed/memo_file/memofile.conduit.in
--- gnome-pilot-conduits-2.0.10/memo_file/memofile.conduit.in	2001-06-28 03:05:07.000000000 -0500
+++ gnome-pilot-conduits-2.0.10.fixed/memo_file/memofile.conduit.in	2004-06-01 21:11:05.547141169 -0500
@@ -1,5 +1,5 @@
 <gnome-pilot-conduit version="1.0">
-     <conduit id="gpmemo1" type="shlib" location="@prefix@/lib/gnome-pilot/conduits/libmemo_file_conduit.so"/>
+     <conduit id="gpmemo1" type="shlib" location="@libdir@/gnome-pilot/conduits/libmemo_file_conduit.so"/>
      <name value="MemoFile"/>
      <conduit-attribute name="description" value="Synchronizes your memos to local file"/>
      <conduit-attribute name="default-synctype" value="synchronize"/>
diff -urN gnome-pilot-conduits-2.0.10/time/Makefile.in gnome-pilot-conduits-2.0.10.fixed/time/Makefile.in
--- gnome-pilot-conduits-2.0.10/time/Makefile.in	2003-07-16 11:44:41.000000000 -0500
+++ gnome-pilot-conduits-2.0.10.fixed/time/Makefile.in	2004-06-01 21:56:38.391657058 -0500
@@ -382,7 +382,9 @@
 
 
 time.conduit: time.conduit.in Makefile
-	sed -e 's^\@prefix\@^$(prefix)^g' < $(srcdir)/time.conduit.in > time.conduit.tmp \
+	sed -e 's^\@libdir\@^$(libdir)^g' \
+	    -e 's^\@prefix\@^$(prefix)^g' \
+	    < $(srcdir)/time.conduit.in > time.conduit.tmp \
 	&& mv time.conduit.tmp time.conduit
 
 install-data-local:
diff -urN gnome-pilot-conduits-2.0.10/time/time.conduit.in gnome-pilot-conduits-2.0.10.fixed/time/time.conduit.in
--- gnome-pilot-conduits-2.0.10/time/time.conduit.in	2003-03-25 14:22:34.000000000 -0600
+++ gnome-pilot-conduits-2.0.10.fixed/time/time.conduit.in	2004-06-01 21:20:12.441380265 -0500
@@ -1,5 +1,5 @@
 <gnome-pilot-conduit version="1.0">
-     <conduit id="gptime1" type="shlib" location="@prefix@/lib/gnome-pilot/conduits/libtime_conduit.so"/>
+     <conduit id="gptime1" type="shlib" location="@libdir@/gnome-pilot/conduits/libtime_conduit.so"/>
      <name value="Time"/>
      <conduit-attribute name="description" value="Synchronizes the devices clock to the desktop"/>
      <conduit-attribute name="default-synctype" value="custom"/>
