diff -r -U 10 gtk+-2.8.4/configure.in gtk+-2.8.4_patch/configure.in
--- gtk+-2.8.4/configure.in	2005-08-31 16:22:33.000000000 +0200
+++ gtk+-2.8.4_patch/configure.in	2005-10-02 13:20:34.115180288 +0200
@@ -1349,21 +1349,21 @@
 else
   AM_CONDITIONAL(USE_LINUX_FB, false)
   AM_CONDITIONAL(ENABLE_FB_MANAGER, false)
 fi
 
 # Check for Pango flags
 
 if test "x$gdktarget" = "xwin32"; then
 	PANGO_PACKAGES="pangowin32 pangocairo"
 else
-	PANGO_PACKAGES="pango pangocairo"
+	PANGO_PACKAGES="pango pangox pangocairo"
 fi
 
 AC_MSG_CHECKING(Pango flags)
 if $PKG_CONFIG --exists $PANGO_PACKAGES ; then
         PANGO_CFLAGS=`$PKG_CONFIG --cflags $PANGO_PACKAGES`
         PANGO_LIBS=`$PKG_CONFIG --libs $PANGO_PACKAGES`
 
         AC_MSG_RESULT($PANGO_CFLAGS $PANGO_LIBS)
 else
         AC_MSG_ERROR([
