diff -Naur gtk+-2.11.6.orig/gtk/gtksettings.c gtk+-2.11.6/gtk/gtksettings.c
--- gtk+-2.11.6.orig/gtk/gtksettings.c	2007-07-20 16:12:29.000000000 -0400
+++ gtk+-2.11.6/gtk/gtksettings.c	2007-08-24 10:31:11.000000000 -0400
@@ -573,7 +573,7 @@
  							       P_("Tooltip timeout"),
  							       P_("Timeout before tooltip is shown"),
  							       0, G_MAXINT,
-							       1500,
+							       500,
  							       GTK_PARAM_READWRITE),
 					     NULL);
 
