diff -r 66a12ff27ba7 data/desktop_gnome_font_rendering.schemas.in
--- a/data/desktop_gnome_font_rendering.schemas.in	Mon Apr 20 15:10:29 2009 -0400
+++ b/data/desktop_gnome_font_rendering.schemas.in	Mon Apr 20 15:11:39 2009 -0400
@@ -19,7 +19,7 @@
 	    <applyto>/desktop/gnome/font_rendering/antialiasing</applyto>
             <owner>gnome</owner>
 	    <type>string</type>
-            <default>grayscale</default>
+            <default>rgba</default>
             <locale name="C">
 	        <short>Antialiasing</short>
                 <long>
@@ -35,7 +35,7 @@
 	    <applyto>/desktop/gnome/font_rendering/hinting</applyto>
             <owner>gnome</owner>
 	    <type>string</type>
-            <default>medium</default>
+            <default>slight</default>
             <locale name="C">
 	        <short>Hinting</short>
                 <long>
