diff -r 85b2f664b03d data/desktop_gnome_font_rendering.schemas.in
--- a/data/desktop_gnome_font_rendering.schemas.in	Mon Apr 06 14:47:06 2009 -0400
+++ b/data/desktop_gnome_font_rendering.schemas.in	Mon Apr 06 14:49:57 2009 -0400
@@ -19,7 +19,7 @@
 	    <applyto>/desktop/gnome/font_rendering/antialiasing</applyto>
             <owner>gnome</owner>
 	    <type>string</type>
-            <default>grayscale</default>
+            <default>rgba</default>
             <locale name="C">
 	        <short>Antialiasing</short>
                 <long>
