diff -Naur ndisgtk-0.6.orig/ndisgtk ndisgtk-0.6/ndisgtk
--- ndisgtk-0.6.orig/ndisgtk	2005-12-10 01:57:43.000000000 -0500
+++ ndisgtk-0.6/ndisgtk	2007-01-04 22:06:43.000000000 -0500
@@ -149,7 +149,7 @@
 		self.driver_list = {}
 
 		# Run the ndiswrapper list command
-		output = commands.getoutput("ndiswrapper -l")
+		output = commands.getoutput("/usr/sbin/ndiswrapper -l")
 
 		if "WARNING" in output:
 			error_dialog(_("Unable to see if hardware is present."), self.window)
@@ -247,8 +247,8 @@
 			error_dialog(_("Not a valid driver .inf file."), self.install_dialog)
 		else:
 			# Attempt to install driver
-			output = commands.getoutput("ndiswrapper -i %s" % inf_file)
-			commands.getoutput("ndiswrapper -m")
+			output = commands.getoutput("gksudo '/usr/sbin/ndiswrapper -i %s'" % inf_file)
+			commands.getoutput("gksudo '/usr/sbin/ndiswrapper -m'")
 
 			# Attempt to detect errors
 			if "already" in output:
@@ -256,8 +256,8 @@
 				error_dialog(_("Driver <b>%s</b> is already installed.") % driver_name, self.install_dialog)
 			else:
 				# Assume driver installed successfully. Set up and load module
-				os.system("ndiswrapper -m")
-				os.system("modprobe ndiswrapper")
+				os.system("gksudo '/usr/sbin/ndiswrapper -m'")
+				os.system("gksudo '/sbin/modprobe ndiswrapper'")
 
 				self.get_driver_list()
 				self.install_dialog_close()
@@ -279,7 +279,7 @@
 
 		if result == gtk.RESPONSE_YES:
 			# Remove driver
-			output = commands.getoutput("ndiswrapper -e %s" % driver_name)
+			output = commands.getoutput("gksudo '/usr/sbin/ndiswrapper -e %s'" % driver_name)
 
 			# Reload driver list
 			self.get_driver_list()
@@ -297,11 +297,6 @@
 
 
 if __name__ == '__main__':
-	# Check for root privileges
-	if os.getuid() != 0:
-		error_dialog(_("Root or sudo privileges required!"))
-		sys.exit(1)
-
 	# Parse options and load GUI
 	if "--kde" in sys.argv:
 		NdisGTK(kde=True)
diff -Naur ndisgtk-0.6.orig/ndisgtk.desktop ndisgtk-0.6/ndisgtk.desktop
--- ndisgtk-0.6.orig/ndisgtk.desktop	2005-12-07 21:47:02.000000000 -0500
+++ ndisgtk-0.6/ndisgtk.desktop	2007-01-04 22:14:27.000000000 -0500
@@ -3,7 +3,7 @@
 Name[fr]=Pilotes sans fil Windows
 Comment=Ndiswrapper driver installation tool
 Comment[fr]=Outil d'installation de pilotes Ndiswrapper
-Exec=gksudo /usr/bin/ndisgtk
+Exec=/usr/bin/ndisgtk
 Icon=/usr/share/ndisgtk/ndisgtk.png
 Terminal=false
 Type=Application
