--- at-3.1.7/parsetime.y.noon	Tue May 13 08:46:16 1997
+++ at-3.1.7/parsetime.y	Thu Mar 18 15:39:06 1999
@@ -181,7 +181,16 @@
 		;
 
 am_pm		: AM
-		| PM
+		    {
+		      if (exectm.tm_hour > 12) {
+			  yyerror("Hour too large for AM");
+			  YYERROR;
+		      }
+		      else if (exectm.tm_hour == 12) {
+			  exectm.tm_hour -=12;
+		      }
+		    }
+		  | PM
 		    {
 			if (exectm.tm_hour > 12) {
 			    yyerror("Hour too large for PM");
