--- boto/ec2/connection.py.old	2010-10-21 11:06:33.000000000 -0400
+++ boto/ec2/connection.py	2010-10-21 11:05:27.000000000 -0400
@@ -176,14 +176,14 @@
         except IndexError: # None of those images available
             return None
 
-    def register_image(self, name, description=None, image_location=None,
+    def register_image(self, name=None, description=None, image_location=None,
                        architecture=None, kernel_id=None, ramdisk_id=None,
                        root_device_name=None, block_device_map=None):
         """
         Register an image.
 
         :type name: string
-        :param name: The name of the AMI.
+        :param name: The name of the AMI.  Valid only for EBS-based images.
 
         :type description: string
         :param description: The description of the AMI.
@@ -210,7 +210,9 @@
         :rtype: string
         :return: The new image id
         """
-        params = {'Name': name}
+        params = {}
+        if name:
+            params['Name'] = name
         if description:
             params['Description'] = description
         if architecture:
@@ -220,7 +222,7 @@
         if ramdisk_id:
             params['RamdiskId'] = ramdisk_id
         if image_location:
-            params['Location'] = image_location
+            params['ImageLocation'] = image_location
         if root_device_name:
             params['RootDeviceName'] = root_device_name
         if block_device_map:
