diff -r e4aa96ac6acf src/canberra-gtk-play.c
--- a/src/canberra-gtk-play.c	Thu Dec 11 09:37:59 2008 -0500
+++ b/src/canberra-gtk-play.c	Thu Dec 11 09:39:01 2008 -0500
@@ -26,9 +26,16 @@
 #include <canberra-gtk.h>
 #include <locale.h>
 
+#if HAVE_UNISTD_H
+#include <unistd.h>
+#include <stdlib.h>
+#include <errno.h>
+#endif
+
 static int ret = 0;
 static ca_proplist *proplist = NULL;
 static int n_loops = 1;
+static gboolean background = FALSE;
 
 static void callback(ca_context *c, uint32_t id, int error, void *userdata);
 
@@ -88,6 +95,9 @@
         { "id",            'i', 0, G_OPTION_ARG_STRING, &event_id,          "Event sound identifier",  "STRING" },
         { "file",          'f', 0, G_OPTION_ARG_STRING, &filename,          "Play file",  "PATH" },
         { "description",   'd', 0, G_OPTION_ARG_STRING, &event_description, "Event sound description", "STRING" },
+#if HAVE_UNISTD_H
+        { "background",    'b', 0, G_OPTION_ARG_NONE,   &background,        "Run in background", NULL },
+#endif
         { "cache-control", 'c', 0, G_OPTION_ARG_STRING, &cache_control,     "Cache control (permanent, volatile, never)", "STRING" },
         { "loop",          'l', 0, G_OPTION_ARG_INT,    &n_loops,           "Loop how many times (detault: 1)", "INTEGER" },
 	{ "version",       'v', 0, G_OPTION_ARG_NONE,   &version,           "Display version number and quit", NULL },
@@ -115,6 +125,24 @@
         g_printerr("No event id or file specified.\n");
         return 1;
     }
+
+#if HAVE_UNISTD_H
+    if (background) {
+        switch (fork ()) {
+        case -1:
+                g_error ("Fork failed: %s", g_strerror (errno));
+                exit (EXIT_FAILURE);
+
+        case 0:
+                /* child */
+		break;
+
+         default:
+                /* parent */
+                exit (EXIT_SUCCESS);
+        }
+    }
+#endif
 
     ca_context_change_props(ca_gtk_context_get(),
                             CA_PROP_APPLICATION_NAME, "canberra-gtk-play",
diff -r e4aa96ac6acf src/libcanberra-login-sound.desktop.in
--- a/src/libcanberra-login-sound.desktop.in	Thu Dec 11 09:37:59 2008 -0500
+++ b/src/libcanberra-login-sound.desktop.in	Thu Dec 11 09:39:01 2008 -0500
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=GNOME Login Sound
-Exec=@bindir@/canberra-gtk-play --id="desktop-login" --description="GNOME Login"
+Exec=@bindir@/canberra-gtk-play --background --id="desktop-login" --description="GNOME Login"
 OnlyShowIn=GNOME;
 AutostartCondition=GNOME /desktop/gnome/sound/event_sounds
 X-GNOME-Autostart-Phase=Desktop
