--- libcanberra-0.25/src/Makefile.am		2010-06-13 18:41:49.000000000 +0100
+++ libcanberra-0.25.new/src/Makefile.am	2010-09-09 15:58:59.000000000 +0100
@@ -356,11 +356,11 @@ endif
 if HAVE_GTK3
 
 canberra_gtk_play_LDADD = \
-	$(GTK3_LIBS) \
+	$(GTK_LIBS) \
 	libcanberra.la \
-	libcanberra-gtk3.la
+	libcanberra-gtk.la
 canberra_gtk_play_CFLAGS = \
-	$(GTK3_CFLAGS)
+	$(GTK_CFLAGS)
 
 else
 if HAVE_GTK
