Index: kdepimlibs-4.4.85/doc/CMakeLists.txt
===================================================================
--- kdepimlibs-4.4.85.orig/doc/CMakeLists.txt	2010-06-06 04:02:35.876150946 +0200
+++ kdepimlibs-4.4.85/doc/CMakeLists.txt	2010-06-06 04:02:42.354900643 +0200
@@ -1,2 +1,2 @@
-add_subdirectory(kcontrol)
-add_subdirectory(kioslave)
+#add_subdirectory(kcontrol)
+#add_subdirectory(kioslave)
