diff --git a/debug_toolbar/panels/timer.py b/debug_toolbar/panels/timer.py
index d866166..6146e70 100644
--- a/debug_toolbar/panels/timer.py
+++ b/debug_toolbar/panels/timer.py
@@ -52,7 +52,7 @@ class TimerDebugPanel(DebugPanel):
     def _elapsed_ru(self, name):
         return getattr(self._end_rusage, name) - getattr(self._start_rusage, name)
 
-    def content(self):
+    def get_context(self):
 
         utime = 1000 * self._elapsed_ru('ru_utime')
         stime = 1000 * self._elapsed_ru('ru_stime')
@@ -90,4 +90,7 @@ class TimerDebugPanel(DebugPanel):
             'rows': rows,
         })
 
-        return render_to_string('debug_toolbar/panels/timer.html', context)
+        return context
+
+    def content(self):
+        return render_to_string('debug_toolbar/panels/timer.html', self.get_context())
diff --git a/debug_toolbar/panels/version.py b/debug_toolbar/panels/version.py
index 098ab62..ce1b483 100644
--- a/debug_toolbar/panels/version.py
+++ b/debug_toolbar/panels/version.py
@@ -8,7 +8,6 @@ from django.utils.translation import ugettext_lazy as _
 import debug_toolbar
 from debug_toolbar.panels import DebugPanel
 
-
 class VersionDebugPanel(DebugPanel):
     """
     Panel that displays the Django version.
@@ -28,7 +27,7 @@ class VersionDebugPanel(DebugPanel):
     def title(self):
         return _('Versions')
 
-    def content(self):
+    def get_context(self):
         versions = {}
         for app in settings.INSTALLED_APPS + ['django']:
             name = app.split('.')[-1].replace('_', ' ').capitalize()
@@ -56,4 +55,9 @@ class VersionDebugPanel(DebugPanel):
             'paths': sys.path,
         })
 
-        return render_to_string('debug_toolbar/panels/versions.html', context)
+        return context
+
+    def content(self):
+        return render_to_string('debug_toolbar/panels/versions.html',
+            self.get_context())
+
