Parse UUID values as strings.

diff --git a/pgsql.c b/pgsql.c
--- a/pgsql.c
+++ b/pgsql.c
@@ -85,6 +85,9 @@
 #define LARGE_OBJECTS	1		/* enables large objects support */
 #endif   /* NO_LARGE */
 
+/* This isn't defined in pg_type.h -- probably for a good reason. Oh well! */
+#define UUIDOID			2950
+
 /* --------------------------------------------------------------------- */
 
 /* MODULE GLOBAL VARIABLES */
@@ -985,6 +988,7 @@
 	case NAMEOID:
 	case CHAROID:
 	case TEXTOID:
+	case UUIDOID:
 	    /* It is assumed that null character terminates these strings */
 	    ret = PyString_FromStringAndSize(cell, cellsize);
 	    break;
