diff -r 2908f7bc767b pam/gkr-pam-module.c
--- a/pam/gkr-pam-module.c	Thu Oct 08 18:20:47 2009 +0200
+++ b/pam/gkr-pam-module.c	Thu Oct 08 18:22:25 2009 +0200
@@ -870,13 +870,21 @@
 	/* Look up the password */
 	ret = pam_get_item (ph, PAM_AUTHTOK, (const void**)&password);
 	if (ret != PAM_SUCCESS || password == NULL) {
-		if (ret == PAM_SUCCESS)
-			syslog (GKR_LOG_WARN, "gkr-pam: (auth) no password is available for user (NULL!)");
-		else
-			syslog (GKR_LOG_WARN, "gkr-pam: (auth) no password is available for user: %s", 
-			        pam_strerror (ph, ret));
-		return PAM_SUCCESS;
+	// begin OpenSUSE patch grafting
+               ret = prompt_password (ph);
+               if (ret != PAM_SUCCESS) {
+                       syslog (GKR_LOG_ERR, "gkr-pam: (auth) couldn't get the password from user: %s",
+                                pam_strerror (ph, ret));
+                       return PAM_AUTH_ERR;
+               }
+               ret = pam_get_item (ph, PAM_AUTHTOK, (const void**)&password);
+               if (ret != PAM_SUCCESS || password == NULL) {
+                       syslog (GKR_LOG_ERR, "gkr-pam: (auth) couldn't get the password from user: %s",
+                               ret == PAM_SUCCESS ? "password was null" : pam_strerror (ph, ret));
+                       return PAM_AUTH_ERR;
+               }	
 	}
+	// end OpenSUSE patch grafting
 	syslog (GKR_LOG_INFO, "gkr-pam: (auth) successfully looked up password for %s", user);
 
 	started_daemon = 0;
