Index: src/messageview.c
===================================================================
RCS file: //claws/src/messageview.c,v
retrieving revision 1.94.2.215
retrieving revision 1.94.2.216
diff -u -r1.94.2.215 -r1.94.2.216
--- src/messageview.c	10 Apr 2011 14:05:43 -0000	1.94.2.215
+++ src/messageview.c	10 Apr 2011 14:09:54 -0000	1.94.2.216
@@ -2558,7 +2558,7 @@
 	if (item) {
 		ac = account_find_from_item(item);
 		if (ac && ac->protocol == A_NNTP &&
-		    item->stype == F_NEWS) {
+		    FOLDER_TYPE(item->folder) == F_NEWS) {
 			compose_new(ac, item->path, NULL);
 			return;
 		}

