--- src/printing.c	2009-04-25 10:19:26.000000000 +0200
+++ src/printing.c.new	2009-10-05 14:38:41.410180616 +0200
@@ -279,6 +279,10 @@
 	/* Config for printing */
 	gtk_print_operation_set_print_settings(op, settings);
 	gtk_print_operation_set_default_page_setup(op, page_setup);
+#if GTK_CHECK_VERSION(2, 18, 0)
+        /* enable Page Size and Orientation in the print dialog */
+	gtk_print_operation_set_embed_page_setup(op, TRUE);
+#endif
 
 	/* signals */
 	g_signal_connect(op, "begin_print", G_CALLBACK(renderer->cb_begin_print), print_data);
