--- configure.sh.orig	2008-06-01 20:03:50.000000000 +0200
+++ configure.sh	2008-06-01 20:04:19.000000000 +0200
@@ -212,12 +212,12 @@
 #	mkdir -p \$(MANDIR)/man1
 #	cp vobcopy   \$(BINDIR)/vobcopy
 #	cp vobcopy.1 \$(MANDIR)/man1/vobcopy.1
-	install -d -m 755 \$(BINDIR)
-	install -d -m 755 \$(MANDIR)/man1
-	install -d -m 755 \$(MANDIR)/de/man1
-	install -m 755 vobcopy \$(BINDIR)/vobcopy
-	install -m 644 vobcopy.1 \$(MANDIR)/man1/vobcopy.1
-	install -m 644 vobcopy.1.de \$(MANDIR)/de/man1/vobcopy.1
+	install -d -m 755 \$(DESTDIR)\$(BINDIR)
+	install -d -m 755 \$(DESTDIR)\$(MANDIR)/man1
+	install -d -m 755 \$(DESTDIR)\$(MANDIR)/de/man1
+	install -m 755 vobcopy \$(DESTDIR)\$(BINDIR)/vobcopy
+	install -m 644 vobcopy.1 \$(DESTDIR)\$(MANDIR)/man1/vobcopy.1
+	install -m 644 vobcopy.1.de \$(DESTDIR)\$(MANDIR)/de/man1/vobcopy.1
 
 uninstall:
 	rm -f \$(BINDIR)/vobcopy
