diff -Naur NetworkManager-ppp_vpn-0.7.0-beta.orig/nm-ppp.desktop.in NetworkManager-ppp_vpn-0.7.0-beta/nm-ppp.desktop.in
--- NetworkManager-ppp_vpn-0.7.0-beta.orig/nm-ppp.desktop.in	2006-06-06 15:58:24.000000000 -0400
+++ NetworkManager-ppp_vpn-0.7.0-beta/nm-ppp.desktop.in	2007-05-15 17:14:42.000000000 -0400
@@ -4,9 +4,10 @@
 _GenericName=VPN Connection Manager (PPP generic)
 _Comment=Add, Remove, and Edit VPN Connections
 Exec=nm-vpn-properties --import-service org.freedesktop.NetworkManager.ppp_starter --import-file %f
-Icon=gnome-mime-application-x-ppp-settings
+Icon=gnome-mime-application-x-ppp-settings.png
 Terminal=false
 Type=Application
+NoDisplay=true
 DocPath=
 Categories=GNOME;Application;Network;
 MimeType=application/x-ppp-settings
