diff -Naur banshee-0.9.12.orig/data/banshee.desktop.in.in banshee-0.9.12/data/banshee.desktop.in.in
--- banshee-0.9.12.orig/data/banshee.desktop.in.in	2005-11-16 00:52:31.000000000 -0500
+++ banshee-0.9.12/data/banshee.desktop.in.in	2005-11-18 10:31:01.000000000 -0500
@@ -4,7 +4,6 @@
 _Name=Banshee Music Player
 _GenericName=Music Player
 Comment=A Music Management and Playback Application
-TryExec=banshee %U
 Exec=banshee
 Icon=music-player-banshee
 StartupNotify=true
