diff -r 6698f6036d68 data/desktop-files/banshee-1.desktop.mime
--- a/data/desktop-files/banshee-1.desktop.mime	Wed Jan 28 09:42:23 2009 -0500
+++ b/data/desktop-files/banshee-1.desktop.mime	Wed Jan 28 09:48:31 2009 -0500
@@ -98,37 +98,3 @@
 text/google-video-pointer
 text/x-google-video-pointer
 text/x-opml+xml
-video/3gpp
-video/avi
-video/dv
-video/fli
-video/flv
-video/mp4
-video/mp4v-es
-video/mpeg
-video/msvideo
-video/ogg
-video/quicktime
-video/vivo
-video/vnd.divx
-video/vnd.vivo
-video/x-anim
-video/x-avi
-video/x-flc
-video/x-fli
-video/x-flic
-video/x-flv
-video/x-m4v
-video/x-matroska
-video/x-mpeg
-video/x-mpg
-video/x-ms-asf
-video/x-msvideo
-video/x-ms-wm
-video/x-ms-wmv
-video/x-ms-wmx
-video/x-ms-wvx
-video/x-nsv
-video/x-ogm+ogg
-video/x-theora
-video/x-theora+ogg
