diff -r 6a6a33ff09f6 src/AssemblyInfo.cs
--- a/src/AssemblyInfo.cs	Thu Feb 19 22:12:12 2009 -0500
+++ b/src/AssemblyInfo.cs	Fri Feb 20 15:50:48 2009 -0500
@@ -3,7 +3,7 @@
 using System.Runtime.CompilerServices;
 
 [assembly: ApplicationVersion ("1.4.2", "1.4.2")]
-[assembly: ApplicationBuildInformation ("svn-checkout", "linux-gnu", "x86_64", "2009-01-20 18:40:37 EST")]
+[assembly: ApplicationBuildInformation ("source-tarball", "linux-gnu", "i686", "2009-02-20 14:38:36 EST")]
 
 [assembly: AssemblyVersion ("1.4.0.0")]
 [assembly: AssemblyTitle ("Banshee")]
diff -r 6a6a33ff09f6 src/Extensions/Banshee.NotificationArea/Notifications/Notification.cs
--- a/src/Extensions/Banshee.NotificationArea/Notifications/Notification.cs	Thu Feb 19 22:12:12 2009 -0500
+++ b/src/Extensions/Banshee.NotificationArea/Notifications/Notification.cs	Fri Feb 20 15:50:48 2009 -0500
@@ -282,10 +282,13 @@
 		}
 
 		public void AddAction (string action, string label, ActionHandler handler) {
-			lock (action_map) {
-				action_map[action] = new ActionTuple (label, handler);
+			if (Notifications.Global.Capabilities != null &&
+			    Array.IndexOf (Notifications.Global.Capabilities, "actions") > -1) {
+				lock (action_map) {
+					action_map[action] = new ActionTuple (label, handler);
+				}
+				Update ();
 			}
-			Update ();
 		}
 
 		public void RemoveAction (string action) {
