diff -Naur banshee-0.12.1.orig/ext/taglib-sharp/TagLib/Id3v2/Frames/TextIdentificationFrame.cs banshee-0.12.1/ext/taglib-sharp/TagLib/Id3v2/Frames/TextIdentificationFrame.cs
--- banshee-0.12.1.orig/ext/taglib-sharp/TagLib/Id3v2/Frames/TextIdentificationFrame.cs	2007-03-31 18:14:59.000000000 -0400
+++ banshee-0.12.1/ext/taglib-sharp/TagLib/Id3v2/Frames/TextIdentificationFrame.cs	2007-07-12 11:05:52.000000000 -0400
@@ -117,6 +117,11 @@
          field_list = new StringList ();
          text_encoding = StringType.UTF8;
          ParseFields (FieldData (data, offset));
+         
+         // Remove the last string if it's empty. (Workaround for defective
+         // nul terminated tags.
+         if (field_list.Count != 0 && field_list [field_list.Count - 1] == "")
+            field_list.RemoveAt (field_list.Count - 1);
       }
    }
 
