--- configure.old	2007-02-25 04:52:47.000000000 +0000
+++ configure	2007-02-25 04:53:08.000000000 +0000
@@ -23250,50 +23250,6 @@
 
 
 
-
-	gst_version=0.10
-	gst_pkg_name="gstreamer-$gst_version"
-	gst_inspect="gst-inspect-$gst_version"
-	gst_toolsdir=`$PKG_CONFIG --variable=toolsdir $gst_pkg_name`
-	gst_inspect="$gst_toolsdir/$gst_inspect"
-
-	{ echo "$as_me:$LINENO: checking for gst-inspect" >&5
-echo $ECHO_N "checking for gst-inspect... $ECHO_C" >&6; }
-	if ! test -x $gst_inspect; then
-		{ echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6; }
-		{ { echo "$as_me:$LINENO: error: Cannot find required gst-inspect tool." >&5
-echo "$as_me: error: Cannot find required gst-inspect tool." >&2;}
-   { (exit 1); exit 1; }; }
-	else
-		{ echo "$as_me:$LINENO: result: $gst_inspect" >&5
-echo "${ECHO_T}$gst_inspect" >&6; }
-	fi
-
-	for element in $(echo "0.10,decodebin
-	playbin
-	cdparanoia
-	gnomevfssink
-	gnomevfssrc
-	audioconvert
-	gconfaudiosink
-" | cut -d, -f2- | sed 's/\,/ /g'); do
-		{ echo "$as_me:$LINENO: checking for GStreamer $gst_version $element plugin" >&5
-echo $ECHO_N "checking for GStreamer $gst_version $element plugin... $ECHO_C" >&6; }
-		if $gst_inspect $element > /dev/null 2>/dev/null; then
-			{ echo "$as_me:$LINENO: result: yes" >&5
-echo "${ECHO_T}yes" >&6; }
-		else
-			{ echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6; }
-			{ { echo "$as_me:$LINENO: error: Cannot find required GStreamer-$gst_version plugin '$element'." >&5
-echo "$as_me: error: Cannot find required GStreamer-$gst_version plugin '$element'." >&2;}
-   { (exit 1); exit 1; }; }
-		fi
-	done;
-
-
-
 # Extract the first word of "gconftool-2", so it can be a program name with args.
 set dummy gconftool-2; ac_word=$2
 { echo "$as_me:$LINENO: checking for $ac_word" >&5
