Index: src/IconView.cs
===================================================================
RCS file: /cvs/gnome/f-spot/src/IconView.cs,v
retrieving revision 1.130
diff -u -r1.130 IconView.cs
--- src/IconView.cs	31 Aug 2006 22:01:26 -0000	1.130
+++ src/IconView.cs	10 Sep 2006 23:01:08 -0000
@@ -850,10 +850,10 @@
 			Pango.Layout layout = (Pango.Layout)date_layouts [date];
 			if (layout == null) {
 				layout = new Pango.Layout (this.PangoContext);
-				layout.SetText (date);
+				layout.SetMarkup ("<span foreground=\"white\">" + date + "</span>");
 				date_layouts [date] = layout;
 			}
-			
+
 			layout.GetPixelSize (out layout_bounds.Width, out layout_bounds.Height);
 
 			layout_bounds.Y = bounds.Y + bounds.Height - cell_border_width - layout_bounds.Height;
@@ -1282,11 +1282,12 @@
 		}
 	}
 
+    private static Gdk.Color gdk_black = new Gdk.Color (0, 0, 0);
 	private void SetColors () 
 	{
 		if (IsRealized) {
-			BinWindow.Background = Style.BaseColors [(int)State];
-			GdkWindow.Background = Style.BaseColors [(int)State];
+			BinWindow.Background = gdk_black;
+			GdkWindow.Background = gdk_black;
 		}
 	}
 
Index: src/PhotoView.cs
===================================================================
RCS file: /cvs/gnome/f-spot/src/PhotoView.cs,v
retrieving revision 1.94
diff -u -r1.94 PhotoView.cs
--- src/PhotoView.cs	24 May 2006 06:59:15 -0000	1.94
+++ src/PhotoView.cs	10 Sep 2006 23:01:09 -0000
@@ -450,7 +450,6 @@
 		description_delay = new FSpot.Delay (1000, new GLib.IdleHandler (CommitPendingChanges));
 		this.Destroyed += HandleDestroy;
 
-
 		Box vbox = new VBox (false, 6);
 		Add (vbox);
 
@@ -566,6 +565,8 @@
 		FSpot.Global.ModifyColors (photo_view);
 		FSpot.Global.ModifyColors (background);
 		FSpot.Global.ModifyColors (photo_view_scrolled);
+
+        photo_view.ModifyBg (Gtk.StateType.Normal, photo_view.Style.Black);
 	}
 
 	protected override void OnStyleSet (Style previous)
