diff -Naur f-spot-0.2.0.orig/src/f-spot.in f-spot-0.2.0/src/f-spot.in
--- f-spot-0.2.0.orig/src/f-spot.in	2006-03-01 02:51:19.000000000 -0500
+++ f-spot-0.2.0/src/f-spot.in	2006-08-29 12:28:02.000000000 -0400
@@ -4,6 +4,9 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 
+#Run db conversion if needed
+/usr/share/f-spot/f-spot_migrate.sh
+
 EXE_TO_RUN="$libdir/f-spot/f-spot.exe"
 
 case "x--uninstalled" in ("x$1" | "x$2")
