From 5c0458cb0fda7ea5258d6f253ff8d662230d5e59 Mon Sep 17 00:00:00 2001
From: Guido Berhoerster <guido+xfce@berhoerster.name>
Date: Thu, 27 Sep 2012 15:41:30 +0000
Subject: Fix a typo

---
diff --git a/libxfce4mixer/xfce-mixer-debug.c b/libxfce4mixer/xfce-mixer-debug.c
index d45b92d..54449b3 100644
--- a/libxfce4mixer/xfce-mixer-debug.c
+++ b/libxfce4mixer/xfce-mixer-debug.c
@@ -82,10 +82,7 @@ xfce_mixer_debug_init (const gchar *log_domain,
     }
 #else
   if (!debug_mode)
-    {
-      g_log_set_handler (log_domain, G_LOG_LEVEL_DEBUG, xfce4_mixer_dummy_log_handler, NULL);
-      g_log_set_handler (G_LOG_DOMAIN, G_LOG_LEVEL_DEBUG, xfce4_mixer_dummy_log_handler, NULL);
-    }
+    g_log_set_handler (log_domain, G_LOG_LEVEL_DEBUG, xfce_mixer_dummy_log_handler, NULL);
 #endif /* GLIB_CHECK_VERSION (2,32,0) */
 }
 
--
cgit v0.9.0.3
