Index: kdebase-runtime-4.2.85/kdebugdialog/klistdebugdialog.cpp
===================================================================
--- kdebase-runtime-4.2.85.orig/kdebugdialog/klistdebugdialog.cpp	2009-05-13 09:20:32.000000000 -0400
+++ kdebase-runtime-4.2.85/kdebugdialog/klistdebugdialog.cpp	2009-05-13 09:20:47.000000000 -0400
@@ -117,7 +117,7 @@
     QTreeWidgetItem* item = m_areaWidget->topLevelItem(i);
     KConfigGroup group = pConfig->group( item->data(0, Qt::UserRole).toByteArray() ); // Group name = debug area code = cb's name
 
-    int setting = group.readEntry("InfoOutput", -1);
+    int setting = group.readEntry("InfoOutput", 4);
 
     switch (setting) {
       case 4: // off
