Index: kdebase-runtime-4.4.92/phonon/kded-module/phononserver.cpp
===================================================================
--- kdebase-runtime-4.4.92.orig/phonon/kded-module/phononserver.cpp	2010-07-23 12:55:41.247724000 -0400
+++ kdebase-runtime-4.4.92/phonon/kded-module/phononserver.cpp	2010-07-23 12:56:25.023724002 -0400
@@ -781,8 +781,17 @@
 
 void PhononServer::askToRemoveDevices(const QStringList &devList, const QList<int> &indexes)
 {
-    const QString &dontAskAgainName = QLatin1String("phonon_forget_devices_") +
-        devList.join(QLatin1String("_"));
+    KSharedConfigPtr config = KGlobal::config();
+    KConfigGroup phononGroup = config->group("phonon");
+    bool alwaysForget = phononGroup.readEntry("phononAlwaysForget", false);
+    QString dontAskAgainName;
+    if (alwaysForget) {
+        dontAskAgainName = QLatin1String("phonon_forget_devices_always");
+        KMessageBox::saveDontShowAgainYesNo(dontAskAgainName, KMessageBox::Yes);
+    } else {
+        dontAskAgainName = QLatin1String("phonon_forget_devices_") +
+            devList.join(QLatin1String("_"));
+    }
     KMessageBox::ButtonCode result;
     if (!KMessageBox::shouldBeShownYesNo(dontAskAgainName, result)) {
         if (result == KMessageBox::Yes) {
