Index: runtime/nepomuk/services/storage/nepomukcore.cpp
===================================================================
--- runtime/nepomuk/services/storage/nepomukcore.cpp	(revision 1083505)
+++ runtime/nepomuk/services/storage/nepomukcore.cpp	(working copy)
@@ -23,6 +23,7 @@
 #include <KSharedConfig>
 #include <KConfigGroup>
 #include <KStandardDirs>
+#include <KProcess>
 
 #include <QtCore/QTimer>
 
@@ -51,6 +52,11 @@
 {
     // TODO: export the main model on org.kde.NepomukRepository via Soprano::Server::DBusExportModel
 
+    KProcess p;
+    p << "virtuosoconverter" << "--auto";
+    p.start();
+    p.waitForFinished(-1);
+
     // we have only the one repository
     model( QLatin1String( s_repositoryName ) );
 }