New search engines, put into kdebase after 4.5 release
Index: kdebase-runtime-4.4.92/kurifilter-plugins/ikws/searchproviders/baidu.desktop
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ kdebase-runtime-4.4.92/kurifilter-plugins/ikws/searchproviders/baidu.desktop	2010-07-23 14:08:41.039368002 +0100
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Type=Service
+Name=Baidu
+Name[zh_CN]=百度
+X-KDE-ServiceTypes=SearchProvider
+Keys=baidu
+Query=http://www.baidu.com/s?wd=\\{@}&tn=ubuntuu_cb&cl=3
Index: kdebase-runtime-4.4.92/kurifilter-plugins/ikws/searchproviders/bing.desktop
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ kdebase-runtime-4.4.92/kurifilter-plugins/ikws/searchproviders/bing.desktop	2010-07-23 13:25:40.239368002 +0100
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Type=Service
+Name=Bing
+X-KDE-ServiceTypes=SearchProvider
+Keys=bing
+Query=http://www.bing.com/search?q=\\{@}
