Index: kdebase-runtime-4.3.80/knetattach/knetattach.desktop
===================================================================
--- kdebase-runtime-4.3.80.orig/knetattach/knetattach.desktop	2009-12-03 14:53:41.000000000 -0500
+++ kdebase-runtime-4.3.80/knetattach/knetattach.desktop	2009-12-03 14:54:05.000000000 -0500
@@ -173,3 +173,4 @@
 X-KDE-StartupNotify=true
 Categories=Qt;KDE;Network;RemoteAccess;
 X-DocPath=knetattach/index.html
+NoDisplay=true
