Index: kdebase-runtime-4.2.0/knetattach/knetattach.cpp
===================================================================
--- kdebase-runtime-4.2.0.orig/knetattach/knetattach.cpp	2009-01-22 21:18:54.000000000 -0330
+++ kdebase-runtime-4.2.0/knetattach/knetattach.cpp	2009-01-22 21:19:03.000000000 -0330
@@ -31,6 +31,8 @@
 #include <kstandarddirs.h>
 #include <kdirnotify.h>
 #include <kcharsets.h>
+#include <kservice.h>
+#include <kmimetypetrader.h>
 
 KNetAttach::KNetAttach( QWidget* parent )
     : Q3Wizard( parent ), Ui_KNetAttach()
@@ -215,7 +217,9 @@
 	return;
     }
 
-    KToolInvocation::invokeBrowser(url.url());
+    KService::Ptr fb = KMimeTypeTrader::self()->preferredService("inode/directory",
+                                                                 "Application");
+    KToolInvocation::startServiceByDesktopName(fb->desktopEntryName(),url.url());
 
     QString name = _connectionName->text().trimmed();
 
