Index: kdebase-runtime-4.4.1/nepomuk/services/strigi/eventmonitor.cpp
===================================================================
--- kdebase-runtime-4.4.1.orig/nepomuk/services/strigi/eventmonitor.cpp	2010-03-21 10:33:02.272974330 -0400
+++ kdebase-runtime-4.4.1/nepomuk/services/strigi/eventmonitor.cpp	2010-03-21 10:37:35.632975122 -0400
@@ -66,9 +66,9 @@
         m_initialIndexTime.start();
 
         // inform the user about the initial indexing
-        sendEvent( "initialIndexingStarted",
-                   i18n( "Indexing files for fast searching. This process may take a while." ),
-                   "nepomuk" );
+//         sendEvent( "initialIndexingStarted",
+//                    i18n( "Indexing files for fast searching. This process may take a while." ),
+//                    "nepomuk" );
 
         // connect to get the end of initial indexing
         // (make it a queued connection since KNotification does not like multithreading
@@ -96,7 +96,7 @@
         kDebug() << "Resuming indexer due to power management";
         m_pauseState = NotPaused;
         m_indexScheduler->resume();
-        sendEvent( "indexingResumed", i18n("Resuming indexing of files for fast searching."), "battery-charging" );
+//         sendEvent( "indexingResumed", i18n("Resuming indexing of files for fast searching."), "battery-charging" );
     }
     else if ( conserveResources &&
               m_indexScheduler->isRunning() &&
@@ -104,7 +104,7 @@
         kDebug() << "Pausing indexer due to power management";
         m_pauseState = PausedDueToPowerManagement;
         m_indexScheduler->suspend();
-        sendEvent( "indexingSuspended", i18n("Suspending the indexing of files to preserve resources."), "battery-100" );
+//         sendEvent( "indexingSuspended", i18n("Suspending the indexing of files to preserve resources."), "battery-100" );
     }
 }
 
@@ -143,11 +143,11 @@
     // inform the user about the end of initial indexing. This will only be called once
     if ( !m_indexScheduler->isSuspended() ) {
         kDebug() << "initial indexing took" << m_initialIndexTime.elapsed();
-        sendEvent( "initialIndexingFinished",
-                   i18nc( "@info %1 is a duration formatted using KLocale::formatDuration",
-                          "Initial indexing of files for fast searching finished in %1",
-                          KGlobal::locale()->formatDuration( m_initialIndexTime.elapsed() ) ),
-                   "nepomuk" );
+//         sendEvent( "initialIndexingFinished",
+//                    i18nc( "@info %1 is a duration formatted using KLocale::formatDuration",
+//                           "Initial indexing of files for fast searching finished in %1",
+//                           KGlobal::locale()->formatDuration( m_initialIndexTime.elapsed() ) ),
+//                    "nepomuk" );
         m_indexScheduler->disconnect( this );
 
         m_periodicUpdateTimer.start();
Index: kdebase-runtime-4.4.1/nepomuk/services/strigi/systray.cpp
===================================================================
--- kdebase-runtime-4.4.1.orig/nepomuk/services/strigi/systray.cpp	2010-03-21 10:39:50.852984986 -0400
+++ kdebase-runtime-4.4.1/nepomuk/services/strigi/systray.cpp	2010-03-21 10:44:14.836975409 -0400
@@ -77,7 +77,10 @@
     setToolTip("nepomuk", i18n("Search Service"),  m_service->userStatusString() );
     m_suspendResumeAction->setChecked( m_service->indexScheduler()->isSuspended() );
     // a manually suspended service should not be passive
-    setStatus( m_service->isIdle() && !m_service->isSuspended() ? Passive : Active );
+    // Kubuntu patch:
+    // But since the average user doesn't really care, we're just going to never
+    // let it become active in the first place
+    // setStatus( m_service->isIdle() && !m_service->isSuspended() ? Passive : Active );
 }
 
 
