Index: kdebase-runtime-4.3.80/knetattach/knetattach.cpp
===================================================================
--- kdebase-runtime-4.3.80.orig/knetattach/knetattach.cpp	2009-12-03 14:53:41.000000000 -0500
+++ kdebase-runtime-4.3.80/knetattach/knetattach.cpp	2009-12-03 14:54:05.000000000 -0500
@@ -189,7 +189,7 @@
     KConfig config("kio_fishrc");
     KConfigGroup cg(&config, _host->text().trimmed());
     cg.writeEntry("Charset", KGlobal::charsets()->encodingForName(_encoding->currentText()));
-	url.setProtocol("fish");
+	url.setProtocol("sftp");
 	url.setPort(_port->value());
     } else if (_type == "FTP") {
 	url.setProtocol("ftp");
