diff -u ndiswrapper-1.54/debian/rules ndiswrapper-1.54/debian/rules
--- ndiswrapper-1.54/debian/rules
+++ ndiswrapper-1.54/debian/rules
@@ -1,22 +1,23 @@
 #!/usr/bin/make -f
+include /usr/share/quilt/quilt.make
 # Uncomment this to turn on verbose mode.
 #export DH_VERBOSE=1
 
 build: build-indep build-arch
 
 build-indep: build-indep-stamp
-build-indep-stamp:
+build-indep-stamp: $(QUILT_STAMPFN)
 	touch $@
 
 build-arch: build-arch-stamp
-build-arch-stamp:
+build-arch-stamp: $(QUILT_STAMPFN)
 	dh_testdir
 
 	$(MAKE) -C utils
 
 	touch $@
 
-clean:
+clean: unpatch
 	dh_testdir
 	dh_testroot
 
diff -u ndiswrapper-1.54/debian/control ndiswrapper-1.54/debian/control
--- ndiswrapper-1.54/debian/control
+++ ndiswrapper-1.54/debian/control
@@ -4,7 +4,7 @@
 Maintainer: Julian Andres Klode <jak@debian.org>
 Uploaders: Kel Modderman <kel@otaku42.de>
 Standards-Version: 3.8.1
-Build-Depends: debhelper (>= 5), bzip2
+Build-Depends: debhelper (>= 5), bzip2, quilt
 Homepage: http://ndiswrapper.sourceforge.net/
 
 Package: ndiswrapper-common
only in patch2:
unchanged:
--- ndiswrapper-1.54.orig/debian/patches/fix-typo-in-netdev-ops-struct.patch
+++ ndiswrapper-1.54/debian/patches/fix-typo-in-netdev-ops-struct.patch
@@ -0,0 +1,16 @@
+fix C syntax error and field name in conditional netdev ops struct, 
+triggering on kernel >= 2.6.29 and CONFIG_NET_POLL_CONTROLLER=y.
+
+Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
+
+--- a/driver/wrapndis.c
++++ b/driver/wrapndis.c
+@@ -1744,7 +1744,7 @@ static const struct net_device_ops ndis_
+ 	.ndo_set_mac_address = ndis_set_mac_address,
+ 	.ndo_get_stats = ndis_get_stats,
+ #ifdef CONFIG_NET_POLL_CONTROLLER
+-	.poll_controller = ndis_poll_controller;
++	.ndo_poll_controller = ndis_poll_controller,
+ #endif
+ };
+ #endif
only in patch2:
unchanged:
--- ndiswrapper-1.54.orig/debian/patches/series
+++ ndiswrapper-1.54/debian/patches/series
@@ -0,0 +1 @@
+fix-typo-in-netdev-ops-struct.patch
