diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -164,16 +164,6 @@
 [AC_MSG_RESULT([found])],
 [AC_MSG_ERROR([not found, cannot proceed])])
 
-dnl libevent < 1.3b crashes on event_base_free()
-dnl no good way to check libevent version.  use hack:
-dnl evhttp.h defines HTTP_SERVUNAVAIL only since 1.3b
-AC_MSG_CHECKING([whether libevent version >= 1.3b])
-AC_EGREP_CPP([HTTP_SERVUNAVAIL],
-[#include <evhttp.h>
-  HTTP_SERVUNAVAIL ],
-[AC_MSG_ERROR([no, cannot proceed])],
-[AC_MSG_RESULT([yes])])
-
 AC_CHECK_FUNCS(event_loopbreak)
 
 # autoconf does not want to find 'install', if not using automake...
