diff -r 72c82acac04e src/applet.c
--- a/src/applet.c	Tue Mar 17 23:52:46 2009 -0400
+++ b/src/applet.c	Tue Mar 17 23:56:09 2009 -0400
@@ -70,6 +70,7 @@
 #include "utils.h"
 #include "gconf-helpers.h"
 
+#define NOTIFY_CAPS_ACTIONS_KEY "actions"
 
 G_DEFINE_TYPE(NMApplet, nma, G_TYPE_OBJECT)
 
@@ -396,7 +397,8 @@
 	notify_notification_set_urgency (notify, urgency);
 	notify_notification_set_timeout (notify, NOTIFY_EXPIRES_DEFAULT);
 
-	if (action1) {
+
+        if (applet->notify_with_actions && action1) {
 		notify_notification_add_action (notify, action1, action1_label,
 		                                action1_cb, action1_user_data, NULL);
 	}
@@ -2477,6 +2479,7 @@
 {
 	NMApplet *applet;
 	AppletDBusManager *dbus_mgr;
+        GList *server_caps;
 
 	applet = NM_APPLET (G_OBJECT_CLASS (nma_parent_class)->constructor (type, n_props, construct_props));
 
@@ -2507,6 +2510,15 @@
 	if (!notify_is_initted ())
 		notify_init ("NetworkManager");
 
+        server_caps = notify_get_server_caps();
+        if (g_list_find (server_caps, NOTIFY_CAPS_ACTIONS_KEY))
+                applet->notify_with_actions = TRUE;
+        else
+                applet->notify_with_actions = FALSE;
+ 
+        g_list_foreach (server_caps, (GFunc) g_free, NULL);
+        g_list_free (server_caps);
+
 	dbus_mgr = applet_dbus_manager_get ();
 	if (dbus_mgr == NULL) {
 		nm_warning ("Couldn't initialize the D-Bus manager.");
diff -r 72c82acac04e src/applet.h
--- a/src/applet.h	Tue Mar 17 23:52:46 2009 -0400
+++ b/src/applet.h	Tue Mar 17 23:56:09 2009 -0400
@@ -143,6 +143,10 @@
 
 	GladeXML *		info_dialog_xml;
 	NotifyNotification*	notification;
+
+        /* Notification stuff */
+        gboolean                notify_with_actions;
+
 } NMApplet;
 
 
