diff -ru libnl-1.0-pre5-64bit-issues/lib/Makefile libnl-1.0-pre5/lib/Makefile
--- libnl-1.0-pre5-64bit-issues/lib/Makefile	2005-11-02 12:29:06.000000000 -0500
+++ libnl-1.0-pre5/lib/Makefile	2005-11-02 17:07:47.000000000 -0500
@@ -13,7 +13,7 @@
     include ../Makefile.opts
 endif
 
-LIBDIR=$(prefix)/lib
+LIBDIR=$(libdir)
 
 # Core
 CIN      := $(wildcard *.c)
diff -ru libnl-1.0-pre5-64bit-issues/Makefile.opts.in libnl-1.0-pre5/Makefile.opts.in
--- libnl-1.0-pre5-64bit-issues/Makefile.opts.in	2005-11-02 12:29:06.000000000 -0500
+++ libnl-1.0-pre5/Makefile.opts.in	2005-11-02 17:06:50.000000000 -0500
@@ -19,6 +19,7 @@
 LIBNL_LIB        := @LIBNL_LIB@
 
 prefix           := @prefix@
+libdir           := @libdir@
 mandir           := @mandir@
 sysconfdir       := @sysconfdir@
 
