--- SConstruct	2009-07-20 17:55:10.000000000 +0200
+++ SConstruct.new	2009-07-31 16:22:02.320704438 +0200
@@ -512,7 +512,7 @@
 
         if conf.CheckHeader ('fftw3.h') == False:
             print ('Ardour cannot be compiled without the FFTW3 headers, which do not seem to be installed')
-            sys.exit (1)            
+        #    sys.exit (1)            
         conf.Finish()
 
 if env['FREESOUND']:

