diff --git a/settings/xfpm-settings.c b/settings/xfpm-settings.c
index 67cc5a1..5748091 100644
--- a/settings/xfpm-settings.c
+++ b/settings/xfpm-settings.c
@@ -168,9 +168,9 @@ void        _cursor_changed_cb 			   (GtkTreeView *view,
 
 void brightness_level_on_ac (GtkSpinButton *w,  XfconfChannel *channel)
 {
-     gint val = (gint) gtk_spin_button_get_value (w);
+    guint val = (guint) gtk_spin_button_get_value (w);
     
-    if (!xfconf_channel_set_int (channel, PROPERTIES_PREFIX BRIGHTNESS_LEVEL_ON_AC, val) )
+    if (!xfconf_channel_set_uint (channel, PROPERTIES_PREFIX BRIGHTNESS_LEVEL_ON_AC, val) )
     {
 	g_critical ("Unable to set value %d for property %s\n", val, BRIGHTNESS_LEVEL_ON_AC);
     }
@@ -178,9 +178,9 @@ void brightness_level_on_ac (GtkSpinButton *w,  XfconfChannel *channel)
 
 void brightness_level_on_battery (GtkSpinButton *w,  XfconfChannel *channel)
 {
-     gint val = (gint) gtk_spin_button_get_value (w);
+     guint val = (guint) gtk_spin_button_get_value (w);
     
-    if (!xfconf_channel_set_int (channel, PROPERTIES_PREFIX BRIGHTNESS_LEVEL_ON_BATTERY, val) )
+    if (!xfconf_channel_set_uint (channel, PROPERTIES_PREFIX BRIGHTNESS_LEVEL_ON_BATTERY, val) )
     {
 	g_critical ("Unable to set value %d for property %s\n", val, BRIGHTNESS_LEVEL_ON_BATTERY);
     }
diff --git a/src/xfpm-xfconf.c b/src/xfpm-xfconf.c
index 9bef947..c82205d 100644
--- a/src/xfpm-xfconf.c
+++ b/src/xfpm-xfconf.c
@@ -339,7 +339,7 @@ xfpm_xfconf_class_init (XfpmXfconfClass *klass)
      **/
     g_object_class_install_property (object_class,
                                      PROP_BRIGHTNESS_LEVEL_ON_AC,
-                                     g_param_spec_int  (BRIGHTNESS_LEVEL_ON_AC,
+                                     g_param_spec_uint  (BRIGHTNESS_LEVEL_ON_AC,
                                                         NULL, NULL,
 							1,
 							100,
