commit 841ba525c2fa30d5ae11473b60d2855b68e1e84d
Author: Mark Trompell <mark@foresightlinux.org>
Date:   Wed Nov 30 15:35:50 2011 +0100

    dimmed_level should always be lower than current

diff --git a/src/xfpm-backlight.c b/src/xfpm-backlight.c
index d69c588..2555bc6 100644
--- a/src/xfpm-backlight.c
+++ b/src/xfpm-backlight.c
@@ -84,29 +84,31 @@ G_DEFINE_TYPE (XfpmBacklight, xfpm_backlight, G_TYPE_OBJECT)
 static void
 xfpm_backlight_dim_brightness (XfpmBacklight *backlight)
 {
-    gboolean ret;
-    
-    if (xfpm_power_get_mode (backlight->priv->power) == XFPM_POWER_MODE_NORMAL )
+    if ( !backlight->priv->dimmed )
     {
-	gint dim_level;
-	
-	g_object_get (G_OBJECT (backlight->priv->conf),
-		      backlight->priv->on_battery ? BRIGHTNESS_LEVEL_ON_BATTERY : BRIGHTNESS_LEVEL_ON_AC, &dim_level,
-		      NULL);
-	
-	ret = xfpm_brightness_get_level (backlight->priv->brightness, &backlight->priv->last_level);
-	
-	if ( !ret )
-	{
-	    g_warning ("Unable to get current brightness level");
-	    return;
-	}
-	
-	dim_level = dim_level * backlight->priv->max_level / 100;
-	
-	XFPM_DEBUG ("Current brightness level before dimming : %i, new %i", backlight->priv->last_level, dim_level);
-	
-	backlight->priv->dimmed = xfpm_brightness_set_level (backlight->priv->brightness, dim_level);
+        gboolean ret;
+        
+        if (xfpm_power_get_mode (backlight->priv->power) == XFPM_POWER_MODE_NORMAL )
+        {
+            glong dim_level;
+            
+            g_object_get (G_OBJECT (backlight->priv->conf),
+                      backlight->priv->on_battery ? BRIGHTNESS_LEVEL_ON_BATTERY : BRIGHTNESS_LEVEL_ON_AC, &dim_level,
+                      NULL);
+            
+            ret = xfpm_brightness_get_level (backlight->priv->brightness, &backlight->priv->last_level);
+        
+            if ( !ret )
+            {
+                g_warning ("Unable to get current brightness level");
+                return;
+            }
+            
+            dim_level = dim_level * backlight->priv->last_level / 100;
+            
+            XFPM_DEBUG ("Current brightness level before dimming : %li, new %li", backlight->priv->last_level, dim_level);
+            backlight->priv->dimmed = xfpm_brightness_set_level (backlight->priv->brightness, dim_level);
+        }
     }
 }
 
