From ab07509cdf0fa05c56ae61a020abce2ac1fb28cb Mon Sep 17 00:00:00 2001
From: Ali Abdallah <aliov@xfce.org>
Date: Mon, 27 Jun 2011 14:55:28 +0000
Subject: Only dim brightness if the current level is brighter.

---
diff --git a/src/xfpm-backlight.c b/src/xfpm-backlight.c
index 94a4f44..73bd5f6 100644
--- a/src/xfpm-backlight.c
+++ b/src/xfpm-backlight.c
@@ -104,9 +104,15 @@ xfpm_backlight_dim_brightness (XfpmBacklight *backlight)
 	
 	dim_level = dim_level * backlight->priv->max_level / 100;
 	
-	XFPM_DEBUG ("Current brightness level before dimming : %i, new %i", backlight->priv->last_level, dim_level);
-	
-	backlight->priv->dimmed = xfpm_brightness_set_level (backlight->priv->brightness, dim_level);
+	/**
+	 * Only reduce if the current level is brighter than
+	 * the configured dim_level
+	 **/
+	if (backlight->priv->last_level > dim_level)
+	{
+	    XFPM_DEBUG ("Current brightness level before dimming : %li, new %li", backlight->priv->last_level, dim_level);
+	    backlight->priv->dimmed = xfpm_brightness_set_level (backlight->priv->brightness, dim_level);
+	}
     }
 }
 
--
cgit 
