--- xorg-x11-6.8.2/xc/programs/Xserver/GL/dri/dri.c.orig	2005-03-03 09:58:05.335820408 -0500
+++ xorg-x11-6.8.2/xc/programs/Xserver/GL/dri/dri.c	2005-03-03 09:58:09.055254968 -0500
@@ -1145,7 +1145,7 @@
     WindowPtr		pWin, pOldWin;
     int			i;
 
-    printf("maxDrawableTableEntry = %d\n", pDRIPriv->pDriverInfo->maxDrawableTableEntry);
+    /* printf("maxDrawableTableEntry = %d\n", pDRIPriv->pDriverInfo->maxDrawableTableEntry); */
 
     if (pDrawable->type == DRAWABLE_WINDOW) {
 	pWin = (WindowPtr)pDrawable;
