diff -ruN qemu-kvm-0.11.1.orig/net.h qemu-kvm-0.11.1/net.h
--- qemu-kvm-0.11.1.orig/net.h	2010-05-25 16:40:07.000000000 -0400
+++ qemu-kvm-0.11.1/net.h	2010-05-25 16:40:28.000000000 -0400
@@ -164,8 +164,8 @@
 void net_host_device_add(Monitor *mon, const char *device, const char *opts);
 void net_host_device_remove(Monitor *mon, int vlan_id, const char *device);
 
-#define DEFAULT_NETWORK_SCRIPT "/etc/qemu-ifup"
-#define DEFAULT_NETWORK_DOWN_SCRIPT "/etc/qemu-ifdown"
+#define DEFAULT_NETWORK_SCRIPT "/etc/kvm/qemu-ifup"
+#define DEFAULT_NETWORK_DOWN_SCRIPT "/etc/kvm/qemu-ifdown"
 #ifdef __sun__
 #define SMBD_COMMAND "/usr/sfw/sbin/smbd"
 #else
