From 4c894a7a5cde4a5e8d8d3aa8c644030efdf77e3a Mon Sep 17 00:00:00 2001
From: Alexander Graf <agraf@suse.de>
Date: Fri, 14 Aug 2009 00:30:53 +0200
Subject: [PATCH 3/3] Enable protocol layering

---
 block.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

Index: qemu-kvm-0.11.0/block.c
===================================================================
--- qemu-kvm-0.11.0.orig/block.c
+++ qemu-kvm-0.11.0/block.c
@@ -358,11 +358,11 @@ int bdrv_open2(BlockDriverState *bs, con
             return ret;
         }
         total_size = bdrv_getlength(bs1) >> SECTOR_BITS;
+        bdrv_delete(bs1);
 
-        if (bs1->drv && bs1->drv->protocol_name)
+        if (find_protocol(filename) != bdrv_find_format("raw"))
             is_protocol = 1;
 
-        bdrv_delete(bs1);
 
         get_tmp_filename(tmp_filename, sizeof(tmp_filename));
 
