diff -ur sudo-1.6.7p5.orig/sudoers sudo-1.6.7p5/sudoers
--- sudo-1.6.7p5.orig/sudoers	2001-12-16 23:45:52.000000000 -0500
+++ sudo-1.6.7p5/sudoers	2005-02-26 15:15:09.340908520 -0500
@@ -17,7 +17,7 @@
 root	ALL=(ALL) ALL
 
 # Uncomment to allow people in group wheel to run all commands
-# %wheel	ALL=(ALL)	ALL
+%wheel	ALL=(ALL)	ALL
 
 # Same thing without a password
 # %wheel	ALL=(ALL)	NOPASSWD: ALL
