--- configure.in-ecb	2006-08-27 18:54:56.000000000 -0400
+++ configure.in	2008-04-08 22:32:57.000000000 -0400
@@ -115,8 +115,8 @@
 PILOT_LIBS="-lpisock"
 
 if test $pilotlibs = no ; then
-for pilot_libs in $pilot_prefix/lib /usr/lib /usr/local/lib/ \
-  /usr/extra/pilot/lib ; do
+for pilot_libs in $pilot_prefix/lib64 /usr/lib64 /usr/local/lib64/ \
+  /usr/extra/pilot/lib64 ; do
    if test -r $pilot_libs/libpisock.so >/dev/null 2>&1 ; then
       pilotlibs=yes
       PILOT_LIBS="-L$pilot_libs $PILOT_LIBS"
