diff -Naur Makefile.in.1 Makefile.in
--- Makefile.in.1	2003-07-11 16:00:57.000000000 +0200
+++ Makefile.in	2003-07-11 16:01:02.000000000 +0200
@@ -40,7 +40,7 @@
 FOOMATIC_RIP=@FOOMATIC_RIP@
 
 # foomatic-rip path for a "make inplace" installation
-LOCAL_FOOMATIC_RIP=@LOCAL_FOOMATIC_RIP@
+LOCAL_FOOMATIC_RIP=@FOOMATIC_RIP@
 
 # Locations for installing the components of this package
 BINDIR=$(bindir)
