--- builds/liferea/liferea-1.7.1/opml/feedlist.opml	2009-06-14 10:04:14.000000000 +0200
+++ builds/liferea/liferea-1.7.1/opml/feedlist-copy.opml	2009-09-30 20:58:48.000000000 +0200
@@ -12,6 +12,7 @@
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
 			<outline text="Liferea Blog" htmlUrl="http://liferea.blogspot.com/" xmlUrl="http://liferea.blogspot.com/feeds/posts/default" />
 			<outline text="mozillaZine" htmlUrl="http://www.mozillazine.org/" xmlUrl="http://www.mozillazine.org/atom.xml" />
+			<outline text="Planet Foresight" htmlUrl="http://planet.foresightlinux.org" xmlUrl="http://planet.foresightlinux.org/atom.xml" />  
 			<outline text="Planet GNOME" htmlUrl="http://planet.gnome.org/" xmlUrl="http://planet.gnome.org/atom.xml" />	
 			<outline text="Slashdot" htmlUrl="http://slashdot.org"  xmlUrl="http://slashdot.org/slashdot.rss" />
 			<outline text="TuxMobil" htmlUrl="http://tuxmobil.org" xmlUrl="http://tuxmobil.org/tuxmobil_rss.rdf" />      
