--- builds/liferea/liferea-1.7.1/opml/feedlist.opml-orig	2009-06-14 10:04:14.000000000 +0200
+++ builds/liferea/liferea-1.7.1/opml/feedlist.opml	2009-09-30 20:58:48.000000000 +0200
@@ -7,6 +7,7 @@
 	<!-- lets keep this default English block in sync over all feed lists! -->
 	<outline text="Example Feeds">		<!-- translate this title for localized feed list -->
 		<outline text="News">
+		        <outline text="Planet Foresight" htmlUrl="http://planet.foresightlinux.org" xmlUrl="http://planet.foresightlinux.org/atom.xml" />
 			<outline text="Ars Technica" htmlUrl="http://arstechnica.com" xmlUrl="http://feeds.arstechnica.com/arstechnica/index"/>
 			<outline text="Slashdot" htmlUrl="http://slashdot.org"  xmlUrl="http://slashdot.org/slashdot.rss" />
 			<outline text="BBC" htmlUrl="http://newsrss.bbc.co.uk/" xmlUrl="http://newsrss.bbc.co.uk/rss/newsonline_world_edition/front_page/rss.xml"/>

