--- gnome-cups-manager-0.24/libgnomecups/gnome-cups-permission.c.orig	2004-08-18 23:02:59.000000000 +0200
+++ gnome-cups-manager-0.24/libgnomecups/gnome-cups-permission.c	2004-09-06 16:38:16.372455992 +0200
@@ -28,12 +28,30 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <string.h>
+#include <grp.h>
 
 static char *gnome_cups_app_path = NULL;
 
 gboolean
 gnome_cups_can_admin ()
 {
+	gid_t groups[NGROUPS_MAX];
+	int numgrp, i;
+	struct group *lpadm_grp;
+
+	/* find group id of lpadmin */
+	lpadm_grp = getgrnam( "lpadmin" );
+
+	if( lpadm_grp ) {
+		/* find out whether user is in group lpadmin */
+		numgrp = getgroups( sizeof( groups ), groups );
+		if( numgrp >= 0 )
+			for( i = 0; i < numgrp; ++i ) {
+				if( groups[i] == lpadm_grp->gr_gid )
+					return 1;
+			}
+	}
+
 	return (geteuid () == 0);
 }
 
