diff -Nur gnome-cups-manager-0.30/gnome-cups-manager/gnome-cups-manager.glade gnome-cups-manager-0.30.new/gnome-cups-manager/gnome-cups-manager.glade
--- gnome-cups-manager-0.30/gnome-cups-manager/gnome-cups-manager.glade	2005-03-18 18:41:33.035487000 +0200
+++ gnome-cups-manager-0.30.new/gnome-cups-manager/gnome-cups-manager.glade	2005-03-18 18:41:55.684044448 +0200
@@ -18,6 +18,7 @@
   <property name="skip_pager_hint">False</property>
   <property name="type_hint">GDK_WINDOW_TYPE_HINT_NORMAL</property>
   <property name="gravity">GDK_GRAVITY_NORTH_WEST</property>
+  <property name="focus_on_map">True</property>
 
   <child>
     <widget class="GtkVBox" id="vbox13">
@@ -284,6 +285,7 @@
   <property name="skip_pager_hint">False</property>
   <property name="type_hint">GDK_WINDOW_TYPE_HINT_NORMAL</property>
   <property name="gravity">GDK_GRAVITY_NORTH_WEST</property>
+  <property name="focus_on_map">True</property>
 
   <child>
     <widget class="GtkVBox" id="vbox14">
@@ -331,8 +333,8 @@
 		  <property name="single_line_mode">False</property>
 		  <property name="angle">0</property>
 		  <accessibility>
-		    <atkrelation target="location_label" type="label-for"/>
 		    <atkrelation target="location_entry" type="label-for"/>
+		    <atkrelation target="location_label" type="label-for"/>
 		  </accessibility>
 		</widget>
 		<packing>
@@ -364,8 +366,8 @@
 		  <property name="single_line_mode">False</property>
 		  <property name="angle">0</property>
 		  <accessibility>
-		    <atkrelation target="description_label" type="label-for"/>
 		    <atkrelation target="description_entry" type="label-for"/>
+		    <atkrelation target="description_label" type="label-for"/>
 		  </accessibility>
 		</widget>
 		<packing>
@@ -1445,6 +1447,7 @@
   <property name="skip_pager_hint">False</property>
   <property name="type_hint">GDK_WINDOW_TYPE_HINT_NORMAL</property>
   <property name="gravity">GDK_GRAVITY_NORTH_WEST</property>
+  <property name="focus_on_map">True</property>
 
   <child>
     <widget class="GtkVBox" id="vbox19">
@@ -1472,7 +1475,7 @@
 		      <property name="use_underline">True</property>
 
 		      <child internal-child="image">
-			<widget class="GtkImage" id="image117">
+			<widget class="GtkImage" id="image132">
 			  <property name="visible">True</property>
 			  <property name="stock">gtk-add</property>
 			  <property name="icon_size">1</property>
@@ -1501,7 +1504,7 @@
 		      <accelerator key="Delete" modifiers="0" signal="activate"/>
 
 		      <child internal-child="image">
-			<widget class="GtkImage" id="image118">
+			<widget class="GtkImage" id="image133">
 			  <property name="visible">True</property>
 			  <property name="stock">gtk-remove</property>
 			  <property name="icon_size">1</property>
@@ -1547,7 +1550,7 @@
 		      <property name="use_underline">True</property>
 
 		      <child internal-child="image">
-			<widget class="GtkImage" id="image119">
+			<widget class="GtkImage" id="image134">
 			  <property name="visible">True</property>
 			  <property name="stock">gnome-stock-timer-stop</property>
 			  <property name="icon_size">1</property>
@@ -1567,7 +1570,7 @@
 		      <property name="use_underline">True</property>
 
 		      <child internal-child="image">
-			<widget class="GtkImage" id="image120">
+			<widget class="GtkImage" id="image135">
 			  <property name="visible">True</property>
 			  <property name="stock">gnome-stock-timer</property>
 			  <property name="icon_size">1</property>
@@ -1612,6 +1615,28 @@
 	      </child>
 	    </widget>
 	  </child>
+
+	  <child>
+	    <widget class="GtkMenuItem" id="global_settings1">
+	      <property name="visible">True</property>
+	      <property name="label" translatable="yes">Global Settings</property>
+	      <property name="use_underline">True</property>
+
+	      <child>
+		<widget class="GtkMenu" id="global_settings1_menu">
+
+		  <child>
+		    <widget class="GtkCheckMenuItem" id="detect_lan_printers1">
+		      <property name="visible">True</property>
+		      <property name="label" translatable="yes">Detect LAN Printers</property>
+		      <property name="use_underline">True</property>
+		      <property name="active">False</property>
+		    </widget>
+		  </child>
+		</widget>
+	      </child>
+	    </widget>
+	  </child>
 	</widget>
 	<packing>
 	  <property name="padding">0</property>
diff -Nur gnome-cups-manager-0.30/gnome-cups-manager/view-printers.c gnome-cups-manager-0.30.new/gnome-cups-manager/view-printers.c
--- gnome-cups-manager-0.30/gnome-cups-manager/view-printers.c	2005-03-18 18:41:33.036487000 +0200
+++ gnome-cups-manager-0.30.new/gnome-cups-manager/view-printers.c	2005-03-18 18:41:55.685044296 +0200
@@ -41,6 +41,11 @@
 #include <stdlib.h>
 #include <cups/cups.h>
 
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <gtk/gtk.h>
+#include <libgnome/gnome-i18n.h>
+
 static void
 printer_view_weak_notify (gpointer user_data, GObject *old_spot)
 {
@@ -49,6 +54,92 @@
 	*widget = NULL;
 }
 
+gint browsing_status()
+{
+	gchar *argv_stat[] = { "/usr/share/cups/browsing_status", NULL};
+     GSpawnFlags flags = G_SPAWN_FILE_AND_ARGV_ZERO;
+	GError *error=NULL;
+	gint spawn_exit_code, my_return_code;	
+	gboolean ok;
+
+	ok = g_spawn_sync (NULL, argv_stat, NULL, flags,NULL, NULL, NULL, NULL, &spawn_exit_code, &error);
+
+	if (ok)
+		my_return_code = WEXITSTATUS (spawn_exit_code);
+	else
+	{
+		my_return_code = -1;
+		fprintf (stderr, "Error occured spawning: %s\n", error->message);
+	}
+	return (my_return_code);
+}
+
+gint browsing_set(gboolean enable)
+{
+     char msg[1024];
+	char const *argv_enable[] = { "gksudo", msg, "/usr/share/cups/enable_browsing 1", NULL };
+	char const *argv_disable[] = { "gksudo", msg, "/usr/share/cups/enable_browsing 0", NULL };
+     GSpawnFlags flags = G_SPAWN_SEARCH_PATH;
+	GError *error=NULL;
+	gint spawn_exit_code, my_return_code;	
+	gboolean ok;
+	
+	if (enable)
+	{
+	     snprintf(msg , sizeof(msg), "-m %s", _("Enter your password to enable LAN printers detection."));
+	     ok = g_spawn_sync (NULL,(gchar**) argv_enable, NULL, flags,NULL, NULL, NULL, NULL, &spawn_exit_code, &error);
+	}
+	else
+	{
+	     snprintf(msg , sizeof(msg), "-m %s", _("Enter your password to disable LAN printers detection."));		   
+	     ok = g_spawn_sync (NULL, (gchar**) argv_disable, NULL, flags,NULL, NULL, NULL, NULL, &spawn_exit_code, &error);
+	}
+
+	if (ok)
+		my_return_code = WEXITSTATUS (spawn_exit_code);
+	else
+	{
+	     my_return_code = -1;
+	     fprintf (stderr, "Error occured spawning: %s\n", error->message);
+	}
+	return (my_return_code);
+}
+
+static void
+browsing_toggled_cb (GtkWidget *item, gpointer user_data)
+{
+	GladeXML *xml = GLADE_XML (user_data);
+	gboolean bstat, checkbox_state;
+	GtkWidget *dialog, *window;
+	item = glade_xml_get_widget (xml, "detect_lan_printers1");
+	window = glade_xml_get_widget (xml, "printer_window");
+	char *warning_msg =
+		  _("Enabling LAN printers detection will open port 631 on your system. This exposes you to the "
+		   "risk of exploiting any security vulnerabilities of the printing system. Remote attackers "
+		   "could potentially gain access to your system with the privileges of the \"cupsys\" user.");
+
+	bstat = browsing_status();
+	checkbox_state = gtk_check_menu_item_get_active (GTK_CHECK_MENU_ITEM (item));
+	
+	if (bstat != checkbox_state)
+	{
+		   if (checkbox_state)
+		   {
+				 dialog = gtk_message_dialog_new (GTK_WINDOW (window),
+										    GTK_DIALOG_MODAL,
+										    GTK_MESSAGE_QUESTION,
+										    GTK_BUTTONS_OK_CANCEL,
+										    "%s", warning_msg);
+				 gint result = gtk_dialog_run (GTK_DIALOG (dialog));
+				 gtk_widget_destroy (dialog);				 
+				 if (result == GTK_RESPONSE_OK) browsing_set(TRUE);
+		   }
+		   else
+				 browsing_set(FALSE);
+		   gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM (item), browsing_status());
+	}
+}
+
 static void
 close_cb (GtkWidget *item, 
 	  gpointer user_data)
@@ -201,6 +292,9 @@
 
 	item = glade_xml_get_widget (xml, "delete_item");
 	g_signal_connect (item, "activate", G_CALLBACK (delete_cb), xml);
+	
+	item = glade_xml_get_widget (xml, "detect_lan_printers1");
+	g_signal_connect (item, "activate", G_CALLBACK (browsing_toggled_cb), xml);	
 }
 
 static void
@@ -215,6 +309,7 @@
 	gboolean delete;
 	gboolean properties;
 	gboolean status;
+	gboolean bstat;	
 
 	list = GNOME_PRINTER_LIST (g_object_get_data (G_OBJECT (xml), "icon_list"));
 
@@ -244,6 +339,16 @@
 
 	item = glade_xml_get_widget (xml, "properties_item");
 	gtk_widget_set_sensitive (item, properties);
+
+	item = glade_xml_get_widget (xml, "detect_lan_printers1");
+	bstat = browsing_status();
+	if ((bstat == 2)||(bstat==-1))
+	{
+		   item = glade_xml_get_widget (xml, "global_settings1");
+		   gtk_widget_hide (GTK_WIDGET (item));
+	}
+	else
+		   gtk_check_menu_item_set_active  (GTK_CHECK_MENU_ITEM (item), bstat);		
 }
 
 static void
