diff -ur redland-1.0.9.orig/src/Makefile.in redland-1.0.9/src/Makefile.in
--- redland-1.0.9.orig/src/Makefile.in	2009-04-13 05:04:20.000000000 +0400
+++ redland-1.0.9/src/Makefile.in	2009-07-27 21:08:39.000000000 +0400
@@ -463,19 +463,19 @@
 @MODULAR_LIBRDF_TRUE@	$(am__append_6)
 @MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_SOURCES = rdf_storage_tstore.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_CPPFLAGS = @TSTORE_CPPFLAGS@
-@MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_LIBADD = @TSTORE_LIBS@
+@MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_LIBADD = @TSTORE_LIBS@ librdf.la
 @MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@librdf_storage_tstore_la_LDFLAGS = -module -avoid-version
 @MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_SOURCES = rdf_storage_mysql.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_CPPFLAGS = @MYSQL_CPPFLAGS@
-@MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_LIBADD = @MYSQL_LIBS@
+@MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_LIBADD = @MYSQL_LIBS@ librdf.la
 @MODULAR_LIBRDF_TRUE@@STORAGE_MYSQL_TRUE@librdf_storage_mysql_la_LDFLAGS = -module -avoid-version
 @MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_SOURCES = rdf_storage_sqlite.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_CPPFLAGS = @SQLITE_CPPFLAGS@
-@MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_LIBADD = @SQLITE_LIBS@
+@MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_LIBADD = @SQLITE_LIBS@ librdf.la
 @MODULAR_LIBRDF_TRUE@@STORAGE_SQLITE_TRUE@librdf_storage_sqlite_la_LDFLAGS = -module -avoid-version
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_SOURCES = rdf_storage_postgresql.c
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_CPPFLAGS = @POSTGRESQL_CPPFLAGS@
-@MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_LIBADD = @POSTGRESQL_LIBS@
+@MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_LIBADD = @POSTGRESQL_LIBS@ librdf.la
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_LDFLAGS = -module -avoid-version
 librdf_la_LIBADD = \
 @LTLIBOBJS@ @DIGEST_OBJS@ @HASH_OBJS@ \
