--- include/linux/kvm_host.h.orig	2009-10-27 17:15:36.620350835 +0000
+++ include/linux/kvm_host.h	2009-10-27 17:16:11.219351130 +0000
@@ -518,17 +518,12 @@
 	trace_mark(kvm_trace_##name, "%u %p %u %u %u %u %u %u", KVM_TRC_##evt, \
 						vcpu, 0, 0, 0, 0, 0, 0)
 
-#ifdef CONFIG_KVM_TRACE
-int kvm_trace_ioctl(unsigned int ioctl, unsigned long arg);
-void kvm_trace_cleanup(void);
-#else
 static inline
 int kvm_trace_ioctl(unsigned int ioctl, unsigned long arg)
 {
 	return -EINVAL;
 }
 #define kvm_trace_cleanup() ((void)0)
-#endif
 
 #ifdef KVM_ARCH_WANT_MMU_NOTIFIER
 static inline int mmu_notifier_retry(struct kvm_vcpu *vcpu, unsigned long mmu_seq)
