Exclude QtUiTools symbols from libplasma dynamic symbols table

Index: kde4libs-4.3.85/plasma/CMakeLists.txt
===================================================================
--- kde4libs-4.3.85.orig/plasma/CMakeLists.txt	2009-12-18 09:25:20.000000000 -0500
+++ kde4libs-4.3.85/plasma/CMakeLists.txt	2009-12-18 09:25:25.000000000 -0500
@@ -208,6 +208,10 @@
                              ${QT_QTUITOOLS_LIBRARY} ${QT_QTWEBKIT_LIBRARY}
                              kdnssd threadweaver ${KDE4_SOLID_LIBS} )
 
+# Do not export QtUiTools internal symbols
+set_target_properties(plasma PROPERTIES LINK_FLAGS_DEBIAN "-Wl,--exclude-libs -Wl,libQtUiTools.a")
+
+
 if(QCA2_FOUND)
    target_link_libraries(plasma ${QCA2_LIBRARIES})
 endif(QCA2_FOUND)
