--- dhcp-3.1.2/includes/dhcpd.h.orig	2009-03-06 16:17:19.000000000 -0500
+++ dhcp-3.1.2/includes/dhcpd.h	2009-03-06 16:17:20.000000000 -0500
@@ -1136,7 +1136,7 @@
 			  const unsigned char *, unsigned, struct universe *);
 int
 store_options(int *ocount,
-	      unsigned char *buffer, unsigned buflen, unsigned index,
+	      unsigned char *buffer, unsigned index, unsigned buflen,
 	      struct packet *packet, struct lease *lease,
 	      struct client_state *client_state,
 	      struct option_state *in_options,
