--- dhcp-3.0pl2/server/dhcpd.c.bak	2004-04-05 18:04:52.000000000 -0400
+++ dhcp-3.0pl2/server/dhcpd.c	2004-04-05 18:04:22.000000000 -0400
@@ -1066,10 +1066,10 @@
 
 static isc_result_t dhcp_io_shutdown_countdown (void *vlp)
 {
-	dhcp_failover_state_t *state;
 #if defined (FAILOVER_PROTOCOL)
-	int failover_connection_count = 0;
+	dhcp_failover_state_t *state;
 #endif
+	int failover_connection_count = 0;
 
       oncemore:
 	if (shutdown_state == shutdown_listeners ||
