--- dhcp-3.1.2/includes/dhcp.h.orig	2009-03-06 13:18:01.000000000 -0500
+++ dhcp-3.1.2/includes/dhcp.h	2009-03-06 13:18:23.000000000 -0500
@@ -45,6 +45,8 @@
 #define DHCP_MAX_OPTION_LEN	(DHCP_MTU_MAX - DHCP_FIXED_LEN)
 #define DHCP_MIN_OPTION_LEN     (DHCP_MTU_MIN - DHCP_FIXED_LEN)
 
+#define DHCP_OPTION_LEN         DHCP_MAX_OPTION_LEN
+
 struct dhcp_packet {
  u_int8_t  op;		/* 0: Message opcode/type */
 	u_int8_t  htype;	/* 1: Hardware addr type (net/if_types.h) */
