--- dhcp-3.0pl2/client/dhclient.c.div0	2003-09-05 16:13:24.060636975 -0400
+++ dhcp-3.0pl2/client/dhclient.c	2003-09-05 16:14:47.440120495 -0400
@@ -789,9 +789,10 @@ void dhcpack (packet)
 		client -> new -> renewal = TIME_MAX;
 
 	/* Now introduce some randomness to the renewal time: */
+	i = (client -> new -> renewal + 3) / 4;
+	if (i < 1) i = 1;
 	if (client -> new -> renewal <= TIME_MAX / 3 - 3)
 		client -> new -> renewal =
 				(((client -> new -> renewal + 3) * 3 / 4) +
-				    (random () % /* XXX NUMS */
-				     ((client -> new -> renewal + 3) / 4)));
+				    (random () % /* XXX NUMS */ i ));
 
 	/* Same deal with the rebind time. */
 	oc = lookup_option (&dhcp_universe, client -> new -> options,
