--- common/parse.c.orig	2008-10-08 17:14:25.000000000 +0300
+++ common/parse.c	2008-10-08 17:15:11.000000000 +0300
@@ -1041,7 +1041,7 @@ parse_option_name (cfile, allocate, know
 			option = new_option(val, MDL);
 			option->universe = universe;
 			option->code = code;
-			option->format = "X";
+			option->format = default_option_format;
 			option_reference(opt, option, MDL);
 		} else
 			log_info("option %s has been redefined as option %s.  "
