--- dhcp-3.0.5/includes/site.h.jau	2007-12-18 15:46:25.000000000 -0500
+++ dhcp-3.0.5/includes/site.h	2007-12-18 17:36:25.000000000 -0500
@@ -183,7 +183,5 @@
 
 #define LDAP_CONFIGURATION
 
-#define _PATH_DHCPD_DB    "/var/lib/dhcpd/dhcpd.leases"
-#define _PATH_DHCLIENT_DB "/var/lib/dhclient/dhclient.leases"
-#define _PATH_DHCPD_DB    "/var/lib/dhcpd/dhcpd.leases"
+#define _PATH_DHCPD_DB    "/var/lib/dhcp/dhcpd.leases"
 #define _PATH_DHCLIENT_DB "/var/lib/dhclient/dhclient.leases"
