--- configure.ac.orig	2008-07-18 21:03:30.000000000 +0300
+++ configure.ac	2008-07-18 21:07:04.000000000 +0300
@@ -288,7 +288,6 @@ AC_SEARCH_LIBS([socket],[socket])
 AC_SEARCH_LIBS([gethostbyname],[nsl])
 AC_SEARCH_LIBS([dlopen],[dl],[AC_DEFINE(HAVE_DLOPEN, 1, [have dlopen])])
 AC_SEARCH_LIBS([res_9_search],[resolv],[AC_DEFINE(HAVE_RES_9_SEARCH, 1, [have res_9_search])])
-AC_SEARCH_LIBS([res_search],[resolv bind],[AC_DEFINE(HAVE_RES_SEARCH, 1, [have res_search])])
 AC_CHECK_DECLS([res_search],,, [
      #include <stdio.h>
      #include <sys/types.h>
@@ -296,6 +295,16 @@ AC_CHECK_DECLS([res_search],,, [
      #include <arpa/nameser.h>
      #include <resolv.h>
 ])
+AC_CHECK_FUNC([res_search], [:],
+	[AC_MSG_CHECKING([for res_search])
+	 old_LIBS="$LIBS"
+	 LIBS="$LIBS -lresolv"
+	 AC_LINK_IFELSE([AC_LANG_PROGRAM(
+			   [[#include <resolv.h>]],
+			   [[res_search((void *)0,0,0,0,0);]])],
+			[AC_MSG_RESULT([found])],
+			[LIBS="$old_LIBS"
+			 AC_MSG_FAILURE([not found])])])
 
 AC_ARG_WITH(libresolv, AS_HELP_STRING([--without-libresolv], [don't use libresolv]),
       [with_libresolv=$withval], [with_libresolv=yes])
