From 506851c47e6cf415d71d45a5c67ca05984423717 Mon Sep 17 00:00:00 2001
From: Jannis Pohlmann <jannis@xfce.org>
Date: Sat, 19 Feb 2011 14:35:33 +0000
Subject: Fix crash when trying to thumbnail a file with unknown content type.

---
diff --git a/NEWS b/NEWS
index 38c4449..b324c95 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,7 @@
+1.2.x
+=====
+- Fix crash when trying to thumbnail a file with unknown content type.
+
 1.2.1
 =====
 - Paste files in correct order (bug #6504).
diff --git a/thunar/thunar-thumbnailer.c b/thunar/thunar-thumbnailer.c
index 8146d29..93127a0 100644
--- a/thunar/thunar-thumbnailer.c
+++ b/thunar/thunar-thumbnailer.c
@@ -480,6 +480,17 @@ thunar_thumbnailer_file_is_supported (ThunarThumbnailer *thumbnailer,
       return FALSE;
     }
 
+  /* determine the content type of the passed file */
+  content_type = thunar_file_get_content_type (file);
+
+  /* abort if the content type is unknown */
+  if (content_type == NULL)
+    {
+      /* release the thumbnailer lock */
+      g_mutex_unlock (thumbnailer->lock);
+      return FALSE;
+    }
+
   /* request the supported types on demand */
   if (thumbnailer->supported_schemes == NULL 
       || thumbnailer->supported_types == NULL)
@@ -496,9 +507,6 @@ thunar_thumbnailer_file_is_supported (ThunarThumbnailer *thumbnailer,
   if (thumbnailer->supported_schemes != NULL
       && thumbnailer->supported_types != NULL)
     {
-      /* determine the content type of the passed file */
-      content_type = thunar_file_get_content_type (file);
-
       /* go through all the URI schemes we support */
       for (n = 0; !supported && thumbnailer->supported_schemes[n] != NULL; ++n)
         {
--
cgit 
