From 9efd4b71e449406a90131b6a14b9bac318e204c1 Mon Sep 17 00:00:00 2001
From: Jannis Pohlmann <jannis@xfce.org>
Date: Fri, 20 May 2011 00:05:06 +0000
Subject: Treat backup files as hidden, as with thunar-vfs (bug #7615).

---
diff --git a/NEWS b/NEWS
index 6ccac5d..e7a470f 100644
--- a/NEWS
+++ b/NEWS
@@ -6,6 +6,7 @@
   Reported and solved by Ambroz Bizjak.
 - Fix crash when hovering a CD drive with a special icon (bug #7309).
   Reported and solved by Ambroz Bizjak.
+- Treat backup files as hidden as with thunar-vfs (bug #7615).
 
 1.2.1
 =====
diff --git a/thunar/thunar-file.c b/thunar/thunar-file.c
index 8caa422..d7f3e8e 100644
--- a/thunar/thunar-file.c
+++ b/thunar/thunar-file.c
@@ -2301,7 +2301,8 @@ thunar_file_is_hidden (const ThunarFile *file)
   if (file->info == NULL)
     return FALSE;
 
-  return g_file_info_get_is_hidden (file->info);
+  return g_file_info_get_is_hidden (file->info)
+         || g_file_info_get_is_backup (file->info);
 }
 
 
--
cgit 
