diff -r 5daa7016f981 applets/clock/clock.schemas.in
--- a/applets/clock/clock.schemas.in	Mon Dec 01 12:18:46 2008 -0500
+++ b/applets/clock/clock.schemas.in	Mon Dec 01 14:28:42 2008 -0500
@@ -18,7 +18,7 @@
                if you translate this to "24 sata" or anything else that isn't "24-hour" or
                "12-hour", things will not work.
               -->
-            24-hour
+            custom
           </default>
           <short>Hour format</short>
           <long>
@@ -36,7 +36,7 @@
           </long>
         </locale>
 	<locale name="en_US">
-	  <default>12-hour</default>
+	  <default>custom</default>
 	</locale>
       </schema>
 
@@ -44,7 +44,6 @@
         <key>/schemas/apps/clock_applet/prefs/custom_format</key>
         <owner>clock-applet</owner>
         <type>string</type>
-        <default></default>
         <locale name="C">
           <short>Custom format of the clock</short>
           <long>
@@ -53,7 +52,11 @@
 	    understood by strftime() to obtain a specific format. See the
 	    strftime() manual for more information.
           </long>
+        <default>&lt;span color=&quot;#999999&quot;&gt;%a %d %b&lt;/span&gt; &lt;b&gt;%I:%M&lt;/b&gt;</default>
         </locale>
+	<locale name="en_US">
+            <default>&lt;span color=&quot;#999999&quot;&gt;%a %d %b&lt;/span&gt; &lt;b&gt;%l:%M %P&lt;/b&gt;</default>
+	</locale>
       </schema>
 
       <schema>
