diff -r 5daa7016f981 applets/clock/clock.c
--- a/applets/clock/clock.c	Mon Dec 01 12:18:46 2008 -0500
+++ b/applets/clock/clock.c	Tue Dec 02 15:42:32 2008 -0500
@@ -419,9 +419,9 @@
 	char       *clock_format;
 
 	if (cd->format == CLOCK_FORMAT_12)
-		time_format = cd->showseconds ? _("%l:%M:%S %p") : _("%l:%M %p");
+		time_format = cd->showseconds ? _("<span size=\"larger\"><b>%l:%M:%S %P</b></span>") : _("<span size=\"larger\"><b>%l:%M %P</b></span>");
 	else
-		time_format = cd->showseconds ? _("%H:%M:%S") : _("%H:%M");
+		time_format = cd->showseconds ? _("<span size=\"larger\"><b>%H:%M:%S</b></span>") : _("<span size=\"larger\"><b>%H:%M</b></span>");
 
 	if (!safe) {
 		/* FIXME: we need this for bug #410169 in pango. We'll be able
@@ -441,7 +441,7 @@
 		 *              should begin with a 0 in your locale (e.g. "May
 		 *              01" instead of "May  1").
 		 */
-		date_format = _("%a %b %e");
+		date_format = _("<span color=\"#999999\">%a %d %b</span> ");
 
 		if (use_two_line_format (cd))
 			/* translators: reverse the order of these arguments
@@ -456,7 +456,7 @@
 			 *              if the time should come before the
 			 *              date on a clock in your locale.
 			 */
-			clock_format = g_strdup_printf (_("%1$s, %2$s"),
+			clock_format = g_strdup_printf (_("%1$s %2$s"),
 							date_format,
 							time_gravity_format);
 	}
