diff -Naur _ROOT_.orig/etc/gconf/schemas/panel-default-setup.entries _ROOT_/etc/gconf/schemas/panel-default-setup.entries
--- _ROOT_.orig/etc/gconf/schemas/panel-default-setup.entries	2007-08-09 23:20:14.000000000 -0400
+++ _ROOT_/etc/gconf/schemas/panel-default-setup.entries	2008-02-04 13:51:49.000000000 -0500
@@ -262,7 +262,7 @@
       <key>objects/browser_launcher/launcher_location</key>
       <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
       <value>
-        <string>epiphany.desktop</string>
+        <string>net-firefox.desktop</string>
       </value>
     </entry>
 
