diff -Naur gnome-panel-2.12.0.orig/gnome-panel/panel-default-setup.entries.in gnome-panel-2.12.0/gnome-panel/panel-default-setup.entries.in
--- gnome-panel-2.12.0.orig/gnome-panel/panel-default-setup.entries.in	2005-08-24 16:51:26.000000000 -0400
+++ gnome-panel-2.12.0/gnome-panel/panel-default-setup.entries.in	2005-09-15 15:14:07.000000000 -0400
@@ -252,7 +252,7 @@
       <key>objects/browser_launcher/launcher_location</key>
       <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
       <value>
-        <string>@DATADIR@/applications/epiphany.desktop</string>
+        <string>@DATADIR@/applications/net-firefox.desktop</string>
       </value>
     </entry>
 
@@ -306,7 +306,7 @@
       <key>objects/email_launcher/launcher_location</key>
       <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
       <value>
-        <string>@DATADIR@/applications/evolution.desktop</string>
+        <string>@DATADIR@/applications/evolution-2.4.desktop</string>
       </value>
     </entry>
 
