diff -Naur gnome-panel-2.12.2.orig/gnome-panel/panel-default-setup.entries.in gnome-panel-2.12.2/gnome-panel/panel-default-setup.entries.in
--- gnome-panel-2.12.2.orig/gnome-panel/panel-default-setup.entries.in	2005-08-24 16:51:26.000000000 -0400
+++ gnome-panel-2.12.2/gnome-panel/panel-default-setup.entries.in	2005-12-09 15:27:15.000000000 -0500
@@ -29,12 +29,6 @@
           <value>
             <string>menu_bar</string>
           </value>
-          <value>
-            <string>browser_launcher</string>
-          </value>
-          <value>
-            <string>email_launcher</string>
-          </value>
         </list>
       </value>
     </entry>
@@ -50,6 +44,9 @@
             <string>window_menu</string>
           </value>
           <value>
+            <string>quick-lounge-applet</string>
+          </value>
+          <value>
             <string>mixer</string>
           </value>
           <value>
@@ -211,113 +208,67 @@
     <entry><key>objects/menu_bar/launcher_location</key><schema_key>/schemas/apps/panel/objects/launcher_location</schema_key></entry>
     <entry><key>objects/menu_bar/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
 
-  <!-- Browser Launcher -->
+  <!-- Quick Lounge Applet -->
 
     <entry>
-      <key>objects/browser_launcher/object_type</key>
+      <key>applets/quick-lounge-applet/object_type</key>
       <schema_key>/schemas/apps/panel/objects/object_type</schema_key>
       <value>
-        <string>launcher-object</string>
+        <string>bonobo-applet</string>
       </value>
     </entry>
     <entry>
-      <key>objects/browser_launcher/toplevel_id</key>
+      <key>applets/quick-lounge-applet/toplevel_id</key>
       <schema_key>/schemas/apps/panel/objects/toplevel_id</schema_key>
       <value>
         <string>top_panel</string>
       </value>
     </entry>
     <entry>
-      <key>objects/browser_launcher/position</key>
+      <key>applets/quick-lounge-applet/position</key>
       <schema_key>/schemas/apps/panel/objects/position</schema_key>
       <value>
         <int>1</int>
       </value>
     </entry>
     <entry>
-      <key>objects/browser_launcher/panel_right_stick</key>
+      <key>applets/quick-lounge-applet/panel_right_stick</key>
       <schema_key>/schemas/apps/panel/objects/panel_right_stick</schema_key>
       <value>
         <bool>false</bool>
       </value>
     </entry>
     <entry>
-      <key>objects/browser_launcher/locked</key>
+      <key>applets/quick-lounge-applet/locked</key>
       <schema_key>/schemas/apps/panel/objects/locked</schema_key>
       <value>
         <bool>true</bool>
       </value>
     </entry>
     <entry>
-      <key>objects/browser_launcher/launcher_location</key>
-      <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
-      <value>
-        <string>@DATADIR@/applications/epiphany.desktop</string>
-      </value>
-    </entry>
-
-    <entry><key>objects/browser_launcher/bonobo_iid</key><schema_key>/schemas/apps/panel/objects/bonobo_iid</schema_key></entry>
-    <entry><key>objects/browser_launcher/attached_toplevel_id</key><schema_key>/schemas/apps/panel/objects/attached_toplevel_id</schema_key> </entry>
-    <entry><key>objects/browser_launcher/tooltip</key><schema_key>/schemas/apps/panel/objects/tooltip</schema_key></entry>
-    <entry><key>objects/browser_launcher/use_custom_icon</key><schema_key>/schemas/apps/panel/objects/use_custom_icon</schema_key></entry>
-    <entry><key>objects/browser_launcher/custom_icon</key><schema_key>/schemas/apps/panel/objects/custom_icon</schema_key></entry>
-    <entry><key>objects/browser_launcher/use_menu_path</key><schema_key>/schemas/apps/panel/objects/use_menu_path</schema_key></entry>
-    <entry><key>objects/browser_launcher/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
-    <entry><key>objects/browser_launcher/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
-
-  <!-- Email Launcher -->
-
-    <entry>
-      <key>objects/email_launcher/object_type</key>
-      <schema_key>/schemas/apps/panel/objects/object_type</schema_key>
-      <value>
-        <string>launcher-object</string>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/email_launcher/toplevel_id</key>
-      <schema_key>/schemas/apps/panel/objects/toplevel_id</schema_key>
+      <key>applet/quick-lounge-applet/prefs/min_size</key>
+      <schema_key>schemas/apps/quick-lounge-applet/prefs/min_size</schema_key>
       <value>
-        <string>top_panel</string>
+        <int>9</int>
       </value>
     </entry>
     <entry>
-      <key>objects/email_launcher/position</key>
-      <schema_key>/schemas/apps/panel/objects/position</schema_key>
-      <value>
-        <int>2</int>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/email_launcher/panel_right_stick</key>
-      <schema_key>/schemas/apps/panel/objects/panel_right_stick</schema_key>
-      <value>
-        <bool>false</bool>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/email_launcher/locked</key>
-      <schema_key>/schemas/apps/panel/objects/locked</schema_key>
-      <value>
-        <bool>true</bool>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/email_launcher/launcher_location</key>
-      <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
+      <key>applets/quick-lounge-applet/bonobo_iid</key>
+      <schema_key>/schemas/apps/panel/objects/bonobo_iid</schema_key>
       <value>
-        <string>@DATADIR@/applications/evolution.desktop</string>
+        <string>OAFIID:GNOME_QuickLoungeApplet</string>
       </value>
     </entry>
 
-    <entry><key>objects/email_launcher/bonobo_iid</key><schema_key>/schemas/apps/panel/objects/bonobo_iid</schema_key></entry>
-    <entry><key>objects/email_launcher/attached_toplevel_id</key><schema_key>/schemas/apps/panel/objects/attached_toplevel_id</schema_key> </entry>
-    <entry><key>objects/email_launcher/tooltip</key><schema_key>/schemas/apps/panel/objects/tooltip</schema_key></entry>
-    <entry><key>objects/email_launcher/use_custom_icon</key><schema_key>/schemas/apps/panel/objects/use_custom_icon</schema_key></entry>
-    <entry><key>objects/email_launcher/custom_icon</key><schema_key>/schemas/apps/panel/objects/custom_icon</schema_key></entry>
-    <entry><key>objects/email_launcher/use_menu_path</key><schema_key>/schemas/apps/panel/objects/use_menu_path</schema_key></entry>
-    <entry><key>objects/email_launcher/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
-    <entry><key>objects/email_launcher/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
+    <entry><key>applets/quick-lounge-applet/attached_toplevel_id</key><schema_key>/schemas/apps/panel/objects/attached_toplevel_id</schema_key> </entry>
+    <entry><key>applets/quick-lounge-applet/tooltip</key><schema_key>/schemas/apps/panel/objects/tooltip</schema_key></entry>
+    <entry><key>applets/quick-lounge-applet/use_custom_icon</key><schema_key>/schemas/apps/panel/objects/use_custom_icon</schema_key></entry>
+    <entry><key>applets/quick-lounge-applet/custom_icon</key><schema_key>/schemas/apps/panel/objects/custom_icon</schema_key></entry>
+    <entry><key>applets/quick-lounge-applet/use_menu_path</key><schema_key>/schemas/apps/panel/objects/use_menu_path</schema_key></entry>
+    <entry><key>applets/quick-lounge-applet/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
+    <entry><key>applets/quick-lounge-applet/launcher_location</key><schema_key>/schemas/apps/panel/objects/launcher_location</schema_key></entry>
+    <entry><key>applets/quick-lounge-applet/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
+
 
   <!-- Window Menu Applet -->
 
@@ -410,6 +361,33 @@
         <bool>true</bool>
       </value>
     </entry>
+
+<!-- Modification for sound stuff -->
+
+    <entry>
+      <key>applets/mixer/prefs/channel</key>
+      <schema_key>/schemas/apps/mixer_applet/prefs/channel</schema_key>
+      <value>
+        <int>0</int>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/mixer/prefs/mute</key>
+      <schema_key>/schemas/apps/mixer_applet/prefs/mute</schema_key>
+      <value>
+        <bool>false</bool>
+      </value>
+    </entry>
+    <entry>
+      <key>applets/mixer/prefs/vol</key>
+      <schema_key>/schemas/apps/mixer_applet/prefs/vol</schema_key>
+      <value>
+        <int>50</int>
+      </value>
+    </entry>
+
+<!-- End Modification for sound stuff -->
+
     <entry>
       <key>applets/mixer/bonobo_iid</key>
       <schema_key>/schemas/apps/panel/objects/bonobo_iid</schema_key>
