diff -r 47dbc7e88f53 gnomedesktop/gnome-desktop-sharp-2.0.pc.in
--- a/gnomedesktop/gnome-desktop-sharp-2.0.pc.in	Wed Jun 11 00:24:07 2008 -0400
+++ b/gnomedesktop/gnome-desktop-sharp-2.0.pc.in	Wed Jun 11 00:26:38 2008 -0400
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 gapidir=${prefix}/share/gnomedesktop-sharp/2.20
 assemblies_dir=${libdir}/mono/gnomedesktop-sharp-2.20
 
diff -r 47dbc7e88f53 gtkhtml/gtkhtml-sharp-3.14.pc.in
--- a/gtkhtml/gtkhtml-sharp-3.14.pc.in	Wed Jun 11 00:24:07 2008 -0400
+++ b/gtkhtml/gtkhtml-sharp-3.14.pc.in	Wed Jun 11 00:26:38 2008 -0400
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 gapidir=${prefix}/share/gtkhtml-sharp/3.14
 assemblies_dir=${libdir}/mono/gtkhtml-sharp-3.14
 
diff -r 47dbc7e88f53 nautilusburn/nautilusburn-sharp.pc.in
--- a/nautilusburn/nautilusburn-sharp.pc.in	Wed Jun 11 00:24:07 2008 -0400
+++ b/nautilusburn/nautilusburn-sharp.pc.in	Wed Jun 11 00:26:38 2008 -0400
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 gapidir=${prefix}/share/nautilusburn-sharp/2.20
 
 
diff -r 47dbc7e88f53 rsvg/rsvg2-sharp-2.0.pc.in
--- a/rsvg/rsvg2-sharp-2.0.pc.in	Wed Jun 11 00:24:07 2008 -0400
+++ b/rsvg/rsvg2-sharp-2.0.pc.in	Wed Jun 11 00:26:38 2008 -0400
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 gapidir=${prefix}/share/rsvg2-sharp/2.0
 
 
diff -r 47dbc7e88f53 vte/vte-sharp-0.16.pc.in
--- a/vte/vte-sharp-0.16.pc.in	Wed Jun 11 00:24:07 2008 -0400
+++ b/vte/vte-sharp-0.16.pc.in	Wed Jun 11 00:26:38 2008 -0400
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 gapidir=${prefix}/share/vte-sharp/0.16
 
 
diff -r 47dbc7e88f53 wnck/wnck-sharp-1.0.pc.in
--- a/wnck/wnck-sharp-1.0.pc.in	Wed Jun 11 00:24:07 2008 -0400
+++ b/wnck/wnck-sharp-1.0.pc.in	Wed Jun 11 00:26:38 2008 -0400
@@ -1,6 +1,6 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 gapidir=${prefix}/share/wnck-sharp/2.20
 assemblies_dir=${libdir}/mono/wnck-sharp-2.20
 
