diff -r baa9137832d8 backends/conary/conaryBackend.py
--- a/backends/conary/conaryBackend.py	Wed Apr 08 08:29:00 2009 -0400
+++ b/backends/conary/conaryBackend.py	Wed Apr 08 09:01:25 2009 -0400
@@ -51,6 +51,7 @@
 def ExceptionHandler(func):
     return func
     def display(error):
+        log.info(error)
         return str(error).replace('\n', ' ').replace("\t",'')
     def wrapper(self, *args, **kwargs):
         try:
@@ -104,18 +105,18 @@
     restartpkgs = ("PackageKit","gnome-packagekit")
 
     packages = []
-    #{{{ 
+    #{{{   Packages structure 
     """
     packages = {
         pkg_name: {
             'trove': ( name,version,flavor)
             'metadata': pkgDict,
-            'status' : status
         }
     }
     
     """
     #}}}
+    #{{{ Init
     def __init__(self, args):
         PackageKitBaseBackend.__init__(self, args)
 
@@ -136,7 +137,7 @@
         version = versions.ThawVersion(frzVersion)
         flavor = deps.ThawFlavor(frzFlavor)
         return version, flavor
-
+    #}}}
     def _get_arch(self, flavor):
         return _get_arch(flavor)
  
@@ -422,6 +423,29 @@
         log.info("options: %s searchlist:%s "%(options, searchlist))
         self._do_search(options, searchlist, 'group')
 
+    @ExceptionHandler
+    def search_file(self, filters, search ):
+
+        log.info("============= search_file ========")
+        self.allow_cancel(True)
+        self.percentage(0)
+        self.status(STATUS_QUERY)
+        log.info("options: %s searchlist:%s "%(filters, search))
+        self.percentage(10)
+
+
+        self.percentage(20)
+      
+
+        self.percentage(30)
+        name = self.conary.search_path( search )
+        self.percentage(50)
+        log.info(name)
+        if name:
+            log.info("resolving")
+            if ":" in name:
+                name = name.split(":")[0]
+            self.resolve( filters, [name]) 
 
     @ExceptionHandler
     def search_name(self, options, searchlist):
@@ -520,7 +544,7 @@
         applyList = [ (x[0], (None, None), x[1:], True) for x in updateItems ]
 
         log.info(">>>>>>>>>> get update >>>>>>>>>>>>")
-        #self._get_update(applyList)
+        self._get_update(applyList)
         log.info(">>>>>>>>>> DO Update >>>>>>>>>>>>")
         jobs = self._do_update(applyList)
         log.info(">>>>>>>>>>END DO Update >>>>>>>>>>>>")
@@ -536,24 +560,6 @@
         self.status(STATUS_REFRESH_CACHE)
         cache = Cache()
         cache.refresh()
-    """
-    @ExceptionHandler
-    def update(self, package_ids):
-        '''
-        Implement the {backend}-update functionality
-        '''
-        self.allow_cancel(True)
-        self.percentage(0)
-        self.status(STATUS_RUNNING)
-        
-        for package in package_ids.split(" "):
-            name, version, flavor, installed = self._findPackage(package)
-            if name:
-                cli = ConaryPk()
-                cli.update(name)
-            else:
-                self.error(ERROR_PACKAGE_ALREADY_INSTALLED, 'No available updates')
-    """
 
     def install_packages(self, package_ids):
         """
@@ -587,6 +593,7 @@
                 log.info(">>> end Prepare Update")
                 self._do_package_update(name, version, flavor)
 
+        
     @ExceptionHandler
     def remove_packages(self, allowDeps, package_ids):
         '''
@@ -809,6 +816,15 @@
                 if lic[1:][0].lower() == j.lower():
                     return j
         return ""
+    def _upgrade_from_branch( self, branch):
+        branchList = branch.split("@")
+        if "2-qa" in branchList[1]:
+            return DISTRO_UPGRADE_TESTING
+        elif "2-devel" in branchList[1]:
+            return DISTRO_UPGRADE_UNSTABLE
+        else:
+            return DISTRO_UPGRADE_STABLE
+
 
     def _get_branch(self, branch ):
         branchList = branch.split("@")
diff -r baa9137832d8 backends/conary/conarypk.py
--- a/backends/conary/conarypk.py	Wed Apr 08 08:29:00 2009 -0400
+++ b/backends/conary/conarypk.py	Wed Apr 08 09:01:25 2009 -0400
@@ -52,6 +52,18 @@
             if "foresight.rpath.org" or "conary.rpath.com" in i.asString():
                 labels.append(i.asString())
         return labels
+
+    def search_path(self,path_file ):
+        labels = self.get_labels_from_config()
+        where = self._get_repos()
+        from conary.deps.deps import compatibleFlavors
+        for label in self.default_label:
+            trove = where.getTroveLeavesByPath([path_file], label)
+            if trove.get(path_file):
+                for ( name,version,flavor) in trove[path_file]:
+                    return name
+                
+
 
     def query(self, name):
         """ do a conary query """
@@ -109,7 +121,7 @@
 
 if __name__ == "__main__":
     conary = ConaryPk()
-    print conary.query("vim-scripts")
+    print conary.search_path("/usr/bin/vim")
     #print conary.query("gimpasdas")
     #print conary.request_query("dpaster",'zodyrepo.rpath.org@rpl:devel')
     #print conary.request_query("gimp")
diff -r baa9137832d8 backends/conary/pk-backend-conary.c
--- a/backends/conary/pk-backend-conary.c	Wed Apr 08 08:29:00 2009 -0400
+++ b/backends/conary/pk-backend-conary.c	Wed Apr 08 09:01:25 2009 -0400
@@ -236,6 +236,7 @@
 /**
     pk_backend_search_groups
 */
+/**
 static void
 backend_search_group (PkBackend *backend, PkBitfield filters, const gchar *search)
 {
@@ -244,7 +245,7 @@
 	pk_backend_spawn_helper (spawn, "conaryBackend.py", "search-group", filters_text, search, NULL);
 	g_free (filters_text);
 }
-
+*/
 
 
 /**
@@ -327,8 +328,27 @@
     pk_backend_spawn_helper (spawn, "conaryBackend.py", "get-packages", filters_text, NULL);
     g_free (filters_text);
 }
+/**
+ * pk_backend_search_file:
+ */
+static void
+backend_search_file (PkBackend *backend, PkBitfield filters, const gchar *search)
+{
+        gchar *filters_text;
+        filters_text = pk_filter_bitfield_to_text (filters);
+        pk_backend_spawn_helper (spawn, "conaryBackend.py", "search-file", filters_text, search, NULL);
+        g_free (filters_text);
+}
 
+/**
+ * backend_get_distro_upgrades:
+static void
+backend_get_distro_upgrades (PkBackend *backend)
+{
+    pk_backend_spawn_helper (spawn, "conaryBackend.py", "get-distro-upgrades", NULL);
+}
 
+ */
 
 PK_BACKEND_OPTIONS (
 	"Conary with XMLCache",				/* description */
@@ -344,7 +364,7 @@
 	NULL,					/* get_categories */
 	NULL,					/* get_depends */
 	backend_get_details,			/* get_details */
-	NULL,					/* get_distro_upgrades */
+	NULL,	                /* get_distro_upgrades */
 	backend_get_files,			/* get_files */
 	backend_get_packages,					/* get_packages */
 	NULL,					/* get_repo_list */
@@ -361,8 +381,8 @@
 	backend_resolve,			/* resolve */
 	NULL,					/* rollback */
 	backend_search_details,					/* search_details */
-	NULL,					/* search_file */
-	backend_search_group,					/* search_group */
+	backend_search_file,					/* search_file */
+	NULL,					/* search_group */
 	backend_search_name,			/* search_name */
 	backend_update_packages,		/* update_packages */
 	backend_update_system,			/* update_system */
diff -r baa9137832d8 configure.ac
--- a/configure.ac	Wed Apr 08 08:29:00 2009 -0400
+++ b/configure.ac	Wed Apr 08 09:01:25 2009 -0400
@@ -1,6 +1,6 @@
 AC_PREREQ(2.52)
 
-AC_INIT(PackageKit, 0.4.6)
+AC_INIT(PackageKit, 0.4.7)
 AC_CONFIG_SRCDIR(src)
 AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
 AM_CONFIG_HEADER(config.h)
