diff --git a/backends/conary/pk-backend-conary.c b/backends/conary/pk-backend-conary.c
index 485f61d..261bd25 100644
--- a/backends/conary/pk-backend-conary.c
+++ b/backends/conary/pk-backend-conary.c
@@ -348,6 +348,25 @@ backend_get_repo_list (PkBackend *backend, PkBitfield filters)
     pk_backend_spawn_helper (spawn, "conaryBackend.py", "get-repo-list", filters_text, NULL);
     g_free (filters_text);
 }
+static void
+backend_simulate_install_packages (PkBackend *backend, gchar **package_ids)
+{
+
+    pk_backend_finished (backend);
+}
+static void
+backend_simulate_remove_packages (PkBackend *backend, gchar **package_ids)
+{
+
+    pk_backend_finished (backend);
+}
+
+static void
+backend_simulate_update (PkBackend *backend, gchar **package_ids)
+{
+
+    pk_backend_finished (backend);
+}
 
 
 PK_BACKEND_OPTIONS (
@@ -389,8 +408,8 @@ PK_BACKEND_OPTIONS (
 	backend_update_system,			/* update_system */
 	NULL,					/* what_provides */
 	NULL,					/* simulate_install_files */
-	NULL,					/* simulate_install_packages */
-	NULL,					/* simulate_remove_packages */
-	NULL					/* simulate_update_packages */
+	backend_simulate_install_packages,					/* simulate_install_packages */
+	backend_simulate_remove_packages,					/* simulate_remove_packages */
+	backend_simulate_update					/* simulate_update_packages */
 );
 
diff --git a/config.h b/config.h
index 0d2e1ef..228f1e3 100644
--- a/config.h
+++ b/config.h
@@ -2,7 +2,7 @@
 /* config.h.in.  Generated from configure.ac by autoheader.  */
 
 /* default backend prefix */
-#define DEFAULT_BACKEND "yum"
+#define DEFAULT_BACKEND "conary"
 
 /* Build test code */
 #define EGG_BUILD_TESTS 1
@@ -14,7 +14,7 @@
 #define GETTEXT_PACKAGE "PackageKit"
 
 /* Define to 1 if you have the <archive.h> header file. */
-#define HAVE_ARCHIVE_H 1
+/* #undef HAVE_ARCHIVE_H */
 
 /* Define to 1 if you have the `bind_textdomain_codeset' function. */
 #define HAVE_BIND_TEXTDOMAIN_CODESET 1
