diff -Naur PackageKit-0.1.5.orig/backends/conary/helpers/conaryBackend.py PackageKit-0.1.5/backends/conary/helpers/conaryBackend.py
--- PackageKit-0.1.5.orig/backends/conary/helpers/conaryBackend.py	2007-12-29 22:27:19.000000000 -0500
+++ PackageKit-0.1.5/backends/conary/helpers/conaryBackend.py	2007-12-21 02:30:54.000000000 -0500
@@ -37,7 +37,7 @@
             self.error(ERROR_DEP_RESOLUTION_FAILED, display(e), exit=True)
         except conaryclient.UpdateError, e:
             # FIXME: Need a enum for UpdateError
-            self.error(ERROR_UNKNOWN, display(e), exit=True)
+            self.error(ERROR_UNKOWN, display(e), exit=True)
         except Exception, e:
             self.error(ERROR_UNKNOWN, display(e), exit=True)
     return wrapper
