From cc6b4e671a490c9c29401a1123ad0c55b878f592 Mon Sep 17 00:00:00 2001
From: Ambroz Bizjak <ambrop7@gmail.com>
Date: Mon, 28 Feb 2011 22:50:44 +0000
Subject: Fix crash if save timeout for a client is triggered (bug #7320).

---
diff --git a/xfce4-session/xfsm-manager.c b/xfce4-session/xfsm-manager.c
index 07c5369..0c9ba05 100644
--- a/xfce4-session/xfsm-manager.c
+++ b/xfce4-session/xfsm-manager.c
@@ -1571,11 +1571,11 @@ xfsm_manager_save_timeout (gpointer user_data)
                 "   Client will be disconnected now.\n\n",
                 xfsm_client_get_id (stdata->client));
 
-  xfsm_manager_close_connection (stdata->manager, stdata->client, TRUE);
-
   /* returning FALSE below will free the data */
   g_object_steal_data (G_OBJECT (stdata->client), "--save-timeout-id");
 
+  xfsm_manager_close_connection (stdata->manager, stdata->client, TRUE);
+
   return FALSE;
 }
 
--
cgit v0.8.3.4
