diff -r e697752393b6 Demo/comparisons/regextest.py
--- a/Demo/comparisons/regextest.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/comparisons/regextest.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # 1)  Regular Expressions Test
 #
diff -r e697752393b6 Demo/comparisons/sortingtest.py
--- a/Demo/comparisons/sortingtest.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/comparisons/sortingtest.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # 2)  Sorting Test
 #
diff -r e697752393b6 Demo/comparisons/systemtest.py
--- a/Demo/comparisons/systemtest.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/comparisons/systemtest.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # 3)  System Test
 #
diff -r e697752393b6 Demo/curses/life.py
--- a/Demo/curses/life.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/curses/life.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # life.py -- A curses-based version of Conway's Game of Life.
 # Contributed by AMK
 #
diff -r e697752393b6 Demo/curses/ncurses.py
--- a/Demo/curses/ncurses.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/curses/ncurses.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # $Id: ncurses.py 66424 2008-09-13 01:22:08Z andrew.kuchling $
 #
diff -r e697752393b6 Demo/curses/rain.py
--- a/Demo/curses/rain.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/curses/rain.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # $Id: rain.py 46625 2006-06-03 23:02:15Z andrew.kuchling $
 #
diff -r e697752393b6 Demo/curses/repeat.py
--- a/Demo/curses/repeat.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/curses/repeat.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """repeat <shell-command>
 
diff -r e697752393b6 Demo/curses/tclock.py
--- a/Demo/curses/tclock.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/curses/tclock.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # $Id: tclock.py 46626 2006-06-03 23:07:21Z andrew.kuchling $
 #
diff -r e697752393b6 Demo/parser/test_parser.py
--- a/Demo/parser/test_parser.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/parser/test_parser.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 #  (Force the script to use the latest build.)
 #
 #  test_parser.py
diff -r e697752393b6 Demo/pdist/RCSProxy.py
--- a/Demo/pdist/RCSProxy.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/pdist/RCSProxy.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """RCS Proxy.
 
diff -r e697752393b6 Demo/pdist/makechangelog.py
--- a/Demo/pdist/makechangelog.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/pdist/makechangelog.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Turn a pile of RCS log output into ChangeLog file entries.
 
diff -r e697752393b6 Demo/pdist/rcvs.py
--- a/Demo/pdist/rcvs.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/pdist/rcvs.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Remote CVS -- command line interface"""
 
diff -r e697752393b6 Demo/pdist/rrcs.py
--- a/Demo/pdist/rrcs.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/pdist/rrcs.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 "Remote RCS -- command line interface"
 
diff -r e697752393b6 Demo/pysvr/pysvr.py
--- a/Demo/pysvr/pysvr.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/pysvr/pysvr.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """A multi-threaded telnet-like server that gives a Python prompt.
 
diff -r e697752393b6 Demo/scripts/beer.py
--- a/Demo/scripts/beer.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/beer.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 # By GvR, demystified after a version by Fredrik Lundh.
 import sys
 n = 100
diff -r e697752393b6 Demo/scripts/eqfix.py
--- a/Demo/scripts/eqfix.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/eqfix.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Fix Python source files to use the new equality test operator, i.e.,
 #       if x = y: ...
diff -r e697752393b6 Demo/scripts/fact.py
--- a/Demo/scripts/fact.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/fact.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Factorize numbers.
 # The algorithm is not efficient, but easy to understand.
diff -r e697752393b6 Demo/scripts/find-uname.py
--- a/Demo/scripts/find-uname.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/find-uname.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """
 For each argument on the command line, look for it in the set of all Unicode
diff -r e697752393b6 Demo/scripts/from.py
--- a/Demo/scripts/from.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/from.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Print From and Subject of messages in $MAIL.
 # Extension to multiple mailboxes and other bells & whistles are left
diff -r e697752393b6 Demo/scripts/ftpstats.py
--- a/Demo/scripts/ftpstats.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/ftpstats.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Extract statistics from ftp daemon log.
 
diff -r e697752393b6 Demo/scripts/lpwatch.py
--- a/Demo/scripts/lpwatch.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/lpwatch.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Watch line printer queue(s).
 # Intended for BSD 4.3 lpq.
diff -r e697752393b6 Demo/scripts/makedir.py
--- a/Demo/scripts/makedir.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/makedir.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Like mkdir, but also make intermediate directories if necessary.
 # It is not an error if the given directory already exists (as long
diff -r e697752393b6 Demo/scripts/markov.py
--- a/Demo/scripts/markov.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/markov.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 class Markov:
     def __init__(self, histsize, choice):
diff -r e697752393b6 Demo/scripts/mboxconvert.py
--- a/Demo/scripts/mboxconvert.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/mboxconvert.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Convert  MH directories (1 message per file) or MMDF mailboxes (4x^A
 # delimited) to unix mailbox (From ... delimited) on stdout.
diff -r e697752393b6 Demo/scripts/mkrcs.py
--- a/Demo/scripts/mkrcs.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/mkrcs.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # A rather specialized script to make sure that a symbolic link named
 # RCS exists pointing to a real RCS directory in a parallel tree
diff -r e697752393b6 Demo/scripts/newslist.py
--- a/Demo/scripts/newslist.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/newslist.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 #######################################################################
 # Newslist  $Revision: 66429 $
 #
diff -r e697752393b6 Demo/scripts/pi.py
--- a/Demo/scripts/pi.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/pi.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Print digits of pi forever.
 #
diff -r e697752393b6 Demo/scripts/pp.py
--- a/Demo/scripts/pp.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/pp.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Emulate some Perl command line options.
 # Usage: pp [-a] [-c] [-d] [-e scriptline] [-F fieldsep] [-n] [-p] [file] ...
diff -r e697752393b6 Demo/scripts/primes.py
--- a/Demo/scripts/primes.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/primes.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Print prime numbers in a given range
 
diff -r e697752393b6 Demo/scripts/queens.py
--- a/Demo/scripts/queens.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/queens.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """N queens problem.
 
diff -r e697752393b6 Demo/scripts/script.py
--- a/Demo/scripts/script.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/script.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 # script.py -- Make typescript of terminal session.
 # Usage:
 #       -a      Append to typescript.
diff -r e697752393b6 Demo/scripts/unbirthday.py
--- a/Demo/scripts/unbirthday.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/unbirthday.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Calculate your unbirthday count (see Alice in Wonderland).
 # This is defined as the number of days from your birth until today
diff -r e697752393b6 Demo/scripts/update.py
--- a/Demo/scripts/update.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/scripts/update.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Update a bunch of files according to a script.
 # The input file contains lines of the form <filename>:<lineno>:<text>,
diff -r e697752393b6 Demo/sockets/echosvr.py
--- a/Demo/sockets/echosvr.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/sockets/echosvr.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Python implementation of an 'echo' tcp server: echo all data it receives.
 #
diff -r e697752393b6 Demo/sockets/finger.py
--- a/Demo/sockets/finger.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/sockets/finger.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Python interface to the Internet finger daemon.
 #
diff -r e697752393b6 Demo/sockets/gopher.py
--- a/Demo/sockets/gopher.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/sockets/gopher.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # A simple gopher client.
 #
diff -r e697752393b6 Demo/sockets/telnet.py
--- a/Demo/sockets/telnet.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/sockets/telnet.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Minimal interface to the Internet telnet protocol.
 #
diff -r e697752393b6 Demo/sockets/throughput.py
--- a/Demo/sockets/throughput.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/sockets/throughput.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Test network throughput.
 #
diff -r e697752393b6 Demo/sockets/udpecho.py
--- a/Demo/sockets/udpecho.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/sockets/udpecho.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Client and server for udp (datagram) echo.
 #
diff -r e697752393b6 Demo/tkinter/guido/MimeViewer.py
--- a/Demo/tkinter/guido/MimeViewer.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/MimeViewer.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # View a single MIME multipart message.
 # Display each part as a box.
diff -r e697752393b6 Demo/tkinter/guido/canvasevents.py
--- a/Demo/tkinter/guido/canvasevents.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/canvasevents.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 from Tkinter import *
 from Canvas import Oval, Group, CanvasText
diff -r e697752393b6 Demo/tkinter/guido/dialog.py
--- a/Demo/tkinter/guido/dialog.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/dialog.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # A Python function that generates dialog boxes with a text message,
 # optional bitmap, and any number of buttons.
diff -r e697752393b6 Demo/tkinter/guido/electrons.py
--- a/Demo/tkinter/guido/electrons.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/electrons.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Simulate "electrons" migrating across the screen.
 # An optional bitmap file in can be in the background.
diff -r e697752393b6 Demo/tkinter/guido/kill.py
--- a/Demo/tkinter/guido/kill.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/kill.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 # Tkinter interface to Linux `kill' command.
 
 from Tkinter import *
diff -r e697752393b6 Demo/tkinter/guido/mbox.py
--- a/Demo/tkinter/guido/mbox.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/mbox.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Scan MH folder, display results in window
 
diff -r e697752393b6 Demo/tkinter/guido/newmenubardemo.py
--- a/Demo/tkinter/guido/newmenubardemo.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/newmenubardemo.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Play with the new Tk 8.0 toplevel menu option."""
 
diff -r e697752393b6 Demo/tkinter/guido/rmt.py
--- a/Demo/tkinter/guido/rmt.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/rmt.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # A Python program implementing rmt, an application for remotely
 # controlling other Tk applications.
diff -r e697752393b6 Demo/tkinter/guido/solitaire.py
--- a/Demo/tkinter/guido/solitaire.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/solitaire.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Solitaire game, much like the one that comes with MS Windows.
 
diff -r e697752393b6 Demo/tkinter/guido/sortvisu.py
--- a/Demo/tkinter/guido/sortvisu.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/sortvisu.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Sorting algorithms visualizer using Tkinter.
 
diff -r e697752393b6 Demo/tkinter/guido/svkill.py
--- a/Demo/tkinter/guido/svkill.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/svkill.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Tkinter interface to SYSV `ps' and `kill' commands.
 
diff -r e697752393b6 Demo/tkinter/guido/tkman.py
--- a/Demo/tkinter/guido/tkman.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/tkinter/guido/tkman.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Tk man page browser -- currently only shows the Tcl/Tk man pages
 
diff -r e697752393b6 Demo/zlib/minigzip.py
--- a/Demo/zlib/minigzip.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/zlib/minigzip.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # Demo program for zlib; it compresses or decompresses files, but *doesn't*
 # delete the original.  This doesn't support all of gzip's options.
 #
diff -r e697752393b6 Demo/zlib/zlibdemo.py
--- a/Demo/zlib/zlibdemo.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Demo/zlib/zlibdemo.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 # Takes an optional filename, defaulting to this file itself.
 # Reads the file and compresses the content using level 1 and level 9
diff -r e697752393b6 Doc/includes/email-dir.py
--- a/Doc/includes/email-dir.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Doc/includes/email-dir.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """Send the contents of a directory as a MIME message."""
 
diff -r e697752393b6 Doc/includes/email-unpack.py
--- a/Doc/includes/email-unpack.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Doc/includes/email-unpack.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """Unpack a MIME message into a directory of files."""
 
diff -r e697752393b6 Lib/Cookie.py
--- a/Lib/Cookie.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/Cookie.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 
 ####
diff -r e697752393b6 Lib/UserString.py
--- a/Lib/UserString.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/UserString.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 ## vim:ts=4:et:nowrap
 """A user-defined wrapper around string objects
 
diff -r e697752393b6 Lib/base64.py
--- a/Lib/base64.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/base64.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """RFC 3548: Base16, Base32, Base64 Data Encodings"""
 
diff -r e697752393b6 Lib/bsddb/dbshelve.py
--- a/Lib/bsddb/dbshelve.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/bsddb/dbshelve.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/bin/env python
+#!/bin/env python2.6
 #------------------------------------------------------------------------
 #           Copyright (c) 1997-2001 by Total Control Software
 #                         All Rights Reserved
diff -r e697752393b6 Lib/bsddb/test/test_dbtables.py
--- a/Lib/bsddb/test/test_dbtables.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/bsddb/test/test_dbtables.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 #-----------------------------------------------------------------------
 # A test suite for the table interface built on bsddb.db
diff -r e697752393b6 Lib/cProfile.py
--- a/Lib/cProfile.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/cProfile.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Python interface for the 'lsprof' profiler.
    Compatible with the 'profile' module.
diff -r e697752393b6 Lib/cgi.py
--- a/Lib/cgi.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/cgi.py	Thu Oct 02 12:07:55 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/python2.6
 
 # NOTE: the above "/usr/local/bin/python" is NOT a mistake.  It is
 # intentionally NOT "/usr/bin/env python".  On many systems
diff -r e697752393b6 Lib/difflib.py
--- a/Lib/difflib.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/difflib.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """
 Module difflib -- helpers for computing deltas between objects.
diff -r e697752393b6 Lib/encodings/rot_13.py
--- a/Lib/encodings/rot_13.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/encodings/rot_13.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 """ Python Character Mapping Codec for ROT13.
 
     See http://ucsub.colorado.edu/~kominek/rot13/ for details.
diff -r e697752393b6 Lib/idlelib/PyShell.py
--- a/Lib/idlelib/PyShell.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/idlelib/PyShell.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 import os
 import os.path
diff -r e697752393b6 Lib/keyword.py
--- a/Lib/keyword.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/keyword.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Keywords (from "graminit.c")
 
diff -r e697752393b6 Lib/lib2to3/pgen2/token.py
--- a/Lib/lib2to3/pgen2/token.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/lib2to3/pgen2/token.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Token constants (from "token.h")."""
 
diff -r e697752393b6 Lib/lib2to3/refactor.py
--- a/Lib/lib2to3/refactor.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/lib2to3/refactor.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.5
+#!/usr/bin/env python2.6
 # Copyright 2006 Google, Inc. All Rights Reserved.
 # Licensed to PSF under a Contributor Agreement.
 
diff -r e697752393b6 Lib/lib2to3/tests/benchmark.py
--- a/Lib/lib2to3/tests/benchmark.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/lib2to3/tests/benchmark.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.5
+#!/usr/bin/env python2.6
 """
 This is a benchmarking script to test the speed of 2to3's pattern matching
 system. It's equivalent to "refactor.py -f all" for every Python module
diff -r e697752393b6 Lib/lib2to3/tests/pytree_idempotency.py
--- a/Lib/lib2to3/tests/pytree_idempotency.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/lib2to3/tests/pytree_idempotency.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.5
+#!/usr/bin/env python2.6
 # Copyright 2006 Google, Inc. All Rights Reserved.
 # Licensed to PSF under a Contributor Agreement.
 
diff -r e697752393b6 Lib/lib2to3/tests/test_all_fixers.py
--- a/Lib/lib2to3/tests/test_all_fixers.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/lib2to3/tests/test_all_fixers.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.5
+#!/usr/bin/env python2.6
 """Tests that run all fixer modules over an input stream.
 
 This has been broken out into its own test module because of its
diff -r e697752393b6 Lib/lib2to3/tests/test_fixers.py
--- a/Lib/lib2to3/tests/test_fixers.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/lib2to3/tests/test_fixers.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.5
+#!/usr/bin/env python2.6
 """ Test suite for the fixer modules """
 # Author: Collin Winter
 
diff -r e697752393b6 Lib/lib2to3/tests/test_parser.py
--- a/Lib/lib2to3/tests/test_parser.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/lib2to3/tests/test_parser.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.5
+#!/usr/bin/env python2.6
 """Test suite for 2to3's parser and grammar files.
 
 This is the place to add tests for changes to 2to3's grammar, such as those
diff -r e697752393b6 Lib/lib2to3/tests/test_pytree.py
--- a/Lib/lib2to3/tests/test_pytree.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/lib2to3/tests/test_pytree.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.5
+#!/usr/bin/env python2.6
 # Copyright 2006 Google, Inc. All Rights Reserved.
 # Licensed to PSF under a Contributor Agreement.
 
diff -r e697752393b6 Lib/lib2to3/tests/test_util.py
--- a/Lib/lib2to3/tests/test_util.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/lib2to3/tests/test_util.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.5
+#!/usr/bin/env python2.6
 """ Test suite for the code in fixes.util """
 # Author: Collin Winter
 
diff -r e697752393b6 Lib/mailbox.py
--- a/Lib/mailbox.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/mailbox.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Read/write support for Maildir, mbox, MH, Babyl, and MMDF mailboxes."""
 
diff -r e697752393b6 Lib/mimify.py
--- a/Lib/mimify.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/mimify.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Mimification and unmimification of mail messages.
 
diff -r e697752393b6 Lib/pdb.py
--- a/Lib/pdb.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/pdb.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """A Python debugger."""
 
diff -r e697752393b6 Lib/plat-mac/appletrunner.py
--- a/Lib/plat-mac/appletrunner.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/plat-mac/appletrunner.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # This file is meant as an executable script for running applets.
 # BuildApplet will use it as the main executable in the .app bundle if
 # we are not running in a framework build.
diff -r e697752393b6 Lib/profile.py
--- a/Lib/profile.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/profile.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 #
 # Class for profiling python code. rev 1.0  6/2/94
 #
diff -r e697752393b6 Lib/pydoc.py
--- a/Lib/pydoc.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/pydoc.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # -*- coding: latin-1 -*-
 """Generate Python documentation in HTML or text for interactive use.
 
diff -r e697752393b6 Lib/quopri.py
--- a/Lib/quopri.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/quopri.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Conversions to/from quoted-printable transport encoding as per RFC 1521."""
 
diff -r e697752393b6 Lib/smtpd.py
--- a/Lib/smtpd.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/smtpd.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """An RFC 2821 smtp proxy.
 
 Usage: %(program)s [options] [localhost:localport [remotehost:remoteport]]
diff -r e697752393b6 Lib/smtplib.py
--- a/Lib/smtplib.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/smtplib.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 '''SMTP/ESMTP client class.
 
diff -r e697752393b6 Lib/symbol.py
--- a/Lib/symbol.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/symbol.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Non-terminal symbols of Python grammar (from "graminit.h")."""
 
diff -r e697752393b6 Lib/tabnanny.py
--- a/Lib/tabnanny.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/tabnanny.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """The Tab Nanny despises ambiguous indentation.  She knows no mercy.
 
diff -r e697752393b6 Lib/tarfile.py
--- a/Lib/tarfile.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/tarfile.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # -*- coding: iso-8859-1 -*-
 #-------------------------------------------------------------------
 # tarfile.py
diff -r e697752393b6 Lib/test/crashers/recursive_call.py
--- a/Lib/test/crashers/recursive_call.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/crashers/recursive_call.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 # No bug report AFAIK, mail on python-dev on 2006-01-10
 
diff -r e697752393b6 Lib/test/curses_tests.py
--- a/Lib/test/curses_tests.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/curses_tests.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # $Id: ncurses.py 36559 2004-07-18 05:56:09Z tim_one $
 #
diff -r e697752393b6 Lib/test/pystone.py
--- a/Lib/test/pystone.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/pystone.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """
 "PYSTONE" Benchmark Program
diff -r e697752393b6 Lib/test/re_tests.py
--- a/Lib/test/re_tests.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/re_tests.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # -*- mode: python -*-
 
 # Re test suite and benchmark suite v1.5
diff -r e697752393b6 Lib/test/regrtest.py
--- a/Lib/test/regrtest.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/regrtest.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Regression test.
 
diff -r e697752393b6 Lib/test/test___future__.py
--- a/Lib/test/test___future__.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test___future__.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 import unittest
 from test import test_support
 import __future__
diff -r e697752393b6 Lib/test/test_al.py
--- a/Lib/test/test_al.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_al.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Whimpy test script for the al module
    Roger E. Masse
 """
diff -r e697752393b6 Lib/test/test_anydbm.py
--- a/Lib/test/test_anydbm.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_anydbm.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Test script for the anydbm module
    based on testdumbdbm.py
 """
diff -r e697752393b6 Lib/test/test_array.py
--- a/Lib/test/test_array.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_array.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Test the arraymodule.
    Roger E. Masse
 """
diff -r e697752393b6 Lib/test/test_binhex.py
--- a/Lib/test/test_binhex.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_binhex.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Test script for the binhex C module
 
    Uses the mechanism of the python binhex module
diff -r e697752393b6 Lib/test/test_bsddb.py
--- a/Lib/test/test_bsddb.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_bsddb.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Test script for the bsddb C module by Roger E. Masse
    Adapted to unittest format and expanded scope by Raymond Hettinger
 """
diff -r e697752393b6 Lib/test/test_cd.py
--- a/Lib/test/test_cd.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_cd.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Whimpy test script for the cd module
    Roger E. Masse
 """
diff -r e697752393b6 Lib/test/test_cl.py
--- a/Lib/test/test_cl.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_cl.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Whimpy test script for the cl module
    Roger E. Masse
 """
diff -r e697752393b6 Lib/test/test_cmd.py
--- a/Lib/test/test_cmd.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_cmd.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 """
 Test script for the 'cmd' module
 Original by Michael Schneider
diff -r e697752393b6 Lib/test/test_codecencodings_cn.py
--- a/Lib/test/test_codecencodings_cn.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_codecencodings_cn.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_codecencodings_cn.py
 #   Codec encoding tests for PRC encodings.
diff -r e697752393b6 Lib/test/test_codecencodings_hk.py
--- a/Lib/test/test_codecencodings_hk.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_codecencodings_hk.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_codecencodings_hk.py
 #   Codec encoding tests for HongKong encodings.
diff -r e697752393b6 Lib/test/test_codecencodings_jp.py
--- a/Lib/test/test_codecencodings_jp.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_codecencodings_jp.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_codecencodings_jp.py
 #   Codec encoding tests for Japanese encodings.
diff -r e697752393b6 Lib/test/test_codecencodings_kr.py
--- a/Lib/test/test_codecencodings_kr.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_codecencodings_kr.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_codecencodings_kr.py
 #   Codec encoding tests for ROK encodings.
diff -r e697752393b6 Lib/test/test_codecencodings_tw.py
--- a/Lib/test/test_codecencodings_tw.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_codecencodings_tw.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_codecencodings_tw.py
 #   Codec encoding tests for ROC encodings.
diff -r e697752393b6 Lib/test/test_codecmaps_cn.py
--- a/Lib/test/test_codecmaps_cn.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_codecmaps_cn.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_codecmaps_cn.py
 #   Codec mapping tests for PRC encodings
diff -r e697752393b6 Lib/test/test_codecmaps_hk.py
--- a/Lib/test/test_codecmaps_hk.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_codecmaps_hk.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_codecmaps_hk.py
 #   Codec mapping tests for HongKong encodings
diff -r e697752393b6 Lib/test/test_codecmaps_jp.py
--- a/Lib/test/test_codecmaps_jp.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_codecmaps_jp.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_codecmaps_jp.py
 #   Codec mapping tests for Japanese encodings
diff -r e697752393b6 Lib/test/test_codecmaps_kr.py
--- a/Lib/test/test_codecmaps_kr.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_codecmaps_kr.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_codecmaps_kr.py
 #   Codec mapping tests for ROK encodings
diff -r e697752393b6 Lib/test/test_codecmaps_tw.py
--- a/Lib/test/test_codecmaps_tw.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_codecmaps_tw.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_codecmaps_tw.py
 #   Codec mapping tests for ROC encodings
diff -r e697752393b6 Lib/test/test_dl.py
--- a/Lib/test/test_dl.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_dl.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Test dlmodule.c
    Roger E. Masse  revised strategy by Barry Warsaw
 """
diff -r e697752393b6 Lib/test/test_dumbdbm.py
--- a/Lib/test/test_dumbdbm.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_dumbdbm.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Test script for the dumbdbm module
    Original by Roger E. Masse
 """
diff -r e697752393b6 Lib/test/test_eof.py
--- a/Lib/test/test_eof.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_eof.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """test script for a few new invalid token catches"""
 
 import unittest
diff -r e697752393b6 Lib/test/test_errno.py
--- a/Lib/test/test_errno.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_errno.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Test the errno module
    Roger E. Masse
 """
diff -r e697752393b6 Lib/test/test_gl.py
--- a/Lib/test/test_gl.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_gl.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Very simple test script for the SGI gl library extension module
     taken mostly from the documentation.
     Roger E. Masse
diff -r e697752393b6 Lib/test/test_gzip.py
--- a/Lib/test/test_gzip.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_gzip.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Test script for the gzip module.
 """
 
diff -r e697752393b6 Lib/test/test_imageop.py
--- a/Lib/test/test_imageop.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_imageop.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Test script for the imageop module.  This has the side
    effect of partially testing the imgfile module as well.
diff -r e697752393b6 Lib/test/test_imgfile.py
--- a/Lib/test/test_imgfile.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_imgfile.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Simple test script for imgfile.c
    Roger E. Masse
diff -r e697752393b6 Lib/test/test_logging.py
--- a/Lib/test/test_logging.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_logging.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # Copyright 2001-2004 by Vinay Sajip. All Rights Reserved.
 #
diff -r e697752393b6 Lib/test/test_marshal.py
--- a/Lib/test/test_marshal.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_marshal.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # -*- coding: iso-8859-1 -*-
 
 from test import test_support
diff -r e697752393b6 Lib/test/test_multibytecodec.py
--- a/Lib/test/test_multibytecodec.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_multibytecodec.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_multibytecodec.py
 #   Unit test for multibytecodec itself
diff -r e697752393b6 Lib/test/test_multibytecodec_support.py
--- a/Lib/test/test_multibytecodec_support.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_multibytecodec_support.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # test_multibytecodec_support.py
 #   Common Unittest Routines for CJK codecs
diff -r e697752393b6 Lib/test/test_multiprocessing.py
--- a/Lib/test/test_multiprocessing.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_multiprocessing.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 #
 # Unit tests for the multiprocessing package
diff -r e697752393b6 Lib/test/test_popen.py
--- a/Lib/test/test_popen.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_popen.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Basic tests for os.popen()
 
   Particularly useful for platforms that fake popen.
diff -r e697752393b6 Lib/test/test_popen2.py
--- a/Lib/test/test_popen2.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_popen2.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Test script for popen2.py"""
 
 import warnings
diff -r e697752393b6 Lib/test/test_random.py
--- a/Lib/test/test_random.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_random.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 import unittest
 import random
diff -r e697752393b6 Lib/test/test_sets.py
--- a/Lib/test/test_sets.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_sets.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 import warnings
 warnings.filterwarnings("ignore", "the sets module is deprecated",
diff -r e697752393b6 Lib/test/test_socket.py
--- a/Lib/test/test_socket.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_socket.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 import unittest
 from test import test_support
diff -r e697752393b6 Lib/test/test_tcl.py
--- a/Lib/test/test_tcl.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_tcl.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 import unittest
 import os
diff -r e697752393b6 Lib/test/test_urllib2_localnet.py
--- a/Lib/test/test_urllib2_localnet.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_urllib2_localnet.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 import mimetools
 import threading
diff -r e697752393b6 Lib/test/test_urllib2net.py
--- a/Lib/test/test_urllib2net.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_urllib2net.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 import unittest
 from test import test_support
diff -r e697752393b6 Lib/test/test_urllibnet.py
--- a/Lib/test/test_urllibnet.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_urllibnet.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 import unittest
 from test import test_support
diff -r e697752393b6 Lib/test/test_urlparse.py
--- a/Lib/test/test_urlparse.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_urlparse.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 from test import test_support
 import unittest
diff -r e697752393b6 Lib/test/test_userstring.py
--- a/Lib/test/test_userstring.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_userstring.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # UserString is a wrapper around the native builtin string type.
 # UserString instances should behave similar to builtin string objects.
 
diff -r e697752393b6 Lib/test/test_whichdb.py
--- a/Lib/test/test_whichdb.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_whichdb.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Test script for the whichdb module
    based on test_anydbm.py
 """
diff -r e697752393b6 Lib/test/test_with.py
--- a/Lib/test/test_with.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/test/test_with.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """Unit tests for the with statement specified in PEP 343."""
 
diff -r e697752393b6 Lib/timeit.py
--- a/Lib/timeit.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/timeit.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Tool for measuring execution time of small code snippets.
 
diff -r e697752393b6 Lib/token.py
--- a/Lib/token.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/token.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Token constants (from "token.h")."""
 
diff -r e697752393b6 Lib/trace.py
--- a/Lib/trace.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/trace.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 # portions copyright 2001, Autonomous Zones Industries, Inc., all rights...
 # err...  reserved and offered to the public under the terms of the
diff -r e697752393b6 Lib/unittest.py
--- a/Lib/unittest.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/unittest.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 '''
 Python unit testing framework, based on Erich Gamma's JUnit and Kent Beck's
 Smalltalk testing framework.
diff -r e697752393b6 Lib/uu.py
--- a/Lib/uu.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/uu.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Copyright 1994 by Lance Ellinghouse
 # Cathedral City, California Republic, United States of America.
diff -r e697752393b6 Lib/webbrowser.py
--- a/Lib/webbrowser.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/webbrowser.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Interfaces for launching and remotely controlling Web browsers."""
 # Maintained by Georg Brandl.
 
diff -r e697752393b6 Lib/whichdb.py
--- a/Lib/whichdb.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Lib/whichdb.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-# !/usr/bin/env python
+# !/usr/bin/env python2.6
 """Guess which db package to use to open a db file."""
 
 import os
diff -r e697752393b6 Misc/find_recursionlimit.py
--- a/Misc/find_recursionlimit.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Misc/find_recursionlimit.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Find the maximum recursion limit that prevents interpreter termination.
 
 This script finds the maximum safe recursion limit on a particular
diff -r e697752393b6 Parser/asdl_c.py
--- a/Parser/asdl_c.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Parser/asdl_c.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """Generate C code from an ASDL description."""
 
 # TO DO
diff -r e697752393b6 Tools/compiler/demo.py
--- a/Tools/compiler/demo.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/compiler/demo.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Print names of all methods defined in module
 
diff -r e697752393b6 Tools/compiler/dumppyc.py
--- a/Tools/compiler/dumppyc.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/compiler/dumppyc.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 import marshal
 import dis
diff -r e697752393b6 Tools/freeze/freeze.py
--- a/Tools/freeze/freeze.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/freeze/freeze.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Freeze a Python script into a binary.
 
diff -r e697752393b6 Tools/i18n/makelocalealias.py
--- a/Tools/i18n/makelocalealias.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/i18n/makelocalealias.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 """
     Convert the X11 locale.alias file into a mapping dictionary suitable
     for locale.py.
diff -r e697752393b6 Tools/i18n/msgfmt.py
--- a/Tools/i18n/msgfmt.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/i18n/msgfmt.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 # -*- coding: iso-8859-1 -*-
 # Written by Martin v. Löwis <loewis@informatik.hu-berlin.de>
 
diff -r e697752393b6 Tools/i18n/pygettext.py
--- a/Tools/i18n/pygettext.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/i18n/pygettext.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 # -*- coding: iso-8859-1 -*-
 # Originally written by Barry Warsaw <barry@zope.com>
 #
diff -r e697752393b6 Tools/modulator/Tkextra.py
--- a/Tools/modulator/Tkextra.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/modulator/Tkextra.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # A Python function that generates dialog boxes with a text message,
 # optional bitmap, and any number of buttons.
diff -r e697752393b6 Tools/modulator/modulator.py
--- a/Tools/modulator/modulator.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/modulator/modulator.py	Thu Oct 02 12:07:56 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 #
 # Modulator - Generate skeleton modules.
 #
diff -r e697752393b6 Tools/pybench/clockres.py
--- a/Tools/pybench/clockres.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/pybench/clockres.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """ clockres - calculates the resolution in seconds of a given timer.
 
diff -r e697752393b6 Tools/pybench/systimes.py
--- a/Tools/pybench/systimes.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/pybench/systimes.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """ systimes() user and system timer implementations for use by
     pybench.
diff -r e697752393b6 Tools/scripts/byext.py
--- a/Tools/scripts/byext.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/byext.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Show file statistics by extension."""
 
diff -r e697752393b6 Tools/scripts/byteyears.py
--- a/Tools/scripts/byteyears.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/byteyears.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Print the product of age and size of each file, in suitable units.
 #
diff -r e697752393b6 Tools/scripts/checkappend.py
--- a/Tools/scripts/checkappend.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/checkappend.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Released to the public domain, by Tim Peters, 28 February 2000.
 
diff -r e697752393b6 Tools/scripts/checkpyc.py
--- a/Tools/scripts/checkpyc.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/checkpyc.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 # Check that all ".pyc" files exist and are up-to-date
 # Uses module 'os'
 
diff -r e697752393b6 Tools/scripts/classfix.py
--- a/Tools/scripts/classfix.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/classfix.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # This script is obsolete -- it is kept for historical purposes only.
 #
diff -r e697752393b6 Tools/scripts/cleanfuture.py
--- a/Tools/scripts/cleanfuture.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/cleanfuture.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """cleanfuture [-d][-r][-v] path ...
 
diff -r e697752393b6 Tools/scripts/combinerefs.py
--- a/Tools/scripts/combinerefs.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/combinerefs.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """
 combinerefs path
diff -r e697752393b6 Tools/scripts/copytime.py
--- a/Tools/scripts/copytime.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/copytime.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Copy one file's atime and mtime to another
 
diff -r e697752393b6 Tools/scripts/crlf.py
--- a/Tools/scripts/crlf.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/crlf.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 "Replace CRLF with LF in argument files.  Print names of changed files."
 
 import sys, os
diff -r e697752393b6 Tools/scripts/cvsfiles.py
--- a/Tools/scripts/cvsfiles.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/cvsfiles.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Print a list of files that are mentioned in CVS directories.
 
diff -r e697752393b6 Tools/scripts/db2pickle.py
--- a/Tools/scripts/db2pickle.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/db2pickle.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """
 Synopsis: %(prog)s [-h|-g|-b|-r|-a] dbfile [ picklefile ]
diff -r e697752393b6 Tools/scripts/dutree.py
--- a/Tools/scripts/dutree.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/dutree.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 # Format du output in a tree shape
 
 import os, sys, errno
diff -r e697752393b6 Tools/scripts/finddiv.py
--- a/Tools/scripts/finddiv.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/finddiv.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """finddiv - a grep-like tool that looks for division operators.
 
diff -r e697752393b6 Tools/scripts/findlinksto.py
--- a/Tools/scripts/findlinksto.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/findlinksto.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # findlinksto
 #
diff -r e697752393b6 Tools/scripts/findnocoding.py
--- a/Tools/scripts/findnocoding.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/findnocoding.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """List all those Python files that require a coding directive
 
diff -r e697752393b6 Tools/scripts/fixcid.py
--- a/Tools/scripts/fixcid.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/fixcid.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Perform massive identifier substitution on C source files.
 # This actually tokenizes the files (to some extent) so it can
diff -r e697752393b6 Tools/scripts/fixdiv.py
--- a/Tools/scripts/fixdiv.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/fixdiv.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """fixdiv - tool to fix division operators.
 
diff -r e697752393b6 Tools/scripts/fixheader.py
--- a/Tools/scripts/fixheader.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/fixheader.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Add some standard cpp magic to a header file
 
diff -r e697752393b6 Tools/scripts/fixnotice.py
--- a/Tools/scripts/fixnotice.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/fixnotice.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """(Ostensibly) fix copyright notices in files.
 
diff -r e697752393b6 Tools/scripts/fixps.py
--- a/Tools/scripts/fixps.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/fixps.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 # Fix Python script(s) to reference the interpreter via /usr/bin/env python.
 # Warning: this overwrites the file without making a backup.
diff -r e697752393b6 Tools/scripts/ftpmirror.py
--- a/Tools/scripts/ftpmirror.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/ftpmirror.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Mirror a remote ftp subtree into a local directory tree.
 
diff -r e697752393b6 Tools/scripts/google.py
--- a/Tools/scripts/google.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/google.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 import sys, webbrowser
 
diff -r e697752393b6 Tools/scripts/h2py.py
--- a/Tools/scripts/h2py.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/h2py.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Read #define's and translate to Python code.
 # Handle #include statements.
diff -r e697752393b6 Tools/scripts/hotshotmain.py
--- a/Tools/scripts/hotshotmain.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/hotshotmain.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # -*- coding: iso-8859-1 -*-
 
 """
diff -r e697752393b6 Tools/scripts/ifdef.py
--- a/Tools/scripts/ifdef.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/ifdef.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Selectively preprocess #ifdef / #ifndef statements.
 # Usage:
diff -r e697752393b6 Tools/scripts/lfcr.py
--- a/Tools/scripts/lfcr.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/lfcr.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 "Replace LF with CRLF in argument files.  Print names of changed files."
 
diff -r e697752393b6 Tools/scripts/linktree.py
--- a/Tools/scripts/linktree.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/linktree.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # linktree
 #
diff -r e697752393b6 Tools/scripts/lll.py
--- a/Tools/scripts/lll.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/lll.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Find symbolic links and show where they point to.
 # Arguments are directories to search; default is current directory.
diff -r e697752393b6 Tools/scripts/logmerge.py
--- a/Tools/scripts/logmerge.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/logmerge.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Consolidate a bunch of CVS or RCS logs read from stdin.
 
diff -r e697752393b6 Tools/scripts/md5sum.py
--- a/Tools/scripts/md5sum.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/md5sum.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Python utility to print MD5 checksums of argument files.
 """
diff -r e697752393b6 Tools/scripts/methfix.py
--- a/Tools/scripts/methfix.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/methfix.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Fix Python source files to avoid using
 #       def method(self, (arg1, ..., argn)):
diff -r e697752393b6 Tools/scripts/mkreal.py
--- a/Tools/scripts/mkreal.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/mkreal.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # mkreal
 #
diff -r e697752393b6 Tools/scripts/ndiff.py
--- a/Tools/scripts/ndiff.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/ndiff.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Module ndiff version 1.7.0
 # Released to the public domain 08-Dec-2000,
diff -r e697752393b6 Tools/scripts/nm2def.py
--- a/Tools/scripts/nm2def.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/nm2def.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 """nm2def.py
 
 Helpers to extract symbols from Unix libs and auto-generate
diff -r e697752393b6 Tools/scripts/objgraph.py
--- a/Tools/scripts/objgraph.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/objgraph.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # objgraph
 #
diff -r e697752393b6 Tools/scripts/pathfix.py
--- a/Tools/scripts/pathfix.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/pathfix.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Change the #! line occurring in Python scripts.  The new interpreter
 # pathname must be given with a -i option.
diff -r e697752393b6 Tools/scripts/pdeps.py
--- a/Tools/scripts/pdeps.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/pdeps.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # pdeps
 #
diff -r e697752393b6 Tools/scripts/pickle2db.py
--- a/Tools/scripts/pickle2db.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/pickle2db.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """
 Synopsis: %(prog)s [-h|-b|-g|-r|-a|-d] [ picklefile ] dbfile
diff -r e697752393b6 Tools/scripts/pindent.py
--- a/Tools/scripts/pindent.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/pindent.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # This file contains a class and a main program that perform three
 # related (though complimentary) formatting operations on Python
diff -r e697752393b6 Tools/scripts/pysource.py
--- a/Tools/scripts/pysource.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/pysource.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """\
 List python source files.
diff -r e697752393b6 Tools/scripts/reindent.py
--- a/Tools/scripts/reindent.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/reindent.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Released to the public domain, by Tim Peters, 03 October 2000.
 
diff -r e697752393b6 Tools/scripts/rgrep.py
--- a/Tools/scripts/rgrep.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/rgrep.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Reverse grep.
 
diff -r e697752393b6 Tools/scripts/suff.py
--- a/Tools/scripts/suff.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/suff.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # suff
 #
diff -r e697752393b6 Tools/scripts/texi2html.py
--- a/Tools/scripts/texi2html.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/texi2html.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Convert GNU texinfo files into HTML, one file per node.
 # Based on Texinfo 2.14.
diff -r e697752393b6 Tools/scripts/treesync.py
--- a/Tools/scripts/treesync.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/treesync.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Script to synchronize two source trees.
 
diff -r e697752393b6 Tools/scripts/untabify.py
--- a/Tools/scripts/untabify.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/untabify.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 "Replace tabs with spaces in argument files.  Print names of changed files."
 
diff -r e697752393b6 Tools/scripts/which.py
--- a/Tools/scripts/which.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/which.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Variant of "which".
 # On stderr, near and total misses are reported.
diff -r e697752393b6 Tools/scripts/xxci.py
--- a/Tools/scripts/xxci.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/scripts/xxci.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # xxci
 #
diff -r e697752393b6 Tools/ssl/get-remote-certificate.py
--- a/Tools/ssl/get-remote-certificate.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/ssl/get-remote-certificate.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 #
 # fetch the certificate that the server(s) are providing in PEM form
 #
diff -r e697752393b6 Tools/unicode/comparecodecs.py
--- a/Tools/unicode/comparecodecs.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/unicode/comparecodecs.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 
 """ Compare the output of two codecs.
 
diff -r e697752393b6 Tools/webchecker/wcgui.py
--- a/Tools/webchecker/wcgui.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/webchecker/wcgui.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """GUI interface to webchecker.
 
diff -r e697752393b6 Tools/webchecker/webchecker.py
--- a/Tools/webchecker/webchecker.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/webchecker/webchecker.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 # Original code by Guido van Rossum; extensive changes by Sam Bayer,
 # including code to check URL fragments.
diff -r e697752393b6 Tools/webchecker/websucker.py
--- a/Tools/webchecker/websucker.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/webchecker/websucker.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """A variant on webchecker that creates a mirror copy of a remote site."""
 
diff -r e697752393b6 Tools/webchecker/wsgui.py
--- a/Tools/webchecker/wsgui.py	Thu Oct 02 11:33:43 2008 -0400
+++ b/Tools/webchecker/wsgui.py	Thu Oct 02 12:07:57 2008 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.6
 
 """Tkinter-based GUI for websucker.
 
diff -ruN Python-2.6.orig/Lib/platform.py Python-2.6/Lib/platform.py
--- Python-2.6.orig/Lib/platform.py	2008-10-20 18:40:31.000000000 -0400
+++ Python-2.6/Lib/platform.py	2008-10-20 18:41:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 """ This module tries to retrieve as much platform-identifying data as
     possible. It makes this information available via function APIs.
diff -ruN Python-2.6.orig/Lib/test/test_bz2.py Python-2.6/Lib/test/test_bz2.py
--- Python-2.6.orig/Lib/test/test_bz2.py	2008-10-20 18:40:31.000000000 -0400
+++ Python-2.6/Lib/test/test_bz2.py	2008-10-20 18:41:30.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 from test import test_support
 from test.test_support import TESTFN
 
diff -ruN Python-2.6.orig/Lib/test/test_optparse.py Python-2.6/Lib/test/test_optparse.py
--- Python-2.6.orig/Lib/test/test_optparse.py	2008-10-20 18:40:31.000000000 -0400
+++ Python-2.6/Lib/test/test_optparse.py	2008-10-20 18:41:46.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 
 #
 # Test suite for Optik.  Supplied by Johannes Gijsbers
