--- Modules/Setup.dist.orig	2008-10-21 11:24:12.000000000 -0400
+++ Modules/Setup.dist	2008-10-21 12:44:00.000000000 -0400
@@ -73,7 +73,7 @@
 
 # Directories where library files get installed.
 # DESTLIB is for Python modules; MACHDESTLIB for shared libraries.
-DESTLIB=$(LIBDEST)
+DESTLIB=$(BINLIBDEST)
 MACHDESTLIB=$(BINLIBDEST)
 
 # NOTE: all the paths are now relative to the prefix that is computed
