--- mozilla/gfx/src/gtk/nsFontMetricsPango.cpp.foo	2005-03-28 14:20:47.000000000 -0500
+++ mozilla/gfx/src/gtk/nsFontMetricsPango.cpp	2005-03-28 14:23:45.000000000 -0500
@@ -1258,6 +1258,10 @@
     mLTRPangoContext = pango_xft_get_context(GDK_DISPLAY(), 0);
     mPangoContext = mLTRPangoContext;
 
+    // Make sure that this is set LTR - RTL text will be set using the
+    // RTL context.
+    pango_context_set_base_dir(mPangoContext, PANGO_DIRECTION_LTR);
+
     // Set the color map so we can draw later.
     gdk_pango_context_set_colormap(mPangoContext, gdk_rgb_get_cmap());
 
