Index: widget/src/gtk2/nsLookAndFeel.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/gtk2/nsLookAndFeel.cpp,v
retrieving revision 1.20
diff -d -u -p -r1.20 nsLookAndFeel.cpp
--- widget/src/gtk2/nsLookAndFeel.cpp	17 Dec 2004 05:39:19 -0000	1.20
+++ widget/src/gtk2/nsLookAndFeel.cpp	22 Feb 2005 20:13:35 -0000
@@ -78,17 +78,16 @@ nsLookAndFeel::~nsLookAndFeel()
 nsresult nsLookAndFeel::NativeGetColor(const nsColorID aID, nscolor& aColor)
 {
     nsresult res = NS_OK;
-    aColor = 0; // default color black
 
     switch (aID) {
         // These colors don't seem to be used for anything anymore in Mozilla
         // (except here at least TextSelectBackground and TextSelectForeground)
         // The CSS2 colors below are used.
     case eColor_WindowBackground:
-        aColor = GDK_COLOR_TO_NS_RGB(mStyle->bg[GTK_STATE_NORMAL]);
+        aColor = GDK_COLOR_TO_NS_RGB(mStyle->base[GTK_STATE_NORMAL]);
         break;
     case eColor_WindowForeground:
-        aColor = GDK_COLOR_TO_NS_RGB(mStyle->fg[GTK_STATE_NORMAL]);
+        aColor = GDK_COLOR_TO_NS_RGB(mStyle->text[GTK_STATE_NORMAL]);
         break;
     case eColor_WidgetBackground:
         aColor = GDK_COLOR_TO_NS_RGB(mStyle->bg[GTK_STATE_NORMAL]);
