--- mozilla/browser/app/Makefile.in	2005-04-09 01:29:22.000000000 +0200
+++ mozilla/browser/app/Makefile.in_patch	2005-10-03 18:58:48.652416610 +0200
@@ -79,20 +79,24 @@
 	$(XPCOM_LIBS) \
 	$(NSPR_LIBS) \
 	$(TK_LIBS) \
 	$(NULL)
 
 # Add explicit X11 dependency when building against X11 toolkits
 ifneq (,$(filter gtk gtk2 xlib,$(MOZ_WIDGET_TOOLKIT)))
 LIBS += $(XLDFLAGS) $(XLIBS)
 endif
 
+ifdef MOZ_ENABLE_XFT
+LIBS += $(MOZ_XFT_LIBS)
+endif
+
 ifdef MOZ_JPROF
 LIBS += -ljprof
 endif
 
 ifndef MOZ_WINCONSOLE
 ifdef MOZ_DEBUG
 MOZ_WINCONSOLE = 1
 else
 MOZ_WINCONSOLE = 0
 endif
