diff -Naur mozilla.orig/browser/app/firefox-branding.js mozilla/browser/app/firefox-branding.js
--- mozilla.orig/browser/app/firefox-branding.js	2006-09-12 11:50:19.000000000 -0400
+++ mozilla/browser/app/firefox-branding.js	2006-11-12 08:37:36.000000000 -0500
@@ -1,5 +1,5 @@
-pref("startup.homepage_override_url","http://www.mozilla.org/projects/%APP%/%VERSION%/whatsnew/");
-pref("startup.homepage_welcome_url","http://www.mozilla.org/projects/%APP%/%VERSION%/firstrun/");
+pref("startup.homepage_override_url","http://localhost:8004/");
+pref("startup.homepage_welcome_url","http://localhost:8004/");
 // URL user can browse to manually if for some reason all update installation
 // attempts fail.
 pref("app.update.url.manual", "http://www.mozilla.org/products/%APP%/");
diff -Naur mozilla.orig/other-licenses/branding/firefox/pref/firefox-branding.js mozilla/other-licenses/branding/firefox/pref/firefox-branding.js
--- mozilla.orig/other-licenses/branding/firefox/pref/firefox-branding.js	2006-09-15 18:55:01.000000000 -0400
+++ mozilla/other-licenses/branding/firefox/pref/firefox-branding.js	2006-11-12 10:11:35.000000000 -0500
@@ -1,5 +1,5 @@
-pref("startup.homepage_override_url","http://%LOCALE%.www.mozilla.com/%LOCALE%/%APP%/%VERSION%/whatsnew/");
-pref("startup.homepage_welcome_url","http://%LOCALE%.www.mozilla.com/%LOCALE%/%APP%/%VERSION%/firstrun/");
+pref("startup.homepage_override_url","http://localhost:8004/");
+pref("startup.homepage_welcome_url","http://localhost:8004/");
 // URL user can browse to manually if for some reason all update installation
 // attempts fail.
 pref("app.update.url.manual", "http://%LOCALE%.www.mozilla.com/%LOCALE%/%APP%/");
