diff -r dcd782cbd278 rmake/lib/recipeutil.py
--- a/rmake/lib/recipeutil.py   Wed Mar 12 11:42:21 2008 -0400
+++ b/rmake/lib/recipeutil.py   Fri Mar 14 12:26:51 2008 -0400
@@ -355,6 +355,7 @@ def getSourceTrovesFromJob(job, serverCf
         loadInstalledList.append(repos)
         loadInstalledSource = trovesource.stack(buildTroveSource,
                                                 *loadInstalledList)
+        loadInstalledSource.iterTrovesByPath = lambda *args, **kw: []
         loadInstalledList = [ trovesource.TroveListTroveSource(repos, x)
                                 for x in resolveTroveTups ]
         loadInstalledList.append(repos)
