diff --git a/lib/python/mod_python/Session.py b/lib/python/mod_python/Session.py
--- a/lib/python/mod_python/Session.py
+++ b/lib/python/mod_python/Session.py
@@ -26,11 +26,14 @@
 import time
 import anydbm, whichdb
 import random
-import md5
 import cPickle, cStringIO
 import tempfile
 import traceback
 import re
+try:
+    from hashlib import md5
+except ImportError:
+    from md5 import md5
 
 COOKIE_NAME="pysid"
 DFT_TIMEOUT=30*60 # 30 min
@@ -105,7 +108,7 @@
     rnd2 = g.randint(0, 999999999)
     ip = req.connection.remote_ip
 
-    return md5.new("%d%d%d%d%s" % (t, pid, rnd1, rnd2, ip)).hexdigest()
+    return md5("%d%d%d%d%s" % (t, pid, rnd1, rnd2, ip)).hexdigest()
     
 validate_sid_re = re.compile('[0-9a-f]{32}$')
 
diff --git a/lib/python/mod_python/importer.py b/lib/python/mod_python/importer.py
--- a/lib/python/mod_python/importer.py
+++ b/lib/python/mod_python/importer.py
@@ -29,12 +29,15 @@
 import types
 import pdb
 import imp
-import md5
 import time
 import string
 import StringIO
 import traceback
 import cgi
+try:
+    from hashlib import md5
+except ImportError:
+    from md5 import md5
 
 try:
     import threading
@@ -969,7 +972,7 @@
         # name which is a filesystem path. Hope MD5 hex
         # digest is okay.
 
-        return self._prefix + md5.new(file).hexdigest()
+        return self._prefix + md5(file).hexdigest()
 
 
 _global_modules_cache = _ModuleCache()
