diff -Naupr sshmenu-3.15.orig/Makefile sshmenu-3.15/Makefile
--- sshmenu-3.15.orig/Makefile	2008-01-14 22:59:16.000000000 +0100
+++ sshmenu-3.15/Makefile	2008-01-14 23:00:25.000000000 +0100
@@ -12,11 +12,11 @@ TEST:
 	RUBYLIB=lib ruby test/ts_all.rb test/tc*.rb
 
 install:
-	install -d $(BIN) $(LIB)/ruby/1.8 $(LIB)/bonobo/servers
+	install -d $(BIN) $(LIB)/ruby/site_ruby/1.8 $(LIB)/bonobo/servers
 	install -d $(LIBEXEC) $(SHARE)/icons/hicolor/48x48/apps
 	install -d $(MAN)/man1
-	install -m644 lib/sshmenu.rb $(LIB)/ruby/1.8/sshmenu.rb
-	install -m644 lib/gnome-sshmenu.rb $(LIB)/ruby/1.8/gnome-sshmenu.rb
+	install -m644 lib/sshmenu.rb $(LIB)/ruby/site_ruby/1.8/sshmenu.rb
+	install -m644 lib/gnome-sshmenu.rb $(LIB)/ruby/site_ruby/1.8/gnome-sshmenu.rb
 	install -m644 sshmenu-applet.server $(LIB)/bonobo/servers/sshmenu-applet.server
 	install -m755 sshmenu-applet $(LIBEXEC)/sshmenu-applet
 	install -m644 sshmenu.1 $(MAN)/man1/sshmenu.1
