--- module-init-tools/module-init-tools-3.2/depmod.c	2005-10-10 15:10:09.000000000 -0500
+++ module-init-tools-noescape/depmod.c	2007-01-11 15:59:44.000000000 -0600
@@ -607,13 +607,8 @@
 	else
 		afterslash++;
 
-	/* Convert to underscores, stop at first . */
-	for (i = 0; afterslash[i] && afterslash[i] != '.'; i++) {
-		if (afterslash[i] == '-')
-			modname[i] = '_';
-		else
+	for (i = 0; afterslash[i] && afterslash[i] != '.'; i++)
 			modname[i] = afterslash[i];
-	}
 	modname[i] = '\0';
 }
 
