--- LVM2.2.02.51.old/tools/toollib.c	2009-07-21 13:10:49.000000000 +0200
+++ LVM2.2.02.51/tools/toollib.c	2009-08-24 14:15:08.000000000 +0200
@@ -643,7 +643,7 @@ int process_each_pv(struct cmd_context *
 
 	dm_list_init(&tags);
 
-	if (lock_global && !lock_vol(cmd, VG_GLOBAL, LCK_READ)) {
+	if (lock_global && !lock_vol(cmd, VG_GLOBAL, LCK_VG_READ)) {
 		log_error("Unable to obtain global lock.");
 		return ECMD_FAILED;
 	}
--- LVM2.2.02.51.old/WHATS_NEW	2009-08-24 14:14:58.000000000 +0200
+++ LVM2.2.02.51/WHATS_NEW	2009-08-24 14:15:25.000000000 +0200
@@ -1,5 +1,6 @@
 Version 2.02.52 -
 =================================
+  Fix global locking in PV reporting commands (2.02.49).
   Fix pvcreate on a partition (2.02.51).
 
 Version 2.02.51 - 6th August 2009
