--- LVM2.2.02.51.old/lib/device/dev-md.c	2009-08-01 19:14:52.000000000 +0200
+++ LVM2.2.02.51/lib/device/dev-md.c	2009-08-24 14:13:39.000000000 +0200
@@ -137,13 +137,15 @@ static int _md_sysfs_attribute_snprintf(
 	if (!sysfs_dir || !*sysfs_dir)
 		return ret;
 
-check_md_major:
-	if (MAJOR(dev) != md_major()) {
-		if (get_primary_dev(sysfs_dir, blkdev, &dev))
-			goto check_md_major;
-		return ret;
+	if (MAJOR(dev) == blkext_major()) {
+		/* lookup parent MD device from blkext partition */
+		if (!get_primary_dev(sysfs_dir, blkdev, &dev))
+			return ret;
 	}
 
+	if (MAJOR(dev) != md_major())
+		return ret;
+
 	ret = dm_snprintf(path, size, "%s/dev/block/%d:%d/md/%s", sysfs_dir,
 			  (int)MAJOR(dev), (int)MINOR(dev), attribute);
 	if (ret < 0) {
--- LVM2.2.02.51.old/lib/filters/filter.c	2009-07-10 00:34:02.000000000 +0200
+++ LVM2.2.02.51/lib/filters/filter.c	2009-08-24 14:13:39.000000000 +0200
@@ -38,6 +38,7 @@ typedef struct {
 } device_info_t;
 
 static int _md_major = -1;
+static int _blkext_major = -1;
 static int _device_mapper_major = -1;
 
 int md_major(void)
@@ -45,6 +46,11 @@ int md_major(void)
 	return _md_major;
 }
 
+int blkext_major(void)
+{
+	return _blkext_major;
+}
+
 /*
  * Devices are only checked for partition tables if their minor number
  * is a multiple of the number corresponding to their type below
@@ -197,6 +203,10 @@ static int _scan_proc_dev(const char *pr
 		if (!strncmp("md", line + i, 2) && isspace(*(line + i + 2)))
 			_md_major = line_maj;
 
+		/* Look for blkext device */
+		if (!strncmp("blkext", line + i, 6) && isspace(*(line + i + 6)))
+			_blkext_major = line_maj;
+
 		/* Look for device-mapper device */
 		/* FIXME Cope with multiple majors */
 		if (!strncmp("device-mapper", line + i, 13) && isspace(*(line + i + 13)))
--- LVM2.2.02.51.old/lib/filters/filter.h	2007-08-20 22:55:25.000000000 +0200
+++ LVM2.2.02.51/lib/filters/filter.h	2009-08-24 14:13:39.000000000 +0200
@@ -36,6 +36,7 @@ struct dev_filter *lvm_type_filter_creat
 void lvm_type_filter_destroy(struct dev_filter *f);
 
 int md_major(void);
+int blkext_major(void);
 int max_partitions(int major);
 
 #endif
--- LVM2.2.02.51.old/WHATS_NEW	2009-08-06 19:08:01.000000000 +0200
+++ LVM2.2.02.51/WHATS_NEW	2009-08-24 14:14:06.000000000 +0200
@@ -1,3 +1,7 @@
+Version 2.02.52 -
+=================================
+  Fix pvcreate on a partition (2.02.51).
+
 Version 2.02.51 - 6th August 2009
 =================================
   Fix locking in clvmd (2.02.50).
