--- ./bind-9.4.1-P1/bin/named/include/named/globals.h.jau	2006-03-01 19:37:21.000000000 -0500
+++ ./bind-9.4.1-P1/bin/named/include/named/globals.h	2007-12-15 13:42:23.000000000 -0500
@@ -107,9 +107,9 @@
 EXTERN isc_boolean_t		ns_g_logstderr		INIT(ISC_FALSE);
 
 EXTERN const char *		ns_g_defaultpidfile 	INIT(NS_LOCALSTATEDIR
-							     "/run/named.pid");
+							     "/run/named/named.pid");
 EXTERN const char *		lwresd_g_defaultpidfile INIT(NS_LOCALSTATEDIR
-							    "/run/lwresd.pid");
+							    "/run/named/lwresd.pid");
 EXTERN const char *		ns_g_username		INIT(NULL);
 
 EXTERN int			ns_g_listen		INIT(3);
