diff -ur meld-1.1.4.orig/help/C/GNUmakefile meld-1.1.4/help/C/GNUmakefile
--- meld-1.1.4.orig/help/C/GNUmakefile	2006-01-28 16:43:57.000000000 -0500
+++ meld-1.1.4/help/C/GNUmakefile	2006-11-05 17:34:55.000000000 -0500
@@ -10,7 +10,7 @@
 
 .PHONY : all
 all $(OMF_NAME).install : $(OMF_NAME)
-	scrollkeeper-preinstall $(INST_XML) $(OMF_NAME) $(OMF_NAME).install
+	scrollkeeper-preinstall $(XML_DIR_)/meld.xml $(OMF_NAME) $(OMF_NAME).install
 
 .PHONY : install
 install : $(OMF_NAME).install
