diff -r b87b6a520379 filediff.py
--- a/filediff.py	Sat Dec 20 08:32:21 2008 -0500
+++ b/filediff.py	Sat Dec 20 08:36:57 2008 -0500
@@ -568,7 +568,11 @@
                     bnew.label = bold.label
                 self.bufferdata[i] = bnew
         self.recompute_label()
-        self.textview[1 if len(files) >= 2 else 0].grab_focus()
+        if len(files) >= 2:
+            self.textview[1].grab_focus()
+        else:
+            self.textview[0].grab_focus()
+
         self.scheduler.add_task( self._set_files_internal(files).next )
 
     def _set_files_internal(self, files):
