diff -uHr ipw2100-1.1.2.old/ipw2100.h ipw2100-1.1.2/ipw2100.h
--- ipw2100-1.1.2.old/ipw2100.h	2005-07-17 13:21:14.000000000 -0700
+++ ipw2100-1.1.2/ipw2100.h	2005-07-17 13:25:42.000000000 -0700
@@ -786,7 +786,7 @@
 
 
 
-#define IPW_HEADER_802_11_SIZE		 sizeof(struct ieee80211_header_data)
+#define IPW_HEADER_802_11_SIZE		 sizeof(struct ieee80211_hdr_3addr)
 #define IPW_MAX_80211_PAYLOAD_SIZE              2304U
 #define IPW_MAX_802_11_PAYLOAD_LENGTH		2312
 #define IPW_MAX_ACCEPTABLE_TX_FRAME_LENGTH	1536
@@ -801,7 +801,7 @@
 		IPW_802_11_FCS_LENGTH)
 
 #define IPW_802_11_PAYLOAD_OFFSET \
-        (sizeof(struct ieee80211_header_data) + \
+        (sizeof(struct ieee80211_hdr_3addr) + \
          sizeof(struct ieee80211_snap_hdr))
 
 struct ipw2100_rx {
diff -uHr ipw2100-1.1.2.old/Makefile ipw2100-1.1.2/Makefile
--- ipw2100-1.1.2.old/Makefile	2005-07-17 13:23:55.000000000 -0700
+++ ipw2100-1.1.2/Makefile	2005-07-17 13:20:30.000000000 -0700
@@ -89,7 +89,7 @@
 # modules are installed into KMISC
 KVER  := $(shell uname -r)
 KSRC := /lib/modules/$(KVER)/build
-KMISC := /lib/modules/$(KVER)/kernel/drivers/net/wireless/
+KMISC := $(DESTDIR)/lib/modules/$(KVER)/kernel/drivers/net/wireless/
 KMISC_INC := /lib/modules/$(KVER)/include
 
 # KSRC_OUTPUT should be overridden if you are using a 2.6 kernel that
@@ -156,7 +156,6 @@
 install: modules
 	install -d $(KMISC)
 	install -m 644 -c $(addsuffix .ko,$(list-m)) $(KMISC)
-	/sbin/depmod -a
 	@echo "Don't forget to copy firmware to your hotplug's firmware directory and have the "
 	@echo "hotplug tools in place."
 	@echo "See INSTALL for more information."
