Index: src/rbgnome-canvas-path-def.c
===================================================================
RCS file: /cvsroot/ruby-gnome2/ruby-gnome2/gnomecanvas/src/rbgnome-canvas-path-def.c,v
retrieving revision 1.11
diff -u -p -r1.11 rbgnome-canvas-path-def.c
--- gnomecanvas/src/rbgnome-canvas-path-def.c	1 Aug 2005 13:25:22 -0000	1.11
+++ gnomecanvas/src/rbgnome-canvas-path-def.c	11 Sep 2005 13:22:02 -0000
@@ -28,6 +28,8 @@ static VALUE gnoCanvasPathDef;
 #define _SELF(self) (GnomeCanvasPathDef *)RVAL2BOXED(self, rbgno_canvas_path_def_get_type())
 #define _WRAP(self) BOXED2RVAL(self, rbgno_canvas_path_def_get_type())
 
+#ifndef GNOME_TYPE_CANVAS_PATH_DEF
+/* this is for libgnomecanvas-2.0 < 2.12 */
 static GnomeCanvasPathDef *
 rbgno_canvas_path_def_copy(path_def)
     GnomeCanvasPathDef *path_def;
@@ -48,6 +50,9 @@ rbgno_canvas_path_def_get_type()
     return our_type;
 }
 
+#define GNOME_TYPE_CANVAS_PATH_DEF (rbgno_canvas_path_def_get_type())
+#endif /* GNOME_TYPE_CANVAS_PATH_DEF */
+
 static void
 cpathdef_r2g_func(from, to)
     VALUE from;
@@ -345,7 +350,7 @@ void
 Init_gnome_canvas_path_def(mGnome)
     VALUE mGnome;
 {
-    gnoCanvasPathDef = G_DEF_CLASS(rbgno_canvas_path_def_get_type(), "CanvasPathDef", mGnome);
+    gnoCanvasPathDef = G_DEF_CLASS(GNOME_TYPE_CANVAS_PATH_DEF, "CanvasPathDef", mGnome);
     rbgobj_register_property_setter(GNOME_TYPE_CANVAS_BPATH, "bpath", cpathdef_r2g_func);
     rbgobj_register_property_getter(GNOME_TYPE_CANVAS_BPATH, "bpath", cpathdef_g2r_func);
 
