--- chromium/home/chrome-svn/tarball/chromium/src/net/base/mime_util.cc	2009-12-04 11:04:18.000000000 +0100
+++ chromium/home/chrome-svn/tarball/chromium/src/net/base/mime_util-copy.cc	2010-02-06 19:57:41.000000000 +0100
@@ -188,7 +188,6 @@
   "audio/ogg",
   "application/ogg",
 
-#if defined(GOOGLE_CHROME_BUILD)
   // MPEG-4.
   "video/mp4",
   "video/x-m4v",
@@ -199,7 +198,7 @@
   "audio/mp3",
   "audio/x-mp3",
   "audio/mpeg",
-#endif
+
 };
 
 // List of supported codecs when passed in with <source type="...">.
@@ -207,10 +206,10 @@
 // Refer to http://wiki.whatwg.org/wiki/Video_type_parameters#Browser_Support
 // for more information.
 static const char* const supported_media_codecs[] = {
-#if defined(GOOGLE_CHROME_BUILD)
+
   "avc1",
   "mp4a",
-#endif
+
   "theora",
   "vorbis",
 };

