Index: chrome-svn/tarball/chromium/src/build/common.gypi
===================================================================
--- chrome-svn/tarball/chromium/src/build/common.gypi   (revision 23812)
+++ chrome-svn/tarball/chromium/src/build/common.gypi   (working copy)
@@ -355,7 +355,6 @@
           'no_strict_aliasing%': 0,
         },
         'cflags': [
-          '<(werror)',  # See note above about the werror variable.
           '-pthread',
           '-fno-exceptions',
           '-Wall',
Index: chrome-svn/tarball/chromium/src/build/external_code.gypi
===================================================================
--- chrome-svn/tarball/chromium/src/build/external_code.gypi    (revision 23812)
+++ chrome-svn/tarball/chromium/src/build/external_code.gypi    (working copy)
@@ -8,7 +8,6 @@
       'target_defaults': {
         'cflags!': [
           '-Wall',
-          '-Werror',
         ],
       },
     }],
