diff --git a/kalarm/kalarmapp.cpp b/kalarm/kalarmapp.cpp
index 8d6b898..f4d3bda 100644
--- a/kalarm/kalarmapp.cpp
+++ b/kalarm/kalarmapp.cpp
@@ -56,6 +56,7 @@
 #include <netwm.h>
 #include <kdebug.h>
 #include <kshell.h>
+#include <ksystemtrayicon.h>
 
 #include <QObject>
 #include <QTimer>
@@ -762,7 +763,6 @@ bool KAlarmApp::displayTrayIcon(bool show, MainWindow* parent)
 			}
 			mTrayWindow = new TrayWindow(parent ? parent : MainWindow::firstWindow());
 			connect(mTrayWindow, SIGNAL(deleted()), SIGNAL(trayIconToggled()));
-			mTrayWindow->show();
 			emit trayIconToggled();
 
 			if (!checkSystemTray())
diff --git a/kalarm/kalarmconfig.kcfg b/kalarm/kalarmconfig.kcfg
index 3bef849..55f9a91 100644
--- a/kalarm/kalarmconfig.kcfg
+++ b/kalarm/kalarmconfig.kcfg
@@ -47,6 +47,8 @@
   <signal name="feb29TypeChanged">
     <argument type="Enum">DefaultFeb29Type</argument>
   </signal>
+  <signal name="tooltipPreferencesChanged">
+  </signal>
 
   <group name="General">
     <entry name="Version" type="String" hidden="true">
@@ -125,21 +127,25 @@
       </whatsthis>
       <default>5</default>
       <min>-1</min>
+      <emit signal="tooltipPreferencesChanged"/>
     </entry>
     <entry name="ShowTooltipAlarmTime" type="Bool">
       <label context="@label">Show alarm times in system tray tooltip</label>
       <whatsthis context="@info:whatsthis">Specify whether to show in the system tray tooltip, the time at which each alarm is due.</whatsthis>
       <default>true</default>
+      <emit signal="tooltipPreferencesChanged"/>
     </entry>
     <entry name="ShowTooltipTimeToAlarm" type="Bool">
       <label context="@label">Show time to alarms in system tray tooltip</label>
       <whatsthis context="@info:whatsthis">Specify whether to show in the system tray tooltip, how long until each alarm is due.</whatsthis>
       <default>true</default>
+      <emit signal="tooltipPreferencesChanged"/>
     </entry>
     <entry name="TooltipTimeToPrefix" type="String">
       <label context="@label">Time-to-alarm prefix in system tray tooltip</label>
       <whatsthis context="@info:whatsthis">Enter the text to be displayed in front of the time until the alarm, in the system tray tooltip.</whatsthis>
       <default code="true">QLatin1String("+")</default>
+      <emit signal="tooltipPreferencesChanged"/>
     </entry>
     <entry name="EmailClient" key="EmailClient" type="Enum">
       <label context="@label">Email client</label>
diff --git a/kalarm/mainwindow.cpp b/kalarm/mainwindow.cpp
index 9ad679b..3355e97 100644
--- a/kalarm/mainwindow.cpp
+++ b/kalarm/mainwindow.cpp
@@ -60,6 +60,7 @@
 #include <kaction.h>
 #include <kactioncollection.h>
 #include <kinputdialog.h>
+#include <ksystemtrayicon.h>
 
 #include <kstandardaction.h>
 #include <kiconloader.h>
diff --git a/kalarm/pixmaps/ox22-emblems-partdisabled.png b/kalarm/pixmaps/ox22-emblems-partdisabled.png
deleted file mode 100644
index 6e40203..0000000
Binary files a/kalarm/pixmaps/ox22-emblems-partdisabled.png and /dev/null differ
diff --git a/kalarm/pixmaps/ox8-emblems-disabled.png b/kalarm/pixmaps/ox8-emblems-disabled.png
new file mode 100644
index 0000000..ed3ec7c
Binary files /dev/null and b/kalarm/pixmaps/ox8-emblems-disabled.png differ
diff --git a/kalarm/pixmaps/ox8-emblems-partdisabled.png b/kalarm/pixmaps/ox8-emblems-partdisabled.png
new file mode 100644
index 0000000..5f2c1a9
Binary files /dev/null and b/kalarm/pixmaps/ox8-emblems-partdisabled.png differ
diff --git a/kalarm/traywindow.cpp b/kalarm/traywindow.cpp
index df2790d..347edc7 100644
--- a/kalarm/traywindow.cpp
+++ b/kalarm/traywindow.cpp
@@ -32,6 +32,7 @@
 #include "newalarmaction.h"
 #include "prefdlg.h"
 #include "preferences.h"
+#include "synchtimer.h"
 #include "templatemenuaction.h"
 
 #include <stdlib.h>
@@ -69,34 +70,14 @@ struct TipItem
 =============================================================================*/
 
 TrayWindow::TrayWindow(MainWindow* parent)
-	: KSystemTrayIcon(parent),
+	: KStatusNotifierItem(parent),
 	  mAssocMainWindow(parent),
 	  mHaveDisabledAlarms(false)
 {
 	kDebug();
-	// Set up GUI icons
-	mIconEnabled  = loadIcon("kalarm");
-	if (mIconEnabled.isNull())
-		KMessageBox::sorry(parent, i18nc("@info", "Cannot load system tray icon."));
-	else
-	{
-		// Create the all alarms disabled icon, by converting the normal icon to grey
-		KIconLoader* loader = KIconLoader::global();
-		QImage icon = mIconEnabled.pixmap(loader->currentSize(KIconLoader::Panel)).toImage();
-		QImage iconDisabled = icon;
-		KIconEffect::toGray(iconDisabled, 1.0);
-		mIconDisabled = QIcon(QPixmap::fromImage(iconDisabled));
-
-		// Create the partially disabled icon, by overlaying the normal icon
-		// with a disabled indication
-		QImage disabled = loader->loadIcon("partdisabled", KIconLoader::Panel, icon.width(), KIconLoader::DefaultState, QStringList("emblems")).toImage();
-		KIconEffect::overlay(icon, disabled);
-		mIconSomeDisabled = QIcon(QPixmap::fromImage(icon));
-	}
-#ifdef __GNUC__
-#warning How to implement drag-and-drop?
-#endif
-	//setAcceptDrops(true);         // allow drag-and-drop onto this window
+	setToolTipIconByName("kalarm");
+	setToolTipTitle(KGlobal::mainComponent().aboutData()->programName());
+	setIconByName("kalarm");
 
 	// Set up the context menu
 	KActionCollection* actions = actionCollection();
@@ -135,14 +116,40 @@ TrayWindow::TrayWindow(MainWindow* parent)
 	a = KStandardAction::quit(0, 0, actions);
 	connect(a, SIGNAL(triggered(bool)), SLOT(slotQuit()), Qt::QueuedConnection);
 
-	// Set icon to correspond with the alarms enabled menu status
-	setEnabledStatus(theApp()->alarmsEnabled());
-
 	connect(AlarmResources::instance(), SIGNAL(resourceStatusChanged(AlarmResource*, AlarmResources::Change)), SLOT(slotResourceStatusChanged()));
 	connect(AlarmCalendar::resources(), SIGNAL(haveDisabledAlarmsChanged(bool)), SLOT(slotHaveDisabledAlarms(bool)));
-	connect(this, SIGNAL(activated(QSystemTrayIcon::ActivationReason)), SLOT(slotActivated(QSystemTrayIcon::ActivationReason)));
+	connect(this, SIGNAL(activateRequested(bool, const QPoint&)), SLOT(slotActivateRequested()));
+	connect(this, SIGNAL(secondaryActivateRequested(const QPoint&)), SLOT(slotSecondaryActivateRequested()));
 	slotResourceStatusChanged();   // initialise action states
 	slotHaveDisabledAlarms(AlarmCalendar::resources()->haveDisabledAlarms());
+
+	// Hack: KSNI does not let us know when it is about to show the tooltip,
+	// so we need to update it whenever something change in it.
+
+	// This timer ensure updateToolTip() is not called several times in a row
+	mToolTipUpdateTimer = new QTimer(this);
+	mToolTipUpdateTimer->setInterval(0);
+	mToolTipUpdateTimer->setSingleShot(true);
+	connect(mToolTipUpdateTimer, SIGNAL(timeout()), SLOT(updateToolTip()));
+
+	// Update every minute to show accurate deadlines
+	MinuteTimer::connect(mToolTipUpdateTimer, SLOT(start()));
+
+	// Update when alarms are modified
+	connect(EventListModel::alarms(), SIGNAL(dataChanged(const QModelIndex&, const QModelIndex&)),
+		mToolTipUpdateTimer, SLOT(start()));
+	connect(EventListModel::alarms(), SIGNAL(rowsInserted(const QModelIndex&, int, int)),
+		mToolTipUpdateTimer, SLOT(start()));
+	connect(EventListModel::alarms(), SIGNAL(rowsMoved(const QModelIndex&, int, int, const QModelIndex&, int)),
+		mToolTipUpdateTimer, SLOT(start()));
+	connect(EventListModel::alarms(), SIGNAL(rowsRemoved(const QModelIndex&, int, int)),
+		mToolTipUpdateTimer, SLOT(start()));
+	connect(EventListModel::alarms(), SIGNAL(modelReset()),
+		mToolTipUpdateTimer, SLOT(start()));
+
+	// Update when tooltip preferences are modified
+	Preferences::connect(SIGNAL(tooltipPreferencesChanged()),
+		mToolTipUpdateTimer, SLOT(start()));
 }
 
 TrayWindow::~TrayWindow()
@@ -196,7 +203,8 @@ void TrayWindow::slotPreferences()
 */
 void TrayWindow::slotQuit()
 {
-	theApp()->doQuit(parentWidget());
+	// FIXME: Do we really need a slotQuit()?
+	theApp()->doQuit(static_cast<QWidget*>(parent()));
 }
 
 /******************************************************************************
@@ -206,7 +214,8 @@ void TrayWindow::slotQuit()
 void TrayWindow::setEnabledStatus(bool status)
 {
 	kDebug() << (int)status;
-	setIcon(status ? (mHaveDisabledAlarms ? mIconSomeDisabled : mIconEnabled) : mIconDisabled);
+	updateIcon();
+	updateToolTip();
 }
 
 /******************************************************************************
@@ -217,75 +226,74 @@ void TrayWindow::slotHaveDisabledAlarms(bool haveDisabled)
 {
 	kDebug() << haveDisabled;
 	mHaveDisabledAlarms = haveDisabled;
-	if (mActionEnabled->isChecked())
-		setIcon(haveDisabled ? mIconSomeDisabled : mIconEnabled);
+	updateIcon();
+	updateToolTip();
 }
 
 /******************************************************************************
-*  Called when the mouse is clicked over the panel icon.
 *  A left click displays the KAlarm main window.
-*  A middle button click displays the New Alarm window.
 */
-void TrayWindow::slotActivated(QSystemTrayIcon::ActivationReason reason)
+void TrayWindow::slotActivateRequested()
 {
-	if (reason == QSystemTrayIcon::Trigger)
-	{
-		// Left click: display/hide the first main window
-		if (mAssocMainWindow  &&  mAssocMainWindow->isVisible())
-		{
-			mAssocMainWindow->raise();
-			mAssocMainWindow->activateWindow();
-		}
-	}
-	else if (reason == QSystemTrayIcon::MiddleClick)
+	// Left click: display/hide the first main window
+	if (mAssocMainWindow  &&  mAssocMainWindow->isVisible())
 	{
-		if (mActionNew->isEnabled())
-			mActionNew->trigger();    // display a New Alarm dialog
+		mAssocMainWindow->raise();
+		mAssocMainWindow->activateWindow();
 	}
 }
 
 /******************************************************************************
-*  Called when the drag cursor enters the panel icon.
+*  A middle button click displays the New Alarm window.
 */
-void TrayWindow::dragEnterEvent(QDragEnterEvent* e)
+void TrayWindow::slotSecondaryActivateRequested()
 {
-	MainWindow::executeDragEnterEvent(e);
+	if (mActionNew->isEnabled())
+		mActionNew->trigger();    // display a New Alarm dialog
 }
 
 /******************************************************************************
-*  Called when an object is dropped on the panel icon.
-*  If the object is recognised, the edit alarm dialog is opened appropriately.
+*  Adjust tooltip according to the app state.
+*  The tooltip text shows alarms due in the next 24 hours. The limit of 24
+*  hours is because only times, not dates, are displayed.
 */
-void TrayWindow::dropEvent(QDropEvent* e)
+void TrayWindow::updateToolTip()
 {
-	MainWindow::executeDropEvent(0, e);
+	bool enabled = theApp()->alarmsEnabled();
+	QString subTitle;
+	if (enabled  &&  Preferences::tooltipAlarmCount())
+		subTitle = tooltipAlarmText();
+	if (!enabled) {
+		//text = i18nc("@info:tooltip 'KAlarm - disabled'", "%1 - disabled", KGlobal::mainComponent().aboutData()->programName());
+		subTitle = i18n("Disabled");
+	} else if (mHaveDisabledAlarms) {
+		if (!subTitle.isEmpty()) {
+			subTitle += "<br/>";
+		}
+		#if 0
+		subTitle += i18nc("@info:tooltip Brief: some alarms are disabled", "(Some alarms disabled)");
+		#else
+		// FIXME: Hack to avoid introducing new strings
+		QString text = i18nc("@info:tooltip Brief: some alarms are disabled", "%1<nl/>(Some alarms disabled)%2", QString(), QString());
+		// i18n() turns "<nl/>" into "<br/>"
+		text = text.mid(text.indexOf("<br/>") + 5);
+		subTitle += text;
+		#endif
+	}
+	kDebug() << subTitle;
+	setToolTipSubTitle(subTitle);
 }
 
 /******************************************************************************
-*  Called when any event occurs.
-*  If it's a tooltip event, display the tooltip text showing alarms due in the
-*  next 24 hours. The limit of 24 hours is because only times, not dates, are
-*  displayed.
+*  Adjust icon according to the app state.
 */
-bool TrayWindow::event(QEvent* e)
+void TrayWindow::updateIcon()
 {
-	if (e->type() != QEvent::ToolTip)
-		return KSystemTrayIcon::event(e);
-	QHelpEvent* he = (QHelpEvent*)e;
-	bool enabled = theApp()->alarmsEnabled();
-	QString altext;
-	if (enabled  &&  Preferences::tooltipAlarmCount())
-		altext = tooltipAlarmText();
-	QString text;
-	if (!enabled)
-		text = i18nc("@info:tooltip 'KAlarm - disabled'", "%1 - disabled", KGlobal::mainComponent().aboutData()->programName());
-	else if (mHaveDisabledAlarms)
-		text = i18nc("@info:tooltip Brief: some alarms are disabled", "%1<nl/>(Some alarms disabled)%2", KGlobal::mainComponent().aboutData()->programName(), altext);
-	else
-		text = i18nc("@info:tooltip", "%1%2", KGlobal::mainComponent().aboutData()->programName(), altext);
-	kDebug() << text;
-	QToolTip::showText(he->globalPos(), text);
-	return true;
+	if (theApp()->alarmsEnabled()) {
+		setOverlayIconByName(mHaveDisabledAlarms ? "partdisabled" : QString());
+	} else {
+		setOverlayIconByName("disabled");
+	}
 }
 
 /******************************************************************************
@@ -353,7 +361,9 @@ QString TrayWindow::tooltipAlarmText() const
 	for (i = 0, iend = items.count();  i < iend;  ++i)
 	{
 		kDebug() << "--" << (count+1) << ")" << items[i].text;
-		text += "<br />" + items[i].text;
+		if (i > 0)
+			text += "<br />";
+		text += items[i].text;
 		if (++count == maxCount)
 			break;
 	}
diff --git a/kalarm/traywindow.h b/kalarm/traywindow.h
index 9d402c5..cfcdddb 100644
--- a/kalarm/traywindow.h
+++ b/kalarm/traywindow.h
@@ -23,7 +23,7 @@
 
 #include "editdlg.h"
 #include "kaevent.h"
-#include <ksystemtrayicon.h>
+#include <kstatusnotifieritem.h>
 #include <QIcon>
 
 class QEvent;
@@ -35,7 +35,7 @@ class KAEvent;
 class MainWindow;
 class NewAlarmAction;
 
-class TrayWindow : public KSystemTrayIcon
+class TrayWindow : public KStatusNotifierItem
 {
 		Q_OBJECT
 	public:
@@ -48,13 +48,9 @@ class TrayWindow : public KSystemTrayIcon
 	signals:
 		void         deleted();
 
-	protected:
-		virtual void dragEnterEvent(QDragEnterEvent*);
-		virtual void dropEvent(QDropEvent*);
-		virtual bool event(QEvent*);
-
 	private slots:
-		void         slotActivated(QSystemTrayIcon::ActivationReason reason);
+		void         slotActivateRequested();
+		void         slotSecondaryActivateRequested();
 		void         slotNewAlarm(EditAlarmDlg::Type);
 		void         slotNewFromTemplate(const KAEvent*);
 		void         slotPreferences();
@@ -62,18 +58,18 @@ class TrayWindow : public KSystemTrayIcon
 		void         slotHaveDisabledAlarms(bool disabled);
 		void         slotResourceStatusChanged();
 		void         slotQuit();
+		void         updateToolTip();
 
 	private:
 		QString      tooltipAlarmText() const;
+		void         updateIcon();
 
 		MainWindow*     mAssocMainWindow;     // main window associated with this, or null
-		QIcon           mIconEnabled;         // normal status icon
-		QIcon           mIconDisabled;        // icon indicating all alarms disabled
-		QIcon           mIconSomeDisabled;    // icon indicating individual alarms disabled
 		KToggleAction*  mActionEnabled;
 		NewAlarmAction* mActionNew;
 		KAction*        mActionNewFromTemplate;
 		bool            mHaveDisabledAlarms;  // some individually disabled alarms exist
+		QTimer*         mToolTipUpdateTimer;
 };
 
 #endif // TRAYWINDOW_H
