diff -ur linuxwacom-0.7.5-4/src/xdrv/xf86Wacom.h linuxwacom-0.7.5-4.new/src/xdrv/xf86Wacom.h
--- linuxwacom-0.7.5-4/src/xdrv/xf86Wacom.h	2006-09-27 18:21:23.000000000 -0400
+++ linuxwacom-0.7.5-4.new/src/xdrv/xf86Wacom.h	2006-11-16 15:19:50.000000000 -0500
@@ -67,7 +67,6 @@
 #define NEED_XF86_TYPES
 #if !defined(DGUX)
 #include <xf86_ansic.h>
-#include <xisb.h>
 #endif
 #include <xf86_OSproc.h>
 #include <xf86Xinput.h>
