diff -r 934719bb0c0f ev_calibrate.c
--- a/ev_calibrate.c	Mon Apr 06 14:24:58 2009 -0400
+++ b/ev_calibrate.c	Mon Apr 06 14:31:26 2009 -0400
@@ -287,7 +287,7 @@
         }
 
         rc = XReadBitmapFile(display, win,
-                             CURSORDIRECTORY"/empty_cursor.xbm",
+                             "/usr/share/xf86-input-evtouch/empty_cursor.xbm",
                              &cursor_width, &cursor_height,
                              &cursor_pic,
                              &hotspot_x, &hotspot_y);
