--- instantbird/instantbird-0.2-src/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc	2010-07-13 18:05:06.000000000 +0200
+++ instantbird/instantbird-0.2-src/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id-copy.cc	2011-05-13 22:20:57.000000000 +0200
@@ -42,6 +42,7 @@
 #include <link.h>
 #include <string.h>
 #include <sys/mman.h>
+#include <sys/stat.h>
 #include <unistd.h>
 
 #include <algorithm>
