--- instantbird/instantbird-0.2-src/mozilla/toolkit/system/gnome/Makefile.in	2010-07-13 18:05:08.000000000 +0200
+++ instantbird/instantbird-0.2-src/mozilla/toolkit/system/gnome/Makefile-copy.in	2011-05-13 23:32:26.000000000 +0200
@@ -80,6 +80,7 @@
 		$(NSPR_LIBS) \
 		$(MOZ_GCONF_LIBS) \
 		$(MOZ_GNOMEVFS_LIBS) \
+        $(MOZ_GNOMEUI_LIBS) \
 		$(GLIB_LIBS) \
 		$(MOZ_LIBNOTIFY_LIBS) \
 		$(NULL)
@@ -91,6 +92,7 @@
 CXXFLAGS += \
 	$(MOZ_GCONF_CFLAGS) \
 	$(MOZ_GNOMEVFS_CFLAGS) \
+    $(MOZ_GNOMEUI_CFLAGS) \
 	$(GLIB_CFLAGS) \
 	$(MOZ_LIBNOTIFY_CFLAGS) \
 	$(NULL)
