diff -r 8aefe5e505f5 conary/build/build.py
--- a/conary/build/build.py	Wed Oct 22 11:25:32 2008 -0400
+++ b/conary/build/build.py	Wed Nov 05 16:20:59 2008 -0500
@@ -1214,9 +1214,9 @@
     """
     _actionPathBuildRequires = set(['python'])
     template = (
-	"""python -c 'from compileall import *; compile_dir("""
+	"""python2.6 -c 'from compileall import *; compile_dir("""
 	""""%%(destdir)s/%%(dir)s", 10, "%%(dir)s")'; """
-	"""python -O -c 'from compileall import *; compile_dir("""
+	"""python2.6 -O -c 'from compileall import *; compile_dir("""
 	""""%%(destdir)s/%%(dir)s", 10, "%%(dir)s")'"""
     )
 
