diff -r 08f60483e00b conary/build/packagepolicy.py
--- a/conary/build/packagepolicy.py	Tue Dec 16 18:29:41 2008 -0500
+++ b/conary/build/packagepolicy.py	Mon Dec 29 19:43:07 2008 +0000
@@ -2659,6 +2659,7 @@
             sys.path = []
             sys.prefix = destdir + sys.prefix
             sys.exec_prefix = destdir + sys.exec_prefix
+            site.PREFIXES=[sys.prefix, sys.exec_prefix]  
             site.addsitepackages(None)
             systemPaths.update(self._stripDestDir(sys.path, destdir))
 
@@ -3472,5 +3473,6 @@
             sys.path = [destdir + pythonDir]
             sys.prefix = destdir + sys.prefix
             sys.exec_prefix = destdir + sys.exec_prefix
+            site.PREFIXES=[sys.prefix, sys.exec_prefix]  
             site.addsitepackages(None)
             systemPaths = sys.path + systemPaths
