diff -r b0074c59ad1e conary/build/build.py
--- a/conary/build/build.py	Wed Jan 21 10:29:27 2009 +0300
+++ b/conary/build/build.py	Wed Jan 21 10:30:09 2009 +0300
@@ -31,7 +31,6 @@
 
 import os
 import re
-import sha
 import shutil
 import stat
 import sys
@@ -41,6 +40,7 @@
 #conary imports
 from conary.build import action, errors
 from conary.lib import fixedglob, log, util
+from conary.lib.crypto import sha
 from conary.build.use import Use
 from conary.build.manifest import Manifest, ExplicitManifest
 
diff -r b0074c59ad1e conary/build/source.py
--- a/conary/build/source.py	Wed Jan 21 10:29:27 2009 +0300
+++ b/conary/build/source.py	Wed Jan 21 10:30:09 2009 +0300
@@ -22,7 +22,6 @@
 import os
 import re
 import shutil, subprocess
-import sha
 import shlex
 import sys
 import tempfile
@@ -31,6 +30,7 @@
 from conary.build import lookaside
 from conary import rpmhelper
 from conary.lib import openpgpfile, util
+from conary.lib.crypto import sha
 from conary.build import action, errors
 from conary.build.errors import RecipeFileError
 from conary.build.manifest import Manifest
diff -r b0074c59ad1e conary/conaryclient/filetypes.py
--- a/conary/conaryclient/filetypes.py	Wed Jan 21 10:29:27 2009 +0300
+++ b/conary/conaryclient/filetypes.py	Wed Jan 21 10:30:09 2009 +0300
@@ -20,12 +20,12 @@
 
 import itertools
 import os
-import sha
 import time
 
 from conary import errors, files
 from conary.deps import deps
 from conary.lib import sha1helper, util
+from conary.lib.crypto import sha
 from conary.repository import filecontents
 
 class ParameterError(errors.ClientError):
diff -r b0074c59ad1e conary/files.py
--- a/conary/files.py	Wed Jan 21 10:29:27 2009 +0300
+++ b/conary/files.py	Wed Jan 21 10:30:09 2009 +0300
@@ -16,7 +16,6 @@
 import gzip
 import os
 import pwd
-import sha
 import socket
 import stat
 import string
@@ -26,6 +25,7 @@
 
 from conary import errors, streams
 from conary.lib import util, sha1helper, log
+from conary.lib.crypto import sha
 
 _FILE_FLAG_CONFIG = 1 << 0
 _FILE_FLAG_PATH_DEPENDENCY_TARGET = 1 << 1
diff -r b0074c59ad1e conary/lib/crypto.py
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/conary/lib/crypto.py	Wed Jan 21 10:30:09 2009 +0300
@@ -0,0 +1,18 @@
+
+__all__ = ["md5", "sha"]
+
+try:
+    hashlib = __import__("hashlib")
+    class md5:
+        @classmethod
+        def new(cls, string=''):
+            return hashlib.md5(string)
+    class sha:
+        @classmethod
+        def new(cls, string=''):
+            return hashlib.sha1(string)
+
+except ImportError:
+    from md5 import md5
+    from sha import sha
+
diff -r b0074c59ad1e conary/lib/openpgpfile.py
--- a/conary/lib/openpgpfile.py	Wed Jan 21 10:29:27 2009 +0300
+++ b/conary/lib/openpgpfile.py	Wed Jan 21 10:30:09 2009 +0300
@@ -17,9 +17,7 @@
 import errno
 import fcntl
 import itertools
-import md5
 import os
-import sha
 import stat
 import struct
 import sys
@@ -44,6 +42,7 @@
 
 from conary import constants
 from conary.lib import util
+from conary.lib.crypto import md5, sha
 
 # key types defined in RFC 2440 page 49
 PK_ALGO_RSA                  = 1
diff -r b0074c59ad1e conary/lib/sha1helper.py
--- a/conary/lib/sha1helper.py	Wed Jan 21 10:29:27 2009 +0300
+++ b/conary/lib/sha1helper.py	Wed Jan 21 10:30:09 2009 +0300
@@ -12,13 +12,13 @@
 # full details.
 #
 
-import sha
-import md5
 import os
 import stat
 import struct
 from Crypto.Hash import SHA256
 
+from conary.lib.crypto import md5, sha
+
 def sha1FileBin(path):
     oldmode = None
     mode = os.lstat(path)[stat.ST_MODE]
diff -r b0074c59ad1e conary/local/localrep.py
--- a/conary/local/localrep.py	Wed Jan 21 10:29:27 2009 +0300
+++ b/conary/local/localrep.py	Wed Jan 21 10:30:09 2009 +0300
@@ -13,13 +13,14 @@
 #
 
 import gzip
-import sha
 import zlib
 try:
     from cStringIO import StringIO
 except ImportError:
     from StringIO import StringIO
 
+from conary.lib.crypto import sha
+
 from conary.repository import errors, repository, datastore
 from conary.local import schema
 from conary import files
diff -r b0074c59ad1e conary/local/update.py
--- a/conary/local/update.py	Wed Jan 21 10:29:27 2009 +0300
+++ b/conary/local/update.py	Wed Jan 21 10:30:09 2009 +0300
@@ -20,7 +20,6 @@
 import itertools
 import os
 import select
-import sha
 import stat
 import sys
 import tempfile
@@ -31,6 +30,7 @@
 from conary.callbacks import UpdateCallback
 from conary.deps import deps
 from conary.lib import log, patch, sha1helper, util, fixedglob
+from conary.lib.crypto import sha
 from conary.local.errors import *
 from conary.local.journal import NoopJobJournal
 from conary.repository import changeset, filecontents
diff -r b0074c59ad1e conary/repository/datastore.py
--- a/conary/repository/datastore.py	Wed Jan 21 10:29:27 2009 +0300
+++ b/conary/repository/datastore.py	Wed Jan 21 10:30:09 2009 +0300
@@ -24,11 +24,11 @@
 import gzip
 import itertools
 import os
-import sha
 import tempfile
 
 from conary.lib import misc, util
 from conary.lib import sha1helper
+from conary.lib.crypto import sha
 from conary.repository import errors, filecontents
 
 class AbstractDataStore:
diff -r b0074c59ad1e conary/repository/netrepos/netauth.py
--- a/conary/repository/netrepos/netauth.py	Wed Jan 21 10:29:27 2009 +0300
+++ b/conary/repository/netrepos/netauth.py	Wed Jan 21 10:30:09 2009 +0300
@@ -12,7 +12,6 @@
 # full details.
 #
 import itertools
-import md5
 import os
 import time
 import urllib, urllib2
@@ -21,6 +20,7 @@
 from conary import conarycfg, versions
 from conary.repository import errors
 from conary.lib import sha1helper, tracelog
+from conary.lib.crypto import md5
 from conary.dbstore import sqlerrors
 from conary.repository.netrepos import items, versionops, accessmap
 from conary.server.schema import resetTable
