diff -r 8aefe5e505f5 Make.rules
--- a/Make.rules	Wed Oct 22 11:25:32 2008 -0400
+++ b/Make.rules	Wed Nov 05 16:24:58 2008 -0500
@@ -13,7 +13,7 @@
 # full details.
 #
 
-PYVER=`python -c 'import sys; print sys.version[0:3]'`
+PYVER=`python2.6 -c 'import sys; print sys.version[0:3]'`
 PYTHON = /usr/bin/python${PYVER}
 PYVERSION = $(shell $(PYTHON) -c 'import os, sys; print sys.version[:3]')
 PYINCLUDE = $(shell $(PYTHON) -c 'import os, sys; print os.sep.join((sys.prefix, "include", "python" + sys.version[:3]))')
diff -r 8aefe5e505f5 commands/conary
--- a/commands/conary	Wed Oct 22 11:25:32 2008 -0400
+++ b/commands/conary	Wed Nov 05 16:24:58 2008 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # -*- mode: python -*-
 #
 # Copyright (c) 2004-2007 rPath, Inc.
diff -r 8aefe5e505f5 commands/cvc
--- a/commands/cvc	Wed Oct 22 11:25:32 2008 -0400
+++ b/commands/cvc	Wed Nov 05 16:24:58 2008 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.6
 # -*- python -*-
 #
 # Copyright (c) 2004 rPath, Inc.
diff -r 8aefe5e505f5 conary/lib/numstream.c
--- a/conary/lib/numstream.c	Wed Oct 22 11:25:32 2008 -0400
+++ b/conary/lib/numstream.c	Wed Nov 05 16:24:58 2008 -0500
@@ -16,6 +16,10 @@
 
 #include <Python.h>
 #include <netinet/in.h>
+
+#if PY_MINOR_VERSION < 5
+typedef int Py_ssize_t
+#endif
 
 #include "cstreams.h"
 
@@ -273,7 +277,7 @@
     if (o->isNone)
         return PyString_FromString("");
 
-    return PyString_FromStringAndSize(&o->val, 1);
+    return (PyObject *) PyString_FromStringAndSize((const char *) &o->val, 1);
 }
 
 static inline PyObject * raw_LongLongStream_Freeze(LongLongStreamObject * o) {
@@ -330,7 +334,7 @@
 
     if (initObj && PyString_CheckExact(initObj)) {
         char * frozen;
-        int frozenLen;
+        Py_ssize_t frozenLen;
 
         PyString_AsStringAndSize(initObj, &frozen, &frozenLen);
         if (!raw_NumericStream_Thaw(o, frozen, frozenLen))
