diff -r ec4ab08378f9 conary/lib/http/opener.py
--- a/conary/lib/http/opener.py Mon Nov 05 10:31:04 2012 -0500
+++ b/conary/lib/http/opener.py Wed Nov 14 15:31:09 2012 -0500
@@ -114,6 +114,9 @@
             #fp = DecompressFileObj(fp)
             fp = util.decompressStream(fp)
             fp.seek(0)
+        if encoding == 'gzip':
+            fp = util.GzipFile(fileobj=fp)
+            fp.seek(0)
         return ResponseWrapper(fp, response)
 
     def _handleError(self, req, response):
