diff -r 8aefe5e505f5 conary/lib/pydeps.py
--- a/conary/lib/pydeps.py	Wed Oct 22 11:25:32 2008 -0400
+++ b/conary/lib/pydeps.py	Wed Nov 05 10:39:23 2008 -0500
@@ -58,15 +58,28 @@
         else:
             return modulefinder.ModuleFinder.scan_code(self, co, m)
 
-    def import_hook(self, name, caller=None, fromlist=None):
+    def import_hook(self, *args, **kwargs):
+        assert len(args) < 4
+
+        if len(args) > 1:
+            kwargs['caller'] = args[1]
+
+            if len(args) > 2:
+                kwargs['fromlist'] = args[2]
+
+            if len(args) > 3:
+                kwargs['level'] = args[3]
+
+            args = (args[0], )
+
         oldCaller = self.caller
-        if caller:
-            self.caller = caller.__file__
+        if 'caller' in kwargs:
+            self.caller = kwargs['caller'].__file__
         else:
             self.caller = None
 
         try:
-            modulefinder.ModuleFinder.import_hook(self, name, caller, fromlist)
+            modulefinder.ModuleFinder.import_hook(self, *args, **kwargs)
         finally:
             self.caller = oldCaller
 
@@ -216,7 +229,12 @@
 
         try:
             inspector(path)
-        except:
+        except Exception, e:
+            f = open('/tmp/pydeps.errors', 'a')
+            f.write(path)
+            f.write('\n%s' % e.__class__.__name__)
+            f.write('\n%s\n' % e)
+            f.close()
             putData(sys.stdout, '///invalid')
             continue
 
