--- Makefile.dist	Wed Mar 11 20:44:08 1992
+++ Makefile	Wed Oct 23 16:50:56 2002
@@ -50,17 +50,17 @@
 #  If you are running on a Unix System V machine, then you should
 #  uncomment the next definition.
 #
-#SYSV           = -DSYSV
+SYSV           = -DSYSV
 #=====================================================================
 #
 #  Default locations where mp files will be installed.
 #  You might wish to alter these values.
 #
-MPPATH     = /usr/local
+MPPATH     = /usr
 MPBINDIR   = $(MPPATH)/bin
-MPLIBDIR   = $(MPPATH)/lib
+MPLIBDIR   = $(MPPATH)/lib/mp
 MPMANDIR   = $(MPPATH)/man/man$(MPMANSECT)
-MPMANSECT  = l
+MPMANSECT  = 1
 #---------------------------------------------------------------------
 #  Alternate names for the mailp frontend
 #
@@ -100,10 +100,11 @@
 
 mp:	$(OBJS) mp.man mailp.man
 	$(CC) $(LDFLAGS) -o mp $(OBJS)
-	-sed -e s,LIBDIR,$(MPLIBDIR),g \
+	sed -e s,LIBDIR,$(MPLIBDIR),g \
 	     -e s,BINDIR,$(MPBINDIR),g mp.man > mp.1
-	-sed -e s,LIBDIR,$(MPLIBDIR),g \
+	sed -e s,LIBDIR,$(MPLIBDIR),g \
 	     -e s,BINDIR,$(MPBINDIR),g mailp.man > mailp.1
+	sed -e s,BINDIR,$(MPBINDIR),g mailp.in > mailp
 
 SHELL = /bin/sh
 install-a4: common-install
@@ -120,7 +121,7 @@
 	   @echo "**IMPORTANT NOTE**"
 	   @echo
 	   @echo "Mp is setup by default to work with US Letter sized paper."
-	   @echo "If you are using A4 sized paper, you should reinstall, by
+	   @echo "If you are using A4 sized paper, you should reinstall, by"
 	   @echo "doing a:"
 	   @echo
 	   @echo "make install-a4"
@@ -129,7 +130,7 @@
 common-install: $(BINARIES)
 #                                                     mp
 	   -cp mp $(MPBINDIR)
-	   -chmod 751 $(MPBINDIR)/mp
+	   -chmod 755 $(MPBINDIR)/mp
 	   -strip $(MPBINDIR)/mp
 #                                                     mp.1
 	   -cp mp.1 $(MPMANDIR)/mp.$(MPMANSECT)
@@ -166,7 +167,7 @@
 	   -chmod 644 $(MPLIBDIR)/mp.pro.ts.ps
 #                                                     mailp
 #
-	   -sed -e s,BINDIR,$(MPBINDIR),g mailp >$(MPBINDIR)/mailp
+	   -cp mailp $(MPBINDIR)/mailp
 	   -chmod 755 $(MPBINDIR)/mailp
 #
 #  NOTE: If you don't want the newsp, filep ... shell scripts installed,
@@ -176,7 +177,7 @@
 	   -@for name in $(MAILPNAMES); do \
 		echo -n $$name " "; \
 		rm -f $(MPBINDIR)/$$name; \
-		ln -s $(MPBINDIR)/mailp $(MPBINDIR)/$$name; \
+		ln $(MPBINDIR)/mailp $(MPBINDIR)/$$name; \
 	   done
 	   @echo
 
