--- libpng/libpng-1.2.12/configure.ac	2006-06-27 15:22:43.000000000 -0500
+++ libpng-1.2.12-asmfix/configure.ac	2006-08-28 13:04:47.000000000 -0500
@@ -66,11 +66,14 @@
   LIBPNG_DEFINES=-DPNG_CONFIGURE_LIBPNG\ -DPNG_NO_ASSEMBLER_CODE)
 if test "$LIBPNG_DEFINES"; then
     AC_MSG_RESULT(no)
+    LIBPNG_NO_ASSEMBLER=true
 else
     AC_MSG_RESULT(yes)
     LIBPNG_DEFINES=-DPNG_CONFIGURE_LIBPNG
+    LIBPNG_NO_ASSEMBLER=
 fi
 AC_SUBST(LIBPNG_DEFINES)
+AC_SUBST(LIBPNG_NO_ASSEMBLER)
 
 AC_MSG_CHECKING([if libraries can be versioned])
 GLD=`$LD --help < /dev/null 2>/dev/null | grep version-script`
--- libpng/libpng-1.2.12/Makefile.am	2006-06-27 15:22:43.000000000 -0500
+++ libpng-1.2.12-asmfix/Makefile.am	2006-08-28 13:04:47.000000000 -0500
@@ -88,6 +88,16 @@
 CLEANFILES= pngout.png libpng12.pc libpng12-config libpng.vers \
 libpng.sym
 
+pngconf.h.in:
+	cat pngconf.h > pngconf.h.in
+
+pngconf.h: pngconf.h.in
+	sed "/Makefile-supplied defines go here:/q" pngconf.h.in > pngconf.h
+	@if test -n "@LIBPNG_NO_ASSEMBLER@"; then \
+	echo "#define PNG_NO_ASSEMBLER_CODE" >> pngconf.h; \
+	fi
+	sed "1,/Makefile-supplied defines go here:/d" pngconf.h.in >> pngconf.h
+
 $(PNGLIB_BASENAME).pc: libpng.pc
 	cp libpng.pc $@
 
--- libpng/libpng-1.2.12/pngconf.h	2006-06-27 15:22:34.000000000 -0500
+++ libpng-1.2.12-asmfix/pngconf.h	2006-08-28 13:14:10.000000000 -0500
@@ -19,6 +19,10 @@
 
 #define PNG_1_2_X
 
+/* backported from libpng-1.4.0 */
+/* Makefile-supplied defines go here: */
+/* End of Makefile-supplied defines. */
+
 /* 
  * PNG_USER_CONFIG has to be defined on the compiler command line. This
  * includes the resource compiler for Windows DLL configurations.
