diff -r a7145884a9be data/js/ui.main.js
--- a/data/js/ui.main.js	Wed May 04 23:31:36 2011 +0800
+++ b/data/js/ui.main.js	Sat May 07 00:52:38 2011 +0800
@@ -381,8 +381,29 @@
             ui.Main.block_info[pagename].since_id 
                 = json_obj[tweet_count - 1].id_str;  
             var last_id = json_obj[0].id_str;
-            if (ui.Main.block_info[pagename].max_id == null)
+            if (ui.Main.block_info[pagename].max_id == null) {
                 ui.Main.block_info[pagename].max_id = last_id - 1;
+            }
+            if (pagename == '#home_timeline' || pagename == '#mentions' || pagename == '#direct_messages_inbox') {
+                var current_profile = conf.get_current_profile();
+                var prefs = current_profile.preferences;
+                var latest_id = prefs[container.pagename + '_latest_id'] || "0";
+                var last_id = json_obj[json_obj.length - 1].id_str;
+                if (util.compare_id(last_id, latest_id) > 0) {
+                    prefs[container.pagename + '_latest_id'] = last_id;
+                    conf.save_prefs(conf.current_name);
+                }
+                var i = json_obj.length - 1;
+                for ( ; i >= 0 ; i -= 1) {
+                if (util.compare_id(json_obj[i].id_str, latest_id) <= 0) {
+                        break;
+                    }
+                }
+                tweet_count = json_obj.length - i - 1;
+                if (tweet_count == 0) {
+                    return;
+                }
+            }
         }
         if (ui.Main.block_info[pagename].use_notify) {
             switch (ui.Main.block_info[pagename].use_notify_type) {
@@ -411,12 +432,12 @@
                         $(window).dequeue('_notify');
                     });
                 }
-                for ( ; 0 <= i && cnt < 4; i -= 1, cnt += 1) {
+                for ( ; 0 <= i && cnt < 4 && cnt < tweet_count; i -= 1, cnt += 1) {
                     load(i)
                 }
                 $(window).queue('_notify', proc);
                 $(window).dequeue('_notify');
-                if (3 < json_obj.length) {
+                if (3 < tweet_count) {
                     hotot_notify(
                           "Update page " + pagename
                         , "and " 
