diff -ur bootsplash-3.1.orig/Scripts/bootanim bootsplash-3.1/Scripts/bootanim
--- bootsplash-3.1.orig/Scripts/bootanim	2005-04-18 10:08:44.000000000 -0400
+++ bootsplash-3.1/Scripts/bootanim	2005-04-18 10:10:45.000000000 -0400
@@ -43,14 +43,14 @@
 THEME="No theme selected"
 test -f /etc/sysconfig/bootsplash && . /etc/sysconfig/bootsplash
 
-if [ ! -d /etc/bootsplash/themes/$THEME ]
+if [ ! -d /usr/share/bootsplash/themes/$THEME ]
 then
-  echo "$0: could not find theme $THEME in /etc/bootsplash/themes."
+  echo "$0: could not find theme $THEME in /usr/share/bootsplash/themes."
   exit 0
 fi
 
 MODE=`/sbin/fbresolution 2>/dev/null`
-DIRECTORY=/etc/bootsplash/themes/$THEME/animations
+DIRECTORY=/usr/share/bootsplash/themes/$THEME/animations
 OPTIONS="-b -c 1"
 FILES=""
 
@@ -122,8 +122,8 @@
 function box() { true; } # ignore box descriptions in the config file
 
 # Sourcing config file
-if [ -f /etc/bootsplash/themes/$THEME/config/bootsplash-$MODE.cfg ]; then
-    . /etc/bootsplash/themes/$THEME/config/bootsplash-$MODE.cfg
+if [ -f /usr/share/bootsplash/themes/$THEME/config/bootsplash-$MODE.cfg ]; then
+    . /usr/share/bootsplash/themes/$THEME/config/bootsplash-$MODE.cfg
 else
     echo "$0: No config file found for theme $THEME ($MODE)."
     exit 1
