--- shell/e-user-creatable-items-handler.c.orig	Wed Jan 18 14:28:13 2006
+++ shell/e-user-creatable-items-handler.c	Wed Jan 18 13:16:54 2006
@@ -646,7 +646,7 @@ gtk_menu_item_func (EUserCreatableItemsH
 					    handler->priv->accel_group,
 					    'n', GDK_CONTROL_MASK,
 					    GTK_ACCEL_VISIBLE);
-	} else {
+	} else if (item->shortcut != '\0') {
 		gtk_widget_add_accelerator (menuitem, "activate",
 					    handler->priv->accel_group,
 					    item->shortcut,
