--- evolution-2.0.2/camel/providers/local/camel-local-folder.c.servicedir	2004-12-07 13:55:58.709528968 -0500
+++ evolution-2.0.2/camel/providers/local/camel-local-folder.c	2004-12-07 13:56:25.322483184 -0500
@@ -390,8 +390,8 @@
 					/* $HOME relative path + protocol string */
 					folder->description = g_strdup_printf(_("~%%s (%%s)"), path+strlen(tmp),
 									      ((CamelService *)folder->parent_store)->url->protocol);
-				else if ((tmp = "/var/spool/mail") && strncmp(tmp, path, strlen(tmp)) == 0)
-					/* /var/spool/mail relative path + protocol */
+				else if ((tmp = "%(servicedir)s/spool/mail") && strncmp(tmp, path, strlen(tmp)) == 0)
+					/* %(servicedir)s/spool/mail relative path + protocol */
 					folder->description = g_strdup_printf(_("mailbox:%%s (%%s)"), path+strlen(tmp),
 									      ((CamelService *)folder->parent_store)->url->protocol);
 				else if ((tmp = "/var/mail") && strncmp(tmp, path, strlen(tmp)) == 0)
