From 8dd741057f7b935d8e7c327017d850388ba77767 Mon Sep 17 00:00:00 2001
From: Stephan Arts <stephan@xfce.org>
Date: Sun, 04 Mar 2012 10:22:22 +0000
Subject: Provide mime-type when loading image

Do not let gdk guess it, it makes mistakes.
---
diff --git a/src/image_viewer.c b/src/image_viewer.c
index aed2fa2..f954b88 100644
--- a/src/image_viewer.c
+++ b/src/image_viewer.c
@@ -1525,7 +1525,12 @@ rstto_image_viewer_load_image (
         viewer->priv->transaction = NULL;
     }
 
-    transaction->loader = gdk_pixbuf_loader_new();
+    transaction->loader = gdk_pixbuf_loader_new_with_mime_type (rstto_file_get_content_type (file), NULL);
+    /* HACK HACK HACK */
+    if (transaction->loader == NULL)
+    {
+        transaction->loader = gdk_pixbuf_loader_new();
+    }
     transaction->cancellable = g_cancellable_new();
     transaction->buffer = g_new0 (guchar, RSTTO_IMAGE_VIEWER_BUFFER_SIZE);
     transaction->file = file;
--
cgit v0.9.0.2
