From 81defc54001ac65dd3f955bb8206fb23542a70e9 Mon Sep 17 00:00:00 2001
From: Stephan Arts <stephan@xfce.org>
Date: Sat, 03 Mar 2012 20:30:57 +0000
Subject: Clear tooltip if image is closed

---
diff --git a/src/image_viewer.c b/src/image_viewer.c
index 7e5baf0..aed2fa2 100644
--- a/src/image_viewer.c
+++ b/src/image_viewer.c
@@ -1501,6 +1501,8 @@ rstto_image_viewer_set_file (
                     widget->window,
                     NULL,
                     FALSE);
+
+            gtk_widget_set_tooltip_text (GTK_WIDGET (viewer), NULL);
         }
     }
 }
--
cgit v0.9.0.2
