=== modified file 'src/main.c'
--- src/main.c	2008-09-24 11:03:29 +0000
+++ src/main.c	2008-10-08 16:52:57 +0000
@@ -223,7 +223,7 @@
 
   /* Set the theme */
   error = NULL;
-  gconf_client_set_string (client, THEME, "Human-Murrine-Netbook", &error);
+  gconf_client_set_string (client, THEME, "Foresight", &error);
   if (error)
   {
     g_warning ("Can't change the theme: %s", 
@@ -324,7 +324,7 @@
 
   /* Set the theme */
   error = NULL;
-  gconf_client_set_string (client, THEME, "Human", &error);
+  gconf_client_set_string (client, THEME, "Foresight", &error);
   if (error)
   {
     g_warning ("Can't change the theme: %s", 

