diff -r dcb334be7daa dulwich/pack.py
--- a/dulwich/pack.py	Thu Apr 30 13:54:33 2009 -0700
+++ b/dulwich/pack.py	Thu Apr 30 18:47:29 2009 -0400
@@ -500,8 +500,10 @@
                 offset += total_size
                 if progress:
                     progress(i, num)
-        finally:
             map.close()
+        except:
+            map.close()
+            raise
   
     def iterentries(self, ext_resolve_ref=None, progress=None):
         found = {}
