diff -Naur mono-zeroconf-0.7.3.orig/src/mono-zeroconf.pc.in mono-zeroconf-0.7.3/src/mono-zeroconf.pc.in
--- mono-zeroconf-0.7.3.orig/src/mono-zeroconf.pc.in	2007-10-09 12:42:44.000000000 -0400
+++ mono-zeroconf-0.7.3/src/mono-zeroconf.pc.in	2007-12-27 09:37:47.000000000 -0500
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 
 Name: Mono.Zeroconf
 Description: Mono library for universal cross platform Zeroconf support via a unified API with built in support for mdnsd and optional support for Avahi.
