--- /usr/share/xemacs/xemacs-packages/lisp/vm/vm-crypto.el	2003-04-01 18:41:40.000000000 -0500
+++ vm-crypto.el	2008-02-26 10:30:11.000000000 -0500
@@ -191,6 +191,7 @@
 	    (insert "RNDfile = " vm-stunnel-random-data-file "\n")
 	    (insert "RNDoverwrite = no\n")
 	    (insert "connect = " (format "%s:%s" host port) "\n")
+	    (insert "debug = local6.crit\n")
 	    (write-region (point-min) (point-max) workfile nil 0))
 	(and work-buffer (kill-buffer work-buffer)))
       (list workfile) )))
--- /usr/share/xemacs/xemacs-packages/lisp/vm/vm-imap.el	2003-09-02 20:43:45.000000000 -0400
+++ vm-imap.el	2007-06-12 20:46:01.000000000 -0400
@@ -1034,10 +1034,10 @@
     t ))
 
 (defun vm-imap-delete-message (process n)
-  (vm-imap-send-command process (format "STORE %d +FLAGS.SILENT (\\Deleted)"
+  (vm-imap-send-command process (format "STORE %d +FLAGS (\\Deleted)"
 					n))
   (if (null (vm-imap-read-ok-response process))
-      (vm-imap-protocol-error "STORE ... +FLAGS.SILENT (\\Deleted) failed")))
+      (vm-imap-protocol-error "STORE ... +FLAGS (\\Deleted) failed")))
 
 (defun vm-imap-get-message-size (process n)
   (let ((list nil)
@@ -1797,7 +1797,9 @@
     (vm-imap-check-connection process)
     (save-excursion
       (set-buffer (process-buffer process))
-      (vm-imap-send-command process "LIST \"\" \"*\"")
+      (vm-imap-send-command process (format "LIST %s \"*\""
+					    (vm-imap-quote-string
+					     vm-imap-server-folder-base)))
       (setq need-ok t)
       (while need-ok
 	(setq response (vm-imap-read-response process))
@@ -1820,7 +1822,7 @@
 		 (if (memq (car p) '(atom string))
 		     (setq c-list (cons (buffer-substring (nth 1 p) (nth 2 p))
 					c-list)))))))
-      c-list )))
+      (cons "inbox" c-list ))))
 
 (defun vm-imap-read-boolean-response (process)
   (let ((need-ok t) retval response)
--- /usr/share/xemacs/xemacs-packages/lisp/vm/vm-vars.el	2003-09-02 20:43:45.000000000 -0400
+++ vm-vars.el	2005-09-28 18:21:53.000000000 -0400
@@ -514,6 +514,11 @@
 variable."
   :type '(choice (const nil) directory))
 
+(defcustom vm-imap-server-folder-base nil
+  "*Directory on server where IMAP folder hierarchy begins.
+Among multiple servers, one size fits all (for now)."
+  :type '(choice (const nil) directory))
+
 (defcustom vm-auto-get-new-mail t
   "*Non-nil value causes VM to automatically move mail from spool files
 to a mail folder when the folder is first visited.  Nil means
--- /usr/share/xemacs/xemacs-packages/lisp/vm/vm-version.el	2003-09-02 20:43:45.000000000 -0400
+++ vm-version.el	2007-06-12 11:17:30.000000000 -0400
@@ -1,7 +1,7 @@
 ;;(provide 'vm)
 ;;(provide 'vm-version)
 
-(defconst vm-version "7.17"
+(defconst vm-version "7.17-Up.3"
   "Version number of VM.")
 
 (defun vm-version ()
