--- xemacs-21.4.19/lib-src/Makefile.in.in.upe	2004-01-26 21:58:10.000000000 -0500
+++ xemacs-21.4.19/lib-src/Makefile.in.in	2006-08-31 15:28:21.000000000 -0400
@@ -48,12 +48,12 @@
 
 ## ==================== Where To Install Things ====================
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-bindir=@bindir@
-libdir=@libdir@
+prefix=$(DESTDIR)/@prefix@
+exec_prefix=$(DESTDIR)/@exec_prefix@
+bindir=$(DESTDIR)/@bindir@
+libdir=$(DESTDIR)/@libdir@
 srcdir=@srcdir@
-datadir=@datadir@
+datadir=$(DESTDIR)/@datadir@
 instvardir=@instvardir@
 top_srcdir=@top_srcdir@
 archlibdir=@archlibdir@
--- xemacs-21.4.19/netinstall/Makefile.in.in.upe	2002-08-20 07:36:04.000000000 -0400
+++ xemacs-21.4.19/netinstall/Makefile.in.in	2006-08-31 15:30:43.000000000 -0400
@@ -49,12 +49,12 @@
 
 ## ==================== Where To Install Things ====================
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-bindir=@bindir@
-libdir=@libdir@
+prefix=$(DESTDIR)/@prefix@
+exec_prefix=$(DESTDIR)/@exec_prefix@
+bindir=$(DESTDIR)/@bindir@
+libdir=$(DESTDIR)/@libdir@
 srcdir=@srcdir@
-datadir=@datadir@
+datadir=$(DESTDIR)/@datadir@
 instvardir=@instvardir@
 top_srcdir=@top_srcdir@
 archlibdir=@archlibdir@
--- xemacs-21.4.19/src/Makefile.in.in.upe	2005-01-30 21:55:02.000000000 -0500
+++ xemacs-21.4.19/src/Makefile.in.in	2006-08-31 15:39:24.000000000 -0400
@@ -61,7 +61,7 @@
 etcdir = ../etc/
 
 ## Here are the things that we expect ../configure to edit.
-prefix=@prefix@
+prefix=$(DESTDIR)/@prefix@
 srcdir=@srcdir@
 blddir=@blddir@
 version=@version@
--- xemacs-21.4.19/Makefile.in.in.upe	2005-01-30 21:54:47.000000000 -0500
+++ xemacs-21.4.19/Makefile.in.in	2006-08-31 15:41:56.000000000 -0400
@@ -97,7 +97,7 @@
 ## subdirectories of this directory.  The default values for many of
 ## the variables below are expressed in terms of this one, so you may
 ## not need to change them.  This defaults to /usr/local.
-prefix=@prefix@
+prefix=$(DESTDIR)/@prefix@
 
 ## Variable data (as per each program update) goes here
 ## The default is ${PROGNAME}
@@ -108,15 +108,15 @@
 instvardir=@instvardir@
 
 ## Like `prefix', but used for architecture-specific files.
-exec_prefix=@exec_prefix@
+exec_prefix=$(DESTDIR)/@exec_prefix@
 
 ## Where to install XEmacs and other binaries that people will want to
 ## run directly (like etags).
-bindir=@bindir@
+bindir=$(DESTDIR)/@bindir@
 
 ## Where to install architecture-independent data files.
 ## ${lispdir} and ${etcdir} are subdirectories of this.
-datadir=@datadir@
+datadir=$(DESTDIR)/@datadir@
 
 ## Where to find XEmacs packages.
 pkgdir=@pkgdir@
@@ -129,17 +129,17 @@
 ## Where to install and expect executable files to be run by XEmacs
 ## rather than directly by users, and other architecture-dependent data
 ## ${archlibdir} is a subdirectory of this.
-libdir=@libdir@
+libdir=$(DESTDIR)/@libdir@
 
 ## Where to install XEmacs's man pages, and what extension they should have.
-mandir=@mandir@
+mandir=$(DESTDIR)/@mandir@
 manext=.1
 
 ## Where to install and expect the info files describing XEmacs.  In the
 ## past, this defaulted to a subdirectory of ${prefix}/lib/xemacs, but
 ## since there are now many packages documented with the texinfo
 ## system, it is inappropriate to imply that it is part of XEmacs.
-infodir=@infodir@
+infodir=$(DESTDIR)/@infodir@
 
 ## Document me.
 ## See callproc.c for code which references this.
