--- /home/juphoff/configure.upe	2005-12-23 19:47:36.000000000 -0500
+++ xemacs-21.4.19/configure	2006-08-28 21:36:35.000000000 -0400
@@ -213,9 +213,14 @@
 prefix='/usr/local'
 exec_prefix='${prefix}'
 bindir='${exec_prefix}/bin'
+sbindir='${exec_prefix}/sbin'
 datadir='${prefix}/lib'
 statedir='${prefix}/lib'
+sharedstatedir='${prefix}/com'
+localstatedir='/var'
 libdir='${exec_prefix}/lib'
+libexecdir='${exec_prefix}/libexec'
+includedir='${prefix}/include'
 mandir='${prefix}/man/man1'
 inststaticdir='${PROGNAME}'
 instvardir='${PROGNAME}-${version}'
@@ -229,6 +234,7 @@
 pkgdir='${datadir}/${instvardir}/lisp'
 package_path=''
 etcdir='${datadir}/${instvardir}/etc'
+sysconfdir='/etc'
 archlibdir='${libdir}/${instvardir}/${configuration}'
 docdir='${archlibdir}'
 with_netinstall="no"
@@ -584,8 +590,8 @@
 	;;
 
 				
-		prefix | exec_prefix | bindir | datadir | statedir | libdir | \
-	mandir | infodir | infopath | lispdir | etcdir | pkgdir | \
+		prefix | exec_prefix | bindir | sbindir | datadir | statedir | localstatedir | sharedstatedir | libdir | libexecdir | includedir | \
+	mandir | infodir | infopath | lispdir | etcdir | sysconfdir | pkgdir | \
 	archlibdir | docdir | package_path | moduledir )
 	   	   if test "$valomitted" = "yes"; then
 	     if test "$#" = 0; then
@@ -16907,6 +16913,7 @@
 s%@SITEMODULEDIR_USER_DEFINED@%$SITEMODULEDIR_USER_DEFINED%g
 s%@SITEMODULEDIR@%$SITEMODULEDIR%g
 s%@etcdir@%$etcdir%g
+s%@sysconfdir@%$sysconfdir%g
 s%@ETCDIR_USER_DEFINED@%$ETCDIR_USER_DEFINED%g
 s%@ETCDIR@%$ETCDIR%g
 s%@docdir@%$docdir%g
