--- xemacs-21.5.27/etc/xemacs.1.paths	2001-04-12 14:20:52.000000000 -0400
+++ xemacs-21.5.27/etc/xemacs.1	2006-08-10 21:46:10.000000000 -0400
@@ -293,7 +293,7 @@
 Sets the color of the text.
 
 See the file
-.I /usr/lib/X11/rgb.txt
+.I /usr/share/X11/rgb.txt
 for a list of valid
 color names.
 .TP
--- xemacs-21.5.27/etc/Emacs.ad.paths	2003-03-27 07:57:44.000000000 -0500
+++ xemacs-21.5.27/etc/Emacs.ad	2006-08-10 21:46:10.000000000 -0400
@@ -9,7 +9,7 @@
 ! changes to .../etc/Emacs.ad made after XEmacs has been built will have no
 ! effect.
 ! 
-! However, you may copy .../etc/Emacs.ad to /usr/lib/X11/app-defaults/Emacs
+! However, you may copy .../etc/Emacs.ad to /usr/share/X11/app-defaults/Emacs
 ! (or whatever the standard app-defaults directory is at your site) to cause
 ! it to be consulted at run-time.  (Do this only for site-wide customizations:
 ! personal customizations should be put into ~/.Xresources instead.)
--- xemacs-21.5.27/lisp/find-paths.el.paths	2005-01-15 10:17:36.000000000 -0500
+++ xemacs-21.5.27/lisp/find-paths.el	2006-08-10 21:47:24.000000000 -0400
@@ -220,8 +220,8 @@
 				  roots
 				  (file-name-as-directory
 				   (paths-construct-path (list
-							  "lib"
-							  emacs-program-name)))
+							  "share"
+							  (construct-emacs-version-name))))
 				  base
 				  envvar default))
 
--- xemacs-21.5.27/lisp/setup-paths.el.paths	2006-08-10 21:47:33.000000000 -0400
+++ xemacs-21.5.27/lisp/setup-paths.el	2006-08-10 21:47:55.000000000 -0400
@@ -94,10 +94,10 @@
   (or
    ;; installed
    (paths-file-readable-directory-p (paths-construct-path (list directory
-								"lib"
+								"lib"
 								emacs-program-name)))
    (paths-file-readable-directory-p (paths-construct-path (list directory
-								"lib"
+								"lib"
 								(construct-emacs-version-name))))
    ;; in-place or windows-nt
    (and
--- xemacs-21.5.27/configure.ac.libdir	2006-08-10 23:17:00.000000000 -0400
+++ xemacs-21.5.27/configure.ac	2006-08-10 23:17:41.000000000 -0400
@@ -625,10 +625,10 @@
 	[AC_DEFINE(LISPDIR_USER_DEFINED)], [with_lispdir='${datadir}/${instvardir}/lisp'])
 XE_MERGED_ARG([archlibdir],
 	AS_HELP_STRING([--with-archlibdir=DIR],[]),
-	[AC_DEFINE(ARCHLIBDIR_USER_DEFINED)], [with_archlibdir='${libdir}/${instvardir}/${configuration}'])
+	[AC_DEFINE(ARCHLIBDIR_USER_DEFINED)], [with_archlibdir='${libdir}/${instvardir}'])
 XE_MERGED_ARG([moduledir],
 	AS_HELP_STRING([--with-moduledir=DIR],[]),
-	[AC_DEFINE(MODULEDIR_USER_DEFINED)], [with_moduledir='${libdir}/${instvardir}/${configuration}/modules'])
+	[AC_DEFINE(MODULEDIR_USER_DEFINED)], [with_moduledir='${libdir}/${instvardir}/modules'])
 XE_MERGED_ARG([etcdir],
 	AS_HELP_STRING([--with-etcdir=DIR],[]),
 	[AC_DEFINE(ETCDIR_USER_DEFINED)], [with_etcdir='${datadir}/${instvardir}/etc'])
--- xemacs-21.5.27/configure.libdir	2006-08-10 23:15:54.000000000 -0400
+++ xemacs-21.5.27/configure	2006-08-10 23:17:52.000000000 -0400
@@ -2122,7 +2122,7 @@
 _ACEOF
 
 else
-  with_archlibdir='${libdir}/${instvardir}/${configuration}'
+  with_archlibdir='${libdir}/${instvardir}'
 fi;
 # If --with-moduledir or --without-moduledir were given then copy the value to the
 # equivalent enable_moduledir variable.
@@ -2143,7 +2143,7 @@
 _ACEOF
 
 else
-  with_moduledir='${libdir}/${instvardir}/${configuration}/modules'
+  with_moduledir='${libdir}/${instvardir}/modules'
 fi;
 # If --with-etcdir or --without-etcdir were given then copy the value to the
 # equivalent enable_etcdir variable.
