--- usr/share/xemacs/xemacs-packages/lisp/mailcrypt/mc-gpg.el.jau	2002-10-03 21:58:28.000000000 -0400
+++ usr/share/xemacs/xemacs-packages/lisp/mailcrypt/mc-gpg.el	2007-04-05 12:08:41.000000000 -0400
@@ -931,7 +931,8 @@
 	  ;; in particular, they cause an infinite loop. If the key doesn't
 	  ;; have a passphrase, the decryption should have worked the first
 	  ;; time around.
-	  ))
+	  )
+      (setq passwd ""))			; Suppress 'pinentry' on first pass.
     (setq args '("--batch"))
     (if mc-gpg-alternate-keyring
 	(setq args (append args (list "--keyring" mc-gpg-alternate-keyring))))
