--- xemacs-21.4.19/configure.archpaths	2006-08-31 16:30:25.000000000 -0400
+++ xemacs-21.4.19/configure	2006-08-31 16:31:17.000000000 -0400
@@ -228,14 +228,14 @@
 infopath=''
 install_pp=''
 lispdir='${datadir}/${instvardir}/lisp'
-moduledir='${libdir}/${instvardir}/${configuration}/modules'
+moduledir='${libdir}/${instvardir}/modules'
 sitelispdir='${datadir}/${inststaticdir}/site-lisp'
 sitemoduledir='${libdir}/${inststaticdir}/site-modules'
 pkgdir='${datadir}/${instvardir}/lisp'
 package_path=''
 etcdir='${datadir}/${instvardir}/etc'
 sysconfdir='/etc'
-archlibdir='${libdir}/${instvardir}/${configuration}'
+archlibdir='${libdir}/${instvardir}'
 docdir='${archlibdir}'
 with_netinstall="no"
 with_prefix='yes'
