diff -ur firstboot-1.3.33.orig/src/modules/create_user.py firstboot-1.3.33/src/modules/create_user.py
--- firstboot-1.3.33.orig/src/modules/create_user.py	2004-10-15 13:38:33.000000000 -0400
+++ firstboot-1.3.33/src/modules/create_user.py	2005-02-26 14:44:44.286358984 -0500
@@ -230,7 +230,7 @@
         args = [ "/usr/bin/chfn", "-f", fullName , username ]
         executil.execWithRedirect(args[0], args, stdout = None, stderr = None)
 
-        args = [ "/usr/sbin/usermod", "-p", cryptPassword(password) , username ]
+        args = [ "/usr/sbin/usermod -G wheel", "-p", cryptPassword(password) , username ]
         executil.execWithRedirect(args[0], args, stdout = None, stderr = None)
 
         return 0
