diff --git a/src/task-item.c b/src/task-item.c
--- a/src/task-item.c
+++ b/src/task-item.c
@@ -205,8 +205,6 @@
                                    WnckWindow *window,
                                    gint size)
 {
-  GdkPixbuf *pbuf;
-  
   g_return_val_if_fail (WNCK_IS_WINDOW (window), NULL);
   
   if (wnck_window_has_icon_name (window))
@@ -221,6 +219,7 @@
                                                       size,
                                                       GTK_ICON_LOOKUP_FORCE_SIZE,
                                                       NULL);
+      GdkPixbuf *pbuf;
       pbuf = gdk_pixbuf_copy (internal);
       g_object_unref (internal);
     }
