diff -r 07abf2519454 data/GNOME_WindowPicker.server.in.in
--- a/data/GNOME_WindowPicker.server.in.in	Fri Jun 13 15:16:24 2008 -0400
+++ b/data/GNOME_WindowPicker.server.in.in	Fri Jun 13 15:41:06 2008 -0400
@@ -1,13 +1,13 @@
 <oaf_info>
 <oaf_server iid="OAFIID:GNOME_WindowPicker_Factory" type="exe"
-            location="@LIBEXECDIR@/window-picker-applet/window-picker-applet">
+            location="/usr/libexec/window-picker-applet">
 
         <oaf_attribute name="repo_ids" type="stringv">
                 <item value="IDL:Bonobo/GenericFactory:1.0"/>
                 <item value="IDL:Bonobo/Unknown:1.0"/>
         </oaf_attribute>
-        <oaf_attribute name="name" type="string" _value="Window Picker"/>
-        <oaf_attribute name="description" type="string" _value="Window Picker"/>
+        <oaf_attribute name="name" type="string" value="Window Picker"/>
+        <oaf_attribute name="description" type="string" value="Window Picker"/>
 </oaf_server>
 
 <oaf_server iid="OAFIID:GNOME_WindowPicker" type="factory" location="OAFIID:GNOME_WindowPicker_Factory">
@@ -17,8 +17,10 @@
                 <item value="IDL:Bonobo/Control:1.0"/>
                 <item value="IDL:Bonobo/Unknown:1.0"/>
         </oaf_attribute>
-        <oaf_attribute name="name" type="string" _value="Window Picker"/>
-        <oaf_attribute name="description" type="string" _value="Window Picker"/>
+
+        <oaf_attribute name="name" type="string" value="Window Picker"/>
+        <oaf_attribute name="description" type="string" value="Window Picker"/>
         <oaf_attribute name="panel:icon" type="string" value="preferences-system-windows"/>
 </oaf_server>
 </oaf_info>
+
