--- net-snmp-5.4.1/configure.orig	2007-08-07 13:31:31.000000000 +0200
+++ net-snmp-5.4.1/configure	2007-08-07 13:33:29.000000000 +0200
@@ -27922,7 +27922,7 @@ if test "${with_libwrap+set}" = set; the
   if test "$with_libwrap" != "no"; then
     if test "$with_libwrap" != "yes"; then
       CPPFLAGS="${CPPFLAGS} -I$with_libwrap/include"
-      LDFLAGS="${LDFLAGS} -L$with_libwrap/lib"
+      LDFLAGS="${LDFLAGS} -L$with_libwrap"
     fi
 
     _libs=${LIBS}
@@ -28126,7 +28126,7 @@ echo "${ECHO_T}yes" >&6
 #define NETSNMP_USE_LIBWRAP 1
 _ACEOF
 
-	    test "$with_libwrap" != no -a "$with_libwrap" != yes && _wraplibs="-L$with_libwrap/lib"
+	    test "$with_libwrap" != no -a "$with_libwrap" != yes && _wraplibs="-L$with_libwrap"
 	    _wraplibs="$_wraplibs -lwrap"
 else
   echo "$as_me: failed program was:" >&5
@@ -28354,7 +28354,7 @@ echo "${ECHO_T}yes" >&6
 #define NETSNMP_USE_LIBWRAP 1
 _ACEOF
 
-			test "$with_libwrap" != no -a "$with_libwrap" != yes && _wraplibs="-L$with_libwrap/lib"
+			test "$with_libwrap" != no -a "$with_libwrap" != yes && _wraplibs="-L$with_libwrap"
 			_wraplibs="$_wraplibs -lwrap -lnsl"
 else
   echo "$as_me: failed program was:" >&5
--- net-snmp-5.4.1/configure.in.orig	2007-07-27 19:02:00.000000000 +0200
+++ net-snmp-5.4.1/configure.in	2007-08-07 13:33:55.000000000 +0200
@@ -2714,7 +2714,7 @@ AC_ARG_WITH(libwrap,
   if test "$with_libwrap" != "no"; then
     if test "$with_libwrap" != "yes"; then
       CPPFLAGS="${CPPFLAGS} -I$with_libwrap/include"
-      LDFLAGS="${LDFLAGS} -L$with_libwrap/lib"
+      LDFLAGS="${LDFLAGS} -L$with_libwrap"
     fi
 
     _libs=${LIBS}
@@ -2732,7 +2732,7 @@ AC_ARG_WITH(libwrap,
 
 	    AC_MSG_RESULT([yes])
 	    AC_DEFINE(NETSNMP_USE_LIBWRAP)
-	    test "$with_libwrap" != no -a "$with_libwrap" != yes && _wraplibs="-L$with_libwrap/lib"
+	    test "$with_libwrap" != no -a "$with_libwrap" != yes && _wraplibs="-L$with_libwrap"
 	    _wraplibs="$_wraplibs -lwrap"],
         [
 	    AC_MSG_RESULT([no])
@@ -2748,7 +2748,7 @@ AC_ARG_WITH(libwrap,
 
 			AC_MSG_RESULT(yes)
 			AC_DEFINE(NETSNMP_USE_LIBWRAP)
-			test "$with_libwrap" != no -a "$with_libwrap" != yes && _wraplibs="-L$with_libwrap/lib"
+			test "$with_libwrap" != no -a "$with_libwrap" != yes && _wraplibs="-L$with_libwrap"
 			_wraplibs="$_wraplibs -lwrap -lnsl",
 
 			AC_MSG_RESULT(no)
