--- branches/V5-4-patches/net-snmp/snmplib/snmpUDPIPv6Domain.c	2007/09/29 06:26:51	16712
+++ branches/V5-4-patches/net-snmp/snmplib/snmpUDPIPv6Domain.c	2007/09/29 13:10:22	16713
@@ -535,6 +535,13 @@
              * Fall through.  
              */
         }
+
+        if (peername[0] == '\0') {
+          DEBUGMSGTL(("netsnmp_sockaddr_in6", "empty hostname\n"));
+          free(peername);
+          return 0;
+        }
+
 #if HAVE_GETADDRINFO
         memset(&hint, 0, sizeof hint);
         hint.ai_flags = 0;
@@ -545,10 +552,11 @@
         err = getaddrinfo(peername, NULL, &hint, &addrs);
         if (err != 0) {
 #if HAVE_GAI_STRERROR
-            snmp_log(LOG_ERR, "getaddrinfo: %s %s\n", peername,
+            snmp_log(LOG_ERR, "getaddrinfo(\"%s\", NULL, ...): %s\n", peername,
                      gai_strerror(err));
 #else
-            snmp_log(LOG_ERR, "getaddrinfo: %s (error %d)\n", peername, err);
+            snmp_log(LOG_ERR, "getaddrinfo(\"%s\", NULL, ...): (error %d)\n",
+                     peername, err);
 #endif
             free(peername);
             return 0;

