diff -ur gtranslator-1.9.5.orig/data/desktop/Makefile.am gtranslator-1.9.5/data/desktop/Makefile.am
--- gtranslator-1.9.5.orig/data/desktop/Makefile.am	2009-04-11 05:03:51.000000000 +0400
+++ gtranslator-1.9.5/data/desktop/Makefile.am	2009-05-30 16:41:48.000000000 +0400
@@ -22,9 +22,9 @@
 CLEANFILES = $(Application_DATA)
 
 UPDATE_DESKTOP = \
-	if [ -f $(Applicationdir)/defaults.list ] ; then \
+	if [ -f $(DESTDIR)$(Applicationdir)/defaults.list ] ; then \
 		if which update-desktop-database>/dev/null 2>&1 ; then \
-			update-desktop-database $(Applicationdir); \
+			update-desktop-database $(DESTDIR)$(Applicationdir); \
 		fi \
 	fi
 
@@ -32,4 +32,4 @@
 	$(UPDATE_DESKTOP)
 
 uninstall-hook:
-	$(UPDATE_DESKTOP) 
\ В конце файла нет новой строки
+	$(UPDATE_DESKTOP) 
diff -ur gtranslator-1.9.5.orig/data/desktop/Makefile.in gtranslator-1.9.5/data/desktop/Makefile.in
--- gtranslator-1.9.5.orig/data/desktop/Makefile.in	2009-04-11 05:38:53.000000000 +0400
+++ gtranslator-1.9.5/data/desktop/Makefile.in	2009-05-30 16:42:04.000000000 +0400
@@ -252,9 +252,9 @@
 
 CLEANFILES = $(Application_DATA)
 UPDATE_DESKTOP = \
-	if [ -f $(Applicationdir)/defaults.list ] ; then \
+	if [ -f $(DESTDIR)$(Applicationdir)/defaults.list ] ; then \
 		if which update-desktop-database>/dev/null 2>&1 ; then \
-			update-desktop-database $(Applicationdir); \
+			update-desktop-database $(DESTDIR)$(Applicationdir); \
 		fi \
 	fi
 
