diff -r fa66f9729dc9 Makefile
--- a/Makefile	Wed Feb 20 00:48:44 2008 -0500
+++ b/Makefile	Wed Feb 20 00:50:30 2008 -0500
@@ -11,10 +11,13 @@ all:
 
 # install the Screenlets
 install:
-	python setup.py install
-	cp desktop-menu/screenlets.svg /usr/share/icons
-	cp desktop-menu/screenlets-manager.desktop /usr/share/applications
-	cp desktop-menu/screenlets-daemon.desktop $(HOME)/.config/autostart
+	python setup.py install --root=$(DESTDIR)
+	mkdir -p $(DESTDIR)/usr/share/icons/
+	cp desktop-menu/screenlets.svg $(DESTDIR)/usr/share/icons/
+	mkdir -p  $(DESTDIR)/usr/share/applications/
+	cp desktop-menu/screenlets-manager.desktop $(DESTDIR)/usr/share/applications/
+	mkdir -p $(DESTDIR)/usr/share/gnome/autostart
+	cp desktop-menu/screenlets-daemon.desktop $(DESTDIR)/usr/share/gnome/autostart/
 
 # uninstall the Screenlets (NOT FINISHED)
 uninstall:
