--- ddclient-3.8.0/ddclient.orig	2009-01-27 14:14:02.000000000 -0500
+++ ddclient-3.8.0/ddclient	2010-06-07 22:25:25.000000000 -0400
@@ -1853,6 +1853,7 @@
 ######################################################################
 ## get_ip
 ######################################################################
+my $stored_ip;
 sub get_ip {
     my $use = lc shift;
     my ($ip, $arg, $reply, $url, $skip) = (undef, opt($use), '');
@@ -1905,6 +1906,14 @@
 	$reply = geturl('', $url, opt('fw-login'), opt('fw-password')) || '';
 	$arg   = $url;
 
+    } elsif ($use eq 'store') {
+        if (defined $stored_ip) {
+            $ip = $stored_ip;
+        } else {
+            return get_ip(opt('use'));
+        }
+    }
+
     } else {
 	$url  = opt('fw')       || '';
 	$skip = opt('fw-skip')  || '';
@@ -1934,6 +1943,7 @@
     }
 
     debug("get_ip: using %s, %s reports %s", $use, $arg, define($ip, "<undefined>"));
+    $stored_ip = $ip;
     return $ip;
 }
 
@@ -2387,6 +2397,11 @@
 
 		my ($status, $ip) = split / /, lc $line;
 		my $h = shift @hosts;
+
+		## Some service providers breaks the convention of appending the IP.
+		if (!defined $ip) {
+			$ip = get_ip('stored');
+		}
 	    
 		$config{$h}{'status'} = $status;
 		if ($status eq 'good') {
