diff -Naur setup-2.5.48.orig/profile setup-2.5.48/profile
--- setup-2.5.48.orig/profile	2006-01-31 08:06:08.000000000 -0500
+++ setup-2.5.48/profile	2006-02-19 23:09:36.000000000 -0500
@@ -20,11 +20,11 @@
 fi
 
 # Path manipulation
-if [ "$EUID" = "0" ]; then
-	pathmunge /sbin
-	pathmunge /usr/sbin
-	pathmunge /usr/local/sbin
-fi
+pathmunge /sbin
+pathmunge /usr/sbin
+pathmunge /usr/local/sbin
+
+pathmunge /usr/X11R6/bin after
 
 # No core files by default
 ulimit -S -c 0 > /dev/null 2>&1
