diff -ur setup-2.5.27.orig/profile setup-2.5.27/profile
--- setup-2.5.27.orig/profile	2002-03-11 12:50:05.000000000 -0500
+++ setup-2.5.27/profile	2005-03-10 15:49:49.016289096 -0500
@@ -14,11 +14,9 @@
 }
 
 # Path manipulation
-if [ `id -u` = 0 ]; then
-	pathmunge /sbin
-	pathmunge /usr/sbin
-	pathmunge /usr/local/sbin
-fi
+pathmunge /sbin
+pathmunge /usr/sbin
+pathmunge /usr/local/sbin
 
 pathmunge /usr/X11R6/bin after
 
