--- plugins/notify-osd/notify-osd.c.orig	2011-01-25 17:54:43.000000000 +0800
+++ plugins/notify-osd/notify-osd.c	2011-01-25 17:55:14.000000000 +0800
@@ -67,6 +67,9 @@
 	gchar *escaped;
 
 	escaped = g_markup_escape_text (message, strlen(message));
+#ifndef NOTIFY_CHECK_VERSION
+# define NOTIFY_CHECK_VERSION(_x,_y,_z) 0
+#endif
 #if !NOTIFY_CHECK_VERSION(0,7,0)
 	notify = notify_notification_new (summary, escaped, NULL, NULL);
 #else
