--- asclock-2.0.12/asclock.h.defs	1999-03-14 18:08:08.000000000 -0500
+++ asclock-2.0.12/asclock.h	2007-01-25 20:31:22.000000000 -0500
@@ -38,11 +38,7 @@
 #define INT_TYPE 0;
 
 /* the xpm data */
-extern char *clock_xpm[];
-extern char *month_xpm[];
-extern char *weekday_xpm[];
 extern char *led_xpm[];
-extern char *date_xpm[];
 extern char *hour_xpm[];
 extern char *minute_xpm[];
 extern char *second_xpm[];
