diff -Nuar mysql-workbench-gpl-5.2.40-src-old/configure.in mysql-workbench-gpl-5.2.40-src/configure.in
--- mysql-workbench-gpl-5.2.40-src-old/configure.in     2012-05-03 15:41:06.000000000 -0400
+++ mysql-workbench-gpl-5.2.40-src/configure.in 2012-07-14 09:52:54.847655344 -0400
@@ -372,7 +372,8 @@
 ac_savecpp=$CPPFLAGS
 LIBS=`pcre-config --libs`
 CPPFLAS=`pcre-config --cflags`
-AC_RUN_IFELSE([[#include <pcre.h>
+AC_RUN_IFELSE([
+AC_LANG_SOURCE([#include <pcre.h>
 int main()
 {
   char *err= NULL;
@@ -381,7 +382,7 @@
     return 0;
   return 1;
 }
-]], [HAVE_UNICODE_PCRE=1], [HAVE_UNICODE_PCRE=0])
+])], [HAVE_UNICODE_PCRE=1], [HAVE_UNICODE_PCRE=0])
 if test $HAVE_UNICODE_PCRE; then
   AC_MSG_RESULT(yes)
 else

