diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/backend/wbprivate/Makefile.am mysql-workbench-gpl-5.2.36-src/backend/wbprivate/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/backend/wbprivate/Makefile.am	2011-12-02 04:37:03.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/backend/wbprivate/Makefile.am	2011-12-21 01:21:35.000000000 -0700
@@ -2,7 +2,7 @@
 
 noinst_LIBRARIES=libwbprivate.a
 
-INCLUDES=@GLIB_CFLAGS@ @SIGC_CFLAGS@ @GRT_CFLAGS@ @CAIRO_CFLAGS@ @GNOME_CFLAGS@\
+INCLUDES=@GLIB_CFLAGS@ @SIGC_CFLAGS@ @GRT_CFLAGS@ @CAIRO_CFLAGS@ @GNOME_CFLAGS@ @PCRE_CFLAGS@\
 	-I$(top_srcdir)/library/grt/src -I$(top_srcdir)/library/base\
 	-I$(top_srcdir)/library/forms\
 	-I$(top_srcdir)/library/forms/mforms\
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/backend/wbpublic/Makefile.am mysql-workbench-gpl-5.2.36-src/backend/wbpublic/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/backend/wbpublic/Makefile.am	2011-12-02 04:36:59.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/backend/wbpublic/Makefile.am	2011-12-21 01:20:35.000000000 -0700
@@ -9,7 +9,7 @@
 pkglib_LTLIBRARIES=libwbpublic.la
 
 
-INCLUDES=@GLIB_CFLAGS@ @SIGC_CFLAGS@ @GRT_CFLAGS@ @CAIRO_CFLAGS@ @MYSQL_INCLUDE@ @SQLITE3_CFLAGS@ @CTEMPLATE_CFLAGS@\
+INCLUDES=@GLIB_CFLAGS@ @SIGC_CFLAGS@ @GRT_CFLAGS@ @CAIRO_CFLAGS@ @MYSQL_INCLUDE@ @SQLITE3_CFLAGS@ @CTEMPLATE_CFLAGS@ @PCRE_CFLAGS@\
 	-I$(top_srcdir)/library/grt/src -I$(top_srcdir)/library/base\
 	-I$(top_srcdir)/library/forms\
 	-I$(top_srcdir)/library/forms/mforms/\
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/frontend/linux/workbench/Makefile.am mysql-workbench-gpl-5.2.36-src/frontend/linux/workbench/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/frontend/linux/workbench/Makefile.am	2011-12-02 04:37:03.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/frontend/linux/workbench/Makefile.am	2011-12-21 18:43:59.000000000 -0700
@@ -22,7 +22,7 @@
 
 #BUILT_SOURCES=precompiled.h.gch
 
-INCLUDES=@GLIB_CFLAGS@ @SIGC_CFLAGS@ @GRT_CFLAGS@ @CAIRO_CFLAGS@ @GNOME_CFLAGS@\
+INCLUDES=@GLIB_CFLAGS@ @SIGC_CFLAGS@ @GRT_CFLAGS@ @CAIRO_CFLAGS@ @GNOME_CFLAGS@ @PCRE_CFLAGS@\
 	-I$(top_srcdir)/library/grt/src -I$(top_srcdir)/library/base\
 	-I$(top_srcdir)/backend -I$(top_srcdir)/generated -I$(top_srcdir)/library/canvas/src\
 	-I$(top_srcdir)/backend/wbprivate/workbench\
@@ -60,6 +60,7 @@
 	$(top_srcdir)/library/canvas/src/libmdcanvas.la\
 	@CTEMPLATE_LIBS@\
 	@GNOME_LIBS@ @GRT_LIBS@ @PCRE_LIBS@ @GLIB_LIBS@ @CAIRO_LIBS@ @ZIP_LIBS@ @MYSQL_LIBS@ @SQLITE3_LIBS@\
+        @PCRE_LIBS@\
 	-lX11
 
 #precompiled.h.gch: precompiled.h
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/modules/db.checks/Makefile.am mysql-workbench-gpl-5.2.36-src/modules/db.checks/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/modules/db.checks/Makefile.am	2011-12-02 04:37:03.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/modules/db.checks/Makefile.am	2011-12-21 16:31:18.000000000 -0700
@@ -12,7 +12,7 @@
                 -I$(top_srcdir)/library/grt\
                 -I$(top_srcdir)/library/base\
                 -I$(top_srcdir)/modules/db.mysql/src\
-                @GRT_CFLAGS@ @GLIB_CFLAGS@ @GNOME_CFLAGS@ @SIGC_CFLAGS@
+                @GRT_CFLAGS@ @GLIB_CFLAGS@ @GNOME_CFLAGS@ @SIGC_CFLAGS@ @PCRE_CFLAGS@
 
 db_checks_wbp_la_LDFLAGS=-module
 
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/modules/db.mysql/Makefile.am mysql-workbench-gpl-5.2.36-src/modules/db.mysql/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/modules/db.mysql/Makefile.am	2011-12-02 04:36:56.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/modules/db.mysql/Makefile.am	2011-12-21 03:02:01.000000000 -0700
@@ -23,7 +23,7 @@
                 -I$(top_srcdir)/library/sql-parser/include\
                 -I$(top_srcdir)/library/sql-parser/source\
                 @CTEMPLATE_CFLAGS@\
-                @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@
+                @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@ @PCRE_CFLAGS@
 
 
 #db_mysql_grt_la_SOURCES=db_mysql_reveng.cpp\
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/modules/db.mysql.sqlparser/Makefile.am mysql-workbench-gpl-5.2.36-src/modules/db.mysql.sqlparser/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/modules/db.mysql.sqlparser/Makefile.am	2011-12-02 04:36:54.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/modules/db.mysql.sqlparser/Makefile.am	2011-12-21 13:55:23.000000000 -0700
@@ -16,7 +16,7 @@
 	-I$(top_srcdir)/ext/vsqlite++/include\
 	-I$(top_srcdir)/ext/vsqlite\
 	-I$(top_srcdir)/library/base\
-	@MYSQL_INCLUDE@ @GRT_CFLAGS@ @GLIB_CFLAGS@ @PCRE_CFLAGS@ @SIGC_CFLAGS@
+	@MYSQL_INCLUDE@ @GRT_CFLAGS@ @GLIB_CFLAGS@ @PCRE_CFLAGS@ @SIGC_CFLAGS@ @PCRE_CFLAGS@
 
 db_mysql_sqlparser_grt_la_LDFLAGS=-module
 
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/modules/wb.model/Makefile.am mysql-workbench-gpl-5.2.36-src/modules/wb.model/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/modules/wb.model/Makefile.am	2011-12-02 04:36:54.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/modules/wb.model/Makefile.am	2011-12-21 13:09:28.000000000 -0700
@@ -70,7 +70,7 @@
 	-I$(top_srcdir)/ext/scintilla/lexlib\
 	-I$(top_srcdir)/ext\
 	-I$(top_srcdir)/modules/db.mysql.sqlide/src\
-	@CTEMPLATE_CFLAGS@ @MYSQL_INCLUDE@ @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@
+	@CTEMPLATE_CFLAGS@ @MYSQL_INCLUDE@ @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@ @PCRE_CFLAGS@
 
 
 wb_model_grt_la_LDFLAGS=-module
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/modules/wb.mysql.import/Makefile.am mysql-workbench-gpl-5.2.36-src/modules/wb.mysql.import/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/modules/wb.mysql.import/Makefile.am	2011-12-02 04:37:03.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/modules/wb.mysql.import/Makefile.am	2011-12-21 13:56:11.000000000 -0700
@@ -18,7 +18,7 @@
 	-I$(top_srcdir)/ext/vsqlite\
 	-I$(top_srcdir)/ext/vsqlite++/include\
 	-I$(top_srcdir)/ext/tinyxml\
-	@MYSQL_INCLUDE@ @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@
+	@MYSQL_INCLUDE@ @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@ @PCRE_CFLAGS@
 
 
 wb_mysql_import_grt_la_LDFLAGS=-module
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/modules/wb.validation/Makefile.am mysql-workbench-gpl-5.2.36-src/modules/wb.validation/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/modules/wb.validation/Makefile.am	2011-12-02 04:36:59.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/modules/wb.validation/Makefile.am	2011-12-21 13:56:29.000000000 -0700
@@ -15,7 +15,7 @@
 	-I$(top_srcdir)/library/sql-parser/source\
 	-I$(top_srcdir)/library/sql-parser/source/linux\
 	-I$(top_srcdir)/ext/vsqlite\
-	@MYSQL_INCLUDE@ @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@
+	@MYSQL_INCLUDE@ @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@ @PCRE_CFLAGS@
 
 
 wb_validation_grt_la_LDFLAGS=-module
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/plugins/db.mysql/frontend/Makefile.am mysql-workbench-gpl-5.2.36-src/plugins/db.mysql/frontend/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/plugins/db.mysql/frontend/Makefile.am	2011-12-02 04:36:59.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/plugins/db.mysql/frontend/Makefile.am	2011-12-21 19:57:45.000000000 -0700
@@ -22,7 +22,7 @@
 		-I$(top_srcdir)/plugins/db/backend\
 		-I$(top_srcdir)/plugins/db.mysql.sync/backend\
 	  -I../backend\
-    @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@  @CAIRO_CFLAGS@
+    @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@  @CAIRO_CFLAGS@ @PCRE_CFLAGS@
 
 
 db_mysql_wbp_la_LDFLAGS=-module
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/plugins/db.mysql.diff.reporting/frontend/Makefile.am mysql-workbench-gpl-5.2.36-src/plugins/db.mysql.diff.reporting/frontend/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/plugins/db.mysql.diff.reporting/frontend/Makefile.am	2011-12-02 04:37:02.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/plugins/db.mysql.diff.reporting/frontend/Makefile.am	2011-12-21 23:52:32.000000000 -0700
@@ -19,7 +19,7 @@
 	  -I$(top_srcdir)/ext/vsqlite++/include\
     -I$(top_srcdir)/plugins\
 	  -I../backend\
-    @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@ @CAIRO_CFLAGS@
+    @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@ @CAIRO_CFLAGS@ @PCRE_CFLAGS@
 
 
 db_mysql_diff_reporting_wbp_la_LDFLAGS=-module
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/plugins/db.mysql.editors/backend/Makefile.am mysql-workbench-gpl-5.2.36-src/plugins/db.mysql.editors/backend/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/plugins/db.mysql.editors/backend/Makefile.am	2011-12-02 04:37:02.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/plugins/db.mysql.editors/backend/Makefile.am	2011-12-21 20:01:13.000000000 -0700
@@ -2,7 +2,7 @@
 
 noinst_LIBRARIES=libplugbe.a
 
-INCLUDES=@GLIB_CFLAGS@ @SIGC_CFLAGS@ @GRT_CFLAGS@\
+INCLUDES=@GLIB_CFLAGS@ @SIGC_CFLAGS@ @GRT_CFLAGS@ @PCRE_CFLAGS@\
 	-I$(top_srcdir)/library/grt/src -I$(top_srcdir)/library/base\
 	-I$(top_srcdir)/backend -I$(top_srcdir)/generated -I$(top_srcdir)/modules
 
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/plugins/db.mysql.editors/linux/Makefile.am mysql-workbench-gpl-5.2.36-src/plugins/db.mysql.editors/linux/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/plugins/db.mysql.editors/linux/Makefile.am	2011-12-02 04:37:03.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/plugins/db.mysql.editors/linux/Makefile.am	2011-12-21 20:01:38.000000000 -0700
@@ -30,7 +30,7 @@
                 -I$(top_srcdir)/frontend/linux/linux_utilities\
                 -I$(top_srcdir)/frontend/linux/sqlide\
                 -I$(top_srcdir)/ext/scintilla/include\
-                @GRT_CFLAGS@ @GLIB_CFLAGS@ @GNOME_CFLAGS@ @SIGC_CFLAGS@
+                @GRT_CFLAGS@ @GLIB_CFLAGS@ @GNOME_CFLAGS@ @SIGC_CFLAGS@ @PCRE_CFLAGS@
 
 
 db_mysql_editors_wbp_la_LDFLAGS=-module
diff -Nuar mysql-workbench-gpl-5.2.36-src.orig/plugins/wb.model.reporting/frontend/Makefile.am mysql-workbench-gpl-5.2.36-src/plugins/wb.model.reporting/frontend/Makefile.am
--- mysql-workbench-gpl-5.2.36-src.orig/plugins/wb.model.reporting/frontend/Makefile.am	2011-12-02 04:36:54.000000000 -0700
+++ mysql-workbench-gpl-5.2.36-src/plugins/wb.model.reporting/frontend/Makefile.am	2011-12-22 00:00:26.000000000 -0700
@@ -14,7 +14,7 @@
 		-I$(top_srcdir)/plugins\
 		-I$(top_srcdir)/modules\
 		-I../backend @CAIRO_CFLAGS@\
-                @GRT_CFLAGS@ @GLIB_CFLAGS@ @GNOME_CFLAGS@ @SIGC_CFLAGS@
+                @GRT_CFLAGS@ @GLIB_CFLAGS@ @GNOME_CFLAGS@ @SIGC_CFLAGS@ @PCRE_CFLAGS@
 
 
 wb_model_reporting_wbp_la_LDFLAGS=-module
