--- latex2html-2002-2-1/config/install.pl.bak	2005-09-22 15:23:23.000000000 -0400
+++ latex2html-2002-2-1/config/install.pl	2005-09-22 15:23:52.000000000 -0400
@@ -293,13 +293,14 @@
   split(/\s+/,$cfg{'IMAGE_TYPES'}) : qw(gif);
 my $iconrx = join('|', @icon_types);
 
+my $destdir = ''.$ENV{'DESTDIR'};
 my $dest1 = "$cfg{'SHLIBDIR'}${dd}icons";
-if((-d $dest1 && !-w _) || (-d $cfg{'SHLIBDIR'} && !-w _)) {
+if((-d $destdir.$dest1 && !-w _) || (-d $destdir.$cfg{'SHLIBDIR'} && !-w _)) {
   print STDERR "Error: Cannot install icons in '$dest1': No write permission.\n";
   $dest1 = '';
 }
 my $dest2 = $cfg{'ICONSTORAGE'} || '';
-if(-d $dest2 && !-w $dest2) {
+if(-d $destdir.$dest2 && !-w _) {
   print STDERR "Error: Cannot install icons in '$dest2': No write permission.\n";
   $dest2 = '';
 }
@@ -371,8 +372,8 @@
 #-----------------------------------------------------------------------------
 
 if($cfg{TEXPATH}) {
-  print "\nNote: trying to install LaTeX2HTML style files in TeX directory tree\n     ($cfg{TEXPATH})\n";
-  unless(mkpath($cfg{TEXPATH})) {
+  print "\nNote: trying to install LaTeX2HTML style files in TeX directory tree\n     ($destdir$cfg{TEXPATH})\n";
+  unless(mkpath($destdir.$cfg{TEXPATH})) {
   #my $testpath = $cfg{TEXPATH}; # to strip (latex2)html
   #$testpath =~ s/[$dd$dd][^$dd$dd]*$//;
   #if((-d $cfg{TEXPATH} && !-w _) || (-d $testpath && !-w _)) {
