Index: kdemultimedia-4.4.80/libkcompactdisc/CMakeLists.txt
===================================================================
--- kdemultimedia-4.4.80.orig/libkcompactdisc/CMakeLists.txt	2009-12-31 08:30:00.000000000 -0500
+++ kdemultimedia-4.4.80/libkcompactdisc/CMakeLists.txt	2010-05-25 08:31:39.509499004 -0400
@@ -47,7 +47,7 @@
 
 kde4_add_library(kcompactdisc SHARED ${kcompactdisc_LIB_SRCS})
 
-target_link_libraries(kcompactdisc ${KDE4_KDECORE_LIBS} ${KDE4_PHONON_LIBS} ${KDE4_SOLID_LIBS})
+target_link_libraries(kcompactdisc ${KDE4_KDECORE_LIBS} ${KDE4_PHONON_LIBS} ${KDE4_SOLID_LIBS} ${CMAKE_THREAD_LIBS_INIT})
 if (HAVE_LIBASOUND2)
     target_link_libraries(kcompactdisc ${ASOUND_LIBRARY})
 endif (HAVE_LIBASOUND2)
