--- sysklogd-1.4.1/syslog.c.orig	2007-08-24 12:02:43.000000000 -0400
+++ sysklogd-1.4.1/syslog.c	2007-08-24 12:05:40.000000000 -0400
@@ -78,6 +78,14 @@
 static const char *LogTag = "syslog";	/* string to tag the entry with */
 static int	LogFacility = LOG_USER;	/* default facility code */
 
+#ifdef syslog
+#undef syslog
+#endif
+
+#ifdef vsyslog
+#undef vsyslog
+#endif
+
 void
 syslog(int pri, const char *fmt, ...)
 {
