--- schemas/desktop_gnome_url_handlers.schemas.in.orig	2004-09-15 11:47:20.000000000 +0200
+++ schemas/desktop_gnome_url_handlers.schemas.in	2004-09-15 11:47:47.000000000 +0200
@@ -153,7 +153,7 @@
       <applyto>/desktop/gnome/url-handlers/http/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>firefox %s</default>
       <locale name="C">
         <short>The handler for "http" URLs</short>
         <long>The command used to handle "http" URLs, if enabled.</long>
@@ -187,7 +187,7 @@
       <applyto>/desktop/gnome/url-handlers/https/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>firefox %s</default>
       <locale name="C">
         <short>The handler for "https" URLs</short>
         <long>The command used to handle "https" URLs, if enabled.</long>
