# upstream http://bugzilla.gnome.org/show_bug.cgi?id=170910

--- libgnomevfs/gnome-vfs-hal-mounts.c.orig	2005-05-10 15:32:09.146734040 +0200
+++ libgnomevfs/gnome-vfs-hal-mounts.c	2005-05-10 15:34:15.151578392 +0200
@@ -448,7 +448,7 @@
 		target_mount_point = gnome_vfs_get_local_path_from_uri (drive->priv->activation_uri);
 
 	/* if we don't use removable media and the volume shouldn't be visible, then hide the drive */
-	if(!libhal_drive_uses_removable_media (hal_drive) && 
+	if(hal_volume && !libhal_drive_uses_removable_media (hal_drive) &&
 	   !libhal_volume_policy_should_be_visible (hal_drive, hal_volume, hal_storage_policy, target_mount_point))
 		drive->priv->is_user_visible = FALSE;
 
