--- modules/cdda-cddb.c	2004-07-25 17:40:35.000000000 +0200
+++ /home/sjoerd/source/gnome-vfs/modules/cdda-cddb.c	2005-05-02 14:08:02.000000000 +0200
@@ -440,7 +440,7 @@
     query->query_match=MATCH_INEXACT;
     query->query_matches=0;
 
-    while(!CDDBReadLine(socket,inbuffer,256)) {
+    while(query->query_matches < MAX_INEXACT_MATCHES && !CDDBReadLine(socket,inbuffer,256)) {
       query->query_list[query->query_matches].list_genre=
 	CDDBGenreValue(ChopWhite(strtok(inbuffer," ")));
       
