--- libgnomevfs/gnome-vfs-cdrom.c.orig	2005-03-24 14:46:28.000000000 +0100
+++ libgnomevfs/gnome-vfs-cdrom.c	2005-03-24 14:47:37.000000000 +0100
@@ -142,7 +142,7 @@
 
 #endif /* defined(__FreeBSD__) */
 	return type;
-#else
+#elif defined(__linux__)
 	*fd = open (vol_dev_path, O_RDONLY|O_NONBLOCK);
 	if (*fd  < 0) {
 		return -1;
@@ -153,6 +153,8 @@
 		return -1;
 	}
 	return ioctl (*fd, CDROM_DISC_STATUS, CDSL_CURRENT);
+#else
+	return CDS_NO_INFO;
 #endif
 }
 
