diff -U 10 -r gnome-vfs-2.12.0/libgnomevfs/gnome-vfs-volume-ops.c gnome-vfs-2.12.0-patch/libgnomevfs/gnome-vfs-volume-ops.c
--- gnome-vfs-2.12.0/libgnomevfs/gnome-vfs-volume-ops.c	2005-06-30 09:27:33.000000000 +0200
+++ gnome-vfs-2.12.0-patch/libgnomevfs/gnome-vfs-volume-ops.c	2005-09-28 12:22:34.685325083 +0200
@@ -54,29 +54,29 @@
 };
 
 #define MOUNT_COMMAND volrmmount_locations
 #define MOUNT_SEPARATOR " -i "
 #define UMOUNT_COMMAND volrmmount_locations
 #define UMOUNT_SEPARATOR " -e "
 
 #else
 
 static const char *mount_known_locations [] = {
-	"/sbin/mount", "/bin/mount",
-	"/usr/sbin/mount", "/usr/bin/mount",
+	"/usr/bin/pmount", "/usr/bin/pmount",
+	"/usr/bin/pmount", "/usr/bin/pmount",
 	NULL
 };
 
 static const char *umount_known_locations [] = {
-	"/usr/bin/pumount", "/bin/pumount",
-	"/sbin/umount", "/bin/umount",
-	"/usr/sbin/umount", "/usr/bin/umount",
+	"/usr/bin/pumount", "/usr/bin/pumount",
+	"/usr/bin/pumount", "/usr/bin/pumount",
+	"/usr/bin/pumount", "/usr/bin/pumount",
 	NULL
 };
 
 #define MOUNT_COMMAND mount_known_locations
 #define MOUNT_SEPARATOR " "
 #define UMOUNT_COMMAND umount_known_locations
 #define UMOUNT_SEPARATOR " "
 
 #endif /* USE_VOLRMMOUNT */
 
Nur in gnome-vfs-2.12.0/po: POTFILES.
