diff -ruN VFlib2-2.25.6.orig/configure.in VFlib2-2.25.6/configure.in
--- VFlib2-2.25.6.orig/configure.in	2002-11-12 17:49:55.000000000 +0900
+++ VFlib2-2.25.6/configure.in	2002-11-12 18:02:51.000000000 +0900
@@ -103,7 +103,7 @@
 AC_SUBST(sony_fslib)
 AC_ARG_WITH(fslib, 
   [  --with-fslib            Compile VFlib 2 with Sony FSlib library], 
-  sony_fslib="-L/usr/sony/lib -lFSlib"; AC_DEFINE(WITH_FSLIB),
+  sony_fslib="-L /usr/sony/%(lib)s -L/usr/sony/lib -lFSlib"; AC_DEFINE(WITH_FSLIB),
   sony_fslib="")
 if test -n "${sony_fslib}" ; then
       echo "Enabled the Sony FSlib driver"; 
@@ -129,8 +129,8 @@
 AC_ARG_WITH(
  gettext_prefix,
  [  --with-gettext-prefix=DIR      GNU gettext preffix direcotry is DIR],
- gettext_lib="-L${withval}/lib"; gettext_include="-I${withval}/include",
- gettext_lib='-L${prefix}/lib';  gettext_include='-I${prefix}/include')
+ gettext_lib="-L${withval}/%(lib)s -L${withval}/lib"; gettext_include="-I${withval}/include",
+ gettext_lib='-L${prefix}/%(lib)s -L${prefix}/lib';  gettext_include='-I${prefix}/include')
 fi
 AC_TRY_LINK([#include <libintl.h>], 
  [ bindtextdomain("xxx", "/usr/share/locale"); ], 
