Index: lib/stomp_frame.rb
===================================================================
--- lib/stomp_frame.rb	(revision 148)
+++ lib/stomp_frame.rb	(working copy)
@@ -58,7 +58,7 @@
       @frame.command, headers = match.captures
       @buffer = match.post_match
       headers.split(/\n/).each do |data|
-        if data =~ /^\s*(\S+)\s*:\s*(.*?)\s*$/
+        if data =~ /^\s*([\w-]+)\s*:\s*(.*?)\s*$/
           @frame.headers[$1] = $2
         end
       end
