Index: trunk/lib/queue_manager.rb
===================================================================
--- trunk/lib/queue_manager.rb	(revision 148)
+++ trunk/lib/queue_manager.rb	(working copy)
@@ -55,6 +55,7 @@
     @pending[user].each do |frame|
       sendmsg(frame)
     end
+    @pending.delete(user)
   end
     
   def send_to_user(frame, user)
Index: trunk/lib/topic_manager.rb
===================================================================
--- trunk/lib/topic_manager.rb	(revision 148)
+++ trunk/lib/topic_manager.rb	(working copy)
@@ -10,7 +10,8 @@
   end
   
   def unsubscribe(topic, user)
-    @topics[topic].delete(user) 
+    @topics[topic].delete(user)
+    @topics.delete(topic) if @topics[topic].empty?
   end
   
   def disconnect(user)
