Use "NIL" for logfile to disable logging.
diff -r a396fe916387 lib/stomp_server.rb
--- a/lib/stomp_server.rb	Tue Jun 24 20:07:50 2008 -0400
+++ b/lib/stomp_server.rb	Tue Jun 24 20:33:30 2008 -0400
@@ -66,7 +66,11 @@
       opts[:etcdir] = File.join(opts[:working_dir],'etc')
       opts[:storage] = File.join(opts[:working_dir],opts[:storage])
       opts[:logdir] = File.join(opts[:working_dir],opts[:logdir])
-      opts[:logfile] = File.join(opts[:logdir],opts[:logfile])
+      if opts[:logfile] == "NIL"
+        opts[:logfile] = nil
+      else
+        opts[:logfile] = File.join(opts[:logdir],opts[:logfile])
+      end
       opts[:pidfile] = File.join(opts[:logdir],opts[:pidfile])
       if opts[:auth]
         opts[:passwd] = File.join(opts[:etcdir],'.passwd')
