diff -Naur gdm-2.17.7.orig/gui/gdmsetup.desktop gdm-2.17.7/gui/gdmsetup.desktop
--- gdm-2.17.7.orig/gui/gdmsetup.desktop	2007-02-12 00:44:36.000000000 -0500
+++ gdm-2.17.7/gui/gdmsetup.desktop	2007-02-24 00:52:59.000000000 -0500
@@ -71,8 +71,7 @@
 Comment[sv]=Konfigurera GDM-inloggningsfönstrets utseende och beteende
 Comment[th]=ปรับแต่งรูปโฉมและพฤติกรรมของหน้าจอเข้าระบบ GDM
 Comment[vi]=Cấu hình diện mạo và ứng xử của cửa sổ đăng nhập GDM
-TryExec=/usr/local/sbin/gdmsetup
-Exec=gdmsetup
+Exec=gksudo gdmsetup
 Icon=gdm-setup
 StartupNotify=true
 Terminal=false
diff -Naur gdm-2.17.7.orig/gui/gdmsetup.desktop.in.in gdm-2.17.7/gui/gdmsetup.desktop.in.in
--- gdm-2.17.7.orig/gui/gdmsetup.desktop.in.in	2007-02-12 00:40:13.000000000 -0500
+++ gdm-2.17.7/gui/gdmsetup.desktop.in.in	2007-02-24 00:53:24.000000000 -0500
@@ -2,8 +2,7 @@
 Encoding=UTF-8
 _Name=Login Window
 _Comment=Configure GDM login window appearance and behavior
-TryExec=@sbindir@/gdmsetup
-Exec=gdmsetup
+Exec=gksudo gdmsetup
 Icon=gdm-setup
 StartupNotify=true
 Terminal=false
