Index: gdm/gui/simple-greeter/gdm-simple-greeter.schemas.in
===================================================================
--- gdm/gui/simple-greeter/gdm-simple-greeter.schemas.in	(revision 5874)
+++ gdm/gui/simple-greeter/gdm-simple-greeter.schemas.in	(working copy)
@@ -6,7 +6,7 @@
       <applyto>/apps/gdm/simple-greeter/logo-icon-name</applyto>
       <owner>gdm-simple-greeter</owner>
       <type>string</type>
-      <default>computer</default>
+      <default>distro-logo</default>
       <locale name="C">
         <short>Icon name to use for greeter logo</short>
         <long>Set to the themed icon name to use for the greeter logo.</long>
diff -r 2a6b72de76f6 gui/simple-greeter/gdm-greeter-login-window.glade
--- a/gui/simple-greeter/gdm-greeter-login-window.glade	Fri May 02 14:57:07 2008 -0400
+++ b/gui/simple-greeter/gdm-greeter-login-window.glade	Tue Jun 03 14:44:47 2008 -0400
@@ -18,7 +18,7 @@
             <child>
               <widget class="GtkImage" id="logo-image">
                 <property name="visible">True</property>
-                <property name="pixel_size">64</property>
+                <property name="pixel_size">128</property>
                 <property name="icon_name">computer</property>
               </widget>
             </child>
