diff --exclude='*~' -ur bash-3.2.orig/config-top.h bash-3.2/config-top.h
--- bash-3.2.orig/config-top.h	2005-04-29 16:36:34.000000000 -0400
+++ bash-3.2/config-top.h	2008-10-23 09:35:36.000000000 -0400
@@ -84,4 +84,5 @@
 
 /* Define this if you want bash to try to check whether it's being run by
    sshd and source the .bashrc if so (like the rshd behavior). */
-/* #define SSH_SOURCE_BASHRC */
+#define SSH_SOURCE_BASHRC
+
