--- gnucash-1.8.12/macros/aqbanking.m4.foo	2005-12-22 01:38:10.000000000 -0500
+++ gnucash-1.8.12/macros/aqbanking.m4	2005-12-22 01:43:49.000000000 -0500
@@ -49,7 +49,7 @@
              /"])
 
   for li in $lcc_dir; do
-      if test -x "$li/bin/aqbanking-config"; then
+      if test -x "$li/include/aqbanking"; then
           aqbanking_dir="$li";
           break
       fi
@@ -65,19 +65,19 @@
   else
       AC_MSG_RESULT($aqbanking_dir)
       AC_MSG_CHECKING(for aqbanking libs)
-      aqbanking_libs="`$aqbanking_dir/bin/aqbanking-config --libraries`"
+      aqbanking_libs="`pkg-config --libs aqbanking`"
       AC_MSG_RESULT($aqbanking_libs)
       AC_MSG_CHECKING(for aqbanking C++ libs)
-      aqbanking_libspp="`$aqbanking_dir/bin/aqbanking-config --libraries++`"
+      aqbanking_libspp="`pkg-config --libs aqbanking++`"
       AC_MSG_RESULT($aqbanking_libspp)
       AC_MSG_CHECKING(for aqbanking includes)
-      aqbanking_includes="`$aqbanking_dir/bin/aqbanking-config --includes`"
+      aqbanking_includes="`pkg-config --cflags aqbanking`"
       AC_MSG_RESULT($aqbanking_includes)
       AC_MSG_CHECKING(for aqbanking plugins)
-      aqbanking_plugins="`$aqbanking_dir/bin/aqbanking-config --plugins`"
+      aqbanking_plugins="`pkg-config --variable=plugindir aqbanking`"
       AC_MSG_RESULT($aqbanking_plugins)
       AC_MSG_CHECKING(for aqbanking data)
-      aqbanking_data="`$aqbanking_dir/bin/aqbanking-config --data`"
+      aqbanking_data="`pkg-config --variable=pkgdatadir aqbanking`"
       AC_MSG_RESULT($aqbanking_data)
   fi
   AC_MSG_CHECKING(if aqbanking test desired)
@@ -88,27 +88,9 @@
   AC_MSG_RESULT($enable_aqbanking_test)
   AC_MSG_CHECKING(for AqBanking version >=$vma.$vmi.$vpl.$vbld)
   if test "$enable_aqbanking_test" != "no"; then
-    aqbanking_versionstring="`$aqbanking_dir/bin/aqbanking-config --vstring`.`$aqbanking_dir/bin/aqbanking-config --vbuild`"
-    AC_MSG_RESULT([found $aqbanking_versionstring])
-    if test "$vma" -gt "`$aqbanking_dir/bin/aqbanking-config --vmajor`"; then
-      AC_MSG_ERROR([Your AqBanking version is way too old.
-      Please update from http://www.aquamaniac.de/aqbanking/])
-    elif test "$vma" = "`$aqbanking_dir/bin/aqbanking-config --vmajor`"; then
-      if test "$vmi" -gt "`$aqbanking_dir/bin/aqbanking-config --vminor`"; then
-        AC_MSG_ERROR([Your AqBanking version is too old.
-          Please update from http://www.aquamaniac.de/aqbanking/])
-      elif test "$vmi" = "`$aqbanking_dir/bin/aqbanking-config --vminor`"; then
-          if test "$vpl" -gt "`$aqbanking_dir/bin/aqbanking-config --vpatchlevel`"; then
-            AC_MSG_ERROR([Your AqBanking version is a little bit too old.
-            Please update from http://www.aquamaniac.de/aqbanking/])
-          elif test "$vpl" = "`$aqbanking_dir/bin/aqbanking-config --vpatchlevel`"; then
-            if test "$vbld" -gt "`$aqbanking_dir/bin/aqbanking-config --vbuild`"; then
-              AC_MSG_ERROR([Your AqBanking version is a little bit too old. 
-  Please update to the latest CVS version. Instructions for accessing 
-  CVS can be found on http://www.aquamaniac.de/aqbanking/])
-             fi
-           fi
-      fi
+    if ! pkg-config --atleast-version $vma.$vmi.$vpl.$vbld aqbanking; then
+     AC_MSG_ERROR([Your AqBanking version is too old.
+     Please update from http://www.aquamaniac.de/aqbanking/])
     fi
     have_aqbanking="yes"
     #AC_MSG_RESULT(yes)
