diff -ru openssh-5.3p1/ssh_config openssh-5.3p1/ssh_config
--- openssh-5.3p1/ssh_config	2009-02-20 20:45:02.000000000 -0500
+++ openssh-5.3p1/ssh_config	2010-02-26 13:31:46.000000000 -0500
@@ -44,3 +44,10 @@
 #   TunnelDevice any:any
 #   PermitLocalCommand no
 #   VisualHostKey no
+Host *
+      GSSAPIAuthentication yes
+# If this option is set to yes then remote X11 clients will have full access
+# to the original X11 display. As virtually no X11 client supports the untrusted
+# mode correctly we set this to yes.
+      ForwardX11Trusted yes
+
diff -ru openssh-5.3p1/sshd_config openssh-5.3p1/sshd_config
--- openssh-5.3p1/sshd_config	2008-07-02 08:35:43.000000000 -0400
+++ openssh-5.3p1/sshd_config	2010-02-26 13:30:43.000000000 -0500
@@ -33,6 +33,7 @@
 # Logging
 # obsoletes QuietMode and FascistLogging
 #SyslogFacility AUTH
+SyslogFacility AUTHPRIV
 #LogLevel INFO
 
 # Authentication:
@@ -60,9 +61,11 @@
 # To disable tunneled clear text passwords, change to no here!
 #PasswordAuthentication yes
 #PermitEmptyPasswords no
+PasswordAuthentication yes
 
 # Change to no to disable s/key passwords
 #ChallengeResponseAuthentication yes
+ChallengeResponseAuthentication no
 
 # Kerberos options
 #KerberosAuthentication no
@@ -72,7 +75,9 @@
 
 # GSSAPI options
 #GSSAPIAuthentication no
+GSSAPIAuthentication yes
 #GSSAPICleanupCredentials yes
+GSSAPICleanupCredentials yes
 
 # Set this to 'yes' to enable PAM authentication, account processing, 
 # and session processing. If this is enabled, PAM authentication will 
@@ -84,11 +89,13 @@
 # PAM authentication, then enable this but set PasswordAuthentication
 # and ChallengeResponseAuthentication to 'no'.
 #UsePAM no
+UsePAM yes
 
 #AllowAgentForwarding yes
 #AllowTcpForwarding yes
 #GatewayPorts no
 #X11Forwarding no
+X11Forwarding yes
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PrintMotd yes
