diff -r 0061ac14e758 libnautilus-private/apps_nautilus_preferences.schemas.in
--- a/libnautilus-private/apps_nautilus_preferences.schemas.in	Sat Mar 08 22:53:54 2008 -0500
+++ b/libnautilus-private/apps_nautilus_preferences.schemas.in	Sat Mar 08 22:55:22 2008 -0500
@@ -879,7 +879,7 @@ most cases, this should be left alone. -
       <applyto>/apps/nautilus/desktop/computer_icon_visible</applyto>
       <owner>nautilus</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
          <short>Computer icon visible on desktop</short>
          <long>
