diff -ur gnome-session-2.9.4.orig/gnome-session/default.in gnome-session-2.9.4/gnome-session/default.in
--- gnome-session-2.9.4.orig/gnome-session/default.in	2003-06-06 09:55:38.000000000 -0400
+++ gnome-session-2.9.4/gnome-session/default.in	2005-03-06 10:38:40.509135120 -0500
@@ -21,3 +21,6 @@
 3,id=default3
 3,Priority=40
 3,RestartCommand=nautilus --no-default-window --sm-client-id default3
+4,id=default4
+4,Priority=50
+4,RestartCommand=best --no-default-window --sm-client-id default4
