--- syslinux-3.52/extlinux/extlinux.c.orig	2007-11-14 13:14:39.000000000 -0500
+++ syslinux-3.52/extlinux/extlinux.c	2007-11-14 13:14:41.000000000 -0500
@@ -718,8 +718,13 @@
     /* Didn't find it in /proc/mounts, try /etc/mtab */
     if ( (mtab = setmntent("/etc/mtab", "r")) ) {
       while ( (mnt = getmntent(mtab)) ) {
-	devname = mnt->mnt_fsname;
-	break;
+        if ( (!strcmp(mnt->mnt_type, "ext2") ||
+              !strcmp(mnt->mnt_type, "ext3")) &&
+             !stat(mnt->mnt_fsname, &dst) &&
+             dst.st_rdev == st.st_dev ) {
+	  devname = mnt->mnt_fsname;
+	  break;
+        }
       }
     }
   }
