diff -ruN pmount-0.9.13.orig/src/pmount-hal.c pmount-0.9.13/src/pmount-hal.c
--- pmount-0.9.13.orig/src/pmount-hal.c 2006-12-26 12:08:17.000000000 -0500
+++ pmount-0.9.13/src/pmount-hal.c      2006-12-26 12:14:58.000000000 -0500
@@ -361,7 +361,7 @@
     /* shut down hal connection */
     libhal_ctx_shutdown( hal_ctx, &error );
     libhal_ctx_free( hal_ctx );
-    dbus_connection_close( dbus_conn );
+    //dbus_connection_close( dbus_conn );
     dbus_connection_unref( dbus_conn );
 
     /* go */
