--- django_restapi/resource.py 2010-08-16 10:52:03.000000000 -0400
+++ django_restapi/resource.py 2010-08-16 11:01:27.000000000 -0400
@@ -14,12 +14,9 @@
     Therefore we can use Django's POST data retrieval method 
     for PUT.
     """
-    if request.method == 'PUT':
-        request.method = 'POST'
-        request._load_post_and_files()
-        request.method = 'PUT'
-        request.PUT = request.POST
-        del request._post
+    request._load_post_and_files()
+    request.PUT = request.POST
+    del request._post
 
 def reverse(viewname, args=(), kwargs=None):
     """
