diff -r a278a7b12684 dmesg.c
--- a/dmesg.c	Mon Jun 09 17:37:06 2008 -0400
+++ b/dmesg.c	Mon Jun 09 17:50:15 2008 -0400
@@ -132,7 +132,7 @@
 /*
  * extract_oops tries to find oops signatures in a log
  */
-static void extract_oops(char *buffer, int remove_syslog)
+static void extract_oops(char *buffer, size_t buflen, int remove_syslog)
 {
 	int i;
 	char prevlevel = 0;
@@ -140,10 +140,10 @@
 	int oopsend;
 	int inbacktrace = 0;
 
-	linepointer = calloc(strlen(buffer)+1, sizeof(char*));
+	linepointer = calloc(buflen+1, sizeof(char*));
 	if (!linepointer)
 		return;
-	linelevel = calloc(strlen(buffer)+1, sizeof(char));
+	linelevel = calloc(buflen+1, sizeof(char));
 	if (!linelevel) {
 		free(linepointer);
 		linepointer = NULL;
@@ -195,10 +195,6 @@
 				if (oopsstart != i)
 					printf("    trigger line is -%s-\n", c);
 			}
-			/* give the kernel some time to finish dumping the oops */
-			/* but not in testmode since that makes regression testins slow */
-			if (oopsstart >= 0 && !testmode)
-				sleep(1);
 
 			/* try to find the end marker */
 			if (oopsstart >= 0) {
@@ -343,7 +339,7 @@
 	buffer = calloc(getpagesize()+1, 1);
 
 	syscall(__NR_syslog, 3, buffer, getpagesize());
-	extract_oops(buffer, 0);
+	extract_oops(buffer, strlen(buffer), 0);
 	free(buffer);
 	if (opted_in >= 2)
 		submit_queue();
@@ -384,7 +380,7 @@
 	fclose(file);
 
 	if (ret > 0)
-		extract_oops(buffer, issyslog);
+		extract_oops(buffer, statb.st_size+1023, issyslog);
 	free(buffer);
 	if (opted_in >= 2)
 		submit_queue();
