=== modified file 'gwibber/digg.py'
--- gwibber/digg.py	2008-07-14 03:24:43 +0000
+++ gwibber/digg.py	2008-07-28 04:25:06 +0000
@@ -13,7 +13,7 @@
 LINK_PARSE =  re.compile("<a[^>]+href=\"(https?://[^\"]+)\">[^<]+</a>")
 
 def parse_time(t):
-  return datetime.datetime.strptime(t,"%a, %d %b %Y %H:%M:%S %Z")
+  return datetime.datetime(*(time.strptime(t,"%a, %d %b %Y %H:%M:%S %Z")[0:6]))
 
 def sanitize_text(t):
   return LINK_PARSE.sub("\\1", t.strip())

=== modified file 'gwibber/identica.py'
--- gwibber/identica.py	2008-07-25 03:25:44 +0000
+++ gwibber/identica.py	2008-07-28 03:29:10 +0000
@@ -15,7 +15,7 @@
 NICK_PARSE = re.compile("@([A-Za-z0-9]+)")
 
 def parse_time(t):
-  return datetime.datetime.strptime(t, "%a %b %d %H:%M:%S +0000 %Y")
+  return datetime.datetime(*(time.strptime(t, "%a %b %d %H:%M:%S +0000 %Y")[0:6]))
 
 class Message:
   def __init__(self, client, data):

=== modified file 'gwibber/twitter.py'
--- gwibber/twitter.py	2008-07-19 02:39:29 +0000
+++ gwibber/twitter.py	2008-07-28 03:29:10 +0000
@@ -15,7 +15,7 @@
 NICK_PARSE = re.compile("@([A-Za-z0-9]+)")
 
 def parse_time(t):
-  return datetime.datetime.strptime(t, "%a %b %d %H:%M:%S +0000 %Y")
+    return datetime.datetime(*(time.strptime(t, "%a %b %d %H:%M:%S +0000 %Y")[0:6]))
 
 class Message:
   def __init__(self, client, data):

