diff -up gwibber-867bzr/gwibber/lib/gtk/gowalla.py.NEW gwibber-867bzr/gwibber/lib/gtk/gowalla.py
--- gwibber-867bzr/gwibber/lib/gtk/gowalla.py.NEW	2010-09-23 13:17:52.955184949 -0400
+++ gwibber-867bzr/gwibber/lib/gtk/gowalla.py	2010-09-23 13:22:37.900184989 -0400
@@ -0,0 +1,44 @@
+#
+# Gowalla support for Gwibber
+#
+# Derived from Foursquare support (foursquare.py)
+#
+# Copyright 2010, Tom "spot" Callaway <tcallawa@redhat.com>
+#
+# Permission to use, copy, modify, and distribute this software and its
+# documentation for any purpose and without fee is hereby granted,
+# provided that the above copyright notice appear in all copies and that
+# both that copyright notice and this permission notice appear in
+# supporting documentation.
+#
+# THE AUTHOR PROVIDES THIS SOFTWARE ''AS IS'' AND ANY EXPRESSED OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+# OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+# NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import gtk
+from gtk import Builder
+import gwibber.microblog
+
+class AccountWidget(gtk.VBox):
+  """AccountWidget: A widget that provides a user interface for configuring Gowalla accounts in Gwibber
+  """
+
+  def __init__(self, account=None, dialog=None):
+    """Creates the account pane for configuring Gowalla accounts"""
+    gtk.VBox.__init__( self, False, 20 )
+    self.ui = gtk.Builder()
+    self.ui.set_translation_domain ("gwibber")
+    self.ui.add_from_file (gwibber.resources.get_ui_asset("gwibber-accounts-gowalla.ui"))
+    self.ui.connect_signals(self)
+    self.vbox_settings = self.ui.get_object("vbox_settings")
+    self.pack_start(self.vbox_settings, False, False)
+    self.show_all()
+    if dialog:
+      dialog.get_object("vbox_create").show()
diff -up gwibber-867bzr/gwibber/lib/gtk/__init__.py.NEW gwibber-867bzr/gwibber/lib/gtk/__init__.py
--- gwibber-867bzr/gwibber/lib/gtk/__init__.py.NEW	2010-09-23 13:17:15.513185001 -0400
+++ gwibber-867bzr/gwibber/lib/gtk/__init__.py	2010-09-23 13:17:28.901184999 -0400
@@ -1 +1 @@
-__all__ = ["twitter", "identica", "flickr", "facebook", "friendfeed", "statusnet", "digg", "qaiku", "buzz", "pingfm", "foursquare"]
+__all__ = ["twitter", "identica", "flickr", "facebook", "friendfeed", "statusnet", "digg", "qaiku", "buzz", "pingfm", "foursquare", "gowalla"]
diff -up gwibber-867bzr/gwibber/microblog/dispatcher.py.NEW gwibber-867bzr/gwibber/microblog/dispatcher.py
--- gwibber-867bzr/gwibber/microblog/dispatcher.py.NEW	2010-09-23 13:22:43.941184999 -0400
+++ gwibber-867bzr/gwibber/microblog/dispatcher.py	2010-09-23 13:23:04.068184978 -0400
@@ -4,7 +4,7 @@ import multiprocessing, threading, trace
 import gobject, dbus, dbus.service
 import twitter, identica, statusnet, flickr, facebook
 import qaiku, friendfeed, digg, buzz, pingfm
-import foursquare
+import foursquare, gowalla
 import sqlite3, mx.DateTime, re, uuid
 import urlshorter, storage, network, util, config
 
@@ -42,6 +42,7 @@ PROTOCOLS = {
   "buzz": buzz,
   "pingfm": pingfm,
   "foursquare": foursquare,
+  "gowalla": gowalla,
 }
 
 FEATURES = json.loads(GWIBBER_OPERATIONS)
diff -up gwibber-867bzr/gwibber/microblog/gowalla.py.NEW gwibber-867bzr/gwibber/microblog/gowalla.py
--- gwibber-867bzr/gwibber/microblog/gowalla.py.NEW	2010-09-23 09:44:48.714185000 -0400
+++ gwibber-867bzr/gwibber/microblog/gowalla.py	2010-09-23 13:16:46.095185001 -0400
@@ -0,0 +1,123 @@
+#
+# Gowalla support for Gwibber
+#
+# Derived from Foursquare support (foursquare.py)
+# 
+# Copyright 2010, Tom "spot" Callaway <tcallawa@redhat.com>
+#
+# Permission to use, copy, modify, and distribute this software and its
+# documentation for any purpose and without fee is hereby granted,
+# provided that the above copyright notice appear in all copies and that
+# both that copyright notice and this permission notice appear in
+# supporting documentation.
+# 
+# THE AUTHOR PROVIDES THIS SOFTWARE ''AS IS'' AND ANY EXPRESSED OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+# OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+# NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import network, json, util, htmllib, re
+from util import log
+from util import exceptions
+from gettext import lgettext as _
+log.logger.name = "Gowalla"
+
+PROTOCOL_INFO = {
+  "name": "Gowalla",
+  "version": "1.0",
+  
+  "config": [
+    "private:password",
+    "receive_enabled",
+    "username",
+    "color",
+  ],
+
+  "authtype": "none",
+  "color": "#f7911f",
+
+  "features": [
+    "receive",
+  ],
+
+  "default_streams": [
+    "receive",
+  ],
+}
+
+URL_PREFIX = "https://api.gowalla.com"
+
+class Client:
+  def __init__(self, acct):
+    self.account = acct
+
+  def _message(self, data):
+
+    m = {}; 
+    m["mid"] = str(data["user"]["first_name"]) + str(data["user"]["last_name"])
+    m["service"] = "gowalla"
+    m["account"] = self.account["id"]
+    m["time"] = util.parsetime(data["created_at"])
+
+    messagetext = ""
+    text = ""
+    if data.has_key("spot"):
+        if data.has_key("message"):
+            messagetext += data["message"] + "<br/><br/>"
+            text += data["message"] + "\n"
+	m["url"] = "<a href='http://gowalla.com%s'>" % data["spot"]["url"]
+	m["url"] = "http://gowalla.com%s" % data["spot"]["url"]
+        img = "<table><tr><td><img src='%s'/></td><td>" % data["spot"]["image_url"]
+        messagetext += img + "Checked in at <a href='" + m["url"] + "'>" + data["spot"]["name"] + "</a>"
+	text += "Checked in at " + data["spot"]["name"]
+    else:
+        if data.has_key("message"):
+            messagetext += data["message"] + "<br/><br/>"
+            text += data["message"] + "\n"
+        else:
+            text= "Checked in off the grid"
+            messagetext= "<table><tr><td><img src='http://gowalla.com/favicon.ico'/></td><td>Checked in off the grid"
+
+    m["text"] = text
+    m["content"] = messagetext + "</td></tr></table>"
+    m["html"] = messagetext + "</td></tr></table>"
+
+    m["sender"] = {}
+    m["sender"]["image"] = data["user"]["image_url"]
+    m["sender"]["url"] = "http://gowalla.com%s" % data["user"]["url"]
+    fullname = ""
+    if data["user"].has_key("first_name"):
+        fullname += data["user"]["first_name"] + " "
+    if data["user"].has_key("last_name"):
+        fullname += data["user"]["last_name"]
+
+    m["sender"]["name"] = fullname
+    m["sender"]["nick"] = fullname
+        
+    m["source"] = "<a href='http://gowalla.com/'>Gowalla</a>"
+
+    return m
+
+  def _get(self, path, parse="message", post=False, single=False, **args):
+    url = "/".join((URL_PREFIX, path))
+
+    data = network.Download(url, util.compact(args) or None, post,
+        self.account["username"], self.account["password"]).get_json()
+
+    checkins = data["activity"]
+    if single: return [getattr(self, "_%s" % parse)(checkins)]
+    if parse: return [getattr(self, "_%s" % parse)(m) for m in checkins]
+    else: return []
+
+  def __call__(self, opname, **args):
+    return getattr(self, opname)(**args)
+
+  def receive(self):
+    return self._get("users/%s/activity/friends" % self.account["username"])
+
diff -up gwibber-867bzr/ui/gwibber-accounts-gowalla.ui.NEW gwibber-867bzr/ui/gwibber-accounts-gowalla.ui
--- gwibber-867bzr/ui/gwibber-accounts-gowalla.ui.NEW	2010-09-23 13:23:40.381184999 -0400
+++ gwibber-867bzr/ui/gwibber-accounts-gowalla.ui	2010-09-23 13:23:32.281184974 -0400
@@ -0,0 +1,219 @@
+<?xml version="1.0"?>
+<interface>
+  <requires lib="gtk+" version="2.16"/>
+  <!-- interface-naming-policy toplevel-contextual -->
+  <object class="GtkVBox" id="vbox_settings">
+    <property name="visible">True</property>
+    <property name="orientation">vertical</property>
+    <property name="spacing">6</property>
+    <child>
+      <object class="GtkTable" id="table_common_settings">
+        <property name="visible">True</property>
+        <property name="n_rows">3</property>
+        <property name="n_columns">3</property>
+        <property name="column_spacing">12</property>
+        <property name="row_spacing">6</property>
+        <child>
+          <object class="GtkEntry" id="password">
+            <property name="visible">True</property>
+            <property name="can_focus">True</property>
+            <property name="visibility">False</property>
+            <property name="invisible_char">&#x25CF;</property>
+          </object>
+          <packing>
+            <property name="left_attach">1</property>
+            <property name="right_attach">3</property>
+            <property name="top_attach">2</property>
+            <property name="bottom_attach">3</property>
+            <property name="y_options"></property>
+          </packing>
+        </child>
+        <child>
+          <object class="GtkEntry" id="username">
+            <property name="visible">True</property>
+            <property name="can_focus">True</property>
+            <property name="invisible_char">&#x25CF;</property>
+          </object>
+          <packing>
+            <property name="left_attach">1</property>
+            <property name="right_attach">3</property>
+            <property name="y_options"></property>
+          </packing>
+        </child>
+        <child>
+          <object class="GtkLabel" id="label_username">
+            <property name="visible">True</property>
+            <property name="xalign">0</property>
+            <property name="label" translatable="yes">Login I_D:</property>
+            <property name="use_underline">True</property>
+            <property name="mnemonic_widget">username</property>
+          </object>
+          <packing>
+            <property name="x_options">GTK_FILL</property>
+            <property name="y_options"></property>
+          </packing>
+        </child>
+        <child>
+          <object class="GtkLabel" id="label_password">
+            <property name="visible">True</property>
+            <property name="xalign">0</property>
+            <property name="label" translatable="yes">Pass_word:</property>
+            <property name="use_underline">True</property>
+            <property name="justify">right</property>
+            <property name="mnemonic_widget">password</property>
+          </object>
+          <packing>
+            <property name="top_attach">2</property>
+            <property name="bottom_attach">3</property>
+            <property name="x_options">GTK_FILL</property>
+            <property name="y_options"></property>
+          </packing>
+        </child>
+        <child>
+          <object class="GtkVBox" id="vbox1">
+            <property name="visible">True</property>
+            <property name="orientation">vertical</property>
+            <child>
+              <object class="GtkLabel" id="label_username_example">
+                <property name="visible">True</property>
+                <property name="xalign">0</property>
+                <property name="xpad">3</property>
+                <property name="label" translatable="yes">&lt;span size="small"&gt;&lt;b&gt;Example:&lt;/b&gt; username@email.com&lt;/span&gt;</property>
+                <property name="use_markup">True</property>
+              </object>
+              <packing>
+                <property name="position">0</property>
+              </packing>
+            </child>
+          </object>
+          <packing>
+            <property name="left_attach">1</property>
+            <property name="right_attach">2</property>
+            <property name="top_attach">1</property>
+            <property name="bottom_attach">2</property>
+          </packing>
+        </child>
+        <child>
+          <placeholder/>
+        </child>
+        <child>
+          <placeholder/>
+        </child>
+      </object>
+      <packing>
+        <property name="expand">False</property>
+        <property name="fill">False</property>
+        <property name="position">0</property>
+      </packing>
+    </child>
+    <child>
+      <object class="GtkExpander" id="expander1">
+        <property name="visible">True</property>
+        <property name="can_focus">True</property>
+        <child>
+          <object class="GtkVBox" id="vbox_advanced">
+            <property name="visible">True</property>
+            <property name="orientation">vertical</property>
+            <property name="spacing">6</property>
+            <child>
+              <object class="GtkLabel" id="label3">
+                <property name="visible">True</property>
+                <property name="label" translatable="yes">Account Status</property>
+                <attributes>
+                  <attribute name="weight" value="bold"/>
+                </attributes>
+              </object>
+              <packing>
+                <property name="position">0</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkTable" id="table_advanced_settings">
+                <property name="visible">True</property>
+                <property name="n_rows">2</property>
+                <property name="n_columns">3</property>
+                <property name="column_spacing">12</property>
+                <property name="row_spacing">6</property>
+                <child>
+                  <object class="GtkCheckButton" id="receive_enabled">
+                    <property name="label" translatable="yes">_Receive Messages</property>
+                    <property name="visible">True</property>
+                    <property name="can_focus">True</property>
+                    <property name="receives_default">False</property>
+                    <property name="use_underline">True</property>
+                    <property name="active">True</property>
+                    <property name="draw_indicator">True</property>
+                  </object>
+                  <packing>
+                    <property name="right_attach">3</property>
+                    <property name="x_options">GTK_FILL</property>
+                    <property name="y_options"></property>
+                  </packing>
+                </child>
+              </object>
+              <packing>
+                <property name="position">1</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkLabel" id="label2">
+                <property name="visible">True</property>
+                <property name="label" translatable="yes">Appearance</property>
+                <attributes>
+                  <attribute name="weight" value="bold"/>
+                </attributes>
+              </object>
+              <packing>
+                <property name="position">2</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkHBox" id="hbox1">
+                <property name="visible">True</property>
+                <property name="homogeneous">True</property>
+                <child>
+                  <object class="GtkLabel" id="label4">
+                    <property name="visible">True</property>
+                    <property name="label" translatable="yes">Message Color:</property>
+                  </object>
+                  <packing>
+                    <property name="position">0</property>
+                  </packing>
+                </child>
+                <child>
+                  <object class="GtkColorButton" id="color">
+                    <property name="visible">True</property>
+                    <property name="can_focus">True</property>
+                    <property name="receives_default">True</property>
+                    <property name="color">#000000000000</property>
+                  </object>
+                  <packing>
+                    <property name="expand">False</property>
+                    <property name="position">1</property>
+                  </packing>
+                </child>
+              </object>
+              <packing>
+                <property name="position">3</property>
+              </packing>
+            </child>
+          </object>
+        </child>
+        <child type="label">
+          <object class="GtkLabel" id="label1">
+            <property name="visible">True</property>
+            <property name="label" translatable="yes">Advanced</property>
+            <attributes>
+              <attribute name="weight" value="bold"/>
+            </attributes>
+          </object>
+        </child>
+      </object>
+      <packing>
+        <property name="expand">False</property>
+        <property name="fill">False</property>
+        <property name="position">1</property>
+      </packing>
+    </child>
+  </object>
+</interface>
diff -up gwibber-867bzr/ui/icons/breakdance/16x16/gowalla.png.NEW gwibber-867bzr/ui/icons/breakdance/16x16/gowalla.png
Binary files gwibber-867bzr/ui/icons/breakdance/16x16/gowalla.png.NEW and gwibber-867bzr/ui/icons/breakdance/16x16/gowalla.png differ
diff -up gwibber-867bzr/ui/icons/breakdance/22x22/gowalla.png.NEW gwibber-867bzr/ui/icons/breakdance/22x22/gowalla.png
Binary files gwibber-867bzr/ui/icons/breakdance/22x22/gowalla.png.NEW and gwibber-867bzr/ui/icons/breakdance/22x22/gowalla.png differ
diff -up gwibber-867bzr/ui/icons/breakdance/32x32/gowalla.png.NEW gwibber-867bzr/ui/icons/breakdance/32x32/gowalla.png
Binary files gwibber-867bzr/ui/icons/breakdance/32x32/gowalla.png.NEW and gwibber-867bzr/ui/icons/breakdance/32x32/gowalla.png differ
diff -up gwibber-867bzr/ui/icons/breakdance/scalable/gowalla.png.NEW gwibber-867bzr/ui/icons/breakdance/scalable/gowalla.png
Binary files gwibber-867bzr/ui/icons/breakdance/scalable/gowalla.png.NEW and gwibber-867bzr/ui/icons/breakdance/scalable/gowalla.png differ
