diff -up gwibber-886bzr/gwibber/microblog/util/const.py.fedora gwibber-886bzr/gwibber/microblog/util/const.py
--- gwibber-886bzr/gwibber/microblog/util/const.py.fedora	2010-10-11 15:47:38.029504004 -0400
+++ gwibber-886bzr/gwibber/microblog/util/const.py	2010-10-11 15:47:58.751504002 -0400
@@ -15,7 +15,7 @@ from os import environ
 if environ.has_key("FB_APP_KEY"):
   FB_APP_KEY = environ["FB_APP_KEY"]
 else:
-  FB_APP_KEY = "71b85c6d8cb5bbb9f1a3f8bbdcdd4b05"
+  FB_APP_KEY = "1e51e9b88c054048dc201f81ea5543b7"
 
 TWITTER_OAUTH_KEY = "VDOuA5qCJ1XhjaSa4pl76g"
 TWITTER_OAUTH_SECRET = "BqHlB8sMz5FhZmmFimwgiIdB0RiBr72Y0bio49IVJM"
