diff -up gwibber-865bzr/gwibber/gwui.py.undefined gwibber-865bzr/gwibber/gwui.py
--- gwibber-865bzr/gwibber/gwui.py.undefined	2010-09-21 13:42:39.795185019 -0400
+++ gwibber-865bzr/gwibber/gwui.py	2010-09-21 13:42:50.457185002 -0400
@@ -955,7 +955,7 @@ class InputTextView(gtk.TextView):
             error_handler=error_shortened)
 
   def set_overlay_text(self, text):
-    if not self.pango_overlay:
+    if not hasattr(self, 'pango_overlay'):
       self.pango_overlay = self.create_pango_layout("")
     self.pango_overlay.set_markup(self.overlay_text % (self.overlay_color, text))
 
