diff -up gwibber-867bzr/gwibber/gwui.py.BAD gwibber-867bzr/gwibber/gwui.py
--- gwibber-867bzr/gwibber/gwui.py.BAD	2010-09-24 11:21:19.897185001 -0400
+++ gwibber-867bzr/gwibber/gwui.py	2010-09-24 11:21:44.167184972 -0400
@@ -939,7 +939,7 @@ class InputTextView(gtk.TextView):
             buf.insert(iter_start, shortened_url)
         def error_shortened(dbus_exc):
             "Internal shortening-url-died function: a closure"
-            iter = buf.get_iter_at_mark(mark)
+            iter = buf.get_iter_at_mark(mark_start)
             buf.insert(iter, text) # shortening failed
 
         # set a mark at iter, so that the callback knows where to insert
