diff -up gwibber-832bzr/gwibber/microblog/util/custom.py.oauth gwibber-832bzr/gwibber/microblog/util/custom.py
--- gwibber-832bzr/gwibber/microblog/util/custom.py.oauth	2010-08-31 17:09:53.519134001 -0400
+++ gwibber-832bzr/gwibber/microblog/util/custom.py	2010-08-31 17:10:37.514134003 -0400
@@ -0,0 +1,17 @@
+""" 
+WHY ARE YOU LOOKING IN HERE? THIS IS TOP SEKRIT!
+
+Seriously, though. This is really, really dumb. Twitter doesn't understand how to use oauth.
+
+So, we have to explicitly hardcode a key for this client to use. In plain text.
+
+Dumb, dumbdumbdumb.
+
+Google gets it right, heck, even Facebook gets it right, why can't Twitter?
+
+Tom "spot" Callaway <tcallawa@redhat.com>
+Tuesday Aug 31, 2010
+"""
+
+TWITTER_OAUTH_KEY = "GDbYbywyvbcPfe26MqrifQ"
+TWITTER_OAUTH_SECRET = "U4C34CzUW6YYNFinX8RjSTsWWn2bWX08WVdFsxyQ"
