diff -up gwibber-867bzr/gwibber/gwui.py.BAD gwibber-867bzr/gwibber/gwui.py
--- gwibber-867bzr/gwibber/gwui.py.BAD	2010-09-24 14:01:18.721249996 -0400
+++ gwibber-867bzr/gwibber/gwui.py	2010-09-24 14:05:27.224250053 -0400
@@ -255,6 +255,10 @@ class WebUi(webkit.WebView):
     template = Template(template, lookup=TemplateLookup(directories=lookup_paths))
     content = template.render(theme=util.get_theme_colors(), resources=resources, _=_, **kwargs)
 
+    # Content occasionally ends	up with	nulls (\x00) which causes a traceback
+    # This works around	it by replacing \x00 with a space whenever it is found.
+    content = content.replace('\x00', ' ')
+
     # Avoid navigation redraw crashes
     if isinstance(self, Navigation) and not self.get_property("visible"):
       return content
