diff -up gwibber-894bzr/gwibber/microblog/network.py.no-libproxy gwibber-894bzr/gwibber/microblog/network.py
--- gwibber-894bzr/gwibber/microblog/network.py.no-libproxy	2010-10-13 15:12:13.887908996 -0400
+++ gwibber-894bzr/gwibber/microblog/network.py	2010-10-13 15:12:30.290909130 -0400
@@ -4,10 +4,10 @@ import urllib, pycurl, json, StringIO
 from util import log
 from util import exceptions
 
-try:
-  import libproxy
-except:
-  libproxy = None
+# try:
+#   import libproxy
+# except:
+libproxy = None
 
 class CurlDownloader:
   def __init__(self, url, params=None, post=False, username=None, password=None, header=None, body=None):
diff -up gwibber-894bzr/gwibber/microblog/util/facelib.py.no-libproxy gwibber-894bzr/gwibber/microblog/util/facelib.py
--- gwibber-894bzr/gwibber/microblog/util/facelib.py.no-libproxy	2010-10-13 15:02:57.000000000 -0400
+++ gwibber-894bzr/gwibber/microblog/util/facelib.py	2010-10-13 15:11:27.927909002 -0400
@@ -1,4 +1,3 @@
-#! /usr/bin/env python
 #
 # pyfacebook - Python bindings for the Facebook API
 #
@@ -61,10 +60,10 @@ import mimetypes
 import pycurl
 import StringIO
 
-try:
-    import libproxy
-except:
-    libproxy = None
+# try:
+#     import libproxy
+# except:
+libproxy = None
 
 # try to use simplejson first, otherwise fallback to XML
 RESPONSE_FORMAT = 'JSON'
