diff -up gwibber-875bzr/ui/templates/navigation.mako.TEST gwibber-875bzr/ui/templates/navigation.mako
--- gwibber-875bzr/ui/templates/navigation.mako.TEST	2010-10-06 15:00:44.000000000 -0400
+++ gwibber-875bzr/ui/templates/navigation.mako	2010-10-07 12:29:33.481095835 -0400
@@ -75,10 +75,23 @@ def active(item):
             return false;
         });
 
-        $(".compact .stream").click(function() {
-            selectstream($(this));
-            document.location = generateStreamURL($(this));
+        $(".compact .stream").mousedown(function(event) {
+            switch (event.which) {
+              case 1:
+              selectstream($(this));
+              document.location = generateStreamURL($(this));
+              break;
+              case 2:
+              alert("     @..@  \n      (\\--/) \n   (.>__<.)\n   ^^^  ^^^\n\nRIBBIT!");
+              break;
+              case 3:
+              document.location = "gwibber:/close?transient=" + $(this).attr("transient");
+              return false;
+              break;
+              default:
+            }
         });
+
       });
     </script>
 
