diff -r 4b7feca2c936 gwibber/client.py
--- a/gwibber/client.py	Sun Aug 17 00:44:02 2008 -0400
+++ b/gwibber/client.py	Sun Aug 17 00:46:34 2008 -0400
@@ -10,6 +10,7 @@
 import sys, time, os, threading, datetime
 import gtk, gtk.glade, gobject, table, webkit
 import microblog, gwui, config, gintegration
+import mx.DateTime
 
 gtk.gdk.threads_init()
 
@@ -600,7 +601,7 @@
 
         self.statusbar.pop(0)
         self.statusbar.push(0, "Last update: %s" % time.strftime("%I:%M:%S %p"))
-        self.last_update = datetime.datetime.utcnow()
+        self.last_update = mx.DateTime.gmt()
         
       finally: gobject.idle_add(self.throbber.clear)
     
