diff -up pygtk-2.12.1/gtk/pygtkcellrenderer.c.pygtk-generic-cell-renderer-activate pygtk-2.12.1/gtk/pygtkcellrenderer.c
--- pygtk-2.12.1/gtk/pygtkcellrenderer.c.pygtk-generic-cell-renderer-activate	2007-11-01 12:20:18.000000000 -0400
+++ pygtk-2.12.1/gtk/pygtkcellrenderer.c	2008-03-17 12:24:47.000000000 -0400
@@ -250,7 +250,8 @@ pygtk_generic_cell_renderer_activate (Gt
     Py_DECREF(py_ret);
 
 out:
-    pygtk_boxed_unref_shared(py_event);
+    if (py_event != Py_None)
+        pygtk_boxed_unref_shared(py_event);
     Py_DECREF(py_widget);
     Py_DECREF(py_background_area);
     Py_DECREF(py_cell_area);
