--- src/libreoffice-impress-%(version)s/slideshow/source/engine/OGLTrans/unx/makefile.mk.orig	2011-06-02 08:12:46.340615086 +0400
+++ src/libreoffice-impress-%(version)s/slideshow/source/engine/OGLTrans/unx/makefile.mk	2011-06-02 08:13:23.485614933 +0400
@@ -55,6 +55,7 @@
 SHL1STDLIBS= $(SALLIB) $(VCLLIB) $(CPPULIB) $(CPPUHELPERLIB) $(COMPHELPERLIB) $(CANVASTOOLSLIB)
 
 SHL1STDLIBS+= \
+    -L%(libdir)s/xorg.mesa.3d \
     -lGL \
     -lGLU \
     -lX11
