--- libreoffice-bootstrap-%(version)s/Makefile.orig	2011-06-24 15:59:22.907750518 +0100
+++ libreoffice-bootstrap-%(version)s/Makefile	2011-06-24 15:59:47.010750526 +0100
@@ -47,12 +47,8 @@
         ./bootstrap
 
 src.downloaded: ooo.lst download
-ifeq (,YES)
 	@. ./*[Ee]nv.[Ss]et.sh && \
 	$$SRC_ROOT/download $$SRC_ROOT/ooo.lst && touch $@
-else
-	@echo "Automatic fetching of external tarballs is disabled."
-endif
 
 fetch: src.downloaded
 
