diff -ru solenv/inc.orig/settings.mk solenv/inc/settings.mk
--- solenv/inc.orig/settings.mk	2007-03-20 16:49:27.934114521 +0300
+++ solenv/inc/settings.mk	2007-03-20 16:49:55.143983688 +0300
@@ -33,6 +33,9 @@
 #
 #*************************************************************************
 MKFILENAME:=SETTINGS.MK
+.IF "$(UNIXVERSIONNAMES)"==""
+UNIXVERSIONNAMES=FAKE
+.ENDIF
 
 # smaller/greater arithmetic's like ".IF 400<=200" are an OOo extention to
 # the initial dmake 4.1PL0 (unfortunately called 4.10) version and are
diff -ru solenv/inc.orig/unx.mk solenv/inc/unx.mk
--- solenv/inc.orig/unx.mk	2007-03-20 16:51:06.493648989 +0300
+++ solenv/inc/unx.mk	2007-03-20 16:50:43.363756193 +0300
@@ -197,9 +197,11 @@
 SOLARSHAREDBIN=$(SOLARLIBDIR)
 SONAME_SWITCH*=-h
 
-.IF "$(UNIXVERSIONNAMES)"!=""
+.IF "$(UNIXVERSIONNAMES)" != ""
+.IF "$(UNIXVERSIONNAMES)" != "FAKE"
 DLLPOST!:=$(DLLPOST).$($(UNIXVERSIONNAMES)_MAJOR)
-.ENDIF			# "$(UNIXVERSIONNAMES)"!=""
+.ENDIF
+.ENDIF
 
 # enable building/linking KDE-dependent code in both OOo and SO build environment
 .IF "$(ENABLE_KDE)" != ""
