diff -r 33e3148de518 images/Makefile.am
--- a/images/Makefile.am	Wed Apr 02 13:51:15 2008 -0400
+++ b/images/Makefile.am	Wed Apr 02 15:53:46 2008 -0400
@@ -1,8 +1,8 @@
 imagesdir = $(imagedir)
 images_DATA =	       \
 	icon.png       \
-	freedesktop.png
+	foresight.png
 
 EXTRA_DIST =	       \
 	icon.png       \
-	freedesktop.png
+	foresight.png
diff -r 33e3148de518 images/Makefile.in
--- a/images/Makefile.in	Wed Apr 02 13:51:15 2008 -0400
+++ b/images/Makefile.in	Wed Apr 02 15:53:46 2008 -0400
@@ -274,10 +274,10 @@
 imagesdir = $(imagedir)
 images_DATA = \
 	icon.png       \
-	freedesktop.png
+	foresight.png
 
 EXTRA_DIST = \
 	icon.png       \
-	freedesktop.png
+	foresight.png
 
 all: all-am
diff -r 33e3148de518 metadata/compiz-cube.schemas
--- a/metadata/compiz-cube.schemas	Wed Apr 02 13:51:15 2008 -0400
+++ b/metadata/compiz-cube.schemas	Wed Apr 02 15:53:46 2008 -0400
@@ -39,7 +39,7 @@
       <applyto>/apps/compiz/plugins/cube/screen0/options/color</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>#feffc7ff</default>
+      <default>#ffffff</default>
       <locale name="C">
         <short>Cube Color</short>
         <long>Color of top and bottom sides of the cube</long>
@@ -73,7 +73,7 @@
       <owner>compiz</owner>
       <type>list</type>
       <list_type>string</list_type>
-      <default>[freedesktop]</default>
+      <default>[foresight]</default>
       <locale name="C">
         <short>Image files</short>
         <long>List of PNG and SVG files that should be rendered on top face of cube</long>
@@ -195,7 +195,7 @@
       <applyto>/apps/compiz/plugins/cube/screen0/options/adjust_image</applyto>
       <owner>compiz</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
         <short>Adjust Image</short>
         <long>Adjust top face image to rotation</long>
diff -r 33e3148de518 metadata/compizrc
--- a/metadata/compizrc	Wed Apr 02 13:51:15 2008 -0400
+++ b/metadata/compizrc	Wed Apr 02 15:53:46 2008 -0400
@@ -119,10 +119,10 @@
 prev_slide_key=Backspace
 
 [cube_screen0]
-color=#feffc7ff
+color=#ffffff
 in=false
 scale_image=false
-images=freedesktop
+images=foresight
 skydome=false
 skydome_image=
 skydome_animated=false
diff -r 33e3148de518 metadata/cube.xml.in
--- a/metadata/cube.xml.in	Wed Apr 02 13:51:15 2008 -0400
+++ b/metadata/cube.xml.in	Wed Apr 02 15:53:46 2008 -0400
@@ -33,9 +33,9 @@
 		<_short>Cube Color</_short>
 		<_long>Color of top and bottom sides of the cube</_long>
 		<default>
-		    <red>0xfefe</red>
+		    <red>0xffff</red>
 		    <green>0xffff</green>
-		    <blue>0xc7c7</blue>
+		    <blue>0xffff</blue>
 		</default>
 	    </option>
 	    <option name="in" type="bool">
@@ -53,7 +53,7 @@
 		<_long>List of PNG and SVG files that should be rendered on top face of cube</_long>
 		<type>string</type>
 		<default>
-		    <value>freedesktop</value>
+		    <value>foresight</value>
 		</default>
 	    </option>
 	    <option name="skydome" type="bool">
@@ -83,9 +83,9 @@
 		<_short>Skydome Gradient End Color</_short>
 		<_long>Color to use for the bottom color-stop of the skydome-fallback gradient</_long>
 		<default>
-		    <red>0xfefe</red>
+		    <red>0xffff</red>
 		    <green>0xffff</green>
-		    <blue>0xc7c7</blue>
+		    <blue>0xffff</blue>
 		</default>
 	    </option>
 	    <option name="acceleration" type="float">
@@ -125,7 +125,7 @@
 	    <option name="adjust_image" type="bool">
 		<_short>Adjust Image</_short>
 		<_long>Adjust top face image to rotation</_long>
-		<default>false</default>
+		<default>true</default>
 	    </option>
 	    <option name="active_opacity" type="float">
 		<_short>Opacity During Rotation</_short>
