diff -ruN gnome-volume-manager-1.3.1-old/src/manager.c gnome-volume-manager-1.3.1/src/manager.c
--- gnome-volume-manager-1.3.1-old/src/manager.c	2005-03-31 18:27:07.000000000 +0200
+++ gnome-volume-manager-1.3.1/src/manager.c	2005-04-13 14:08:36.049442384 +0200
@@ -257,31 +257,32 @@
 /*
  * gvm_check_dvd - is this a Video DVD?  If so, do something about it.
  *
- * Returns TRUE if this was a Video DVD and FALSE otherwise.
+ * Returns TRUE if the Video DVD action was taken and FALSE otherwise.
  */
 static gboolean
 gvm_check_dvd (const char *device, const char *mount_point, const char *udi)
 {
 	char *path;
-	gboolean retval;
+	gboolean isdvd;
 
 	path = g_build_path (G_DIR_SEPARATOR_S, mount_point, "video_ts", NULL);
-	retval = g_file_test (path, G_FILE_TEST_IS_DIR);
+	isdvd = g_file_test (path, G_FILE_TEST_IS_DIR);
 	g_free (path);
 
 	/* try the other name, if needed */
-	if (retval == FALSE) {
+	if (isdvd == FALSE) {
 		path = g_build_path (G_DIR_SEPARATOR_S, mount_point,
 				     "VIDEO_TS", NULL);
-		retval = g_file_test (path, G_FILE_TEST_IS_DIR);
+		isdvd = g_file_test (path, G_FILE_TEST_IS_DIR);
 		g_free (path);
 	}
 
-	if (retval && config.autoplay_dvd)
+	if (isdvd && config.autoplay_dvd) {
 		gvm_run_command (device, config.autoplay_dvd_command,
 				 mount_point, udi);
-
-	return retval;
+    return TRUE;
+  }
+  return FALSE;
 }
 
 /*
@@ -289,7 +290,8 @@
  * unit from a digital camera (e.g., a compact flash card).  If it is, then
  * ask the user if he wants to import the photos.
  *
- * Returns TRUE if there were photos on this device, FALSE otherwise
+ * Returns TRUE if there were photos on this device were imported, 
+ * FALSE otherwise
  *
  * FIXME: Should probably not prompt the user and just do it automatically.
  *        This now makes sense, as gphoto added an import mode.
@@ -308,7 +310,6 @@
 	if (!g_file_test (dcim_path, G_FILE_TEST_IS_DIR))
 		goto out;
 
-	retval = TRUE;
 	dbg ("Photos detected: %s\n", dcim_path);
 
 	/* add the "content.photos" capability to this device */
@@ -322,6 +323,7 @@
 	}
 
 	if (config.autophoto) {
+		retval=TRUE;
 		askme = gtk_message_dialog_new (NULL, 0, GTK_MESSAGE_WARNING,
 						GTK_BUTTONS_NONE,
 						_("Import photos from device?"));
