diff -ruN gnome-volume-manager-1.5.14-old/src/manager.c gnome-volume-manager-1.5.14/src/manager.c
--- gnome-volume-manager-1.5.14-old/src/manager.c	2006-02-28 11:56:52.000000000 +0100
+++ gnome-volume-manager-1.5.14/src/manager.c	2006-02-28 12:17:01.000000000 +0100
@@ -3105,9 +3105,20 @@
 	char *path;
 	
 	if (!(exists & 0x2)) {
-		path = g_build_filename (GVM_CONSOLE_AUTH_DIR, g_get_user_name (), NULL);
-		exists = (access (path, F_OK) == 0) ? 0x3 : 0x2;
-		g_free (path);
+                path = g_strdup_printf ("%s:", g_get_user_name ());
+                GDir* consoledir;
+                const gchar* entry;
+                
+                if ( (consoledir = g_dir_open (GVM_CONSOLE_AUTH_DIR, 0, NULL)) != NULL) {
+                    while ((entry = g_dir_read_name (consoledir)) != NULL) {
+                        if (g_str_has_prefix (entry, path)) {
+                            exists |= 0x1;
+                            break;
+                        } 
+                    }
+                    g_dir_close (consoledir);
+                }
+                exists |= 0x2;
 	}
 	
 	return (exists & 0x1);
