diff -ruN gnome-volume-manager-1.3.1-old/src/manager.c gnome-volume-manager-1.3.1/src/manager.c
--- gnome-volume-manager-1.3.1-old/src/manager.c	2005-06-22 14:42:20.000000000 +0200
+++ gnome-volume-manager-1.3.1/src/manager.c	2005-06-22 14:45:27.000000000 +0200
@@ -1153,6 +1153,8 @@
 	char **volumes;
 	char *udi;
 	char *device_file;
+	char *drive_udi = NULL;
+	gboolean do_mount;
 	DBusError error;
 
 	if (!config.automount_media)
@@ -1188,6 +1190,21 @@
 							       NULL), 
 			    "filesystem") != 0) 
 			continue;
+		drive_udi = libhal_device_get_property_string(ctx, udi, "info.parent", NULL);
+  
+		do_mount = TRUE;
+		if (libhal_device_property_exists(ctx, drive_udi, "storage.hotpluggable", NULL) &&
+				libhal_device_get_property_bool(ctx, drive_udi, "storage.hotpluggable", NULL)) {
+			do_mount = config.automount_drives;
+		} else if (
+			libhal_device_property_exists(ctx, drive_udi, "storage.removable", NULL) &&
+			libhal_device_get_property_bool(ctx, drive_udi, "storage.removable", NULL)) {
+			do_mount = config.automount_media;
+		} 
+		libhal_free_string(drive_udi);
+      
+		if (!do_mount) continue;
+
 
 		device_file = libhal_device_get_property_string (ctx, udi,
 				 				 "block.device",
