diff -ruN gnome-volume-manager-1.5.10-old/src/manager.c gnome-volume-manager-1.5.10/src/manager.c
--- gnome-volume-manager-1.5.10-old/src/manager.c	2006-01-26 11:50:10.000000000 +0100
+++ gnome-volume-manager-1.5.10/src/manager.c	2006-01-26 11:50:26.000000000 +0100
@@ -2246,7 +2246,6 @@
 			g_free (policy);
 		} else if (gvm_user_active_at_console ()) {
 			dbg ("not in mount queue: %s\n", udi);
-			gvm_device_mounted (udi);
 		}
 	} else {
 		dbg ("Unmounted: %s\n", udi);
