diff -ruN gnome-volume-manager-1.5.15-old/src/manager.c gnome-volume-manager-1.5.15/src/manager.c
--- gnome-volume-manager-1.5.15-old/src/manager.c	2006-03-14 13:35:42.000000000 +0100
+++ gnome-volume-manager-1.5.15/src/manager.c	2006-03-14 13:39:16.000000000 +0100
@@ -2979,6 +2979,10 @@
 	
 	if (!(mount_point = libhal_device_get_property_string (hal_ctx, udi, "volume.mount_point", NULL)))
 		return TRUE;
+
+	/* Ignore volumes in /media */
+	if (g_str_has_prefix (mount_point, "/media"))
+		return TRUE;
 	
 	if (statfs (mount_point, &buf) != -1) {
 		double free_space;
