diff -ruN gnome-volume-manager-1.5.15-old/src/manager.c gnome-volume-manager-1.5.15/src/manager.c
--- gnome-volume-manager-1.5.15-old/src/manager.c	2006-03-14 13:57:47.000000000 +0100
+++ gnome-volume-manager-1.5.15/src/manager.c	2006-03-14 13:57:53.000000000 +0100
@@ -33,7 +33,7 @@
 #endif
 
 #ifdef ENABLE_NOTIFY
-#include <sys/vfs.h>
+#include <sys/statvfs.h>
 #include <libnotify/notify.h>
 #endif
 
@@ -2975,7 +2975,7 @@
 {
 	char *mount_point = NULL;
 	NotifyNotification *n;
-	struct statfs buf;
+	struct statvfs buf;
 	
 	if (!(mount_point = libhal_device_get_property_string (hal_ctx, udi, "volume.mount_point", NULL)))
 		return TRUE;
@@ -2984,8 +2984,12 @@
 	if (g_str_has_prefix (mount_point, "/media"))
 		return TRUE;
 	
-	if (statfs (mount_point, &buf) != -1) {
+	if (statvfs (mount_point, &buf) != -1) {
 		double free_space;
+
+		/* Ignore read-only mounts */
+		if (buf.f_flag & ST_RDONLY)
+			return TRUE;
 		
 		free_space = (double) buf.f_bavail / (double) buf.f_blocks;
 
