diff -ruN gnome-volume-manager-1.5.15-old/src/manager.c gnome-volume-manager-1.5.15/src/manager.c
--- gnome-volume-manager-1.5.15-old/src/manager.c	2006-03-14 14:06:08.000000000 +0100
+++ gnome-volume-manager-1.5.15/src/manager.c	2006-03-14 14:06:56.000000000 +0100
@@ -3006,10 +3006,10 @@
 			int free_pct;
 			
 			label = libhal_device_get_property_string (hal_ctx, udi, "volume.label", NULL);
-			if (!label || label[0] == '\0')
+			if (!label || label[0] == '\0' || !strcmp (label, mount_point))
 				disk = g_strdup (mount_point);
 			else
-				disk = g_strdup (label);
+				disk = g_strdup_printf ("%s (%s)", label, mount_point);
 			libhal_free_string (label);
 			
 			free_pct = 100 - (free_space * 100);
