diff -ruN gnome-volume-manager-1.5.15-old/src/manager.c gnome-volume-manager-1.5.15/src/manager.c
--- gnome-volume-manager-1.5.15-old/src/manager.c	2006-04-13 14:54:05.000000000 +0200
+++ gnome-volume-manager-1.5.15/src/manager.c	2006-04-13 14:54:21.000000000 +0200
@@ -3027,7 +3027,7 @@
 			msg = g_strdup_printf (_("%d%% of the disk space on %s is in use"), free_pct, disk);
 			g_free (disk);
 			
-			n = notify_notification_new ("Low Disk Space", msg, "gnome-dev-harddisk", NULL);
+			n = notify_notification_new (_("Low Disk Space"), msg, "gnome-dev-harddisk", NULL);
 			notify_notification_set_urgency (n, NOTIFY_URGENCY_CRITICAL);
 			notify_notification_show (n, NULL);
 			g_object_unref (n);
