diff -ruN gnome-volume-manager-1.3.1-old/gnome-volume-properties.glade gnome-volume-manager-1.3.1/gnome-volume-properties.glade
--- gnome-volume-manager-1.3.1-old/gnome-volume-properties.glade	2005-02-06 17:10:39.000000000 +0100
+++ gnome-volume-manager-1.3.1/gnome-volume-properties.glade	2005-06-22 14:29:34.000000000 +0200
@@ -182,25 +182,6 @@
 			      <property name="fill">False</property>
 			    </packing>
 			  </child>
-
-			  <child>
-			    <widget class="GtkCheckButton" id="autorun_cb">
-			      <property name="visible">True</property>
-			      <property name="can_focus">True</property>
-			      <property name="label" translatable="yes">_Auto-run programs on new drives and media</property>
-			      <property name="use_underline">True</property>
-			      <property name="relief">GTK_RELIEF_NORMAL</property>
-			      <property name="focus_on_click">True</property>
-			      <property name="active">False</property>
-			      <property name="inconsistent">False</property>
-			      <property name="draw_indicator">True</property>
-			    </widget>
-			    <packing>
-			      <property name="padding">0</property>
-			      <property name="expand">False</property>
-			      <property name="fill">False</property>
-			    </packing>
-			  </child>
 			</widget>
 			<packing>
 			  <property name="padding">0</property>
@@ -357,6 +338,7 @@
 				      <property name="directory_entry">False</property>
 				      <property name="modal">False</property>
 				      <property name="use_filechooser">True</property>
+				      <property name="filechooser_action">GTK_FILE_CHOOSER_ACTION_OPEN</property>
 
 				      <child internal-child="entry">
 					<widget class="GtkEntry" id="command_cdr_entry">
@@ -367,7 +349,7 @@
 					  <property name="max_length">0</property>
 					  <property name="text" translatable="yes"></property>
 					  <property name="has_frame">True</property>
-					  <property name="invisible_char" translatable="yes">*</property>
+					  <property name="invisible_char">*</property>
 					  <property name="activates_default">False</property>
 					</widget>
 				      </child>
@@ -581,6 +563,7 @@
 				      <property name="directory_entry">False</property>
 				      <property name="modal">False</property>
 				      <property name="use_filechooser">True</property>
+				      <property name="filechooser_action">GTK_FILE_CHOOSER_ACTION_OPEN</property>
 
 				      <child internal-child="entry">
 					<widget class="GtkEntry" id="command_cda_entry">
@@ -591,7 +574,7 @@
 					  <property name="max_length">0</property>
 					  <property name="text" translatable="yes"></property>
 					  <property name="has_frame">True</property>
-					  <property name="invisible_char" translatable="yes">*</property>
+					  <property name="invisible_char">*</property>
 					  <property name="activates_default">False</property>
 					</widget>
 				      </child>
@@ -773,6 +756,7 @@
 				      <property name="directory_entry">False</property>
 				      <property name="modal">False</property>
 				      <property name="use_filechooser">True</property>
+				      <property name="filechooser_action">GTK_FILE_CHOOSER_ACTION_OPEN</property>
 
 				      <child internal-child="entry">
 					<widget class="GtkEntry" id="command_dvd_entry">
@@ -783,7 +767,7 @@
 					  <property name="max_length">0</property>
 					  <property name="text" translatable="yes"></property>
 					  <property name="has_frame">True</property>
-					  <property name="invisible_char" translatable="yes">*</property>
+					  <property name="invisible_char">*</property>
 					  <property name="activates_default">False</property>
 					</widget>
 				      </child>
@@ -965,6 +949,7 @@
 				      <property name="directory_entry">False</property>
 				      <property name="modal">False</property>
 				      <property name="use_filechooser">True</property>
+				      <property name="filechooser_action">GTK_FILE_CHOOSER_ACTION_OPEN</property>
 
 				      <child internal-child="entry">
 					<widget class="GtkEntry" id="command_photo_entry">
@@ -975,7 +960,7 @@
 					  <property name="max_length">0</property>
 					  <property name="text" translatable="yes"></property>
 					  <property name="has_frame">True</property>
-					  <property name="invisible_char" translatable="yes">*</property>
+					  <property name="invisible_char">*</property>
 					  <property name="activates_default">False</property>
 					</widget>
 				      </child>
diff -ruN gnome-volume-manager-1.3.1-old/src/manager.c gnome-volume-manager-1.3.1/src/manager.c
--- gnome-volume-manager-1.3.1-old/src/manager.c	2005-06-22 14:29:34.000000000 +0200
+++ gnome-volume-manager-1.3.1/src/manager.c	2005-06-22 14:30:35.000000000 +0200
@@ -221,6 +221,7 @@
  *
  * Returns TRUE if the user selected 'Run' and FALSE otherwise
  */
+/*
 static gboolean
 gvm_ask_autorun (const char *path)
 {
@@ -257,6 +258,7 @@
 
 	return retval;
 }
+*/
 
 /*
  * gvm_check_dvd - is this a Video DVD?  If so, do something about it.
@@ -440,7 +442,7 @@
 
 	if (gvm_check_photos (udi, device, mount_point))
 		goto out;
-
+/*
 	if (config.autorun == TRUE && config.autorun_path) {
 		char **autorun_fns;
 		int i;
@@ -478,7 +480,7 @@
 
 		g_strfreev (autorun_fns);
 	}
-	
+*/
 	if ((config.autobrowse == TRUE) && (autorun_succeeded == FALSE)) {
 		gvm_run_command (device, NAUTILUS_COMMAND, mount_point, udi);
 	}
diff -ruN gnome-volume-manager-1.3.1-old/src/properties.c gnome-volume-manager-1.3.1/src/properties.c
--- gnome-volume-manager-1.3.1-old/src/properties.c	2005-06-22 14:29:33.000000000 +0200
+++ gnome-volume-manager-1.3.1/src/properties.c	2005-06-22 14:29:34.000000000 +0200
@@ -36,7 +36,7 @@
 GtkWidget *automount_drives_cb;
 GtkWidget *automount_media_cb;
 GtkWidget *autobrowse_cb;
-GtkWidget *autorun_cb;
+//GtkWidget *autorun_cb;
 GtkWidget *autoplay_cda_cb;
 GtkWidget *command_cda_hbox;
 GtkWidget *command_cda_entry;
@@ -65,8 +65,8 @@
 static void
 set_sensitivity (void)
 {
-	gtk_widget_set_sensitive (autorun_cb,
-			config.automount_media || config.automount_drives);
+/*	gtk_widget_set_sensitive (autorun_cb,
+			config.automount_media || config.automount_drives); */
 	gtk_widget_set_sensitive (command_cda_hbox, config.autoplay_cda);
 	gtk_widget_set_sensitive (command_cdr_hbox, config.autoburn_cdr);
 	gtk_widget_set_sensitive (command_dvd_hbox, config.autoplay_dvd);
@@ -85,8 +85,8 @@
 			       config.automount_media, NULL);
 	gconf_client_set_bool (config.client, GCONF_ROOT "autobrowse",
 			       config.autobrowse, NULL);
-	gconf_client_set_bool (config.client, GCONF_ROOT "autorun",
-			       config.autorun, NULL);
+/*	gconf_client_set_bool (config.client, GCONF_ROOT "autorun",
+			       config.autorun, NULL); */
 	gconf_client_set_bool (config.client, GCONF_ROOT "autoburn_cdr",
 			       config.autoburn_cdr, NULL);
 	gconf_client_set_string (config.client,
@@ -120,7 +120,7 @@
 			(automount_media_cb)->active;
 	config.autobrowse = GTK_TOGGLE_BUTTON
 			(autobrowse_cb)->active;
-	config.autorun = GTK_TOGGLE_BUTTON (autorun_cb)->active;
+/*	config.autorun = GTK_TOGGLE_BUTTON (autorun_cb)->active; */
 	config.autoburn_cdr = GTK_TOGGLE_BUTTON (autoburn_cdr_cb)->active;
 	config.autoplay_cda = GTK_TOGGLE_BUTTON (autoplay_cda_cb)->active;
 	config.autoplay_dvd = GTK_TOGGLE_BUTTON (autoplay_dvd_cb)->active;
@@ -170,7 +170,7 @@
 	GError *error = NULL;
 
 	if (!(config.automount_drives || config.automount_media 
-			|| config.autobrowse || config.autorun 
+			|| config.autobrowse /* || config.autorun */
 			|| config.autoplay_cda || config.autoplay_dvd 
 			|| config.autophoto) || check_clipboard ()) 
 		return;
@@ -209,8 +209,8 @@
 				      config.automount_media);
 	gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (autobrowse_cb),
 				      config.autobrowse);
-	gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (autorun_cb),
-				      config.autorun);
+/*	gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (autorun_cb),
+				      config.autorun); */
 	gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (autoplay_cda_cb),
 				      config.autoplay_cda);
 
@@ -252,8 +252,8 @@
 			GCONF_ROOT "automount_media", NULL);
 	config.autobrowse = gconf_client_get_bool (config.client,
 			GCONF_ROOT "autobrowse", NULL);
-	config.autorun = gconf_client_get_bool (config.client,
-			GCONF_ROOT "autorun", NULL);
+/*	config.autorun = gconf_client_get_bool (config.client,
+			GCONF_ROOT "autorun", NULL); */
 	config.autoplay_cda = gconf_client_get_bool (config.client,
 			GCONF_ROOT "autoplay_cda", NULL);
 	config.autoplay_cda_command =
@@ -454,7 +454,7 @@
 	automount_drives_cb = glade_xml_get_widget (xml, "automount_drives_cb");
 	automount_media_cb = glade_xml_get_widget (xml, "automount_media_cb");
 	autobrowse_cb = glade_xml_get_widget (xml, "autobrowse_cb");
-	autorun_cb = glade_xml_get_widget (xml, "autorun_cb");
+/*	autorun_cb = glade_xml_get_widget (xml, "autorun_cb"); */
 
 	autoplay_cda_cb = glade_xml_get_widget (xml, "autoplay_cda_cb");
 	command_cda_entry = glade_xml_get_widget (xml, "command_cda_entry");
@@ -505,8 +505,8 @@
 			    GTK_SIGNAL_FUNC (signal_response), NULL);
 	gtk_signal_connect (GTK_OBJECT (autobrowse_cb), "toggled",
 			    GTK_SIGNAL_FUNC (signal_response), NULL);
-	gtk_signal_connect (GTK_OBJECT (autorun_cb), "toggled",
-			    GTK_SIGNAL_FUNC (signal_response), NULL);
+/*	gtk_signal_connect (GTK_OBJECT (autorun_cb), "toggled",
+			    GTK_SIGNAL_FUNC (signal_response), NULL); */
 
 	gtk_signal_connect (GTK_OBJECT (autoplay_cda_cb), "toggled",
 			    GTK_SIGNAL_FUNC (signal_response), NULL);
