diff -ruN gnome-volume-manager-1.5.15-old/src/manager.c gnome-volume-manager-1.5.15/src/manager.c
--- gnome-volume-manager-1.5.15-old/src/manager.c	2006-05-11 15:14:51.000000000 +0200
+++ gnome-volume-manager-1.5.15/src/manager.c	2006-05-11 15:28:35.000000000 +0200
@@ -3376,8 +3376,19 @@
 gvm_user_active_at_console (void)
 {
 #ifdef ENABLE_MULTIUSER
-	/* FIXME: it is currently impossible to get this information */
-	return TRUE;
+	gchar *argv[] = { "check-foreground-console", NULL };
+	int retcode;
+
+	if (!g_spawn_sync (NULL, argv, NULL, G_SPAWN_SEARCH_PATH, NULL, NULL,
+		    NULL, NULL, &retcode, NULL)  || !WIFEXITED (retcode) ) {
+	    /* if check-foreground-console could not be executed, assume active console */
+	    warn ("gvm_user_active_at_console: could not execute check-foreground-console\n");
+	    return TRUE;
+	}
+
+	dbg ("gvm_user_active_at_console: check-foreground-console returned with %i\n", WEXITSTATUS (retcode));
+	return WEXITSTATUS (retcode) == 0;
+
 #else
 	/* for the non-multiuser configuration, we assume we're always on the active console */
 	return TRUE;
