diff -ruN gnome-volume-manager-1.5.15-old/gnome-volume-manager.desktop.in gnome-volume-manager-1.5.15/gnome-volume-manager.desktop.in
--- gnome-volume-manager-1.5.15-old/gnome-volume-manager.desktop.in	2006-01-23 21:51:14.000000000 +0100
+++ gnome-volume-manager-1.5.15/gnome-volume-manager.desktop.in	2006-05-05 15:38:23.000000000 +0200
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 Categories=
-OnlyShowIn=GNOME;
+OnlyShowIn=GNOME;XFCE;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-volume-manager
 X-GNOME-Bugzilla-Component=general
