diff -Naur gnome-volume-manager-1.4.0.orig/gnome-volume-manager.schemas.in gnome-volume-manager-1.4.0/gnome-volume-manager.schemas.in
--- gnome-volume-manager-1.4.0.orig/gnome-volume-manager.schemas.in	2005-08-19 13:56:30.000000000 -0400
+++ gnome-volume-manager-1.4.0/gnome-volume-manager.schemas.in	2005-09-05 11:14:17.000000000 -0400
@@ -76,7 +76,7 @@
         <key>/schemas/desktop/gnome/volume_manager/autoplay_cda_command</key>
 	<applyto>/desktop/gnome/volume_manager/autoplay_cda_command</applyto>
 	<type>string</type>
-	<default>totem %d</default>
+	<default>banshee --audio-cd %d</default>
 	<locale name="C">
 	  <short>CD play command</short>
 	  <long>Command to run when an audio CD is inserted.</long>
@@ -99,7 +99,7 @@
         <key>/schemas/desktop/gnome/volume_manager/autoburn_audio_cd_command</key>
 	<applyto>/desktop/gnome/volume_manager/autoburn_audio_cd_command</applyto>
 	<type>string</type>
-	<default>nautilus --no-desktop burn:</default>
+	<default>banshee --burn-cd</default>
 	<locale name="C">
 	  <short>Audio CD burn command</short>
 	  <long>Command to run when the user wishes to burn an Audio CD.</long>
@@ -144,7 +144,7 @@
         <key>/schemas/desktop/gnome/volume_manager/autoplay_dvd_command</key>
 	<applyto>/desktop/gnome/volume_manager/autoplay_dvd_command</applyto>
 	<type>string</type>
-	<default>totem %d</default>
+	<default>totem dvd://%d</default>
 	<locale name="C">
 	  <short>DVD play command</short>
 	  <long>Command to run when a video DVD is inserted.</long>
@@ -178,7 +178,7 @@
         <key>/schemas/desktop/gnome/volume_manager/autophoto</key>
 	<applyto>/desktop/gnome/volume_manager/autophoto</applyto>
 	<type>bool</type>
-	<default>false</default>
+	<default>true</default>
 	<locale name="C">
 	  <short>Run photo management command</short>
 	  <long>Whether, when a digital camera is connected or media from a
@@ -191,7 +191,7 @@
         <key>/schemas/desktop/gnome/volume_manager/autophoto_command</key>
 	<applyto>/desktop/gnome/volume_manager/autophoto_command</applyto>
 	<type>string</type>
-	<default></default>
+	<default>f-spot-import %h</default>
 	<locale name="C">
 	  <short>Photo management command</short>
 	  <long>Command to run when a digital camera is connected or media 
@@ -203,7 +203,7 @@
         <key>/schemas/desktop/gnome/volume_manager/autoipod</key>
 	<applyto>/desktop/gnome/volume_manager/autoipod</applyto>
 	<type>bool</type>
-	<default>false</default>
+	<default>true</default>
 	<locale name="C">
 	  <short>Run iPod program</short>
 	  <long>Whether, when an iPod is connected, gnome-volume-manager should run
@@ -215,7 +215,7 @@
         <key>/schemas/desktop/gnome/volume_manager/autoipod_command</key>
 	<applyto>/desktop/gnome/volume_manager/autoipod_command</applyto>
 	<type>string</type>
-	<default></default>
+	<default>banshee --dap=%h</default>
 	<locale name="C">
 	  <short>iPod sync command</short>
 	  <long>Command to run when an iPod is connected.</long>
