--- gnome-volume-manager-1.5.7/src/manager.c.array-pointer-fix	2006-01-04 20:38:50.000000000 -0500
+++ gnome-volume-manager-1.5.7/src/manager.c	2006-01-04 20:40:05.000000000 -0500
@@ -1243,7 +1243,8 @@
 	dbus_uint64_t size;
 	int noptions = 0;
 	DBusError error;
-	
+	char **poptions;
+		
 	dbg ("mounting %s...\n", udi);
 	
 	if (!(dmesg = dbus_message_new_method_call ("org.freedesktop.Hal", udi,
@@ -1261,9 +1262,10 @@
 	if (size <= (256 * 1024 * 1024))
 		options[noptions++] = "sync";
 	options[noptions] = NULL;
-	
+
+	poptions = options;
 	if (!dbus_message_append_args (dmesg, DBUS_TYPE_STRING, &mount_point, DBUS_TYPE_STRING, &fstype,
-				       DBUS_TYPE_ARRAY, DBUS_TYPE_STRING, &options, noptions,
+				       DBUS_TYPE_ARRAY, DBUS_TYPE_STRING, &poptions, noptions,
 				       DBUS_TYPE_INVALID)) {
 		dbg ("mount failed for %s: could not append args to dbus message\n", udi);
 		dbus_message_unref (dmesg);
