--- gnome-volume-manager-0.9.10/gnome-volume-properties.desktop.in.addtobase	2004-08-31 13:30:04.591853000 -0400
+++ gnome-volume-manager-0.9.10/gnome-volume-properties.desktop.in	2004-08-31 13:31:07.369012000 -0400
@@ -6,7 +6,7 @@
 Exec=gnome-volume-properties
 Terminal=false
 Type=Application
-Categories=Application;Settings;
+Categories=Application;Settings;X-Red-Hat-Base
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-volume-manager
