--- devhelp-0.10/src/Makefile.am.org	2005-10-23 08:56:50.000000000 -0400
+++ devhelp-0.10/src/Makefile.am	2005-10-23 08:57:25.000000000 -0400
@@ -76,6 +76,9 @@
 	@LIBDEVHELP_LIBS@				\
 	$(MOZILLA_COMPONENT_LIBS)
 
+libdevhelp_1_la_LDFLAGS =                               \
+	-R$(MOZILLA_HOME)
+
 dh-marshal.h: dh-marshal.list
 	(cd $(srcdir) && \
 	@GLIB_GENMARSHAL@ --prefix=dh_marshal dh-marshal.list --header) > xgen-gmh \
