diff -up pywebkitgtk-1.0.1/Makefile.in.py26 pywebkitgtk-1.0.1/Makefile.in
--- pywebkitgtk-1.0.1/Makefile.in.py26	2008-12-05 13:05:19.000000000 -0500
+++ pywebkitgtk-1.0.1/Makefile.in	2008-12-05 13:05:23.000000000 -0500
@@ -741,6 +741,5 @@ webkit.c: webkit.defs webkit.override
 	    --register $(PYGTK_DEFSDIR)/gtk-types.defs \
 	    --override $*.override \
 	    --prefix py$* $*.defs) 2>&1 >gen-$*.c | tee $*.errors \
-	&& ! grep -q -v "^\*\*\*INFO\*\*\*" $*.errors \
 	&& cp gen-$*.c $*.c \
 	&& rm -f gen-$*.c
