--- tracepath6.c.orig	2008-10-10 17:33:19.000000000 +0300
+++ tracepath6.c	2008-10-10 18:06:06.000000000 +0300
@@ -359,8 +359,8 @@ int main(int argc, char **argv)
 		perror("IP_RECVERR");
 		exit(1);
 	}
-	if (setsockopt(fd, SOL_IPV6, IPV6_HOPLIMIT, &on, sizeof(on))) {
-		perror("IPV6_HOPLIMIT");
+	if (setsockopt(fd, SOL_IPV6, IPV6_RECVHOPLIMIT, &on, sizeof(on))) {
+		perror("IPV6_RECVHOPLIMIT");
 		exit(1);
 	}
 	if (mapped && setsockopt(fd, SOL_IP, IP_RECVTTL, &on, sizeof(on))) {
--- ping6.c.orig	2008-10-10 17:33:10.000000000 +0300
+++ ping6.c	2008-10-10 18:04:45.000000000 +0300
@@ -523,7 +523,7 @@ int main(int argc, char *argv[])
 
 	if (1) {
 		int on = 1;
-		if (setsockopt(icmp_sock, IPPROTO_IPV6, IPV6_HOPLIMIT,
+		if (setsockopt(icmp_sock, IPPROTO_IPV6, IPV6_RECVHOPLIMIT,
 			       &on, sizeof(on)) == -1) {
 			perror ("can't receive hop limit");
 			exit(2);
