diff -Naur gmime-2.2.3.orig/configure gmime-2.2.3/configure
--- gmime-2.2.3.orig/configure	2006-08-02 10:47:44.000000000 -0400
+++ gmime-2.2.3/configure	2006-11-19 14:25:57.000000000 -0500
@@ -25167,35 +25167,6 @@
 echo "${ECHO_T}no" >&6
 fi
 
-	# if we found mcs, make sure it's mono and not something else named
-	# mcs, such as Tru64's /usr/bin/mcs, for manipulating object file
-	# comment sections of eCOFF object files.
-	if test "x$CSC" != "xno" ; then
-		echo "$as_me:$LINENO: checking whether $CSC is GNU Mono" >&5
-echo $ECHO_N "checking whether $CSC is GNU Mono... $ECHO_C" >&6
-		$CSC --version > /dev/null 2>&1
-		if test $? -ne 0 ; then
-			CSC=no
-			echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6
-		else
-			echo "$as_me:$LINENO: result: yes" >&5
-echo "${ECHO_T}yes" >&6
-		fi
-	fi
-else
-	CSC="no"
-fi
-
-if test "x$CSC" = "xno"; then
-		if test "x$enable_mono" = "xyes"; then
-		{ { echo "$as_me:$LINENO: error: Could not find mcs" >&5
-echo "$as_me: error: Could not find mcs" >&2;}
-   { (exit 1); exit 1; }; }
-	fi
-
-	enable_mono="no"
-
 
 if false; then
   ENABLE_MONO_TRUE=
