--- Makefile.config.in.old	2007-11-29 19:29:27.000000000 +0200
+++ Makefile.config.in	2007-11-29 19:30:28.000000000 +0200
@@ -33,7 +33,7 @@
 #CC = gcc
 # Note that 'cc' is usually an alias for whatever is the main compiler
 # on a system, e.g. the GNU Compiler on Linux.
-CC = cc
+CC = %(cc)s
 
 # The linker.
 LD = $(CC)
@@ -96,7 +96,7 @@
 # to use to compile and link build tools.
 CC_FOR_BUILD = $(CC)
 LD_FOR_BUILD = $(LD)
-CFLAGS_FOR_BUILD = $(CFLAGS)
+CFLAGS_FOR_BUILD = %(cflags)s
 
 # MAKE is set automatically by Make to what was used to invoke Make.
 
@@ -112,7 +112,7 @@
 
 # STRIPFLAG is the option you pass to the above install program to make it
 # strip unnecessary information out of binaries.
-STRIPFLAG = -s
+STRIPFLAG = 
 # If you don't want to strip the binaries, just leave it null:
 #STRIPFLAG = 
 
@@ -163,6 +163,7 @@
 # A user of Tru64 5.1A in July 2003 needed NOT to have -DLONG_32.  In
 # theory, you need this if on your system, long is 32 bits and int is not.
 # But it may be completely irrelevant today.
+CFLAGS = %(cflags)s
 #Tru64:
 #CFLAGS = -O2 -std1 -DLONG_32
 #CFLAGS = -O2 -std1
@@ -280,7 +281,7 @@
 # compiler/linker).  Build-time linking fails without it.  I don't
 # know why -- history seems to be repeating itself.  2005.02.23.
 
-CFLAGS_SHLIB = 
+CFLAGS_SHLIB = -fPIC
 # Solaris or SunOS with gcc, and NetBSD:
 #CFLAGS_SHLIB = -fpic
 #CFLAGS_SHLIB = -fPIC
@@ -382,10 +383,10 @@
 # JPEG stuff statically linked in, in which case you won't need 
 # JPEGLIB in order to build the Tiff converters.
 
-JPEGLIB = NONE
-JPEGHDR_DIR =
-#JPEGLIB = libjpeg.so
-#JPEGHDR_DIR = /usr/include/jpeg
+#JPEGLIB = NONE
+#JPEGHDR_DIR =
+JPEGLIB = libjpeg.so
+JPEGHDR_DIR = %(includedir)s
 # Netbsd:
 #JPEGLIB = ${LOCALBASE}/lib/libjpeg.so
 #JPEGHDR_DIR = ${LOCALBASE}/include
@@ -576,8 +577,8 @@
 #and installed in addition to whatever library type you specified by
 #NETPBMLIBTYPE.  If NETPBMLIBTYPE specified a static library,
 #STATICLIB_TOO simply has no effect.
-STATICLIB_TOO = y
-#STATICLIB_TOO = n
+#STATICLIB_TOO = y
+STATICLIB_TOO = n
 
 #STATICLIBSUFFIX is the suffix that static libraries have.  It's
 #meaningless if you aren't building static libraries.
