Index: src/tools/xmlversion/main.cpp
===================================================================
--- src/tools/xmlversion/main.cpp	(revision 850)
+++ src/tools/xmlversion/main.cpp	(working copy)
@@ -12,6 +12,25 @@
 #define snprintf _snprintf
 #endif
 
+static char* strip_destdir( char *filename, char const *destdir )
+{
+	int xx;
+	int destdir_length;
+
+	if( !destdir )
+		return filename;
+	if( !filename )
+		return NULL;
+
+	destdir_length = strlen( destdir );
+
+	for( xx = 0; xx < destdir_length; ++xx )
+		if( destdir[ xx ] != filename[ xx ] )
+			return filename;
+
+	return &filename[ destdir_length ];
+}
+
 static int findIndex( void *versionHandle, const char* dataLocation, const char* userLocation, char* path, bool dataOnly )
 {
 	int nbIndices = GfParmGetEltNb( versionHandle, path ) + 1;
@@ -46,7 +65,7 @@
 	return curIndex;
 }
 
-static int process( const char* xmlVersion, char* dataLocation, char* userLocation )
+static int process( const char* xmlVersion, char* dataLocation, char* userLocation, char const* destdir )
 {
 	void *versionHandle;
 	void *xmlHandle;
@@ -73,7 +92,7 @@
 	path = (char*)malloc( sizeof(char) * 31 );
 	snprintf( path, 30, "versions/%d", index );
 
-	GfParmSetStr( versionHandle, path, "Data location", dataLocation );
+	GfParmSetStr( versionHandle, path, "Data location", strip_destdir( dataLocation, destdir ) );
 	GfParmSetStr( versionHandle, path, "Local location", userLocation );
 	GfParmSetNum( versionHandle, path, "Major version", NULL, GfParmGetMajorVersion( xmlHandle ) );
 	GfParmSetNum( versionHandle, path, "Minor version", NULL, GfParmGetMinorVersion( xmlHandle ) );
@@ -91,7 +110,7 @@
 	return 0;
 }
 
-static int add_directory( const char* xmlVersion, char* directoryName )
+static int add_directory( const char* xmlVersion, char* directoryName, char const* destdir )
 {
 	void *versionHandle;
 	char *path;
@@ -110,7 +129,7 @@
 	path = (char*)malloc( sizeof(char) * 31 );
 	snprintf( path, 30, "directories/%d", index );
 
-	GfParmSetStr( versionHandle, path, "Data location", directoryName );
+	GfParmSetStr( versionHandle, path, "Data location", strip_destdir( directoryName, destdir ) );
 
 	free( pathStart );
 	free( path );
@@ -129,6 +148,7 @@
 	char *xmlversion;
 	char *dataLocation;
 	char *userLocation;
+	char const *destdir;
 	int ret;
 
 	if( argc <= 3 )
@@ -153,15 +173,16 @@
 	xmlversion = strdup( argv[1] );
 	dataLocation = strdup( argv[2] );
 	userLocation = strdup( argv[3] );
+	destdir = getenv( "DESTDIR" );
 
 	if( strcmp( xmlversion, "-d" ) == 0 )
-		ret = add_directory( dataLocation, userLocation );
+		ret = add_directory( dataLocation, userLocation, destdir );
 	else if( strcmp( dataLocation, "-d" ) == 0 )
-		ret = add_directory( xmlversion, userLocation );
+		ret = add_directory( xmlversion, userLocation, destdir );
 	else if( strcmp( userLocation, "-d" ) == 0 )
-		ret = add_directory( xmlversion, dataLocation );
+		ret = add_directory( xmlversion, dataLocation, destdir );
 	else
-		ret = process( xmlversion, dataLocation, userLocation );
+		ret = process( xmlversion, dataLocation, userLocation, destdir );
 
 	free( xmlversion );
 	free( dataLocation );
